// @splunk/themes/enterprise/dark.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _light = _interopRequireDefault(require("./light"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var dragHandleDark = "iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAA1SURBVHgB7dKhEQAgDAPAhHmwSKZHYtmHVtZVVNTkXS53UeG57yPYazLmrB8o6h8QgPqBOAOboRAPJUGIOAAAAABJRU5ErkJggg=="; // see babel-plugin-base64-png
var dark = {
  borderColor: '#75797a',
  borderColorWeak: '#404547',
  borderColorStrong: '#9c9fa0'
};
var notificationColors = {
  notificationColorInfoWeak: '#2E1B4B',
  notificationColorInfo: '#AC75FF',
  notificationColorInfoStrong: '#D1B2FF',
  notificationColorPositiveWeak: '#092635',
  notificationColorPositive: '#1AB2FF',
  notificationColorPositiveStrong: '#99DDFF',
  notificationColorCautionWeak: '#331E00',
  notificationColorCaution: '#FFAF0F',
  notificationColorCautionStrong: '#FDDC9B',
  notificationColorNegativeWeak: '#391313',
  notificationColorNegative: '#FF5252',
  notificationColorNegativeStrong: '#FF9999'
};
var severityColors = {
  severityColorCustom: '#868EC2',
  severityColorUnknown: '#889099',
  severityColorInfo: '#A974F7',
  severityColorNormal: '#04A4B0',
  severityColorNotice: '#D6900D',
  severityColorWarning: '#F26722',
  severityColorAlert: '#FA5762',
  severityColorCritical: '#F2638C',
  severityColorEmergency: '#E85FC6'
};
var syntaxColors = {
  syntaxBlue: '#6cd0f0',
  syntaxBrown: '#fccf87',
  syntaxGray: '#b5b5b5',
  syntaxGreen: '#cef06c',
  syntaxOrange: '#f7994a',
  syntaxPink: '#f494e5',
  syntaxPurple: '#c99eff',
  syntaxRed: '#fa94aa',
  syntaxTeal: '#45d4ba'
};
var contentColors = {
  contentColorActive: _light["default"].white,
  contentColorDefault: _light["default"].gray92,
  contentColorDisabled: _light["default"].gray45,
  contentColorInverted: _light["default"].black,
  contentColorMuted: _light["default"].gray60,
  contentColorAccent: '#20ac33',
  contentColorAccentStrong: '#a4e9ae',
  contentColorAccentWeak: '#0b4613',
  contentColorNegative: '#e37267',
  contentColorNegativeWeak: '#562a25',
  contentColorLink: _light["default"].accentColorL10,
  contentColorLinkDisabled: _light["default"].accentColorD30,
  contentColorLinkVisited: '#a974f7',
  contentColorLinkVisitedDisabled: '#6732b8'
};
var actionColors = {
  actionColorBackgroundPrimary: '#1a8929',
  actionColorBackgroundPrimaryActive: '#6eba78',
  actionColorBackgroundPrimaryDisabled: '#163b1b',
  actionColorBackgroundPrimaryHover: '#6eba78',
  actionColorBackgroundSecondary: 'transparent',
  actionColorBackgroundSecondaryActive: 'rgba(225, 225, 225, 0.16)',
  actionColorBackgroundSecondaryDisabled: 'transparent',
  actionColorBackgroundSecondaryHover: 'rgba(255, 255, 255, 0.05)',
  actionColorBackgroundSubtle: 'transparent',
  actionColorBackgroundSubtleActive: '#214526',
  actionColorBackgroundSubtleDisabled: 'transparent',
  actionColorBackgroundSubtleHover: '#214526',
  actionColorBackgroundDestructive: '#dc4e41',
  actionColorBackgroundDestructiveActive: '#ff776b',
  actionColorBackgroundDestructiveDisabled: '#3b1a16',
  actionColorBackgroundDestructiveHover: '#ff776b',
  actionColorBackgroundDestructiveSecondary: 'transparent',
  actionColorBackgroundDestructiveSecondaryActive: '#47221f',
  actionColorBackgroundDestructiveSecondaryDisabled: 'transparent',
  actionColorBackgroundDestructiveSecondaryHover: '#47221f',
  actionColorBorderSecondary: _light["default"].gray60,
  actionColorBorderSecondaryActive: _light["default"].gray60,
  actionColorBorderSecondaryDisabled: 'rgba(255, 255, 255, 0.3)',
  actionColorBorderSecondaryHover: 'rgba(255, 255, 255, 0.7)',
  actionColorBorderDestructiveSecondary: '#ff776b',
  actionColorBorderDestructiveSecondaryActive: '#ff8a80',
  actionColorBorderDestructiveSecondaryDisabled: '#562a25',
  actionColorBorderDestructiveSecondaryHover: '#ff8a80',
  actionColorContentPrimary: contentColors.contentColorInverted,
  actionColorContentPrimaryDisabled: contentColors.contentColorInverted,
  actionColorContentSecondary: contentColors.contentColorActive,
  actionColorContentSecondaryDisabled: contentColors.contentColorDisabled,
  actionColorContentSubtleActive: contentColors.contentColorAccentStrong,
  actionColorContentSubtleHover: contentColors.contentColorAccentStrong,
  actionColorContentStandalone: contentColors.contentColorAccent,
  actionColorContentStandaloneActive: contentColors.contentColorAccentStrong,
  actionColorContentStandaloneDisabled: contentColors.contentColorAccentWeak,
  actionColorContentStandaloneHover: contentColors.contentColorAccentStrong,
  actionColorContentDestructive: contentColors.contentColorInverted,
  actionColorContentDestructiveDisabled: contentColors.contentColorInverted,
  actionColorContentDestructiveSecondary: contentColors.contentColorNegative,
  actionColorContentDestructiveSecondaryDisabled: contentColors.contentColorNegativeWeak
};
var interactiveColors = {
  interactiveColorBorder: _light["default"].gray60,
  interactiveColorBorderActive: _light["default"].gray60,
  interactiveColorBorderHover: 'rgba(255, 255, 255, 0.7)',
  interactiveColorBorderDisabled: 'rgba(255, 255, 255, 0.3)',
  interactiveColorOverlaySelected: 'rgba(255, 255, 255, 0.1)',
  interactiveColorOverlayHover: 'rgba(255, 255, 255, 0.05)',
  interactiveColorOverlayActive: 'rgba(225, 225, 225, 0.16)',
  interactiveColorOverlayDrag: 'rgba(57, 147, 255, 0.16)',
  interactiveColorBackground: 'transparent',
  interactiveColorBackgroundDisabled: 'transparent',
  interactiveColorAccent: '#1A8929',
  interactiveColorAccentError: _light["default"].errorColorL10,
  interactiveColorAccentErrorStrong: '#f1b9b3',
  interactiveColorAccentErrorWeak: '#6f261d'
};
var neutralColors = {
  white: '#ffffff',
  neutral50: '#20262a',
  neutral100: _light["default"].gray25,
  neutral200: _light["default"].gray30,
  neutral300: _light["default"].gray45,
  neutral400: _light["default"].gray60,
  neutral500: _light["default"].gray80,
  black: '#000000',
  transparent: 'transparent'
};
var backgroundColors = {
  backgroundColorDialog: _light["default"].gray20,
  backgroundColorFloating: _light["default"].white,
  backgroundColorNavigation: _light["default"].black,
  backgroundColorPage: _light["default"].gray20,
  backgroundColorPopup: _light["default"].gray22,
  backgroundColorScrim: "rgba(0, 0, 0, 0.8)",
  backgroundColorSection: '#1E2427',
  backgroundColorSidebar: _light["default"].black
};
var shadows = {
  focusShadow: "0 0 0 2px ".concat(backgroundColors.backgroundColorPage, ", 0 0 0 5px ").concat(_light["default"].focusColor),
  focusShadowInset: "inset 0 0 0 3px ".concat(_light["default"].focusColor),
  embossShadow: '0px 1px 5px rgba(0, 0, 0, 0.35), 0px 0px 1px rgba(0, 0, 0, 0.35)',
  overlayShadow: '0px 26px 103px rgba(0, 0, 0, 0.64), 0px 11px 18px rgba(0, 0, 0, 0.32), 0px 3px 6px rgba(0, 0, 0, 0.3)',
  dragShadow: '0px 26px 103px rgba(0, 0, 0, 0.64), 0px 11px 18px rgba(0, 0, 0, 0.32), 0px 3px 6px rgba(0, 0, 0, 0.3)',
  modalShadow: '0px 50px 200px #000000, 0px 29px 66px rgba(0, 0, 0, 0.41), 0px 14px 47px rgba(0, 0, 0, 0.17), 0px 5px 10px rgba(0, 0, 0, 0.15)'
};
var deprecated = {
  contentColorInfo: '#66a7c4',
  // @deprecated, SUI-8251
  contentColorNegativeStrong: '#ffa69e',
  contentColorPositive: '#7ecd7e',
  contentColorWarning: '#f8be34',
  backgroundColor: _light["default"].gray20,
  // @deprecated SUI-6656
  backgroundColorHover: _light["default"].gray30,
  borderActiveColor: "rgba(225,225,225, 0.5)",
  borderLightColor: _light["default"].gray60,
  // @deprecated, SUI-5981
  // @deprecated, SUI-6160
  textColor: _light["default"].white,
  textGray: _light["default"].gray92,
  textDisabledColor: _light["default"].gray45,
  // deprecated
  accentColorPositive: _light["default"].successColor,
  accentColorWarning: _light["default"].warningColor,
  accentColorAlert: _light["default"].alertColor,
  accentColorNegative: _light["default"].errorColor,
  linkColor: _light["default"].accentColorL10,
  linkColorHover: _light["default"].accentColorL20,
  border: "1px solid ".concat(_light["default"].gray22),
  borderDark: "1px solid ".concat(_light["default"].black),
  borderLight: "1px solid ".concat(_light["default"].gray60),
  interactiveColorPrimary: _light["default"].brandColor,
  statusColorInfo: _light["default"].infoColor,
  statusColorInfoWeak: _light["default"].infoColorD10,
  statusColorInfoStrong: _light["default"].infoColorL10,
  statusColorNormal: _light["default"].successColor,
  statusColorNormalWeak: _light["default"].successColorD10,
  statusColorNormalStrong: _light["default"].successColorL10,
  statusColorLow: _light["default"].warningColor,
  statusColorLowWeak: _light["default"].warningColorD10,
  statusColorLowStrong: _light["default"].warningColorL10,
  statusColorMedium: _light["default"].alertColor,
  statusColorMediumWeak: _light["default"].alertColorD10,
  statusColorMediumStrong: _light["default"].alertColorL10,
  statusColorHigh: _light["default"].errorColor,
  statusColorHighWeak: _light["default"].errorColorD10,
  statusColorHighStrong: _light["default"].errorColorL10,
  statusColorCritical: _light["default"].errorColorD10,
  statusColorCriticalWeak: _light["default"].errorColorD20,
  statusColorCriticalStrong: _light["default"].errorColor,
  draggableBackground: "url('data:image/png;base64,".concat(dragHandleDark, "') 0 0 / 8px 8px repeat")
};
var theme = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, _light["default"]), dark), actionColors), notificationColors), severityColors), backgroundColors), neutralColors), syntaxColors), contentColors), shadows), interactiveColors), deprecated);
var _default = theme;
exports["default"] = _default;
