// @splunk/themes/enterprise/light.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _dataViz = _interopRequireDefault(require("./dataViz"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var dragHandle = "iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAA2SURBVHgB7dKhEQAgDAPAhDnxDMAcDIBnT1pZV1FRk3e53EWFc+2P4N3DmLN+oKh/QADqB+IMUKEQD/CeueAAAAAASUVORK5CYII="; // see babel-plugin-base64-pngimport
/**
 * To avoid duplicating hex codes and keeping them in sync, for deprecated tokens and
 * colors that are used for themes 1.0 tokens, add this object to store the colors that
 * are used across enterprise themes.
 */
var deprecatedEnterpriseColors = {
  gray30: '#3c444d',
  gray45: '#5c6773',
  gray60: '#818d99',
  gray80: '#c3cbd4',
  gray92: '#e1e6eb',
  gray96: '#f2f4f5',
  gray98: '#f7f8fa',
  successColor: '#53a051',
  warningColor: '#f8be34',
  alertColor: '#f1813f',
  errorColor: '#dc4e41',
  accentColorD10: '#006eaa',
  accentColor: '#007abd',
  accentColorL30: '#7ed2ff',
  infoColor: '#006d9c',
  infoColorL10: '#338ab0',
  infoColorD10: '#00577c',
  successColorL10: '#76b374',
  successColorD10: '#479144',
  warningColorL10: '#facb5d',
  warningColorD10: '#e0ac16',
  alertColorL10: '#f49b66',
  alertColorD10: '#da742e',
  errorColorL10: '#e37267',
  errorColorD10: '#c84535',
  errorColorD20: '#b23d30',
  brandColor: '#5cc05c'
};
var neutralColors = {
  white: '#ffffff',
  neutral50: '#e8ebee',
  neutral100: '#e1e5ea',
  neutral200: '#dae0e7',
  neutral300: '#d5dbe2',
  neutral400: '#c9d1da',
  neutral500: '#bbc4ce',
  black: '#000000',
  transparent: 'transparent'
};
var usageColors = {
  borderColor: '#8c8c8c',
  borderColorWeak: '#d6d6d6',
  borderColorStrong: '#666666',
  focusColor: deprecatedEnterpriseColors.accentColorD10
};
var notificationColors = {
  notificationColorInfoWeak: '#F0E5FF',
  notificationColorInfo: '#8C4CEB',
  notificationColorInfoStrong: '#40226D',
  notificationColorPositiveWeak: '#CCEEFF',
  notificationColorPositive: '#0076B2',
  notificationColorPositiveStrong: '#054362',
  notificationColorCautionWeak: '#FEE7B9',
  notificationColorCaution: '#996600',
  notificationColorCautionStrong: '#664400',
  notificationColorNegativeWeak: '#FFE5E5',
  notificationColorNegative: '#CF1717',
  notificationColorNegativeStrong: '#5D0A0A'
};
var severityColors = {
  severityColorCustom: '#5D6596',
  severityColorUnknown: '#656C75',
  severityColorInfo: '#864AE0',
  severityColorNormal: '#017580',
  severityColorNotice: '#A65503',
  severityColorWarning: '#BD410B',
  severityColorAlert: '#CC2D37',
  severityColorCritical: '#C2306F',
  severityColorEmergency: '#B53394'
};
var elevationShadows = {
  embossShadow: '0px 1px 5px rgba(0, 0, 0, 0.07), 0px 0px 1px rgba(0, 0, 0, 0.07)',
  overlayShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  dragShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  modalShadow: '0px 50px 200px rgba(0, 0, 0, 0.3), 0px 29px 66px rgba(0, 0, 0, 0.08), 0px 29px 47px rgba(0, 0, 0, 0.08), 0px 5px 10px rgba(0, 0, 0, 0.03)'
};
var syntaxColors = {
  syntaxBlue: '#006aa3',
  syntaxBrown: '#905b04',
  syntaxGray: '#5c6773',
  syntaxGreen: '#2f612e',
  syntaxOrange: '#a44b0e',
  syntaxPink: '#b9139e',
  syntaxPurple: '#5317f2',
  syntaxRed: '#ca163d',
  syntaxTeal: '#1a7060'
};
var backgroundColors = {
  backgroundColorDialog: neutralColors.white,
  backgroundColorFloating: neutralColors.black,
  backgroundColorNavigation: '#e4e8ec',
  backgroundColorPage: '#f2f4f5',
  backgroundColorPopup: neutralColors.white,
  backgroundColorScrim: "rgba(255, 255, 255, 0.75)",
  backgroundColorSection: neutralColors.white,
  backgroundColorSidebar: '#eaedf1'
};
var shadows = {
  focusShadow: "0 0 0 2px ".concat(backgroundColors.backgroundColorPage, ", 0 0 0 5px ").concat(usageColors.focusColor),
  focusShadowInset: "inset 0 0 0 3px ".concat(usageColors.focusColor)
};
var contentColors = {
  contentColorActive: deprecatedEnterpriseColors.gray30,
  contentColorDefault: deprecatedEnterpriseColors.gray45,
  contentColorDisabled: deprecatedEnterpriseColors.gray80,
  contentColorInverted: neutralColors.white,
  contentColorMuted: deprecatedEnterpriseColors.gray60,
  contentColorAccent: '#1a8929',
  contentColorAccentStrong: '#0d4515',
  contentColorAccentWeak: '#b5e3b5',
  contentColorNegative: '#c84535',
  contentColorNegativeWeak: '#f5aca3',
  contentColorLink: '#006297',
  contentColorLinkDisabled: deprecatedEnterpriseColors.accentColorL30,
  contentColorLinkVisited: '#864ae0',
  contentColorLinkVisitedDisabled: '#e0ccff'
};
var actionColors = {
  actionColorBackgroundPrimary: '#1a8929',
  actionColorBackgroundPrimaryActive: '#125e1c',
  actionColorBackgroundPrimaryDisabled: '#98d798',
  actionColorBackgroundPrimaryHover: '#125e1c',
  actionColorBackgroundSecondary: 'transparent',
  actionColorBackgroundSecondaryActive: 'rgba(0, 0, 0, 0.12)',
  actionColorBackgroundSecondaryDisabled: 'transparent',
  actionColorBackgroundSecondaryHover: 'rgba(0, 0, 0, 0.03)',
  actionColorBackgroundSubtle: 'transparent',
  actionColorBackgroundSubtleActive: '#d8f3dc',
  actionColorBackgroundSubtleDisabled: 'transparent',
  actionColorBackgroundSubtleHover: '#d8f3dc',
  actionColorBackgroundDestructive: '#c84535',
  actionColorBackgroundDestructiveActive: '#a1382b',
  actionColorBackgroundDestructiveDisabled: '#f5aca3',
  actionColorBackgroundDestructiveHover: '#a1382b',
  actionColorBackgroundDestructiveSecondary: 'transparent',
  actionColorBackgroundDestructiveSecondaryActive: '#fad6d1',
  actionColorBackgroundDestructiveSecondaryDisabled: 'transparent',
  actionColorBackgroundDestructiveSecondaryHover: '#fad6d1',
  actionColorBorderSecondary: deprecatedEnterpriseColors.gray60,
  actionColorBorderSecondaryActive: deprecatedEnterpriseColors.gray45,
  actionColorBorderSecondaryDisabled: 'rgba(0, 0, 0, 0.3)',
  actionColorBorderSecondaryHover: 'rgba(0, 0, 0, 0.6)',
  actionColorBorderDestructiveSecondary: '#c84535',
  actionColorBorderDestructiveSecondaryActive: '#a1382b',
  actionColorBorderDestructiveSecondaryDisabled: '#f5aca3',
  actionColorBorderDestructiveSecondaryHover: '#a1382b',
  actionColorContentPrimary: contentColors.contentColorInverted,
  actionColorContentPrimaryDisabled: contentColors.contentColorInverted,
  actionColorContentSecondary: contentColors.contentColorActive,
  actionColorContentSecondaryDisabled: contentColors.contentColorDisabled,
  actionColorContentSubtleActive: contentColors.contentColorAccentStrong,
  actionColorContentSubtleHover: contentColors.contentColorAccentStrong,
  actionColorContentStandalone: contentColors.contentColorAccent,
  actionColorContentStandaloneActive: contentColors.contentColorAccentStrong,
  actionColorContentStandaloneDisabled: contentColors.contentColorAccentWeak,
  actionColorContentStandaloneHover: contentColors.contentColorAccentStrong,
  actionColorContentDestructive: contentColors.contentColorInverted,
  actionColorContentDestructiveDisabled: contentColors.contentColorInverted,
  actionColorContentDestructiveSecondary: contentColors.contentColorNegative,
  actionColorContentDestructiveSecondaryDisabled: contentColors.contentColorNegativeWeak
};
var interactiveColors = {
  interactiveColorBorder: deprecatedEnterpriseColors.gray60,
  interactiveColorBorderActive: deprecatedEnterpriseColors.gray45,
  interactiveColorBorderHover: 'rgba(0, 0, 0, 0.6)',
  interactiveColorBorderDisabled: 'rgba(0, 0, 0, 0.3)',
  interactiveColorOverlaySelected: 'rgba(0, 0, 0, 0.04)',
  interactiveColorOverlayHover: 'rgba(0, 0, 0, 0.03)',
  interactiveColorOverlayActive: 'rgba(0, 0, 0, 0.12)',
  interactiveColorOverlayDrag: 'rgba(2, 100, 215, 0.16)',
  interactiveColorBackground: 'transparent',
  interactiveColorBackgroundDisabled: 'transparent',
  interactiveColorAccent: '#1A8929',
  interactiveColorAccentError: deprecatedEnterpriseColors.errorColorD20,
  interactiveColorAccentErrorStrong: '#852d24',
  interactiveColorAccentErrorWeak: '#ea958d'
};
var sansFontFamily = "'Splunk Platform Sans', 'Proxima Nova', Roboto, Droid, 'Helvetica Neue', Helvetica, Arial, sans-serif";
var typography = {
  sansFontFamily: sansFontFamily,
  serifFontFamily: "Georgia, 'Times New Roman', Times, serif",
  monoFontFamily: "'Splunk Platform Mono', Inconsolata, Consolas, 'Droid Sans Mono', Monaco, 'Courier New', Courier, monospace",
  fontFamily: sansFontFamily,
  fontFamilyTitle: sansFontFamily,
  fontSizeSmall: '0.75rem',
  fontSize: '0.875rem',
  fontSizeLarge: '1rem',
  fontSizeXLarge: '1.25rem',
  fontSizeXXLarge: '1.5rem',
  fontWeightLight: 300,
  fontWeightNormal: 400,
  fontWeightSemiBold: 500,
  fontWeightBold: 700,
  fontWeightHeavy: 800,
  fontWeightExtraBold: 900,
  lineLengthMinimum: '34ch',
  lineLength: '80ch',
  lineHeight: '1.5',
  lineHeightSingle: '1',
  lineHeightTight: '1.2',
  lineHeightSnug: '1.25',
  lineHeightComfortable: '1.333',
  lineHeightRelaxed: '1.375',
  lineHeightSpacious: '1.429',
  lineHeightNormal: '1.5'
};
var zindexes = {
  zindexLayer: 1000,
  zindexFixedNavbar: 1030,
  zindexModalBackdrop: 1040,
  zindexModal: 1050,
  zindexPopover: 1060,
  zindexToastMessages: 2000
};
var deprecated = {
  contentColorInfo: '#006d9c',
  // @eprecated, SUI-8251
  contentColorNegativeStrong: '#812d22',
  contentColorPositive: '#307b30',
  contentColorWarning: '#7d600f',
  backgroundColor: neutralColors.white,
  // @deprecated SUI-6656
  backgroundColorHover: deprecatedEnterpriseColors.gray96,
  borderActiveColor: "rgba(0,0,0, 0.5)",
  borderLightColor: deprecatedEnterpriseColors.gray92,
  // @deprecated SUI-5981
  // @deprecated SUI-6160
  textColor: deprecatedEnterpriseColors.gray30,
  textGray: '#6b7785',
  textDisabledColor: deprecatedEnterpriseColors.gray80,
  // deprecated
  accentColorPositive: deprecatedEnterpriseColors.successColor,
  accentColorWarning: deprecatedEnterpriseColors.warningColor,
  accentColorAlert: deprecatedEnterpriseColors.alertColor,
  accentColorNegative: deprecatedEnterpriseColors.errorColor,
  linkColor: deprecatedEnterpriseColors.accentColorD10,
  linkColorHover: deprecatedEnterpriseColors.accentColor,
  border: "1px solid ".concat(usageColors.borderColor),
  brandColorL50: '#f5fbf5',
  brandColorL40: '#dff2df',
  brandColorL30: '#bee6be',
  brandColorL20: '#9ed99e',
  brandColorL10: '#7ecd7e',
  brandColorD10: '#49b849',
  brandColorD20: '#40a540',
  brandColorD30: '#389038',
  brandColorD40: '#307b30',
  brandColorD50: '#286728',
  gray25: '#31373e',
  gray22: '#2b3033',
  gray20: '#171d21',
  accentColorL50: '#ecf8ff',
  accentColorL40: '#bfe9ff',
  accentColorL20: '#3ebcff',
  accentColorL10: '#00a4fd',
  accentColorD20: '#006297',
  accentColorD30: '#005684',
  accentColorD40: '#004a71',
  accentColorD50: '#003d5e',
  errorColorL50: '#fcedec',
  errorColorL40: '#f8dcd9',
  errorColorL30: '#f1b9b3',
  errorColorL20: '#ea958d',
  errorColorD30: '#9c3529',
  errorColorD40: '#852d24',
  errorColorD50: '#6f261d',
  alertColorL50: '#fef3ec',
  alertColorL40: '#fde6d9',
  alertColorL30: '#facdb3',
  alertColorL20: '#f7b48c',
  alertColorD20: '#c2672a',
  alertColorD30: '#aa5a25',
  alertColorD40: '#914d1f',
  alertColorD50: '#79401a',
  warningColorL50: '#fff9eb',
  warningColorL40: '#fef2d7',
  warningColorL30: '#fde5ae',
  warningColorL20: '#fbd886',
  warningColorD20: '#c79915',
  warningColorD30: '#ae8613',
  warningColorD40: '#957312',
  warningColorD50: '#7d600f',
  successColorL50: '#eef6ee',
  successColorL40: '#ddecdd',
  successColorL30: '#bbd9ba',
  successColorL20: '#98c697',
  successColorD20: '#40813d',
  successColorD30: '#387135',
  successColorD40: '#2f612e',
  successColorD50: '#275126',
  infoColorL50: '#e5f0f5',
  infoColorL40: '#cce2eb',
  infoColorL30: '#99c5d7',
  infoColorL20: '#66a7c4',
  infoColorD20: '#004c6c',
  infoColorD30: '#00415d',
  infoColorD40: '#00364d',
  infoColorD50: '#002b3e',
  diverging1ColorA: '#006d9c',
  diverging1ColorB: '#ec9960',
  diverging2ColorA: '#af575a',
  diverging2ColorB: '#62b3b2',
  diverging3ColorA: '#4fa484',
  diverging3ColorB: '#f8be34',
  diverging4ColorA: '#5a4575',
  diverging4ColorB: '#708794',
  diverging5ColorA: '#294e70',
  diverging5ColorB: '#b6c75a',
  cat1Color: '#297ba5',
  cat1ColorL: '#78b9d6',
  cat2Color: '#4fa484',
  cat2ColorL: '#74d5c2',
  cat3Color: '#b6c75a',
  cat3ColorL: '#dce6a5',
  cat4Color: '#3c6188',
  cat4ColorL: '#a0b2ca',
  cat5Color: '#ec9960',
  cat5ColorL: '#fac9a7',
  cat6Color: '#a65c7d',
  cat6ColorL: '#d3a7ba',
  cat7Color: '#708794',
  cat7ColorL: '#b2c0c8',
  cat8Color: '#38b8bf',
  cat8ColorL: '#92dde2',
  cat9Color: '#ffde63',
  cat9ColorL: '#ffeeae',
  cat10Color: '#c19975',
  cat10ColorL: '#d7bfab',
  cat11Color: '#5a4575',
  cat11ColorL: '#b7acca',
  cat12Color: '#7ea77b',
  cat12ColorL: '#b2cab0',
  cat13Color: '#576d83',
  cat13ColorL: '#a5b2bf',
  cat14Color: '#d7c6b7',
  cat14ColorL: '#e9ddd4',
  cat15Color: '#339bb2',
  cat15ColorL: '#66c3d0',
  cat16Color: '#236d9b',
  cat16ColorL: '#66a7c2',
  cat17Color: '#e5dc80',
  cat17ColorL: '#f1eab7',
  cat18Color: '#96907f',
  cat18ColorL: '#c1bcb3',
  cat19Color: '#87bc65',
  cat19ColorL: '#b6d7a3',
  cat20Color: '#cf7e60',
  cat20ColorL: '#e1b2a1',
  cat21Color: '#7b5547',
  cat21ColorL: '#dec4ba',
  cat22Color: '#77d6d8',
  cat22ColorL: '#abe6e8',
  cat23Color: '#4a7f2c',
  cat23ColorL: '#91b282',
  cat24Color: '#f589ad',
  cat24ColorL: '#f8b7ce',
  cat25Color: '#6a2c5d',
  cat25ColorL: '#cba3c2',
  cat26Color: '#aaabae',
  cat26ColorL: '#cccdce',
  cat27Color: '#9a7438',
  cat27ColorL: '#c3ab89',
  cat28Color: '#a4d563',
  cat28ColorL: '#c7e6a3',
  cat29Color: '#7672a4',
  cat29ColorL: '#ada9c8',
  cat30Color: '#184b81',
  cat30ColorL: '#a4bbe0',
  syntaxBlueLight: '#006d9c',
  syntaxGreenLight: '#5ba383',
  syntaxPurpleLight: '#b19cd9',
  syntaxRedLight: '#af575a',
  interactiveColorPrimary: deprecatedEnterpriseColors.brandColor,
  statusColorInfo: deprecatedEnterpriseColors.infoColor,
  statusColorInfoWeak: deprecatedEnterpriseColors.infoColorL10,
  statusColorInfoStrong: deprecatedEnterpriseColors.infoColorD10,
  statusColorNormal: deprecatedEnterpriseColors.successColor,
  statusColorNormalWeak: deprecatedEnterpriseColors.successColorL10,
  statusColorNormalStrong: deprecatedEnterpriseColors.successColorD10,
  statusColorLow: deprecatedEnterpriseColors.warningColor,
  statusColorLowWeak: deprecatedEnterpriseColors.warningColorL10,
  statusColorLowStrong: deprecatedEnterpriseColors.warningColorD10,
  statusColorMedium: deprecatedEnterpriseColors.alertColor,
  statusColorMediumWeak: deprecatedEnterpriseColors.alertColorL10,
  statusColorMediumStrong: deprecatedEnterpriseColors.alertColorD10,
  statusColorHigh: deprecatedEnterpriseColors.errorColor,
  statusColorHighWeak: deprecatedEnterpriseColors.errorColorL10,
  statusColorHighStrong: deprecatedEnterpriseColors.errorColorD10,
  statusColorCritical: deprecatedEnterpriseColors.errorColorD10,
  statusColorCriticalWeak: deprecatedEnterpriseColors.errorColor,
  statusColorCriticalStrong: deprecatedEnterpriseColors.errorColorD20,
  draggableBackground: "url('data:image/png;base64,".concat(dragHandle, "') 0 0 / 8px 8px repeat")
};
var theme = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, deprecatedEnterpriseColors), backgroundColors), actionColors), neutralColors), notificationColors), severityColors), elevationShadows), syntaxColors), usageColors), shadows), typography), contentColors), interactiveColors), zindexes), _dataViz["default"]), deprecated);
var _default = theme;
exports["default"] = _default;
