// @splunk/webpack-configs/component.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.create = create;
exports["default"] = void 0;
var _webpack = _interopRequireDefault(require("webpack"));
var _webpackMerge = _interopRequireDefault(require("webpack-merge"));
var _terserWebpackPlugin = _interopRequireDefault(require("terser-webpack-plugin"));
var _base = require("./base.config");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function create(options) {
  return (0, _webpackMerge["default"])((0, _base.create)(options), {
    devtool: false,
    externals: /^[^/.][a-zA-Z\-0-9./]+$/,
    optimization: {
      // minimization works around a Webpack 5 + Webpack 4 compatiblity issue
      // see https://github.com/webpack/webpack/issues/17730
      // and https://splunk.atlassian.net/browse/SUI-5802
      minimize: true,
      minimizer: [new _terserWebpackPlugin["default"]({
        terserOptions: {
          compress: false,
          format: {
            beautify: true,
            comments: 'all'
          }
        },
        extractComments: false
      })]
    },
    output: {
      filename: '[name].js',
      libraryTarget: 'commonjs2'
    },
    plugins: [
    // This enables developers to use the `__DEV__` variable to create code paths that will only
    // execute in dev mode and will be removed in production.
    new _webpack["default"].DefinePlugin({
      __DEV__: 'process.env.NODE_ENV !== "production"'
    })]
  });
}
var _default = create();
exports["default"] = _default;
