// @splunk/themes/splunk-magnetic/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
var _magneticTokensCore = _interopRequireDefault(require("./magneticTokensCore"));
var _magneticTokensDark = _interopRequireDefault(require("./magneticTokensDark"));
var _magneticTokensLight = _interopRequireDefault(require("./magneticTokensLight"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 *  Handles mapping both our light and dark categorical colors to their appropriate Magnetic token value
 *  Note: Unlike other tokens dark + light tokens for these exists on either colorSchem so both sets need mapped.
 */
var mapMagneticCategoricalColors = function mapMagneticCategoricalColors() {
  var tokenMap = {
    categorical1D1: '',
    categorical1D2: '',
    categorical1D3: '',
    categorical1D4: '',
    categorical1D5: '',
    categorical1D6: '',
    categorical1D7: '',
    categorical2D1: '',
    categorical2D2: '',
    categorical2D3: '',
    categorical2D4: '',
    categorical2D5: '',
    categorical2D6: '',
    categorical2D7: '',
    categorical1L1: '',
    categorical1L2: '',
    categorical1L3: '',
    categorical1L4: '',
    categorical1L5: '',
    categorical1L6: '',
    categorical1L7: '',
    categorical2L1: '',
    categorical2L2: '',
    categorical2L3: '',
    categorical2L4: '',
    categorical2L5: '',
    categorical2L6: '',
    categorical2L7: ''
  };
  var magneticTokenLetters = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
  for (var i = 0; i < 7; i += 1) {
    var magneticCat1TokenName = "accent-".concat(magneticTokenLetters[i], "-default");
    var magneticCat2TokenName = "accent-".concat(magneticTokenLetters[i], "-weak-default");

    // Splunk light mode tokens (yes D + L are opposite of what you would think)
    var splunkLightCat1TokenName = "categorical1D".concat(i + 1);
    tokenMap[splunkLightCat1TokenName] = _magneticTokensLight["default"][magneticCat1TokenName];
    var splunkLightCat2TokenName = "categorical2D".concat(i + 1);
    tokenMap[splunkLightCat2TokenName] = _magneticTokensLight["default"][magneticCat2TokenName];

    // Splunk dark mode tokens (yes D + L are opposite of what you would think)
    var splunkDarkCat1TokenName = "categorical1L".concat(i + 1);
    tokenMap[splunkDarkCat1TokenName] = _magneticTokensDark["default"][magneticCat1TokenName];
    var splunkDarkCat2TokenName = "categorical2L".concat(i + 1);
    tokenMap[splunkDarkCat2TokenName] = _magneticTokensDark["default"][magneticCat2TokenName];
  }
  return tokenMap;
};
var magneticTypography = {
  fontFamily: "Inter, -apple-system, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif",
  fontFamilyTitle: "'Sharp Sans', -apple-system, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif",
  sansFontFamily: "Inter, -apple-system, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, sans-serif",
  monoFontFamily: "'Roboto Mono', ui-monospace, 'Courier New', Menlo, Monaco"
};
var splunkMagneticThemeCustomizer = function splunkMagneticThemeCustomizer(currentTheme) {
  var pickMagnetic = function pickMagnetic(magneticTokenName) {
    switch (currentTheme.colorScheme) {
      case 'light':
        return _magneticTokensLight["default"][magneticTokenName];
      case 'dark':
        return _magneticTokensDark["default"][magneticTokenName];
      default:
        if (false) {
          // ensure all possible known types had case statements: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking
          var exhaustiveCheck = currentTheme.colorScheme;
          // eslint-disable-next-line no-console
          console.log("Unknown theme colorScheme, ".concat(exhaustiveCheck, "."));
        }
        return '';
    }
  };
  return _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, currentTheme), magneticTypography), mapMagneticCategoricalColors()), {}, {
    accentColor: pickMagnetic('brand-accent-border-default'),
    actionColorBackgroundDestructive: pickMagnetic('negative-bg-default'),
    actionColorBackgroundDestructiveActive: pickMagnetic('negative-bg-active'),
    actionColorBackgroundDestructiveDisabled: pickMagnetic('negative-bg-disabled'),
    actionColorBackgroundDestructiveHover: pickMagnetic('negative-bg-hover'),
    actionColorBackgroundDestructiveSecondary: pickMagnetic('interact-bg-weak-disabled'),
    actionColorBackgroundDestructiveSecondaryActive: pickMagnetic('negative-bg-weak-active'),
    actionColorBackgroundDestructiveSecondaryDisabled: pickMagnetic('negative-bg-weak-disabled'),
    actionColorBackgroundDestructiveSecondaryHover: pickMagnetic('negative-bg-weak-hover'),
    actionColorBackgroundPrimary: pickMagnetic('interact-bg-default'),
    actionColorBackgroundPrimaryActive: pickMagnetic('interact-bg-active'),
    actionColorBackgroundPrimaryDisabled: pickMagnetic('interact-bg-disabled'),
    actionColorBackgroundPrimaryHover: pickMagnetic('interact-bg-hover'),
    actionColorBackgroundSecondary: pickMagnetic('interact-bg-weak-default'),
    actionColorBackgroundSecondaryActive: pickMagnetic('interact-bg-weak-active'),
    actionColorBackgroundSecondaryDisabled: pickMagnetic('interact-bg-weak-disabled'),
    actionColorBackgroundSecondaryHover: pickMagnetic('interact-bg-weak-hover'),
    actionColorBackgroundSubtle: pickMagnetic('interact-bg-weak-default'),
    actionColorBackgroundSubtleActive: pickMagnetic('interact-bg-weak-active'),
    actionColorBackgroundSubtleDisabled: pickMagnetic('interact-bg-weak-disabled'),
    actionColorBackgroundSubtleHover: pickMagnetic('interact-bg-weak-hover'),
    actionColorBorder: pickMagnetic('interact-border-default'),
    actionColorBorderDestructiveSecondary: pickMagnetic('negative-border-default'),
    actionColorBorderDestructiveSecondaryActive: pickMagnetic('negative-border-active'),
    actionColorBorderDestructiveSecondaryDisabled: pickMagnetic('negative-border-disabled'),
    actionColorBorderDestructiveSecondaryHover: pickMagnetic('negative-border-hover'),
    actionColorBorderSecondary: pickMagnetic('interact-border-default'),
    actionColorBorderSecondaryActive: pickMagnetic('interact-border-active'),
    actionColorBorderSecondaryDisabled: pickMagnetic('interact-border-disabled'),
    actionColorBorderSecondaryHover: pickMagnetic('interact-border-hover'),
    actionColorBorderWeak: pickMagnetic('interact-border-weak-default'),
    actionColorPrimary: pickMagnetic('interact-bg-default'),
    actionColorDestructive: pickMagnetic('negative-bg-default'),
    actionColorDestructiveStrong: pickMagnetic('negative-bg-strong-default'),
    actionColorDestructiveWeak: pickMagnetic('negative-bg-weak-default'),
    actionColorContentPrimary: pickMagnetic('interact-text-in-default'),
    actionColorContentPrimaryDisabled: pickMagnetic('interact-text-in-disabled'),
    actionColorContentSecondary: pickMagnetic('interact-text-default'),
    actionColorContentSecondaryDisabled: pickMagnetic('interact-text-disabled'),
    actionColorContentSubtleActive: pickMagnetic('interact-text-active'),
    actionColorContentSubtleHover: pickMagnetic('interact-text-hover'),
    actionColorContentStandalone: pickMagnetic('interact-text-default'),
    actionColorContentStandaloneActive: pickMagnetic('interact-text-active'),
    actionColorContentStandaloneDisabled: pickMagnetic('interact-text-disabled'),
    actionColorContentStandaloneHover: pickMagnetic('interact-text-hover'),
    actionColorContentDestructive: pickMagnetic('negative-text-in-default'),
    actionColorContentDestructiveDisabled: pickMagnetic('negative-text-in-disabled'),
    actionColorContentDestructiveSecondary: pickMagnetic('negative-text-default'),
    actionColorContentDestructiveSecondaryDisabled: pickMagnetic('negative-text-disabled'),
    backgroundColorDialog: pickMagnetic('control-bg-weak-default'),
    backgroundColorInverted: pickMagnetic('inverse-bg-default'),
    backgroundColorPage: pickMagnetic('base-bg-default'),
    backgroundColorNavigation: currentTheme.colorScheme === 'light' ? "linear-gradient(90deg, ".concat(_magneticTokensLight["default"]['inverse-bg-gradient-start'], ", ").concat(_magneticTokensLight["default"]['inverse-bg-gradient-end'], ")") : "linear-gradient(90deg, ".concat(_magneticTokensDark["default"]['inverse-bg-gradient-start'], ", ").concat(_magneticTokensDark["default"]['inverse-bg-gradient-end'], ")"),
    backgroundColorPopup: pickMagnetic('base-bg-weak-default'),
    backgroundColorScrim: pickMagnetic('control-bg-weak-default'),
    backgroundColorSidebar: pickMagnetic('base-bg-default'),
    backgroundColorSection: pickMagnetic('base-bg-weak-default'),
    borderColor: pickMagnetic('base-border-default'),
    borderColorStrong: pickMagnetic('base-border-strong-default'),
    borderColorWeak: pickMagnetic('base-border-weak-default'),
    borderRadius: '6px',
    contentColorActive: pickMagnetic('base-text-default'),
    contentColorDefault: pickMagnetic('base-text-default'),
    contentColorDisabled: pickMagnetic('base-text-disabled'),
    contentColorInverted: pickMagnetic('base-text-in-default'),
    contentColorMuted: pickMagnetic('base-text-weak-default'),
    contentColorNegative: pickMagnetic('negative-text-default'),
    embossShadow: '0px 2px 5px 0px rgba(0, 0, 0, 0.05)',
    focusColor: pickMagnetic('control-border-focus'),
    focusShadow: currentTheme.colorScheme === 'light' ? "0 0 0 2px ".concat(_magneticTokensLight["default"]['interact-bg-weak-active'], ",0 0 0 4px ").concat(_magneticTokensLight["default"]['control-border-focus']) : "0 0 0 2px ".concat(_magneticTokensDark["default"]['interact-bg-weak-active'], ",0 0 0 4px ").concat(_magneticTokensDark["default"]['control-border-focus']),
    inputBorderWidth: '2px',
    interactiveColorAccent: pickMagnetic('control-bg-strong-default'),
    interactiveColorAccentError: pickMagnetic('negative-border-medium-default'),
    interactiveColorAccentErrorStrong: pickMagnetic('negative-bg-strong-default'),
    interactiveColorAccentErrorWeak: pickMagnetic('negative-bg-weak-default'),
    interactiveColorBackground: pickMagnetic('control-bg-default'),
    interactiveColorBackgroundDisabled: pickMagnetic('control-bg-disabled'),
    interactiveColorBorder: pickMagnetic('control-border-default'),
    interactiveColorBorderDisabled: pickMagnetic('control-border-disabled'),
    interactiveColorBorderDisabledError: pickMagnetic('negative-border-weak-disabled'),
    interactiveColorBorderHover: pickMagnetic('control-border-hover'),
    linkColor: pickMagnetic('interact-text-default'),
    modalShadow: '0px 4px 12px 0px rgba(0, 0, 0, 0.12)',
    neutral50: currentTheme.colorScheme === 'light' ? '#EFF0F1' : '#171B1F',
    neutral100: currentTheme.colorScheme === 'light' ? _magneticTokensCore["default"]['neutral-95'] : _magneticTokensCore["default"]['neutral-25'],
    neutral200: currentTheme.colorScheme === 'light' ? _magneticTokensCore["default"]['neutral-90'] : _magneticTokensCore["default"]['neutral-35'],
    neutral300: currentTheme.colorScheme === 'light' ? _magneticTokensCore["default"]['neutral-85'] : _magneticTokensCore["default"]['neutral-40'],
    neutral400: currentTheme.colorScheme === 'light' ? _magneticTokensCore["default"]['neutral-80'] : _magneticTokensCore["default"]['neutral-50'],
    neutral500: currentTheme.colorScheme === 'light' ? _magneticTokensCore["default"]['neutral-70'] : _magneticTokensCore["default"]['neutral-60'],
    notificationColorNegativeWeak: pickMagnetic('negative-bg-weak-default'),
    notificationColorNegativeStrong: pickMagnetic('negative-bg-strong-default'),
    notificationColorNegative: pickMagnetic('negative-bg-default'),
    overlayShadow: '0px 4px 12px 0px rgba(0,0,0,0.18)',
    statusColorHigh: pickMagnetic('negative-icon-default'),
    statusColorHighStrong: pickMagnetic('negative-border-strong-default'),
    statusColorHighWeak: pickMagnetic('negative-bg-medium-default'),
    statusColorInfo: pickMagnetic('info-icon-default'),
    statusColorInfoStrong: currentTheme.colorScheme === 'light' ? (0, _tinycolor["default"])(pickMagnetic('info-icon-default')).darken(10).toRgbString() : (0, _tinycolor["default"])(pickMagnetic('info-icon-default')).lighten(10).toRgbString(),
    statusColorInfoWeak: pickMagnetic('info-bg-medium-default'),
    statusColorMedium: pickMagnetic('warning-icon-strong-default'),
    statusColorMediumStrong: pickMagnetic('warning-border-strong-default'),
    statusColorMediumWeak: pickMagnetic('warning-bg-medium-default'),
    statusColorNormal: pickMagnetic('positive-icon-strong-default'),
    statusColorNormalStrong: pickMagnetic('positive-border-strong-default'),
    statusColorNormalWeak: pickMagnetic('positive-bg-medium-default'),
    syntaxBlue: currentTheme.colorScheme === 'light' ? '#0f7190' : '#6cd0f0',
    syntaxBrown: currentTheme.colorScheme === 'light' ? '#8b5804' : '#fccf87',
    syntaxGray: currentTheme.colorScheme === 'light' ? '#666666' : '#909090',
    syntaxGreen: currentTheme.colorScheme === 'light' ? '#556f0b' : '#cef06c',
    syntaxOrange: currentTheme.colorScheme === 'light' ? '#a64708' : '#f7933f',
    syntaxPink: currentTheme.colorScheme === 'light' ? '#b9139e' : '#f494e5',
    syntaxPurple: currentTheme.colorScheme === 'light' ? '#703ef4' : '#ab74f1',
    syntaxRed: currentTheme.colorScheme === 'light' ? '#ca1c41' : '#e9627f',
    syntaxTeal: currentTheme.colorScheme === 'light' ? '#1b7464' : '#45d4ba'
  });
};
var _default = splunkMagneticThemeCustomizer;
exports["default"] = _default;
