// intl-tel-input/svelte/build/IntlTelInput.mjs
var Fn = Object.defineProperty;
var Te = (e) => {
  throw TypeError(e);
};
var Hn = (e, t, n) => t in e ? Fn(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var tt = (e, t, n) => Hn(e, typeof t != "symbol" ? t + "" : t, n), te = (e, t, n) => t.has(e) || Te("Cannot " + n);
var A = (e, t, n) => (te(e, t, "read from private field"), n ? n.call(e) : t.get(e)), Z = (e, t, n) => t.has(e) ? Te("Cannot add the same private member more than once") : t instanceof WeakSet ? t.add(e) : t.set(e, n), bt = (e, t, n, i) => (te(e, t, "write to private field"), i ? i.call(e, n) : t.set(e, n), n), lt = (e, t, n) => (te(e, t, "access private method"), n);
import { onMount as Vn, onDestroy as Wn } from "svelte";
const jn = "5";
var qe;
typeof window < "u" && ((qe = window.__svelte ?? (window.__svelte = {})).v ?? (qe.v = /* @__PURE__ */ new Set())).add(jn);
const Gn = 1, zn = 4, Kn = 8, Yn = 16, qn = 2, P = Symbol(), $n = "http://www.w3.org/1999/xhtml", Xn = "@attach", ce = !1;
var $e = Array.isArray, Qn = Array.prototype.indexOf, Zn = Object.defineProperty, Wt = Object.getOwnPropertyDescriptor, Jn = Object.getOwnPropertyDescriptors, ti = Object.prototype, ei = Array.prototype, Xe = Object.getPrototypeOf;
function ni(e) {
  for (var t = 0; t < e.length; t++)
    e[t]();
}
function Qe() {
  var e, t, n = new Promise((i, s) => {
    e = i, t = s;
  });
  return { promise: n, resolve: e, reject: t };
}
const k = 2, Gt = 4, ye = 8, ve = 1 << 24, mt = 16, yt = 32, Pt = 64, Ze = 128, z = 512, U = 1024, W = 2048, rt = 4096, Nt = 8192, J = 16384, be = 32768, Pe = 65536, Oe = 1 << 17, Je = 1 << 18, Zt = 1 << 19, ii = 1 << 20, Ct = 32768, de = 1 << 21, we = 1 << 22, et = 1 << 23, ft = Symbol("$state"), si = Symbol("legacy props"), ri = Symbol(""), wt = new class extends Error {
  constructor() {
    super(...arguments);
    tt(this, "name", "StaleReactionError");
    tt(this, "message", "The reaction that called `getAbortSignal()` was re-run or destroyed");
  }
}();
function oi() {
  throw new Error("https://svelte.dev/e/async_derived_orphan");
}
function li(e) {
  throw new Error("https://svelte.dev/e/effect_in_teardown");
}
function ai() {
  throw new Error("https://svelte.dev/e/effect_in_unowned_derived");
}
function ui(e) {
  throw new Error("https://svelte.dev/e/effect_orphan");
}
function ci() {
  throw new Error("https://svelte.dev/e/effect_update_depth_exceeded");
}
function di(e) {
  throw new Error("https://svelte.dev/e/props_invalid_value");
}
function fi() {
  throw new Error("https://svelte.dev/e/state_descriptors_fixed");
}
function hi() {
  throw new Error("https://svelte.dev/e/state_prototype_fixed");
}
function pi() {
  throw new Error("https://svelte.dev/e/state_unsafe_mutation");
}
function _i() {
  console.warn("https://svelte.dev/e/select_multiple_invalid_value");
}
function tn(e) {
  return e === this.v;
}
function Ci(e, t) {
  return e != e ? t == t : e !== t || e !== null && typeof e == "object" || typeof e == "function";
}
function gi(e) {
  return !Ci(e, this.v);
}
let mi = !1, X = null;
function zt(e) {
  X = e;
}
function yi(e, t = !1, n) {
  X = {
    p: X,
    i: !1,
    c: null,
    e: null,
    s: e,
    x: null,
    l: null
  };
}
function vi(e) {
  var t = (
    /** @type {ComponentContext} */
    X
  ), n = t.e;
  if (n !== null) {
    t.e = null;
    for (var i of n)
      _n(i);
  }
  return e !== void 0 && (t.x = e), t.i = !0, X = t.p, e ?? /** @type {T} */
  {};
}
function en() {
  return !0;
}
let It = [];
function bi() {
  var e = It;
  It = [], ni(e);
}
function Jt(e) {
  if (It.length === 0) {
    var t = It;
    queueMicrotask(() => {
      t === It && bi();
    });
  }
  It.push(e);
}
function wi(e) {
  var t = N;
  if (t === null)
    return w.f |= et, e;
  if ((t.f & be) === 0) {
    if ((t.f & Ze) === 0)
      throw e;
    t.b.error(e);
  } else
    Kt(e, t);
}
function Kt(e, t) {
  for (; t !== null; ) {
    if ((t.f & Ze) !== 0)
      try {
        t.b.error(e);
        return;
      } catch (n) {
        e = n;
      }
    t = t.parent;
  }
  throw e;
}
const Ii = -7169;
function B(e, t) {
  e.f = e.f & Ii | t;
}
function Ie(e) {
  (e.f & z) !== 0 || e.deps === null ? B(e, U) : B(e, rt);
}
function nn(e) {
  if (e !== null)
    for (const t of e)
      (t.f & k) === 0 || (t.f & Ct) === 0 || (t.f ^= Ct, nn(
        /** @type {Derived} */
        t.deps
      ));
}
function Ei(e, t, n) {
  (e.f & W) !== 0 ? t.add(e) : (e.f & rt) !== 0 && n.add(e), nn(e.deps), B(e, U);
}
const Bt = /* @__PURE__ */ new Set();
let S = null, R = null, $ = [], Ee = null, fe = !1;
var Dt, At, ct, dt, Ut, Lt, St, K, he, pe, sn, rn;
const Qt = class Qt {
  constructor() {
    Z(this, K);
    tt(this, "committed", !1);
    /**
     * The current values of any sources that are updated in this batch
     * They keys of this map are identical to `this.#previous`
     * @type {Map<Source, any>}
     */
    tt(this, "current", /* @__PURE__ */ new Map());
    /**
     * The values of any sources that are updated in this batch _before_ those updates took place.
     * They keys of this map are identical to `this.#current`
     * @type {Map<Source, any>}
     */
    tt(this, "previous", /* @__PURE__ */ new Map());
    /**
     * When the batch is committed (and the DOM is updated), we need to remove old branches
     * and append new ones by calling the functions added inside (if/each/key/etc) blocks
     * @type {Set<() => void>}
     */
    Z(this, Dt, /* @__PURE__ */ new Set());
    /**
     * If a fork is discarded, we need to destroy any effects that are no longer needed
     * @type {Set<(batch: Batch) => void>}
     */
    Z(this, At, /* @__PURE__ */ new Set());
    /**
     * The number of async effects that are currently in flight
     */
    Z(this, ct, 0);
    /**
     * The number of async effects that are currently in flight, _not_ inside a pending boundary
     */
    Z(this, dt, 0);
    /**
     * A deferred that resolves when the batch is committed, used with `settled()`
     * TODO replace with Promise.withResolvers once supported widely enough
     * @type {{ promise: Promise<void>, resolve: (value?: any) => void, reject: (reason: unknown) => void } | null}
     */
    Z(this, Ut, null);
    /**
     * Deferred effects (which run after async work has completed) that are DIRTY
     * @type {Set<Effect>}
     */
    Z(this, Lt, /* @__PURE__ */ new Set());
    /**
     * Deferred effects that are MAYBE_DIRTY
     * @type {Set<Effect>}
     */
    Z(this, St, /* @__PURE__ */ new Set());
    /**
     * A set of branches that still exist, but will be destroyed when this batch
     * is committed — we skip over these during `process`
     * @type {Set<Effect>}
     */
    tt(this, "skipped_effects", /* @__PURE__ */ new Set());
    tt(this, "is_fork", !1);
  }
  is_deferred() {
    return this.is_fork || A(this, dt) > 0;
  }
  /**
   *
   * @param {Effect[]} root_effects
   */
  process(t) {
    var s;
    $ = [], this.apply();
    var n = [], i = [];
    for (const r of t)
      lt(this, K, he).call(this, r, n, i);
    this.is_fork || lt(this, K, sn).call(this), this.is_deferred() ? (lt(this, K, pe).call(this, i), lt(this, K, pe).call(this, n)) : (S = null, Re(i), Re(n), (s = A(this, Ut)) == null || s.resolve()), R = null;
  }
  /**
   * Associate a change to a given source with the current
   * batch, noting its previous and current values
   * @param {Source} source
   * @param {any} value
   */
  capture(t, n) {
    n !== P && !this.previous.has(t) && this.previous.set(t, n), (t.f & et) === 0 && (this.current.set(t, t.v), R == null || R.set(t, t.v));
  }
  activate() {
    S = this, this.apply();
  }
  deactivate() {
    S === this && (S = null, R = null);
  }
  flush() {
    if (this.activate(), $.length > 0) {
      if (Ni(), S !== null && S !== this)
        return;
    } else A(this, ct) === 0 && this.process([]);
    this.deactivate();
  }
  discard() {
    for (const t of A(this, At)) t(this);
    A(this, At).clear();
  }
  /**
   *
   * @param {boolean} blocking
   */
  increment(t) {
    bt(this, ct, A(this, ct) + 1), t && bt(this, dt, A(this, dt) + 1);
  }
  /**
   *
   * @param {boolean} blocking
   */
  decrement(t) {
    bt(this, ct, A(this, ct) - 1), t && bt(this, dt, A(this, dt) - 1), this.revive();
  }
  revive() {
    for (const t of A(this, Lt))
      A(this, St).delete(t), B(t, W), Tt(t);
    for (const t of A(this, St))
      B(t, rt), Tt(t);
    this.flush();
  }
  /** @param {() => void} fn */
  oncommit(t) {
    A(this, Dt).add(t);
  }
  /** @param {(batch: Batch) => void} fn */
  ondiscard(t) {
    A(this, At).add(t);
  }
  settled() {
    return (A(this, Ut) ?? bt(this, Ut, Qe())).promise;
  }
  static ensure() {
    if (S === null) {
      const t = S = new Qt();
      Bt.add(S), Qt.enqueue(() => {
        S === t && t.flush();
      });
    }
    return S;
  }
  /** @param {() => void} task */
  static enqueue(t) {
    Jt(t);
  }
  apply() {
  }
};
Dt = new WeakMap(), At = new WeakMap(), ct = new WeakMap(), dt = new WeakMap(), Ut = new WeakMap(), Lt = new WeakMap(), St = new WeakMap(), K = new WeakSet(), /**
 * Traverse the effect tree, executing effects or stashing
 * them for later execution as appropriate
 * @param {Effect} root
 * @param {Effect[]} effects
 * @param {Effect[]} render_effects
 */
he = function(t, n, i) {
  t.f ^= U;
  for (var s = t.first, r = null; s !== null; ) {
    var o = s.f, l = (o & (yt | Pt)) !== 0, a = l && (o & U) !== 0, u = a || (o & Nt) !== 0 || this.skipped_effects.has(s);
    if (!u && s.fn !== null) {
      l ? s.f ^= U : r !== null && (o & (Gt | ye | ve)) !== 0 ? r.b.defer_effect(s) : (o & Gt) !== 0 ? n.push(s) : xt(s) && ((o & mt) !== 0 && A(this, Lt).add(s), Mt(s));
      var c = s.first;
      if (c !== null) {
        s = c;
        continue;
      }
    }
    var d = s.parent;
    for (s = s.next; s === null && d !== null; )
      d === r && (r = null), s = d.next, d = d.parent;
  }
}, /**
 * @param {Effect[]} effects
 */
pe = function(t) {
  for (var n = 0; n < t.length; n += 1)
    Ei(t[n], A(this, Lt), A(this, St));
}, sn = function() {
  if (A(this, dt) === 0) {
    for (const t of A(this, Dt)) t();
    A(this, Dt).clear();
  }
  A(this, ct) === 0 && lt(this, K, rn).call(this);
}, rn = function() {
  var s;
  if (Bt.size > 1) {
    this.previous.clear();
    var t = R, n = !0;
    for (const r of Bt) {
      if (r === this) {
        n = !1;
        continue;
      }
      const o = [];
      for (const [a, u] of this.current) {
        if (r.current.has(a))
          if (n && u !== r.current.get(a))
            r.current.set(a, u);
          else
            continue;
        o.push(a);
      }
      if (o.length === 0)
        continue;
      const l = [...r.current.keys()].filter((a) => !this.current.has(a));
      if (l.length > 0) {
        var i = $;
        $ = [];
        const a = /* @__PURE__ */ new Set(), u = /* @__PURE__ */ new Map();
        for (const c of o)
          on(c, l, a, u);
        if ($.length > 0) {
          S = r, r.apply();
          for (const c of $)
            lt(s = r, K, he).call(s, c, [], []);
          r.deactivate();
        }
        $ = i;
      }
    }
    S = null, R = t;
  }
  this.committed = !0, Bt.delete(this);
};
let Yt = Qt;
function Ni() {
  var e = ht;
  fe = !0;
  var t = null;
  try {
    var n = 0;
    for ($t(!0); $.length > 0; ) {
      var i = Yt.ensure();
      if (n++ > 1e3) {
        var s, r;
        Di();
      }
      i.process($), nt.clear();
    }
  } finally {
    fe = !1, $t(e), Ee = null;
  }
}
function Di() {
  try {
    ci();
  } catch (e) {
    Kt(e, Ee);
  }
}
let Y = null;
function Re(e) {
  var t = e.length;
  if (t !== 0) {
    for (var n = 0; n < t; ) {
      var i = e[n++];
      if ((i.f & (J | Nt)) === 0 && xt(i) && (Y = /* @__PURE__ */ new Set(), Mt(i), i.deps === null && i.first === null && i.nodes === null && (i.teardown === null && i.ac === null ? vn(i) : i.fn = null), (Y == null ? void 0 : Y.size) > 0)) {
        nt.clear();
        for (const s of Y) {
          if ((s.f & (J | Nt)) !== 0) continue;
          const r = [s];
          let o = s.parent;
          for (; o !== null; )
            Y.has(o) && (Y.delete(o), r.push(o)), o = o.parent;
          for (let l = r.length - 1; l >= 0; l--) {
            const a = r[l];
            (a.f & (J | Nt)) === 0 && Mt(a);
          }
        }
        Y.clear();
      }
    }
    Y = null;
  }
}
function on(e, t, n, i) {
  if (!n.has(e) && (n.add(e), e.reactions !== null))
    for (const s of e.reactions) {
      const r = s.f;
      (r & k) !== 0 ? on(
        /** @type {Derived} */
        s,
        t,
        n,
        i
      ) : (r & (we | mt)) !== 0 && (r & W) === 0 && ln(s, t, i) && (B(s, W), Tt(
        /** @type {Effect} */
        s
      ));
    }
}
function ln(e, t, n) {
  const i = n.get(e);
  if (i !== void 0) return i;
  if (e.deps !== null)
    for (const s of e.deps) {
      if (t.includes(s))
        return !0;
      if ((s.f & k) !== 0 && ln(
        /** @type {Derived} */
        s,
        t,
        n
      ))
        return n.set(
          /** @type {Derived} */
          s,
          !0
        ), !0;
    }
  return n.set(e, !1), !1;
}
function Tt(e) {
  for (var t = Ee = e; t.parent !== null; ) {
    t = t.parent;
    var n = t.f;
    if (fe && t === N && (n & mt) !== 0 && (n & Je) === 0)
      return;
    if ((n & (Pt | yt)) !== 0) {
      if ((n & U) === 0) return;
      t.f ^= U;
    }
  }
  $.push(t);
}
function Ai(e, t, n, i) {
  const s = Ne;
  if (n.length === 0 && e.length === 0) {
    i(t.map(s));
    return;
  }
  var r = S, o = (
    /** @type {Effect} */
    N
  ), l = Li();
  function a() {
    Promise.all(n.map((u) => /* @__PURE__ */ Si(u))).then((u) => {
      l();
      try {
        i([...t.map(s), ...u]);
      } catch (c) {
        (o.f & J) === 0 && Kt(c, o);
      }
      r == null || r.deactivate(), qt();
    }).catch((u) => {
      Kt(u, o);
    });
  }
  e.length > 0 ? Promise.all(e).then(() => {
    l();
    try {
      return a();
    } finally {
      r == null || r.deactivate(), qt();
    }
  }) : a();
}
function Li() {
  var e = N, t = w, n = X, i = S;
  return function(r = !0) {
    st(e), Q(t), zt(n), r && (i == null || i.activate());
  };
}
function qt() {
  st(null), Q(null), zt(null);
}
// @__NO_SIDE_EFFECTS__
function Ne(e) {
  var t = k | W, n = w !== null && (w.f & k) !== 0 ? (
    /** @type {Derived} */
    w
  ) : null;
  return N !== null && (N.f |= Zt), {
    ctx: X,
    deps: null,
    effects: null,
    equals: tn,
    f: t,
    fn: e,
    reactions: null,
    rv: 0,
    v: (
      /** @type {V} */
      P
    ),
    wv: 0,
    parent: n ?? N,
    ac: null
  };
}
// @__NO_SIDE_EFFECTS__
function Si(e, t, n) {
  let i = (
    /** @type {Effect | null} */
    N
  );
  i === null && oi();
  var s = (
    /** @type {Boundary} */
    i.b
  ), r = (
    /** @type {Promise<V>} */
    /** @type {unknown} */
    void 0
  ), o = dn(
    /** @type {V} */
    P
  ), l = !w, a = /* @__PURE__ */ new Map();
  return ji(() => {
    var p;
    var u = Qe();
    r = u.promise;
    try {
      Promise.resolve(e()).then(u.resolve, u.reject).then(() => {
        c === S && c.committed && c.deactivate(), qt();
      });
    } catch (C) {
      u.reject(C), qt();
    }
    var c = (
      /** @type {Batch} */
      S
    );
    if (l) {
      var d = s.is_rendered();
      s.update_pending_count(1), c.increment(d), (p = a.get(c)) == null || p.reject(wt), a.delete(c), a.set(c, u);
    }
    const f = (C, y = void 0) => {
      if (c.activate(), y)
        y !== wt && (o.f |= et, Ce(o, y));
      else {
        (o.f & et) !== 0 && (o.f ^= et), Ce(o, C);
        for (const [b, g] of a) {
          if (a.delete(b), b === c) break;
          g.reject(wt);
        }
      }
      l && (s.update_pending_count(-1), c.decrement(d));
    };
    u.promise.then(f, (C) => f(null, C || "unknown"));
  }), pn(() => {
    for (const u of a.values())
      u.reject(wt);
  }), new Promise((u) => {
    function c(d) {
      function f() {
        d === r ? u(o) : c(r);
      }
      d.then(f, f);
    }
    c(r);
  });
}
// @__NO_SIDE_EFFECTS__
function Ti(e) {
  const t = /* @__PURE__ */ Ne(e);
  return t.equals = gi, t;
}
function an(e) {
  var t = e.effects;
  if (t !== null) {
    e.effects = null;
    for (var n = 0; n < t.length; n += 1)
      gt(
        /** @type {Effect} */
        t[n]
      );
  }
}
function Pi(e) {
  for (var t = e.parent; t !== null; ) {
    if ((t.f & k) === 0)
      return (t.f & J) === 0 ? (
        /** @type {Effect} */
        t
      ) : null;
    t = t.parent;
  }
  return null;
}
function De(e) {
  var t, n = N;
  st(Pi(e));
  try {
    e.f &= ~Ct, an(e), t = En(e);
  } finally {
    st(n);
  }
  return t;
}
function un(e) {
  var t = De(e);
  if (!e.equals(t) && (e.wv = wn(), (!(S != null && S.is_fork) || e.deps === null) && (e.v = t, e.deps === null))) {
    B(e, U);
    return;
  }
  it || (R !== null ? (hn() || S != null && S.is_fork) && R.set(e, t) : Ie(e));
}
let _e = /* @__PURE__ */ new Set();
const nt = /* @__PURE__ */ new Map();
let cn = !1;
function dn(e, t) {
  var n = {
    f: 0,
    // TODO ideally we could skip this altogether, but it causes type errors
    v: e,
    reactions: null,
    equals: tn,
    rv: 0,
    wv: 0
  };
  return n;
}
// @__NO_SIDE_EFFECTS__
function G(e, t) {
  const n = dn(e);
  return Yi(n), n;
}
function V(e, t, n = !1) {
  w !== null && // since we are untracking the function inside `$inspect.with` we need to add this check
  // to ensure we error if state is set inside an inspect effect
  (!q || (w.f & Oe) !== 0) && en() && (w.f & (k | mt | we | Oe)) !== 0 && !(x != null && x.includes(e)) && pi();
  let i = n ? Et(t) : t;
  return Ce(e, i);
}
function Ce(e, t) {
  if (!e.equals(t)) {
    var n = e.v;
    it ? nt.set(e, t) : nt.set(e, n), e.v = t;
    var i = Yt.ensure();
    if (i.capture(e, n), (e.f & k) !== 0) {
      const s = (
        /** @type {Derived} */
        e
      );
      (e.f & W) !== 0 && De(s), Ie(s);
    }
    e.wv = wn(), fn(e, W), N !== null && (N.f & U) !== 0 && (N.f & (yt | Pt)) === 0 && (j === null ? qi([e]) : j.push(e)), !i.is_fork && _e.size > 0 && !cn && Oi();
  }
  return t;
}
function Oi() {
  cn = !1;
  var e = ht;
  $t(!0);
  const t = Array.from(_e);
  try {
    for (const n of t)
      (n.f & U) !== 0 && B(n, rt), xt(n) && Mt(n);
  } finally {
    $t(e);
  }
  _e.clear();
}
function ee(e) {
  V(e, e.v + 1);
}
function fn(e, t) {
  var n = e.reactions;
  if (n !== null)
    for (var i = n.length, s = 0; s < i; s++) {
      var r = n[s], o = r.f, l = (o & W) === 0;
      if (l && B(r, t), (o & k) !== 0) {
        var a = (
          /** @type {Derived} */
          r
        );
        R == null || R.delete(a), (o & Ct) === 0 && (o & z && (r.f |= Ct), fn(a, rt));
      } else l && ((o & mt) !== 0 && Y !== null && Y.add(
        /** @type {Effect} */
        r
      ), Tt(
        /** @type {Effect} */
        r
      ));
    }
}
function Et(e) {
  if (typeof e != "object" || e === null || ft in e)
    return e;
  const t = Xe(e);
  if (t !== ti && t !== ei)
    return e;
  var n = /* @__PURE__ */ new Map(), i = $e(e), s = /* @__PURE__ */ G(0), r = pt, o = (l) => {
    if (pt === r)
      return l();
    var a = w, u = pt;
    Q(null), Ue(r);
    var c = l();
    return Q(a), Ue(u), c;
  };
  return i && n.set("length", /* @__PURE__ */ G(
    /** @type {any[]} */
    e.length
  )), new Proxy(
    /** @type {any} */
    e,
    {
      defineProperty(l, a, u) {
        (!("value" in u) || u.configurable === !1 || u.enumerable === !1 || u.writable === !1) && fi();
        var c = n.get(a);
        return c === void 0 ? c = o(() => {
          var d = /* @__PURE__ */ G(u.value);
          return n.set(a, d), d;
        }) : V(c, u.value, !0), !0;
      },
      deleteProperty(l, a) {
        var u = n.get(a);
        if (u === void 0) {
          if (a in l) {
            const c = o(() => /* @__PURE__ */ G(P));
            n.set(a, c), ee(s);
          }
        } else
          V(u, P), ee(s);
        return !0;
      },
      get(l, a, u) {
        var p;
        if (a === ft)
          return e;
        var c = n.get(a), d = a in l;
        if (c === void 0 && (!d || (p = Wt(l, a)) != null && p.writable) && (c = o(() => {
          var C = Et(d ? l[a] : P), y = /* @__PURE__ */ G(C);
          return y;
        }), n.set(a, c)), c !== void 0) {
          var f = I(c);
          return f === P ? void 0 : f;
        }
        return Reflect.get(l, a, u);
      },
      getOwnPropertyDescriptor(l, a) {
        var u = Reflect.getOwnPropertyDescriptor(l, a);
        if (u && "value" in u) {
          var c = n.get(a);
          c && (u.value = I(c));
        } else if (u === void 0) {
          var d = n.get(a), f = d == null ? void 0 : d.v;
          if (d !== void 0 && f !== P)
            return {
              enumerable: !0,
              configurable: !0,
              value: f,
              writable: !0
            };
        }
        return u;
      },
      has(l, a) {
        var f;
        if (a === ft)
          return !0;
        var u = n.get(a), c = u !== void 0 && u.v !== P || Reflect.has(l, a);
        if (u !== void 0 || N !== null && (!c || (f = Wt(l, a)) != null && f.writable)) {
          u === void 0 && (u = o(() => {
            var p = c ? Et(l[a]) : P, C = /* @__PURE__ */ G(p);
            return C;
          }), n.set(a, u));
          var d = I(u);
          if (d === P)
            return !1;
        }
        return c;
      },
      set(l, a, u, c) {
        var h;
        var d = n.get(a), f = a in l;
        if (i && a === "length")
          for (var p = u; p < /** @type {Source<number>} */
          d.v; p += 1) {
            var C = n.get(p + "");
            C !== void 0 ? V(C, P) : p in l && (C = o(() => /* @__PURE__ */ G(P)), n.set(p + "", C));
          }
        if (d === void 0)
          (!f || (h = Wt(l, a)) != null && h.writable) && (d = o(() => /* @__PURE__ */ G(void 0)), V(d, Et(u)), n.set(a, d));
        else {
          f = d.v !== P;
          var y = o(() => Et(u));
          V(d, y);
        }
        var b = Reflect.getOwnPropertyDescriptor(l, a);
        if (b != null && b.set && b.set.call(c, u), !f) {
          if (i && typeof a == "string") {
            var g = (
              /** @type {Source<number>} */
              n.get("length")
            ), m = Number(a);
            Number.isInteger(m) && m >= g.v && V(g, m + 1);
          }
          ee(s);
        }
        return !0;
      },
      ownKeys(l) {
        I(s);
        var a = Reflect.ownKeys(l).filter((d) => {
          var f = n.get(d);
          return f === void 0 || f.v !== P;
        });
        for (var [u, c] of n)
          c.v !== P && !(u in l) && a.push(u);
        return a;
      },
      setPrototypeOf() {
        hi();
      }
    }
  );
}
function ke(e) {
  try {
    if (e !== null && typeof e == "object" && ft in e)
      return e[ft];
  } catch {
  }
  return e;
}
function Ri(e, t) {
  return Object.is(ke(e), ke(t));
}
var ki, Mi, Ui;
// @__NO_SIDE_EFFECTS__
function xi(e) {
  return (
    /** @type {TemplateNode | null} */
    Mi.call(e)
  );
}
// @__NO_SIDE_EFFECTS__
function Bi(e) {
  return (
    /** @type {TemplateNode | null} */
    Ui.call(e)
  );
}
function Fi(e, t) {
  if (t) {
    const n = document.body;
    e.autofocus = !0, Jt(() => {
      document.activeElement === n && e.focus();
    });
  }
}
function Ae(e) {
  var t = w, n = N;
  Q(null), st(null);
  try {
    return e();
  } finally {
    Q(t), st(n);
  }
}
function Hi(e) {
  N === null && (w === null && ui(), ai()), it && li();
}
function Vi(e, t) {
  var n = t.last;
  n === null ? t.last = t.first = e : (n.next = e, e.prev = n, t.last = e);
}
function vt(e, t, n) {
  var i = N;
  i !== null && (i.f & Nt) !== 0 && (e |= Nt);
  var s = {
    ctx: X,
    deps: null,
    nodes: null,
    f: e | W | z,
    first: null,
    fn: t,
    last: null,
    next: null,
    parent: i,
    b: i && i.b,
    prev: null,
    teardown: null,
    wv: 0,
    ac: null
  };
  if (n)
    try {
      Mt(s), s.f |= be;
    } catch (l) {
      throw gt(s), l;
    }
  else t !== null && Tt(s);
  var r = s;
  if (n && r.deps === null && r.teardown === null && r.nodes === null && r.first === r.last && // either `null`, or a singular child
  (r.f & Zt) === 0 && (r = r.first, (e & mt) !== 0 && (e & Pe) !== 0 && r !== null && (r.f |= Pe)), r !== null && (r.parent = i, i !== null && Vi(r, i), w !== null && (w.f & k) !== 0 && (e & Pt) === 0)) {
    var o = (
      /** @type {Derived} */
      w
    );
    (o.effects ?? (o.effects = [])).push(r);
  }
  return s;
}
function hn() {
  return w !== null && !q;
}
function pn(e) {
  const t = vt(ye, null, !1);
  return B(t, U), t.teardown = e, t;
}
function Wi(e) {
  Hi();
  var t = (
    /** @type {Effect} */
    N.f
  ), n = !w && (t & yt) !== 0 && (t & be) === 0;
  if (n) {
    var i = (
      /** @type {ComponentContext} */
      X
    );
    (i.e ?? (i.e = [])).push(e);
  } else
    return _n(e);
}
function _n(e) {
  return vt(Gt | ii, e, !1);
}
function Le(e) {
  return vt(Gt, e, !1);
}
function ji(e) {
  return vt(we | Zt, e, !0);
}
function Gi(e, t = 0) {
  return vt(ye | t, e, !0);
}
function Cn(e, t = 0) {
  var n = vt(ve | t, e, !0);
  return n;
}
function gn(e) {
  return vt(yt | Zt, e, !0);
}
function mn(e) {
  var t = e.teardown;
  if (t !== null) {
    const n = it, i = w;
    Me(!0), Q(null);
    try {
      t.call(null);
    } finally {
      Me(n), Q(i);
    }
  }
}
function yn(e, t = !1) {
  var n = e.first;
  for (e.first = e.last = null; n !== null; ) {
    const s = n.ac;
    s !== null && Ae(() => {
      s.abort(wt);
    });
    var i = n.next;
    (n.f & Pt) !== 0 ? n.parent = null : gt(n, t), n = i;
  }
}
function zi(e) {
  for (var t = e.first; t !== null; ) {
    var n = t.next;
    (t.f & yt) === 0 && gt(t), t = n;
  }
}
function gt(e, t = !0) {
  var n = !1;
  (t || (e.f & Je) !== 0) && e.nodes !== null && e.nodes.end !== null && (Ki(
    e.nodes.start,
    /** @type {TemplateNode} */
    e.nodes.end
  ), n = !0), yn(e, t && !n), Xt(e, 0), B(e, J);
  var i = e.nodes && e.nodes.t;
  if (i !== null)
    for (const r of i)
      r.stop();
  mn(e);
  var s = e.parent;
  s !== null && s.first !== null && vn(e), e.next = e.prev = e.teardown = e.ctx = e.deps = e.fn = e.nodes = e.ac = null;
}
function Ki(e, t) {
  for (; e !== null; ) {
    var n = e === t ? null : /* @__PURE__ */ Bi(e);
    e.remove(), e = n;
  }
}
function vn(e) {
  var t = e.parent, n = e.prev, i = e.next;
  n !== null && (n.next = i), i !== null && (i.prev = n), t !== null && (t.first === e && (t.first = i), t.last === e && (t.last = n));
}
let ht = !1;
function $t(e) {
  ht = e;
}
let it = !1;
function Me(e) {
  it = e;
}
let w = null, q = !1;
function Q(e) {
  w = e;
}
let N = null;
function st(e) {
  N = e;
}
let x = null;
function Yi(e) {
  w !== null && (x === null ? x = [e] : x.push(e));
}
let F = null, H = 0, j = null;
function qi(e) {
  j = e;
}
let bn = 1, ut = 0, pt = ut;
function Ue(e) {
  pt = e;
}
function wn() {
  return ++bn;
}
function xt(e) {
  var t = e.f;
  if ((t & W) !== 0)
    return !0;
  if (t & k && (e.f &= ~Ct), (t & rt) !== 0) {
    for (var n = (
      /** @type {Value[]} */
      e.deps
    ), i = n.length, s = 0; s < i; s++) {
      var r = n[s];
      if (xt(
        /** @type {Derived} */
        r
      ) && un(
        /** @type {Derived} */
        r
      ), r.wv > e.wv)
        return !0;
    }
    (t & z) !== 0 && // During time traveling we don't want to reset the status so that
    // traversal of the graph in the other batches still happens
    R === null && B(e, U);
  }
  return !1;
}
function In(e, t, n = !0) {
  var i = e.reactions;
  if (i !== null && !(x != null && x.includes(e)))
    for (var s = 0; s < i.length; s++) {
      var r = i[s];
      (r.f & k) !== 0 ? In(
        /** @type {Derived} */
        r,
        t,
        !1
      ) : t === r && (n ? B(r, W) : (r.f & U) !== 0 && B(r, rt), Tt(
        /** @type {Effect} */
        r
      ));
    }
}
function En(e) {
  var C;
  var t = F, n = H, i = j, s = w, r = x, o = X, l = q, a = pt, u = e.f;
  F = /** @type {null | Value[]} */
  null, H = 0, j = null, w = (u & (yt | Pt)) === 0 ? e : null, x = null, zt(e.ctx), q = !1, pt = ++ut, e.ac !== null && (Ae(() => {
    e.ac.abort(wt);
  }), e.ac = null);
  try {
    e.f |= de;
    var c = (
      /** @type {Function} */
      e.fn
    ), d = c(), f = e.deps;
    if (F !== null) {
      var p;
      if (Xt(e, H), f !== null && H > 0)
        for (f.length = H + F.length, p = 0; p < F.length; p++)
          f[H + p] = F[p];
      else
        e.deps = f = F;
      if (hn() && (e.f & z) !== 0)
        for (p = H; p < f.length; p++)
          ((C = f[p]).reactions ?? (C.reactions = [])).push(e);
    } else f !== null && H < f.length && (Xt(e, H), f.length = H);
    if (en() && j !== null && !q && f !== null && (e.f & (k | rt | W)) === 0)
      for (p = 0; p < /** @type {Source[]} */
      j.length; p++)
        In(
          j[p],
          /** @type {Effect} */
          e
        );
    if (s !== null && s !== e) {
      if (ut++, s.deps !== null)
        for (let y = 0; y < n; y += 1)
          s.deps[y].rv = ut;
      if (t !== null)
        for (const y of t)
          y.rv = ut;
      j !== null && (i === null ? i = j : i.push(.../** @type {Source[]} */
      j));
    }
    return (e.f & et) !== 0 && (e.f ^= et), d;
  } catch (y) {
    return wi(y);
  } finally {
    e.f ^= de, F = t, H = n, j = i, w = s, x = r, zt(o), q = l, pt = a;
  }
}
function $i(e, t) {
  let n = t.reactions;
  if (n !== null) {
    var i = Qn.call(n, e);
    if (i !== -1) {
      var s = n.length - 1;
      s === 0 ? n = t.reactions = null : (n[i] = n[s], n.pop());
    }
  }
  if (n === null && (t.f & k) !== 0 && // Destroying a child effect while updating a parent effect can cause a dependency to appear
  // to be unused, when in fact it is used by the currently-updating parent. Checking `new_deps`
  // allows us to skip the expensive work of disconnecting and immediately reconnecting it
  (F === null || !F.includes(t))) {
    var r = (
      /** @type {Derived} */
      t
    );
    (r.f & z) !== 0 && (r.f ^= z, r.f &= ~Ct), Ie(r), an(r), Xt(r, 0);
  }
}
function Xt(e, t) {
  var n = e.deps;
  if (n !== null)
    for (var i = t; i < n.length; i++)
      $i(e, n[i]);
}
function Mt(e) {
  var t = e.f;
  if ((t & J) === 0) {
    B(e, U);
    var n = N, i = ht;
    N = e, ht = !0;
    try {
      (t & (mt | ve)) !== 0 ? zi(e) : yn(e), mn(e);
      var s = En(e);
      e.teardown = typeof s == "function" ? s : null, e.wv = bn;
      var r;
      ce && mi && (e.f & W) !== 0 && e.deps;
    } finally {
      ht = i, N = n;
    }
  }
}
function I(e) {
  var t = e.f, n = (t & k) !== 0;
  if (w !== null && !q) {
    var i = N !== null && (N.f & J) !== 0;
    if (!i && !(x != null && x.includes(e))) {
      var s = w.deps;
      if ((w.f & de) !== 0)
        e.rv < ut && (e.rv = ut, F === null && s !== null && s[H] === e ? H++ : F === null ? F = [e] : F.push(e));
      else {
        (w.deps ?? (w.deps = [])).push(e);
        var r = e.reactions;
        r === null ? e.reactions = [w] : r.includes(w) || r.push(w);
      }
    }
  }
  if (it && nt.has(e))
    return nt.get(e);
  if (n) {
    var o = (
      /** @type {Derived} */
      e
    );
    if (it) {
      var l = o.v;
      return ((o.f & U) === 0 && o.reactions !== null || Dn(o)) && (l = De(o)), nt.set(o, l), l;
    }
    var a = (o.f & z) === 0 && !q && w !== null && (ht || (w.f & z) !== 0), u = o.deps === null;
    xt(o) && (a && (o.f |= z), un(o)), a && !u && Nn(o);
  }
  if (R != null && R.has(e))
    return R.get(e);
  if ((e.f & et) !== 0)
    throw e.v;
  return e.v;
}
function Nn(e) {
  if (e.deps !== null) {
    e.f |= z;
    for (const t of e.deps)
      (t.reactions ?? (t.reactions = [])).push(e), (t.f & k) !== 0 && (t.f & z) === 0 && Nn(
        /** @type {Derived} */
        t
      );
  }
}
function Dn(e) {
  if (e.v === P) return !0;
  if (e.deps === null) return !1;
  for (const t of e.deps)
    if (nt.has(t) || (t.f & k) !== 0 && Dn(
      /** @type {Derived} */
      t
    ))
      return !0;
  return !1;
}
function An(e) {
  var t = q;
  try {
    return q = !0, e();
  } finally {
    q = t;
  }
}
const Xi = /* @__PURE__ */ new Set(), Qi = /* @__PURE__ */ new Set();
function Zi(e, t, n, i = {}) {
  function s(r) {
    if (i.capture || ts.call(t, r), !r.cancelBubble)
      return Ae(() => n == null ? void 0 : n.call(this, r));
  }
  return e.startsWith("pointer") || e.startsWith("touch") || e === "wheel" ? Jt(() => {
    t.addEventListener(e, s, i);
  }) : t.addEventListener(e, s, i), s;
}
function Ji(e) {
  for (var t = 0; t < e.length; t++)
    Xi.add(e[t]);
  for (var n of Qi)
    n(e);
}
let xe = null;
function ts(e) {
  var b;
  var t = this, n = (
    /** @type {Node} */
    t.ownerDocument
  ), i = e.type, s = ((b = e.composedPath) == null ? void 0 : b.call(e)) || [], r = (
    /** @type {null | Element} */
    s[0] || e.target
  );
  xe = e;
  var o = 0, l = xe === e && e.__root;
  if (l) {
    var a = s.indexOf(l);
    if (a !== -1 && (t === document || t === /** @type {any} */
    window)) {
      e.__root = t;
      return;
    }
    var u = s.indexOf(t);
    if (u === -1)
      return;
    a <= u && (o = a);
  }
  if (r = /** @type {Element} */
  s[o] || e.target, r !== t) {
    Zn(e, "currentTarget", {
      configurable: !0,
      get() {
        return r || n;
      }
    });
    var c = w, d = N;
    Q(null), st(null);
    try {
      for (var f, p = []; r !== null; ) {
        var C = r.assignedSlot || r.parentNode || /** @type {any} */
        r.host || null;
        try {
          var y = r["__" + i];
          y != null && (!/** @type {any} */
          r.disabled || // DOM could've been updated already by the time this is reached, so we check this as well
          // -> the target could not have been disabled because it emits the event in the first place
          e.target === r) && y.call(r, e);
        } catch (g) {
          f ? p.push(g) : f = g;
        }
        if (e.cancelBubble || C === t || C === null)
          break;
        r = C;
      }
      if (f) {
        for (let g of p)
          queueMicrotask(() => {
            throw g;
          });
        throw f;
      }
    } finally {
      e.__root = t, delete e.currentTarget, Q(c), st(d);
    }
  }
}
function es(e) {
  var t = document.createElement("template");
  return t.innerHTML = e.replaceAll("<!>", "<!---->"), t.content;
}
function ns(e, t) {
  var n = (
    /** @type {Effect} */
    N
  );
  n.nodes === null && (n.nodes = { start: e, end: t, a: null, t: null });
}
// @__NO_SIDE_EFFECTS__
function is(e, t) {
  var n = (t & qn) !== 0, i, s = !e.startsWith("<!>");
  return () => {
    i === void 0 && (i = es(s ? e : "<!>" + e), i = /** @type {TemplateNode} */
    /* @__PURE__ */ xi(i));
    var r = (
      /** @type {TemplateNode} */
      n || ki ? document.importNode(i, !0) : i.cloneNode(!0)
    );
    return ns(r, r), r;
  };
}
function ss(e, t) {
  e !== null && e.before(
    /** @type {Node} */
    t
  );
}
function rs(e) {
  return e.endsWith("capture") && e !== "gotpointercapture" && e !== "lostpointercapture";
}
const os = [
  "beforeinput",
  "click",
  "change",
  "dblclick",
  "contextmenu",
  "focusin",
  "focusout",
  "input",
  "keydown",
  "keyup",
  "mousedown",
  "mousemove",
  "mouseout",
  "mouseover",
  "mouseup",
  "pointerdown",
  "pointermove",
  "pointerout",
  "pointerover",
  "pointerup",
  "touchend",
  "touchmove",
  "touchstart"
];
function ls(e) {
  return os.includes(e);
}
const as = {
  // no `class: 'className'` because we handle that separately
  formnovalidate: "formNoValidate",
  ismap: "isMap",
  nomodule: "noModule",
  playsinline: "playsInline",
  readonly: "readOnly",
  defaultvalue: "defaultValue",
  defaultchecked: "defaultChecked",
  srcobject: "srcObject",
  novalidate: "noValidate",
  allowfullscreen: "allowFullscreen",
  disablepictureinpicture: "disablePictureInPicture",
  disableremoteplayback: "disableRemotePlayback"
};
function us(e) {
  return e = e.toLowerCase(), as[e] ?? e;
}
function cs(e, t) {
  var n = void 0, i;
  Cn(() => {
    n !== (n = t()) && (i && (gt(i), i = null), n && (i = gn(() => {
      Le(() => (
        /** @type {(node: Element) => void} */
        n(e)
      ));
    })));
  });
}
function Ln(e) {
  var t, n, i = "";
  if (typeof e == "string" || typeof e == "number") i += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var s = e.length;
    for (t = 0; t < s; t++) e[t] && (n = Ln(e[t])) && (i && (i += " "), i += n);
  } else for (n in e) e[n] && (i && (i += " "), i += n);
  return i;
}
function ds() {
  for (var e, t, n = 0, i = "", s = arguments.length; n < s; n++) (e = arguments[n]) && (t = Ln(e)) && (i && (i += " "), i += t);
  return i;
}
function fs(e) {
  return typeof e == "object" ? ds(e) : e ?? "";
}
const Be = [...` 	
\r\f \v\uFEFF`];
function hs(e, t, n) {
  var i = e == null ? "" : "" + e;
  if (n) {
    for (var s in n)
      if (n[s])
        i = i ? i + " " + s : s;
      else if (i.length)
        for (var r = s.length, o = 0; (o = i.indexOf(s, o)) >= 0; ) {
          var l = o + r;
          (o === 0 || Be.includes(i[o - 1])) && (l === i.length || Be.includes(i[l])) ? i = (o === 0 ? "" : i.substring(0, o)) + i.substring(l + 1) : o = l;
        }
  }
  return i === "" ? null : i;
}
function Fe(e, t = !1) {
  var n = t ? " !important;" : ";", i = "";
  for (var s in e) {
    var r = e[s];
    r != null && r !== "" && (i += " " + s + ": " + r + n);
  }
  return i;
}
function ne(e) {
  return e[0] !== "-" || e[1] !== "-" ? e.toLowerCase() : e;
}
function ps(e, t) {
  if (t) {
    var n = "", i, s;
    if (Array.isArray(t) ? (i = t[0], s = t[1]) : i = t, e) {
      e = String(e).replaceAll(/\s*\/\*.*?\*\/\s*/g, "").trim();
      var r = !1, o = 0, l = !1, a = [];
      i && a.push(...Object.keys(i).map(ne)), s && a.push(...Object.keys(s).map(ne));
      var u = 0, c = -1;
      const y = e.length;
      for (var d = 0; d < y; d++) {
        var f = e[d];
        if (l ? f === "/" && e[d - 1] === "*" && (l = !1) : r ? r === f && (r = !1) : f === "/" && e[d + 1] === "*" ? l = !0 : f === '"' || f === "'" ? r = f : f === "(" ? o++ : f === ")" && o--, !l && r === !1 && o === 0) {
          if (f === ":" && c === -1)
            c = d;
          else if (f === ";" || d === y - 1) {
            if (c !== -1) {
              var p = ne(e.substring(u, c).trim());
              if (!a.includes(p)) {
                f !== ";" && d++;
                var C = e.substring(u, d).trim();
                n += " " + C + ";";
              }
            }
            u = d + 1, c = -1;
          }
        }
      }
    }
    return i && (n += Fe(i)), s && (n += Fe(s, !0)), n = n.trim(), n === "" ? null : n;
  }
  return e == null ? null : String(e);
}
function _s(e, t, n, i, s, r) {
  var o = e.__className;
  if (o !== n || o === void 0) {
    var l = hs(n, i, r);
    l == null ? e.removeAttribute("class") : t ? e.className = l : e.setAttribute("class", l), e.__className = n;
  } else if (r && s !== r)
    for (var a in r) {
      var u = !!r[a];
      (s == null || u !== !!s[a]) && e.classList.toggle(a, u);
    }
  return r;
}
function ie(e, t = {}, n, i) {
  for (var s in n) {
    var r = n[s];
    t[s] !== r && (n[s] == null ? e.style.removeProperty(s) : e.style.setProperty(s, r, i));
  }
}
function Cs(e, t, n, i) {
  var s = e.__style;
  if (s !== t) {
    var r = ps(t, i);
    r == null ? e.removeAttribute("style") : e.style.cssText = r, e.__style = t;
  } else i && (Array.isArray(i) ? (ie(e, n == null ? void 0 : n[0], i[0]), ie(e, n == null ? void 0 : n[1], i[1], "important")) : ie(e, n, i));
  return i;
}
function ge(e, t, n = !1) {
  if (e.multiple) {
    if (t == null)
      return;
    if (!$e(t))
      return _i();
    for (var i of e.options)
      i.selected = t.includes(He(i));
    return;
  }
  for (i of e.options) {
    var s = He(i);
    if (Ri(s, t)) {
      i.selected = !0;
      return;
    }
  }
  (!n || t !== void 0) && (e.selectedIndex = -1);
}
function gs(e) {
  var t = new MutationObserver(() => {
    ge(e, e.__value);
  });
  t.observe(e, {
    // Listen to option element changes
    childList: !0,
    subtree: !0,
    // because of <optgroup>
    // Listen to option element value attribute changes
    // (doesn't get notified of select value changes,
    // because that property is not reflected as an attribute)
    attributes: !0,
    attributeFilter: ["value"]
  }), pn(() => {
    t.disconnect();
  });
}
function He(e) {
  return "__value" in e ? e.__value : e.value;
}
const Ot = Symbol("class"), Rt = Symbol("style"), Sn = Symbol("is custom element"), Tn = Symbol("is html");
function ms(e, t) {
  t ? e.hasAttribute("selected") || e.setAttribute("selected", "") : e.removeAttribute("selected");
}
function Ve(e, t, n, i) {
  var s = Pn(e);
  s[t] !== (s[t] = n) && (t === "loading" && (e[ri] = n), n == null ? e.removeAttribute(t) : typeof n != "string" && On(e).includes(t) ? e[t] = n : e.setAttribute(t, n));
}
function ys(e, t, n, i, s = !1, r = !1) {
  var o = Pn(e), l = o[Sn], a = !o[Tn], u = t || {}, c = e.tagName === "OPTION";
  for (var d in t)
    d in n || (n[d] = null);
  n.class ? n.class = fs(n.class) : n[Ot] && (n.class = null), n[Rt] && (n.style ?? (n.style = null));
  var f = On(e);
  for (const h in n) {
    let v = n[h];
    if (c && h === "value" && v == null) {
      e.value = e.__value = "", u[h] = v;
      continue;
    }
    if (h === "class") {
      var p = e.namespaceURI === "http://www.w3.org/1999/xhtml";
      _s(e, p, v, i, t == null ? void 0 : t[Ot], n[Ot]), u[h] = v, u[Ot] = n[Ot];
      continue;
    }
    if (h === "style") {
      Cs(e, v, t == null ? void 0 : t[Rt], n[Rt]), u[h] = v, u[Rt] = n[Rt];
      continue;
    }
    var C = u[h];
    if (!(v === C && !(v === void 0 && e.hasAttribute(h)))) {
      u[h] = v;
      var y = h[0] + h[1];
      if (y !== "$$")
        if (y === "on") {
          const T = {}, ot = "$$" + h;
          let O = h.slice(2);
          var b = ls(O);
          if (rs(O) && (O = O.slice(0, -7), T.capture = !0), !b && C) {
            if (v != null) continue;
            e.removeEventListener(O, u[ot], T), u[ot] = null;
          }
          if (v != null)
            if (b)
              e[`__${O}`] = v, Ji([O]);
            else {
              let xn = function(Bn) {
                u[h].call(this, Bn);
              };
              u[ot] = Zi(O, e, xn, T);
            }
          else b && (e[`__${O}`] = void 0);
        } else if (h === "style")
          Ve(e, h, v);
        else if (h === "autofocus")
          Fi(
            /** @type {HTMLElement} */
            e,
            !!v
          );
        else if (!l && (h === "__value" || h === "value" && v != null))
          e.value = e.__value = v;
        else if (h === "selected" && c)
          ms(
            /** @type {HTMLOptionElement} */
            e,
            v
          );
        else {
          var g = h;
          a || (g = us(g));
          var m = g === "defaultValue" || g === "defaultChecked";
          if (v == null && !l && !m)
            if (o[h] = null, g === "value" || g === "checked") {
              let T = (
                /** @type {HTMLInputElement} */
                e
              );
              const ot = t === void 0;
              if (g === "value") {
                let O = T.defaultValue;
                T.removeAttribute(g), T.defaultValue = O, T.value = T.__value = ot ? O : null;
              } else {
                let O = T.defaultChecked;
                T.removeAttribute(g), T.defaultChecked = O, T.checked = ot ? O : !1;
              }
            } else
              e.removeAttribute(h);
          else m || f.includes(g) && (l || typeof v != "string") ? (e[g] = v, g in o && (o[g] = P)) : typeof v != "function" && Ve(e, g, v);
        }
    }
  }
  return u;
}
function vs(e, t, n = [], i = [], s = [], r, o = !1, l = !1) {
  Ai(s, n, i, (a) => {
    var u = void 0, c = {}, d = e.nodeName === "SELECT", f = !1;
    if (Cn(() => {
      var C = t(...a.map(I)), y = ys(
        e,
        u,
        C,
        r,
        o,
        l
      );
      f && d && "value" in C && ge(
        /** @type {HTMLSelectElement} */
        e,
        C.value
      );
      for (let g of Object.getOwnPropertySymbols(c))
        C[g] || gt(c[g]);
      for (let g of Object.getOwnPropertySymbols(C)) {
        var b = C[g];
        g.description === Xn && (!u || b !== u[g]) && (c[g] && gt(c[g]), c[g] = gn(() => cs(e, () => b))), y[g] = b;
      }
      u = y;
    }), d) {
      var p = (
        /** @type {HTMLSelectElement} */
        e
      );
      Le(() => {
        ge(
          p,
          /** @type {Record<string | symbol, any>} */
          u.value,
          !0
        ), gs(p);
      });
    }
    f = !0;
  });
}
function Pn(e) {
  return (
    /** @type {Record<string | symbol, unknown>} **/
    // @ts-expect-error
    e.__attributes ?? (e.__attributes = {
      [Sn]: e.nodeName.includes("-"),
      [Tn]: e.namespaceURI === $n
    })
  );
}
var We = /* @__PURE__ */ new Map();
function On(e) {
  var t = e.getAttribute("is") || e.nodeName, n = We.get(t);
  if (n) return n;
  We.set(t, n = []);
  for (var i, s = e, r = Element.prototype; r !== s; ) {
    i = Jn(s);
    for (var o in i)
      i[o].set && n.push(o);
    s = Xe(s);
  }
  return n;
}
function je(e, t) {
  return e === t || (e == null ? void 0 : e[ft]) === t;
}
function bs(e = {}, t, n, i) {
  return Le(() => {
    var s, r;
    return Gi(() => {
      s = r, r = [], An(() => {
        e !== n(...r) && (t(e, ...r), s && je(n(...s), e) && t(null, ...s));
      });
    }), () => {
      Jt(() => {
        r && je(n(...r), e) && t(null, ...r);
      });
    };
  }), e;
}
let Ft = !1;
function ws(e) {
  var t = Ft;
  try {
    return Ft = !1, [e(), Ft];
  } finally {
    Ft = t;
  }
}
function Ht(e, t, n, i) {
  var m;
  var s = (n & Kn) !== 0, r = (n & Yn) !== 0, o = (
    /** @type {V} */
    i
  ), l = !0, a = () => (l && (l = !1, o = r ? An(
    /** @type {() => V} */
    i
  ) : (
    /** @type {V} */
    i
  )), o), u;
  if (s) {
    var c = ft in e || si in e;
    u = ((m = Wt(e, t)) == null ? void 0 : m.set) ?? (c && t in e ? (h) => e[t] = h : void 0);
  }
  var d, f = !1;
  s ? [d, f] = ws(() => (
    /** @type {V} */
    e[t]
  )) : d = /** @type {V} */
  e[t], d === void 0 && i !== void 0 && (d = a(), u && (di(), u(d)));
  var p;
  if (p = () => {
    var h = (
      /** @type {V} */
      e[t]
    );
    return h === void 0 ? a() : (l = !0, h);
  }, (n & zn) === 0)
    return p;
  if (u) {
    var C = e.$$legacy;
    return (
      /** @type {() => V} */
      (function(h, v) {
        return arguments.length > 0 ? ((!v || C || f) && u(v ? p() : h), h) : p();
      })
    );
  }
  var y = !1, b = ((n & Gn) !== 0 ? Ne : Ti)(() => (y = !1, p()));
  s && I(b);
  var g = (
    /** @type {Effect} */
    N
  );
  return (
    /** @type {() => V} */
    (function(h, v) {
      if (arguments.length > 0) {
        const T = v ? I(b) : s ? Et(h) : h;
        return V(b, T), y = !0, o !== void 0 && (o = T), h;
      }
      return it && y || (g.f & J) !== 0 ? b.v : I(b);
    })
  );
}
const Is = [
  [
    "af",
    // Afghanistan
    "93",
    0,
    null,
    "0"
  ],
  [
    "ax",
    // Åland Islands
    "358",
    1,
    ["18", "4"],
    // (4 is a mobile range shared with FI)
    "0"
  ],
  [
    "al",
    // Albania
    "355",
    0,
    null,
    "0"
  ],
  [
    "dz",
    // Algeria
    "213",
    0,
    null,
    "0"
  ],
  [
    "as",
    // American Samoa
    "1",
    5,
    ["684"],
    "1"
  ],
  [
    "ad",
    // Andorra
    "376"
  ],
  [
    "ao",
    // Angola
    "244"
  ],
  [
    "ai",
    // Anguilla
    "1",
    6,
    ["264"],
    "1"
  ],
  [
    "ag",
    // Antigua and Barbuda
    "1",
    7,
    ["268"],
    "1"
  ],
  [
    "ar",
    // Argentina
    "54",
    0,
    null,
    "0"
  ],
  [
    "am",
    // Armenia
    "374",
    0,
    null,
    "0"
  ],
  [
    "aw",
    // Aruba
    "297"
  ],
  [
    "ac",
    // Ascension Island
    "247"
  ],
  [
    "au",
    // Australia
    "61",
    0,
    ["4"],
    // (mobile range shared with CX and CC)
    "0"
  ],
  [
    "at",
    // Austria
    "43",
    0,
    null,
    "0"
  ],
  [
    "az",
    // Azerbaijan
    "994",
    0,
    null,
    "0"
  ],
  [
    "bs",
    // Bahamas
    "1",
    8,
    ["242"],
    "1"
  ],
  [
    "bh",
    // Bahrain
    "973"
  ],
  [
    "bd",
    // Bangladesh
    "880",
    0,
    null,
    "0"
  ],
  [
    "bb",
    // Barbados
    "1",
    9,
    ["246"],
    "1"
  ],
  [
    "by",
    // Belarus
    "375",
    0,
    null,
    "8"
  ],
  [
    "be",
    // Belgium
    "32",
    0,
    null,
    "0"
  ],
  [
    "bz",
    // Belize
    "501"
  ],
  [
    "bj",
    // Benin
    "229"
  ],
  [
    "bm",
    // Bermuda
    "1",
    10,
    ["441"],
    "1"
  ],
  [
    "bt",
    // Bhutan
    "975"
  ],
  [
    "bo",
    // Bolivia
    "591",
    0,
    null,
    "0"
  ],
  [
    "ba",
    // Bosnia and Herzegovina
    "387",
    0,
    null,
    "0"
  ],
  [
    "bw",
    // Botswana
    "267"
  ],
  [
    "br",
    // Brazil
    "55",
    0,
    null,
    "0"
  ],
  [
    "io",
    // British Indian Ocean Territory
    "246"
  ],
  [
    "vg",
    // British Virgin Islands
    "1",
    11,
    ["284"],
    "1"
  ],
  [
    "bn",
    // Brunei
    "673"
  ],
  [
    "bg",
    // Bulgaria
    "359",
    0,
    null,
    "0"
  ],
  [
    "bf",
    // Burkina Faso
    "226"
  ],
  [
    "bi",
    // Burundi
    "257"
  ],
  [
    "kh",
    // Cambodia
    "855",
    0,
    null,
    "0"
  ],
  [
    "cm",
    // Cameroon
    "237"
  ],
  [
    "ca",
    // Canada
    "1",
    1,
    ["204", "226", "236", "249", "250", "257", "263", "289", "306", "343", "354", "365", "367", "368", "382", "403", "416", "418", "428", "431", "437", "438", "450", "468", "474", "506", "514", "519", "548", "579", "581", "584", "587", "604", "613", "639", "647", "672", "683", "705", "709", "742", "753", "778", "780", "782", "807", "819", "825", "867", "873", "879", "902", "905", "942"],
    "1"
  ],
  [
    "cv",
    // Cape Verde
    "238"
  ],
  [
    "bq",
    // Caribbean Netherlands
    "599",
    1,
    ["3", "4", "7"]
  ],
  [
    "ky",
    // Cayman Islands
    "1",
    12,
    ["345"],
    "1"
  ],
  [
    "cf",
    // Central African Republic
    "236"
  ],
  [
    "td",
    // Chad
    "235"
  ],
  [
    "cl",
    // Chile
    "56"
  ],
  [
    "cn",
    // China
    "86",
    0,
    null,
    "0"
  ],
  [
    "cx",
    // Christmas Island
    "61",
    2,
    ["4", "89164"],
    // (4 is a mobile range shared with AU and CC)
    "0"
  ],
  [
    "cc",
    // Cocos (Keeling) Islands
    "61",
    1,
    ["4", "89162"],
    // (4 is a mobile range shared with AU and CX)
    "0"
  ],
  [
    "co",
    // Colombia
    "57",
    0,
    null,
    "0"
  ],
  [
    "km",
    // Comoros
    "269"
  ],
  [
    "cg",
    // Congo (Brazzaville)
    "242"
  ],
  [
    "cd",
    // Congo (Kinshasa)
    "243",
    0,
    null,
    "0"
  ],
  [
    "ck",
    // Cook Islands
    "682"
  ],
  [
    "cr",
    // Costa Rica
    "506"
  ],
  [
    "ci",
    // Côte d'Ivoire
    "225"
  ],
  [
    "hr",
    // Croatia
    "385",
    0,
    null,
    "0"
  ],
  [
    "cu",
    // Cuba
    "53",
    0,
    null,
    "0"
  ],
  [
    "cw",
    // Curaçao
    "599",
    0
  ],
  [
    "cy",
    // Cyprus
    "357"
  ],
  [
    "cz",
    // Czech Republic
    "420"
  ],
  [
    "dk",
    // Denmark
    "45"
  ],
  [
    "dj",
    // Djibouti
    "253"
  ],
  [
    "dm",
    // Dominica
    "1",
    13,
    ["767"],
    "1"
  ],
  [
    "do",
    // Dominican Republic
    "1",
    2,
    ["809", "829", "849"],
    "1"
  ],
  [
    "ec",
    // Ecuador
    "593",
    0,
    null,
    "0"
  ],
  [
    "eg",
    // Egypt
    "20",
    0,
    null,
    "0"
  ],
  [
    "sv",
    // El Salvador
    "503"
  ],
  [
    "gq",
    // Equatorial Guinea
    "240"
  ],
  [
    "er",
    // Eritrea
    "291",
    0,
    null,
    "0"
  ],
  [
    "ee",
    // Estonia
    "372"
  ],
  [
    "sz",
    // Eswatini
    "268"
  ],
  [
    "et",
    // Ethiopia
    "251",
    0,
    null,
    "0"
  ],
  [
    "fk",
    // Falkland Islands (Malvinas)
    "500"
  ],
  [
    "fo",
    // Faroe Islands
    "298"
  ],
  [
    "fj",
    // Fiji
    "679"
  ],
  [
    "fi",
    // Finland
    "358",
    0,
    ["4"],
    // (mobile range shared with AX)
    "0"
  ],
  [
    "fr",
    // France
    "33",
    0,
    null,
    "0"
  ],
  [
    "gf",
    // French Guiana
    "594",
    0,
    null,
    "0"
  ],
  [
    "pf",
    // French Polynesia
    "689"
  ],
  [
    "ga",
    // Gabon
    "241"
  ],
  [
    "gm",
    // Gambia
    "220"
  ],
  [
    "ge",
    // Georgia
    "995",
    0,
    null,
    "0"
  ],
  [
    "de",
    // Germany
    "49",
    0,
    null,
    "0"
  ],
  [
    "gh",
    // Ghana
    "233",
    0,
    null,
    "0"
  ],
  [
    "gi",
    // Gibraltar
    "350"
  ],
  [
    "gr",
    // Greece
    "30"
  ],
  [
    "gl",
    // Greenland
    "299"
  ],
  [
    "gd",
    // Grenada
    "1",
    14,
    ["473"],
    "1"
  ],
  [
    "gp",
    // Guadeloupe
    "590",
    0,
    null,
    "0"
  ],
  [
    "gu",
    // Guam
    "1",
    15,
    ["671"],
    "1"
  ],
  [
    "gt",
    // Guatemala
    "502"
  ],
  [
    "gg",
    // Guernsey
    "44",
    1,
    ["1481", "7781", "7839", "7911"],
    "0"
  ],
  [
    "gn",
    // Guinea
    "224"
  ],
  [
    "gw",
    // Guinea-Bissau
    "245"
  ],
  [
    "gy",
    // Guyana
    "592"
  ],
  [
    "ht",
    // Haiti
    "509"
  ],
  [
    "hn",
    // Honduras
    "504"
  ],
  [
    "hk",
    // Hong Kong SAR China
    "852"
  ],
  [
    "hu",
    // Hungary
    "36",
    0,
    null,
    "06"
  ],
  [
    "is",
    // Iceland
    "354"
  ],
  [
    "in",
    // India
    "91",
    0,
    null,
    "0"
  ],
  [
    "id",
    // Indonesia
    "62",
    0,
    null,
    "0"
  ],
  [
    "ir",
    // Iran
    "98",
    0,
    null,
    "0"
  ],
  [
    "iq",
    // Iraq
    "964",
    0,
    null,
    "0"
  ],
  [
    "ie",
    // Ireland
    "353",
    0,
    null,
    "0"
  ],
  [
    "im",
    // Isle of Man
    "44",
    2,
    ["1624", "74576", "7524", "7624", "7924"],
    "0"
  ],
  [
    "il",
    // Israel
    "972",
    0,
    null,
    "0"
  ],
  [
    "it",
    // Italy
    "39",
    0,
    ["3"]
    // (mobile range shared with VA)
  ],
  [
    "jm",
    // Jamaica
    "1",
    4,
    ["658", "876"],
    "1"
  ],
  [
    "jp",
    // Japan
    "81",
    0,
    null,
    "0"
  ],
  [
    "je",
    // Jersey
    "44",
    3,
    ["1534", "7509", "7700", "7797", "7829", "7937"],
    "0"
  ],
  [
    "jo",
    // Jordan
    "962",
    0,
    null,
    "0"
  ],
  [
    "kz",
    // Kazakhstan
    "7",
    1,
    ["33", "7"],
    // (33 is shared with RU)
    "8"
  ],
  [
    "ke",
    // Kenya
    "254",
    0,
    null,
    "0"
  ],
  [
    "ki",
    // Kiribati
    "686",
    0,
    null,
    "0"
  ],
  [
    "xk",
    // Kosovo
    "383",
    0,
    null,
    "0"
  ],
  [
    "kw",
    // Kuwait
    "965"
  ],
  [
    "kg",
    // Kyrgyzstan
    "996",
    0,
    null,
    "0"
  ],
  [
    "la",
    // Laos
    "856",
    0,
    null,
    "0"
  ],
  [
    "lv",
    // Latvia
    "371"
  ],
  [
    "lb",
    // Lebanon
    "961",
    0,
    null,
    "0"
  ],
  [
    "ls",
    // Lesotho
    "266"
  ],
  [
    "lr",
    // Liberia
    "231",
    0,
    null,
    "0"
  ],
  [
    "ly",
    // Libya
    "218",
    0,
    null,
    "0"
  ],
  [
    "li",
    // Liechtenstein
    "423",
    0,
    null,
    "0"
  ],
  [
    "lt",
    // Lithuania
    "370",
    0,
    null,
    "0"
  ],
  [
    "lu",
    // Luxembourg
    "352"
  ],
  [
    "mo",
    // Macao SAR China
    "853"
  ],
  [
    "mg",
    // Madagascar
    "261",
    0,
    null,
    "0"
  ],
  [
    "mw",
    // Malawi
    "265",
    0,
    null,
    "0"
  ],
  [
    "my",
    // Malaysia
    "60",
    0,
    null,
    "0"
  ],
  [
    "mv",
    // Maldives
    "960"
  ],
  [
    "ml",
    // Mali
    "223"
  ],
  [
    "mt",
    // Malta
    "356"
  ],
  [
    "mh",
    // Marshall Islands
    "692",
    0,
    null,
    "1"
  ],
  [
    "mq",
    // Martinique
    "596",
    0,
    null,
    "0"
  ],
  [
    "mr",
    // Mauritania
    "222"
  ],
  [
    "mu",
    // Mauritius
    "230"
  ],
  [
    "yt",
    // Mayotte
    "262",
    1,
    ["269", "639"],
    "0"
  ],
  [
    "mx",
    // Mexico
    "52"
  ],
  [
    "fm",
    // Micronesia
    "691"
  ],
  [
    "md",
    // Moldova
    "373",
    0,
    null,
    "0"
  ],
  [
    "mc",
    // Monaco
    "377",
    0,
    null,
    "0"
  ],
  [
    "mn",
    // Mongolia
    "976",
    0,
    null,
    "0"
  ],
  [
    "me",
    // Montenegro
    "382",
    0,
    null,
    "0"
  ],
  [
    "ms",
    // Montserrat
    "1",
    16,
    ["664"],
    "1"
  ],
  [
    "ma",
    // Morocco
    "212",
    0,
    ["6", "7"],
    // (mobile ranges shared with EH)
    "0"
  ],
  [
    "mz",
    // Mozambique
    "258"
  ],
  [
    "mm",
    // Myanmar (Burma)
    "95",
    0,
    null,
    "0"
  ],
  [
    "na",
    // Namibia
    "264",
    0,
    null,
    "0"
  ],
  [
    "nr",
    // Nauru
    "674"
  ],
  [
    "np",
    // Nepal
    "977",
    0,
    null,
    "0"
  ],
  [
    "nl",
    // Netherlands
    "31",
    0,
    null,
    "0"
  ],
  [
    "nc",
    // New Caledonia
    "687"
  ],
  [
    "nz",
    // New Zealand
    "64",
    0,
    null,
    "0"
  ],
  [
    "ni",
    // Nicaragua
    "505"
  ],
  [
    "ne",
    // Niger
    "227"
  ],
  [
    "ng",
    // Nigeria
    "234",
    0,
    null,
    "0"
  ],
  [
    "nu",
    // Niue
    "683"
  ],
  [
    "nf",
    // Norfolk Island
    "672"
  ],
  [
    "kp",
    // North Korea
    "850",
    0,
    null,
    "0"
  ],
  [
    "mk",
    // North Macedonia
    "389",
    0,
    null,
    "0"
  ],
  [
    "mp",
    // Northern Mariana Islands
    "1",
    17,
    ["670"],
    "1"
  ],
  [
    "no",
    // Norway
    "47",
    0,
    ["4", "9"]
    // (mobile ranges shared with SJ)
  ],
  [
    "om",
    // Oman
    "968"
  ],
  [
    "pk",
    // Pakistan
    "92",
    0,
    null,
    "0"
  ],
  [
    "pw",
    // Palau
    "680"
  ],
  [
    "ps",
    // Palestinian Territories
    "970",
    0,
    null,
    "0"
  ],
  [
    "pa",
    // Panama
    "507"
  ],
  [
    "pg",
    // Papua New Guinea
    "675"
  ],
  [
    "py",
    // Paraguay
    "595",
    0,
    null,
    "0"
  ],
  [
    "pe",
    // Peru
    "51",
    0,
    null,
    "0"
  ],
  [
    "ph",
    // Philippines
    "63",
    0,
    null,
    "0"
  ],
  [
    "pl",
    // Poland
    "48"
  ],
  [
    "pt",
    // Portugal
    "351"
  ],
  [
    "pr",
    // Puerto Rico
    "1",
    3,
    ["787", "939"],
    "1"
  ],
  [
    "qa",
    // Qatar
    "974"
  ],
  [
    "re",
    // Réunion
    "262",
    0,
    null,
    "0"
  ],
  [
    "ro",
    // Romania
    "40",
    0,
    null,
    "0"
  ],
  [
    "ru",
    // Russia
    "7",
    0,
    ["33"],
    // (shared with KZ)
    "8"
  ],
  [
    "rw",
    // Rwanda
    "250",
    0,
    null,
    "0"
  ],
  [
    "ws",
    // Samoa
    "685"
  ],
  [
    "sm",
    // San Marino
    "378"
  ],
  [
    "st",
    // São Tomé & Príncipe
    "239"
  ],
  [
    "sa",
    // Saudi Arabia
    "966",
    0,
    null,
    "0"
  ],
  [
    "sn",
    // Senegal
    "221"
  ],
  [
    "rs",
    // Serbia
    "381",
    0,
    null,
    "0"
  ],
  [
    "sc",
    // Seychelles
    "248"
  ],
  [
    "sl",
    // Sierra Leone
    "232",
    0,
    null,
    "0"
  ],
  [
    "sg",
    // Singapore
    "65"
  ],
  [
    "sx",
    // Sint Maarten
    "1",
    21,
    ["721"],
    "1"
  ],
  [
    "sk",
    // Slovakia
    "421",
    0,
    null,
    "0"
  ],
  [
    "si",
    // Slovenia
    "386",
    0,
    null,
    "0"
  ],
  [
    "sb",
    // Solomon Islands
    "677"
  ],
  [
    "so",
    // Somalia
    "252",
    0,
    null,
    "0"
  ],
  [
    "za",
    // South Africa
    "27",
    0,
    null,
    "0"
  ],
  [
    "kr",
    // South Korea
    "82",
    0,
    null,
    "0"
  ],
  [
    "ss",
    // South Sudan
    "211",
    0,
    null,
    "0"
  ],
  [
    "es",
    // Spain
    "34"
  ],
  [
    "lk",
    // Sri Lanka
    "94",
    0,
    null,
    "0"
  ],
  [
    "bl",
    // St. Barthélemy
    "590",
    1,
    null,
    "0"
  ],
  [
    "sh",
    // St. Helena
    "290"
  ],
  [
    "kn",
    // St. Kitts & Nevis
    "1",
    18,
    ["869"],
    "1"
  ],
  [
    "lc",
    // St. Lucia
    "1",
    19,
    ["758"],
    "1"
  ],
  [
    "mf",
    // St. Martin
    "590",
    2,
    null,
    "0"
  ],
  [
    "pm",
    // St. Pierre & Miquelon
    "508",
    0,
    null,
    "0"
  ],
  [
    "vc",
    // St. Vincent & Grenadines
    "1",
    20,
    ["784"],
    "1"
  ],
  [
    "sd",
    // Sudan
    "249",
    0,
    null,
    "0"
  ],
  [
    "sr",
    // Suriname
    "597"
  ],
  [
    "sj",
    // Svalbard & Jan Mayen
    "47",
    1,
    ["4", "79", "9"]
    // (4 and 9 are mobile ranges shared with NO)
  ],
  [
    "se",
    // Sweden
    "46",
    0,
    null,
    "0"
  ],
  [
    "ch",
    // Switzerland
    "41",
    0,
    null,
    "0"
  ],
  [
    "sy",
    // Syria
    "963",
    0,
    null,
    "0"
  ],
  [
    "tw",
    // Taiwan
    "886",
    0,
    null,
    "0"
  ],
  [
    "tj",
    // Tajikistan
    "992"
  ],
  [
    "tz",
    // Tanzania
    "255",
    0,
    null,
    "0"
  ],
  [
    "th",
    // Thailand
    "66",
    0,
    null,
    "0"
  ],
  [
    "tl",
    // Timor-Leste
    "670"
  ],
  [
    "tg",
    // Togo
    "228"
  ],
  [
    "tk",
    // Tokelau
    "690"
  ],
  [
    "to",
    // Tonga
    "676"
  ],
  [
    "tt",
    // Trinidad & Tobago
    "1",
    22,
    ["868"],
    "1"
  ],
  [
    "tn",
    // Tunisia
    "216"
  ],
  [
    "tr",
    // Turkey
    "90",
    0,
    null,
    "0"
  ],
  [
    "tm",
    // Turkmenistan
    "993",
    0,
    null,
    "8"
  ],
  [
    "tc",
    // Turks & Caicos Islands
    "1",
    23,
    ["649"],
    "1"
  ],
  [
    "tv",
    // Tuvalu
    "688"
  ],
  [
    "vi",
    // U.S. Virgin Islands
    "1",
    24,
    ["340"],
    "1"
  ],
  [
    "ug",
    // Uganda
    "256",
    0,
    null,
    "0"
  ],
  [
    "ua",
    // Ukraine
    "380",
    0,
    null,
    "0"
  ],
  [
    "ae",
    // United Arab Emirates
    "971",
    0,
    null,
    "0"
  ],
  [
    "gb",
    // United Kingdom
    "44",
    0,
    null,
    "0"
  ],
  [
    "us",
    // United States
    "1",
    0,
    null,
    "1"
  ],
  [
    "uy",
    // Uruguay
    "598",
    0,
    null,
    "0"
  ],
  [
    "uz",
    // Uzbekistan
    "998"
  ],
  [
    "vu",
    // Vanuatu
    "678"
  ],
  [
    "va",
    // Vatican City
    "39",
    1,
    ["06698", "3"]
    // (3 is a mobile range shared with IT)
  ],
  [
    "ve",
    // Venezuela
    "58",
    0,
    null,
    "0"
  ],
  [
    "vn",
    // Vietnam
    "84",
    0,
    null,
    "0"
  ],
  [
    "wf",
    // Wallis & Futuna
    "681"
  ],
  [
    "eh",
    // Western Sahara
    "212",
    1,
    ["5288", "5289", "6", "7"],
    // (6 and 7 are mobile ranges shared with MA)
    "0"
  ],
  [
    "ye",
    // Yemen
    "967",
    0,
    null,
    "0"
  ],
  [
    "zm",
    // Zambia
    "260",
    0,
    null,
    "0"
  ],
  [
    "zw",
    // Zimbabwe
    "263",
    0,
    null,
    "0"
  ]
], _t = [];
for (const e of Is)
  _t.push({
    name: "",
    // populated in the plugin
    iso2: e[0],
    dialCode: e[1],
    priority: e[2] || 0,
    areaCodes: e[3] || null,
    nodeById: {},
    // populated by the plugin
    nationalPrefix: e[4] || null,
    normalisedName: "",
    // populated in the plugin
    initials: "",
    // populated in the plugin
    dialCodePlus: ""
    // populated in the plugin
  });
const Es = {
  ad: "Andorra",
  ae: "United Arab Emirates",
  af: "Afghanistan",
  ag: "Antigua & Barbuda",
  ai: "Anguilla",
  al: "Albania",
  am: "Armenia",
  ao: "Angola",
  ar: "Argentina",
  as: "American Samoa",
  at: "Austria",
  au: "Australia",
  aw: "Aruba",
  ax: "Åland Islands",
  az: "Azerbaijan",
  ba: "Bosnia & Herzegovina",
  bb: "Barbados",
  bd: "Bangladesh",
  be: "Belgium",
  bf: "Burkina Faso",
  bg: "Bulgaria",
  bh: "Bahrain",
  bi: "Burundi",
  bj: "Benin",
  bl: "St. Barthélemy",
  bm: "Bermuda",
  bn: "Brunei",
  bo: "Bolivia",
  bq: "Caribbean Netherlands",
  br: "Brazil",
  bs: "Bahamas",
  bt: "Bhutan",
  bw: "Botswana",
  by: "Belarus",
  bz: "Belize",
  ca: "Canada",
  cc: "Cocos (Keeling) Islands",
  cd: "Congo - Kinshasa",
  cf: "Central African Republic",
  cg: "Congo - Brazzaville",
  ch: "Switzerland",
  ci: "Côte d’Ivoire",
  ck: "Cook Islands",
  cl: "Chile",
  cm: "Cameroon",
  cn: "China",
  co: "Colombia",
  cr: "Costa Rica",
  cu: "Cuba",
  cv: "Cape Verde",
  cw: "Curaçao",
  cx: "Christmas Island",
  cy: "Cyprus",
  cz: "Czechia",
  de: "Germany",
  dj: "Djibouti",
  dk: "Denmark",
  dm: "Dominica",
  do: "Dominican Republic",
  dz: "Algeria",
  ec: "Ecuador",
  ee: "Estonia",
  eg: "Egypt",
  eh: "Western Sahara",
  er: "Eritrea",
  es: "Spain",
  et: "Ethiopia",
  fi: "Finland",
  fj: "Fiji",
  fk: "Falkland Islands",
  fm: "Micronesia",
  fo: "Faroe Islands",
  fr: "France",
  ga: "Gabon",
  gb: "United Kingdom",
  gd: "Grenada",
  ge: "Georgia",
  gf: "French Guiana",
  gg: "Guernsey",
  gh: "Ghana",
  gi: "Gibraltar",
  gl: "Greenland",
  gm: "Gambia",
  gn: "Guinea",
  gp: "Guadeloupe",
  gq: "Equatorial Guinea",
  gr: "Greece",
  gt: "Guatemala",
  gu: "Guam",
  gw: "Guinea-Bissau",
  gy: "Guyana",
  hk: "Hong Kong SAR China",
  hn: "Honduras",
  hr: "Croatia",
  ht: "Haiti",
  hu: "Hungary",
  id: "Indonesia",
  ie: "Ireland",
  il: "Israel",
  im: "Isle of Man",
  in: "India",
  io: "British Indian Ocean Territory",
  iq: "Iraq",
  ir: "Iran",
  is: "Iceland",
  it: "Italy",
  je: "Jersey",
  jm: "Jamaica",
  jo: "Jordan",
  jp: "Japan",
  ke: "Kenya",
  kg: "Kyrgyzstan",
  kh: "Cambodia",
  ki: "Kiribati",
  km: "Comoros",
  kn: "St. Kitts & Nevis",
  kp: "North Korea",
  kr: "South Korea",
  kw: "Kuwait",
  ky: "Cayman Islands",
  kz: "Kazakhstan",
  la: "Laos",
  lb: "Lebanon",
  lc: "St. Lucia",
  li: "Liechtenstein",
  lk: "Sri Lanka",
  lr: "Liberia",
  ls: "Lesotho",
  lt: "Lithuania",
  lu: "Luxembourg",
  lv: "Latvia",
  ly: "Libya",
  ma: "Morocco",
  mc: "Monaco",
  md: "Moldova",
  me: "Montenegro",
  mf: "St. Martin",
  mg: "Madagascar",
  mh: "Marshall Islands",
  mk: "North Macedonia",
  ml: "Mali",
  mm: "Myanmar (Burma)",
  mn: "Mongolia",
  mo: "Macao SAR China",
  mp: "Northern Mariana Islands",
  mq: "Martinique",
  mr: "Mauritania",
  ms: "Montserrat",
  mt: "Malta",
  mu: "Mauritius",
  mv: "Maldives",
  mw: "Malawi",
  mx: "Mexico",
  my: "Malaysia",
  mz: "Mozambique",
  na: "Namibia",
  nc: "New Caledonia",
  ne: "Niger",
  nf: "Norfolk Island",
  ng: "Nigeria",
  ni: "Nicaragua",
  nl: "Netherlands",
  no: "Norway",
  np: "Nepal",
  nr: "Nauru",
  nu: "Niue",
  nz: "New Zealand",
  om: "Oman",
  pa: "Panama",
  pe: "Peru",
  pf: "French Polynesia",
  pg: "Papua New Guinea",
  ph: "Philippines",
  pk: "Pakistan",
  pl: "Poland",
  pm: "St. Pierre & Miquelon",
  pr: "Puerto Rico",
  ps: "Palestinian Territories",
  pt: "Portugal",
  pw: "Palau",
  py: "Paraguay",
  qa: "Qatar",
  re: "Réunion",
  ro: "Romania",
  rs: "Serbia",
  ru: "Russia",
  rw: "Rwanda",
  sa: "Saudi Arabia",
  sb: "Solomon Islands",
  sc: "Seychelles",
  sd: "Sudan",
  se: "Sweden",
  sg: "Singapore",
  sh: "St. Helena",
  si: "Slovenia",
  sj: "Svalbard & Jan Mayen",
  sk: "Slovakia",
  sl: "Sierra Leone",
  sm: "San Marino",
  sn: "Senegal",
  so: "Somalia",
  sr: "Suriname",
  ss: "South Sudan",
  st: "São Tomé & Príncipe",
  sv: "El Salvador",
  sx: "Sint Maarten",
  sy: "Syria",
  sz: "Eswatini",
  tc: "Turks & Caicos Islands",
  td: "Chad",
  tg: "Togo",
  th: "Thailand",
  tj: "Tajikistan",
  tk: "Tokelau",
  tl: "Timor-Leste",
  tm: "Turkmenistan",
  tn: "Tunisia",
  to: "Tonga",
  tr: "Turkey",
  tt: "Trinidad & Tobago",
  tv: "Tuvalu",
  tw: "Taiwan",
  tz: "Tanzania",
  ua: "Ukraine",
  ug: "Uganda",
  us: "United States",
  uy: "Uruguay",
  uz: "Uzbekistan",
  va: "Vatican City",
  vc: "St. Vincent & Grenadines",
  ve: "Venezuela",
  vg: "British Virgin Islands",
  vi: "U.S. Virgin Islands",
  vn: "Vietnam",
  vu: "Vanuatu",
  wf: "Wallis & Futuna",
  ws: "Samoa",
  ye: "Yemen",
  yt: "Mayotte",
  za: "South Africa",
  zm: "Zambia",
  zw: "Zimbabwe"
}, Ns = {
  selectedCountryAriaLabel: "Change country, selected ${countryName} (${dialCode})",
  noCountrySelected: "Select country",
  countryListAriaLabel: "List of countries",
  searchPlaceholder: "Search",
  clearSearchAriaLabel: "Clear search",
  zeroSearchResults: "No results found",
  oneSearchResult: "1 result found",
  multipleSearchResults: "${count} results found",
  // additional countries (not supported by country-list library)
  ac: "Ascension Island",
  xk: "Kosovo"
}, Rn = { ...Es, ...Ns }, Vt = {
  OPEN_COUNTRY_DROPDOWN: "open:countrydropdown",
  CLOSE_COUNTRY_DROPDOWN: "close:countrydropdown",
  COUNTRY_CHANGE: "countrychange",
  INPUT: "input"
  // used for synthetic input trigger
}, E = {
  HIDE: "iti__hide",
  V_HIDE: "iti__v-hide",
  ARROW_UP: "iti__arrow--up",
  GLOBE: "iti__globe",
  FLAG: "iti__flag",
  COUNTRY_ITEM: "iti__country",
  HIGHLIGHT: "iti__highlight"
}, M = {
  ARROW_UP: "ArrowUp",
  ARROW_DOWN: "ArrowDown",
  SPACE: " ",
  ENTER: "Enter",
  ESC: "Escape",
  TAB: "Tab"
}, Ge = {
  PASTE: "insertFromPaste",
  DELETE_FWD: "deleteContentForward"
}, at = {
  ALPHA_UNICODE: new RegExp("\\p{L}", "u"),
  // any kind of letter from any language
  NON_PLUS_NUMERIC: /[^+0-9]/,
  // chars that are NOT + or digit
  NON_PLUS_NUMERIC_GLOBAL: /[^+0-9]/g,
  // chars that are NOT + or digit (global)
  HIDDEN_SEARCH_CHAR: /^[a-zA-ZÀ-ÿа-яА-Я ]$/
  // single acceptable hidden-search char
}, Ds = {
  HIDDEN_SEARCH_RESET_MS: 1e3
}, ze = {
  UNKNOWN_NUMBER_TYPE: -99,
  UNKNOWN_VALIDATION_ERROR: -99
}, se = {
  SANE_SELECTED_WITH_DIAL_WIDTH: 78,
  // px width fallback when separateDialCode enabled
  SANE_SELECTED_NO_DIAL_WIDTH: 42,
  // px width fallback when no separate dial code
  INPUT_PADDING_EXTRA_LEFT: 6
  // px gap between selected country container and input text
}, kn = {
  NANP: "1"
  // North American Numbering Plan
}, re = {
  DIAL_CODE: "44",
  // +44 United Kingdom
  MOBILE_PREFIX: "7",
  // UK mobile numbers start with 7 after national trunk (0) or core section
  MOBILE_CORE_LENGTH: 10
  // core number length (excluding dial code / national prefix) for mobiles
}, As = {
  ISO2: "us"
}, me = {
  AGGRESSIVE: "aggressive",
  POLITE: "polite"
}, oe = {
  AUTO: "auto"
}, Ke = {
  COUNTRY_CODE: "countryCode",
  DIAL_CODE: "dialCode"
}, D = {
  EXPANDED: "aria-expanded",
  LABEL: "aria-label",
  SELECTED: "aria-selected",
  ACTIVE_DESCENDANT: "aria-activedescendant",
  HASPOPUP: "aria-haspopup",
  CONTROLS: "aria-controls",
  HIDDEN: "aria-hidden",
  AUTOCOMPLETE: "aria-autocomplete",
  MODAL: "aria-modal"
}, le = (e) => typeof window < "u" && typeof window.matchMedia == "function" && window.matchMedia(e).matches, Ls = () => {
  if (typeof navigator < "u" && typeof window < "u") {
    const e = /Android.+Mobile|webOS|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
      navigator.userAgent
    ), t = le("(max-width: 500px)"), n = le("(max-height: 600px)"), i = le("(pointer: coarse)");
    return e || t || i && n;
  }
  return !1;
}, Mn = {
  // Allow alphanumeric "phonewords" (e.g. +1 800 FLOWERS) as valid numbers
  allowPhonewords: !1,
  //* Whether or not to allow the dropdown.
  allowDropdown: !0,
  //* Add a placeholder in the input with an example number for the selected country.
  autoPlaceholder: me.POLITE,
  //* Modify the parentClass.
  containerClass: "",
  //* The order of the countries in the dropdown. Defaults to alphabetical.
  countryOrder: null,
  //* Add a country search input at the top of the dropdown.
  countrySearch: !0,
  //* Modify the auto placeholder.
  customPlaceholder: null,
  //* Append menu to specified element.
  dropdownContainer: null,
  //* Don't display these countries.
  excludeCountries: [],
  //* Fix the dropdown width to the input width (rather than being as wide as the longest country name).
  fixDropdownWidth: !0,
  //* Format the number as the user types
  formatAsYouType: !0,
  //* Format the input value during initialisation and on setNumber.
  formatOnDisplay: !0,
  //* geoIp lookup function.
  geoIpLookup: null,
  //* Inject a hidden input with the name returned from this function, and on submit, populate it with the result of getNumber.
  hiddenInput: null,
  //* Internationalise the plugin text e.g. search input placeholder, country names.
  i18n: {},
  //* Initial country.
  initialCountry: "",
  //* A function to load the utils script.
  loadUtils: null,
  //* National vs international formatting for numbers e.g. placeholders and displaying existing numbers.
  nationalMode: !0,
  //* Display only these countries.
  onlyCountries: [],
  //* Number type to use for placeholders.
  placeholderNumberType: "MOBILE",
  //* Show flags - for both the selected country, and in the country dropdown
  showFlags: !0,
  //* Display the international dial code next to the selected flag.
  separateDialCode: !1,
  //* Only allow certain chars e.g. a plus followed by numeric digits, and cap at max valid length.
  strictMode: !1,
  //* Use full screen popup instead of dropdown for country list.
  useFullscreenPopup: Ls(),
  //* The number type to enforce during validation.
  validationNumberTypes: ["MOBILE"]
}, Ss = (e, t) => {
  e.useFullscreenPopup && (e.fixDropdownWidth = !1), e.onlyCountries.length === 1 && (e.initialCountry = e.onlyCountries[0]), e.separateDialCode && (e.nationalMode = !1), e.allowDropdown && !e.showFlags && !e.separateDialCode && (e.nationalMode = !1), e.useFullscreenPopup && !e.dropdownContainer && (e.dropdownContainer = document.body), e.i18n = { ...t, ...e.i18n };
}, jt = (e) => e.replace(/\D/g, ""), Un = (e = "") => e.normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase(), Ts = (e, t) => {
  const n = Un(t), i = [], s = [], r = [], o = [], l = [], a = [];
  for (const c of e)
    c.iso2 === n ? i.push(c) : c.normalisedName.startsWith(n) ? s.push(c) : c.normalisedName.includes(n) ? r.push(c) : n === c.dialCode || n === c.dialCodePlus ? o.push(c) : c.dialCodePlus.includes(n) ? l.push(c) : c.initials.includes(n) && a.push(c);
  const u = (c, d) => c.priority - d.priority;
  return [
    ...i.sort(u),
    ...s.sort(u),
    ...r.sort(u),
    ...o.sort(u),
    ...l.sort(u),
    ...a.sort(u)
  ];
}, Ps = (e, t) => {
  const n = t.toLowerCase();
  for (const i of e)
    if (i.name.toLowerCase().startsWith(n))
      return i;
  return null;
}, ae = (e) => Object.keys(e).filter((t) => !!e[t]).join(" "), L = (e, t, n) => {
  const i = document.createElement(e);
  return t && Object.entries(t).forEach(
    ([s, r]) => i.setAttribute(s, r)
  ), n && n.appendChild(i), i;
}, Os = () => `
  <svg class="iti__search-icon-svg" width="14" height="14" viewBox="0 0 24 24" focusable="false" ${D.HIDDEN}="true">
    <circle cx="11" cy="11" r="7" />
    <line x1="21" y1="21" x2="16.65" y2="16.65" />
  </svg>`, Rs = (e) => {
  const t = `iti-${e}-clear-mask`;
  return `
    <svg class="iti__search-clear-svg" width="12" height="12" viewBox="0 0 16 16" ${D.HIDDEN}="true" focusable="false">
      <mask id="${t}" maskUnits="userSpaceOnUse">
        <rect width="16" height="16" fill="white" />
        <path d="M5.2 5.2 L10.8 10.8 M10.8 5.2 L5.2 10.8" stroke="black" stroke-linecap="round" class="iti__search-clear-x" />
      </mask>
      <circle cx="8" cy="8" r="8" class="iti__search-clear-bg" mask="url(#${t})" />
    </svg>`;
};
class ks {
  constructor(t, n, i) {
    this.highlightedItem = null, this.selectedItem = null, t.dataset.intlTelInputId = i.toString(), this.telInput = t, this.options = n, this.id = i, this.hadInitialPlaceholder = !!t.getAttribute("placeholder"), this.isRTL = !!this.telInput.closest("[dir=rtl]"), this.options.separateDialCode && (this.originalPaddingLeft = this.telInput.style.paddingLeft);
  }
  //* Generate all of the markup for the plugin: the selected country overlay, and the dropdown.
  generateMarkup(t) {
    this.countries = t, this._prepareTelInput();
    const n = this._createWrapperAndInsert();
    this._maybeBuildCountryContainer(n), n.appendChild(this.telInput), this._maybeUpdateInputPaddingAndReveal(), this._maybeBuildHiddenInputs(n);
  }
  _prepareTelInput() {
    var t;
    this.telInput.classList.add("iti__tel-input"), !this.telInput.hasAttribute("autocomplete") && !((t = this.telInput.form) != null && t.hasAttribute("autocomplete")) && this.telInput.setAttribute("autocomplete", "off");
  }
  _createWrapperAndInsert() {
    const { allowDropdown: t, showFlags: n, containerClass: i, useFullscreenPopup: s } = this.options, r = ae({
      iti: !0,
      "iti--allow-dropdown": t,
      "iti--show-flags": n,
      "iti--inline-dropdown": !s,
      [i]: !!i
    }), o = L("div", { class: r });
    return this.isRTL && o.setAttribute("dir", "ltr"), this.telInput.before(o), o;
  }
  _maybeBuildCountryContainer(t) {
    const { allowDropdown: n, separateDialCode: i, showFlags: s } = this.options;
    if (n || s || i) {
      this.countryContainer = L(
        "div",
        // visibly hidden until we measure it's width to set the input padding correctly
        { class: `iti__country-container ${E.V_HIDE}` },
        t
      ), n ? (this.selectedCountry = L(
        "button",
        {
          type: "button",
          class: "iti__selected-country",
          [D.EXPANDED]: "false",
          [D.LABEL]: this.options.i18n.noCountrySelected,
          [D.HASPOPUP]: "dialog",
          [D.CONTROLS]: `iti-${this.id}__dropdown-content`
        },
        this.countryContainer
      ), this.telInput.disabled && this.selectedCountry.setAttribute("disabled", "true")) : this.selectedCountry = L(
        "div",
        { class: "iti__selected-country" },
        this.countryContainer
      );
      const r = L(
        "div",
        { class: "iti__selected-country-primary" },
        this.selectedCountry
      );
      this.selectedCountryInner = L(
        "div",
        { class: E.FLAG },
        r
      ), n && (this.dropdownArrow = L(
        "div",
        { class: "iti__arrow", [D.HIDDEN]: "true" },
        r
      )), i && (this.selectedDialCode = L(
        "div",
        { class: "iti__selected-dial-code" },
        this.selectedCountry
      )), n && this._buildDropdownContent();
    }
  }
  _buildDropdownContent() {
    const {
      fixDropdownWidth: t,
      useFullscreenPopup: n,
      countrySearch: i,
      i18n: s,
      dropdownContainer: r,
      containerClass: o
    } = this.options, l = t ? "" : "iti--flexible-dropdown-width";
    if (this.dropdownContent = L("div", {
      id: `iti-${this.id}__dropdown-content`,
      class: `iti__dropdown-content ${E.HIDE} ${l}`,
      role: "dialog",
      [D.MODAL]: "true"
    }), this.isRTL && this.dropdownContent.setAttribute("dir", "rtl"), i && this._buildSearchUI(), this.countryList = L(
      "ul",
      {
        class: "iti__country-list",
        id: `iti-${this.id}__country-listbox`,
        role: "listbox",
        [D.LABEL]: s.countryListAriaLabel
      },
      this.dropdownContent
    ), this._appendListItems(), i && this.updateSearchResultsA11yText(), r) {
      const a = ae({
        iti: !0,
        "iti--container": !0,
        "iti--fullscreen-popup": n,
        "iti--inline-dropdown": !n,
        [o]: !!o
      });
      this.dropdown = L("div", { class: a }), this.dropdown.appendChild(this.dropdownContent);
    } else
      this.countryContainer.appendChild(this.dropdownContent);
  }
  _buildSearchUI() {
    const { i18n: t } = this.options, n = L(
      "div",
      { class: "iti__search-input-wrapper" },
      this.dropdownContent
    );
    this.searchIcon = L(
      "span",
      {
        class: "iti__search-icon",
        [D.HIDDEN]: "true"
      },
      n
    ), this.searchIcon.innerHTML = Os(), this.searchInput = L(
      "input",
      {
        id: `iti-${this.id}__search-input`,
        // Chrome says inputs need either a name or an id
        type: "search",
        class: "iti__search-input",
        placeholder: t.searchPlaceholder,
        // role=combobox + aria-autocomplete=list + aria-activedescendant allows maintaining focus on the search input while allowing users to navigate search results with up/down keyboard keys
        role: "combobox",
        [D.EXPANDED]: "true",
        [D.LABEL]: t.searchPlaceholder,
        [D.CONTROLS]: `iti-${this.id}__country-listbox`,
        [D.AUTOCOMPLETE]: "list",
        autocomplete: "off"
      },
      n
    ), this.searchClearButton = L(
      "button",
      {
        type: "button",
        class: `iti__search-clear ${E.HIDE}`,
        [D.LABEL]: t.clearSearchAriaLabel,
        tabindex: "-1"
      },
      n
    ), this.searchClearButton.innerHTML = Rs(this.id), this.searchResultsA11yText = L(
      "span",
      { class: "iti__a11y-text" },
      this.dropdownContent
    ), this.searchNoResults = L(
      "div",
      {
        class: `iti__no-results ${E.HIDE}`,
        [D.HIDDEN]: "true"
        // all a11y messaging happens in this.searchResultsA11yText
      },
      this.dropdownContent
    ), this.searchNoResults.textContent = t.zeroSearchResults;
  }
  _maybeUpdateInputPaddingAndReveal() {
    this.countryContainer && (this.updateInputPadding(), this.countryContainer.classList.remove(E.V_HIDE));
  }
  _maybeBuildHiddenInputs(t) {
    var i, s;
    const { hiddenInput: n } = this.options;
    if (n) {
      const r = this.telInput.getAttribute("name") || "", o = n(r);
      if (o.phone) {
        const l = (i = this.telInput.form) == null ? void 0 : i.querySelector(
          `input[name="${o.phone}"]`
        );
        l ? this.hiddenInput = l : (this.hiddenInput = L("input", {
          type: "hidden",
          name: o.phone
        }), t.appendChild(this.hiddenInput));
      }
      if (o.country) {
        const l = (s = this.telInput.form) == null ? void 0 : s.querySelector(
          `input[name="${o.country}"]`
        );
        l ? this.hiddenInputCountry = l : (this.hiddenInputCountry = L("input", {
          type: "hidden",
          name: o.country
        }), t.appendChild(this.hiddenInputCountry));
      }
    }
  }
  //* For each country: add a country list item <li> to the countryList <ul> container.
  _appendListItems() {
    const t = document.createDocumentFragment();
    for (let n = 0; n < this.countries.length; n++) {
      const i = this.countries[n], s = ae({
        [E.COUNTRY_ITEM]: !0
      }), r = L("li", {
        id: `iti-${this.id}__item-${i.iso2}`,
        class: s,
        tabindex: "-1",
        role: "option",
        [D.SELECTED]: "false"
      });
      r.dataset.dialCode = i.dialCode, r.dataset.countryCode = i.iso2, i.nodeById[this.id] = r, this.options.showFlags && L("div", { class: `${E.FLAG} iti__${i.iso2}` }, r);
      const o = L("span", { class: "iti__country-name" }, r);
      o.textContent = i.name;
      const l = L("span", { class: "iti__dial-code" }, r);
      this.isRTL && l.setAttribute("dir", "ltr"), l.textContent = `+${i.dialCode}`, t.appendChild(r);
    }
    this.countryList.appendChild(t);
  }
  //* Update the input padding to make space for the selected country/dial code.
  updateInputPadding() {
    if (this.selectedCountry) {
      const t = this.options.separateDialCode ? se.SANE_SELECTED_WITH_DIAL_WIDTH : se.SANE_SELECTED_NO_DIAL_WIDTH, i = (this.selectedCountry.offsetWidth || this._getHiddenSelectedCountryWidth() || t) + se.INPUT_PADDING_EXTRA_LEFT;
      this.telInput.style.paddingLeft = `${i}px`;
    }
  }
  //* When input is in a hidden container during init, we cannot calculate the selected country width.
  //* Fix: clone the markup, make it invisible, add it to the end of the DOM, and then measure it's width.
  //* To get the right styling to apply, all we need is a shallow clone of the container,
  //* and then to inject a deep clone of the selectedCountry element.
  _getHiddenSelectedCountryWidth() {
    if (this.telInput.parentNode) {
      let t;
      try {
        t = window.top.document.body;
      } catch {
        t = document.body;
      }
      const n = this.telInput.parentNode.cloneNode(
        !1
      );
      n.style.visibility = "hidden", t.appendChild(n);
      const i = this.countryContainer.cloneNode();
      n.appendChild(i);
      const s = this.selectedCountry.cloneNode(
        !0
      );
      i.appendChild(s);
      const r = s.offsetWidth;
      return t.removeChild(n), r;
    }
    return 0;
  }
  //* Update search results text (for a11y).
  updateSearchResultsA11yText() {
    const { i18n: t } = this.options, n = this.countryList.childElementCount;
    let i;
    n === 0 ? i = t.zeroSearchResults : t.searchResultsText ? i = t.searchResultsText(n) : n === 1 ? i = t.oneSearchResult : i = t.multipleSearchResults.replace(
      "${count}",
      n.toString()
    ), this.searchResultsA11yText.textContent = i;
  }
  //* Check if an element is visible within it's container, else scroll until it is.
  scrollTo(t) {
    const n = this.countryList, i = document.documentElement.scrollTop, s = n.offsetHeight, r = n.getBoundingClientRect().top + i, o = r + s, l = t.offsetHeight, a = t.getBoundingClientRect().top + i, u = a + l, c = a - r + n.scrollTop;
    if (a < r)
      n.scrollTop = c;
    else if (u > o) {
      const d = s - l;
      n.scrollTop = c - d;
    }
  }
  //* Remove highlighting from the previous list item and highlight the new one.
  highlightListItem(t, n) {
    const i = this.highlightedItem;
    if (i && i.classList.remove(E.HIGHLIGHT), this.highlightedItem = t, this.highlightedItem && (this.highlightedItem.classList.add(E.HIGHLIGHT), this.options.countrySearch)) {
      const s = this.highlightedItem.getAttribute("id") || "";
      this.searchInput.setAttribute(D.ACTIVE_DESCENDANT, s);
    }
    n && this.highlightedItem.focus();
  }
  updateSelectedItem(t) {
    if (this.selectedItem && this.selectedItem.dataset.countryCode !== t && (this.selectedItem.setAttribute(D.SELECTED, "false"), this.selectedItem = null), t && !this.selectedItem) {
      const n = this.countryList.querySelector(
        `[data-country-code="${t}"]`
      );
      n && (n.setAttribute(D.SELECTED, "true"), this.selectedItem = n);
    }
  }
  //* Country search: Filter the country list to the given array of countries.
  filterCountries(t) {
    this.countryList.innerHTML = "";
    let n = !0;
    for (const i of t) {
      const s = i.nodeById[this.id];
      s && (this.countryList.appendChild(s), n && (this.highlightListItem(s, !1), n = !1));
    }
    n ? (this.highlightListItem(null, !1), this.searchNoResults && this.searchNoResults.classList.remove(E.HIDE)) : this.searchNoResults && this.searchNoResults.classList.add(E.HIDE), this.countryList.scrollTop = 0, this.updateSearchResultsA11yText();
  }
  destroy() {
    this.telInput.iti = void 0, delete this.telInput.dataset.intlTelInputId, this.options.separateDialCode && (this.telInput.style.paddingLeft = this.originalPaddingLeft);
    const t = this.telInput.parentNode;
    t.before(this.telInput), t.remove(), this.telInput = null, this.countryContainer = null, this.selectedCountry = null, this.selectedCountryInner = null, this.selectedDialCode = null, this.dropdownArrow = null, this.dropdownContent = null, this.searchInput = null, this.searchIcon = null, this.searchClearButton = null, this.searchNoResults = null, this.searchResultsA11yText = null, this.countryList = null, this.dropdown = null, this.hiddenInput = null, this.hiddenInputCountry = null, this.highlightedItem = null, this.selectedItem = null;
    for (const n of this.countries)
      delete n.nodeById[this.id];
    this.countries = null;
  }
}
const Ms = (e) => {
  const { onlyCountries: t, excludeCountries: n } = e;
  if (t.length) {
    const i = t.map(
      (s) => s.toLowerCase()
    );
    return _t.filter(
      (s) => i.includes(s.iso2)
    );
  } else if (n.length) {
    const i = n.map(
      (s) => s.toLowerCase()
    );
    return _t.filter(
      (s) => !i.includes(s.iso2)
    );
  }
  return _t;
}, Us = (e, t) => {
  for (const n of e) {
    const i = n.iso2.toLowerCase();
    t.i18n[i] && (n.name = t.i18n[i]);
  }
}, xs = (e) => {
  const t = /* @__PURE__ */ new Set();
  let n = 0;
  const i = {}, s = (o, l) => {
    if (!o || !l)
      return;
    l.length > n && (n = l.length), i.hasOwnProperty(l) || (i[l] = []);
    const a = i[l];
    a.includes(o) || a.push(o);
  }, r = [...e].sort((o, l) => o.priority - l.priority);
  for (const o of r) {
    t.has(o.dialCode) || t.add(o.dialCode);
    for (let l = 1; l < o.dialCode.length; l++) {
      const a = o.dialCode.substring(0, l);
      s(o.iso2, a);
    }
    if (s(o.iso2, o.dialCode), o.areaCodes) {
      const l = i[o.dialCode][0];
      for (const a of o.areaCodes) {
        for (let u = 1; u < a.length; u++) {
          const c = a.substring(0, u), d = o.dialCode + c;
          s(l, d), s(o.iso2, d);
        }
        s(o.iso2, o.dialCode + a);
      }
    }
  }
  return { dialCodes: t, dialCodeMaxLen: n, dialCodeToIso2Map: i };
}, Bs = (e, t) => {
  t.countryOrder && (t.countryOrder = t.countryOrder.map(
    (n) => n.toLowerCase()
  )), e.sort((n, i) => {
    const { countryOrder: s } = t;
    if (s) {
      const r = s.indexOf(n.iso2), o = s.indexOf(i.iso2), l = r > -1, a = o > -1;
      if (l || a)
        return l && a ? r - o : l ? -1 : 1;
    }
    return n.name.localeCompare(i.name);
  });
}, Fs = (e) => {
  for (const t of e)
    t.normalisedName = Un(t.name), t.initials = t.normalisedName.split(/[^a-z]/).map((n) => n[0]).join(""), t.dialCodePlus = `+${t.dialCode}`;
}, Hs = (e, t, n, i) => {
  let s = e;
  if (n && t) {
    t = `+${i.dialCode}`;
    const r = s[t.length] === " " || s[t.length] === "-" ? t.length + 1 : t.length;
    s = s.substring(r);
  }
  return s;
}, Vs = (e, t, n, i, s) => {
  const r = n ? n.formatNumberAsYouType(e, i.iso2) : e, { dialCode: o } = i;
  return s && t.charAt(0) !== "+" && r.includes(`+${o}`) ? (r.split(`+${o}`)[1] || "").trim() : r;
}, Ws = (e, t, n, i) => {
  if (n === 0 && !i)
    return 0;
  let s = 0;
  for (let r = 0; r < t.length; r++) {
    if (/[+0-9]/.test(t[r]) && s++, s === e && !i)
      return r + 1;
    if (i && s === e + 1)
      return r;
  }
  return t.length;
}, js = [
  "800",
  "822",
  "833",
  "844",
  "855",
  "866",
  "877",
  "880",
  "881",
  "882",
  "883",
  "884",
  "885",
  "886",
  "887",
  "888",
  "889"
], Ye = (e) => {
  const t = jt(e);
  if (t.startsWith(kn.NANP) && t.length >= 4) {
    const n = t.substring(1, 4);
    return js.includes(n);
  }
  return !1;
};
for (const e of _t)
  e.name = Rn[e.iso2];
let Gs = 0;
const zs = new Set(_t.map((e) => e.iso2)), ue = (e) => zs.has(e);
class Se {
  constructor(t, n = {}) {
    this.id = Gs++, this.options = { ...Mn, ...n }, Ss(this.options, Rn), this.ui = new ks(t, this.options, this.id), this.isAndroid = Se._getIsAndroid(), this.promise = this._createInitPromises(), this.countries = Ms(this.options);
    const { dialCodes: i, dialCodeMaxLen: s, dialCodeToIso2Map: r } = xs(
      this.countries
    );
    this.dialCodes = i, this.dialCodeMaxLen = s, this.dialCodeToIso2Map = r, this.countryByIso2 = new Map(this.countries.map((o) => [o.iso2, o])), this._init();
  }
  static _getIsAndroid() {
    return typeof navigator < "u" ? /Android/i.test(navigator.userAgent) : !1;
  }
  _updateNumeralSet(t) {
    /[\u0660-\u0669]/.test(t) ? this.userNumeralSet = "arabic-indic" : /[\u06F0-\u06F9]/.test(t) ? this.userNumeralSet = "persian" : this.userNumeralSet = "ascii";
  }
  _mapAsciiToUserNumerals(t) {
    if (this.userNumeralSet || this._updateNumeralSet(this.ui.telInput.value), this.userNumeralSet === "ascii")
      return t;
    const n = this.userNumeralSet === "arabic-indic" ? 1632 : 1776;
    return t.replace(/[0-9]/g, (i) => String.fromCharCode(n + Number(i)));
  }
  // Normalize Eastern Arabic (U+0660-0669) and Persian/Extended Arabic-Indic (U+06F0-06F9) numerals to ASCII 0-9
  _normaliseNumerals(t) {
    if (!t)
      return "";
    if (this._updateNumeralSet(t), this.userNumeralSet === "ascii")
      return t;
    const n = this.userNumeralSet === "arabic-indic" ? 1632 : 1776, i = this.userNumeralSet === "arabic-indic" ? /[\u0660-\u0669]/g : /[\u06F0-\u06F9]/g;
    return t.replace(i, (s) => String.fromCharCode(48 + (s.charCodeAt(0) - n)));
  }
  _getTelInputValue() {
    const t = this.ui.telInput.value.trim();
    return this._normaliseNumerals(t);
  }
  _setTelInputValue(t) {
    this.ui.telInput.value = this._mapAsciiToUserNumerals(t);
  }
  _createInitPromises() {
    const t = new Promise((i, s) => {
      this.resolveAutoCountryPromise = i, this.rejectAutoCountryPromise = s;
    }), n = new Promise((i, s) => {
      this.resolveUtilsScriptPromise = i, this.rejectUtilsScriptPromise = s;
    });
    return Promise.all([t, n]);
  }
  //* Can't be private as it's called from intlTelInput convenience wrapper.
  _init() {
    this.selectedCountryData = {}, this.abortController = new AbortController(), this._processCountryData(), this.ui.generateMarkup(this.countries), this._setInitialState(), this._initListeners(), this._initRequests();
  }
  //********************
  //*  PRIVATE METHODS
  //********************
  //* Prepare all of the country data, including onlyCountries, excludeCountries, countryOrder options.
  _processCountryData() {
    Us(this.countries, this.options), Bs(this.countries, this.options), Fs(this.countries);
  }
  //* Set the initial state of the input value and the selected country by:
  //* 1. Extracting a dial code from the given number
  //* 2. Using explicit initialCountry
  _setInitialState(t = !1) {
    const n = this.ui.telInput.getAttribute("value"), i = this._normaliseNumerals(n), s = this._getTelInputValue(), o = i && i.startsWith("+") && (!s || !s.startsWith("+")) ? i : s, l = this._getDialCode(o), a = Ye(o), { initialCountry: u, geoIpLookup: c } = this.options, d = u === oe.AUTO && c;
    if (l && !a)
      this._updateCountryFromNumber(o);
    else if (!d || t) {
      const f = u ? u.toLowerCase() : "";
      ue(f) ? this._setCountry(f) : l && a ? this._setCountry(As.ISO2) : this._setCountry("");
    }
    o && this._updateValFromNumber(o);
  }
  //* Initialise the main event listeners: input keyup, and click selected country.
  _initListeners() {
    this._initTelInputListeners(), this.options.allowDropdown && this._initDropdownListeners(), (this.ui.hiddenInput || this.ui.hiddenInputCountry) && this.ui.telInput.form && this._initHiddenInputListener();
  }
  //* Update hidden input on form submit.
  _initHiddenInputListener() {
    var n;
    const t = () => {
      this.ui.hiddenInput && (this.ui.hiddenInput.value = this.getNumber()), this.ui.hiddenInputCountry && (this.ui.hiddenInputCountry.value = this.selectedCountryData.iso2 || "");
    };
    (n = this.ui.telInput.form) == null || n.addEventListener("submit", t, {
      signal: this.abortController.signal
    });
  }
  //* initialise the dropdown listeners.
  _initDropdownListeners() {
    const t = this.abortController.signal, n = (o) => {
      this.ui.dropdownContent.classList.contains(E.HIDE) ? this.ui.telInput.focus() : o.preventDefault();
    }, i = this.ui.telInput.closest("label");
    i && i.addEventListener("click", n, { signal: t });
    const s = () => {
      this.ui.dropdownContent.classList.contains(
        E.HIDE
      ) && !this.ui.telInput.disabled && !this.ui.telInput.readOnly && this._openDropdown();
    };
    this.ui.selectedCountry.addEventListener(
      "click",
      s,
      {
        signal: t
      }
    );
    const r = (o) => {
      this.ui.dropdownContent.classList.contains(
        E.HIDE
      ) && [M.ARROW_UP, M.ARROW_DOWN, M.SPACE, M.ENTER].includes(o.key) && (o.preventDefault(), o.stopPropagation(), this._openDropdown()), o.key === M.TAB && this._closeDropdown();
    };
    this.ui.countryContainer.addEventListener(
      "keydown",
      r,
      { signal: t }
    );
  }
  //* Init many requests: utils script / geo ip lookup.
  _initRequests() {
    const { loadUtils: t, initialCountry: n, geoIpLookup: i } = this.options;
    if (t && !_.utils) {
      const r = () => {
        var o;
        (o = _.attachUtils(t)) == null || o.catch(() => {
        });
      };
      if (_.documentReady())
        r();
      else {
        const o = () => {
          r();
        };
        window.addEventListener("load", o, {
          signal: this.abortController.signal
        });
      }
    } else
      this.resolveUtilsScriptPromise();
    n === oe.AUTO && i && !this.selectedCountryData.iso2 ? this._loadAutoCountry() : this.resolveAutoCountryPromise();
  }
  //* Perform the geo ip lookup.
  _loadAutoCountry() {
    _.autoCountry ? this.handleAutoCountry() : _.startedLoadingAutoCountry || (_.startedLoadingAutoCountry = !0, typeof this.options.geoIpLookup == "function" && this.options.geoIpLookup(
      (t = "") => {
        const n = t.toLowerCase();
        ue(n) ? (_.autoCountry = n, setTimeout(() => kt("handleAutoCountry"))) : (this._setInitialState(!0), kt("rejectAutoCountryPromise"));
      },
      () => {
        this._setInitialState(!0), kt("rejectAutoCountryPromise");
      }
    ));
  }
  _openDropdownWithPlus() {
    this._openDropdown(), this.ui.searchInput.value = "+", this._filterCountriesByQuery("");
  }
  //* Initialize the tel input listeners.
  _initTelInputListeners() {
    this._bindInputListener(), this._maybeBindKeydownListener(), this._maybeBindPasteListener();
  }
  _bindInputListener() {
    const {
      strictMode: t,
      formatAsYouType: n,
      separateDialCode: i,
      allowDropdown: s,
      countrySearch: r
    } = this.options;
    let o = !1;
    at.ALPHA_UNICODE.test(this._getTelInputValue()) && (o = !0);
    const l = (a) => {
      const u = this._getTelInputValue();
      if (this.isAndroid && (a == null ? void 0 : a.data) === "+" && i && s && r) {
        const C = this.ui.telInput.selectionStart || 0, y = u.substring(0, C - 1), b = u.substring(C);
        this._setTelInputValue(y + b), this._openDropdownWithPlus();
        return;
      }
      this._updateCountryFromNumber(u) && this._triggerCountryChange();
      const c = (a == null ? void 0 : a.data) && at.NON_PLUS_NUMERIC.test(a.data), d = (a == null ? void 0 : a.inputType) === Ge.PASTE && u;
      c || d && !t ? o = !0 : at.NON_PLUS_NUMERIC.test(u) || (o = !1);
      const f = (a == null ? void 0 : a.detail) && a.detail.isSetNumber, p = this.userNumeralSet === "ascii";
      if (n && !o && !f && p) {
        const C = this.ui.telInput.selectionStart || 0, b = u.substring(
          0,
          C
        ).replace(
          at.NON_PLUS_NUMERIC_GLOBAL,
          ""
        ).length, g = (a == null ? void 0 : a.inputType) === Ge.DELETE_FWD, m = this._getFullNumber(), h = Vs(
          m,
          u,
          _.utils,
          this.selectedCountryData,
          this.options.separateDialCode
        ), v = Ws(
          b,
          h,
          C,
          g
        );
        this._setTelInputValue(h), this.ui.telInput.setSelectionRange(v, v);
      }
    };
    this.ui.telInput.addEventListener(
      "input",
      l,
      {
        signal: this.abortController.signal
      }
    );
  }
  _maybeBindKeydownListener() {
    const { strictMode: t, separateDialCode: n, allowDropdown: i, countrySearch: s } = this.options;
    if (t || n) {
      const r = (o) => {
        if (o.key && o.key.length === 1 && !o.altKey && !o.ctrlKey && !o.metaKey) {
          if (n && i && s && o.key === "+") {
            o.preventDefault(), this._openDropdownWithPlus();
            return;
          }
          if (t) {
            const l = this._getTelInputValue(), u = !l.startsWith("+") && this.ui.telInput.selectionStart === 0 && o.key === "+", c = this._normaliseNumerals(o.key), d = /^[0-9]$/.test(c), f = n ? d : u || d, p = this.ui.telInput, C = p.selectionStart, y = p.selectionEnd, b = l.slice(0, C), g = l.slice(y), m = b + o.key + g, h = this._getFullNumber(m), v = _.utils.getCoreNumber(
              h,
              this.selectedCountryData.iso2
            ), T = this.maxCoreNumberLength && v.length > this.maxCoreNumberLength, O = this._getNewCountryFromNumber(h) !== null;
            (!f || T && !O && !u) && o.preventDefault();
          }
        }
      };
      this.ui.telInput.addEventListener("keydown", r, {
        signal: this.abortController.signal
      });
    }
  }
  _maybeBindPasteListener() {
    if (this.options.strictMode) {
      const t = (n) => {
        n.preventDefault();
        const i = this.ui.telInput, s = i.selectionStart, r = i.selectionEnd, o = this._getTelInputValue(), l = o.slice(0, s), a = o.slice(r), u = this.selectedCountryData.iso2, c = n.clipboardData.getData("text"), d = this._normaliseNumerals(c), f = s === 0 && r > 0, p = !o.startsWith("+") || f, C = d.replace(at.NON_PLUS_NUMERIC_GLOBAL, ""), y = C.startsWith("+"), b = C.replace(/\+/g, ""), g = y && p ? `+${b}` : b;
        let m = l + g + a;
        if (m.length > 5) {
          let v = _.utils.getCoreNumber(m, u);
          for (; v.length === 0 && m.length > 0; )
            m = m.slice(0, -1), v = _.utils.getCoreNumber(m, u);
          if (!v)
            return;
          if (this.maxCoreNumberLength && v.length > this.maxCoreNumberLength)
            if (i.selectionEnd === o.length) {
              const T = v.length - this.maxCoreNumberLength;
              m = m.slice(0, m.length - T);
            } else
              return;
        }
        this._setTelInputValue(m);
        const h = s + g.length;
        i.setSelectionRange(h, h), i.dispatchEvent(new InputEvent("input", { bubbles: !0 }));
      };
      this.ui.telInput.addEventListener("paste", t, {
        signal: this.abortController.signal
      });
    }
  }
  //* Adhere to the input's maxlength attr.
  _cap(t) {
    const n = Number(this.ui.telInput.getAttribute("maxlength"));
    return n && t.length > n ? t.substring(0, n) : t;
  }
  //* Trigger a custom event on the input (typed via ItiEventMap).
  _trigger(t, n = {}) {
    const i = new CustomEvent(t, {
      bubbles: !0,
      cancelable: !0,
      detail: n
    });
    this.ui.telInput.dispatchEvent(i);
  }
  //* Open the dropdown.
  _openDropdown() {
    const { fixDropdownWidth: t, countrySearch: n } = this.options;
    if (this.dropdownAbortController = new AbortController(), t && (this.ui.dropdownContent.style.width = `${this.ui.telInput.offsetWidth}px`), this.ui.dropdownContent.classList.remove(E.HIDE), this.ui.selectedCountry.setAttribute(D.EXPANDED, "true"), this._setDropdownPosition(), n) {
      const i = this.ui.countryList.firstElementChild;
      i && (this.ui.highlightListItem(i, !1), this.ui.countryList.scrollTop = 0), this.ui.searchInput.focus();
    }
    this._bindDropdownListeners(), this.ui.dropdownArrow.classList.add(E.ARROW_UP), this._trigger(Vt.OPEN_COUNTRY_DROPDOWN);
  }
  //* Set the dropdown position
  _setDropdownPosition() {
    if (this.options.dropdownContainer && this.options.dropdownContainer.appendChild(this.ui.dropdown), !this.options.useFullscreenPopup) {
      const t = this.ui.telInput.getBoundingClientRect(), n = this.ui.telInput.offsetHeight;
      if (this.options.dropdownContainer) {
        this.ui.dropdown.style.top = `${t.top + n}px`, this.ui.dropdown.style.left = `${t.left}px`;
        const i = () => this._closeDropdown();
        window.addEventListener("scroll", i, {
          signal: this.dropdownAbortController.signal
        });
      }
    }
  }
  //* We only bind dropdown listeners when the dropdown is open.
  _bindDropdownListeners() {
    const t = this.dropdownAbortController.signal;
    this._bindDropdownMouseoverListener(t), this._bindDropdownCountryClickListener(t), this._bindDropdownClickOffListener(t), this._bindDropdownKeydownListener(t), this.options.countrySearch && this._bindDropdownSearchListeners(t);
  }
  //* When mouse over a list item, just highlight that one
  //* we add the class "highlight", so if they hit "enter" we know which one to select.
  _bindDropdownMouseoverListener(t) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${E.COUNTRY_ITEM}`
      );
      s && this.ui.highlightListItem(s, !1);
    };
    this.ui.countryList.addEventListener(
      "mouseover",
      n,
      {
        signal: t
      }
    );
  }
  //* Listen for country selection.
  _bindDropdownCountryClickListener(t) {
    const n = (i) => {
      var r;
      const s = (r = i.target) == null ? void 0 : r.closest(
        `.${E.COUNTRY_ITEM}`
      );
      s && this._selectListItem(s);
    };
    this.ui.countryList.addEventListener("click", n, {
      signal: t
    });
  }
  //* Click off to close (except when this initial opening click is bubbling up).
  //* We cannot just stopPropagation as it may be needed to close another instance.
  _bindDropdownClickOffListener(t) {
    const n = (i) => {
      !!i.target.closest(
        `#iti-${this.id}__dropdown-content`
      ) || this._closeDropdown();
    };
    setTimeout(() => {
      document.documentElement.addEventListener(
        "click",
        n,
        { signal: t }
      );
    }, 0);
  }
  //* Listen for up/down scrolling, enter to select, or escape to close.
  //* Use keydown as keypress doesn't fire for non-char keys and we want to catch if they
  //* just hit down and hold it to scroll down (no keyup event).
  //* Listen on the document because that's where key events are triggered if no input has focus.
  _bindDropdownKeydownListener(t) {
    let n = "", i = null;
    const s = (r) => {
      [
        M.ARROW_UP,
        M.ARROW_DOWN,
        M.ENTER,
        M.ESC
      ].includes(r.key) && (r.preventDefault(), r.stopPropagation(), r.key === M.ARROW_UP || r.key === M.ARROW_DOWN ? this._handleUpDownKey(r.key) : r.key === M.ENTER ? this._handleEnterKey() : r.key === M.ESC && (this._closeDropdown(), this.ui.selectedCountry.focus())), !this.options.countrySearch && at.HIDDEN_SEARCH_CHAR.test(r.key) && (r.stopPropagation(), i && clearTimeout(i), n += r.key.toLowerCase(), this._searchForCountry(n), i = setTimeout(() => {
        n = "";
      }, Ds.HIDDEN_SEARCH_RESET_MS));
    };
    document.addEventListener("keydown", s, { signal: t });
  }
  //* Search input listeners when countrySearch enabled.
  _bindDropdownSearchListeners(t) {
    const n = () => {
      const o = this.ui.searchInput.value.trim();
      this._filterCountriesByQuery(o), this.ui.searchInput.value ? this.ui.searchClearButton.classList.remove(E.HIDE) : this.ui.searchClearButton.classList.add(E.HIDE);
    };
    let i = null;
    const s = () => {
      i && clearTimeout(i), i = setTimeout(() => {
        n(), i = null;
      }, 100);
    };
    this.ui.searchInput.addEventListener("input", s, {
      signal: t
    });
    const r = () => {
      this.ui.searchInput.value = "", this.ui.searchInput.focus(), n();
    };
    this.ui.searchClearButton.addEventListener("click", r, {
      signal: t
    });
  }
  //* Hidden search (countrySearch disabled): Find the first list item whose name starts with the query string.
  _searchForCountry(t) {
    const n = Ps(this.countries, t);
    if (n) {
      const i = n.nodeById[this.id];
      this.ui.highlightListItem(i, !1), this.ui.scrollTo(i);
    }
  }
  //* Country search: Filter the countries according to the search query.
  _filterCountriesByQuery(t) {
    let n;
    t === "" ? n = this.countries : n = Ts(this.countries, t), this.ui.filterCountries(n);
  }
  //* Highlight the next/prev item in the list (and ensure it is visible).
  _handleUpDownKey(t) {
    var i, s;
    let n = t === M.ARROW_UP ? (i = this.ui.highlightedItem) == null ? void 0 : i.previousElementSibling : (s = this.ui.highlightedItem) == null ? void 0 : s.nextElementSibling;
    !n && this.ui.countryList.childElementCount > 1 && (n = t === M.ARROW_UP ? this.ui.countryList.lastElementChild : this.ui.countryList.firstElementChild), n && (this.ui.scrollTo(n), this.ui.highlightListItem(n, !1));
  }
  //* Select the currently highlighted item.
  _handleEnterKey() {
    this.ui.highlightedItem && this._selectListItem(this.ui.highlightedItem);
  }
  //* Update the input's value to the given val (format first if possible)
  //* NOTE: this is called from _setInitialState, handleUtils and setNumber.
  _updateValFromNumber(t) {
    let n = t;
    if (this.options.formatOnDisplay && _.utils && this.selectedCountryData) {
      const i = this.options.nationalMode || !n.startsWith("+") && !this.options.separateDialCode, { NATIONAL: s, INTERNATIONAL: r } = _.utils.numberFormat, o = i ? s : r;
      n = _.utils.formatNumber(
        n,
        this.selectedCountryData.iso2,
        o
      );
    }
    n = this._beforeSetNumber(n), this._setTelInputValue(n);
  }
  //* Check if need to select a new country based on the given number
  //* Note: called from _setInitialState, keyup handler, setNumber.
  _updateCountryFromNumber(t) {
    const n = this._getNewCountryFromNumber(t);
    return n !== null ? this._setCountry(n) : !1;
  }
  // if there is a selected country, and the number doesn't start with a dial code, then add it
  _ensureHasDialCode(t) {
    const { dialCode: n, nationalPrefix: i } = this.selectedCountryData;
    if (t.startsWith("+") || !n)
      return t;
    const o = i && t.startsWith(i) && !this.options.separateDialCode ? t.substring(1) : t;
    return `+${n}${o}`;
  }
  // Get the country ISO2 code from the given number
  // BUT ONLY IF ITS CHANGED FROM THE CURRENTLY SELECTED COUNTRY
  // NOTE: consider refactoring this to be more clear
  _getNewCountryFromNumber(t) {
    const n = t.indexOf("+");
    let i = n ? t.substring(n) : t;
    const s = this.selectedCountryData.iso2, r = this.selectedCountryData.dialCode;
    i = this._ensureHasDialCode(i);
    const o = this._getDialCode(i, !0), l = jt(i);
    if (o) {
      const a = jt(o), u = this.dialCodeToIso2Map[a];
      if (u.length === 1)
        return u[0] === s ? null : u[0];
      if (!s && this.defaultCountry && u.includes(this.defaultCountry))
        return this.defaultCountry;
      if (r === kn.NANP && Ye(l))
        return null;
      const { areaCodes: d, priority: f } = this.selectedCountryData;
      if (d) {
        const g = d.map(
          (m) => `${r}${m}`
        );
        for (const m of g)
          if (l.startsWith(m))
            return null;
      }
      const C = d && !(f === 0) && l.length > a.length, y = s && u.includes(s) && !C, b = s === u[0];
      if (!y && !b)
        return u[0];
    } else if (i.startsWith("+") && l.length) {
      const a = this.selectedCountryData.dialCode || "";
      return a && a.startsWith(l) ? null : "";
    } else if ((!i || i === "+") && !s)
      return this.defaultCountry;
    return null;
  }
  //* Update the selected country, dial code (if separateDialCode), placeholder, title, and selected list item.
  //* Note: called from _setInitialState, _updateCountryFromNumber, _selectListItem, setCountry.
  _setCountry(t) {
    const { separateDialCode: n, showFlags: i, i18n: s, allowDropdown: r } = this.options, o = this.selectedCountryData.iso2 || "";
    if (r && this.ui.updateSelectedItem(t), this.selectedCountryData = t ? this.countryByIso2.get(t) : {}, this.selectedCountryData.iso2 && (this.defaultCountry = this.selectedCountryData.iso2), this.ui.selectedCountry) {
      const l = t && i ? `${E.FLAG} iti__${t}` : `${E.FLAG} ${E.GLOBE}`;
      let a, u;
      if (t) {
        const { name: c, dialCode: d } = this.selectedCountryData;
        u = c, a = s.selectedCountryAriaLabel.replace("${countryName}", c).replace("${dialCode}", `+${d}`);
      } else
        u = s.noCountrySelected, a = s.noCountrySelected;
      this.ui.selectedCountryInner.className = l, this.ui.selectedCountry.setAttribute("title", u), this.ui.selectedCountry.setAttribute(D.LABEL, a);
    }
    if (n) {
      const l = this.selectedCountryData.dialCode ? `+${this.selectedCountryData.dialCode}` : "";
      this.ui.selectedDialCode.textContent = l, this.ui.updateInputPadding();
    }
    return this._updatePlaceholder(), this._updateMaxLength(), o !== t;
  }
  //* Update the maximum valid number length for the currently selected country.
  _updateMaxLength() {
    const { strictMode: t, placeholderNumberType: n, validationNumberTypes: i } = this.options, { iso2: s } = this.selectedCountryData;
    if (t && _.utils)
      if (s) {
        const r = _.utils.numberType[n];
        let o = _.utils.getExampleNumber(
          s,
          !1,
          r,
          !0
        ), l = o;
        for (; _.utils.isPossibleNumber(
          o,
          s,
          i
        ); )
          l = o, o += "0";
        const a = _.utils.getCoreNumber(l, s);
        this.maxCoreNumberLength = a.length, s === "by" && (this.maxCoreNumberLength = a.length + 1);
      } else
        this.maxCoreNumberLength = null;
  }
  //* Update the input placeholder to an example number from the currently selected country.
  _updatePlaceholder() {
    const {
      autoPlaceholder: t,
      placeholderNumberType: n,
      nationalMode: i,
      customPlaceholder: s
    } = this.options, r = t === me.AGGRESSIVE || !this.ui.hadInitialPlaceholder && t === me.POLITE;
    if (_.utils && r) {
      const o = _.utils.numberType[n];
      let l = this.selectedCountryData.iso2 ? _.utils.getExampleNumber(
        this.selectedCountryData.iso2,
        i,
        o
      ) : "";
      l = this._beforeSetNumber(l), typeof s == "function" && (l = s(l, this.selectedCountryData)), this.ui.telInput.setAttribute("placeholder", l);
    }
  }
  //* Called when the user selects a list item from the dropdown.
  _selectListItem(t) {
    const n = t.dataset[Ke.COUNTRY_CODE], i = this._setCountry(n);
    this._closeDropdown();
    const s = t.dataset[Ke.DIAL_CODE];
    if (this._updateDialCode(s), this.options.formatOnDisplay) {
      const r = this._getTelInputValue();
      this._updateValFromNumber(r);
    }
    this.ui.telInput.focus(), i && this._triggerCountryChange();
  }
  //* Close the dropdown and unbind any listeners.
  _closeDropdown() {
    this.ui.dropdownContent.classList.contains(E.HIDE) || (this.ui.dropdownContent.classList.add(E.HIDE), this.ui.selectedCountry.setAttribute(D.EXPANDED, "false"), this.options.countrySearch && (this.ui.searchInput.removeAttribute(D.ACTIVE_DESCENDANT), this.ui.highlightedItem && (this.ui.highlightedItem.classList.remove(E.HIGHLIGHT), this.ui.highlightedItem = null)), this.ui.dropdownArrow.classList.remove(E.ARROW_UP), this.dropdownAbortController.abort(), this.dropdownAbortController = null, this.options.dropdownContainer && this.ui.dropdown.remove(), this._trigger(Vt.CLOSE_COUNTRY_DROPDOWN));
  }
  //* Replace any existing dial code with the new one
  //* Note: called from _selectListItem and setCountry
  _updateDialCode(t) {
    const n = this._getTelInputValue(), i = `+${t}`;
    let s;
    if (n.startsWith("+")) {
      const r = this._getDialCode(n);
      r ? s = n.replace(r, i) : s = i, this._setTelInputValue(s);
    }
  }
  //* Try and extract a valid international dial code from a full telephone number.
  //* Note: returns the raw string inc plus character and any whitespace/dots etc.
  _getDialCode(t, n) {
    let i = "";
    if (t.startsWith("+")) {
      let s = "", r = !1;
      for (let o = 0; o < t.length; o++) {
        const l = t.charAt(o);
        if (/[0-9]/.test(l)) {
          if (s += l, !!!this.dialCodeToIso2Map[s])
            break;
          if (this.dialCodes.has(s)) {
            if (i = t.substring(0, o + 1), r = !0, !n)
              break;
          } else n && r && (i = t.substring(0, o + 1));
          if (s.length === this.dialCodeMaxLen)
            break;
        }
      }
    }
    return i;
  }
  //* Get the input val, adding the dial code if separateDialCode is enabled.
  _getFullNumber(t) {
    const n = t ? this._normaliseNumerals(t) : this._getTelInputValue(), { dialCode: i } = this.selectedCountryData;
    let s;
    const r = jt(n);
    return this.options.separateDialCode && !n.startsWith("+") && i && r ? s = `+${i}` : s = "", s + n;
  }
  //* Remove the dial code if separateDialCode is enabled also cap the length if the input has a maxlength attribute
  _beforeSetNumber(t) {
    const n = this._getDialCode(t), i = Hs(
      t,
      n,
      this.options.separateDialCode,
      this.selectedCountryData
    );
    return this._cap(i);
  }
  //* Trigger the 'countrychange' event.
  _triggerCountryChange() {
    this._trigger(Vt.COUNTRY_CHANGE);
  }
  //**************************
  //*  SECRET PUBLIC METHODS
  //**************************
  //* This is called when the geoip call returns.
  handleAutoCountry() {
    this.options.initialCountry === oe.AUTO && _.autoCountry && (this.defaultCountry = _.autoCountry, this.selectedCountryData.iso2 || this.ui.selectedCountryInner.classList.contains(E.GLOBE) || this.setCountry(this.defaultCountry), this.resolveAutoCountryPromise());
  }
  //* This is called when the utils request completes.
  handleUtils() {
    if (_.utils) {
      const t = this._getTelInputValue();
      t && this._updateValFromNumber(t), this.selectedCountryData.iso2 && (this._updatePlaceholder(), this._updateMaxLength());
    }
    this.resolveUtilsScriptPromise();
  }
  //********************
  //*  PUBLIC METHODS
  //********************
  //* Remove plugin.
  destroy() {
    this.ui.telInput && (this.options.allowDropdown && this._closeDropdown(), this.abortController.abort(), this.abortController = null, this.ui.destroy(), _.instances instanceof Map ? _.instances.delete(this.id) : delete _.instances[this.id]);
  }
  //* Get the extension from the current number.
  getExtension() {
    return _.utils ? _.utils.getExtension(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : "";
  }
  //* Format the number to the given format.
  getNumber(t) {
    if (_.utils) {
      const { iso2: n } = this.selectedCountryData, i = this._getFullNumber(), s = _.utils.formatNumber(
        i,
        n,
        t
      );
      return this._mapAsciiToUserNumerals(s);
    }
    return "";
  }
  //* Get the type of the entered number e.g. landline/mobile.
  getNumberType() {
    return _.utils ? _.utils.getNumberType(
      this._getFullNumber(),
      this.selectedCountryData.iso2
    ) : ze.UNKNOWN_NUMBER_TYPE;
  }
  //* Get the country data for the currently selected country.
  getSelectedCountryData() {
    return this.selectedCountryData;
  }
  //* Get the validation error.
  getValidationError() {
    if (_.utils) {
      const { iso2: t } = this.selectedCountryData;
      return _.utils.getValidationError(this._getFullNumber(), t);
    }
    return ze.UNKNOWN_VALIDATION_ERROR;
  }
  //* Validate the input val using number length only
  isValidNumber() {
    const { dialCode: t, iso2: n } = this.selectedCountryData;
    if (t === re.DIAL_CODE && _.utils) {
      const i = this._getFullNumber(), s = _.utils.getCoreNumber(i, n);
      if (s[0] === re.MOBILE_PREFIX && s.length !== re.MOBILE_CORE_LENGTH)
        return !1;
    }
    return this._validateNumber(!1);
  }
  //* Validate the input val with precise validation
  isValidNumberPrecise() {
    return this._validateNumber(!0);
  }
  _utilsIsPossibleNumber(t) {
    return _.utils ? _.utils.isPossibleNumber(
      t,
      this.selectedCountryData.iso2,
      this.options.validationNumberTypes
    ) : null;
  }
  //* Shared internal validation logic to handle alpha character extension rules.
  _validateNumber(t) {
    if (!_.utils)
      return null;
    if (!this.selectedCountryData.iso2)
      return !1;
    const n = (o) => t ? this._utilsIsValidNumber(o) : this._utilsIsPossibleNumber(o), i = this._getFullNumber(), s = i.search(at.ALPHA_UNICODE);
    if (s > -1 && !this.options.allowPhonewords) {
      const o = i.substring(0, s), l = n(o), a = n(i);
      return l && a;
    }
    return n(i);
  }
  _utilsIsValidNumber(t) {
    return _.utils ? _.utils.isValidNumber(
      t,
      this.selectedCountryData.iso2,
      this.options.validationNumberTypes
    ) : null;
  }
  //* Update the selected country, and update the input val accordingly.
  setCountry(t) {
    const n = t == null ? void 0 : t.toLowerCase();
    if (!ue(n))
      throw new Error(`Invalid country code: '${n}'`);
    const i = this.selectedCountryData.iso2;
    if (t && n !== i || !t && i) {
      if (this._setCountry(n), this._updateDialCode(this.selectedCountryData.dialCode), this.options.formatOnDisplay) {
        const r = this._getTelInputValue();
        this._updateValFromNumber(r);
      }
      this._triggerCountryChange();
    }
  }
  //* Set the input value and update the country.
  setNumber(t) {
    const n = this._normaliseNumerals(t), i = this._updateCountryFromNumber(n);
    this._updateValFromNumber(n), i && this._triggerCountryChange(), this._trigger(Vt.INPUT, { isSetNumber: !0 });
  }
  //* Set the placeholder number typ
  setPlaceholderNumberType(t) {
    this.options.placeholderNumberType = t, this._updatePlaceholder();
  }
  setDisabled(t) {
    this.ui.telInput.disabled = t, t ? this.ui.selectedCountry.setAttribute("disabled", "true") : this.ui.selectedCountry.removeAttribute("disabled");
  }
}
const Ks = (e) => {
  if (!_.utils && !_.startedLoadingUtilsScript) {
    let t;
    if (typeof e == "function")
      try {
        t = Promise.resolve(e());
      } catch (n) {
        return Promise.reject(n);
      }
    else
      return Promise.reject(
        new TypeError(
          `The argument passed to attachUtils must be a function that returns a promise for the utilities module, not ${typeof e}`
        )
      );
    return _.startedLoadingUtilsScript = !0, t.then((n) => {
      const i = n == null ? void 0 : n.default;
      if (!i || typeof i != "object")
        throw new TypeError(
          "The loader function passed to attachUtils did not resolve to a module object with utils as its default export."
        );
      return _.utils = i, kt("handleUtils"), !0;
    }).catch((n) => {
      throw kt("rejectUtilsScriptPromise", n), n;
    });
  }
  return null;
}, kt = (e, ...t) => {
  Object.values(_.instances).forEach((n) => {
    const i = n[e];
    typeof i == "function" && i.apply(n, t);
  });
}, _ = Object.assign(
  (e, t) => {
    const n = new Se(e, t);
    return _.instances[n.id] = n, e.iti = n, n;
  },
  {
    defaults: Mn,
    //* Using a static var like this allows us to mock it in the tests.
    documentReady: () => document.readyState === "complete",
    //* Get the country data object.
    getCountryData: () => _t,
    //* A getter for the plugin instance.
    getInstance: (e) => {
      const t = e.dataset.intlTelInputId;
      return t ? _.instances[t] : null;
    },
    //* A map from instance ID to instance object.
    instances: {},
    attachUtils: Ks,
    startedLoadingUtilsScript: !1,
    startedLoadingAutoCountry: !1,
    version: "25.15.1"
  }
);
var Ys = /* @__PURE__ */ is("<input/>");
function Qs(e, t) {
  yi(t, !0);
  let n = Ht(t, "disabled", 3, !1), i = Ht(t, "inputProps", 19, () => ({})), s = Ht(t, "options", 19, () => ({})), r = Ht(t, "value", 3, ""), o = /* @__PURE__ */ G(void 0), l = /* @__PURE__ */ G(void 0), a = /* @__PURE__ */ G(null), u = /* @__PURE__ */ G(!1);
  const c = () => I(l) ? s().strictMode ? I(l).isValidNumberPrecise() : I(l).isValidNumber() : null, d = () => {
    var h, v;
    const m = c();
    I(a) !== m && (V(a, m, !0), (h = t.onChangeValidity) == null || h.call(t, !!m), (v = t.onChangeErrorCode) == null || v.call(t, m ? null : I(l).getValidationError()));
  }, f = () => {
    var m, h;
    (h = t.onChangeNumber) == null || h.call(t, ((m = I(l)) == null ? void 0 : m.getNumber()) ?? ""), d();
  }, p = () => {
    var m, h;
    (h = t.onChangeCountry) == null || h.call(t, ((m = I(l)) == null ? void 0 : m.getSelectedCountryData().iso2) ?? ""), f();
  };
  Vn(() => {
    I(o) && (V(l, _(I(o), s()), !0), I(o).addEventListener("countrychange", p), r() && I(l).setNumber(r()), n() && I(l).setDisabled(n()), V(a, c(), !0), V(u, !0));
  }), Wn(() => {
    var m;
    I(o) && I(o).removeEventListener("countrychange", p), (m = I(l)) == null || m.destroy();
  }), Wi(() => {
    I(u) && I(l) && I(l).setDisabled(n());
  });
  function C() {
    return I(l);
  }
  function y() {
    return I(o);
  }
  var b = { getInstance: C, getInput: y }, g = Ys();
  return vs(g, () => ({ type: "tel", oninput: f, ...i() }), void 0, void 0, void 0, void 0, !0), bs(g, (m) => V(o, m), () => I(o)), ss(e, g), vi(b);
}
export {
  Qs as default
};
