// @splunk/react-ui/Accordion.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Panel: () => /* reexport */ x,
        default: () => /* reexport */ D
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const l = require("@splunk/react-ui/useControlled");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const u = require("@splunk/react-ui/usePrevious");
    var c =  e.n(u);
    // CONCATENATED MODULE: ./src/Accordion/AccordionContext.tsx
    var s =  (0, r.createContext)({});
    s.displayName = "Accordion";
    /* harmony default export */    const f = s;
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var p =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const v = require("@splunk/react-ui/Box");
    var m =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Accordion/AccordionStyles.ts
    var b = p()(m()).withConfig({
        displayName: "AccordionStyles__StyledBox",
        componentId: "m39hyh-0"
    })([ "", ";flex-direction:column;margin-bottom:0;" ], y.mixins.reset("flex"));
    // CONCATENATED MODULE: external "@splunk/react-ui/CollapsiblePanel"
    const g = require("@splunk/react-ui/CollapsiblePanel");
    var h =  e.n(g);
    // CONCATENATED MODULE: ./src/Accordion/Panel.tsx
    function O() {
        return O = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, O.apply(null, arguments);
    }
    function P(e, n) {
        if (null == e) return {};
        var r, t, a = I(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function I(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var j = {
        description: o().string,
        elementRef: o().oneOfType([ o().func, o().object ]),
        inset: o().bool,
        panelId: o().any.isRequired,
        title: o().node.isRequired
    };
    var S = {
        flexShrink: 1
    };
    /**
 * `Accordion.Panel` operates as a container component for content in an `Accordion`.
 */    function C(e) {
        var n = e.panelId, a = e.inset, o = P(e, [ "panelId", "inset" ]);
        // @docs-props-type AccordionPanelPropsBase
                var l = (0, r.useContext)(s), i = l.onChange, u = l.openPanelId, c = l.inset;
        var f = u === n;
        var d = a !== undefined ? a : c;
        // Props are different based on controlled/uncontrolled
        // For controlled, onChange required prop for controlled cases and must be defined.
                var p = i ? {
            onChange: i,
            open: f
        } : {
            onChange: undefined
        };
        
        return t().createElement(h(), O({
            panelId: n,
            "data-test": "panel",
            inset: d,
            "data-test-panel-id": n,
            style: S
        }, o, p));
    }
    C.propTypes = j;
    /* harmony default export */    const x = C;
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var w = "is deprecated and will be removed in the next major version.";
    var A = function e(n) {
        var t = n.additionalMessage, a = t === void 0 ? "" : t, o = n.componentName;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ a, o ]);
    };
    var E = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, a = n.componentName, o = n.propName, l = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ t, a, o, l ]);
    };
    var N = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, a = n.componentName, o = n.deprecatedPropValue, l = n.propName, i = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ t, a, l, i, o ]);
    };
    // CONCATENATED MODULE: ./src/Accordion/Accordion.tsx
    function q() {
        return q = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, q.apply(null, arguments);
    }
    function M(e, n) {
        return V(e) || _(e, n) || T(e, n) || k();
    }
    function k() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function T(e, n) {
        if (e) {
            if ("string" == typeof e) return R(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? R(e, n) : void 0;
        }
    }
    function R(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function _(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, a, o, l, i = [], u = !0, c = !1;
            try {
                if (o = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = o.call(r)).done) && (i.push(t.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (l = r["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function V(e) {
        if (Array.isArray(e)) return e;
    }
    function B(e, n) {
        if (null == e) return {};
        var r, t, a = U(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function U(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var $ = {
        children: o().node,
        defaultOpenPanelId: o().any,
        elementRef: o().oneOfType([ o().func, o().object ]),
        inset: o().bool,
        onChange: o().func,
        openPanelId: o().any
    };
    /**
 * @deprecated
 * Accordion has been deprecated and will be removed in a future major version. Use Collapsible Panel's SingleOpenPanelGroup API instead.
 */    function z(e) {
        var n = e.children, a = e.defaultOpenPanelId, o = e.elementRef, l = e.inset, u = l === void 0 ? true : l, s = e.onChange, d = e.openPanelId, p = B(e, [ "children", "defaultOpenPanelId", "elementRef", "inset", "onChange", "openPanelId" ]);
        // @docs-props-type AccordionPropsBase
                A({
            additionalMessage: "Use Collapsible Panel instead.",
            componentName: "Accordion"
        });
        var v = c()(a);
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ a, v ]);
        var m = i()({
            componentName: "Accordion",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultOpenPanelId",
            valuePropName: "openPanelId"
        });
        var y = (0, r.useState)(a), g = M(y, 2), h = g[0], O = g[1];
        var P = m ? d : h;
        var I = r.Children.toArray(n).filter(r.isValidElement);
 // remove false, null, 0, etc
                var j = (0, r.useCallback)((function(e, n) {
            var r = n.panelId, t = n.reason;
            // if the open panel is clicked on, collapse it
                        var a = P === r ? undefined : r;
            if (!m) {
                O(a);
            }
            s === null || s === void 0 ? void 0 : s(e, {
                event: e,
                panelId: a,
                reason: t
            });
        }), [ m, s, P ]);
        var S = (0, r.useMemo)((function() {
            return {
                inset: u,
                onChange: j,
                openPanelId: P
            };
        }), [ j, u, P ]);
        
        return t().createElement(b, q({
            "data-test": "accordion",
            "data-test-open-panel-id": P,
            ref: o
        }, p),  t().createElement(f.Provider, {
            value: S
        }, I));
    }
    z.propTypes = $;
    z.Panel = x;
    /* harmony default export */    const D = z;
    // CONCATENATED MODULE: ./src/Accordion/index.ts
    module.exports = n;
    /******/})();
