// @splunk/react-ui/AnchorMenu.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ H
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const a = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const o = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/AnchorMenu/AnchorMenuContext.tsx
    var c =  (0, r.createContext)({
        hideSpine: false
    });
    /* harmony default export */    const u = c;
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var d =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const f = require("@splunk/react-ui/Heading");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const v = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/AnchorMenu/AnchorMenuStyles.ts
    var m = d().nav.withConfig({
        displayName: "AnchorMenuStyles__StyledAnchorMenu",
        componentId: "peh8x9-0"
    })([ "display:flex;flex-direction:column;min-width:200px;gap:", ";" ], y.variables.spacingSmall);
    var h = d()(p()).withConfig({
        displayName: "AnchorMenuStyles__StyledHeading",
        componentId: "peh8x9-1"
    })([ "flex-shrink:0;" ]);
    var b = d()(v.Item).withConfig({
        displayName: "AnchorMenuStyles__StyledItem",
        componentId: "peh8x9-2"
    })([ "display:inline-block;width:100%;&:focus{box-shadow:", ";z-index:1;}color:", ";font-weight:", ";", " &[aria-current]{background:", ";color:", ";border-inline-start:2px solid ", ";}" ], y.variables.focusShadow, y.variables.contentColorDefault, y.variables.fontWeightSemiBold, (function(e) {
        var n = e.$hideSpine;
        return !n ? (0, s.css)([ "border-inline-start:2px solid ", ";" ], y.variables.borderColorWeak) : "";
    }), y.variables.interactiveColorOverlaySelected, y.variables.contentColorActive, y.variables.interactiveColorAccent);
    var g = d().ul.withConfig({
        displayName: "AnchorMenuStyles__StyledAnchorMenuList",
        componentId: "peh8x9-3"
    })([ "--parent-level:0;list-style:none;margin:0;padding:0;" ]);
    var S = d().li.withConfig({
        displayName: "AnchorMenuStyles__StyledAnchorMenuItem",
        componentId: "peh8x9-4"
    })([ "--current-level:calc(var(--parent-level) + 1);& ", "{--parent-level:var(--current-level);}& ", "{padding-inline-start:calc(var(--current-level) * ", ");}" ], g, b, y.variables.spacingLarge);
    // CONCATENATED MODULE: ./src/AnchorMenu/Item.tsx
    function O() {
        return O = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, O.apply(null, arguments);
    }
    function I(e, n) {
        if (null == e) return {};
        var r, t, l = w(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (l[r] = e[r]);
            }
        }
        return l;
    }
    function w(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var x = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        itemId: i().string,
        label: i().string,
        onClick: i().func,
        to: i().string
    };
    function j(e) {
        var n = e.children, l = e.elementRef, i = e.itemId, a = e.label, o = e.onClick, c = e.to, s = I(e, [ "children", "elementRef", "itemId", "label", "onClick", "to" ]);
        // @docs-props-type AnchorMenuItemPropsBase
                var d = (0, r.useContext)(u), f = d.activeItemId, p = d.hideSpine;
        var v = i !== null && i !== void 0 ? i : c;
        var y = v != null && v === f;
        
        return t().createElement(S, O({
            ref: l
        }, s),  t().createElement(b, {
            $hideSpine: p,
            "aria-current": y ? true : undefined,
            onClick: o,
            role: "link",
            to: c
        }, a),  t().createElement(g, null, n));
    }
    j.propTypes = x;
    /* harmony default export */    const C = j;
    // CONCATENATED MODULE: ./src/AnchorMenu/AnchorMenu.tsx
    function A() {
        return A = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, A.apply(null, arguments);
    }
    function M(e, n) {
        return q(e) || E(e, n) || k(e, n) || _();
    }
    function _() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function k(e, n) {
        if (e) {
            if ("string" == typeof e) return P(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? P(e, n) : void 0;
        }
    }
    function P(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function E(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, l, i, a, o = [], c = !0, u = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (t = i.call(r)).done) && (o.push(t.value), o.length !== n); c = !0) {
                }
            } catch (e) {
                u = !0, l = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (u) throw l;
                }
            }
            return o;
        }
    }
    function q(e) {
        if (Array.isArray(e)) return e;
    }
    function T(e, n) {
        if (null == e) return {};
        var r, t, l = R(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (l[r] = e[r]);
            }
        }
        return l;
    }
    function R(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var N = {
        activeItemId: i().string,
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        hideSpine: i().bool,
        label: i().string
    };
    var D = (0, a._)("On this page:");
    function $(e) {
        var n = e.activeItemId, l = e.children, i = e.elementRef, a = e.hideSpine, c = e.label, s = c === void 0 ? D : c, d = T(e, [ "activeItemId", "children", "elementRef", "hideSpine", "label" ]);
        // @docs-props-type AnchorMenuPropsBase
                var f = (0, r.useMemo)((function() {
            return {
                activeItemId: n,
                hideSpine: a
            };
        }), [ n, a ]);
        var p = (0, r.useState)((0, o.createDOMID)("anchor-menu-heading")), v = M(p, 1), y = v[0];
        
        return t().createElement(m, A({
            "aria-labelledby": y,
            "data-test": "anchor-menu",
            ref: i
        }, d), s &&  t().createElement(h, {
            level: 4,
            id: y
        }, s),  t().createElement(u.Provider, {
            value: f
        },  t().createElement(g, null, l)));
    }
    $.propTypes = N;
    $.Item = C;
    /* harmony default export */    const H = $;
    // CONCATENATED MODULE: ./src/AnchorMenu/index.ts
    module.exports = n;
    /******/})();
