// @splunk/react-ui/Anchor.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ y
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var c =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const l = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Anchor/AnchorStyles.ts
    // TODO: name is deprecated, it's only defined here to satisfy ts - remove/replace
    var f = c().a.withConfig({
        displayName: "AnchorStyles__StyledAnchor",
        componentId: "sc-1m9yzet-0"
    })([ "", ";color:inherit;font-size:inherit;font-family:inherit;font-weight:inherit;line-height:inherit;text-decoration:inherit;text-rendering:inherit;text-transform:inherit;cursor:pointer;margin:1px;&:hover::after{content:' #';opacity:0.5;}&:focus,&:active{box-shadow:", ";&::after{content:' #';opacity:0.5;}}" ], l.mixins.reset("inline"), l.variables.focusShadow);
    // CONCATENATED MODULE: ./src/Anchor/Anchor.tsx
    function s() {
        return s = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, s.apply(null, arguments);
    }
    function u(e, r) {
        if (null == e) return {};
        var t, n, o = d(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function d(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var p = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        name: i().string.isRequired
    };
    /**
 * `Anchor` is a utility component used to add a fragment link to an element.
 */    function h(e) {
        var r = e.children, t = e.elementRef, o = e.name, i = u(e, [ "children", "elementRef", "name" ]);
        // @docs-props-type AnchorPropsBase
                
        return n().createElement(f, s({
            "data-test": "anchor",
            ref: t,
            id: o
        }, i, {
            name: o,
            href: "#".concat(o)
        }), r);
    }
    h.propTypes = p;
    /* harmony default export */    const y = h;
    // CONCATENATED MODULE: ./src/Anchor/index.ts
    module.exports = r;
    /******/})();
