// @splunk/react-ui/Avatar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ Y,
        getInitials: () => /* reexport */ h
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/omit"
    const o = require("lodash/omit");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var s =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const f = require("@splunk/react-ui/Clickable");
    var d =  e.n(f);
    // CONCATENATED MODULE: ./src/Avatar/AvatarStyles.ts
    var v = (0, c.css)([ "", ";align-items:center;justify-content:center;border-radius:50%;font-weight:normal;overflow:visible;vertical-align:middle;position:relative;background-color:", ";", ";& > img{object-fit:cover;height:100%;width:100%;border-radius:50%;}" ], u.mixins.reset("inline-flex"), (function(e) {
        var r = e.$backgroundColor;
        return r;
    }), (function(e) {
        var r = e.$size;
        if (typeof r === "string") {
            if (r === "small") {
                return (0, c.css)([ "height:24px;width:24px;" ]);
            }
            if (r === "medium") {
                return (0, c.css)([ "height:32px;width:32px;" ]);
            }
            if (r === "large") {
                return (0, c.css)([ "height:48px;width:48px;" ]);
            }
        }
        return (0, c.css)([ "height:", ";width:", ";" ], r, r);
    }));
    var b = s().div.withConfig({
        displayName: "AvatarStyles__Styled",
        componentId: "sc-1t4nch-0"
    })([ "", "" ], v);
    var y = s()(d()).withConfig({
        displayName: "AvatarStyles__StyledClickable",
        componentId: "sc-1t4nch-1"
    })([ "", " cursor:pointer;position:relative;border:1px solid transparent;border-radius:50%;:hover{border-color:", ";}:focus{box-shadow:", ";z-index:1;}:active{border-color:", ";}" ], v, u.variables.actionColorBorderSecondaryHover, u.variables.focusShadow, u.variables.actionColorBorderSecondaryActive);
    var p = s().svg.withConfig({
        displayName: "AvatarStyles__StyledSVG",
        componentId: "sc-1t4nch-2"
    })([ "position:absolute;top:0;left:0;border-radius:50%;" ]);
    var m = s().text.withConfig({
        displayName: "AvatarStyles__StyledText",
        componentId: "sc-1t4nch-3"
    })([ "font-weight:", ";text-transform:capitalize;text-anchor:middle;dominant-baseline:central;fill:", ";font-size:", ";" ], u.variables.fontWeightBold, (function(e) {
        var r = e.$textColor;
        return r;
    }), u.variables.fontSize);
    // CONCATENATED MODULE: ./src/Avatar/getInitials.ts
    /**
 * Returns a suitable set of initials for a name. Uses the first character of each
 * name segment and omits middle segments if the segment count is greater than three.
 * @param {string} name - The full name.
 * @returns {string} Limited to three characters. Empty if `name` is empty.
 * @public
 */
    function g(e) {
        var r = (e.match(/[^\s-]+/g) || []).map((function(e) {
            return e[0];
        })).join("");
        return r.length > 3 ? "".concat(r[0]).concat(r[r.length - 1]) : r;
    }
    /* harmony default export */    const h = g;
    // CONCATENATED MODULE: external "@splunk/ui-utils/color"
    const O = require("@splunk/ui-utils/color");
    // CONCATENATED MODULE: ./src/utils/determineTextBrightness.ts
    function S(e, r) {
        return k(e) || C(e, r) || x(e, r) || w();
    }
    function w() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function x(e, r) {
        if (e) {
            if ("string" == typeof e) return j(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? j(e, r) : void 0;
        }
    }
    function j(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function C(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (n = i.call(t)).done) && (l.push(n.value), l.length !== r); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function k(e) {
        if (Array.isArray(e)) return e;
    }
    var A = "lightText";
    var P = "darkText";
    function E(e) {
        var r = S(e, 3), t = r[0], n = r[1], o = r[2];
        var i = (t * 299 + n * 587 + o * 114) / 1e3;
 // formula from https://www.w3.org/TR/AERT/#color-contrast
                return i > 125 ? P : A;
    }
    /**
 * Determines if light or dark text should be used over a specific background color by determining the color brightness of a
 * supplied background color.
 * @param isLight - True if colorScheme is light. False if colorScheme is dark.
 * @param backgroundColor - accepts hex color, rgb, rgba, common color keywords
 * @returns DARK_TEXT or LIGHT_TEXT
 *
 * Some things to watch out for:
 * 1. determineTextBrightness helps you come close to accessible contrast, but it is not guaranteed to always work.
 * It might fail say for a color that is in the middle in terms of brightness/ near the borderline of the cutoff.
 * For example if a background color has a text brightness of 124 when the cutoff is determined by: brightness > 125 ? DARK_TEXT : LIGHT_TEXT
 * 2. If you use a color that is not black for dark text or white for light text that would decrease the likelihood that it would pass contrast with the background color
 * if you use determineTextBrightness to decide whether to use dark or light text.
 */    function T(e, r) {
        if (typeof r === "string") {
            if ((0, O.isValidHexColor)(r)) {
                // hex
                var t = r;
                if (t.length === 3 || t.length === 4) {
                    // shorthand hex
                    t = (0, O.expandShortHandHex)(t);
                }
                var n = (0, O.hexToRgb)(t);
                if (n != null) {
                    return E(n);
                }
            } else if (r.includes(",")) {
                if ((0, O.isValidRgb)(r)) {
                    // rgb
                    var o = r.replace(/[^\d,]/g, "").split(",").slice(0, 3);
                    var i = o.map((function(e) {
                        return parseInt(e, 10);
                    })), a = S(i, 3), l = a[0], u = a[1], c = a[2];
                    return E([ l, u, c ]);
                }
            } else if (O.extendedColorKeywords.includes(r.toLowerCase())) {
                // named color
                var s = (0, O.namedColorToHex)(r.toLowerCase());
                if (s != null) {
                    var f = (0, O.hexToRgb)(s);
                    if (f != null) {
                        return E(f);
                    }
                }
            }
        }
        // TODO: Need to add support for auto text color to work with theme variable background color (SUI-5706).
        /**
   * We know the backgroundColor defaults to variables.neutral200 so to save on computing we hard code the text brightness.
   * TODO: Convert theme variable color variables.neutral200 to rgb dynamically and memoize it (SUI-5731).
   */
        // backgroundColor is either not provided or an accepted representation of color
                return e ? P : A;
    }
    // CONCATENATED MODULE: ./src/Avatar/Avatar.tsx
    function I(e) {
        "@babel/helpers - typeof";
        return I = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, I(e);
    }
    function z(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function _(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? z(Object(t), !0).forEach((function(r) {
                R(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : z(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function R(e, r, t) {
        return (r = q(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function q(e) {
        var r = $(e, "string");
        return "symbol" == I(r) ? r : r + "";
    }
    function $(e, r) {
        if ("object" != I(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != I(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function H() {
        return H = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, H.apply(null, arguments);
    }
    function M(e, r) {
        return V(e) || L(e, r) || B(e, r) || N();
    }
    function N() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function B(e, r) {
        if (e) {
            if ("string" == typeof e) return D(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? D(e, r) : void 0;
        }
    }
    function D(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function L(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (n = i.call(t)).done) && (l.push(n.value), l.length !== r); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function V(e) {
        if (Array.isArray(e)) return e;
    }
    function G(e, r) {
        if (null == e) return {};
        var t, n, o = U(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function U(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */    var K = {
        backgroundColor: l().oneOfType([ l().string, l().func ]),
        elementRef: l().oneOfType([ l().func, l().object ]),
        image: l().node,
        initials: l().string.isRequired,
        label: l().string.isRequired,
        onClick: l().func,
        size: l().oneOfType([ l().oneOf([ "small", "medium", "large" ]), l().string ]),
        to: l().string,
        /** @private */
        textColor: l().oneOfType([ l().oneOf([ "auto" ]), l().string, l().func ]),
        value: l().string
    };
    function W(e) {
        return !!e.onClick || !!e.to;
    }
    function F(e) {
        var r = e.backgroundColor, t = e.label, n = e.size, o = G(e, [ "backgroundColor", "label", "size" ]);
        var i = r == null ? u.variables.neutral200 : r;
        var a = {
            "aria-label": t,
            "data-test": "avatar",
            $backgroundColor: i,
            $size: n
        };
        return [ a, o ];
    }
    function J(e) {
        var r = F(e), o = M(r, 2), a = o[0], l = o[1];
        var u = l.children, c = l.elementRef, s = l.value, f = l.onClick, d = l.to, v = G(l, [ "children", "elementRef", "value", "onClick", "to" ]);
        var b = (0, t.useCallback)((function(e) {
            f === null || f === void 0 ? void 0 : f(e, {
                value: s
            });
        }), [ f, s ]);
        
        return n().createElement(y, H({
            elementRef: c,
            onClick: b,
            to: d
        }, a, i()(v, Object.keys(K))), u);
    }
    function Q(e) {
        var r = F(e), t = M(r, 2), o = t[0], a = t[1];
        var l = a.children, u = a.elementRef, c = G(a, [ "children", "elementRef" ]);
        
        return n().createElement(b, H({
            ref: u
        }, o, i()(c, Object.keys(K))), l);
    }
    function X(e) {
        var r = e.backgroundColor, o = e.textColor, i = o === void 0 ? "auto" : o, a = e.initials, l = e.image, c = e.size, s = c === void 0 ? "medium" : c, f = G(e, [ "backgroundColor", "textColor", "initials", "image", "size" ]);
        // @docs-props-type AvatarPropsBase
                var d = (0, u.useSplunkTheme)(), v = d.isLight;
        var b = i;
        if (i === "auto") {
            var y = T(v, r);
            b = y === P ? u.variables.black : u.variables.white;
        }
        // SVG attributes to improve rendering of text when initials are longer than 2 characters
                var g = a.length > 2 ? {
            textLength: "90%",
            lengthAdjust: "spacingAndGlyphs"
        } : {};
        var h =  n().createElement(p, {
            viewBox: "0 0 32 32"
        },  n().createElement(m, H({
            "data-test": "initials",
            x: "50%",
            y: "50%",
            dy: "-0.5",
            $textColor: b
        }, g), a));
        var O = l !== null && l !== void 0 ? l : h;
        var S = (0, t.useState)("loading"), w = M(S, 2), x = w[0], j = w[1];
        if ( (0, t.isValidElement)(l)) {
            if (l.type === "img") {
                var C = l.props, k = C.src, A = C.alt, E = C.onError, I = G(C, [ "src", "alt", "onError" ]);
                var z =  n().createElement("img", H({
                    src: k,
                    alt: A,
                    onError: function e() {
                        j("error");
                        if (E) {
                            E();
                        }
                    }
                }, I));
                O = z;
            }
        }
        // fallback to initials if img fails to load
                var R = x !== "error" ? O : h;
        var q = _({
            backgroundColor: r,
            size: s,
            initials: a,
            image: l,
            textColor: i
        }, f);
        return W(q) ?  n().createElement(J, q, R) :  n().createElement(Q, q, R);
    }
    X.propTypes = K;
    /* harmony default export */    const Y = X;
    // CONCATENATED MODULE: ./src/Avatar/index.ts
    module.exports = r;
    /******/})();
