// @splunk/react-ui/Badge.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var o in n) {
                /******/ if (e.o(n, o) && !e.o(r, o)) {
                    /******/ Object.defineProperty(r, o, {
                        enumerable: true,
                        get: n[o]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ m
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var o =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const t = require("prop-types");
    var l =  e.n(t);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const c = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/themes/mixins"
    const s = require("@splunk/themes/mixins");
    var u =  e.n(s);
    // CONCATENATED MODULE: ./src/Badge/BadgeStyles.ts
    var d = i().span.withConfig({
        displayName: "BadgeStyles__Styled",
        componentId: "sc-14wnqqw-0"
    })([ "", ";", ";gap:", ";align-items:center;justify-content:center;padding:", " ", ";width:min-content;border-radius:500px;", " ", "" ], u().reset("inline-flex"), u().typography("smallBody", {
        color: "active",
        weight: "semiBold"
    }), c.variables.spacingXSmall, c.variables.spacingXSmall, c.variables.spacingMedium, (function(e) {
        var r = e.$backgroundColor;
        return r ? (0, a.css)([ "background-color:", ";" ], r) : (0, a.css)([ "background-color:", ";" ], c.variables.neutral100);
    }), (function(e) {
        var r = e.$foregroundColor;
        return (0, a.css)([ "color:", ";" ], r);
    }));
    var f = i().div.withConfig({
        displayName: "BadgeStyles__StyledLabel",
        componentId: "sc-14wnqqw-1"
    })([ "", "" ], u().ellipsis());
    // CONCATENATED MODULE: ./src/Badge/Badge.tsx
    function p() {
        return p = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var o in n) {
                    ({}).hasOwnProperty.call(n, o) && (e[o] = n[o]);
                }
            }
            return e;
        }, p.apply(null, arguments);
    }
    function g(e, r) {
        if (null == e) return {};
        var n, o, t = b(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (o = 0; o < l.length; o++) {
                n = l[o], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function b(e, r) {
        if (null == e) return {};
        var n = {};
        for (var o in e) {
            if ({}.hasOwnProperty.call(e, o)) {
                if (-1 !== r.indexOf(o)) continue;
                n[o] = e[o];
            }
        }
        return n;
    }
    var y = {
        backgroundColor: l().oneOfType([ l().string, l().func ]),
        elementRef: l().oneOfType([ l().func, l().object ]),
        label: l().oneOfType([ l().string ]).isRequired,
        icon: l().node,
        foregroundColor: l().oneOfType([ l().string, l().func ])
    };
    function v(e) {
        var r = e.backgroundColor, n = e.elementRef, t = e.label, l = e.icon, a = e.foregroundColor, i = g(e, [ "backgroundColor", "elementRef", "label", "icon", "foregroundColor" ]);
        // @docs-props-type BadgePropsBase
                
        return o().createElement(d, p({
            $backgroundColor: r,
            $foregroundColor: a,
            "data-test": "badge",
            ref: n
        }, i), l,  o().createElement(f, null, t));
    }
    v.propTypes = y;
    /* harmony default export */    const m = v;
    // CONCATENATED MODULE: ./src/Badge/index.ts
    module.exports = r;
    /******/})();
