// @splunk/react-ui/Box.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ h
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Box/BoxStyles.ts
    var d = l().div.withConfig({
        displayName: "BoxStyles__Styled",
        componentId: "sc-1h4b5f6-0"
    })([ "", ";align-items:stretch;align-content:stretch;flex-flow:row nowrap;flex-grow:0;flex-shrink:0;justify-content:flex-start;width:auto;max-width:100%;&[data-inline]{display:inline-block;vertical-align:middle;}&[data-flex]{display:flex;}&[data-flex][data-inline]{display:inline-flex;}" ], f.mixins.reset("block"));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function s(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var c = "is deprecated and will be removed in the next major version.";
    var u = function e(n) {
        var t = n.additionalMessage, r = t === void 0 ? "" : t, a = n.componentName;
        useEffect((function() {
            if (false) {}
        }), [ r, a ]);
    };
    var p = function e(n) {
        var r = n.additionalMessage, a = r === void 0 ? "" : r, o = n.componentName, i = n.propName, l = n.propValue;
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ a, o, i, l ]);
    };
    var v = function e(n) {
        var t = n.additionalMessage, r = t === void 0 ? "" : t, a = n.componentName, o = n.deprecatedPropValue, i = n.propName, l = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ r, a, i, l, o ]);
    };
    // CONCATENATED MODULE: ./src/Box/Box.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function y(e, n) {
        if (null == e) return {};
        var t, r, a = b(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (r = 0; r < o.length; r++) {
                t = o[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function b(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var g = {
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        flex: o().bool,
        inline: o().bool
    };
    /**
 * Box is a utility component to normalize and abstract common layout styles.
 */    var x =  r().forwardRef((function(e, n) {
        var a = e.children, o = e.elementRef, i = e.flex, l = i === void 0 ? false : i, f = e.tag, c = f === void 0 ? "div" : f, u = e.inline, v = u === void 0 ? false : u, b = y(e, [ "children", "elementRef", "flex", "tag", "inline" ]);
        // @docs-props-type BoxPropsBase
                p({
            additionalMessage: "Use `ref` instead.",
            componentName: "Box",
            propName: "elementRef",
            propValue: o
        });
        var g = (0, t.useCallback)((function(e) {
            s(o, e);
            s(n, e);
        }), [ o, n ]);
        
        return r().createElement(d, m({
            as: c,
            ref: g
        }, b, {
            "data-inline": v || undefined,
            "data-flex": l || undefined
        }), a);
    }));
    x.propTypes = g;
    /* harmony default export */    const h = x;
    // CONCATENATED MODULE: ./src/Box/index.ts
    module.exports = n;
    /******/})();
