// @splunk/react-ui/Breadcrumbs.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        BreadcrumbsContext: () => /* reexport */ c,
        Item: () => /* reexport */ S,
        default: () => /* reexport */ E
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const o = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/Breadcrumbs/BreadcrumbsContext.tsx
    var i =  (0, n.createContext)({});
    i.displayName = "Breadcrumbs";
    /* harmony default export */    const c = i;
    // CONCATENATED MODULE: external "styled-components"
    const u = require("styled-components");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Breadcrumbs/BreadcrumbsStyles.ts
    var f = s().ol.withConfig({
        displayName: "BreadcrumbsStyles__StyledList",
        componentId: "sc-1maeyfk-0"
    })([ "", " flex-wrap:wrap;" ], d.mixins.reset("flex"));
    var p = s().li.withConfig({
        displayName: "BreadcrumbsStyles__StyledListItem",
        componentId: "sc-1maeyfk-1"
    })([ "display:flex;min-width:max-content;a[aria-current]{font-weight:", ";color:", ";cursor:default;}" ], d.variables.fontWeightSemiBold, d.variables.contentColorActive);
    var m = s().span.withConfig({
        displayName: "BreadcrumbsStyles__StyledSeparator",
        componentId: "sc-1maeyfk-2"
    })([ "cursor:default;padding:0 ", ";color:", ";" ], d.variables.spacingSmall, d.variables.contentColorDefault);
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const b = require("@splunk/react-ui/Link");
    var v =  e.n(b);
    // CONCATENATED MODULE: ./src/Breadcrumbs/ItemStyles.ts
    var y = s()(v()).withConfig({
        displayName: "ItemStyles__StyledLink",
        componentId: "sc-1ywtawf-0"
    })([ "display:flex;align-items:center;gap:", ";white-space:nowrap;color:", ";&:where(:hover,:focus):not([disabled],[aria-disabled='true']){color:", ";}" ], d.variables.spacingXSmall, d.variables.contentColorDefault, d.variables.contentColorActive);
    // CONCATENATED MODULE: ./src/Breadcrumbs/Item.tsx
    function g() {
        return g = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, g.apply(null, arguments);
    }
    function C(e, r) {
        if (null == e) return {};
        var n, t, a = O(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function O(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var h = {
        elementRef: l().oneOfType([ l().func, l().object ]),
        enableCurrentPage: l().bool,
        endAdornment: l().node,
        isCurrent: l().bool,
        label: l().string.isRequired,
        onClick: l().func,
        startAdornment: l().node,
        to: l().string.isRequired
    };
    function w(e) {
        var r = e.enableCurrentPage, a = e.endAdornment, l = e.isCurrent, o = e.label, i = e.onClick, u = e.startAdornment, s = e.to, d = C(e, [ "enableCurrentPage", "endAdornment", "isCurrent", "label", "onClick", "startAdornment", "to" ]);
        // @docs-props-type ItemPropsBase
                var f = (0, n.useContext)(c), p = f.onClick;
        var m = {
            to: s
        };
        var b = (0, n.useCallback)((function(e) {
            i === null || i === void 0 ? void 0 : i(e, {
                label: o,
                to: s
            });
            p === null || p === void 0 ? void 0 : p(e, {
                label: o,
                to: s
            });
        }), [ i, p, o, s ]);
        if (l) {
            m["aria-current"] = "page";
            if (!r) {
                m.disabled = true;
            }
        }
        var v = i || p ? {
            onClick: b
        } : {};
        
        return t().createElement(y, g({
            appearance: "standalone",
            "data-test": "item"
        }, v, m, d), u && u, o, a && a);
    }
    w.propTypes = h;
    /* harmony default export */    const S = w;
    // CONCATENATED MODULE: ./src/Breadcrumbs/Breadcrumbs.tsx
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, P.apply(null, arguments);
    }
    function k(e, r) {
        if (null == e) return {};
        var n, t, a = j(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function j(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */    var x = {
        children: l().node.isRequired,
        elementRef: l().oneOfType([ l().func, l().object ]),
        enableCurrentPage: l().bool,
        onClick: l().func
    };
    function _(e) {
        var r = e.children, a = e.elementRef, l = e.enableCurrentPage, i = e.onClick, u = k(e, [ "children", "elementRef", "enableCurrentPage", "onClick" ]);
        // @docs-props-type BreadcrumbsPropsBase
                var s;
        var d = n.Children.toArray(r).filter(n.isValidElement).map((function(e, a) {
            if (false) {}
            if (false) {}
            // Set the last child as the current Breadcrumbs.Item
                        var o = a === n.Children.count(r) - 1;
            var i =  (0, n.cloneElement)(e, {
                isCurrent: o,
                enableCurrentPage: l
            });
            
            return t().createElement(p, {
                key: e.props.to
            }, i, !o &&  t().createElement(m, {
                "aria-hidden": "true"
            }, "/"));
        }));
        var b = (0, n.useMemo)((function() {
            return {
                onClick: i
            };
        }), [ i ]);
        
        return t().createElement(c.Provider, {
            value: b
        },  t().createElement("nav", P({
            "data-test": "breadcrumbs",
            "aria-label": (0, o._)("Breadcrumbs"),
            ref: a
        }, u),  t().createElement(f, null, d)));
    }
    _.propTypes = x;
    _.Item = S;
    /* harmony default export */    const E = _;
    // CONCATENATED MODULE: ./src/Breadcrumbs/index.ts
    module.exports = r;
    /******/})();
