// @splunk/react-ui/Button.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ _
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "lodash/isString"
    const a = require("lodash/isString");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var o =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRightInset"
    const s = require("@splunk/react-icons/ArrowSquareTopRightInset");
    var u =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-icons/CaretSmallDown"
    const c = require("@splunk/react-icons/CaretSmallDown");
    var p =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const d = require("@splunk/react-ui/ScreenReaderContent");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const v = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/style"
    const b = require("@splunk/ui-utils/style");
    // CONCATENATED MODULE: external "styled-components"
    const m = require("styled-components");
    var y =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const g = require("@splunk/react-ui/ButtonSimple");
    var h =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/themes"
    const w = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Button/ButtonStyles.ts
    var x = y()(h()).withConfig({
        displayName: "ButtonStyles__StyledButtonSimple",
        componentId: "eqxqs2-0"
    })([ "vertical-align:middle;text-decoration:none;white-space:nowrap;flex-grow:1;min-width:", ";max-width:100%;min-height:", ";padding:2px ", ";line-height:", ";", " ", " &[data-inline]{display:inline-flex;}" ], w.variables.inputHeight, w.variables.inputHeight, w.variables.spacingSmall, w.variables.lineHeightSpacious, (function(e) {
        var n = e.$inline, t = e.append, r = e.prepend;
        return !n && !t && !r && (0, m.css)([ "width:100%;" ]);
    }), (function(e) {
        var n = e.$isIconOnly;
        return n && (0, m.css)([ "width:", ";" ], w.variables.inputHeight);
    }));
    var O = y().span.withConfig({
        displayName: "ButtonStyles__StyledContentWrapper",
        componentId: "eqxqs2-1"
    })([ "display:flex;flex-direction:row;align-items:center;justify-content:center;flex:1 0 auto;max-width:100%;gap:", ";box-sizing:border-box;" ], w.variables.spacingXSmall);
    var S = y().span.withConfig({
        displayName: "ButtonStyles__StyledLabel",
        componentId: "eqxqs2-2"
    })([ "text-overflow:ellipsis;overflow:hidden;flex:0 1 auto;", "" ], (function(e) {
        var n = e.$isMenu;
        return n && (0, m.css)([ "flex:1 1 auto;text-align:start;" ]);
    }));
    // CONCATENATED MODULE: ./src/Button/Button.tsx
    function q() {
        return q = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, q.apply(null, arguments);
    }
    function C(e, n) {
        if (null == e) return {};
        var t, r, a = N(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function N(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */    var k = {
        action: o().string,
        appearance: o().oneOf([ "default", "secondary", "primary", "destructive", "destructiveSecondary", "standalone", "subtle" ]),
        append: o().bool,
        children: o().node,
        /** @private An additional className to add to the button. */
        classNamePrivate: o().string,
        disabled: o().oneOfType([ o().bool, o().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: o().oneOfType([ o().func, o().object ]),
        /** @private */
        error: o().bool,
        icon: o().node,
        inline: o().bool,
        isMenu: o().bool,
        label: o().node,
        onClick: o().func,
        openInNewContext: o().oneOfType([ o().bool, o().string ]),
        prepend: o().bool,
        to: o().string,
        value: o().any
    };
    var I = (0, v._)("(Opens new window)");
    var P =  r().forwardRef((function(e, 
    // SUI-7623 - needs to be specified for react-docgen versions less than 7.1.0
    n) {
        var a = e.action, l = e.appearance, o = l === void 0 ? "default" : l, s = e.append, c = e.children, d = e.className, v = e.classNamePrivate, m = e.disabled, y = e.error, g = e.icon, h = e.inline, w = h === void 0 ? true : h, N = e.isMenu, k = e.label, P = e.onClick, _ = e.openInNewContext, j = e.prepend, M = e.value, E = C(e, [ "action", "appearance", "append", "children", "className", "classNamePrivate", "disabled", "error", "icon", "inline", "isMenu", "label", "onClick", "openInNewContext", "prepend", "value" ]);
        // @docs-props-type ButtonPropsBase
                var T = (0, t.useCallback)((function(e) {
            P === null || P === void 0 ? void 0 : P(e, {
                action: a,
                icon: g,
                label: k,
                value: M
            });
        }), [ a, g, k, P, M ]);
        var $ = !k && i()(c);
        var B = $ ? c : k;
        var H = false;
        var R;
        if (_) {
            H = true;
            R = typeof _ === "string" ? _ : I;
        }
        var z = !!(g && !k && !N && !c);
        
        return r().createElement(x, q({
            $inline: w,
            $isIconOnly: z,
            appearance: o,
            append: s,
            "aria-haspopup": N || undefined,
            className: (0, b.toClassName)(d, v),
            "data-action": a,
            "data-inline": w || null,
            "data-test": "button",
            disabled: m,
            error: y,
            isMenu: N,
            onClick: P ? T : undefined,
            openInNewContext: H,
            prepend: j,
            ref: n,
            value: M
        }, E),  r().createElement(O, null, g, B &&  r().createElement(S, {
            "data-test": "label",
            $isMenu: N
        }, B), !$ && c, N &&  r().createElement(p(), null), _ &&  r().createElement(r().Fragment, null,  r().createElement(u(), null),  r().createElement(f(), null, R))));
    }));
    P.propTypes = k;
    /* harmony default export */    const _ = P;
    // CONCATENATED MODULE: ./src/Button/index.ts
    module.exports = n;
    /******/})();
