// @splunk/react-ui/ButtonSimple.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var r = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ r.n = e => {
            /******/ var o = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            r.d(o, {
                a: o
            });
            /******/            return o;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ r.d = (e, o) => {
            /******/ for (var a in o) {
                /******/ if (r.o(o, a) && !r.o(e, a)) {
                    /******/ Object.defineProperty(e, a, {
                        enumerable: true,
                        get: o[a]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ r.o = (r, e) => Object.prototype.hasOwnProperty.call(r, e)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ r.r = r => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(r, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(r, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var e = {};
    // ESM COMPAT FLAG
        r.r(e);
    // EXPORTS
        r.d(e, {
        buttonMixin: () => /* reexport */ C,
        default: () => /* reexport */ $
    });
    // CONCATENATED MODULE: external "react"
    const o = require("react");
    var a =  r.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const t = require("prop-types");
    var n =  r.n(t);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var c =  r.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const l = require("@splunk/react-ui/Clickable");
    var s =  r.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: external "lodash/merge"
    const b = require("lodash/merge");
    var u =  r.n(b);
    // CONCATENATED MODULE: ./src/ButtonSimple/mixin.ts
    function v(r) {
        "@babel/helpers - typeof";
        return v = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(r) {
            return typeof r;
        } : function(r) {
            return r && "function" == typeof Symbol && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
        }, v(r);
    }
    var y = [ "primary", "secondary", "destructive", "destructiveSecondary", "subtle", "standalone" ];
    function p(r) {
        var e = "transparent";
        var o = "transparent";
        var a = "transparent";
        var t = "transparent";
        var n = "transparent";
        var i;
        var c;
        var l;
        var s = d.variables.actionColorContentSecondary;
        var b;
        var u = d.variables.actionColorContentSecondaryDisabled;
        var v;
        switch (r) {
          case "primary":
            {
                e = d.variables.actionColorBackgroundPrimary;
                o = d.variables.actionColorBackgroundPrimaryActive;
                a = d.variables.actionColorBackgroundPrimaryDisabled;
                t = d.variables.actionColorBackgroundPrimaryHover;
                s = d.variables.actionColorContentPrimary;
                u = d.variables.actionColorContentPrimaryDisabled;
                break;
            }

          case "destructive":
            {
                e = d.variables.actionColorBackgroundDestructive;
                o = d.variables.actionColorBackgroundDestructiveActive;
                a = d.variables.actionColorBackgroundDestructiveDisabled;
                t = d.variables.actionColorBackgroundDestructiveHover;
                s = d.variables.actionColorContentDestructive;
                u = d.variables.actionColorContentDestructiveDisabled;
                break;
            }

          case "destructiveSecondary":
            {
                n = d.variables.actionColorBorderDestructiveSecondary;
                i = d.variables.actionColorBorderDestructiveSecondaryActive;
                c = d.variables.actionColorBorderDestructiveSecondaryDisabled;
                l = d.variables.actionColorBorderDestructiveSecondaryHover;
                e = d.variables.actionColorBackgroundDestructiveSecondary;
                o = d.variables.actionColorBackgroundDestructiveSecondaryActive;
                a = d.variables.actionColorBackgroundDestructiveSecondaryDisabled;
                t = d.variables.actionColorBackgroundDestructiveSecondaryHover;
                s = d.variables.actionColorContentDestructiveSecondary;
                u = d.variables.actionColorContentDestructiveSecondaryDisabled;
                break;
            }

          case "secondary":
            n = d.variables.actionColorBorderSecondary;
            i = d.variables.actionColorBorderSecondaryActive;
            c = d.variables.actionColorBorderSecondaryDisabled;
            l = d.variables.actionColorBorderSecondaryHover;
            e = d.variables.actionColorBackgroundSecondary;
            o = d.variables.actionColorBackgroundSecondaryActive;
            t = d.variables.actionColorBackgroundSecondaryHover;
            break;

          case "subtle":
            o = d.variables.actionColorBackgroundSubtleActive;
            t = d.variables.actionColorBackgroundSubtleHover;
            b = d.variables.actionColorContentSubtleActive;
            v = d.variables.actionColorContentSubtleHover;
            break;

          case "standalone":
            e = d.variables.actionColorBackgroundSubtle;
            o = d.variables.actionColorBackgroundSubtleActive;
            t = d.variables.actionColorBackgroundSubtleHover;
            s = d.variables.actionColorContentStandalone;
            b = d.variables.actionColorContentStandaloneActive;
            u = d.variables.actionColorContentStandaloneDisabled;
            v = d.variables.actionColorContentStandaloneHover;
            break;

          default:
        }
        var y = {
            background: e,
            backgroundActive: o,
            backgroundDisabled: a,
            backgroundHover: t,
            borderColor: n,
            borderColorActive: i,
            borderColorDisabled: c,
            borderColorHover: l,
            color: s,
            colorActive: b,
            colorDisabled: u,
            colorHover: v
        };
        return y;
    }
    function f(r, e) {
        var o = typeof r === "string" && y.includes(r) ? r : undefined;
        var a = {};
        if (o && e !== undefined) {
            a = e;
        } else if (o === undefined && v(r) === "object" && e === undefined) {
            a = r;
        }
        var t = p(o);
        var n = u()(t, a), c = n.background, l = n.backgroundActive, s = n.backgroundDisabled, b = n.backgroundDisabledSelected, f = n.backgroundHover, C = n.borderColor, g = n.borderColorActive, S = n.borderColorDisabled, m = n.borderColorHover, k = n.color, D = n.colorActive, O = n.colorDisabled, B = n.colorHover;
        var h = c != null && c !== "transparent";
        return function() {
            return (0, i.css)([ "border:", " solid ", ";border-radius:", ";", " ", " cursor:pointer;position:relative;font-weight:", ";&:focus,&:hover{z-index:3;}&:focus{box-shadow:", ";}&:not([disabled],[aria-disabled='true']){background-color:", ";color:", ";transition:background-color 0.2s,box-shadow 0.2s,text-decoration 0.2s;&:hover{background-color:", ";border-color:", ";color:", ";}&:active{background-color:", ";border-color:", ";color:", ";transition:none;}}&[disabled],&[aria-disabled='true']{border-color:", ";color:", ";", "}" ], d.variables.inputBorderWidth, C, d.variables.borderRadius, (function(r) {
                var e = r.$append;
                return e && (0, i.css)([ "border-top-right-radius:0.1px;border-bottom-right-radius:0.1px;border-right:none;" ]);
            }), (function(r) {
                var e = r.$prepend;
                return e && (0, i.css)([ "border-top-left-radius:0.1px;border-bottom-left-radius:0.1px;" ]);
            }), d.variables.fontWeightSemiBold, d.variables.focusShadow, c, k, f, m, B, l, g, D, S, O, (function(r) {
                var e = r.$selected;
                return h && (0, i.css)([ "background-color:", ";" ], e ? b : s);
            }));
        };
    }
    /* harmony default export */    const C = f;
    // CONCATENATED MODULE: ./src/ButtonSimple/ButtonSimpleStyles.ts
    function g(r) {
        "@babel/helpers - typeof";
        return g = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(r) {
            return typeof r;
        } : function(r) {
            return r && "function" == typeof Symbol && r.constructor === Symbol && r !== Symbol.prototype ? "symbol" : typeof r;
        }, g(r);
    }
    function S(r, e) {
        var o = Object.keys(r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(r);
            e && (a = a.filter((function(e) {
                return Object.getOwnPropertyDescriptor(r, e).enumerable;
            }))), o.push.apply(o, a);
        }
        return o;
    }
    function m(r) {
        for (var e = 1; e < arguments.length; e++) {
            var o = null != arguments[e] ? arguments[e] : {};
            e % 2 ? S(Object(o), !0).forEach((function(e) {
                k(r, e, o[e]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(r, Object.getOwnPropertyDescriptors(o)) : S(Object(o)).forEach((function(e) {
                Object.defineProperty(r, e, Object.getOwnPropertyDescriptor(o, e));
            }));
        }
        return r;
    }
    function k(r, e, o) {
        return (e = D(e)) in r ? Object.defineProperty(r, e, {
            value: o,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : r[e] = o, r;
    }
    function D(r) {
        var e = O(r, "string");
        return "symbol" == g(e) ? e : e + "";
    }
    function O(r, e) {
        if ("object" != g(r) || !r) return r;
        var o = r[Symbol.toPrimitive];
        if (void 0 !== o) {
            var a = o.call(r, e || "default");
            if ("object" != g(a)) return a;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === e ? String : Number)(r);
    }
    var B = {
        borderColor: d.variables.interactiveColorAccentError,
        borderColorActive: d.variables.interactiveColorAccentError,
        borderColorHover: d.mixins.overlayColors(d.variables.interactiveColorAccentError, d.variables.interactiveColorBorderHover)
    };
    /**
 * When `isMenu` is true, the button should use interactive tokens (e.g. Select) rather than action tokens (e.g. Button).
 */    var h = {
        color: d.variables.contentColorActive,
        colorDisabled: d.variables.contentColorDisabled,
        borderColor: d.variables.interactiveColorBorder,
        borderColorActive: d.variables.interactiveColorBorderActive,
        borderColorHover: d.variables.interactiveColorBorderHover,
        borderColorDisabled: d.variables.interactiveColorBorderDisabled
    };
    var j = c()(s()).withConfig({
        displayName: "ButtonSimpleStyles__StyledClickable",
        componentId: "vlarwe-0"
    })([ "", " align-items:center;justify-content:center;&:not([disabled],[aria-disabled='true']){", "}", "" ], d.mixins.reset("flex"), (function(r) {
        var e = r.$error, o = r.$isMenu;
        return e && (0, i.css)([ "", "" ], (0, d.pickVariant)("$variant", {
            default: o ? C("secondary", m(m({}, h), B)) : C("secondary", B),
            secondary: o ? C("secondary", m(m({}, h), B)) : C("secondary", B),
            subtle: C("subtle", B)
        }));
    }), (function(r) {
        var e = r.$isMenu;
        return (0, d.pickVariant)("$variant", {
            default: e ? C("secondary", h) : C("secondary"),
            primary: C("primary"),
            secondary: e ? C("secondary", h) : C("secondary"),
            subtle: C("subtle"),
            destructive: C("destructive"),
            destructiveSecondary: C("destructiveSecondary"),
            standalone: C("standalone")
        });
    }));
    // CONCATENATED MODULE: ./src/ButtonSimple/ButtonSimple.tsx
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(r) {
            for (var e = 1; e < arguments.length; e++) {
                var o = arguments[e];
                for (var a in o) {
                    ({}).hasOwnProperty.call(o, a) && (r[a] = o[a]);
                }
            }
            return r;
        }, P.apply(null, arguments);
    }
    function w(r, e) {
        if (null == r) return {};
        var o, a, t = A(r, e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(r);
            for (a = 0; a < n.length; a++) {
                o = n[a], -1 === e.indexOf(o) && {}.propertyIsEnumerable.call(r, o) && (t[o] = r[o]);
            }
        }
        return t;
    }
    function A(r, e) {
        if (null == r) return {};
        var o = {};
        for (var a in r) {
            if ({}.hasOwnProperty.call(r, a)) {
                if (-1 !== e.indexOf(a)) continue;
                o[a] = r[a];
            }
        }
        return o;
    }
    var H = {
        action: n().string,
        appearance: n().oneOf([ "default", "secondary", "primary", "destructive", "destructiveSecondary", "subtle", "standalone" ]),
        append: n().bool,
        children: n().node,
        disabled: n().oneOfType([ n().bool, n().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: n().oneOfType([ n().func, n().object ]),
        error: n().bool,
        isMenu: n().bool,
        openInNewContext: n().bool,
        prepend: n().bool,
        to: n().string
    };
    var x =  a().forwardRef((function(r, e) {
        var o = r.appearance, t = o === void 0 ? "default" : o, n = r.append, i = r.children, c = r.disabled, l = r.error, s = r.isMenu, d = r.prepend, b = w(r, [ "appearance", "append", "children", "disabled", "error", "isMenu", "prepend" ]);
        // @docs-props-type ButtonSimplePropsBase
                
        return a().createElement(j, P({
            "aria-invalid": l,
            "data-test": "button-simple",
            disabled: c,
            $variant: t,
            $append: n,
            $prepend: d,
            $error: l,
            $isMenu: s,
            ref: e
        }, b), i);
    }));
    x.propTypes = H;
    /* harmony default export */    const $ = x;
    // only for styled-components that wrap ButtonSimple
        // CONCATENATED MODULE: ./src/ButtonSimple/index.ts
    module.exports = e;
    /******/})();
