// @splunk/react-ui/Calendar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = a => {
            /******/ var r = a && a.__esModule ? 
            /******/ () => a["default"]
            /******/ : () => a
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (a, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(a, t)) {
                    /******/ Object.defineProperty(a, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, a) => Object.prototype.hasOwnProperty.call(e, a)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var a = {};
    // ESM COMPAT FLAG
        e.r(a);
    // EXPORTS
        e.d(a, {
        calendarActiveSelector: () => /* reexport */ ce,
        default: () => /* reexport */ Ye,
        focusActiveDay: () => /* reexport */ fe
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "moment"
    const n = require("moment");
    var l =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const u = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Calendar/CalendarStyles.ts
    var s = d().div.withConfig({
        displayName: "CalendarStyles__Styled",
        componentId: "sc-18925wt-0"
    })([ "", ";grid-row-gap:", ";padding:", ";" ], v.mixins.reset("grid"), v.variables.spacingXSmall, v.variables.spacingLarge);
    // CONCATENATED MODULE: external "lodash/times"
    const f = require("lodash/times");
    var y =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const Y = require("@splunk/react-ui/ScreenReaderContent");
    var p =  e.n(Y);
    // CONCATENATED MODULE: ./src/Calendar/DateGridStyles.ts
    var g = d().div.withConfig({
        displayName: "DateGridStyles__Styled",
        componentId: "sc-7cb7ui-0"
    })([ "", ";grid-template-columns:repeat(7,min-content);grid-row-gap:", ";" ], v.mixins.reset("grid"), v.variables.spacingXSmall);
    var b = d().div.withConfig({
        displayName: "DateGridStyles__StyledWeekday",
        componentId: "sc-7cb7ui-1"
    })([ "", ";text-align:center;" ], v.mixins.typography("smallBody"));
    // CONCATENATED MODULE: ./src/Calendar/DayStyles.ts
    var h = d().div.withConfig({
        displayName: "DayStyles__Styled",
        componentId: "sc-1nryhyt-0"
    })([ "", " width:", ";height:", ";align-items:center;justify-content:center;border:1px solid transparent;border-radius:", ";color:", ";cursor:pointer;", " ", " &:focus{box-shadow:", ";font-weight:", ";outline:none;z-index:0;}&:hover{background-color:", ";color:", ";}", "" ], v.mixins.reset("flex"), v.variables.inputHeight, v.variables.inputHeight, v.variables.borderRadius, v.variables.contentColorActive, (function(e) {
        var a = e.$dayOfWeek;
        // We need the + 1 here since the dayOfWeek goes from 0-6 (Sun-Sat) but grid-column values go from 1-7.
        return (0, c.css)([ "grid-column:calc(", " + 1);" ], a);
    }), (function(e) {
        var a = e.$today;
        return a && (0, c.css)([ "border:1px solid ", ";" ], v.variables.interactiveColorBorder);
    }), v.variables.focusShadow, v.variables.fontWeightBold, v.variables.interactiveColorOverlayHover, v.variables.contentColorAccent, (function(e) {
        var a = e.$selected;
        return a && (0, c.css)([ "border:1px solid ", ";font-weight:", ";background-color:", ";&:hover{background-color:", ";}" ], v.variables.contentColorAccent, v.variables.fontWeightBold, v.variables.interactiveColorOverlaySelected, v.mixins.overlayColors(v.variables.interactiveColorOverlaySelected, v.variables.interactiveColorOverlayHover));
    }));
    // CONCATENATED MODULE: ./src/Calendar/Day.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function D(e, a) {
        if (null == e) return {};
        var r, t, n = M(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], -1 === a.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function M(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== a.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var O = {
        active: i().bool,
        /** @private. */
        dayOfWeek: i().number,
        locale: i().string,
        onClick: i().func,
        onKeydown: i().func,
        selected: i().bool,
        today: i().bool,
        value: i().string
    };
    function w(e) {
        var a = e.active, n = a === void 0 ? false : a, o = e.dayOfWeek, i = e.locale, u = i === void 0 ? "en_US" : i, c = e.onClick, d = e.onKeyDown, v = e.selected, s = v === void 0 ? false : v, f = e.today, y = f === void 0 ? false : f, Y = e.value, p = D(e, [ "active", "dayOfWeek", "locale", "onClick", "onKeyDown", "selected", "today", "value" ]);
        // @docs-props-type DayPropsBase
                var g = (0, r.useCallback)((function(e) {
            e.preventDefault();
            c === null || c === void 0 ? void 0 : c(e, {
                value: Y
            });
        }), [ c, Y ]);
        var b = (0, r.useCallback)((function(e) {
            e.preventDefault();
            d === null || d === void 0 ? void 0 : d(e, {
                value: Y
            });
        }), [ d, Y ]);
        var M = Y && l()(Y, "YYYY-MM-DD", u).format("D");
        
        return t().createElement(h, m({
            tabIndex: n ? 0 : -1,
            "aria-selected": s,
            "data-test-day": M,
            "data-test": "day-of-month",
            "data-active-day": n,
            "data-is-today": y,
            onClick: g,
            onKeyDown: b,
            $dayOfWeek: o,
            $today: y,
            $selected: s
        }, p), M);
    }
    w.propTypes = O;
    /* harmony default export */    const C = w;
    // CONCATENATED MODULE: ./src/Calendar/DateGrid.tsx
    function k() {
        return k = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, k.apply(null, arguments);
    }
    function S(e, a) {
        if (null == e) return {};
        var r, t, n = x(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], -1 === a.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function x(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== a.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var I = {
        displayValue: i().string.isRequired,
        highlightToday: i().bool,
        labelledBy: i().string,
        locale: i().string,
        onChange: i().func,
        onInternalChange: i().func,
        selectedValue: i().string.isRequired,
        todayValue: i().string
    };
    function E(e) {
        var a = e.displayValue, r = e.endDate, n = e.highlightToday, o = e.locale, i = e.onChange, u = e.onKeyDown, c = e.selectedDate, d = e.selectedValue, v = e.startDate, s = e.todayValue;
        // @docs-props-type DateGridPropsBase
                var f = l()(v).locale(o).startOf("month");
        var y = [];
        while (f.isSameOrBefore(r)) {
            var Y = f.format("YYYY-MM-DD");
            var p = Y === s;
            var g = Y === d;
            var b = Y === a;
            var h = Number(f.format("e"));
            y.push( t().createElement(C, {
                key: Y,
                locale: o,
                onClick: i,
                onKeyDown: u,
                active: c ? b : p,
                today: n && p,
                value: Y,
                selected: g,
                dayOfWeek: h
            }));
            f.add(1, "day");
        }
        return y;
    }
    function j(e, a) {
        return y()(7, (function(r) {
            var n = l()().locale(e).weekday(r).format(a);
            var o = l()().locale(e).weekday(r).format("dddd");
            
            return t().createElement(b, {
                key: n
            },  t().createElement(p(), null, o),  t().createElement("span", {
                "aria-hidden": true
            }, n));
        }));
    }
    function _(e) {
        var a = e.displayValue, r = e.highlightToday, n = r === void 0 ? false : r, o = e.labelledBy, i = e.locale, u = i === void 0 ? "en_US" : i, c = e.onChange, d = e.onInternalChange, v = e.selectedValue, s = e.todayValue, f = S(e, [ "displayValue", "highlightToday", "labelledBy", "locale", "onChange", "onInternalChange", "selectedValue", "todayValue" ]);
        var y = "dd";
        var Y = l()(a, "YYYY-MM-DD", u);
        var p = l()(v, "YYYY-MM-DD", u);
        var b = l()(Y).startOf("month");
        var h = l()(Y).endOf("month").startOf("day");
        var m = function e(a, r) {
            a.preventDefault();
            c === null || c === void 0 ? void 0 : c(a, r);
        };
        var D = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(-1, "d").format("YYYY-MM-DD")
            });
        };
        var M = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(1, "d").format("YYYY-MM-DD")
            });
        };
        var O = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(-1, "w").format("YYYY-MM-DD")
            });
        };
        var w = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(1, "w").format("YYYY-MM-DD")
            });
        };
        var C = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(-1, "M").format("YYYY-MM-DD")
            });
        };
        var x = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(1, "M").format("YYYY-MM-DD")
            });
        };
        var I = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(-1, "y").format("YYYY-MM-DD")
            });
        };
        var _ = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").add(1, "y").format("YYYY-MM-DD")
            });
        };
        var P = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").startOf("week").format("YYYY-MM-DD")
            });
        };
        var q = function e(a, r) {
            var t = r.value;
            a.preventDefault();
            d === null || d === void 0 ? void 0 : d(a, {
                value: l()(t, "YYYY-MM-DD").endOf("week").format("YYYY-MM-DD")
            });
        };
        var V = function e(a, r) {
            switch (a.key) {
              case "Enter":
              case "Space":
                m(a, r);
                break;

              case "ArrowUp":
                O(a, r);
                break;

              case "ArrowDown":
                w(a, r);
                break;

              case "ArrowLeft":
                D(a, r);
                break;

              case "ArrowRight":
                M(a, r);
                break;

              case "PageUp":
                if (a.shiftKey) {
                    I(a, r);
                } else {
                    C(a, r);
                }
                break;

              case "PageDown":
                if (a.shiftKey) {
                    _(a, r);
                } else {
                    x(a, r);
                }
                break;

              case "Home":
                P(a, r);
                break;

              case "End":
                q(a, r);
                break;

              default:
 // do nothing
                        }
        };
        
        return t().createElement(g, k({
            role: "grid",
            "aria-labelledby": o
        }, f), j(u, y), E({
            displayValue: a,
            endDate: h,
            highlightToday: n,
            locale: u,
            onChange: c,
            onKeyDown: V,
            selectedDate: p,
            selectedValue: v,
            startDate: b,
            todayValue: s
        }));
    }
    _.propTypes = I;
    /* harmony default export */    const P = _;
    // CONCATENATED MODULE: external "lodash/omit"
    const q = require("lodash/omit");
    var V =  e.n(q);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronLeft"
    const B = require("@splunk/react-icons/ChevronLeft");
    var T =  e.n(B);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const A = require("@splunk/react-icons/ChevronRight");
    var R =  e.n(A);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const H = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const K = require("@splunk/react-ui/Button");
    var N =  e.n(K);
    // CONCATENATED MODULE: ./src/Calendar/MonthHeaderStyles.ts
    var W = d().div.withConfig({
        displayName: "MonthHeaderStyles__Styled",
        componentId: "sc-18bmbh7-0"
    })([ "", ";grid-template-columns:min-content 1fr min-content;align-items:center;" ], v.mixins.reset("grid"));
    var U = d().span.withConfig({
        displayName: "MonthHeaderStyles__StyledHeading",
        componentId: "sc-18bmbh7-1"
    })([ "", " text-align:center;" ], v.mixins.typography("title4"));
    var $ = d()(N()).withConfig({
        displayName: "MonthHeaderStyles__StyledButton",
        componentId: "sc-18bmbh7-2"
    })([ "border:none;" ]);
    // CONCATENATED MODULE: ./src/Calendar/MonthHeader.tsx
    function L(e, a) {
        if (null == e) return {};
        var r, t, n = z(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], -1 === a.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function z(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== a.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var G = {
        headingId: i().string,
        locale: i().string,
        onChange: i().func,
        value: i().string.isRequired
    };
    var X = {
        default: "MMM YYYY",
        ja: "YYYY年 MMM",
        ko: "YYYY년 MMMM",
        "zh-cn": "YYYY年 MMM",
        "zh-tw": "YYYY年 MMM"
    };
    function F(e) {
        var a = e.toLowerCase().replace("_", "-");
        return X[a] || 
        // full locale match
        X[a.substr(0, 2)] || 
        // language match
        X["default"];
    }
    function J(e) {
        var a = e.headingId, n = e.locale, o = n === void 0 ? "en_US" : n, i = e.onChange, u = e.value, c = L(e, [ "headingId", "locale", "onChange", "value" ]);
        // @docs-props-type MonthHeaderPropsBase
                var d = (0, r.useCallback)((function(e, a) {
            e.stopPropagation();
            var r = l()(u, "YYYY-MM-DD").add(a, "M").format("YYYY-MM-DD");
            i === null || i === void 0 ? void 0 : i(e, {
                value: r
            });
        }), [ i, u ]);
        var v = (0, r.useCallback)((function(e) {
            d(e, -1);
        }), [ d ]);
        var s = (0, r.useCallback)((function(e) {
            d(e, 1);
        }), [ d ]);
        var f = l()(u, "YYYY-MM-DD", o);
        var y = f.format(F(o));
        var Y = (0, H._)("Previous month");
        var p = (0, H._)("Next month");
        
        return t().createElement(W, V()(c, "onChange"),  t().createElement($, {
            appearance: "secondary",
            icon:  t().createElement(T(), {
                "aria-label": Y
            }),
            "data-test": "previous-month",
            onClick: v
        }),  t().createElement(U, {
            id: a,
            "data-test": "header-label"
        }, y),  t().createElement($, {
            appearance: "secondary",
            icon:  t().createElement(R(), {
                "aria-label": p
            }),
            "data-test": "next-month",
            onClick: s
        }));
    }
    J.propTypes = G;
    /* harmony default export */    const Q = J;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function Z(e, a) {
        if (e) {
            if (typeof e === "function") {
                e(a);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = a;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Calendar/Calendar.tsx
    function ee() {
        return ee = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var r = arguments[a];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, ee.apply(null, arguments);
    }
    function ae(e, a) {
        return oe(e) || le(e, a) || te(e, a) || re();
    }
    function re() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function te(e, a) {
        if (e) {
            if ("string" == typeof e) return ne(e, a);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? ne(e, a) : void 0;
        }
    }
    function ne(e, a) {
        (null == a || a > e.length) && (a = e.length);
        for (var r = 0, t = Array(a); r < a; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function le(e, a) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, n, l, o, i = [], u = !0, c = !1;
            try {
                if (l = (r = r.call(e)).next, 0 === a) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = l.call(r)).done) && (i.push(t.value), i.length !== a); u = !0) {
                }
            } catch (e) {
                c = !0, n = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (o = r["return"](), Object(o) !== o)) return;
                } finally {
                    if (c) throw n;
                }
            }
            return i;
        }
    }
    function oe(e) {
        if (Array.isArray(e)) return e;
    }
    function ie(e, a) {
        if (null == e) return {};
        var r, t, n = ue(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], -1 === a.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (n[r] = e[r]);
            }
        }
        return n;
    }
    function ue(e, a) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== a.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var ce = '[data-active-day="true"]';
    var de = {
        describedBy: i().string,
        highlightToday: i().bool,
        labelledBy: i().string,
        locale: i().string,
        mockDate: i().string,
        onChange: i().func,
        /** @private. */
        required: i().bool,
        value: i().string,
        elementRef: i().oneOfType([ i().func, i().object ])
    };
    var ve = function e(a) {
        return l()().locale(a).format("YYYY-MM-DD");
    };
    var se = function e(a) {
        if (!moment(a, "YYYY-MM-DD").isValid()) {
            return new Error("Invalid value prop ".concat(a, " passed to Calendar. value must be a string date formatted 'YYYY-MM-DD'."));
        }
        return null;
    };
    /**
 * Place focus on the active day.
 */    var fe = function e(a) {
        var r = a === null || a === void 0 ? void 0 : a.querySelector(ce);
        r === null || r === void 0 ? void 0 : r.focus();
    };
    function ye(e) {
        var a = e.describedBy, n = e.elementRef, o = e.highlightToday, i = e.labelledBy, c = e.locale, d = c === void 0 ? "en_US" : c, v = e.mockDate, f = e.onChange, y = e.required, Y = e.value, p = Y === void 0 ? l()().format("YYYY-MM-DD") : Y, g = ie(e, [ "describedBy", "elementRef", "highlightToday", "labelledBy", "locale", "mockDate", "onChange", "required", "value" ]);
        // @docs-props-type CalendarPropsBase
                var b = (0, r.useState)(p), h = ae(b, 2), m = h[0], D = h[1];
        var M = (0, r.useState)(v || ve(d)), O = ae(M, 2), w = O[0], C = O[1];
        var k = (0, r.useState)(null), S = ae(k, 2), x = S[0], I = S[1];
        var E = (0, r.useRef)((0, u.createDOMID)("header"));
        var j = (0, r.useRef)();
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ p ]);
        (0, r.useEffect)((function() {
            if (!v) {
                var e = setInterval((function() {
                    var e = ve(d);
                    if (e !== w) {
                        C(e);
                    }
                }), 1e3);
                return function() {
                    if (e !== undefined) {
                        clearInterval(e);
                    }
                };
            }
            return undefined;
        }), [ d, v, w ]);
        (0, r.useEffect)((function() {
            if (x === "day") {
                fe(j.current);
            }
        }), [ m, x ]);
        var _ = (0, r.useCallback)((function(e, a) {
            var r = a.value;
            D(r);
            I("month");
        }), []);
        var q = (0, r.useCallback)((function(e, a) {
            var r = a.value;
            D(r);
            I("day");
        }), []);
        var V = (0, r.useCallback)((function(e) {
            Z(j, e);
            Z(n, e);
        }), [ n ]);
        
        return t().createElement(s, ee({
            "aria-labelledby": i,
            "aria-describedby": a,
            "aria-required": y,
            "data-test": "calendar",
            ref: V
        }, g),  t().createElement(Q, {
            headingId: E.current,
            locale: d,
            onChange: _,
            value: m
        }),  t().createElement(P, {
            displayValue: m,
            highlightToday: o,
            labelledBy: E.current,
            locale: d,
            onChange: f,
            onInternalChange: q,
            selectedValue: p,
            todayValue: w
        }));
    }
    ye.propTypes = de;
    /* harmony default export */    const Ye = ye;
    // CONCATENATED MODULE: ./src/Calendar/index.ts
    module.exports = a;
    /******/})();
