// @splunk/react-ui/CardLayout.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ j
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var c =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/CardLayout/CardLayoutStyles.ts
    var l = c().div.withConfig({
        displayName: "CardLayoutStyles__Styled",
        componentId: "sc-18kz7zm-0"
    })([ "", ";", " ", "" ], u.mixins.reset("flex"), (function(e) {
        var r = e.$wrapCards;
        return r && (0, a.css)([ "flex-wrap:wrap;" ]);
    }), (0, u.pickVariant)("$alignCards", {
        center: (0, a.css)([ "justify-content:center;" ]),
        right: (0, a.css)([ "justify-content:flex-end;" ])
    }));
    // CONCATENATED MODULE: ./src/CardLayout/CardLayout.tsx
    function f(e) {
        "@babel/helpers - typeof";
        return f = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, f(e);
    }
    function s() {
        return s = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, s.apply(null, arguments);
    }
    function d(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function p(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? d(Object(t), !0).forEach((function(r) {
                y(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : d(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function y(e, r, t) {
        return (r = b(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function b(e) {
        var r = m(e, "string");
        return "symbol" == f(r) ? r : r + "";
    }
    function m(e, r) {
        if ("object" != f(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != f(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function v(e, r) {
        if (null == e) return {};
        var t, n, o = g(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function g(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var O = {
        alignCards: i().oneOf([ "left", "center", "right" ]),
        cardMaxWidth: i().oneOfType([ i().number, i().string ]),
        cardMinWidth: i().oneOfType([ i().number, i().string ]),
        cardWidth: i().oneOfType([ i().number, i().string ]),
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        gutterSize: i().number,
        wrapCards: i().bool
    };
    function h(e) {
        var r = e.alignCards, o = r === void 0 ? "left" : r, i = e.cardMaxWidth, a = e.cardMinWidth, c = e.cardWidth, u = e.children, f = e.elementRef, d = e.gutterSize, y = d === void 0 ? 10 : d, b = e.style, m = e.wrapCards, g = m === void 0 ? true : m, O = v(e, [ "alignCards", "cardMaxWidth", "cardMinWidth", "cardWidth", "children", "elementRef", "gutterSize", "style", "wrapCards" ]);
 // @docs-props-type CardLayoutPropsBase
                var h = t.Children.toArray(u).filter(t.isValidElement).map((function(e) {
            
            return (0, t.cloneElement)(e, {
                minWidth: c || a || undefined,
                maxWidth: c || i || undefined,
                margin: y / 2
            });
        }));
        var j = (0, t.useMemo)((function() {
            return p({
                padding: y / 2
            }, b);
        }), [ y, b ]);
        
        return n().createElement(l, s({
            "data-test": "card-layout",
            $alignCards: o,
            $wrapCards: g,
            ref: f,
            style: j
        }, O), h);
    }
    h.propTypes = O;
    /* harmony default export */    const j = h;
    // CONCATENATED MODULE: ./src/CardLayout/index.ts
    module.exports = r;
    /******/})();
