// @splunk/react-ui/Card.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Body: () => /* reexport */ j,
        Footer: () => /* reexport */ W,
        Header: () => /* reexport */ le,
        default: () => /* reexport */ _e
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const c = require("@splunk/react-ui/Box");
    var s =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Card/HeaderStyles.ts
    // The padding here is for the spacing between the Card and its subcomponents.
    // We apply this to the subcomponents rather than directly to the Card to allow Body's inset prop
    // to remove spacing around it when false and we cannot do that if the padding is on its parent.
    var d = (0, i.css)([ "padding:0 ", ";&:first-child{padding-top:", ";}&:last-child{padding-bottom:", ";}" ], u.variables.spacingXLarge, u.variables.spacingLarge, u.variables.spacingLarge);
    var f = l().div.withConfig({
        displayName: "HeaderStyles__StyledTitle",
        componentId: "eqi6uk-0"
    })([ "", " ", " overflow-wrap:break-word;" ], u.mixins.reset("block"), u.mixins.typography("title3"));
    var p = l().div.withConfig({
        displayName: "HeaderStyles__StyledSubtitle",
        componentId: "eqi6uk-1"
    })([ "", " ", " overflow-wrap:break-word;" ], u.mixins.reset("block"), u.mixins.typography("body", {
        color: "active"
    }));
    var v = l().div.withConfig({
        displayName: "HeaderStyles__StyledTitleContainer",
        componentId: "eqi6uk-2"
    })([ "", " flex:0 1 auto;align-self:center;margin-right:auto;", "" ], u.mixins.reset("block"), (function(e) {
        var t = e.$truncateTitle;
        return t && (0, i.css)([ "overflow:hidden;& > ", ",& > ", ",& > ", " *{overflow:hidden;text-overflow:ellipsis;white-space:nowrap;}" ], /* sc-sel */ f, /* sc-sel */ p, /* sc-sel */ f);
    }));
    var m = l().div.withConfig({
        displayName: "HeaderStyles__StyledActionContainer",
        componentId: "eqi6uk-3"
    })([ "", " align-items:center;flex:0 0 auto;justify-content:space-between;gap:", ";transform:translateY(calc(-1 * (", " - 20px) / 2));" ], u.mixins.reset("flex"), u.variables.spacingSmall, u.variables.inputHeight);
    var y = l().div.withConfig({
        displayName: "HeaderStyles__StyledIconContainer",
        componentId: "eqi6uk-4"
    })([ "", " align-items:center;flex:0 0 auto;" ], u.mixins.reset("flex"));
    var b = l()(s()).withConfig({
        displayName: "HeaderStyles__StyledBox",
        componentId: "eqi6uk-5"
    })([ "", " align-items:start;display:flex;justify-content:space-between;position:relative;min-height:30px;gap:", ";>:not(:is( ", ",", ",", " ) ){flex:1 0 auto;}" ], d, u.variables.spacingSmall, /* sc-sel */ v, /* sc-sel */ m, /* sc-sel */ y);
    // CONCATENATED MODULE: ./src/Card/BodyStyles.ts
    var g = l()(s()).withConfig({
        displayName: "BodyStyles__StyledBox",
        componentId: "sc-1130udp-0"
    })([ "", ";overflow:auto;height:100%;" ], (function(e) {
        var t = e.$inset;
        return t && (0, i.css)([ "", "" ], d);
    }));
    // CONCATENATED MODULE: ./src/Card/Body.tsx
    function h() {
        return h = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, h.apply(null, arguments);
    }
    function O(e, t) {
        if (null == e) return {};
        var r, n, o = w(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function w(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var S = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        inset: a().bool
    };
    /**
 * A styled container for `Card` body content.
 */    function x(e) {
        var t = e.children, r = e.inset, o = r === void 0 ? true : r, a = O(e, [ "children", "inset" ]);
        // @docs-props-type BodyPropsBase
                
        return n().createElement(g, h({
            "data-test": "body",
            $inset: o
        }, a), t);
    }
    x.propTypes = S;
    /* harmony default export */    const j = x;
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const k = require("@splunk/react-ui/Clickable");
    var C =  e.n(k);
    // CONCATENATED MODULE: ./src/Card/CardStyles.ts
    var P = "0.2s";
    var E = (0, i.css)([ "", ";gap:", " 0;grid-template-rows:min-content;background-color:", ";min-width:100px;flex:1;vertical-align:top;border:1px solid ", ";border-radius:", ";box-shadow:", ";transition:height ", ",width ", ",min-width ", ",max-width ", ",margin ", ",box-shadow ", ",border-color ", ";&[data-clickable='true']{cursor:pointer;border-color:", ";", " &:hover{background-color:", ";box-shadow:", ";", "{text-decoration:underline;}}&:focus{box-shadow:", ";", "{text-decoration:underline;}}}" ], u.mixins.reset("inline-grid"), u.variables.spacingLarge, u.variables.backgroundColorSection, u.variables.transparent, u.variables.borderRadius, u.variables.embossShadow, P, P, P, P, P, P, P, u.variables.borderColorWeak, (function(e) {
        var t = e.$selected;
        return t && (0, i.css)([ "background-color:", ";" ], u.mixins.overlayColors(u.variables.backgroundColorSection, u.variables.interactiveColorOverlaySelected));
    }), u.mixins.overlayColors(u.variables.backgroundColorSection, u.variables.interactiveColorOverlayHover), u.variables.overlayShadow, /* sc-sel */ f, u.variables.focusShadow, /* sc-sel */ f);
    var _ = l().article.withConfig({
        displayName: "CardStyles__Styled",
        componentId: "ola3x0-0"
    })([ "", "" ], E);
    var N = l()(C()).withConfig({
        displayName: "CardStyles__StyledClickable",
        componentId: "ola3x0-1"
    })([ "", "" ], E);
    // CONCATENATED MODULE: ./src/Card/FooterStyles.ts
    var T = l()(s()).withConfig({
        displayName: "FooterStyles__StyledBox",
        componentId: "sc-1yu3r4s-0"
    })([ "", " ", " text-align:right;align-self:end;" ], u.mixins.typography("body"), d);
    // CONCATENATED MODULE: ./src/Card/Footer.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function q(e, t) {
        if (null == e) return {};
        var r, n, o = R(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function R(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var H = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ])
    };
    /**
 * A styled container for `Card` footer content.
 */    function A(e) {
        var t = e.children, r = q(e, [ "children" ]);
        // @docs-props-type FooterPropsBase
                
        return n().createElement(T, I({
            "data-test": "footer"
        }, r), t);
    }
    A.propTypes = H;
    /* harmony default export */    const W = A;
    // CONCATENATED MODULE: external "@splunk/react-icons/DotsThreeVertical"
    const B = require("@splunk/react-icons/DotsThreeVertical");
    var M =  e.n(B);
    // CONCATENATED MODULE: external "@splunk/react-ui/Anchor"
    const $ = require("@splunk/react-ui/Anchor");
    var V =  e.n($);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const D = require("@splunk/react-ui/Button");
    var F =  e.n(D);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const L = require("@splunk/react-ui/Dropdown");
    var U =  e.n(L);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const X = require("@splunk/react-ui/Tooltip");
    var Y =  e.n(X);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const z = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var G = "is deprecated and will be removed in the next major version.";
    var J = function e(t) {
        var r = t.additionalMessage, n = r === void 0 ? "" : r, o = t.componentName;
        useEffect((function() {
            if (false) {}
        }), [ n, o ]);
    };
    var K = function e(t) {
        var n = t.additionalMessage, o = n === void 0 ? "" : n, a = t.componentName, i = t.propName, l = t.propValue;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ o, a, i, l ]);
    };
    var Q = function e(t) {
        var r = t.additionalMessage, n = r === void 0 ? "" : r, o = t.componentName, a = t.deprecatedPropValue, i = t.propName, l = t.propValue;
        useEffect((function() {
            if (false) {}
        }), [ n, o, i, l, a ]);
    };
    // CONCATENATED MODULE: ./src/Card/Header.tsx
    function Z(e, t) {
        if (null == e) return {};
        var r, n, o = ee(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function ee(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    function te() {
        return te = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, te.apply(null, arguments);
    }
    var re = {
        actionPrimary: a().node,
        actionsSecondary: a().node,
        anchor: a().string,
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        icon: a().node,
        subtitle: a().node,
        title: a().node,
        truncateTitle: a().bool
    };
    // The order of the interfaces is necessary to maintain proper error messages: HeaderWithActions then HeaderWithActionsRenderProp
    // If the actions comes first the error will report that actions should be undefined
    // when we want actions to be the intended use case
        var ne = "36px";
    var oe =  n().createElement(M(), null);
    // `forwardRef` is used to enable focus on toggle when Dropdown is closing
        var ae =  n().forwardRef((function(e, t) {
        
        return n().createElement(Y(), te({
            content: (0, z._)("Actions"),
            contentRelationship: "label"
        }, e),  n().createElement(F(), {
            appearance: "secondary",
            "data-test": "actions-secondary-toggle",
            icon: oe,
            elementRef: t
        }));
    }));
    /**
 * A styled container for `Card` header content.
 */    function ie(e) {
        var t = e.actions, o = e.actionPrimary, a = e.actionsSecondary, i = e.anchor, l = e.children, c = e.icon, s = e.subtitle, u = e.title, d = e.truncateTitle, g = d === void 0 ? true : d, h = Z(e, [ "actions", "actionPrimary", "actionsSecondary", "anchor", "children", "icon", "subtitle", "title", "truncateTitle" ]);
        // @docs-props-type HeaderPropsBase
                K({
            additionalMessage: "Pass the Anchor component to the 'title' prop instead.",
            componentName: "Card.Header",
            propName: "anchor",
            propValue: i
        });
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ t, o, a ]);
        var O;
        if (t) {
            O = t();
        } else if (o || a) {
            O =  n().createElement(n().Fragment, null, o, a &&  n().createElement(U(), {
                toggle:  n().createElement(ae, null)
            }, a));
        }
        var w =  (0, r.isValidElement)(c) &&  (0, r.cloneElement)(c, {
            width: c.props.width || ne,
            height: c.props.height || ne
        });
        
        return n().createElement(b, te({
            "data-test": "header"
        }, h), w &&  n().createElement(y, null, w), (u || s) &&  n().createElement(v, {
            $truncateTitle: g
        }, u &&  n().createElement(f, null, i ?  n().createElement(V(), {
            name: i
        }, u) : u), s &&  n().createElement(p, null, s)), l, O &&  n().createElement(m, null, O));
    }
    ie.propTypes = re;
    /* harmony default export */    const le = ie;
    // CONCATENATED MODULE: ./src/Card/Card.tsx
    function ce(e) {
        "@babel/helpers - typeof";
        return ce = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, ce(e);
    }
    function se() {
        return se = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, se.apply(null, arguments);
    }
    function ue(e, t) {
        return me(e) || ve(e, t) || fe(e, t) || de();
    }
    function de() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function fe(e, t) {
        if (e) {
            if ("string" == typeof e) return pe(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? pe(e, t) : void 0;
        }
    }
    function pe(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function ve(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, a, i, l = [], c = !0, s = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (n = a.call(r)).done) && (l.push(n.value), l.length !== t); c = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function me(e) {
        if (Array.isArray(e)) return e;
    }
    function ye(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function be(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? ye(Object(r), !0).forEach((function(t) {
                ge(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : ye(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function ge(e, t, r) {
        return (t = he(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function he(e) {
        var t = Oe(e, "string");
        return "symbol" == ce(t) ? t : t + "";
    }
    function Oe(e, t) {
        if ("object" != ce(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != ce(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function we(e, t) {
        if (null == e) return {};
        var r, n, o = Se(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function Se(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */    var xe = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        /** @private */
        margin: a().number,
        /** @private */
        maxWidth: a().oneOfType([ a().number, a().string ]),
        /** @private */
        minWidth: a().oneOfType([ a().number, a().string ]),
        onClick: a().func,
        openInNewContext: a().bool,
        selected: a().bool,
        to: a().string,
        value: a().any,
        tag: a().oneOf([ "article", "div", "aside", "section" ])
    };
    function je(e) {
        return !!e.onClick || !!e.to;
    }
    function ke(e) {
        var t = e.margin, r = e.maxWidth, n = e.minWidth, o = e.style, a = we(e, [ "margin", "maxWidth", "minWidth", "style" ]);
        var i = be({
            minWidth: n,
            maxWidth: r,
            margin: t
        }, o);
        var l = {
            "data-test": "card",
            style: i
        };
        return [ l, a ];
    }
    function Ce(e) {
        var t = ke(e), o = ue(t, 2), a = o[0], i = o[1];
        var l = i.children, c = i.elementRef, s = i.onClick, u = i.openInNewContext, d = i.selected, f = i.to, p = i.value, v = we(i, [ "children", "elementRef", "onClick", "openInNewContext", "selected", "to", "value" ]);
        var m = (0, r.useCallback)((function(e) {
            s === null || s === void 0 ? void 0 : s(e, {
                selected: d || false,
                value: p
            });
        }), [ s, d, p ]);
        
        return n().createElement(N, se({
            "data-clickable": true,
            $selected: d || false,
            elementRef: c
        }, a, v, {
            onClick: s ? m : undefined,
            openInNewContext: u,
            to: f || undefined
        }), l);
    }
    function Pe(e) {
        var t = ke(e), r = ue(t, 2), o = r[0], a = r[1];
        var i = a.children, l = a.elementRef, c = a.tag, s = we(a, [ "children", "elementRef", "tag" ]);
        
        return n().createElement(_, se({
            $selected: false,
            ref: l,
            as: c
        }, o, s, {
            onClick: undefined
        }), i);
    }
    function Ee(e) {
        // @docs-props-type CardPropsBase
        K({
            componentName: "Card",
            propName: "selected",
            propValue: e.selected
        });
        return je(e) ?  n().createElement(Ce, e) :  n().createElement(Pe, e);
    }
    Ee.propTypes = xe;
    Ee.Header = le;
    Ee.Body = j;
    Ee.Footer = W;
    /* harmony default export */    const _e = Ee;
    // CONCATENATED MODULE: ./src/Card/index.ts
    module.exports = t;
    /******/})();
