// @splunk/react-ui/Chip.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var o in n) {
                /******/ if (e.o(n, o) && !e.o(r, o)) {
                    /******/ Object.defineProperty(r, o, {
                        enumerable: true,
                        get: n[o]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ D
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var o =  e.n(n);
    // CONCATENATED MODULE: external "lodash/omit"
    const a = require("lodash/omit");
    var t =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const s = require("@splunk/react-ui/ScreenReaderContent");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/color"
    const u = require("@splunk/ui-utils/color");
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const d = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const v = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const b = require("@splunk/react-icons/Cross");
    var m =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const y = require("@splunk/react-ui/Clickable");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/themes"
    const C = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Chip/ChipStyles.ts
    var h = (0, f.css)([ "", ";", ";align-content:center;align-items:center;height:", ";max-width:100%;padding:", " ", ";column-gap:", ";background-color:", ";color:", ";border-radius:", ";", ";", ";" ], C.mixins.reset("grid"), C.mixins.typography("body"), C.variables.inputHeight, C.variables.spacingXSmall, C.variables.spacingSmall, C.variables.spacingXSmall, (0, 
    C.pickVariant)("$appearance", {
        default: C.variables.neutral200,
        outline: C.variables.transparent,
        info: C.variables.severityColorInfo,
        success: C.variables.severityColorNormal,
        warning: C.variables.severityColorNotice,
        error: C.variables.severityColorAlert,
        custom: function e(r) {
            var n = r.$backgroundColor;
            return n;
        }
    }), (function(e) {
        var r = e.$appearance, n = e.$foregroundColor;
        return n || (r === "default" || r === "outline" ? C.variables.contentColorActive : C.variables.contentColorInverted);
    }), C.variables.borderRadius, (function(e) {
        var r = e.$appearance;
        return r === "outline" && (0, f.css)([ "border:1px solid ", ";" ], C.variables.borderColorStrong);
    }), (function(e) {
        var r = e.$appearance, n = e.$disabled;
        return n && (0, f.css)([ "background-color:", ";color:", ";", "" ], C.variables.neutral100, C.variables.contentColorDisabled, r === "outline" && (0, 
        f.css)([ "background-color:", ";border-color:", ";" ], C.variables.transparent, C.variables.borderColorWeak));
    }));
    var S = p().div.withConfig({
        displayName: "ChipStyles__StyledNonInteractive",
        componentId: "sc-1sd3tsh-0"
    })([ "", " ", ";" ], h, (0, C.pickVariant)("$icon", {
        true: (0, f.css)([ "grid-template-columns:[icon] min-content [label] auto;" ]),
        false: (0, f.css)([ "grid-template-columns:[label] auto;" ])
    }));
    var $ = p().div.withConfig({
        displayName: "ChipStyles__StyledIcon",
        componentId: "sc-1sd3tsh-1"
    })([ "display:grid;grid-column:icon;" ]);
    var k = p()(m()).withConfig({
        displayName: "ChipStyles__StyledRemoveIcon",
        componentId: "sc-1sd3tsh-2"
    })([ "grid-column:remove;" ]);
    var O = p().div.withConfig({
        displayName: "ChipStyles__StyledLabel",
        componentId: "sc-1sd3tsh-3"
    })([ "", " grid-column:label;" ], C.mixins.ellipsis());
    var R = p()(g()).withConfig({
        displayName: "ChipStyles__StyledClickable",
        componentId: "sc-1sd3tsh-4"
    })([ "", " cursor:pointer;", ";", "" ], h, (0, C.pickVariant)("$icon", {
        true: (0, f.css)([ "grid-template-columns:[icon] min-content [label] auto [remove] min-content;" ]),
        false: (0, f.css)([ "grid-template-columns:[label] auto [remove] min-content;" ])
    }), (function(e) {
        var r = e.$disabled;
        return !r && (0, f.css)([ "&:focus{box-shadow:", ";}&:hover{background-color:", ";}" ], C.variables.focusShadow, (0, 
        C.pickVariant)("$appearance", {
            outline: C.variables.neutral100,
            default: C.variables.neutral300,
            info: C.mixins.overlayColors(C.variables.severityColorInfo, C.variables.interactiveColorOverlayHover),
            success: C.mixins.overlayColors(C.variables.severityColorNormal, C.variables.interactiveColorOverlayHover),
            warning: C.mixins.overlayColors(C.variables.severityColorNotice, C.variables.interactiveColorOverlayHover),
            error: C.mixins.overlayColors(C.variables.severityColorAlert, C.variables.interactiveColorOverlayHover),
            custom: function e(r) {
                var n = r.$backgroundColor;
                return n && C.mixins.overlayColors(n, C.variables.interactiveColorOverlayHover);
            }
        }));
    }));
    // CONCATENATED MODULE: ./src/Chip/Chip.tsx
    function w() {
        return w = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var o in n) {
                    ({}).hasOwnProperty.call(n, o) && (e[o] = n[o]);
                }
            }
            return e;
        }, w.apply(null, arguments);
    }
    function q(e, r) {
        return E(e) || I(e, r) || j(e, r) || _();
    }
    function _() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function j(e, r) {
        if (e) {
            if ("string" == typeof e) return x(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? x(e, r) : void 0;
        }
    }
    function x(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, o = Array(r); n < r; n++) {
            o[n] = e[n];
        }
        return o;
    }
    function I(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var o, a, t, l, i = [], s = !0, c = !1;
            try {
                if (t = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    s = !1;
                } else for (;!(s = (o = t.call(n)).done) && (i.push(o.value), i.length !== r); s = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!s && null != n["return"] && (l = n["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function E(e) {
        if (Array.isArray(e)) return e;
    }
    function N(e, r) {
        if (null == e) return {};
        var n, o, a = A(e, r);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            for (o = 0; o < t.length; o++) {
                n = t[o], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function A(e, r) {
        if (null == e) return {};
        var n = {};
        for (var o in e) {
            if ({}.hasOwnProperty.call(e, o)) {
                if (-1 !== r.indexOf(o)) continue;
                n[o] = e[o];
            }
        }
        return n;
    }
    /** @public */    var P = {
        appearance: i().oneOf([ "info", "success", "warning", "error", "outline" ]),
        backgroundColor: i().string,
        children: i().node.isRequired,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        foregroundColor: i().string,
        icon: i().node,
        onRequestRemove: i().func,
        /** Includes this value in `onRequestRemove` callbacks. */
        value: i().any
    };
    function H(e) {
        return !!e.onRequestRemove;
    }
    function T(e) {
        var r = e.appearance, n = e.backgroundColor, o = e.disabled, a = e.foregroundColor, t = e.icon, l = N(e, [ "appearance", "backgroundColor", "disabled", "foregroundColor", "icon" ]);
 // Validate that the color being rendered into the CSS is restricted to only values that are of CSS type <color>.
                var i = a && (0, u.isCSSColor)(a) ? a : undefined;
        var s = n && (0, u.isCSSColor)(n) ? n : undefined;
        var c = {
            "data-test": "chip",
            $appearance: r || (s ? "custom" : "default"),
            $backgroundColor: s,
            $foregroundColor: i,
            disabled: o,
            icon: t
        };
        return [ c, l ];
    }
    function M(e) {
        var r = T(e), a = q(r, 2), l = a[0], i = a[1];
        var s = l.$appearance, u = l.$foregroundColor, f = l.disabled, p = f === void 0 ? false : f, b = l.icon, m = N(l, [ "$appearance", "$foregroundColor", "disabled", "icon" ]);
        var y = i.children, g = i.elementRef, C = i.onRequestRemove, h = i.value, S = N(i, [ "children", "elementRef", "onRequestRemove", "value" ]);
        var _ = typeof y === "string";
        var j = _ ? (0, d.sprintf)((0, v._)("Remove %(children)s"), {
            children: y
        }) : (0, v._)("Remove");
        var x = (0, n.useCallback)((function(e) {
            C(e, {
                value: h
            });
        }), [ C, h ]);
        
        return o().createElement(R, w({
            $appearance: s,
            $icon: !!b,
            $disabled: p,
            $foregroundColor: u,
            "data-test-value": h,
            disabled: p ? "disabled" : undefined,
            elementRef: g,
            onClick: x
        }, m, t()(S, Object.keys(P))),  o().createElement(c(), null, j), b &&  o().createElement($, null, b),  o().createElement(O, {
            "data-test": "label",
            "aria-hidden": _ ? true : undefined
        }, y),  o().createElement(k, {
            "data-test": "cross"
        }));
    }
    function V(e) {
        var r = T(e), n = q(r, 2), a = n[0], l = n[1];
        var i = a.$appearance, s = a.$foregroundColor, c = a.disabled, u = a.icon, d = N(a, [ "$appearance", "$foregroundColor", "disabled", "icon" ]);
        var v = l.children, f = l.elementRef, p = N(l, [ "children", "elementRef" ]);
        
        return o().createElement(S, w({
            $appearance: i,
            $icon: !!u,
            $disabled: c !== null && c !== void 0 ? c : false,
            $foregroundColor: s,
            "data-test-disabled": c ? "disabled" : undefined,
            ref: f
        }, d, t()(p, Object.keys(P))), u &&  o().createElement($, null, u),  o().createElement(O, {
            "data-test": "label"
        }, v));
    }
    function X(e) {
        // @docs-props-type ChipPropsBase
        return H(e) ?  o().createElement(M, e) :  o().createElement(V, e);
    }
    X.propTypes = P;
    /* harmony default export */    const D = X;
    // ChipInteractiveProps and ChipNonInteractiveProps are exported for Storybook
        // CONCATENATED MODULE: ./src/Chip/index.ts
    module.exports = r;
    /******/})();
