// @splunk/react-ui/Clickable.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        NavigationContext: () => /* reexport */ m,
        NavigationProvider: () => /* reexport */ y,
        default: () => /* reexport */ T,
        isInternalLink: () => /* reexport */ j,
        isRootRelativeLink: () => /* reexport */ S
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    // CONCATENATED MODULE: external "lodash/isString"
    const a = require("lodash/isString");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "lodash/omit"
    const o = require("lodash/omit");
    var d =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var c =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Clickable/ClickableStyles.ts
    var v = c().a.withConfig({
        displayName: "ClickableStyles__StyledA",
        componentId: "sc-7al1vw-0"
    })([ "", " cursor:pointer;&[disabled],&[aria-disabled='true']{cursor:not-allowed;color:", ";}&::-moz-focus-inner{border:0;padding:0;}" ], b.mixins.reset("inline"), b.variables.contentColorDisabled);
    // CONCATENATED MODULE: ./src/Clickable/NavigationProvider.tsx
    /** @public */
    var p = {
        children: s().node,
        onClick: s().func,
        onLinkClick: s().func,
        transformUrl: s().func
    };
    var m =  t().createContext({});
    /**
 * Used to provide an override for the `onClick` for links for single page applications so that
 * internal links can navigate without a page reload.
 */    function y(e) {
        var n = e.children, i = e.onClick, a = e.onLinkClick, l = e.transformUrl;
        // @docs-props-type NavigationProviderProps
                var o = (0, r.useMemo)((function() {
            return {
                onClick: i,
                onLinkClick: a,
                transformUrl: l
            };
        }), [ i, a, l ]);
        
        return t().createElement(m.Provider, {
            value: o
        }, n);
    }
    y.propTypes = p;
    // CONCATENATED MODULE: ./src/utils/disabled.ts
    /**
 * Normalizes the disabled prop and returns both the boolean state and normalized string value.
 * Use this function to ensure consistent handling of disabled prop across components.
 *
 * @param disabled - The disabled prop value
 * @param defaultWhenBooleanTrue - Optional normalization override for disabled={true}
 * @returns Object with isDisabled boolean|undefined and disabledValue string
 *
 * TODO (SUI-8600): `disabled` prop should default to 'dimmed' for data entry components
 */
    function k(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "dimmed";
        if (e === "dimmed") {
            return {
                isDisabled: true,
                disabledValue: "dimmed"
            };
        }
        if (e === true) {
            return {
                isDisabled: true,
                disabledValue: n
            };
        }
        if (e === "disabled") {
            return {
                isDisabled: true,
                disabledValue: "disabled"
            };
        }
        if (e === false) {
            return {
                isDisabled: false,
                disabledValue: undefined
            };
        }
        return {
            isDisabled: undefined,
            disabledValue: undefined
        };
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function g(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Clickable/Clickable.tsx
    function C() {
        return C = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, C.apply(null, arguments);
    }
    function O(e, n) {
        if (null == e) return {};
        var r, t, i = h(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function h(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var w = /^[a-z0-9]+:/;
    var x = /^\/\//;
    var P = /^\//;
    var S = function e(n) {
        return n != null && P.test(n);
    };
    var j = function e(n) {
        return n != null && !w.test(n) && !x.test(n);
    };
    var L = {
        children: s().node,
        disabled: s().oneOfType([ s().bool, s().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: s().oneOfType([ s().func, s().object ]),
        onClick: s().func,
        openInNewContext: s().bool,
        navigationLabel: s().string,
        to: s().oneOfType([ s().string, s().oneOf([ undefined ]) ])
    };
    function I(e, n) {
        return e !== "button" && n !== undefined;
    }
    /**
 * `Clickable` renders as a `button` element, or as an `a` element if the `to` prop is set
 * and the `disabled` prop is `false`. This is called link mode.
 */    var R =  t().forwardRef((function(e, 
    // SUI-7623 - needs to be specified for react-docgen versions less than 7.1.0
    n) {
        var i = e.children, a = e.disabled, o = e.elementRef, u = e.navigationLabel, s = e.onClick, f = e.openInNewContext, c = e.tag, b = e.to, p = O(e, [ "children", "disabled", "elementRef", "navigationLabel", "onClick", "openInNewContext", "tag", "to" ]);
        // @docs-props-type ClickablePropsBase
                var y = (0, r.useContext)(m), h = y.onClick, w = y.onLinkClick, x = y.transformUrl;
        (0, r.useEffect)((function() {
            if (false) {}
            // this effect only runs once
            // eslint-disable-next-line react-hooks/exhaustive-deps
                }), []);
        var P = (0, r.useCallback)((function(e) {
            g(o, e);
            g(n, e);
        }), [ o, n ]);
        // memoizing this result causes TS to lose track of the type guard established by isLinkCheck
                var L = I(c, b);
        // ideally we wouldn't need to pass "to" as "toValue" here
        // but without doing that TS thinks the return value of this is
        // string | undefined which doesn't work for usages of this function
                var R = (0, r.useCallback)((function(e) {
            return x != null ? x(e, {
                isInternal: j(e),
                isRootRelative: S(e)
            }) : e;
        }), [ x ]);
        var T = (0, r.useCallback)((function(e) {
            if (L) {
                // when user command-click on mac or ctrl-click on other platforms, and
                // Tag is an <a>, let the click pass through, let the <a> to achieve user's
                // intent of 'open in new context'
                // on mac, ctrl-click will be caught and open option menu even before hitting
                // the DOM, so we're safe to check both metaKey and ctrlKey here
                // without platform sniffing
                if ((e.metaKey || e.ctrlKey) && b) {
                    return;
                }
            }
            s === null || s === void 0 ? void 0 : s(e);
            var n = s == null && h != null;
            if (L && (n || w != null)) {
                var r = u;
                if (!r && l()(i)) {
                    r = i;
                }
                var t = {
                    to: R(b),
                    originalTo: b,
                    openInNewContext: f,
                    label: r
                };
                if (n) {
                    h(e, t);
                }
                w === null || w === void 0 ? void 0 : w(e, t);
            }
        }), [ i, R, L, u, s, h, w, f, b ]);
        var q = k(a), D = q.isDisabled, N = q.disabledValue;
        if (L) {
            if (D) {
                
                return t().createElement(v, C({
                    "data-test": "clickable",
                    "data-test-disabled": N,
                    ref: P,
                    role: "link",
                    "aria-disabled": "true"
                }, p), i);
            }
            var _ = f ? "_blank" : undefined;
            var E = f && j(b) ? "noopener noreferrer" : undefined;
            
            return t().createElement(v, C({
                "data-test": "clickable",
                ref: P,
                href: R(b),
                target: _,
                rel: E,
                onClick: T
            }, d()(p)), i);
        }
        // button variant
                var V = a === true || a === "dimmed" || undefined;
        
        return t().createElement(v, C({
            as: "button",
            "aria-disabled": V,
            "data-test": "clickable",
            "data-test-disabled": N,
            ref: P,
            type: "button",
            disabled: a === "disabled",
            onClick: D ? undefined : T
        }, d()(p, "disabled")), i);
    }));
    R.propTypes = L;
    /* harmony default export */    const T = R;
    // forwardRef appears to prevent TS from inferring ClickableProps in components that import Clickable
    // so we need to export it separately to prevent the TS build from failing with:
    // "... has or is using name 'ClickablePropsBase' from external module "...packages/react-ui/src/Clickable/Clickable" but cannot be named."
        // CONCATENATED MODULE: ./src/Clickable/index.ts
    module.exports = n;
    /******/})();
