// @splunk/react-ui/Code.js
/******/ (() => {
    // webpackBootstrap
    /******/ var e = {
        /***/ 74: 
        /***/ e => {
            (function() {
                var t = undefined;
                var n = undefined;
                /// <reference lib="WebWorker"/>
                                var r = typeof t !== "undefined" ? t : typeof WorkerGlobalScope !== "undefined" && self instanceof WorkerGlobalScope ? self : {};
                /**
 * Prism: Lightweight, robust, elegant syntax highlighting
 *
 * @license MIT <https://opensource.org/licenses/MIT>
 * @author Lea Verou <https://lea.verou.me>
 * @namespace
 * @public
 */                var a = function(e) {
                    // Private helper vars
                    var n = /(?:^|\s)lang(?:uage)?-([\w-]+)(?=\s|$)/i;
                    var r = 0;
                    // The grammar object for plaintext
                                        var a = {};
                    var s = {
                        /**
		 * By default, Prism will attempt to highlight all code elements (by calling {@link Prism.highlightAll}) on the
		 * current page after the page finished loading. This might be a problem if e.g. you wanted to asynchronously load
		 * additional languages or plugins yourself.
		 *
		 * By setting this value to `true`, Prism will not automatically highlight all code elements on the page.
		 *
		 * You obviously have to change this value before the automatic highlighting started. To do this, you can add an
		 * empty Prism object into the global scope before loading the Prism script like this:
		 *
		 * ```js
		 * window.Prism = window.Prism || {};
		 * Prism.manual = true;
		 * // add a new <script> to load Prism's script
		 * ```
		 *
		 * @default false
		 * @type {boolean}
		 * @memberof Prism
		 * @public
		 */
                        manual: e.Prism && e.Prism.manual,
                        /**
		 * By default, if Prism is in a web worker, it assumes that it is in a worker it created itself, so it uses
		 * `addEventListener` to communicate with its parent instance. However, if you're using Prism manually in your
		 * own worker, you don't want it to do this.
		 *
		 * By setting this value to `true`, Prism will not add its own listeners to the worker.
		 *
		 * You obviously have to change this value before Prism executes. To do this, you can add an
		 * empty Prism object into the global scope before loading the Prism script like this:
		 *
		 * ```js
		 * window.Prism = window.Prism || {};
		 * Prism.disableWorkerMessageHandler = true;
		 * // Load Prism's script
		 * ```
		 *
		 * @default false
		 * @type {boolean}
		 * @memberof Prism
		 * @public
		 */
                        disableWorkerMessageHandler: e.Prism && e.Prism.disableWorkerMessageHandler,
                        /**
		 * A namespace for utility methods.
		 *
		 * All function in this namespace that are not explicitly marked as _public_ are for __internal use only__ and may
		 * change or disappear at any time.
		 *
		 * @namespace
		 * @memberof Prism
		 */
                        util: {
                            encode: function e(t) {
                                if (t instanceof i) {
                                    return new i(t.type, e(t.content), t.alias);
                                } else if (Array.isArray(t)) {
                                    return t.map(e);
                                } else {
                                    return t.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/\u00a0/g, " ");
                                }
                            },
                            /**
			 * Returns the name of the type of the given value.
			 *
			 * @param {any} o
			 * @returns {string}
			 * @example
			 * type(null)      === 'Null'
			 * type(undefined) === 'Undefined'
			 * type(123)       === 'Number'
			 * type('foo')     === 'String'
			 * type(true)      === 'Boolean'
			 * type([1, 2])    === 'Array'
			 * type({})        === 'Object'
			 * type(String)    === 'Function'
			 * type(/abc+/)    === 'RegExp'
			 */
                            type: function(e) {
                                return Object.prototype.toString.call(e).slice(8, -1);
                            },
                            /**
			 * Returns a unique number for the given object. Later calls will still return the same number.
			 *
			 * @param {Object} obj
			 * @returns {number}
			 */
                            objId: function(e) {
                                if (!e["__id"]) {
                                    Object.defineProperty(e, "__id", {
                                        value: ++r
                                    });
                                }
                                return e["__id"];
                            },
                            /**
			 * Creates a deep clone of the given object.
			 *
			 * The main intended use of this function is to clone language definitions.
			 *
			 * @param {T} o
			 * @param {Record<number, any>} [visited]
			 * @returns {T}
			 * @template T
			 */
                            clone: function e(t, n) {
                                n = n || {};
                                var r;
                                var a;
                                switch (s.util.type(t)) {
                                  case "Object":
                                    a = s.util.objId(t);
                                    if (n[a]) {
                                        return n[a];
                                    }
                                    r = /** @type {Record<string, any>} */ {};
                                    n[a] = r;
                                    for (var i in t) {
                                        if (t.hasOwnProperty(i)) {
                                            r[i] = e(t[i], n);
                                        }
                                    }
                                    /** @type {any} */
                                    return r;

                                  case "Array":
                                    a = s.util.objId(t);
                                    if (n[a]) {
                                        return n[a];
                                    }
                                    r = [];
                                    n[a] = r;
                                    /** @type {Array} */ /** @type {any} */ t.forEach((function(t, a) {
                                        r[a] = e(t, n);
                                    }));
                                    /** @type {any} */
                                    return r;

                                  default:
                                    return t;
                                }
                            },
                            /**
			 * Returns the Prism language of the given element set by a `language-xxxx` or `lang-xxxx` class.
			 *
			 * If no language is set for the element or the element is `null` or `undefined`, `none` will be returned.
			 *
			 * @param {Element} element
			 * @returns {string}
			 */
                            getLanguage: function(e) {
                                while (e) {
                                    var t = n.exec(e.className);
                                    if (t) {
                                        return t[1].toLowerCase();
                                    }
                                    e = e.parentElement;
                                }
                                return "none";
                            },
                            /**
			 * Sets the Prism `language-xxxx` class of the given element.
			 *
			 * @param {Element} element
			 * @param {string} language
			 * @returns {void}
			 */
                            setLanguage: function(e, t) {
                                // remove all `language-xxxx` classes
                                // (this might leave behind a leading space)
                                e.className = e.className.replace(RegExp(n, "gi"), "");
                                // add the new `language-xxxx` class
                                // (using `classList` will automatically clean up spaces for us)
                                                                e.classList.add("language-" + t);
                            },
                            /**
			 * Returns the script element that is currently executing.
			 *
			 * This does __not__ work for line script element.
			 *
			 * @returns {HTMLScriptElement | null}
			 */
                            currentScript: function() {
                                if (typeof document === "undefined") {
                                    return null;
                                }
                                if ("currentScript" in document && 1 < 2 /* hack to trip TS' flow analysis */) {
                                    /** @type {any} */
                                    return document.currentScript;
                                }
                                // IE11 workaround
                                // we'll get the src of the current script by parsing IE11's error stack trace
                                // this will not work for inline scripts
                                                                try {
                                    throw new Error;
                                } catch (r) {
                                    // Get file src url from stack. Specifically works with the format of stack traces in IE.
                                    // A stack will look like this:
                                    // Error
                                    //    at _.util.currentScript (http://localhost/components/prism-core.js:119:5)
                                    //    at Global code (http://localhost/components/prism-core.js:606:1)
                                    var e = (/at [^(\r\n]*\((.*):[^:]+:[^:]+\)$/i.exec(r.stack) || [])[1];
                                    if (e) {
                                        var t = document.getElementsByTagName("script");
                                        for (var n in t) {
                                            if (t[n].src == e) {
                                                return t[n];
                                            }
                                        }
                                    }
                                    return null;
                                }
                            },
                            /**
			 * Returns whether a given class is active for `element`.
			 *
			 * The class can be activated if `element` or one of its ancestors has the given class and it can be deactivated
			 * if `element` or one of its ancestors has the negated version of the given class. The _negated version_ of the
			 * given class is just the given class with a `no-` prefix.
			 *
			 * Whether the class is active is determined by the closest ancestor of `element` (where `element` itself is
			 * closest ancestor) that has the given class or the negated version of it. If neither `element` nor any of its
			 * ancestors have the given class or the negated version of it, then the default activation will be returned.
			 *
			 * In the paradoxical situation where the closest ancestor contains __both__ the given class and the negated
			 * version of it, the class is considered active.
			 *
			 * @param {Element} element
			 * @param {string} className
			 * @param {boolean} [defaultActivation=false]
			 * @returns {boolean}
			 */
                            isActive: function(e, t, n) {
                                var r = "no-" + t;
                                while (e) {
                                    var a = e.classList;
                                    if (a.contains(t)) {
                                        return true;
                                    }
                                    if (a.contains(r)) {
                                        return false;
                                    }
                                    e = e.parentElement;
                                }
                                return !!n;
                            }
                        },
                        /**
		 * This namespace contains all currently loaded languages and the some helper functions to create and modify languages.
		 *
		 * @namespace
		 * @memberof Prism
		 * @public
		 */
                        languages: {
                            /**
			 * The grammar for plain, unformatted text.
			 */
                            plain: a,
                            plaintext: a,
                            text: a,
                            txt: a,
                            /**
			 * Creates a deep copy of the language with the given id and appends the given tokens.
			 *
			 * If a token in `redef` also appears in the copied language, then the existing token in the copied language
			 * will be overwritten at its original position.
			 *
			 * ## Best practices
			 *
			 * Since the position of overwriting tokens (token in `redef` that overwrite tokens in the copied language)
			 * doesn't matter, they can technically be in any order. However, this can be confusing to others that trying to
			 * understand the language definition because, normally, the order of tokens matters in Prism grammars.
			 *
			 * Therefore, it is encouraged to order overwriting tokens according to the positions of the overwritten tokens.
			 * Furthermore, all non-overwriting tokens should be placed after the overwriting ones.
			 *
			 * @param {string} id The id of the language to extend. This has to be a key in `Prism.languages`.
			 * @param {Grammar} redef The new tokens to append.
			 * @returns {Grammar} The new language created.
			 * @public
			 * @example
			 * Prism.languages['css-with-colors'] = Prism.languages.extend('css', {
			 *     // Prism.languages.css already has a 'comment' token, so this token will overwrite CSS' 'comment' token
			 *     // at its original position
			 *     'comment': { ... },
			 *     // CSS doesn't have a 'color' token, so this token will be appended
			 *     'color': /\b(?:red|green|blue)\b/
			 * });
			 */
                            extend: function(e, t) {
                                var n = s.util.clone(s.languages[e]);
                                for (var r in t) {
                                    n[r] = t[r];
                                }
                                return n;
                            },
                            /**
			 * Inserts tokens _before_ another token in a language definition or any other grammar.
			 *
			 * ## Usage
			 *
			 * This helper method makes it easy to modify existing languages. For example, the CSS language definition
			 * not only defines CSS highlighting for CSS documents, but also needs to define highlighting for CSS embedded
			 * in HTML through `<style>` elements. To do this, it needs to modify `Prism.languages.markup` and add the
			 * appropriate tokens. However, `Prism.languages.markup` is a regular JavaScript object literal, so if you do
			 * this:
			 *
			 * ```js
			 * Prism.languages.markup.style = {
			 *     // token
			 * };
			 * ```
			 *
			 * then the `style` token will be added (and processed) at the end. `insertBefore` allows you to insert tokens
			 * before existing tokens. For the CSS example above, you would use it like this:
			 *
			 * ```js
			 * Prism.languages.insertBefore('markup', 'cdata', {
			 *     'style': {
			 *         // token
			 *     }
			 * });
			 * ```
			 *
			 * ## Special cases
			 *
			 * If the grammars of `inside` and `insert` have tokens with the same name, the tokens in `inside`'s grammar
			 * will be ignored.
			 *
			 * This behavior can be used to insert tokens after `before`:
			 *
			 * ```js
			 * Prism.languages.insertBefore('markup', 'comment', {
			 *     'comment': Prism.languages.markup.comment,
			 *     // tokens after 'comment'
			 * });
			 * ```
			 *
			 * ## Limitations
			 *
			 * The main problem `insertBefore` has to solve is iteration order. Since ES2015, the iteration order for object
			 * properties is guaranteed to be the insertion order (except for integer keys) but some browsers behave
			 * differently when keys are deleted and re-inserted. So `insertBefore` can't be implemented by temporarily
			 * deleting properties which is necessary to insert at arbitrary positions.
			 *
			 * To solve this problem, `insertBefore` doesn't actually insert the given tokens into the target object.
			 * Instead, it will create a new object and replace all references to the target object with the new one. This
			 * can be done without temporarily deleting properties, so the iteration order is well-defined.
			 *
			 * However, only references that can be reached from `Prism.languages` or `insert` will be replaced. I.e. if
			 * you hold the target object in a variable, then the value of the variable will not change.
			 *
			 * ```js
			 * var oldMarkup = Prism.languages.markup;
			 * var newMarkup = Prism.languages.insertBefore('markup', 'comment', { ... });
			 *
			 * assert(oldMarkup !== Prism.languages.markup);
			 * assert(newMarkup === Prism.languages.markup);
			 * ```
			 *
			 * @param {string} inside The property of `root` (e.g. a language id in `Prism.languages`) that contains the
			 * object to be modified.
			 * @param {string} before The key to insert before.
			 * @param {Grammar} insert An object containing the key-value pairs to be inserted.
			 * @param {Object<string, any>} [root] The object containing `inside`, i.e. the object that contains the
			 * object to be modified.
			 *
			 * Defaults to `Prism.languages`.
			 * @returns {Grammar} The new grammar object.
			 * @public
			 */
                            insertBefore: function(e, t, n, r) {
                                r = r || /** @type {any} */ s.languages;
                                var a = r[e];
                                /** @type {Grammar} */                                var i = {};
                                for (var o in a) {
                                    if (a.hasOwnProperty(o)) {
                                        if (o == t) {
                                            for (var l in n) {
                                                if (n.hasOwnProperty(l)) {
                                                    i[l] = n[l];
                                                }
                                            }
                                        }
                                        // Do not insert token which also occur in insert. See #1525
                                                                                if (!n.hasOwnProperty(o)) {
                                            i[o] = a[o];
                                        }
                                    }
                                }
                                var u = r[e];
                                r[e] = i;
                                // Update references in other language definitions
                                                                s.languages.DFS(s.languages, (function(t, n) {
                                    if (n === u && t != e) {
                                        this[t] = i;
                                    }
                                }));
                                return i;
                            },
                            // Traverse a language definition with Depth First Search
                            DFS: function e(t, n, r, a) {
                                a = a || {};
                                var i = s.util.objId;
                                for (var o in t) {
                                    if (t.hasOwnProperty(o)) {
                                        n.call(t, o, t[o], r || o);
                                        var l = t[o];
                                        var u = s.util.type(l);
                                        if (u === "Object" && !a[i(l)]) {
                                            a[i(l)] = true;
                                            e(l, n, null, a);
                                        } else if (u === "Array" && !a[i(l)]) {
                                            a[i(l)] = true;
                                            e(l, n, o, a);
                                        }
                                    }
                                }
                            }
                        },
                        plugins: {},
                        /**
		 * This is the most high-level function in Prism’s API.
		 * It fetches all the elements that have a `.language-xxxx` class and then calls {@link Prism.highlightElement} on
		 * each one of them.
		 *
		 * This is equivalent to `Prism.highlightAllUnder(document, async, callback)`.
		 *
		 * @param {boolean} [async=false] Same as in {@link Prism.highlightAllUnder}.
		 * @param {HighlightCallback} [callback] Same as in {@link Prism.highlightAllUnder}.
		 * @memberof Prism
		 * @public
		 */
                        highlightAll: function(e, t) {
                            s.highlightAllUnder(document, e, t);
                        },
                        /**
		 * Fetches all the descendants of `container` that have a `.language-xxxx` class and then calls
		 * {@link Prism.highlightElement} on each one of them.
		 *
		 * The following hooks will be run:
		 * 1. `before-highlightall`
		 * 2. `before-all-elements-highlight`
		 * 3. All hooks of {@link Prism.highlightElement} for each element.
		 *
		 * @param {ParentNode} container The root element, whose descendants that have a `.language-xxxx` class will be highlighted.
		 * @param {boolean} [async=false] Whether each element is to be highlighted asynchronously using Web Workers.
		 * @param {HighlightCallback} [callback] An optional callback to be invoked on each element after its highlighting is done.
		 * @memberof Prism
		 * @public
		 */
                        highlightAllUnder: function(e, t, n) {
                            var r = {
                                callback: n,
                                container: e,
                                selector: 'code[class*="language-"], [class*="language-"] code, code[class*="lang-"], [class*="lang-"] code'
                            };
                            s.hooks.run("before-highlightall", r);
                            r.elements = Array.prototype.slice.apply(r.container.querySelectorAll(r.selector));
                            s.hooks.run("before-all-elements-highlight", r);
                            for (var a = 0, i; i = r.elements[a++]; ) {
                                s.highlightElement(i, t === true, r.callback);
                            }
                        },
                        /**
		 * Highlights the code inside a single element.
		 *
		 * The following hooks will be run:
		 * 1. `before-sanity-check`
		 * 2. `before-highlight`
		 * 3. All hooks of {@link Prism.highlight}. These hooks will be run by an asynchronous worker if `async` is `true`.
		 * 4. `before-insert`
		 * 5. `after-highlight`
		 * 6. `complete`
		 *
		 * Some the above hooks will be skipped if the element doesn't contain any text or there is no grammar loaded for
		 * the element's language.
		 *
		 * @param {Element} element The element containing the code.
		 * It must have a class of `language-xxxx` to be processed, where `xxxx` is a valid language identifier.
		 * @param {boolean} [async=false] Whether the element is to be highlighted asynchronously using Web Workers
		 * to improve performance and avoid blocking the UI when highlighting very large chunks of code. This option is
		 * [disabled by default](https://prismjs.com/faq.html#why-is-asynchronous-highlighting-disabled-by-default).
		 *
		 * Note: All language definitions required to highlight the code must be included in the main `prism.js` file for
		 * asynchronous highlighting to work. You can build your own bundle on the
		 * [Download page](https://prismjs.com/download.html).
		 * @param {HighlightCallback} [callback] An optional callback to be invoked after the highlighting is done.
		 * Mostly useful when `async` is `true`, since in that case, the highlighting is done asynchronously.
		 * @memberof Prism
		 * @public
		 */
                        highlightElement: function(t, n, r) {
                            // Find language
                            var a = s.util.getLanguage(t);
                            var i = s.languages[a];
                            // Set language on the element, if not present
                                                        s.util.setLanguage(t, a);
                            // Set language on the parent, for styling
                                                        var o = t.parentElement;
                            if (o && o.nodeName.toLowerCase() === "pre") {
                                s.util.setLanguage(o, a);
                            }
                            var l = t.textContent;
                            var u = {
                                element: t,
                                language: a,
                                grammar: i,
                                code: l
                            };
                            function c(e) {
                                u.highlightedCode = e;
                                s.hooks.run("before-insert", u);
                                u.element.innerHTML = u.highlightedCode;
                                s.hooks.run("after-highlight", u);
                                s.hooks.run("complete", u);
                                r && r.call(u.element);
                            }
                            s.hooks.run("before-sanity-check", u);
                            // plugins may change/add the parent/element
                                                        o = u.element.parentElement;
                            if (o && o.nodeName.toLowerCase() === "pre" && !o.hasAttribute("tabindex")) {
                                o.setAttribute("tabindex", "0");
                            }
                            if (!u.code) {
                                s.hooks.run("complete", u);
                                r && r.call(u.element);
                                return;
                            }
                            s.hooks.run("before-highlight", u);
                            if (!u.grammar) {
                                c(s.util.encode(u.code));
                                return;
                            }
                            if (n && e.Worker) {
                                var p = new Worker(s.filename);
                                p.onmessage = function(e) {
                                    c(e.data);
                                };
                                p.postMessage(JSON.stringify({
                                    language: u.language,
                                    code: u.code,
                                    immediateClose: true
                                }));
                            } else {
                                c(s.highlight(u.code, u.grammar, u.language));
                            }
                        },
                        /**
		 * Low-level function, only use if you know what you’re doing. It accepts a string of text as input
		 * and the language definitions to use, and returns a string with the HTML produced.
		 *
		 * The following hooks will be run:
		 * 1. `before-tokenize`
		 * 2. `after-tokenize`
		 * 3. `wrap`: On each {@link Token}.
		 *
		 * @param {string} text A string with the code to be highlighted.
		 * @param {Grammar} grammar An object containing the tokens to use.
		 *
		 * Usually a language definition like `Prism.languages.markup`.
		 * @param {string} language The name of the language definition passed to `grammar`.
		 * @returns {string} The highlighted HTML.
		 * @memberof Prism
		 * @public
		 * @example
		 * Prism.highlight('var foo = true;', Prism.languages.javascript, 'javascript');
		 */
                        highlight: function(e, t, n) {
                            var r = {
                                code: e,
                                grammar: t,
                                language: n
                            };
                            s.hooks.run("before-tokenize", r);
                            if (!r.grammar) {
                                throw new Error('The language "' + r.language + '" has no grammar.');
                            }
                            r.tokens = s.tokenize(r.code, r.grammar);
                            s.hooks.run("after-tokenize", r);
                            return i.stringify(s.util.encode(r.tokens), r.language);
                        },
                        /**
		 * This is the heart of Prism, and the most low-level function you can use. It accepts a string of text as input
		 * and the language definitions to use, and returns an array with the tokenized code.
		 *
		 * When the language definition includes nested tokens, the function is called recursively on each of these tokens.
		 *
		 * This method could be useful in other contexts as well, as a very crude parser.
		 *
		 * @param {string} text A string with the code to be highlighted.
		 * @param {Grammar} grammar An object containing the tokens to use.
		 *
		 * Usually a language definition like `Prism.languages.markup`.
		 * @returns {TokenStream} An array of strings and tokens, a token stream.
		 * @memberof Prism
		 * @public
		 * @example
		 * let code = `var foo = 0;`;
		 * let tokens = Prism.tokenize(code, Prism.languages.javascript);
		 * tokens.forEach(token => {
		 *     if (token instanceof Prism.Token && token.type === 'number') {
		 *         console.log(`Found numeric literal: ${token.content}`);
		 *     }
		 * });
		 */
                        tokenize: function(e, t) {
                            var n = t.rest;
                            if (n) {
                                for (var r in n) {
                                    t[r] = n[r];
                                }
                                delete t.rest;
                            }
                            var a = new u;
                            c(a, a.head, e);
                            l(e, a, t, a.head, 0);
                            return d(a);
                        },
                        /**
		 * @namespace
		 * @memberof Prism
		 * @public
		 */
                        hooks: {
                            all: {},
                            /**
			 * Adds the given callback to the list of callbacks for the given hook.
			 *
			 * The callback will be invoked when the hook it is registered for is run.
			 * Hooks are usually directly run by a highlight function but you can also run hooks yourself.
			 *
			 * One callback function can be registered to multiple hooks and the same hook multiple times.
			 *
			 * @param {string} name The name of the hook.
			 * @param {HookCallback} callback The callback function which is given environment variables.
			 * @public
			 */
                            add: function(e, t) {
                                var n = s.hooks.all;
                                n[e] = n[e] || [];
                                n[e].push(t);
                            },
                            /**
			 * Runs a hook invoking all registered callbacks with the given environment variables.
			 *
			 * Callbacks will be invoked synchronously and in the order in which they were registered.
			 *
			 * @param {string} name The name of the hook.
			 * @param {Object<string, any>} env The environment variables of the hook passed to all callbacks registered.
			 * @public
			 */
                            run: function(e, t) {
                                var n = s.hooks.all[e];
                                if (!n || !n.length) {
                                    return;
                                }
                                for (var r = 0, a; a = n[r++]; ) {
                                    a(t);
                                }
                            }
                        },
                        Token: i
                    };
                    e.Prism = s;
                    // Typescript note:
                    // The following can be used to import the Token type in JSDoc:
                    
                    //   @typedef {InstanceType<import("./prism-core")["Token"]>} Token
                    /**
	 * Creates a new token.
	 *
	 * @param {string} type See {@link Token#type type}
	 * @param {string | TokenStream} content See {@link Token#content content}
	 * @param {string|string[]} [alias] The alias(es) of the token.
	 * @param {string} [matchedStr=""] A copy of the full string this token was created from.
	 * @class
	 * @global
	 * @public
	 */                    function i(e, t, n, r) {
                        /**
		 * The type of the token.
		 *
		 * This is usually the key of a pattern in a {@link Grammar}.
		 *
		 * @type {string}
		 * @see GrammarToken
		 * @public
		 */
                        this.type = e;
                        /**
		 * The strings or tokens contained by this token.
		 *
		 * This will be a token stream if the pattern matched also defined an `inside` grammar.
		 *
		 * @type {string | TokenStream}
		 * @public
		 */                        this.content = t;
                        /**
		 * The alias(es) of the token.
		 *
		 * @type {string|string[]}
		 * @see GrammarToken
		 * @public
		 */                        this.alias = n;
                        // Copy of the full string this token was created from
                                                this.length = (r || "").length | 0;
                    }
                    /**
	 * A token stream is an array of strings and {@link Token Token} objects.
	 *
	 * Token streams have to fulfill a few properties that are assumed by most functions (mostly internal ones) that process
	 * them.
	 *
	 * 1. No adjacent strings.
	 * 2. No empty strings.
	 *
	 *    The only exception here is the token stream that only contains the empty string and nothing else.
	 *
	 * @typedef {Array<string | Token>} TokenStream
	 * @global
	 * @public
	 */
                    /**
	 * Converts the given token or token stream to an HTML representation.
	 *
	 * The following hooks will be run:
	 * 1. `wrap`: On each {@link Token}.
	 *
	 * @param {string | Token | TokenStream} o The token or token stream to be converted.
	 * @param {string} language The name of current language.
	 * @returns {string} The HTML representation of the token or token stream.
	 * @memberof Token
	 * @static
	 */                    i.stringify = function e(t, n) {
                        if (typeof t == "string") {
                            return t;
                        }
                        if (Array.isArray(t)) {
                            var r = "";
                            t.forEach((function(t) {
                                r += e(t, n);
                            }));
                            return r;
                        }
                        var a = {
                            type: t.type,
                            content: e(t.content, n),
                            tag: "span",
                            classes: [ "token", t.type ],
                            attributes: {},
                            language: n
                        };
                        var i = t.alias;
                        if (i) {
                            if (Array.isArray(i)) {
                                Array.prototype.push.apply(a.classes, i);
                            } else {
                                a.classes.push(i);
                            }
                        }
                        s.hooks.run("wrap", a);
                        var o = "";
                        for (var l in a.attributes) {
                            o += " " + l + '="' + (a.attributes[l] || "").replace(/"/g, "&quot;") + '"';
                        }
                        return "<" + a.tag + ' class="' + a.classes.join(" ") + '"' + o + ">" + a.content + "</" + a.tag + ">";
                    };
                    /**
	 * @param {RegExp} pattern
	 * @param {number} pos
	 * @param {string} text
	 * @param {boolean} lookbehind
	 * @returns {RegExpExecArray | null}
	 */                    function o(e, t, n, r) {
                        e.lastIndex = t;
                        var a = e.exec(n);
                        if (a && r && a[1]) {
                            // change the match to remove the text matched by the Prism lookbehind group
                            var s = a[1].length;
                            a.index += s;
                            a[0] = a[0].slice(s);
                        }
                        return a;
                    }
                    /**
	 * @param {string} text
	 * @param {LinkedList<string | Token>} tokenList
	 * @param {any} grammar
	 * @param {LinkedListNode<string | Token>} startNode
	 * @param {number} startPos
	 * @param {RematchOptions} [rematch]
	 * @returns {void}
	 * @private
	 *
	 * @typedef RematchOptions
	 * @property {string} cause
	 * @property {number} reach
	 */                    function l(e, t, n, r, a, u) {
                        for (var d in n) {
                            if (!n.hasOwnProperty(d) || !n[d]) {
                                continue;
                            }
                            var g = n[d];
                            g = Array.isArray(g) ? g : [ g ];
                            for (var f = 0; f < g.length; ++f) {
                                if (u && u.cause == d + "," + f) {
                                    return;
                                }
                                var m = g[f];
                                var b = m.inside;
                                var h = !!m.lookbehind;
                                var v = !!m.greedy;
                                var E = m.alias;
                                if (v && !m.pattern.global) {
                                    // Without the global flag, lastIndex won't work
                                    var y = m.pattern.toString().match(/[imsuy]*$/)[0];
                                    m.pattern = RegExp(m.pattern.source, y + "g");
                                }
                                /** @type {RegExp} */                                var S = m.pattern || m;
                                for (// iterate the token list and keep track of the current token/string position
                                var A = r.next, k = a; A !== t.tail; k += A.value.length, A = A.next) {
                                    if (u && k >= u.reach) {
                                        break;
                                    }
                                    var x = A.value;
                                    if (t.length > e.length) {
                                        // Something went terribly wrong, ABORT, ABORT!
                                        return;
                                    }
                                    if (x instanceof i) {
                                        continue;
                                    }
                                    var T = 1;
 // this is the to parameter of removeBetween
                                                                        var I;
                                    if (v) {
                                        I = o(S, k, e, h);
                                        if (!I || I.index >= e.length) {
                                            break;
                                        }
                                        var R = I.index;
                                        var w = I.index + I[0].length;
                                        var N = k;
                                        // find the node that contains the match
                                                                                N += A.value.length;
                                        while (R >= N) {
                                            A = A.next;
                                            N += A.value.length;
                                        }
                                        // adjust pos (and p)
                                                                                N -= A.value.length;
                                        k = N;
                                        // the current node is a Token, then the match starts inside another Token, which is invalid
                                                                                if (A.value instanceof i) {
                                            continue;
                                        }
                                        // find the last node which is affected by this match
                                                                                for (var O = A; O !== t.tail && (N < w || typeof O.value === "string"); O = O.next) {
                                            T++;
                                            N += O.value.length;
                                        }
                                        T--;
                                        // replace with the new match
                                                                                x = e.slice(k, N);
                                        I.index -= k;
                                    } else {
                                        I = o(S, 0, x, h);
                                        if (!I) {
                                            continue;
                                        }
                                    }
                                    // eslint-disable-next-line no-redeclare
                                                                        var R = I.index;
                                    var _ = I[0];
                                    var L = x.slice(0, R);
                                    var F = x.slice(R + _.length);
                                    var C = k + x.length;
                                    if (u && C > u.reach) {
                                        u.reach = C;
                                    }
                                    var D = A.prev;
                                    if (L) {
                                        D = c(t, D, L);
                                        k += L.length;
                                    }
                                    p(t, D, T);
                                    var P = new i(d, b ? s.tokenize(_, b) : _, E, _);
                                    A = c(t, D, P);
                                    if (F) {
                                        c(t, A, F);
                                    }
                                    if (T > 1) {
                                        // at least one Token object was removed, so we have to do some rematching
                                        // this can only happen if the current pattern is greedy
                                        /** @type {RematchOptions} */
                                        var $ = {
                                            cause: d + "," + f,
                                            reach: C
                                        };
                                        l(e, t, n, A.prev, k, $);
                                        // the reach might have been extended because of the rematching
                                                                                if (u && $.reach > u.reach) {
                                            u.reach = $.reach;
                                        }
                                    }
                                }
                            }
                        }
                    }
                    /**
	 * @typedef LinkedListNode
	 * @property {T} value
	 * @property {LinkedListNode<T> | null} prev The previous node.
	 * @property {LinkedListNode<T> | null} next The next node.
	 * @template T
	 * @private
	 */
                    /**
	 * @template T
	 * @private
	 */                    function u() {
                        /** @type {LinkedListNode<T>} */
                        var e = {
                            value: null,
                            prev: null,
                            next: null
                        };
                        /** @type {LinkedListNode<T>} */                        var t = {
                            value: null,
                            prev: e,
                            next: null
                        };
                        e.next = t;
                        /** @type {LinkedListNode<T>} */                        this.head = e;
                        /** @type {LinkedListNode<T>} */                        this.tail = t;
                        this.length = 0;
                    }
                    /**
	 * Adds a new node with the given value to the list.
	 *
	 * @param {LinkedList<T>} list
	 * @param {LinkedListNode<T>} node
	 * @param {T} value
	 * @returns {LinkedListNode<T>} The added node.
	 * @template T
	 */                    function c(e, t, n) {
                        // assumes that node != list.tail && values.length >= 0
                        var r = t.next;
                        var a = {
                            value: n,
                            prev: t,
                            next: r
                        };
                        t.next = a;
                        r.prev = a;
                        e.length++;
                        return a;
                    }
                    /**
	 * Removes `count` nodes after the given node. The given node will not be removed.
	 *
	 * @param {LinkedList<T>} list
	 * @param {LinkedListNode<T>} node
	 * @param {number} count
	 * @template T
	 */                    function p(e, t, n) {
                        var r = t.next;
                        for (var a = 0; a < n && r !== e.tail; a++) {
                            r = r.next;
                        }
                        t.next = r;
                        r.prev = t;
                        e.length -= a;
                    }
                    /**
	 * @param {LinkedList<T>} list
	 * @returns {T[]}
	 * @template T
	 */                    function d(e) {
                        var t = [];
                        var n = e.head.next;
                        while (n !== e.tail) {
                            t.push(n.value);
                            n = n.next;
                        }
                        return t;
                    }
                    if (!e.document) {
                        if (!e.addEventListener) {
                            // in Node.js
                            return s;
                        }
                        if (!s.disableWorkerMessageHandler) {
                            // In worker
                            e.addEventListener("message", (function(t) {
                                var n = JSON.parse(t.data);
                                var r = n.language;
                                var a = n.code;
                                var i = n.immediateClose;
                                e.postMessage(s.highlight(a, s.languages[r], r));
                                if (i) {
                                    e.close();
                                }
                            }), false);
                        }
                        return s;
                    }
                    // Get current script and highlight
                                        var g = s.util.currentScript();
                    if (g) {
                        s.filename = g.src;
                        if (g.hasAttribute("data-manual")) {
                            s.manual = true;
                        }
                    }
                    function f() {
                        if (!s.manual) {
                            s.highlightAll();
                        }
                    }
                    if (!s.manual) {
                        // If the document state is "loading", then we'll use DOMContentLoaded.
                        // If the document state is "interactive" and the prism.js script is deferred, then we'll also use the
                        // DOMContentLoaded event because there might be some plugins or languages which have also been deferred and they
                        // might take longer one animation frame to execute which can create a race condition where only some plugins have
                        // been loaded when Prism.highlightAll() is executed, depending on how fast resources are loaded.
                        // See https://github.com/PrismJS/prism/issues/2102
                        var m = document.readyState;
                        if (m === "loading" || m === "interactive" && g && g.defer) {
                            document.addEventListener("DOMContentLoaded", f);
                        } else {
                            if (t.requestAnimationFrame) {
                                t.requestAnimationFrame(f);
                            } else {
                                t.setTimeout(f, 16);
                            }
                        }
                    }
                    return s;
                }(r);
                if (true && e.exports) {
                    e.exports = a;
                }
                // hack for components to work correctly in node.js
                                if (typeof n !== "undefined") {
                    n.Prism = a;
                }
                // some additional documentation/types
                /**
 * The expansion of a simple `RegExp` literal to support additional properties.
 *
 * @typedef GrammarToken
 * @property {RegExp} pattern The regular expression of the token.
 * @property {boolean} [lookbehind=false] If `true`, then the first capturing group of `pattern` will (effectively)
 * behave as a lookbehind group meaning that the captured text will not be part of the matched text of the new token.
 * @property {boolean} [greedy=false] Whether the token is greedy.
 * @property {string|string[]} [alias] An optional alias or list of aliases.
 * @property {Grammar} [inside] The nested grammar of this token.
 *
 * The `inside` grammar will be used to tokenize the text value of each token of this kind.
 *
 * This can be used to make nested and even recursive language definitions.
 *
 * Note: This can cause infinite recursion. Be careful when you embed different languages or even the same language into
 * each another.
 * @global
 * @public
 */
                /**
 * @typedef Grammar
 * @type {Object<string, RegExp | GrammarToken | Array<RegExp | GrammarToken>>}
 * @property {Grammar} [rest] An optional grammar object that will be appended to this grammar.
 * @global
 * @public
 */
                /**
 * A function which will invoked after an element was successfully highlighted.
 *
 * @callback HighlightCallback
 * @param {Element} element The element successfully highlighted.
 * @returns {void}
 * @global
 * @public
 */
                /**
 * @callback HookCallback
 * @param {Object<string, any>} env The environment variables of the hook.
 * @returns {void}
 * @global
 * @public
 */
                // https://www.json.org/json-en.html
                                a.languages.json = {
                    property: {
                        pattern: /(^|[^\\])"(?:\\.|[^\\"\r\n])*"(?=\s*:)/,
                        lookbehind: true,
                        greedy: true
                    },
                    string: {
                        pattern: /(^|[^\\])"(?:\\.|[^\\"\r\n])*"(?!\s*:)/,
                        lookbehind: true,
                        greedy: true
                    },
                    comment: {
                        pattern: /\/\/.*|\/\*[\s\S]*?(?:\*\/|$)/,
                        greedy: true
                    },
                    number: /-?\b\d+(?:\.\d+)?(?:e[+-]?\d+)?\b/i,
                    punctuation: /[{}[\],]/,
                    operator: /:/,
                    boolean: /\b(?:false|true)\b/,
                    null: {
                        pattern: /\bnull\b/,
                        alias: "keyword"
                    }
                };
                a.languages.webmanifest = a.languages.json;
                a.languages.markup = {
                    comment: {
                        pattern: /<!--(?:(?!<!--)[\s\S])*?-->/,
                        greedy: true
                    },
                    prolog: {
                        pattern: /<\?[\s\S]+?\?>/,
                        greedy: true
                    },
                    doctype: {
                        // https://www.w3.org/TR/xml/#NT-doctypedecl
                        pattern: /<!DOCTYPE(?:[^>"'[\]]|"[^"]*"|'[^']*')+(?:\[(?:[^<"'\]]|"[^"]*"|'[^']*'|<(?!!--)|<!--(?:[^-]|-(?!->))*-->)*\]\s*)?>/i,
                        greedy: true,
                        inside: {
                            "internal-subset": {
                                pattern: /(^[^\[]*\[)[\s\S]+(?=\]>$)/,
                                lookbehind: true,
                                greedy: true,
                                inside: null
                            },
                            string: {
                                pattern: /"[^"]*"|'[^']*'/,
                                greedy: true
                            },
                            punctuation: /^<!|>$|[[\]]/,
                            "doctype-tag": /^DOCTYPE/i,
                            name: /[^\s<>'"]+/
                        }
                    },
                    cdata: {
                        pattern: /<!\[CDATA\[[\s\S]*?\]\]>/i,
                        greedy: true
                    },
                    tag: {
                        pattern: /<\/?(?!\d)[^\s>\/=$<%]+(?:\s(?:\s*[^\s>\/=]+(?:\s*=\s*(?:"[^"]*"|'[^']*'|[^\s'">=]+(?=[\s>]))|(?=[\s/>])))+)?\s*\/?>/,
                        greedy: true,
                        inside: {
                            tag: {
                                pattern: /^<\/?[^\s>\/]+/,
                                inside: {
                                    punctuation: /^<\/?/,
                                    namespace: /^[^\s>\/:]+:/
                                }
                            },
                            "special-attr": [],
                            "attr-value": {
                                pattern: /=\s*(?:"[^"]*"|'[^']*'|[^\s'">=]+)/,
                                inside: {
                                    punctuation: [ {
                                        pattern: /^=/,
                                        alias: "attr-equals"
                                    }, {
                                        pattern: /^(\s*)["']|["']$/,
                                        lookbehind: true
                                    } ]
                                }
                            },
                            punctuation: /\/?>/,
                            "attr-name": {
                                pattern: /[^\s>\/]+/,
                                inside: {
                                    namespace: /^[^\s>\/:]+:/
                                }
                            }
                        }
                    },
                    entity: [ {
                        pattern: /&[\da-z]{1,8};/i,
                        alias: "named-entity"
                    }, /&#x?[\da-f]{1,8};/i ]
                };
                a.languages.markup["tag"].inside["attr-value"].inside["entity"] = a.languages.markup["entity"];
                a.languages.markup["doctype"].inside["internal-subset"].inside = a.languages.markup;
                // Plugin to make entity title show the real entity, idea by Roman Komarov
                                a.hooks.add("wrap", (function(e) {
                    if (e.type === "entity") {
                        e.attributes["title"] = e.content.replace(/&amp;/, "&");
                    }
                }));
                Object.defineProperty(a.languages.markup.tag, "addInlined", {
                    /**
	 * Adds an inlined language to markup.
	 *
	 * An example of an inlined language is CSS with `<style>` tags.
	 *
	 * @param {string} tagName The name of the tag that contains the inlined language. This name will be treated as
	 * case insensitive.
	 * @param {string} lang The language key.
	 * @example
	 * addInlined('style', 'css');
	 */
                    value: function e(t, n) {
                        var r = {};
                        r["language-" + n] = {
                            pattern: /(^<!\[CDATA\[)[\s\S]+?(?=\]\]>$)/i,
                            lookbehind: true,
                            inside: a.languages[n]
                        };
                        r["cdata"] = /^<!\[CDATA\[|\]\]>$/i;
                        var s = {
                            "included-cdata": {
                                pattern: /<!\[CDATA\[[\s\S]*?\]\]>/i,
                                inside: r
                            }
                        };
                        s["language-" + n] = {
                            pattern: /[\s\S]+/,
                            inside: a.languages[n]
                        };
                        var i = {};
                        i[t] = {
                            pattern: RegExp(/(<__[^>]*>)(?:<!\[CDATA\[(?:[^\]]|\](?!\]>))*\]\]>|(?!<!\[CDATA\[)[\s\S])*?(?=<\/__>)/.source.replace(/__/g, (function() {
                                return t;
                            })), "i"),
                            lookbehind: true,
                            greedy: true,
                            inside: s
                        };
                        a.languages.insertBefore("markup", "cdata", i);
                    }
                });
                Object.defineProperty(a.languages.markup.tag, "addAttribute", {
                    /**
	 * Adds an pattern to highlight languages embedded in HTML attributes.
	 *
	 * An example of an inlined language is CSS with `style` attributes.
	 *
	 * @param {string} attrName The name of the tag that contains the inlined language. This name will be treated as
	 * case insensitive.
	 * @param {string} lang The language key.
	 * @example
	 * addAttribute('style', 'css');
	 */
                    value: function(e, t) {
                        a.languages.markup.tag.inside["special-attr"].push({
                            pattern: RegExp(/(^|["'\s])/.source + "(?:" + e + ")" + /\s*=\s*(?:"[^"]*"|'[^']*'|[^\s'">=]+(?=[\s>]))/.source, "i"),
                            lookbehind: true,
                            inside: {
                                "attr-name": /^[^\s=]+/,
                                "attr-value": {
                                    pattern: /=[\s\S]+/,
                                    inside: {
                                        value: {
                                            pattern: /(^=\s*(["']|(?!["'])))\S[\s\S]*(?=\2$)/,
                                            lookbehind: true,
                                            alias: [ t, "language-" + t ],
                                            inside: a.languages[t]
                                        },
                                        punctuation: [ {
                                            pattern: /^=/,
                                            alias: "attr-equals"
                                        }, /"|'/ ]
                                    }
                                }
                            }
                        });
                    }
                });
                a.languages.html = a.languages.markup;
                a.languages.mathml = a.languages.markup;
                a.languages.svg = a.languages.markup;
                a.languages.xml = a.languages.extend("markup", {});
                a.languages.ssml = a.languages.xml;
                a.languages.atom = a.languages.xml;
                a.languages.rss = a.languages.xml;
                a.languages.clike = {
                    comment: [ {
                        pattern: /(^|[^\\])\/\*[\s\S]*?(?:\*\/|$)/,
                        lookbehind: true,
                        greedy: true
                    }, {
                        pattern: /(^|[^\\:])\/\/.*/,
                        lookbehind: true,
                        greedy: true
                    } ],
                    string: {
                        pattern: /(["'])(?:\\(?:\r\n|[\s\S])|(?!\1)[^\\\r\n])*\1/,
                        greedy: true
                    },
                    "class-name": {
                        pattern: /(\b(?:class|extends|implements|instanceof|interface|new|trait)\s+|\bcatch\s+\()[\w.\\]+/i,
                        lookbehind: true,
                        inside: {
                            punctuation: /[.\\]/
                        }
                    },
                    keyword: /\b(?:break|catch|continue|do|else|finally|for|function|if|in|instanceof|new|null|return|throw|try|while)\b/,
                    boolean: /\b(?:false|true)\b/,
                    function: /\b\w+(?=\()/,
                    number: /\b0x[\da-f]+\b|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:e[+-]?\d+)?/i,
                    operator: /[<>]=?|[!=]=?=?|--?|\+\+?|&&?|\|\|?|[?*/~^%]/,
                    punctuation: /[{}[\];(),.:]/
                };
                a.languages.javascript = a.languages.extend("clike", {
                    "class-name": [ a.languages.clike["class-name"], {
                        pattern: /(^|[^$\w\xA0-\uFFFF])(?!\s)[_$A-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\.(?:constructor|prototype))/,
                        lookbehind: true
                    } ],
                    keyword: [ {
                        pattern: /((?:^|\})\s*)catch\b/,
                        lookbehind: true
                    }, {
                        pattern: /(^|[^.]|\.\.\.\s*)\b(?:as|assert(?=\s*\{)|async(?=\s*(?:function\b|\(|[$\w\xA0-\uFFFF]|$))|await|break|case|class|const|continue|debugger|default|delete|do|else|enum|export|extends|finally(?=\s*(?:\{|$))|for|from(?=\s*(?:['"]|$))|function|(?:get|set)(?=\s*(?:[#\[$\w\xA0-\uFFFF]|$))|if|implements|import|in|instanceof|interface|let|new|null|of|package|private|protected|public|return|static|super|switch|this|throw|try|typeof|undefined|var|void|while|with|yield)\b/,
                        lookbehind: true
                    } ],
                    // Allow for all non-ASCII characters (See http://stackoverflow.com/a/2008444)
                    function: /#?(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*(?:\.\s*(?:apply|bind|call)\s*)?\()/,
                    number: {
                        pattern: RegExp(/(^|[^\w$])/.source + "(?:" + (
                        // constant
                        /NaN|Infinity/.source + "|" + 
                        // binary integer
                        /0[bB][01]+(?:_[01]+)*n?/.source + "|" + 
                        // octal integer
                        /0[oO][0-7]+(?:_[0-7]+)*n?/.source + "|" + 
                        // hexadecimal integer
                        /0[xX][\dA-Fa-f]+(?:_[\dA-Fa-f]+)*n?/.source + "|" + 
                        // decimal bigint
                        /\d+(?:_\d+)*n/.source + "|" + 
                        // decimal number (integer or float) but no bigint
                        /(?:\d+(?:_\d+)*(?:\.(?:\d+(?:_\d+)*)?)?|\.\d+(?:_\d+)*)(?:[Ee][+-]?\d+(?:_\d+)*)?/.source) + ")" + /(?![\w$])/.source),
                        lookbehind: true
                    },
                    operator: /--|\+\+|\*\*=?|=>|&&=?|\|\|=?|[!=]==|<<=?|>>>?=?|[-+*/%&|^!=<>]=?|\.{3}|\?\?=?|\?\.?|[~:]/
                });
                a.languages.javascript["class-name"][0].pattern = /(\b(?:class|extends|implements|instanceof|interface|new)\s+)[\w.\\]+/;
                a.languages.insertBefore("javascript", "keyword", {
                    regex: {
                        pattern: RegExp(
                        // lookbehind
                        // eslint-disable-next-line regexp/no-dupe-characters-character-class
                        /((?:^|[^$\w\xA0-\uFFFF."'\])\s]|\b(?:return|yield))\s*)/.source + 
                        // Regex pattern:
                        // There are 2 regex patterns here. The RegExp set notation proposal added support for nested character
                        // classes if the `v` flag is present. Unfortunately, nested CCs are both context-free and incompatible
                        // with the only syntax, so we have to define 2 different regex patterns.
                        /\//.source + "(?:" + /(?:\[(?:[^\]\\\r\n]|\\.)*\]|\\.|[^/\\\[\r\n])+\/[dgimyus]{0,7}/.source + "|" + 
                        // `v` flag syntax. This supports 3 levels of nested character classes.
                        /(?:\[(?:[^[\]\\\r\n]|\\.|\[(?:[^[\]\\\r\n]|\\.|\[(?:[^[\]\\\r\n]|\\.)*\])*\])*\]|\\.|[^/\\\[\r\n])+\/[dgimyus]{0,7}v[dgimyus]{0,7}/.source + ")" + 
                        // lookahead
                        /(?=(?:\s|\/\*(?:[^*]|\*(?!\/))*\*\/)*(?:$|[\r\n,.;:})\]]|\/\/))/.source),
                        lookbehind: true,
                        greedy: true,
                        inside: {
                            "regex-source": {
                                pattern: /^(\/)[\s\S]+(?=\/[a-z]*$)/,
                                lookbehind: true,
                                alias: "language-regex",
                                inside: a.languages.regex
                            },
                            "regex-delimiter": /^\/|\/$/,
                            "regex-flags": /^[a-z]+$/
                        }
                    },
                    // This must be declared before keyword because we use "function" inside the look-forward
                    "function-variable": {
                        pattern: /#?(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*[=:]\s*(?:async\s*)?(?:\bfunction\b|(?:\((?:[^()]|\([^()]*\))*\)|(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*)\s*=>))/,
                        alias: "function"
                    },
                    parameter: [ {
                        pattern: /(function(?:\s+(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*)?\s*\(\s*)(?!\s)(?:[^()\s]|\s+(?![\s)])|\([^()]*\))+(?=\s*\))/,
                        lookbehind: true,
                        inside: a.languages.javascript
                    }, {
                        pattern: /(^|[^$\w\xA0-\uFFFF])(?!\s)[_$a-z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*=>)/i,
                        lookbehind: true,
                        inside: a.languages.javascript
                    }, {
                        pattern: /(\(\s*)(?!\s)(?:[^()\s]|\s+(?![\s)])|\([^()]*\))+(?=\s*\)\s*=>)/,
                        lookbehind: true,
                        inside: a.languages.javascript
                    }, {
                        pattern: /((?:\b|\s|^)(?!(?:as|async|await|break|case|catch|class|const|continue|debugger|default|delete|do|else|enum|export|extends|finally|for|from|function|get|if|implements|import|in|instanceof|interface|let|new|null|of|package|private|protected|public|return|set|static|super|switch|this|throw|try|typeof|undefined|var|void|while|with|yield)(?![$\w\xA0-\uFFFF]))(?:(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*\s*)\(\s*|\]\s*\(\s*)(?!\s)(?:[^()\s]|\s+(?![\s)])|\([^()]*\))+(?=\s*\)\s*\{)/,
                        lookbehind: true,
                        inside: a.languages.javascript
                    } ],
                    constant: /\b[A-Z](?:[A-Z_]|\dx?)*\b/
                });
                a.languages.insertBefore("javascript", "string", {
                    hashbang: {
                        pattern: /^#!.*/,
                        greedy: true,
                        alias: "comment"
                    },
                    "template-string": {
                        pattern: /`(?:\\[\s\S]|\$\{(?:[^{}]|\{(?:[^{}]|\{[^}]*\})*\})+\}|(?!\$\{)[^\\`])*`/,
                        greedy: true,
                        inside: {
                            "template-punctuation": {
                                pattern: /^`|`$/,
                                alias: "string"
                            },
                            interpolation: {
                                pattern: /((?:^|[^\\])(?:\\{2})*)\$\{(?:[^{}]|\{(?:[^{}]|\{[^}]*\})*\})+\}/,
                                lookbehind: true,
                                inside: {
                                    "interpolation-punctuation": {
                                        pattern: /^\$\{|\}$/,
                                        alias: "punctuation"
                                    },
                                    rest: a.languages.javascript
                                }
                            },
                            string: /[\s\S]+/
                        }
                    },
                    "string-property": {
                        pattern: /((?:^|[,{])[ \t]*)(["'])(?:\\(?:\r\n|[\s\S])|(?!\2)[^\\\r\n])*\2(?=\s*:)/m,
                        lookbehind: true,
                        greedy: true,
                        alias: "property"
                    }
                });
                a.languages.insertBefore("javascript", "operator", {
                    "literal-property": {
                        pattern: /((?:^|[,{])[ \t]*)(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*:)/m,
                        lookbehind: true,
                        alias: "property"
                    }
                });
                if (a.languages.markup) {
                    a.languages.markup.tag.addInlined("script", "javascript");
                    // add attribute support for all DOM events.
                    // https://developer.mozilla.org/en-US/docs/Web/Events#Standard_events
                                        a.languages.markup.tag.addAttribute(/on(?:abort|blur|change|click|composition(?:end|start|update)|dblclick|error|focus(?:in|out)?|key(?:down|up)|load|mouse(?:down|enter|leave|move|out|over|up)|reset|resize|scroll|select|slotchange|submit|unload|wheel)/.source, "javascript");
                }
                a.languages.js = a.languages.javascript;
                (function(e) {
                    var t = e.util.clone(e.languages.javascript);
                    var n = /(?:\s|\/\/.*(?!.)|\/\*(?:[^*]|\*(?!\/))\*\/)/.source;
                    var r = /(?:\{(?:\{(?:\{[^{}]*\}|[^{}])*\}|[^{}])*\})/.source;
                    var a = /(?:\{<S>*\.{3}(?:[^{}]|<BRACES>)*\})/.source;
                    /**
	 * @param {string} source
	 * @param {string} [flags]
	 */                    function s(e, t) {
                        e = e.replace(/<S>/g, (function() {
                            return n;
                        })).replace(/<BRACES>/g, (function() {
                            return r;
                        })).replace(/<SPREAD>/g, (function() {
                            return a;
                        }));
                        return RegExp(e, t);
                    }
                    a = s(a).source;
                    e.languages.jsx = e.languages.extend("markup", t);
                    e.languages.jsx.tag.pattern = s(/<\/?(?:[\w.:-]+(?:<S>+(?:[\w.:$-]+(?:=(?:"(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*'|[^\s{'"/>=]+|<BRACES>))?|<SPREAD>))*<S>*\/?)?>/.source);
                    e.languages.jsx.tag.inside["tag"].pattern = /^<\/?[^\s>\/]*/;
                    e.languages.jsx.tag.inside["attr-value"].pattern = /=(?!\{)(?:"(?:\\[\s\S]|[^\\"])*"|'(?:\\[\s\S]|[^\\'])*'|[^\s'">]+)/;
                    e.languages.jsx.tag.inside["tag"].inside["class-name"] = /^[A-Z]\w*(?:\.[A-Z]\w*)*$/;
                    e.languages.jsx.tag.inside["comment"] = t["comment"];
                    e.languages.insertBefore("inside", "attr-name", {
                        spread: {
                            pattern: s(/<SPREAD>/.source),
                            inside: e.languages.jsx
                        }
                    }, e.languages.jsx.tag);
                    e.languages.insertBefore("inside", "special-attr", {
                        script: {
                            // Allow for two levels of nesting
                            pattern: s(/=<BRACES>/.source),
                            alias: "language-javascript",
                            inside: {
                                "script-punctuation": {
                                    pattern: /^=(?=\{)/,
                                    alias: "punctuation"
                                },
                                rest: e.languages.jsx
                            }
                        }
                    }, e.languages.jsx.tag);
                    // The following will handle plain text inside tags
                                        var i = function(e) {
                        if (!e) {
                            return "";
                        }
                        if (typeof e === "string") {
                            return e;
                        }
                        if (typeof e.content === "string") {
                            return e.content;
                        }
                        return e.content.map(i).join("");
                    };
                    var o = function(t) {
                        var n = [];
                        for (var r = 0; r < t.length; r++) {
                            var a = t[r];
                            var s = false;
                            if (typeof a !== "string") {
                                if (a.type === "tag" && a.content[0] && a.content[0].type === "tag") {
                                    // We found a tag, now find its kind
                                    if (a.content[0].content[0].content === "</") {
                                        // Closing tag
                                        if (n.length > 0 && n[n.length - 1].tagName === i(a.content[0].content[1])) {
                                            // Pop matching opening tag
                                            n.pop();
                                        }
                                    } else {
                                        if (a.content[a.content.length - 1].content === "/>") {
                                            // Autoclosed tag, ignore
                                        } else {
                                            // Opening tag
                                            n.push({
                                                tagName: i(a.content[0].content[1]),
                                                openedBraces: 0
                                            });
                                        }
                                    }
                                } else if (n.length > 0 && a.type === "punctuation" && a.content === "{") {
                                    // Here we might have entered a JSX context inside a tag
                                    n[n.length - 1].openedBraces++;
                                } else if (n.length > 0 && n[n.length - 1].openedBraces > 0 && a.type === "punctuation" && a.content === "}") {
                                    // Here we might have left a JSX context inside a tag
                                    n[n.length - 1].openedBraces--;
                                } else {
                                    s = true;
                                }
                            }
                            if (s || typeof a === "string") {
                                if (n.length > 0 && n[n.length - 1].openedBraces === 0) {
                                    // Here we are inside a tag, and not inside a JSX context.
                                    // That's plain text: drop any tokens matched.
                                    var l = i(a);
                                    // And merge text with adjacent text
                                                                        if (r < t.length - 1 && (typeof t[r + 1] === "string" || t[r + 1].type === "plain-text")) {
                                        l += i(t[r + 1]);
                                        t.splice(r + 1, 1);
                                    }
                                    if (r > 0 && (typeof t[r - 1] === "string" || t[r - 1].type === "plain-text")) {
                                        l = i(t[r - 1]) + l;
                                        t.splice(r - 1, 1);
                                        r--;
                                    }
                                    t[r] = new e.Token("plain-text", l, null, l);
                                }
                            }
                            if (a.content && typeof a.content !== "string") {
                                o(a.content);
                            }
                        }
                    };
                    e.hooks.add("after-tokenize", (function(e) {
                        if (e.language !== "jsx" && e.language !== "tsx") {
                            return;
                        }
                        o(e.tokens);
                    }));
                })(a);
                (function(e) {
                    e.languages.typescript = e.languages.extend("javascript", {
                        "class-name": {
                            pattern: /(\b(?:class|extends|implements|instanceof|interface|new|type)\s+)(?!keyof\b)(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?:\s*<(?:[^<>]|<(?:[^<>]|<[^<>]*>)*>)*>)?/,
                            lookbehind: true,
                            greedy: true,
                            inside: null
                        },
                        builtin: /\b(?:Array|Function|Promise|any|boolean|console|never|number|string|symbol|unknown)\b/
                    });
                    // The keywords TypeScript adds to JavaScript
                                        e.languages.typescript.keyword.push(/\b(?:abstract|declare|is|keyof|readonly|require)\b/, 
                    // keywords that have to be followed by an identifier
                    /\b(?:asserts|infer|interface|module|namespace|type)\b(?=\s*(?:[{_$a-zA-Z\xA0-\uFFFF]|$))/, 
                    // This is for `import type *, {}`
                    /\btype\b(?=\s*(?:[\{*]|$))/);
                    // doesn't work with TS because TS is too complex
                                        delete e.languages.typescript["parameter"];
                    delete e.languages.typescript["literal-property"];
                    // a version of typescript specifically for highlighting types
                                        var t = e.languages.extend("typescript", {});
                    delete t["class-name"];
                    e.languages.typescript["class-name"].inside = t;
                    e.languages.insertBefore("typescript", "function", {
                        decorator: {
                            pattern: /@[$\w\xA0-\uFFFF]+/,
                            inside: {
                                at: {
                                    pattern: /^@/,
                                    alias: "operator"
                                },
                                function: /^[\s\S]+/
                            }
                        },
                        "generic-function": {
                            // e.g. foo<T extends "bar" | "baz">( ...
                            pattern: /#?(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*\s*<(?:[^<>]|<(?:[^<>]|<[^<>]*>)*>)*>(?=\s*\()/,
                            greedy: true,
                            inside: {
                                function: /^#?(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*/,
                                generic: {
                                    pattern: /<[\s\S]+/,
                                    // everything after the first <
                                    alias: "class-name",
                                    inside: t
                                }
                            }
                        }
                    });
                    e.languages.ts = e.languages.typescript;
                })(a);
                (function(e) {
                    var t = e.util.clone(e.languages.typescript);
                    e.languages.tsx = e.languages.extend("jsx", t);
                    // doesn't work with TS because TS is too complex
                                        delete e.languages.tsx["parameter"];
                    delete e.languages.tsx["literal-property"];
                    // This will prevent collisions between TSX tags and TS generic types.
                    // Idea by https://github.com/karlhorky
                    // Discussion: https://github.com/PrismJS/prism/issues/2594#issuecomment-710666928
                                        var n = e.languages.tsx.tag;
                    n.pattern = RegExp(/(^|[^\w$]|(?=<\/))/.source + "(?:" + n.pattern.source + ")", n.pattern.flags);
                    n.lookbehind = true;
                })(a);
                (function(e) {
                    var t = /(?:"(?:\\(?:\r\n|[\s\S])|[^"\\\r\n])*"|'(?:\\(?:\r\n|[\s\S])|[^'\\\r\n])*')/;
                    e.languages.css = {
                        comment: /\/\*[\s\S]*?\*\//,
                        atrule: {
                            pattern: RegExp("@[\\w-](?:" + /[^;{\s"']|\s+(?!\s)/.source + "|" + t.source + ")*?" + /(?:;|(?=\s*\{))/.source),
                            inside: {
                                rule: /^@[\w-]+/,
                                "selector-function-argument": {
                                    pattern: /(\bselector\s*\(\s*(?![\s)]))(?:[^()\s]|\s+(?![\s)])|\((?:[^()]|\([^()]*\))*\))+(?=\s*\))/,
                                    lookbehind: true,
                                    alias: "selector"
                                },
                                keyword: {
                                    pattern: /(^|[^\w-])(?:and|not|only|or)(?![\w-])/,
                                    lookbehind: true
                                }
                            }
                        },
                        url: {
                            // https://drafts.csswg.org/css-values-3/#urls
                            pattern: RegExp("\\burl\\((?:" + t.source + "|" + /(?:[^\\\r\n()"']|\\[\s\S])*/.source + ")\\)", "i"),
                            greedy: true,
                            inside: {
                                function: /^url/i,
                                punctuation: /^\(|\)$/,
                                string: {
                                    pattern: RegExp("^" + t.source + "$"),
                                    alias: "url"
                                }
                            }
                        },
                        selector: {
                            pattern: RegExp("(^|[{}\\s])[^{}\\s](?:[^{};\"'\\s]|\\s+(?![\\s{])|" + t.source + ")*(?=\\s*\\{)"),
                            lookbehind: true
                        },
                        string: {
                            pattern: t,
                            greedy: true
                        },
                        property: {
                            pattern: /(^|[^-\w\xA0-\uFFFF])(?!\s)[-_a-z\xA0-\uFFFF](?:(?!\s)[-\w\xA0-\uFFFF])*(?=\s*:)/i,
                            lookbehind: true
                        },
                        important: /!important\b/i,
                        function: {
                            pattern: /(^|[^-a-z0-9])[-a-z0-9]+(?=\()/i,
                            lookbehind: true
                        },
                        punctuation: /[(){};:,]/
                    };
                    e.languages.css["atrule"].inside.rest = e.languages.css;
                    var n = e.languages.markup;
                    if (n) {
                        n.tag.addInlined("style", "css");
                        n.tag.addAttribute("style", "css");
                    }
                })(a);
                (function(e) {
                    // $ set | grep '^[A-Z][^[:space:]]*=' | cut -d= -f1 | tr '\n' '|'
                    // + LC_ALL, RANDOM, REPLY, SECONDS.
                    // + make sure PS1..4 are here as they are not always set,
                    // - some useless things.
                    var t = "\\b(?:BASH|BASHOPTS|BASH_ALIASES|BASH_ARGC|BASH_ARGV|BASH_CMDS|BASH_COMPLETION_COMPAT_DIR|BASH_LINENO|BASH_REMATCH|BASH_SOURCE|BASH_VERSINFO|BASH_VERSION|COLORTERM|COLUMNS|COMP_WORDBREAKS|DBUS_SESSION_BUS_ADDRESS|DEFAULTS_PATH|DESKTOP_SESSION|DIRSTACK|DISPLAY|EUID|GDMSESSION|GDM_LANG|GNOME_KEYRING_CONTROL|GNOME_KEYRING_PID|GPG_AGENT_INFO|GROUPS|HISTCONTROL|HISTFILE|HISTFILESIZE|HISTSIZE|HOME|HOSTNAME|HOSTTYPE|IFS|INSTANCE|JOB|LANG|LANGUAGE|LC_ADDRESS|LC_ALL|LC_IDENTIFICATION|LC_MEASUREMENT|LC_MONETARY|LC_NAME|LC_NUMERIC|LC_PAPER|LC_TELEPHONE|LC_TIME|LESSCLOSE|LESSOPEN|LINES|LOGNAME|LS_COLORS|MACHTYPE|MAILCHECK|MANDATORY_PATH|NO_AT_BRIDGE|OLDPWD|OPTERR|OPTIND|ORBIT_SOCKETDIR|OSTYPE|PAPERSIZE|PATH|PIPESTATUS|PPID|PS1|PS2|PS3|PS4|PWD|RANDOM|REPLY|SECONDS|SELINUX_INIT|SESSION|SESSIONTYPE|SESSION_MANAGER|SHELL|SHELLOPTS|SHLVL|SSH_AUTH_SOCK|TERM|UID|UPSTART_EVENTS|UPSTART_INSTANCE|UPSTART_JOB|UPSTART_SESSION|USER|WINDOWID|XAUTHORITY|XDG_CONFIG_DIRS|XDG_CURRENT_DESKTOP|XDG_DATA_DIRS|XDG_GREETER_DATA_DIR|XDG_MENU_PREFIX|XDG_RUNTIME_DIR|XDG_SEAT|XDG_SEAT_PATH|XDG_SESSION_DESKTOP|XDG_SESSION_ID|XDG_SESSION_PATH|XDG_SESSION_TYPE|XDG_VTNR|XMODIFIERS)\\b";
                    var n = {
                        pattern: /(^(["']?)\w+\2)[ \t]+\S.*/,
                        lookbehind: true,
                        alias: "punctuation",
                        // this looks reasonably well in all themes
                        inside: null
                    };
                    var r = {
                        bash: n,
                        environment: {
                            pattern: RegExp("\\$" + t),
                            alias: "constant"
                        },
                        variable: [ 
                        // [0]: Arithmetic Environment
                        {
                            pattern: /\$?\(\([\s\S]+?\)\)/,
                            greedy: true,
                            inside: {
                                // If there is a $ sign at the beginning highlight $(( and )) as variable
                                variable: [ {
                                    pattern: /(^\$\(\([\s\S]+)\)\)/,
                                    lookbehind: true
                                }, /^\$\(\(/ ],
                                number: /\b0x[\dA-Fa-f]+\b|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:[Ee]-?\d+)?/,
                                // Operators according to https://www.gnu.org/software/bash/manual/bashref.html#Shell-Arithmetic
                                operator: /--|\+\+|\*\*=?|<<=?|>>=?|&&|\|\||[=!+\-*/%<>^&|]=?|[?~:]/,
                                // If there is no $ sign at the beginning highlight (( and )) as punctuation
                                punctuation: /\(\(?|\)\)?|,|;/
                            }
                        }, 
                        // [1]: Command Substitution
                        {
                            pattern: /\$\((?:\([^)]+\)|[^()])+\)|`[^`]+`/,
                            greedy: true,
                            inside: {
                                variable: /^\$\(|^`|\)$|`$/
                            }
                        }, 
                        // [2]: Brace expansion
                        {
                            pattern: /\$\{[^}]+\}/,
                            greedy: true,
                            inside: {
                                operator: /:[-=?+]?|[!\/]|##?|%%?|\^\^?|,,?/,
                                punctuation: /[\[\]]/,
                                environment: {
                                    pattern: RegExp("(\\{)" + t),
                                    lookbehind: true,
                                    alias: "constant"
                                }
                            }
                        }, /\$(?:\w+|[#?*!@$])/ ],
                        // Escape sequences from echo and printf's manuals, and escaped quotes.
                        entity: /\\(?:[abceEfnrtv\\"]|O?[0-7]{1,3}|U[0-9a-fA-F]{8}|u[0-9a-fA-F]{4}|x[0-9a-fA-F]{1,2})/
                    };
                    e.languages.bash = {
                        shebang: {
                            pattern: /^#!\s*\/.*/,
                            alias: "important"
                        },
                        comment: {
                            pattern: /(^|[^"{\\$])#.*/,
                            lookbehind: true
                        },
                        "function-name": [ 
                        // a) function foo {
                        // b) foo() {
                        // c) function foo() {
                        // but not “foo {”
                        {
                            // a) and c)
                            pattern: /(\bfunction\s+)[\w-]+(?=(?:\s*\(?:\s*\))?\s*\{)/,
                            lookbehind: true,
                            alias: "function"
                        }, {
                            // b)
                            pattern: /\b[\w-]+(?=\s*\(\s*\)\s*\{)/,
                            alias: "function"
                        } ],
                        // Highlight variable names as variables in for and select beginnings.
                        "for-or-select": {
                            pattern: /(\b(?:for|select)\s+)\w+(?=\s+in\s)/,
                            alias: "variable",
                            lookbehind: true
                        },
                        // Highlight variable names as variables in the left-hand part
                        // of assignments (“=” and “+=”).
                        "assign-left": {
                            pattern: /(^|[\s;|&]|[<>]\()\w+(?:\.\w+)*(?=\+?=)/,
                            inside: {
                                environment: {
                                    pattern: RegExp("(^|[\\s;|&]|[<>]\\()" + t),
                                    lookbehind: true,
                                    alias: "constant"
                                }
                            },
                            alias: "variable",
                            lookbehind: true
                        },
                        // Highlight parameter names as variables
                        parameter: {
                            pattern: /(^|\s)-{1,2}(?:\w+:[+-]?)?\w+(?:\.\w+)*(?=[=\s]|$)/,
                            alias: "variable",
                            lookbehind: true
                        },
                        string: [ 
                        // Support for Here-documents https://en.wikipedia.org/wiki/Here_document
                        {
                            pattern: /((?:^|[^<])<<-?\s*)(\w+)\s[\s\S]*?(?:\r?\n|\r)\2/,
                            lookbehind: true,
                            greedy: true,
                            inside: r
                        }, 
                        // Here-document with quotes around the tag
                        // → No expansion (so no “inside”).
                        {
                            pattern: /((?:^|[^<])<<-?\s*)(["'])(\w+)\2\s[\s\S]*?(?:\r?\n|\r)\3/,
                            lookbehind: true,
                            greedy: true,
                            inside: {
                                bash: n
                            }
                        }, 
                        // “Normal” string
                        {
                            // https://www.gnu.org/software/bash/manual/html_node/Double-Quotes.html
                            pattern: /(^|[^\\](?:\\\\)*)"(?:\\[\s\S]|\$\([^)]+\)|\$(?!\()|`[^`]+`|[^"\\`$])*"/,
                            lookbehind: true,
                            greedy: true,
                            inside: r
                        }, {
                            // https://www.gnu.org/software/bash/manual/html_node/Single-Quotes.html
                            pattern: /(^|[^$\\])'[^']*'/,
                            lookbehind: true,
                            greedy: true
                        }, {
                            // https://www.gnu.org/software/bash/manual/html_node/ANSI_002dC-Quoting.html
                            pattern: /\$'(?:[^'\\]|\\[\s\S])*'/,
                            greedy: true,
                            inside: {
                                entity: r.entity
                            }
                        } ],
                        environment: {
                            pattern: RegExp("\\$?" + t),
                            alias: "constant"
                        },
                        variable: r.variable,
                        function: {
                            pattern: /(^|[\s;|&]|[<>]\()(?:add|apropos|apt|apt-cache|apt-get|aptitude|aspell|automysqlbackup|awk|basename|bash|bc|bconsole|bg|bzip2|cal|cargo|cat|cfdisk|chgrp|chkconfig|chmod|chown|chroot|cksum|clear|cmp|column|comm|composer|cp|cron|crontab|csplit|curl|cut|date|dc|dd|ddrescue|debootstrap|df|diff|diff3|dig|dir|dircolors|dirname|dirs|dmesg|docker|docker-compose|du|egrep|eject|env|ethtool|expand|expect|expr|fdformat|fdisk|fg|fgrep|file|find|fmt|fold|format|free|fsck|ftp|fuser|gawk|git|gparted|grep|groupadd|groupdel|groupmod|groups|grub-mkconfig|gzip|halt|head|hg|history|host|hostname|htop|iconv|id|ifconfig|ifdown|ifup|import|install|ip|java|jobs|join|kill|killall|less|link|ln|locate|logname|logrotate|look|lpc|lpr|lprint|lprintd|lprintq|lprm|ls|lsof|lynx|make|man|mc|mdadm|mkconfig|mkdir|mke2fs|mkfifo|mkfs|mkisofs|mknod|mkswap|mmv|more|most|mount|mtools|mtr|mutt|mv|nano|nc|netstat|nice|nl|node|nohup|notify-send|npm|nslookup|op|open|parted|passwd|paste|pathchk|ping|pkill|pnpm|podman|podman-compose|popd|pr|printcap|printenv|ps|pushd|pv|quota|quotacheck|quotactl|ram|rar|rcp|reboot|remsync|rename|renice|rev|rm|rmdir|rpm|rsync|scp|screen|sdiff|sed|sendmail|seq|service|sftp|sh|shellcheck|shuf|shutdown|sleep|slocate|sort|split|ssh|stat|strace|su|sudo|sum|suspend|swapon|sync|sysctl|tac|tail|tar|tee|time|timeout|top|touch|tr|traceroute|tsort|tty|umount|uname|unexpand|uniq|units|unrar|unshar|unzip|update-grub|uptime|useradd|userdel|usermod|users|uudecode|uuencode|v|vcpkg|vdir|vi|vim|virsh|vmstat|wait|watch|wc|wget|whereis|which|who|whoami|write|xargs|xdg-open|yarn|yes|zenity|zip|zsh|zypper)(?=$|[)\s;|&])/,
                            lookbehind: true
                        },
                        keyword: {
                            pattern: /(^|[\s;|&]|[<>]\()(?:case|do|done|elif|else|esac|fi|for|function|if|in|select|then|until|while)(?=$|[)\s;|&])/,
                            lookbehind: true
                        },
                        // https://www.gnu.org/software/bash/manual/html_node/Shell-Builtin-Commands.html
                        builtin: {
                            pattern: /(^|[\s;|&]|[<>]\()(?:\.|:|alias|bind|break|builtin|caller|cd|command|continue|declare|echo|enable|eval|exec|exit|export|getopts|hash|help|let|local|logout|mapfile|printf|pwd|read|readarray|readonly|return|set|shift|shopt|source|test|times|trap|type|typeset|ulimit|umask|unalias|unset)(?=$|[)\s;|&])/,
                            lookbehind: true,
                            // Alias added to make those easier to distinguish from strings.
                            alias: "class-name"
                        },
                        boolean: {
                            pattern: /(^|[\s;|&]|[<>]\()(?:false|true)(?=$|[)\s;|&])/,
                            lookbehind: true
                        },
                        "file-descriptor": {
                            pattern: /\B&\d\b/,
                            alias: "important"
                        },
                        operator: {
                            // Lots of redirections here, but not just that.
                            pattern: /\d?<>|>\||\+=|=[=~]?|!=?|<<[<-]?|[&\d]?>>|\d[<>]&?|[<>][&=]?|&[>&]?|\|[&|]?/,
                            inside: {
                                "file-descriptor": {
                                    pattern: /^\d/,
                                    alias: "important"
                                }
                            }
                        },
                        punctuation: /\$?\(\(?|\)\)?|\.\.|[{}[\];\\]/,
                        number: {
                            pattern: /(^|\s)(?:[1-9]\d*|0)(?:[.,]\d+)?\b/,
                            lookbehind: true
                        }
                    };
                    n.inside = e.languages.bash;
                    /* Patterns in command substitution. */                    var a = [ "comment", "function-name", "for-or-select", "assign-left", "parameter", "string", "environment", "function", "keyword", "builtin", "boolean", "file-descriptor", "operator", "punctuation", "number" ];
                    var s = r.variable[1].inside;
                    for (var i = 0; i < a.length; i++) {
                        s[a[i]] = e.languages.bash[a[i]];
                    }
                    e.languages.sh = e.languages.bash;
                    e.languages.shell = e.languages.bash;
                })(a);
                (function(e) {
                    // https://yaml.org/spec/1.2/spec.html#c-ns-anchor-property
                    // https://yaml.org/spec/1.2/spec.html#c-ns-alias-node
                    var t = /[*&][^\s[\]{},]+/;
                    // https://yaml.org/spec/1.2/spec.html#c-ns-tag-property
                                        var n = /!(?:<[\w\-%#;/?:@&=+$,.!~*'()[\]]+>|(?:[a-zA-Z\d-]*!)?[\w\-%#;/?:@&=+$.~*'()]+)?/;
                    // https://yaml.org/spec/1.2/spec.html#c-ns-properties(n,c)
                                        var r = "(?:" + n.source + "(?:[ \t]+" + t.source + ")?|" + t.source + "(?:[ \t]+" + n.source + ")?)";
                    // https://yaml.org/spec/1.2/spec.html#ns-plain(n,c)
                    // This is a simplified version that doesn't support "#" and multiline keys
                    // All these long scarry character classes are simplified versions of YAML's characters
                                        var a = /(?:[^\s\x00-\x08\x0e-\x1f!"#%&'*,\-:>?@[\]`{|}\x7f-\x84\x86-\x9f\ud800-\udfff\ufffe\uffff]|[?:-]<PLAIN>)(?:[ \t]*(?:(?![#:])<PLAIN>|:<PLAIN>))*/.source.replace(/<PLAIN>/g, (function() {
                        return /[^\s\x00-\x08\x0e-\x1f,[\]{}\x7f-\x84\x86-\x9f\ud800-\udfff\ufffe\uffff]/.source;
                    }));
                    var s = /"(?:[^"\\\r\n]|\\.)*"|'(?:[^'\\\r\n]|\\.)*'/.source;
                    /**
	 *
	 * @param {string} value
	 * @param {string} [flags]
	 * @returns {RegExp}
	 */                    function i(e, t) {
                        t = (t || "").replace(/m/g, "") + "m";
 // add m flag
                                                var n = /([:\-,[{]\s*(?:\s<<prop>>[ \t]+)?)(?:<<value>>)(?=[ \t]*(?:$|,|\]|\}|(?:[\r\n]\s*)?#))/.source.replace(/<<prop>>/g, (function() {
                            return r;
                        })).replace(/<<value>>/g, (function() {
                            return e;
                        }));
                        return RegExp(n, t);
                    }
                    e.languages.yaml = {
                        scalar: {
                            pattern: RegExp(/([\-:]\s*(?:\s<<prop>>[ \t]+)?[|>])[ \t]*(?:((?:\r?\n|\r)[ \t]+)\S[^\r\n]*(?:\2[^\r\n]+)*)/.source.replace(/<<prop>>/g, (function() {
                                return r;
                            }))),
                            lookbehind: true,
                            alias: "string"
                        },
                        comment: /#.*/,
                        key: {
                            pattern: RegExp(/((?:^|[:\-,[{\r\n?])[ \t]*(?:<<prop>>[ \t]+)?)<<key>>(?=\s*:\s)/.source.replace(/<<prop>>/g, (function() {
                                return r;
                            })).replace(/<<key>>/g, (function() {
                                return "(?:" + a + "|" + s + ")";
                            }))),
                            lookbehind: true,
                            greedy: true,
                            alias: "atrule"
                        },
                        directive: {
                            pattern: /(^[ \t]*)%.+/m,
                            lookbehind: true,
                            alias: "important"
                        },
                        datetime: {
                            pattern: i(/\d{4}-\d\d?-\d\d?(?:[tT]|[ \t]+)\d\d?:\d{2}:\d{2}(?:\.\d*)?(?:[ \t]*(?:Z|[-+]\d\d?(?::\d{2})?))?|\d{4}-\d{2}-\d{2}|\d\d?:\d{2}(?::\d{2}(?:\.\d*)?)?/.source),
                            lookbehind: true,
                            alias: "number"
                        },
                        boolean: {
                            pattern: i(/false|true/.source, "i"),
                            lookbehind: true,
                            alias: "important"
                        },
                        null: {
                            pattern: i(/null|~/.source, "i"),
                            lookbehind: true,
                            alias: "important"
                        },
                        string: {
                            pattern: i(s),
                            lookbehind: true,
                            greedy: true
                        },
                        number: {
                            pattern: i(/[+-]?(?:0x[\da-f]+|0o[0-7]+|(?:\d+(?:\.\d*)?|\.\d+)(?:e[+-]?\d+)?|\.inf|\.nan)/.source, "i"),
                            lookbehind: true
                        },
                        tag: n,
                        important: t,
                        punctuation: /---|[:[\]{}\-,|>?]|\.\.\./
                    };
                    e.languages.yml = e.languages.yaml;
                })(a);
                a.languages["splunk-spl"] = {
                    comment: /`comment\("(?:\\.|[^\\"])*"\)`/,
                    string: {
                        pattern: /"(?:\\.|[^\\"])*"/,
                        greedy: true
                    },
                    // https://docs.splunk.com/Documentation/Splunk/7.3.0/SearchReference/ListOfSearchCommands
                    keyword: /\b(?:abstract|accum|addcoltotals|addinfo|addtotals|analyzefields|anomalies|anomalousvalue|anomalydetection|append|appendcols|appendcsv|appendlookup|appendpipe|arules|associate|audit|autoregress|bin|bucket|bucketdir|chart|cluster|cofilter|collect|concurrency|contingency|convert|correlate|datamodel|dbinspect|dedup|delete|delta|diff|erex|eval|eventcount|eventstats|extract|fieldformat|fields|fieldsummary|filldown|fillnull|findtypes|folderize|foreach|format|from|gauge|gentimes|geom|geomfilter|geostats|head|highlight|history|iconify|input|inputcsv|inputlookup|iplocation|join|kmeans|kv|kvform|loadjob|localize|localop|lookup|makecontinuous|makemv|makeresults|map|mcollect|metadata|metasearch|meventcollect|mstats|multikv|multisearch|mvcombine|mvexpand|nomv|outlier|outputcsv|outputlookup|outputtext|overlap|pivot|predict|rangemap|rare|regex|relevancy|reltime|rename|replace|rest|return|reverse|rex|rtorder|run|savedsearch|script|scrub|search|searchtxn|selfjoin|sendemail|set|setfields|sichart|sirare|sistats|sitimechart|sitop|sort|spath|stats|strcat|streamstats|table|tags|tail|timechart|timewrap|top|transaction|transpose|trendline|tscollect|tstats|typeahead|typelearner|typer|union|uniq|untable|where|x11|xmlkv|xmlunescape|xpath|xyseries)\b/i,
                    "operator-word": {
                        pattern: /\b(?:and|as|by|not|or|xor)\b/i,
                        alias: "operator"
                    },
                    function: /\b\w+(?=\s*\()/,
                    property: /\b\w+(?=\s*=(?!=))/,
                    date: {
                        // MM/DD/YYYY(:HH:MM:SS)?
                        pattern: /\b\d{1,2}\/\d{1,2}\/\d{1,4}(?:(?::\d{1,2}){3})?\b/,
                        alias: "number"
                    },
                    number: /\b\d+(?:\.\d+)?\b/,
                    boolean: /\b(?:f|false|t|true)\b/i,
                    operator: /[<>=]=?|[-+*/%|]/,
                    punctuation: /[()[\],]/
                };
                a.languages.sql = {
                    comment: {
                        pattern: /(^|[^\\])(?:\/\*[\s\S]*?\*\/|(?:--|\/\/|#).*)/,
                        lookbehind: true
                    },
                    variable: [ {
                        pattern: /@(["'`])(?:\\[\s\S]|(?!\1)[^\\])+\1/,
                        greedy: true
                    }, /@[\w.$]+/ ],
                    string: {
                        pattern: /(^|[^@\\])("|')(?:\\[\s\S]|(?!\2)[^\\]|\2\2)*\2/,
                        greedy: true,
                        lookbehind: true
                    },
                    identifier: {
                        pattern: /(^|[^@\\])`(?:\\[\s\S]|[^`\\]|``)*`/,
                        greedy: true,
                        lookbehind: true,
                        inside: {
                            punctuation: /^`|`$/
                        }
                    },
                    function: /\b(?:AVG|COUNT|FIRST|FORMAT|LAST|LCASE|LEN|MAX|MID|MIN|MOD|NOW|ROUND|SUM|UCASE)(?=\s*\()/i,
                    // Should we highlight user defined functions too?
                    keyword: /\b(?:ACTION|ADD|AFTER|ALGORITHM|ALL|ALTER|ANALYZE|ANY|APPLY|AS|ASC|AUTHORIZATION|AUTO_INCREMENT|BACKUP|BDB|BEGIN|BERKELEYDB|BIGINT|BINARY|BIT|BLOB|BOOL|BOOLEAN|BREAK|BROWSE|BTREE|BULK|BY|CALL|CASCADED?|CASE|CHAIN|CHAR(?:ACTER|SET)?|CHECK(?:POINT)?|CLOSE|CLUSTERED|COALESCE|COLLATE|COLUMNS?|COMMENT|COMMIT(?:TED)?|COMPUTE|CONNECT|CONSISTENT|CONSTRAINT|CONTAINS(?:TABLE)?|CONTINUE|CONVERT|CREATE|CROSS|CURRENT(?:_DATE|_TIME|_TIMESTAMP|_USER)?|CURSOR|CYCLE|DATA(?:BASES?)?|DATE(?:TIME)?|DAY|DBCC|DEALLOCATE|DEC|DECIMAL|DECLARE|DEFAULT|DEFINER|DELAYED|DELETE|DELIMITERS?|DENY|DESC|DESCRIBE|DETERMINISTIC|DISABLE|DISCARD|DISK|DISTINCT|DISTINCTROW|DISTRIBUTED|DO|DOUBLE|DROP|DUMMY|DUMP(?:FILE)?|DUPLICATE|ELSE(?:IF)?|ENABLE|ENCLOSED|END|ENGINE|ENUM|ERRLVL|ERRORS|ESCAPED?|EXCEPT|EXEC(?:UTE)?|EXISTS|EXIT|EXPLAIN|EXTENDED|FETCH|FIELDS|FILE|FILLFACTOR|FIRST|FIXED|FLOAT|FOLLOWING|FOR(?: EACH ROW)?|FORCE|FOREIGN|FREETEXT(?:TABLE)?|FROM|FULL|FUNCTION|GEOMETRY(?:COLLECTION)?|GLOBAL|GOTO|GRANT|GROUP|HANDLER|HASH|HAVING|HOLDLOCK|HOUR|IDENTITY(?:COL|_INSERT)?|IF|IGNORE|IMPORT|INDEX|INFILE|INNER|INNODB|INOUT|INSERT|INT|INTEGER|INTERSECT|INTERVAL|INTO|INVOKER|ISOLATION|ITERATE|JOIN|KEYS?|KILL|LANGUAGE|LAST|LEAVE|LEFT|LEVEL|LIMIT|LINENO|LINES|LINESTRING|LOAD|LOCAL|LOCK|LONG(?:BLOB|TEXT)|LOOP|MATCH(?:ED)?|MEDIUM(?:BLOB|INT|TEXT)|MERGE|MIDDLEINT|MINUTE|MODE|MODIFIES|MODIFY|MONTH|MULTI(?:LINESTRING|POINT|POLYGON)|NATIONAL|NATURAL|NCHAR|NEXT|NO|NONCLUSTERED|NULLIF|NUMERIC|OFF?|OFFSETS?|ON|OPEN(?:DATASOURCE|QUERY|ROWSET)?|OPTIMIZE|OPTION(?:ALLY)?|ORDER|OUT(?:ER|FILE)?|OVER|PARTIAL|PARTITION|PERCENT|PIVOT|PLAN|POINT|POLYGON|PRECEDING|PRECISION|PREPARE|PREV|PRIMARY|PRINT|PRIVILEGES|PROC(?:EDURE)?|PUBLIC|PURGE|QUICK|RAISERROR|READS?|REAL|RECONFIGURE|REFERENCES|RELEASE|RENAME|REPEAT(?:ABLE)?|REPLACE|REPLICATION|REQUIRE|RESIGNAL|RESTORE|RESTRICT|RETURN(?:ING|S)?|REVOKE|RIGHT|ROLLBACK|ROUTINE|ROW(?:COUNT|GUIDCOL|S)?|RTREE|RULE|SAVE(?:POINT)?|SCHEMA|SECOND|SELECT|SERIAL(?:IZABLE)?|SESSION(?:_USER)?|SET(?:USER)?|SHARE|SHOW|SHUTDOWN|SIMPLE|SMALLINT|SNAPSHOT|SOME|SONAME|SQL|START(?:ING)?|STATISTICS|STATUS|STRIPED|SYSTEM_USER|TABLES?|TABLESPACE|TEMP(?:ORARY|TABLE)?|TERMINATED|TEXT(?:SIZE)?|THEN|TIME(?:STAMP)?|TINY(?:BLOB|INT|TEXT)|TOP?|TRAN(?:SACTIONS?)?|TRIGGER|TRUNCATE|TSEQUAL|TYPES?|UNBOUNDED|UNCOMMITTED|UNDEFINED|UNION|UNIQUE|UNLOCK|UNPIVOT|UNSIGNED|UPDATE(?:TEXT)?|USAGE|USE|USER|USING|VALUES?|VAR(?:BINARY|CHAR|CHARACTER|YING)|VIEW|WAITFOR|WARNINGS|WHEN|WHERE|WHILE|WITH(?: ROLLUP|IN)?|WORK|WRITE(?:TEXT)?|YEAR)\b/i,
                    boolean: /\b(?:FALSE|NULL|TRUE)\b/i,
                    number: /\b0x[\da-f]+\b|\b\d+(?:\.\d*)?|\B\.\d+\b/i,
                    operator: /[-+*\/=%^~]|&&?|\|\|?|!=?|<(?:=>?|<|>)?|>[>=]?|\b(?:AND|BETWEEN|DIV|ILIKE|IN|IS|LIKE|NOT|OR|REGEXP|RLIKE|SOUNDS LIKE|XOR)\b/i,
                    punctuation: /[;[\]()`,.]/
                };
                a.languages.c = a.languages.extend("clike", {
                    comment: {
                        pattern: /\/\/(?:[^\r\n\\]|\\(?:\r\n?|\n|(?![\r\n])))*|\/\*[\s\S]*?(?:\*\/|$)/,
                        greedy: true
                    },
                    string: {
                        // https://en.cppreference.com/w/c/language/string_literal
                        pattern: /"(?:\\(?:\r\n|[\s\S])|[^"\\\r\n])*"/,
                        greedy: true
                    },
                    "class-name": {
                        pattern: /(\b(?:enum|struct)\s+(?:__attribute__\s*\(\([\s\S]*?\)\)\s*)?)\w+|\b[a-z]\w*_t\b/,
                        lookbehind: true
                    },
                    keyword: /\b(?:_Alignas|_Alignof|_Atomic|_Bool|_Complex|_Generic|_Imaginary|_Noreturn|_Static_assert|_Thread_local|__attribute__|asm|auto|break|case|char|const|continue|default|do|double|else|enum|extern|float|for|goto|if|inline|int|long|register|return|short|signed|sizeof|static|struct|switch|typedef|typeof|union|unsigned|void|volatile|while)\b/,
                    function: /\b[a-z_]\w*(?=\s*\()/i,
                    number: /(?:\b0x(?:[\da-f]+(?:\.[\da-f]*)?|\.[\da-f]+)(?:p[+-]?\d+)?|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:e[+-]?\d+)?)[ful]{0,4}/i,
                    operator: />>=?|<<=?|->|([-+&|:])\1|[?:~]|[-+*/%&|^!=<>]=?/
                });
                a.languages.insertBefore("c", "string", {
                    char: {
                        // https://en.cppreference.com/w/c/language/character_constant
                        pattern: /'(?:\\(?:\r\n|[\s\S])|[^'\\\r\n]){0,32}'/,
                        greedy: true
                    }
                });
                a.languages.insertBefore("c", "string", {
                    macro: {
                        // allow for multiline macro definitions
                        // spaces after the # character compile fine with gcc
                        pattern: /(^[\t ]*)#\s*[a-z](?:[^\r\n\\/]|\/(?!\*)|\/\*(?:[^*]|\*(?!\/))*\*\/|\\(?:\r\n|[\s\S]))*/im,
                        lookbehind: true,
                        greedy: true,
                        alias: "property",
                        inside: {
                            string: [ {
                                // highlight the path of the include statement as a string
                                pattern: /^(#\s*include\s*)<[^>]+>/,
                                lookbehind: true
                            }, a.languages.c["string"] ],
                            char: a.languages.c["char"],
                            comment: a.languages.c["comment"],
                            "macro-name": [ {
                                pattern: /(^#\s*define\s+)\w+\b(?!\()/i,
                                lookbehind: true
                            }, {
                                pattern: /(^#\s*define\s+)\w+\b(?=\()/i,
                                lookbehind: true,
                                alias: "function"
                            } ],
                            // highlight macro directives as keywords
                            directive: {
                                pattern: /^(#\s*)[a-z]+/,
                                lookbehind: true,
                                alias: "keyword"
                            },
                            "directive-hash": /^#/,
                            punctuation: /##|\\(?=[\r\n])/,
                            expression: {
                                pattern: /\S[\s\S]*/,
                                inside: a.languages.c
                            }
                        }
                    }
                });
                a.languages.insertBefore("c", "function", {
                    // highlight predefined macros as constants
                    constant: /\b(?:EOF|NULL|SEEK_CUR|SEEK_END|SEEK_SET|__DATE__|__FILE__|__LINE__|__TIMESTAMP__|__TIME__|__func__|stderr|stdin|stdout)\b/
                });
                delete a.languages.c["boolean"];
                (function(e) {
                    var t = /\b(?:alignas|alignof|asm|auto|bool|break|case|catch|char|char16_t|char32_t|char8_t|class|co_await|co_return|co_yield|compl|concept|const|const_cast|consteval|constexpr|constinit|continue|decltype|default|delete|do|double|dynamic_cast|else|enum|explicit|export|extern|final|float|for|friend|goto|if|import|inline|int|int16_t|int32_t|int64_t|int8_t|long|module|mutable|namespace|new|noexcept|nullptr|operator|override|private|protected|public|register|reinterpret_cast|requires|return|short|signed|sizeof|static|static_assert|static_cast|struct|switch|template|this|thread_local|throw|try|typedef|typeid|typename|uint16_t|uint32_t|uint64_t|uint8_t|union|unsigned|using|virtual|void|volatile|wchar_t|while)\b/;
                    var n = /\b(?!<keyword>)\w+(?:\s*\.\s*\w+)*\b/.source.replace(/<keyword>/g, (function() {
                        return t.source;
                    }));
                    e.languages.cpp = e.languages.extend("c", {
                        "class-name": [ {
                            pattern: RegExp(/(\b(?:class|concept|enum|struct|typename)\s+)(?!<keyword>)\w+/.source.replace(/<keyword>/g, (function() {
                                return t.source;
                            }))),
                            lookbehind: true
                        }, 
                        // This is intended to capture the class name of method implementations like:
                        //   void foo::bar() const {}
                        // However! The `foo` in the above example could also be a namespace, so we only capture the class name if
                        // it starts with an uppercase letter. This approximation should give decent results.
                        /\b[A-Z]\w*(?=\s*::\s*\w+\s*\()/, 
                        // This will capture the class name before destructors like:
                        //   Foo::~Foo() {}
                        /\b[A-Z_]\w*(?=\s*::\s*~\w+\s*\()/i, 
                        // This also intends to capture the class name of method implementations but here the class has template
                        // parameters, so it can't be a namespace (until C++ adds generic namespaces).
                        /\b\w+(?=\s*<(?:[^<>]|<(?:[^<>]|<[^<>]*>)*>)*>\s*::\s*\w+\s*\()/ ],
                        keyword: t,
                        number: {
                            pattern: /(?:\b0b[01']+|\b0x(?:[\da-f']+(?:\.[\da-f']*)?|\.[\da-f']+)(?:p[+-]?[\d']+)?|(?:\b[\d']+(?:\.[\d']*)?|\B\.[\d']+)(?:e[+-]?[\d']+)?)[ful]{0,4}/i,
                            greedy: true
                        },
                        operator: />>=?|<<=?|->|--|\+\+|&&|\|\||[?:~]|<=>|[-+*/%&|^!=<>]=?|\b(?:and|and_eq|bitand|bitor|not|not_eq|or|or_eq|xor|xor_eq)\b/,
                        boolean: /\b(?:false|true)\b/
                    });
                    e.languages.insertBefore("cpp", "string", {
                        module: {
                            // https://en.cppreference.com/w/cpp/language/modules
                            pattern: RegExp(/(\b(?:import|module)\s+)/.source + "(?:" + 
                            // header-name
                            /"(?:\\(?:\r\n|[\s\S])|[^"\\\r\n])*"|<[^<>\r\n]*>/.source + "|" + 
                            // module name or partition or both
                            /<mod-name>(?:\s*:\s*<mod-name>)?|:\s*<mod-name>/.source.replace(/<mod-name>/g, (function() {
                                return n;
                            })) + ")"),
                            lookbehind: true,
                            greedy: true,
                            inside: {
                                string: /^[<"][\s\S]+/,
                                operator: /:/,
                                punctuation: /\./
                            }
                        },
                        "raw-string": {
                            pattern: /R"([^()\\ ]{0,16})\([\s\S]*?\)\1"/,
                            alias: "string",
                            greedy: true
                        }
                    });
                    e.languages.insertBefore("cpp", "keyword", {
                        "generic-function": {
                            pattern: /\b(?!operator\b)[a-z_]\w*\s*<(?:[^<>]|<[^<>]*>)*>(?=\s*\()/i,
                            inside: {
                                function: /^\w+/,
                                generic: {
                                    pattern: /<[\s\S]+/,
                                    alias: "class-name",
                                    inside: e.languages.cpp
                                }
                            }
                        }
                    });
                    e.languages.insertBefore("cpp", "operator", {
                        "double-colon": {
                            pattern: /::/,
                            alias: "punctuation"
                        }
                    });
                    e.languages.insertBefore("cpp", "class-name", {
                        // the base clause is an optional list of parent classes
                        // https://en.cppreference.com/w/cpp/language/class
                        "base-clause": {
                            pattern: /(\b(?:class|struct)\s+\w+\s*:\s*)[^;{}"'\s]+(?:\s+[^;{}"'\s]+)*(?=\s*[;{])/,
                            lookbehind: true,
                            greedy: true,
                            inside: e.languages.extend("cpp", {})
                        }
                    });
                    e.languages.insertBefore("inside", "double-colon", {
                        // All untokenized words that are not namespaces should be class names
                        "class-name": /\b[a-z_]\w*\b(?!\s*::)/i
                    }, e.languages.cpp["base-clause"]);
                })(a);
                (function(e) {
                    /**
	 * Replaces all placeholders "<<n>>" of given pattern with the n-th replacement (zero based).
	 *
	 * Note: This is a simple text based replacement. Be careful when using backreferences!
	 *
	 * @param {string} pattern the given pattern.
	 * @param {string[]} replacements a list of replacement which can be inserted into the given pattern.
	 * @returns {string} the pattern with all placeholders replaced with their corresponding replacements.
	 * @example replace(/a<<0>>a/.source, [/b+/.source]) === /a(?:b+)a/.source
	 */
                    function t(e, t) {
                        return e.replace(/<<(\d+)>>/g, (function(e, n) {
                            return "(?:" + t[+n] + ")";
                        }));
                    }
                    /**
	 * @param {string} pattern
	 * @param {string[]} replacements
	 * @param {string} [flags]
	 * @returns {RegExp}
	 */                    function n(e, n, r) {
                        return RegExp(t(e, n), r || "");
                    }
                    /**
	 * Creates a nested pattern where all occurrences of the string `<<self>>` are replaced with the pattern itself.
	 *
	 * @param {string} pattern
	 * @param {number} depthLog2
	 * @returns {string}
	 */                    function r(e, t) {
                        for (var n = 0; n < t; n++) {
                            e = e.replace(/<<self>>/g, (function() {
                                return "(?:" + e + ")";
                            }));
                        }
                        return e.replace(/<<self>>/g, "[^\\s\\S]");
                    }
                    // https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/
                                        var a = {
                        // keywords which represent a return or variable type
                        type: "bool byte char decimal double dynamic float int long object sbyte short string uint ulong ushort var void",
                        // keywords which are used to declare a type
                        typeDeclaration: "class enum interface record struct",
                        // contextual keywords
                        // ("var" and "dynamic" are missing because they are used like types)
                        contextual: "add alias and ascending async await by descending from(?=\\s*(?:\\w|$)) get global group into init(?=\\s*;) join let nameof not notnull on or orderby partial remove select set unmanaged value when where with(?=\\s*{)",
                        // all other keywords
                        other: "abstract as base break case catch checked const continue default delegate do else event explicit extern finally fixed for foreach goto if implicit in internal is lock namespace new null operator out override params private protected public readonly ref return sealed sizeof stackalloc static switch this throw try typeof unchecked unsafe using virtual volatile while yield"
                    };
                    // keywords
                                        function s(e) {
                        return "\\b(?:" + e.trim().replace(/ /g, "|") + ")\\b";
                    }
                    var i = s(a.typeDeclaration);
                    var o = RegExp(s(a.type + " " + a.typeDeclaration + " " + a.contextual + " " + a.other));
                    var l = s(a.typeDeclaration + " " + a.contextual + " " + a.other);
                    var u = s(a.type + " " + a.typeDeclaration + " " + a.other);
                    // types
                                        var c = r(/<(?:[^<>;=+\-*/%&|^]|<<self>>)*>/.source, 2);
 // the idea behind the other forbidden characters is to prevent false positives. Same for tupleElement.
                                        var p = r(/\((?:[^()]|<<self>>)*\)/.source, 2);
                    var d = /@?\b[A-Za-z_]\w*\b/.source;
                    var g = t(/<<0>>(?:\s*<<1>>)?/.source, [ d, c ]);
                    var f = t(/(?!<<0>>)<<1>>(?:\s*\.\s*<<1>>)*/.source, [ l, g ]);
                    var m = /\[\s*(?:,\s*)*\]/.source;
                    var b = t(/<<0>>(?:\s*(?:\?\s*)?<<1>>)*(?:\s*\?)?/.source, [ f, m ]);
                    var h = t(/[^,()<>[\];=+\-*/%&|^]|<<0>>|<<1>>|<<2>>/.source, [ c, p, m ]);
                    var v = t(/\(<<0>>+(?:,<<0>>+)+\)/.source, [ h ]);
                    var E = t(/(?:<<0>>|<<1>>)(?:\s*(?:\?\s*)?<<2>>)*(?:\s*\?)?/.source, [ v, f, m ]);
                    var y = {
                        keyword: o,
                        punctuation: /[<>()?,.:[\]]/
                    };
                    // strings & characters
                    // https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#character-literals
                    // https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#string-literals
                                        var S = /'(?:[^\r\n'\\]|\\.|\\[Uux][\da-fA-F]{1,8})'/.source;
 // simplified pattern
                                        var A = /"(?:\\.|[^\\"\r\n])*"/.source;
                    var k = /@"(?:""|\\[\s\S]|[^\\"])*"(?!")/.source;
                    e.languages.csharp = e.languages.extend("clike", {
                        string: [ {
                            pattern: n(/(^|[^$\\])<<0>>/.source, [ k ]),
                            lookbehind: true,
                            greedy: true
                        }, {
                            pattern: n(/(^|[^@$\\])<<0>>/.source, [ A ]),
                            lookbehind: true,
                            greedy: true
                        } ],
                        "class-name": [ {
                            // Using static
                            // using static System.Math;
                            pattern: n(/(\busing\s+static\s+)<<0>>(?=\s*;)/.source, [ f ]),
                            lookbehind: true,
                            inside: y
                        }, {
                            // Using alias (type)
                            // using Project = PC.MyCompany.Project;
                            pattern: n(/(\busing\s+<<0>>\s*=\s*)<<1>>(?=\s*;)/.source, [ d, E ]),
                            lookbehind: true,
                            inside: y
                        }, {
                            // Using alias (alias)
                            // using Project = PC.MyCompany.Project;
                            pattern: n(/(\busing\s+)<<0>>(?=\s*=)/.source, [ d ]),
                            lookbehind: true
                        }, {
                            // Type declarations
                            // class Foo<A, B>
                            // interface Foo<out A, B>
                            pattern: n(/(\b<<0>>\s+)<<1>>/.source, [ i, g ]),
                            lookbehind: true,
                            inside: y
                        }, {
                            // Single catch exception declaration
                            // catch(Foo)
                            // (things like catch(Foo e) is covered by variable declaration)
                            pattern: n(/(\bcatch\s*\(\s*)<<0>>/.source, [ f ]),
                            lookbehind: true,
                            inside: y
                        }, {
                            // Name of the type parameter of generic constraints
                            // where Foo : class
                            pattern: n(/(\bwhere\s+)<<0>>/.source, [ d ]),
                            lookbehind: true
                        }, {
                            // Casts and checks via as and is.
                            // as Foo<A>, is Bar<B>
                            // (things like if(a is Foo b) is covered by variable declaration)
                            pattern: n(/(\b(?:is(?:\s+not)?|as)\s+)<<0>>/.source, [ b ]),
                            lookbehind: true,
                            inside: y
                        }, {
                            // Variable, field and parameter declaration
                            // (Foo bar, Bar baz, Foo[,,] bay, Foo<Bar, FooBar<Bar>> bax)
                            pattern: n(/\b<<0>>(?=\s+(?!<<1>>|with\s*\{)<<2>>(?:\s*[=,;:{)\]]|\s+(?:in|when)\b))/.source, [ E, u, d ]),
                            inside: y
                        } ],
                        keyword: o,
                        // https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#literals
                        number: /(?:\b0(?:x[\da-f_]*[\da-f]|b[01_]*[01])|(?:\B\.\d+(?:_+\d+)*|\b\d+(?:_+\d+)*(?:\.\d+(?:_+\d+)*)?)(?:e[-+]?\d+(?:_+\d+)*)?)(?:[dflmu]|lu|ul)?\b/i,
                        operator: />>=?|<<=?|[-=]>|([-+&|])\1|~|\?\?=?|[-+*/%&|^!=<>]=?/,
                        punctuation: /\?\.?|::|[{}[\];(),.:]/
                    });
                    e.languages.insertBefore("csharp", "number", {
                        range: {
                            pattern: /\.\./,
                            alias: "operator"
                        }
                    });
                    e.languages.insertBefore("csharp", "punctuation", {
                        "named-parameter": {
                            pattern: n(/([(,]\s*)<<0>>(?=\s*:)/.source, [ d ]),
                            lookbehind: true,
                            alias: "punctuation"
                        }
                    });
                    e.languages.insertBefore("csharp", "class-name", {
                        namespace: {
                            // namespace Foo.Bar {}
                            // using Foo.Bar;
                            pattern: n(/(\b(?:namespace|using)\s+)<<0>>(?:\s*\.\s*<<0>>)*(?=\s*[;{])/.source, [ d ]),
                            lookbehind: true,
                            inside: {
                                punctuation: /\./
                            }
                        },
                        "type-expression": {
                            // default(Foo), typeof(Foo<Bar>), sizeof(int)
                            pattern: n(/(\b(?:default|sizeof|typeof)\s*\(\s*(?!\s))(?:[^()\s]|\s(?!\s)|<<0>>)*(?=\s*\))/.source, [ p ]),
                            lookbehind: true,
                            alias: "class-name",
                            inside: y
                        },
                        "return-type": {
                            // Foo<Bar> ForBar(); Foo IFoo.Bar() => 0
                            // int this[int index] => 0; T IReadOnlyList<T>.this[int index] => this[index];
                            // int Foo => 0; int Foo { get; set } = 0;
                            pattern: n(/<<0>>(?=\s+(?:<<1>>\s*(?:=>|[({]|\.\s*this\s*\[)|this\s*\[))/.source, [ E, f ]),
                            inside: y,
                            alias: "class-name"
                        },
                        "constructor-invocation": {
                            // new List<Foo<Bar[]>> { }
                            pattern: n(/(\bnew\s+)<<0>>(?=\s*[[({])/.source, [ E ]),
                            lookbehind: true,
                            inside: y,
                            alias: "class-name"
                        },
                        /*'explicit-implementation': {
			// int IFoo<Foo>.Bar => 0; void IFoo<Foo<Foo>>.Foo<T>();
			pattern: replace(/\b<<0>>(?=\.<<1>>)/, className, methodOrPropertyDeclaration),
			inside: classNameInside,
			alias: 'class-name'
		},*/
                        "generic-method": {
                            // foo<Bar>()
                            pattern: n(/<<0>>\s*<<1>>(?=\s*\()/.source, [ d, c ]),
                            inside: {
                                function: n(/^<<0>>/.source, [ d ]),
                                generic: {
                                    pattern: RegExp(c),
                                    alias: "class-name",
                                    inside: y
                                }
                            }
                        },
                        "type-list": {
                            // The list of types inherited or of generic constraints
                            // class Foo<F> : Bar, IList<FooBar>
                            // where F : Bar, IList<int>
                            pattern: n(/\b((?:<<0>>\s+<<1>>|record\s+<<1>>\s*<<5>>|where\s+<<2>>)\s*:\s*)(?:<<3>>|<<4>>|<<1>>\s*<<5>>|<<6>>)(?:\s*,\s*(?:<<3>>|<<4>>|<<6>>))*(?=\s*(?:where|[{;]|=>|$))/.source, [ i, g, d, E, o.source, p, /\bnew\s*\(\s*\)/.source ]),
                            lookbehind: true,
                            inside: {
                                "record-arguments": {
                                    pattern: n(/(^(?!new\s*\()<<0>>\s*)<<1>>/.source, [ g, p ]),
                                    lookbehind: true,
                                    greedy: true,
                                    inside: e.languages.csharp
                                },
                                keyword: o,
                                "class-name": {
                                    pattern: RegExp(E),
                                    greedy: true,
                                    inside: y
                                },
                                punctuation: /[,()]/
                            }
                        },
                        preprocessor: {
                            pattern: /(^[\t ]*)#.*/m,
                            lookbehind: true,
                            alias: "property",
                            inside: {
                                // highlight preprocessor directives as keywords
                                directive: {
                                    pattern: /(#)\b(?:define|elif|else|endif|endregion|error|if|line|nullable|pragma|region|undef|warning)\b/,
                                    lookbehind: true,
                                    alias: "keyword"
                                }
                            }
                        }
                    });
                    // attributes
                                        var x = A + "|" + S;
                    var T = t(/\/(?![*/])|\/\/[^\r\n]*[\r\n]|\/\*(?:[^*]|\*(?!\/))*\*\/|<<0>>/.source, [ x ]);
                    var I = r(t(/[^"'/()]|<<0>>|\(<<self>>*\)/.source, [ T ]), 2);
                    // https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/#attribute-targets
                                        var R = /\b(?:assembly|event|field|method|module|param|property|return|type)\b/.source;
                    var w = t(/<<0>>(?:\s*\(<<1>>*\))?/.source, [ f, I ]);
                    e.languages.insertBefore("csharp", "class-name", {
                        attribute: {
                            // Attributes
                            // [Foo], [Foo(1), Bar(2, Prop = "foo")], [return: Foo(1), Bar(2)], [assembly: Foo(Bar)]
                            pattern: n(/((?:^|[^\s\w>)?])\s*\[\s*)(?:<<0>>\s*:\s*)?<<1>>(?:\s*,\s*<<1>>)*(?=\s*\])/.source, [ R, w ]),
                            lookbehind: true,
                            greedy: true,
                            inside: {
                                target: {
                                    pattern: n(/^<<0>>(?=\s*:)/.source, [ R ]),
                                    alias: "keyword"
                                },
                                "attribute-arguments": {
                                    pattern: n(/\(<<0>>*\)/.source, [ I ]),
                                    inside: e.languages.csharp
                                },
                                "class-name": {
                                    pattern: RegExp(f),
                                    inside: {
                                        punctuation: /\./
                                    }
                                },
                                punctuation: /[:,]/
                            }
                        }
                    });
                    // string interpolation
                                        var N = /:[^}\r\n]+/.source;
                    // multi line
                                        var O = r(t(/[^"'/()]|<<0>>|\(<<self>>*\)/.source, [ T ]), 2);
                    var _ = t(/\{(?!\{)(?:(?![}:])<<0>>)*<<1>>?\}/.source, [ O, N ]);
                    // single line
                                        var L = r(t(/[^"'/()]|\/(?!\*)|\/\*(?:[^*]|\*(?!\/))*\*\/|<<0>>|\(<<self>>*\)/.source, [ x ]), 2);
                    var F = t(/\{(?!\{)(?:(?![}:])<<0>>)*<<1>>?\}/.source, [ L, N ]);
                    function C(t, r) {
                        return {
                            interpolation: {
                                pattern: n(/((?:^|[^{])(?:\{\{)*)<<0>>/.source, [ t ]),
                                lookbehind: true,
                                inside: {
                                    "format-string": {
                                        pattern: n(/(^\{(?:(?![}:])<<0>>)*)<<1>>(?=\}$)/.source, [ r, N ]),
                                        lookbehind: true,
                                        inside: {
                                            punctuation: /^:/
                                        }
                                    },
                                    punctuation: /^\{|\}$/,
                                    expression: {
                                        pattern: /[\s\S]+/,
                                        alias: "language-csharp",
                                        inside: e.languages.csharp
                                    }
                                }
                            },
                            string: /[\s\S]+/
                        };
                    }
                    e.languages.insertBefore("csharp", "string", {
                        "interpolation-string": [ {
                            pattern: n(/(^|[^\\])(?:\$@|@\$)"(?:""|\\[\s\S]|\{\{|<<0>>|[^\\{"])*"/.source, [ _ ]),
                            lookbehind: true,
                            greedy: true,
                            inside: C(_, O)
                        }, {
                            pattern: n(/(^|[^@\\])\$"(?:\\.|\{\{|<<0>>|[^\\"{])*"/.source, [ F ]),
                            lookbehind: true,
                            greedy: true,
                            inside: C(F, L)
                        } ],
                        char: {
                            pattern: RegExp(S),
                            greedy: true
                        }
                    });
                    e.languages.dotnet = e.languages.cs = e.languages.csharp;
                })(a);
                (function(e) {
                    var t = e.languages.powershell = {
                        comment: [ {
                            pattern: /(^|[^`])<#[\s\S]*?#>/,
                            lookbehind: true
                        }, {
                            pattern: /(^|[^`])#.*/,
                            lookbehind: true
                        } ],
                        string: [ {
                            pattern: /"(?:`[\s\S]|[^`"])*"/,
                            greedy: true,
                            inside: null
                        }, {
                            pattern: /'(?:[^']|'')*'/,
                            greedy: true
                        } ],
                        // Matches name spaces as well as casts, attribute decorators. Force starting with letter to avoid matching array indices
                        // Supports two levels of nested brackets (e.g. `[OutputType([System.Collections.Generic.List[int]])]`)
                        namespace: /\[[a-z](?:\[(?:\[[^\]]*\]|[^\[\]])*\]|[^\[\]])*\]/i,
                        boolean: /\$(?:false|true)\b/i,
                        variable: /\$\w+\b/,
                        // Cmdlets and aliases. Aliases should come last, otherwise "write" gets preferred over "write-host" for example
                        // Get-Command | ?{ $_.ModuleName -match "Microsoft.PowerShell.(Util|Core|Management)" }
                        // Get-Alias | ?{ $_.ReferencedCommand.Module.Name -match "Microsoft.PowerShell.(Util|Core|Management)" }
                        function: [ /\b(?:Add|Approve|Assert|Backup|Block|Checkpoint|Clear|Close|Compare|Complete|Compress|Confirm|Connect|Convert|ConvertFrom|ConvertTo|Copy|Debug|Deny|Disable|Disconnect|Dismount|Edit|Enable|Enter|Exit|Expand|Export|Find|ForEach|Format|Get|Grant|Group|Hide|Import|Initialize|Install|Invoke|Join|Limit|Lock|Measure|Merge|Move|New|Open|Optimize|Out|Ping|Pop|Protect|Publish|Push|Read|Receive|Redo|Register|Remove|Rename|Repair|Request|Reset|Resize|Resolve|Restart|Restore|Resume|Revoke|Save|Search|Select|Send|Set|Show|Skip|Sort|Split|Start|Step|Stop|Submit|Suspend|Switch|Sync|Tee|Test|Trace|Unblock|Undo|Uninstall|Unlock|Unprotect|Unpublish|Unregister|Update|Use|Wait|Watch|Where|Write)-[a-z]+\b/i, /\b(?:ac|cat|chdir|clc|cli|clp|clv|compare|copy|cp|cpi|cpp|cvpa|dbp|del|diff|dir|ebp|echo|epal|epcsv|epsn|erase|fc|fl|ft|fw|gal|gbp|gc|gci|gcs|gdr|gi|gl|gm|gp|gps|group|gsv|gu|gv|gwmi|iex|ii|ipal|ipcsv|ipsn|irm|iwmi|iwr|kill|lp|ls|measure|mi|mount|move|mp|mv|nal|ndr|ni|nv|ogv|popd|ps|pushd|pwd|rbp|rd|rdr|ren|ri|rm|rmdir|rni|rnp|rp|rv|rvpa|rwmi|sal|saps|sasv|sbp|sc|select|set|shcm|si|sl|sleep|sls|sort|sp|spps|spsv|start|sv|swmi|tee|trcm|type|write)\b/i ],
                        // per http://technet.microsoft.com/en-us/library/hh847744.aspx
                        keyword: /\b(?:Begin|Break|Catch|Class|Continue|Data|Define|Do|DynamicParam|Else|ElseIf|End|Exit|Filter|Finally|For|ForEach|From|Function|If|InlineScript|Parallel|Param|Process|Return|Sequence|Switch|Throw|Trap|Try|Until|Using|Var|While|Workflow)\b/i,
                        operator: {
                            pattern: /(^|\W)(?:!|-(?:b?(?:and|x?or)|as|(?:Not)?(?:Contains|In|Like|Match)|eq|ge|gt|is(?:Not)?|Join|le|lt|ne|not|Replace|sh[lr])\b|-[-=]?|\+[+=]?|[*\/%]=?)/i,
                            lookbehind: true
                        },
                        punctuation: /[|{}[\];(),.]/
                    };
                    // Variable interpolation inside strings, and nested expressions
                                        t.string[0].inside = {
                        function: {
                            // Allow for one level of nesting
                            pattern: /(^|[^`])\$\((?:\$\([^\r\n()]*\)|(?!\$\()[^\r\n)])*\)/,
                            lookbehind: true,
                            inside: t
                        },
                        boolean: t.boolean,
                        variable: t.variable
                    };
                })(a);
                a.languages.python = {
                    comment: {
                        pattern: /(^|[^\\])#.*/,
                        lookbehind: true,
                        greedy: true
                    },
                    "string-interpolation": {
                        pattern: /(?:f|fr|rf)(?:("""|''')[\s\S]*?\1|("|')(?:\\.|(?!\2)[^\\\r\n])*\2)/i,
                        greedy: true,
                        inside: {
                            interpolation: {
                                // "{" <expression> <optional "!s", "!r", or "!a"> <optional ":" format specifier> "}"
                                pattern: /((?:^|[^{])(?:\{\{)*)\{(?!\{)(?:[^{}]|\{(?!\{)(?:[^{}]|\{(?!\{)(?:[^{}])+\})+\})+\}/,
                                lookbehind: true,
                                inside: {
                                    "format-spec": {
                                        pattern: /(:)[^:(){}]+(?=\}$)/,
                                        lookbehind: true
                                    },
                                    "conversion-option": {
                                        pattern: /![sra](?=[:}]$)/,
                                        alias: "punctuation"
                                    },
                                    rest: null
                                }
                            },
                            string: /[\s\S]+/
                        }
                    },
                    "triple-quoted-string": {
                        pattern: /(?:[rub]|br|rb)?("""|''')[\s\S]*?\1/i,
                        greedy: true,
                        alias: "string"
                    },
                    string: {
                        pattern: /(?:[rub]|br|rb)?("|')(?:\\.|(?!\1)[^\\\r\n])*\1/i,
                        greedy: true
                    },
                    function: {
                        pattern: /((?:^|\s)def[ \t]+)[a-zA-Z_]\w*(?=\s*\()/g,
                        lookbehind: true
                    },
                    "class-name": {
                        pattern: /(\bclass\s+)\w+/i,
                        lookbehind: true
                    },
                    decorator: {
                        pattern: /(^[\t ]*)@\w+(?:\.\w+)*/m,
                        lookbehind: true,
                        alias: [ "annotation", "punctuation" ],
                        inside: {
                            punctuation: /\./
                        }
                    },
                    keyword: /\b(?:_(?=\s*:)|and|as|assert|async|await|break|case|class|continue|def|del|elif|else|except|exec|finally|for|from|global|if|import|in|is|lambda|match|nonlocal|not|or|pass|print|raise|return|try|while|with|yield)\b/,
                    builtin: /\b(?:__import__|abs|all|any|apply|ascii|basestring|bin|bool|buffer|bytearray|bytes|callable|chr|classmethod|cmp|coerce|compile|complex|delattr|dict|dir|divmod|enumerate|eval|execfile|file|filter|float|format|frozenset|getattr|globals|hasattr|hash|help|hex|id|input|int|intern|isinstance|issubclass|iter|len|list|locals|long|map|max|memoryview|min|next|object|oct|open|ord|pow|property|range|raw_input|reduce|reload|repr|reversed|round|set|setattr|slice|sorted|staticmethod|str|sum|super|tuple|type|unichr|unicode|vars|xrange|zip)\b/,
                    boolean: /\b(?:False|None|True)\b/,
                    number: /\b0(?:b(?:_?[01])+|o(?:_?[0-7])+|x(?:_?[a-f0-9])+)\b|(?:\b\d+(?:_\d+)*(?:\.(?:\d+(?:_\d+)*)?)?|\B\.\d+(?:_\d+)*)(?:e[+-]?\d+(?:_\d+)*)?j?(?!\w)/i,
                    operator: /[-+%=]=?|!=|:=|\*\*?=?|\/\/?=?|<[<=>]?|>[=>]?|[&|^~]/,
                    punctuation: /[{}[\];(),.:]/
                };
                a.languages.python["string-interpolation"].inside["interpolation"].inside.rest = a.languages.python;
                a.languages.py = a.languages.python;
            })();
            /***/
            /******/        }
    };
    /************************************************************************/
    /******/ // The module cache
    /******/    var t = {};
    /******/
    /******/ // The require function
    /******/    function n(r) {
        /******/ // Check if module is in cache
        /******/ var a = t[r];
        /******/        if (a !== undefined) {
            /******/ return a.exports;
            /******/        }
        /******/ // Create a new module (and put it into the cache)
        /******/        var s = t[r] = {
            /******/ // no module.id needed
            /******/ // no module.loaded needed
            /******/ exports: {}
            /******/        };
        /******/
        /******/ // Execute the module function
        /******/        e[r](s, s.exports, n);
        /******/
        /******/ // Return the exports of the module
        /******/        return s.exports;
        /******/    }
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ n.n = e => {
            /******/ var t = e && e.__esModule ? 
            /******/ () => e["default"]
            /******/ : () => e
            /******/;
            n.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ n.d = (e, t) => {
            /******/ for (var r in t) {
                /******/ if (n.o(t, r) && !n.o(e, r)) {
                    /******/ Object.defineProperty(e, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ n.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ n.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // This entry need to be wrapped in an IIFE because it need to be in strict mode.
        (() => {
        "use strict";
        // ESM COMPAT FLAG
                n.r(r);
        // EXPORTS
                n.d(r, {
            default: () => /* reexport */ se
        });
        // CONCATENATED MODULE: external "react"
        const e = require("react");
        var t =  n.n(e);
        // CONCATENATED MODULE: external "lodash/castArray"
        const a = require("lodash/castArray");
        var s =  n.n(a);
        // CONCATENATED MODULE: external "lodash/flatten"
        const i = require("lodash/flatten");
        var o =  n.n(i);
        // CONCATENATED MODULE: external "lodash/indexOf"
        const l = require("lodash/indexOf");
        var u =  n.n(l);
        // CONCATENATED MODULE: external "lodash/repeat"
        const c = require("lodash/repeat");
        var p =  n.n(c);
        // CONCATENATED MODULE: external "lodash/times"
        const d = require("lodash/times");
        var g =  n.n(d);
        // CONCATENATED MODULE: external "prop-types"
        const f = require("prop-types");
        var m =  n.n(f);
        // CONCATENATED MODULE: external "@splunk/react-ui/useResizeObserver"
        const b = require("@splunk/react-ui/useResizeObserver");
        var h =  n.n(b);
        // CONCATENATED MODULE: external "styled-components"
        const v = require("styled-components");
        var E =  n.n(v);
        // CONCATENATED MODULE: external "@splunk/themes"
        const y = require("@splunk/themes");
        // CONCATENATED MODULE: ./src/Code/CodeStyles.ts
        var S = "lineHeightSpacious";
        var A = "spacingLarge";
        var k = E().code.withConfig({
            displayName: "CodeStyles__StyledCode",
            componentId: "sc-1eq4k68-0"
        })([ "font-family:", ";position:relative;z-index:1;", ";" ], y.variables.monoFontFamily, (function(e) {
            var t = e.$isPlaintext;
            return !t && (0, v.css)([ "color:", ";" ], y.variables.syntaxTeal);
        }));
        var x = E().pre.withConfig({
            displayName: "CodeStyles__StyledPre",
            componentId: "sc-1eq4k68-1"
        })([ "", ";", ";line-height:", ";position:relative;text-align:left;tab-size:4;white-space:pre;word-spacing:normal;word-break:normal;word-wrap:normal;hyphens:none;overflow-x:auto;&:focus-visible{box-shadow:", ";}", " ", ";" ], y.mixins.reset("grid"), y.mixins.typography("monoBody"), y.variables[S], y.variables.focusShadow, (function(e) {
            var t = e.$showLineNumbers, n = e.$counterReset;
            return t && (0, v.css)([ "grid-template-columns:min-content auto;counter-reset:linenumber ", ";" ], n);
        }), (function(e) {
            var t = e.$containerAppearance;
            return t === "none" ? (0, v.css)([ "gap:", ";" ], y.variables.spacingMedium) : (0, 
            v.css)([ "background-color:", ";border-radius:", ";", "{margin:", ";}" ], y.variables.neutral50, y.variables.borderRadius, /* sc-sel */ k, y.variables[A]);
        }));
        var T = E().span.withConfig({
            displayName: "CodeStyles__StyledIndent",
            componentId: "sc-1eq4k68-2"
        })([ "&:not(:last-child){background-image:linear-gradient( to left,", " 1px,transparent 1px,transparent );}" ], y.variables.syntaxGray);
        var I = {
            comment: y.variables.syntaxGray,
            prolog: y.variables.syntaxGray,
            doctype: y.variables.syntaxGray,
            cdata: y.variables.syntaxGray,
            punctuation: y.variables.syntaxGray,
            property: y.variables.syntaxPurple,
            tag: y.variables.syntaxPurple,
            boolean: y.variables.syntaxPurple,
            number: y.variables.syntaxPurple,
            constant: y.variables.syntaxPurple,
            symbol: y.variables.syntaxPurple,
            deleted: y.variables.syntaxPurple,
            key: y.variables.syntaxPurple,
            selector: y.variables.syntaxGreen,
            "attr-name": y.variables.syntaxGreen,
            string: y.variables.syntaxGreen,
            char: y.variables.syntaxGreen,
            builtin: y.variables.syntaxGreen,
            inserted: y.variables.syntaxGreen,
            operator: y.variables.syntaxBrown,
            entity: y.variables.syntaxBrown,
            url: y.variables.syntaxBrown,
            atrule: y.variables.syntaxBlue,
            "attr-value": y.variables.syntaxBlue,
            keyword: y.variables.syntaxBlue,
            function: y.variables.syntaxRed,
            regex: y.variables.syntaxOrange,
            important: y.variables.syntaxOrange,
            variable: y.variables.syntaxOrange
        };
        var R = E().span.withConfig({
            displayName: "CodeStyles__StyledToken",
            componentId: "sc-1eq4k68-3"
        })([ "color:", ";cursor:", ";opacity:", ";font-style:", ";font-weight:", ";" ], (function(e) {
            var t = e.partType;
            return I[t];
        }), (function(e) {
            var t = e.partType;
            return t === "entity" ? "help" : null;
        }), (function(e) {
            var t = e.partType;
            return t === "namespace" ? .7 : null;
        }), (function(e) {
            var t = e.partType;
            return t === "italic" ? "italic" : null;
        }), (function(e) {
            var t = e.partType;
            return t === "important" || t === "bold" ? y.variables.fontWeightBold : null;
        }));
        // CONCATENATED MODULE: ./src/Code/LineHighlightsStyles.ts
        var w = E().div.withConfig({
            displayName: "LineHighlightsStyles__StyledLineHighlight",
            componentId: "sc-1x2b02z-0"
        })([ "position:absolute;left:0;height:1lh;background:", ";border-left:2px solid ", ";pointer-events:none;line-height:inherit;width:", "px;", ";" ], y.variables.notificationColorInfoWeak, y.variables.notificationColorInfo, (function(e) {
            var t = e.$width;
            return t;
        }), (function(e) {
            var t = e.$containerAppearance, n = e.$lineNum;
            return t === "none" ? (0, v.css)([ "top:calc(", " * ", "em);" ], n - 1, y.variables[S]) : (0, 
            v.css)([ "top:calc( ", " * ", "em + ", " );" ], n - 1, y.variables[S], y.variables[A]);
        }));
        // CONCATENATED MODULE: ./src/Code/LineHighlights.tsx
        function N(e, t) {
            return C(e) || F(e, t) || _(e, t) || O();
        }
        function O() {
            throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
        }
        function _(e, t) {
            if (e) {
                if ("string" == typeof e) return L(e, t);
                var n = {}.toString.call(e).slice(8, -1);
                return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? L(e, t) : void 0;
            }
        }
        function L(e, t) {
            (null == t || t > e.length) && (t = e.length);
            for (var n = 0, r = Array(t); n < t; n++) {
                r[n] = e[n];
            }
            return r;
        }
        function F(e, t) {
            var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
            if (null != n) {
                var r, a, s, i, o = [], l = !0, u = !1;
                try {
                    if (s = (n = n.call(e)).next, 0 === t) {
                        if (Object(n) !== n) return;
                        l = !1;
                    } else for (;!(l = (r = s.call(n)).done) && (o.push(r.value), o.length !== t); l = !0) {
                    }
                } catch (e) {
                    u = !0, a = e;
                } finally {
                    try {
                        if (!l && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                    } finally {
                        if (u) throw a;
                    }
                }
                return o;
            }
        }
        function C(e) {
            if (Array.isArray(e)) return e;
        }
        // parseHighlightLines will turn a string like "2,4-5" into an array of ranges like [[2, 2], [4, 5]]
                var D = function e(t) {
            return t.replace(/\s+/g, "").split(",").filter(Boolean).map((function(e) {
                var t = e.split("-").map(Number), n = N(t, 2), r = n[0], a = n[1];
                return [ r, a || r ];
            }));
        };
        // Flatten the array of ranges into a single array of line numbers
        // example: [[2, 2], [4, 5]] will become [2, 4, 5]
                var P = function e(t) {
            var n = t.map((function(e) {
                var t = N(e, 2), n = t[0], r = t[1];
                return Array.from({
                    length: r - n + 1
                }, (function(e, t) {
                    return n + t;
                }));
            })).reduce((function(e, t) {
                return e.concat(t);
            }), []);
            return Array.from(new Set(n));
 // Remove duplicates
                };
        var $ = function e(n) {
            var r = n.containerAppearance, a = n.highlight, s = n.width;
            var i = D(a);
            var o = P(i);
            
            return t().createElement(t().Fragment, null, o.map((function(e) {
                
                return t().createElement(w, {
                    key: e,
                    className: "line-highlight",
                    "aria-hidden": "true",
                    $containerAppearance: r,
                    $lineNum: e,
                    $width: s
                });
            })));
        };
        /* harmony default export */        const U = $;
        // CONCATENATED MODULE: ./src/Code/LineNumbersStyles.ts
        var B = E().span.withConfig({
            displayName: "LineNumbersStyles__StyledLineNumberRows",
            componentId: "sc-1h4k3dh-0"
        })([ "pointer-events:none;user-select:none;", ";& > span{display:block;counter-increment:linenumber;position:relative;z-index:1;&::before{display:block;content:counter(linenumber);color:", ";text-align:right;", "}}" ], (function(e) {
            var t = e.$containerAppearance;
            return t === "none" ? (0, v.css)([ "border-right:1px solid ", ";z-index:1;" ], y.variables.borderColor) : (0, 
            v.css)([ "padding:", ";background-color:", ";" ], y.variables[A], y.variables.neutral200);
        }), y.variables.contentColorDefault, (function(e) {
            var t = e.$containerAppearance;
            return t === "none" && (0, v.css)([ "padding-inline-end:", ";" ], y.variables.spacingSmall);
        }));
        // CONCATENATED MODULE: ./src/Code/LineNumbers.tsx
        function M(e) {
            var n = e.containerAppearance, r = n === void 0 ? "none" : n, a = e.value, s = a === void 0 ? "" : a;
            // Use a regex to see how many lines there are in the value, then output an
            // empty span for each line.  The numbers themselves are handled by the CSS counter.
                        var i = s.match(/\n(?!$)/g);
            var o = i ? i.length + 1 : 1;
            var l = Array.from({
                length: o
            }, (function(e, n) {
                
                return t().createElement("span", {
                    key: n
                });
            }));
            
            return t().createElement(B, {
                $containerAppearance: r,
                "aria-hidden": true
            }, l);
        }
        /* harmony default export */        const G = M;
        // EXTERNAL MODULE: ./src/Code/prism/prism.empty
                var j = n(74);
        var H =  n.n(j);
        // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
        /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
        function z(e, t) {
            if (e) {
                if (typeof e === "function") {
                    e(t);
                } else {
                    // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                    // the intention here is to signal "we will take care of setting 'current', not you".
                    e.current = t;
 // eslint-disable-line no-param-reassign
                                }
            }
        }
        // CONCATENATED MODULE: ./src/Code/Code.tsx
        function W() {
            return W = Object.assign ? Object.assign.bind() : function(e) {
                for (var t = 1; t < arguments.length; t++) {
                    var n = arguments[t];
                    for (var r in n) {
                        ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                    }
                }
                return e;
            }, W.apply(null, arguments);
        }
        function Y(e, t) {
            return V(e) || Z(e, t) || X(e, t) || q();
        }
        function q() {
            throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
        }
        function X(e, t) {
            if (e) {
                if ("string" == typeof e) return K(e, t);
                var n = {}.toString.call(e).slice(8, -1);
                return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? K(e, t) : void 0;
            }
        }
        function K(e, t) {
            (null == t || t > e.length) && (t = e.length);
            for (var n = 0, r = Array(t); n < t; n++) {
                r[n] = e[n];
            }
            return r;
        }
        function Z(e, t) {
            var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
            if (null != n) {
                var r, a, s, i, o = [], l = !0, u = !1;
                try {
                    if (s = (n = n.call(e)).next, 0 === t) {
                        if (Object(n) !== n) return;
                        l = !1;
                    } else for (;!(l = (r = s.call(n)).done) && (o.push(r.value), o.length !== t); l = !0) {
                    }
                } catch (e) {
                    u = !0, a = e;
                } finally {
                    try {
                        if (!l && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                    } finally {
                        if (u) throw a;
                    }
                }
                return o;
            }
        }
        function V(e) {
            if (Array.isArray(e)) return e;
        }
        function J(e, t) {
            if (null == e) return {};
            var n, r, a = Q(e, t);
            if (Object.getOwnPropertySymbols) {
                var s = Object.getOwnPropertySymbols(e);
                for (r = 0; r < s.length; r++) {
                    n = s[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
                }
            }
            return a;
        }
        function Q(e, t) {
            if (null == e) return {};
            var n = {};
            for (var r in e) {
                if ({}.hasOwnProperty.call(e, r)) {
                    if (-1 !== t.indexOf(r)) continue;
                    n[r] = e[r];
                }
            }
            return n;
        }
        // Prism requires special handling to work around issues related to global scope and
        // auto formatting. The import below points to an empty file which is replaced at build
        // time, see ./prism/build-isolated-prism.js for details.
        // typed as 'prismjs'
                var ee = [ "bash", "c", "clike", "cpp", "cs", "csharp", "css", "html", "javascript", "js", "json", "jsx", "markup", "mathml", "plain", "plaintext", "powershell", "py", "python", "sh", "shell", "splunk-spl", "sql", "svg", "text", "ts", "tsx", "txt", "typescript", "xml", "yaml", "yml" ];
        function te(e) {
            return [ "plain", "plaintext", "text", "txt" ].includes(e);
        }
        function ne(e, t) {
            if (te(t)) {
                return H().tokenize(e, {});
            }
            return H().tokenize(e, H().languages[t]);
        }
        var re = {
            containerAppearance: m().oneOf([ "none", "section" ]),
            elementRef: m().oneOfType([ m().func, m().object ]),
            indentChars: m().number,
            language: m().oneOf(ee),
            languageFallback: m().oneOf(ee),
            lineHighlights: m().string,
            lineNumberStart: m().number,
            showIndentGuide: m().bool,
            showLineNumbers: m().bool,
            value: m().string
        };
        function ae(n) {
            var r = n.containerAppearance, a = r === void 0 ? "none" : r, i = n.elementRef, l = n.indentChars, c = l === void 0 ? 4 : l, d = n.language, f = n.languageFallback, m = f === void 0 ? "javascript" : f, b = n.lineHighlights, v = n.lineNumberStart, E = n.showIndentGuide, y = E === void 0 ? true : E, S = n.showLineNumbers, A = n.value, I = A === void 0 ? "" : A, w = J(n, [ "containerAppearance", "elementRef", "indentChars", "language", "languageFallback", "lineHighlights", "lineNumberStart", "showIndentGuide", "showLineNumbers", "value" ]);
            // @docs-props-type CodePropsBase
                        var N = (0, e.useRef)(null);
            var O = (0, e.useState)(undefined), _ = Y(O, 2), L = _[0], F = _[1];
            var C = h()(N), D = C.width, P = C.height;
            var $ = (0, e.useCallback)((function(e) {
                z(N, e);
                z(i, e);
            }), [ i ]);
            (0, e.useEffect)((function() {
                var e = N.current;
                if (e) {
                    var t = e.scrollHeight > Math.round(P);
                    var n = e.scrollWidth > Math.round(D);
                    var r = t || n;
                    F(r ? 0 : undefined);
                }
            }), [ N, D, P ]);
            var B = p()(" ", c);
            var M = function e(n) {
                var r = Math.floor(n / c);
                var a = n % c;
                var s = g()(r, (function(e) {
                    
                    return t().createElement(T, {
                        key: e.toString()
                    }, B);
                }));
                if (a) {
                    s.push(p()(" ", a));
                }
                return s;
            };
            var j = d ? u()(ee, d) > -1 : true;
            (0, e.useEffect)((function() {
                // a fix for SUI-5425, where the Markdown component doesn't pass a valid language in some circumstances
                if (false) {}
            }), [ j, d, m ]);
            var H = j && d ? d : m;
            var q = ne(I, H);
            var X = te(H);
            var K = function e(n) {
                return s()(n).map((function(n, r) {
                    if (typeof n === "string") {
                        if (!y) {
                            return n;
                        }
                        var a = n.split("\n");
                        if (a.length === 1) {
                            return n;
                        }
                        return o()(a.map((function(e, n) {
                            var a = n === 0 ? [] : [ "\n" ];
                            var s = e.match(/^(\s+)(.*)/);
                            if (!s) {
                                a.push(e);
                            } else {
                                a.push( t().createElement("span", {
                                    key: "".concat(r.toString(), ":").concat(n.toString())
                                }, M(s[1].length)), s[2]);
                            }
                            return a;
                        })));
                    }
                    var s = typeof n.content === "string" ? n.content : e(n.content);
                    
                    // eslint-disable-next-line react/no-array-index-key
                    return t().createElement(R, {
                        "data-part-type": n.type,
                        partType: n.type,
                        key: r.toString()
                    }, s);
                }));
            };
            var Z = K(q);
            
            return t().createElement(x, W({
                "data-test": "code",
                ref: $,
                tabIndex: L,
                $containerAppearance: a,
                $showLineNumbers: S,
                $counterReset: v && v - 1
            }, w), S &&  t().createElement(G, {
                containerAppearance: a,
                value: I
            }),  t().createElement(k, {
                className: "language-".concat(H),
                $isPlaintext: X
            }, Z), b &&  t().createElement(U, {
                containerAppearance: a,
                width: D,
                highlight: b
            }));
        }
        ae.propTypes = re;
        /* harmony default export */        const se = ae;
    }) // CONCATENATED MODULE: ./src/Code/index.ts
    ();
    module.exports = r;
    /******/})();
