// @splunk/react-ui/CollapsiblePanel.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        SingleOpenPanelGroup: () => /* reexport */ ne,
        default: () => /* reexport */ Se
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/TransitionOpen"
    const o = require("@splunk/react-ui/TransitionOpen");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const s = require("@splunk/react-ui/useControlled");
    var u =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const c = require("@splunk/react-ui/usePrevious");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const p = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var b =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const v = require("@splunk/react-icons/ChevronRight");
    var y =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const m = require("@splunk/react-ui/Box");
    var g =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const h = require("@splunk/react-ui/ButtonSimple");
    var S =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const O = require("@splunk/react-ui/Clickable");
    var C =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/themes"
    const P = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/CollapsiblePanel/CollapsiblePanelStyles.ts
    var w = b()(g()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledBox",
        componentId: "sc-1phikbp-0"
    })([ "display:flex;flex-direction:column;flex-shrink:1;min-height:28px;background-color:", ";& + &{margin-top:1px;}" ], (0, 
    P.pickVariant)("$appearance", {
        default: P.variables.backgroundColorSection,
        subtle: P.variables.transparent
    }));
    var k = b()(g()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledSingleOpenPanelGroupBox",
        componentId: "sc-1phikbp-1"
    })([ "", ";flex-direction:column;margin-bottom:0;" ], P.mixins.reset("flex"));
    /*
    these styles are consistent for both regular and actions, but applied to different elements
    * in regular mode, they're on the SVG's span wrapper
    * in actionsMode, they're on the button
    applying them directly to the icon wrapper doesn't work because Enterprise has 1px border around the button
*/    var I = (0, f.css)([ "min-width:16px;" ]);
    var j = b().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledIconWrapper",
        componentId: "sc-1phikbp-2"
    })([ "display:inline-block;flex:0 auto;", "" ], I);
    var x = b().div.withConfig({
        displayName: "CollapsiblePanelStyles__StyledActions",
        componentId: "sc-1phikbp-3"
    })([ "display:flex;flex-shrink:0;gap:", ";padding:", " ", ";transition:background-color 0.2s,border 0.2s,box-shadow 0.2s;color:", ";background-color:", ";", "" ], P.variables.spacingSmall, P.variables.spacingXSmall, P.variables.spacingLarge, P.variables.contentColorActive, (0, 
    P.pickVariant)("$appearance", {
        default: P.variables.neutral100,
        subtle: P.variables.transparent
    }), (function(e) {
        var n = e.$disabled;
        return n && (0, f.css)([ "color:", ";" ], P.variables.contentColorDisabled);
    }));
    var E = b()(S()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledToggleButton",
        componentId: "sc-1phikbp-4"
    })([ "flex:0 0 auto;align-self:flex-start;width:", ";height:", ";" ], P.variables.inputHeight, P.variables.inputHeight);
    var _ = b()(C()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledTitleClickable",
        componentId: "sc-1phikbp-5"
    })([ "width:100%;flex-shrink:0;gap:", ";cursor:pointer;display:flex;transition:background-color 0.2s,border 0.2s,box-shadow 0.2s;padding:", " ", ";color:", ";background-color:", ";&[disabled],&[aria-disabled='true']{cursor:not-allowed;color:", ";}&:focus:not([disabled]),&:active:not([disabled],[aria-disabled='true']){box-shadow:", ";z-index:1;}&:hover:not([disabled],[aria-disabled='true']){background-color:", ";}" ], P.variables.spacingSmall, P.variables.spacingXSmall, P.variables.spacingLarge, P.variables.contentColorActive, (0, 
    P.pickVariant)("$appearance", {
        default: P.variables.neutral100,
        subtle: P.variables.transparent
    }), P.variables.contentColorDisabled, P.variables.focusShadowInset, P.mixins.overlayColors(P.variables.neutral100, P.variables.interactiveColorOverlayHover));
    var A = b().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledHeadingContent",
        componentId: "sc-1phikbp-6"
    })([ "display:flex;justify-content:space-between;flex:1 1 auto;font-weight:", ";line-height:", ";", "" ], P.variables.fontWeightSemiBold, P.variables.lineHeight, (function(e) {
        var n = e.$disabled;
        return n && (0, f.css)([ "color:", ";" ], P.variables.contentColorDisabled);
    }));
    var D = b().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledHeadingActions",
        componentId: "sc-1phikbp-7"
    })([ "min-height:", ";" ], P.variables.inputHeight);
    var $ = b().span.withConfig({
        displayName: "CollapsiblePanelStyles__StyledDescription",
        componentId: "sc-1phikbp-8"
    })([ "color:", ";font-size:", ";margin-left:", ";line-height:1.8;", "" ], P.variables.contentColorDefault, P.variables.fontSizeSmall, P.variables.spacingSmall, (function(e) {
        var n = e.$disabled;
        return n && (0, f.css)([ "color:", ";" ], P.variables.contentColorDisabled);
    }));
    var N = b().div.withConfig({
        displayName: "CollapsiblePanelStyles__StyledInsetWrapper",
        componentId: "sc-1phikbp-9"
    })([ "", ";", "" ], P.mixins.reset("block"), (function(e) {
        var n = e.$inset;
        return n && (0, f.css)([ "padding:", ";" ], P.variables.spacingLarge);
    }));
    var q = b()(y()).withConfig({
        displayName: "CollapsiblePanelStyles__StyledExpandPanelIcon",
        componentId: "sc-1phikbp-10"
    })([ "transition:transform 200ms;", " width:100%;vertical-align:text-bottom;" ], (function(e) {
        var n = e.$open;
        return n && (0, f.css)([ "transform:rotate(90deg);" ]);
    }));
    // CONCATENATED MODULE: external "lodash/omit"
    const T = require("lodash/omit");
    var V =  e.n(T);
    // CONCATENATED MODULE: ./src/CollapsiblePanel/SingleOpenPanelGroupContext.tsx
    var M =  (0, t.createContext)({});
    /* harmony default export */    const B = M;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/SingleOpenPanelGroup.tsx
    function R() {
        return R = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, R.apply(null, arguments);
    }
    function H(e) {
        return G(e) || W(e) || X(e) || L();
    }
    function L() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function W(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function G(e) {
        if (Array.isArray(e)) return F(e);
    }
    function U(e, n) {
        return K(e) || J(e, n) || X(e, n) || z();
    }
    function z() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function X(e, n) {
        if (e) {
            if ("string" == typeof e) return F(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? F(e, n) : void 0;
        }
    }
    function F(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function J(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, a, l, o, i = [], s = !0, u = !1;
            try {
                if (l = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    s = !1;
                } else for (;!(s = (r = l.call(t)).done) && (i.push(r.value), i.length !== n); s = !0) {
                }
            } catch (e) {
                u = !0, a = e;
            } finally {
                try {
                    if (!s && null != t["return"] && (o = t["return"](), Object(o) !== o)) return;
                } finally {
                    if (u) throw a;
                }
            }
            return i;
        }
    }
    function K(e) {
        if (Array.isArray(e)) return e;
    }
    function Q(e, n) {
        if (null == e) return {};
        var t, r, a = Y(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                t = l[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function Y(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */    var Z = {
        children: l().node,
        defaultOpenPanelId: l().any,
        elementRef: l().oneOfType([ l().func, l().object ]),
        inset: l().bool,
        onChange: l().func,
        openPanelId: l().any
    };
    function ee(e) {
        var n = e.children, a = e.defaultOpenPanelId, l = e.elementRef, o = e.inset, i = o === void 0 ? true : o, s = e.onChange, c = e.openPanelId, d = Q(e, [ "children", "defaultOpenPanelId", "elementRef", "inset", "onChange", "openPanelId" ]);
        // @docs-props-type SingleOpenPanelGroupPropsBase
                var p = u()({
            componentName: "SingleOpenPanelGroup",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultOpenPanelId",
            valuePropName: "openPanelId"
        });
        var f = !p ? a : undefined;
        var b = (0, t.useState)(f), v = U(b, 2), y = v[0], m = v[1];
        var g = (0, t.useCallback)((function(e, n) {
            var t = n.panelId;
            var r = p ? c : y;
            var a = r === t ? null : t;
            if (!p) {
                m(a);
            }
            s === null || s === void 0 ? void 0 : s(e, n);
        }), [ p, c, y, s ]);
        var h = t.Children.toArray(n).filter(t.isValidElement);
 // remove false, null, 0, etc
                var S = (0, t.useMemo)((function() {
            return {
                onChange: g,
                openPanelId: p ? c : y,
                inset: i
            };
        }), [ g, i, p, c, y ]);
        
        return r().createElement(k, R({
            "data-test": "single-open-panel-group",
            "data-test-open-panel-id": c,
            ref: l
        }, V().apply(void 0, [ d ].concat(H(Object.keys(Z))))),  r().createElement(B.Provider, {
            value: S
        }, h));
    }
    ee.propTypes = Z;
    /* harmony default export */    const ne = ee;
    // CONCATENATED MODULE: ./src/utils/disabled.ts
    /**
 * Normalizes the disabled prop and returns both the boolean state and normalized string value.
 * Use this function to ensure consistent handling of disabled prop across components.
 *
 * @param disabled - The disabled prop value
 * @param defaultWhenBooleanTrue - Optional normalization override for disabled={true}
 * @returns Object with isDisabled boolean|undefined and disabledValue string
 *
 * TODO (SUI-8600): `disabled` prop should default to 'dimmed' for data entry components
 */
    function te(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "dimmed";
        if (e === "dimmed") {
            return {
                isDisabled: true,
                disabledValue: "dimmed"
            };
        }
        if (e === true) {
            return {
                isDisabled: true,
                disabledValue: n
            };
        }
        if (e === "disabled") {
            return {
                isDisabled: true,
                disabledValue: "disabled"
            };
        }
        if (e === false) {
            return {
                isDisabled: false,
                disabledValue: undefined
            };
        }
        return {
            isDisabled: undefined,
            disabledValue: undefined
        };
    }
    // CONCATENATED MODULE: ./src/CollapsiblePanel/CollapsiblePanel.tsx
    function re(e) {
        "@babel/helpers - typeof";
        return re = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, re(e);
    }
    function ae() {
        return ae = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, ae.apply(null, arguments);
    }
    function le(e, n) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            n && (r = r.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), t.push.apply(t, r);
        }
        return t;
    }
    function oe(e) {
        for (var n = 1; n < arguments.length; n++) {
            var t = null != arguments[n] ? arguments[n] : {};
            n % 2 ? le(Object(t), !0).forEach((function(n) {
                ie(e, n, t[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : le(Object(t)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(t, n));
            }));
        }
        return e;
    }
    function ie(e, n, t) {
        return (n = se(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function se(e) {
        var n = ue(e, "string");
        return "symbol" == re(n) ? n : n + "";
    }
    function ue(e, n) {
        if ("object" != re(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != re(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function ce(e, n) {
        return ve(e) || be(e, n) || pe(e, n) || de();
    }
    function de() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function pe(e, n) {
        if (e) {
            if ("string" == typeof e) return fe(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? fe(e, n) : void 0;
        }
    }
    function fe(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function be(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, a, l, o, i = [], s = !0, u = !1;
            try {
                if (l = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    s = !1;
                } else for (;!(s = (r = l.call(t)).done) && (i.push(r.value), i.length !== n); s = !0) {
                }
            } catch (e) {
                u = !0, a = e;
            } finally {
                try {
                    if (!s && null != t["return"] && (o = t["return"](), Object(o) !== o)) return;
                } finally {
                    if (u) throw a;
                }
            }
            return i;
        }
    }
    function ve(e) {
        if (Array.isArray(e)) return e;
    }
    function ye(e, n) {
        if (null == e) return {};
        var t, r, a = me(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                t = l[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function me(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */    var ge = {
        innerBodyStyles: l().object,
        actions: l().node,
        appearance: l().oneOf([ "default", "subtle" ]),
        children: l().node,
        defaultOpen: l().bool,
        description: l().string,
        disabled: l().oneOfType([ l().bool, l().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: l().oneOfType([ l().func, l().object ]),
        headingLevel: l().number,
        inset: l().bool,
        onChange: l().func,
        open: l().bool,
        overflow: l().string,
        panelId: l().any,
        renderChildrenWhenCollapsed: l().bool,
        title: l().node.isRequired
    };
    function he(e) {
        var n = e.innerBodyStyles, a = e.actions, l = e.appearance, o = l === void 0 ? "default" : l, s = e.children, c = e.defaultOpen, f = c === void 0 ? undefined : c, b = e.description, v = e.disabled, y = e.elementRef, m = e.headingLevel, g = e.onChange, h = e.open, S = e.overflow, O = S === void 0 ? "auto" : S, C = e.panelId, P = e.renderChildrenWhenCollapsed, k = e.title, I = e.inset, T = I === void 0 ? true : I, V = ye(e, [ "innerBodyStyles", "actions", "appearance", "children", "defaultOpen", "description", "disabled", "elementRef", "headingLevel", "onChange", "open", "overflow", "panelId", "renderChildrenWhenCollapsed", "title", "inset" ]);
        // @docs-props-type CollapsiblePanelPropsBase
                var M = (0, t.useContext)(B);
        var R = Object.keys(M).length !== 0;
        var H = u()({
            componentName: "CollapsiblePanel",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultOpen",
            valuePropName: "open"
        });
        var L = (0, t.useState)("container-".concat((0, p.createGUID)())), W = ce(L, 1), G = W[0];
        var U = (0, t.useState)("toggle-".concat((0, p.createGUID)())), z = ce(U, 1), X = z[0];
        var F = d()(h);
        var J = !H ? f || false : undefined;
        var K = (0, t.useState)(J), Q = ce(K, 2), Y = Q[0], Z = Q[1];
        var ee = (0, t.useState)(false), ne = ce(ee, 2), re = ne[0], le = ne[1];
        var ie = (0, t.useMemo)((function() {
            if (R) {
                /**
       * SingleOpenPanelGroup already takes into account if it is controlled or not in determining openPanelId
       * so context.openPanelId is used for both the controlled and uncontrolled case.
       */
                return M.openPanelId === C;
            }
            return H ? h : Y;
        }), [ M.openPanelId, R, H, h, Y, C ]);
        // SUI-7877 Must initialize openPanel with value to avoid re-render animation bug
                var se = (0, t.useState)(ie), ue = ce(se, 2), de = ue[0], pe = ue[1];
        (0, t.useEffect)((function() {
            pe(ie);
        }), [ ie ]);
        // Only testing this in the scenario of a SingleOpenPanelGroup.
                (0, t.useEffect)((function() {
            if (false) {}
        }), [ f, R, h ]);
        (0, t.useEffect)((function() {
            if (F != null && h !== F) {
                le(true);
            }
        }), [ h, F ]);
        var fe = (0, t.useCallback)((function(e) {
            var n;
            if (!H) {
                le(true);
                Z(!de);
            }
            var t = de ? "close" : "open";
            g === null || g === void 0 ? void 0 : g(e, {
                panelId: C,
                reason: "toggleClick",
                action: t
            });
            (n = M.onChange) === null || n === void 0 ? void 0 : n.call(M, e, {
                panelId: C,
                reason: "toggleClick",
                action: t
            });
        }), [ M, H, de, g, C ]);
        var be = (0, t.useCallback)((function() {
            le(false);
        }), []);
        var ve = (0, p.createDOMID)("panel-toggle");
        var me = te(v), ge = me.isDisabled, he = me.disabledValue;
        var Se = {
            disabled: v,
            onClick: fe,
            id: X,
            "aria-controls": G,
            "aria-expanded": de,
            "data-test": "toggle",
            "data-test-disabled": he
        };
        var Oe = (0, t.useMemo)((function() {
            return {
                overflow: re ? "hidden" : O
            };
        }), [ re, O ]);
        var Ce = (0, t.useMemo)((function() {
            return oe({
                tableLayout: "fixed"
            }, n);
        }), [ n ]);
        
        return r().createElement(w, ae({
            "data-test": "collapsible-panel",
            ref: y,
            $appearance: o
        }, V), a != null ?  r().createElement(x, {
            $appearance: o,
            $disabled: !!ge
        },  r().createElement(E, ae({
            appearance: "secondary",
            "aria-labelledby": ve
        }, Se),  r().createElement(q, {
            $open: !!de
        })),  r().createElement(A, {
            $disabled: !!ge,
            "data-test": "title",
            id: ve
        }, k),  r().createElement(D, null, a)) : 
        
        /* Clickable component is not allowed to have role="heading" so it is wrapped with a span */
        r().createElement("span", {
            "data-test": "heading",
            role: m ? "heading" : undefined,
            "aria-level": m
        },  r().createElement(_, ae({
            $appearance: o
        }, Se),  r().createElement(j, null,  r().createElement(q, {
            $open: !!de
        })),  r().createElement(A, {
            $disabled: !!ge
        },  r().createElement("span", {
            "data-test": "title"
        }, k), b &&  r().createElement($, {
            $disabled: !!ge,
            "data-test": "description"
        }, b)))),  r().createElement(i(), {
            outerId: G,
            "aria-labelledby": X,
            "data-test": "body",
            outerStyle: Oe,
            innerStyle: Ce,
            open: de,
            onAnimationEnd: be,
            role: a != null ? "region" : undefined,
            renderChildrenWhenCollapsed: P
        },  r().createElement(N, {
            $inset: T
        }, s)));
    }
    he.propTypes = ge;
    /* harmony default export */    const Se = he;
    // CONCATENATED MODULE: ./src/CollapsiblePanel/index.ts
    module.exports = n;
    /******/})();
