// @splunk/react-ui/ColumnLayout.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Column: () => /* reexport */ m,
        Row: () => /* reexport */ _,
        default: () => /* reexport */ T
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var o =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const c = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnStyles.ts
    var u = o().div.withConfig({
        displayName: "ColumnStyles__Styled",
        componentId: "sc-1pchz98-0"
    })([ "", " flex:", ";", "" ], c.mixins.reset("block"), (function(e) {
        var r = e.$span;
        return "".concat(r, " ").concat(r, " auto");
    }), (function(e) {
        var r = e.$gutter, t = e.$span;
        if (r === 0) {
            return (0, l.css)([ "width:", "%;" ], 100 / 12 * t);
        }
        var n = r !== undefined ? "".concat(r, "px") : c.variables.spacingXLarge;
        return (0, l.css)([ "width:calc( (100% - 11 * ", ") * ", " + (", " * ", ") );" ], n, t / 12, n, t - 1);
    }));
    // CONCATENATED MODULE: ./src/ColumnLayout/Column.tsx
    function s() {
        return s = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, s.apply(null, arguments);
    }
    function d(e, r) {
        if (null == e) return {};
        var t, n, a = f(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function f(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var v = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        /** @private. */
        gutter: i().number,
        span: i().number
    };
    function p(e) {
        var r = e.children, t = e.elementRef, a = e.gutter, i = e.span, l = i === void 0 ? 1 : i, o = d(e, [ "children", "elementRef", "gutter", "span" ]);
        // @docs-props-type ColumnPropsBase
                
        return n().createElement(u, s({
            "data-test": "column",
            ref: t,
            $gutter: a,
            $span: l
        }, o), r);
    }
    p.propTypes = v;
    /* harmony default export */    const m = p;
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const g = require("@splunk/react-ui/Divider");
    var y =  e.n(g);
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnLayoutStyles.ts
    var b = o().div.withConfig({
        displayName: "ColumnLayoutStyles__Styled",
        componentId: "sc-3eatxz-0"
    })([ "", ";flex-direction:column;", ";" ], c.mixins.reset("flex"), (function(e) {
        var r = e.$hasDivider, t = e.$gutter;
        var n = t !== undefined ? "".concat(t, "px") : c.variables.spacingXLarge;
        if (r) {
            return (0, l.css)([ "gap:calc(", " / 2);" ], n);
        }
        return (0, l.css)([ "gap:", ";" ], n);
    }));
    var h = o()(y()).withConfig({
        displayName: "ColumnLayoutStyles__StyledDivider",
        componentId: "sc-3eatxz-1"
    })([ "border-color:", ";height:0;" ], c.variables.borderColorWeak);
    // CONCATENATED MODULE: ./src/ColumnLayout/RowStyles.ts
    var O = o().div.withConfig({
        displayName: "RowStyles__Styled",
        componentId: "sc-121ntds-0"
    })([ "", ";flex-flow:row nowrap;", ";&[data-align-items='start']{align-items:flex-start;}&[data-align-items='end']{align-items:flex-end;}&[data-align-items='center']{align-items:center;}&[data-align-items='stretch']{align-items:stretch;}" ], c.mixins.reset("flex"), (function(e) {
        var r = e.$hasDivider, t = e.$gutter;
        var n = t !== undefined ? "".concat(t, "px") : c.variables.spacingXLarge;
        if (r) {
            return (0, l.css)([ "gap:calc(", " / 2);" ], n);
        }
        return (0, l.css)([ "gap:", ";" ], n);
    }));
    var w = o()(y()).withConfig({
        displayName: "RowStyles__StyledDivider",
        componentId: "sc-121ntds-1"
    })([ "border-color:", ";flex:0 0 1;" ], c.variables.borderColorWeak);
    // CONCATENATED MODULE: ./src/ColumnLayout/Row.tsx
    function x() {
        return x = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, x.apply(null, arguments);
    }
    function S(e, r) {
        if (null == e) return {};
        var t, n, a = j(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function j(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var C = {
        alignItems: i().oneOf([ "start", "end", "center", "stretch" ]),
        children: i().node,
        /** @private. */
        divider: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        /** @private. */
        gutter: i().number
    };
    function P(e) {
        var r = e.alignItems, a = r === void 0 ? "stretch" : r, i = e.children, l = e.divider, o = e.elementRef, c = e.gutter, u = S(e, [ "alignItems", "children", "divider", "elementRef", "gutter" ]);
        // @docs-props-type RowPropsBase
                if (false) {
            var s;
        }
        var d = (0, t.useCallback)((function(e) {
            
            return (0, t.cloneElement)(e, {
                gutter: c
            });
        }), [ c ]);
        var f = (0, t.useCallback)((function(e, r, t, a) {
            e.push(r);
            if (l && t < a.length - 1) {
                e.push( n().createElement(w, {
                    orientation: "vertical",
                    decorative: true,
                    key: "".concat(t, "-divider")
                }));
            }
            return e;
        }), [ l ]);
        var v = t.Children.toArray(i).filter(t.isValidElement).map(d).reduce(f, []);
        
        return n().createElement(O, x({
            "data-align-items": a,
            "data-test": "row",
            $hasDivider: l,
            $gutter: c,
            ref: o
        }, u), v);
    }
    P.propTypes = C;
    /* harmony default export */    const _ = P;
    // CONCATENATED MODULE: ./src/ColumnLayout/ColumnLayout.tsx
    function R() {
        return R = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, R.apply(null, arguments);
    }
    function $(e, r) {
        if (null == e) return {};
        var t, n, a = E(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function E(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var k = {
        children: i().node,
        divider: i().oneOf([ "none", "vertical", "horizontal" ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        gutter: i().number
    };
    /**
 * `ColumnLayout` provides simple API for laying out content into columns. It is based on flexbox
 * and has 12 columns.
 */    function I(e) {
        var r = e.children, a = e.divider, i = a === void 0 ? "none" : a, l = e.elementRef, o = e.gutter, c = $(e, [ "children", "divider", "elementRef", "gutter" ]);
        // @docs-props-type ColumnLayoutPropsBase
                var u = (0, t.useCallback)((function(e) {
            
            return (0, t.cloneElement)(e, {
                divider: i === "vertical",
                gutter: o
            });
        }), [ i, o ]);
        var s = (0, t.useCallback)((function(e, r, t, a) {
            e.push(r);
            if (i === "horizontal" && t < a.length - 1) {
                e.push( n().createElement(h, {
                    decorative: true,
                    key: "".concat(t, "-divider")
                }));
            }
            return e;
        }), [ i ]);
        var d = t.Children.toArray(r).filter(t.isValidElement).map(u).reduce(s, []);
        
        return n().createElement(b, R({
            "data-test": "column-layout",
            $gutter: o,
            $hasDivider: i === "horizontal",
            ref: l
        }, c), d);
    }
    I.propTypes = k;
    I.Row = _;
    I.Column = m;
    /* harmony default export */    const T = I;
    // CONCATENATED MODULE: ./src/ColumnLayout/index.ts
    module.exports = r;
    /******/})();
