// @splunk/react-ui/ComboBox.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Divider: () => /* reexport */ d.Divider,
        Heading: () => /* reexport */ d.Heading,
        Option: () => /* reexport */ A,
        default: () => /* reexport */ re
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/extend"
    const o = require("lodash/extend");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/memoize"
    const i = require("lodash/memoize");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/some"
    const u = require("lodash/some");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "prop-types"
    const s = require("prop-types");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const d = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const p = require("@splunk/react-ui/Popover");
    var v =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const b = require("@splunk/react-ui/ResultsMenu");
    var m =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const y = require("@splunk/react-ui/Text");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const h = require("@splunk/react-ui/useControlled");
    var O =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const S = require("@splunk/react-ui/usePrevious");
    var k =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/ui-utils/filter"
    const C = require("@splunk/ui-utils/filter");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const w = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const P = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const j = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/ui-utils/scroll"
    const R = require("@splunk/ui-utils/scroll");
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function M(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/ComboBox/Option.tsx
    function E() {
        return E = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, E.apply(null, arguments);
    }
    function q(e, n) {
        if (null == e) return {};
        var r, t, o = x(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function x(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var B = {
        /** @private */
        active: f().bool,
        description: f().string,
        descriptionPosition: f().oneOf([ "right", "bottom" ]),
        disabled: f().bool,
        elementRef: f().oneOfType([ f().func, f().object ]),
        icon: f().node,
        label: f().string,
        matchRanges: f().arrayOf(f().shape({
            start: f().number.isRequired,
            end: f().number.isRequired
        })),
        /** @private */
        onClick: f().func,
        truncate: f().bool,
        value: f().string.isRequired
    };
    /**
 * An option within a `ComboBox`.
 */    function D(e) {
        var n = e.active, o = e.description, a = e.descriptionPosition, i = a === void 0 ? "bottom" : a, l = e.disabled, u = e.elementRef, c = e.icon, s = e.label, f = e.matchRanges, p = e.onClick, v = e.truncate, b = e.value, m = q(e, [ "active", "description", "descriptionPosition", "disabled", "elementRef", "icon", "label", "matchRanges", "onClick", "truncate", "value" ]);
        // @docs-props-type OptionPropsBase
                var y = (0, r.useCallback)((function(e) {
            if (!l) {
                p === null || p === void 0 ? void 0 : p(e, {
                    value: b
                });
            }
        }), [ l, p, b ]);
        var g = (0, r.useCallback)((function(e) {
            M(u, e);
        }), [ u ]);
        var h = s === undefined ? b : s;
        
        return t().createElement(d.Item, E({
            active: n,
            "data-test": "option",
            "data-test-value": b,
            description: o,
            descriptionPosition: i,
            disabled: l ? "disabled" : undefined,
            elementRef: g,
            startAdornment: c,
            matchRanges: f,
            onClick: y,
            truncate: v
        }, m, {
            role: "option"
        }), h);
    }
    D.propTypes = B;
    D.as = "Item";
    /* harmony default export */    const A = D;
    // CONCATENATED MODULE: ./src/ComboBox/ComboBox.tsx
    function I(e) {
        "@babel/helpers - typeof";
        return I = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, I(e);
    }
    function T() {
        return T = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, T.apply(null, arguments);
    }
    function L(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function H(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? L(Object(r), !0).forEach((function(n) {
                V(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : L(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function V(e, n, r) {
        return (n = z(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function z(e) {
        var n = F(e, "string");
        return "symbol" == I(n) ? n : n + "";
    }
    function F(e, n) {
        if ("object" != I(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != I(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function K(e, n) {
        return X(e) || U(e, n) || N(e, n) || _();
    }
    function _() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function N(e, n) {
        if (e) {
            if ("string" == typeof e) return W(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? W(e, n) : void 0;
        }
    }
    function W(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function U(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, o, a, i, l = [], u = !0, c = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = a.call(r)).done) && (l.push(t.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function X(e) {
        if (Array.isArray(e)) return e;
    }
    function Y(e, n) {
        if (null == e) return {};
        var r, t, o = $(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function $(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var G = {
        animateLoading: f().bool,
        append: f().bool,
        children: f().node,
        controlledFilter: f().bool,
        defaultPlacement: f().oneOf([ "above", "below", "vertical" ]),
        defaultValue: f().string,
        describedBy: f().string,
        disabled: f().bool,
        elementRef: f().oneOfType([ f().func, f().object ]),
        error: f().bool,
        footerMessage: f().node,
        inline: f().bool,
        inputId: f().string,
        inputRef: f().oneOfType([ f().func, f().object ]),
        isLoadingOptions: f().bool,
        labelledBy: f().string,
        loadingMessage: f().node,
        menuStyle: f().object,
        name: f().string,
        noOptionsMessage: f().node,
        onBlur: f().func,
        onChange: f().func,
        onClose: f().func,
        onFocus: f().func,
        onKeyDown: f().func,
        onOpen: f().func,
        onScroll: f().func,
        onScrollBottom: f().func,
        onSelect: f().func,
        placeholder: f().string,
        prepend: f().bool,
        value: f().string,
        /** @private. */
        virtualization: f().number
    };
    function J(e, n) {
        var r = n.clientX, t = n.clientY;
        var o = e.getBoundingClientRect(), a = o.top, i = o.left, l = o.bottom, u = o.right;
        return r > i && r < u && t > a && t < l;
    }
    var Q = l()((function(e) {
        var n = e.anchorWidth, r = e.maxHeight, t = e.menuStyle;
        return a()({
            maxHeight: r,
            overflow: "auto",
            width: Math.max(n !== null && n !== void 0 ? n : 0, 200)
        }, t);
    }));
    var Z = {};
    var ee = (0, w._)("Select...");
    /**
 * `ComboBox` allows the user to select a predefined string or enter a new value. Unlike `Select`
 * and `Multiselect`, `Option` value must always be a string.
 */    function ne(e) {
        var n = e.animateLoading, o = e.append, a = e.children, i = e.controlledFilter, l = e.defaultPlacement, u = l === void 0 ? "vertical" : l, s = e.defaultValue, f = e.describedBy, d = e.disabled, p = e.elementRef, y = e.error, h = e.footerMessage, S = e.inline, E = e.inputId, q = e.inputRef, x = e.isLoadingOptions, B = e.labelledBy, D = e.loadingMessage, I = e.menuStyle, L = I === void 0 ? Z : I, V = e.name, z = e.noOptionsMessage, F = e.onBlur, _ = e.onChange, N = e.onClose, W = e.onFocus, U = e.onKeyDown, X = e.onOpen, $ = e.onScroll, G = e.onScrollBottom, ne = e.onSelect, re = e.placeholder, te = re === void 0 ? ee : re, oe = e.prepend, ae = e.value, ie = e.virtualization, le = Y(e, [ "animateLoading", "append", "children", "controlledFilter", "defaultPlacement", "defaultValue", "describedBy", "disabled", "elementRef", "error", "footerMessage", "inline", "inputId", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onBlur", "onChange", "onClose", "onFocus", "onKeyDown", "onOpen", "onScroll", "onScrollBottom", "onSelect", "placeholder", "prepend", "value", "virtualization" ]);
        // @docs-props-type ComboBoxPropsBase
                var ue = O()({
            componentName: "ComboBox",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var ce = (0, r.useState)(0), se = K(ce, 2), fe = se[0], de = se[1];
        var pe = (0, r.useState)(null), ve = K(pe, 2), be = ve[0], me = ve[1];
        var ye = (0, r.useState)(false), ge = K(ye, 2), he = ge[0], Oe = ge[1];
        var Se = (0, r.useState)(null), ke = K(Se, 2), Ce = ke[0], we = ke[1];
        var Pe = (0, r.useState)(), je = K(Pe, 2), Re = je[0], Me = je[1];
        var Ee = (0, r.useState)(s || ""), qe = K(Ee, 2), xe = qe[0], Be = qe[1];
        var De = k()(fe);
        var Ae = (0, r.useRef)();
        var Ie = (0, r.useRef)(0);
        var Te = (0, r.useRef)(De);
        var Le = (0, P.createDOMID)("popover");
        var He = (0, P.createDOMID)("active-item");
        var Ve = (0, r.useCallback)((function() {
            Oe(false);
            Te.current = null;
            N === null || N === void 0 ? void 0 : N();
        }), [ N ]);
        var ze = (0, r.useCallback)((function() {
            if (!he) {
                Oe(true);
                de(0);
                X === null || X === void 0 ? void 0 : X();
            }
        }), [ he, X ]);
        var Fe = (0, r.useCallback)((function() {
            return ue ? ae : xe;
        }), [ ue, ae, xe ]);
        var Ke = (0, r.useCallback)((function() {
            var e = Fe();
            var n = r.Children.toArray(a).filter(r.isValidElement);
            var t = !Re && n.find((function(n) {
                return n.props.label !== undefined && n.props.value === e;
            }));
            if (t) {
                return t === null || t === void 0 ? void 0 : t.props.label;
            }
            return e;
        }), [ a, Fe, Re ]);
        var _e = (0, r.useCallback)((function(e) {
            we(e);
            M(q, e);
        }), [ q ]);
        var Ne = (0, r.useCallback)((function(e) {
            me(e);
            M(p, e);
        }), [ p ]);
        var We = (0, r.useCallback)((function(e) {
            if (Te.current !== fe) {
                (0, R.scrollIntoViewIfNeeded)(e);
            }
        }), [ fe, Te ]);
        var Ue = (0, r.useCallback)((function(e, n) {
            ze();
            W === null || W === void 0 ? void 0 : W(e, n);
        }), [ W, ze ]);
        var Xe = (0, r.useCallback)((function(e, n) {
            var r = n.value, t = n.selectedOption;
            e.stopPropagation();
            var o = t ? undefined : r;
            de(0);
            Me(o);
            if (!ue) {
                Be(r);
            }
            _ === null || _ === void 0 ? void 0 : _(e, {
                value: r,
                name: V
            });
        }), [ ue, V, _ ]);
        var Ye = (0, r.useCallback)((function(e, n) {
            Xe(e, H(H({}, n), {}, {
                selectedOption: true
            }));
            Ce === null || Ce === void 0 ? void 0 : Ce.focus();
            Ve();
        }), [ Ve, Xe, Ce ]);
        var $e = (0, r.useCallback)((function(e) {
            e.stopPropagation();
            if (!d) {
                ze();
            }
        }), [ d, ze ]);
        var Ge = (0, r.useCallback)((function(e) {
            var n = e.event, r = e.reason;
            var t = r === "offScreen" || r === "escapeKey" || r === "clickAway" && be && !J(be, n);
            if (t) {
                Ve();
            }
        }), [ be, Ve ]);
        var Je = (0, r.useCallback)((function(e) {
            if (he && !x) {
                G === null || G === void 0 ? void 0 : G(e);
            }
        }), [ x, he, G ]);
        var Qe = (0, r.useMemo)((function() {
            var e = Fe();
            var n = r.Children.toArray(a).filter(r.isValidElement);
            var o = c()(n, (function(n) {
                if (n.props.label === undefined) {
                    return n.props.value === e;
                }
                if (n.props.label !== undefined && Re !== undefined) {
                    return n.props.label === e;
                }
                return n.props.value === e;
            }));
            if (!o && e) {
                n.unshift( t().createElement(A, {
                    description: (0, w._)("(New value)"),
                    descriptionPosition: "right",
                    key: "currentValueOption",
                    value: e,
                    "data-test-current-value-option": e
                }));
            }
            // Highlight Active
                        Ie.current = 0;
            Ae.current = undefined;
            var l = Re ? e : Ke();
            var u = l ? (0, C.stringToKeywords)(l) : [];
            return (i ? n : n.filter((function(e) {
                if (e.props.label !== undefined) {
                    return (0, C.testPhrase)(e.props.label, u);
                }
                if (e.props.value !== undefined) {
                    return (0, C.testPhrase)(e.props.value, u);
                }
                return true;
 // Keep all headers and non-interactive options
                        }))).map((function(n, t) {
                if (n.type.as !== "Item") {
                    // ignore non-Option items such as Headings and Dividers
                    return n;
                }
                var a = n.props.label !== undefined ? n.props.label : n.props.value;
                var l = n.props.matchRanges;
                var c = !i && e && (o || t > 0) ? (0, C.keywordLocations)(a, u) || undefined : undefined;
                var s = {
                    onClick: Ye,
                    matchRanges: l || c
                };
                if (n.props.disabled) {
                    
                    return (0, r.cloneElement)(n, s);
                }
                var f = Ie.current === fe;
                Ie.current += 1;
                if (f) {
                    Ae.current = n.props.value;
                    
                    return (0, r.cloneElement)(n, H(H({}, s), {}, {
                        elementRef: We,
                        id: He,
                        active: true
                    }));
                }
                
                return (0, r.cloneElement)(n, s);
            }));
        }), [ fe, He, a, i, Ke, Fe, We, Ye, Re ]);
        var Ze = (0, r.useCallback)((function(e) {
            var n = Ie.current;
            var t = Ae.current;
            var o = e.key, i = e.nativeEvent;
            if (he) {
                switch (o) {
                  case "Enter":
                    {
                        e.preventDefault();
                        if (t) {
                            Ye(e, {
                                value: t
                            });
                        }
                        break;
                    }

                  case "Tab":
                    Ve();
                    break;

                  case "ArrowDown":
                    if (n <= 0) {
                        break;
                    }
                    de((function(e) {
                        return Math.min(e + 1, n - 1);
                    }));
                    if (a && G) {
                        var l = r.Children.count(a) - 2;
                        if (fe === l) {
                            Je(e);
                        }
                    }
                    break;

                  case "ArrowUp":
                    de((function(e) {
                        return Math.max(e - 1, 0);
                    }));
                    break;

                  default:
 // do nothing
                                }
            } else if ((0, j.addsCharacter)(i) !== false || 
            // Safari 9.0 returns undefined
            o === "Enter" || o === "Backspace" || o === "ArrowDown" || o === "ArrowUp") {
                ze();
            }
            U === null || U === void 0 ? void 0 : U(e);
        }), [ fe, a, Ve, Je, Ye, he, U, G, ze ]);
        var en = Fe();
        var nn = Ke();
        var rn = function e(r) {
            var o = r.anchorWidth, a = r.maxHeight;
            var i = Q({
                anchorWidth: o,
                maxHeight: a,
                menuStyle: L
            });
            var l = {
                animateLoading: n,
                controlledExternally: ue,
                footerMessage: h,
                isLoading: x,
                loadingMessage: D,
                noOptionsMessage: z,
                onScroll: $,
                onScrollBottom: G ? Je : undefined,
                style: i
            };
            if (ie) {
                
                return t().createElement(b.VirtualizedResultsMenu, T({
                    virtualization: ie
                }, l), Qe);
            }
            
            return t().createElement(m(), l, Qe);
        };
        
        return t().createElement(g(), T({
            append: o,
            "aria-activedescendant": he && Array.isArray(Qe) && Qe.length > 0 ? He : undefined,
            "aria-expanded": he,
            "aria-controls": he ? Le : undefined,
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            canClear: true,
            "data-test": "combo-box",
            "data-test-popover-id": Le,
            "data-test-label": nn,
            "data-test-value": en,
            "data-test-open": he && !!be,
            describedBy: f,
            disabled: d,
            elementRef: Ne,
            error: y,
            inline: S,
            inputId: E,
            inputRef: _e,
            labelledBy: B,
            name: V,
            onBlur: F,
            onFocus: Ue,
            onChange: Xe,
            onClick: $e,
            onKeyDown: Ze,
            onSelect: ne,
            placeholder: te,
            prepend: oe,
            role: "combobox",
            value: nn
        }, le, {
            spellCheck: false
        }),  t().createElement(v(), {
            anchor: be,
            autoCloseWhenOffScreen: true,
            canCoverAnchor: false,
            defaultPlacement: u,
            id: Le,
            onRequestClose: Ge,
            open: he && !!be,
            repositionMode: "flip"
        }, rn));
    }
    ne.Divider = d.Divider;
    ne.Heading = d.Heading;
    ne.Option = A;
    ne.propTypes = G;
    /* harmony default export */    const re = ne;
    // CONCATENATED MODULE: ./src/ComboBox/index.ts
    module.exports = n;
    /******/})();
