// @splunk/react-ui/ControlGroup.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        ControlGroupContext: () => /* reexport */ p,
        default: () => /* reexport */ ve
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const l = require("@splunk/react-ui/ScreenReaderContent");
    var a =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const s = require("@splunk/react-ui/usePrevious");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const u = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/ControlGroup/ControlGroupContext.tsx
    /**
 * A React context used to inform subcomponets of ControlGroup of parent component values.
 * The context interface is `ControlGroupContextValue`.
 * Defaults to `'{}'`.
 * @public
 */
    var d =  n().createContext({});
    /* harmony default export */    const p = d;
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var y =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const v = require("@splunk/react-ui/Box");
    var b =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const m = require("@splunk/react-ui/Tooltip");
    var h =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/themes"
    const g = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ControlGroup/ControlGroupStyles.ts
    var w = "24px";
    var O = y().div.withConfig({
        displayName: "ControlGroupStyles__StyledHelp",
        componentId: "wjnyif-0"
    })([ "", ";", ";" ], g.mixins.reset("block"), g.mixins.typography("smallBody"));
    var S = y().label.withConfig({
        displayName: "ControlGroupStyles__StyledLabel",
        componentId: "wjnyif-1"
    })([ "", ";word-wrap:break-word;" ], g.mixins.typography("body", {
        color: "active"
    }));
    var x = y()(b()).withConfig({
        displayName: "ControlGroupStyles__StyledBox",
        componentId: "wjnyif-2"
    })([ "", " max-width:600px;margin-block-end:", ";&:last-child{margin-block-end:0;}", " ", "" ], g.mixins.reset("flex"), g.variables.spacingLarge, (function(e) {
        var r = e.$labelPosition;
        return r === "top" ? (0, f.css)([ "flex-direction:column;gap:", ";" ], g.variables.spacingXSmall) : (0, 
        f.css)([ "flex-direction:row;gap:", ";" ], g.variables.spacingMedium);
    }), (function(e) {
        var r = e.$disabled;
        return r && (0, f.css)([ "", "{color:", ";}", "{color:", ";}" ], /* sc-sel */ S, g.variables.contentColorDisabled, /* sc-sel */ O, g.variables.contentColorDisabled);
    }));
    // Aligns label to the top for case where labelPosition="left" and StyledControlsAndMessagingWrapper is taller than label.
        var I = y().div.withConfig({
        displayName: "ControlGroupStyles__StyledLabelWrapper",
        componentId: "wjnyif-3"
    })([ "display:inline-flex;align-items:flex-start;gap:", ";flex-shrink:0;" ], g.variables.spacingXSmall);
    // Vertically centers label & tooltip.
        var C = y().div.withConfig({
        displayName: "ControlGroupStyles__StyledLabelInnerWrapper",
        componentId: "wjnyif-4"
    })([ "display:inline-flex;gap:", ";align-items:center;:has(", ":not(:empty)){min-height:", ";}" ], g.variables.spacingXSmall, /* sc-sel */ S, (0, 
    g.pickVariant)("$labelPosition", {
        /* Ensures label is vertically centered with input */
        left: g.variables.inputHeight,
        /**
   * Ensures inputs in ControlGroups next to each other are vertically aligned regardless of Tooltip presence.
   * Checked by HorizontalLabelTopLayout story.
   */
        top: w
    }));
    var j = y()(h()).withConfig({
        displayName: "ControlGroupStyles__StyledTooltip",
        componentId: "wjnyif-5"
    })([ "flex-shrink:0;& button{min-width:", ";min-height:", ";width:", ";height:", ";}" ], w, w, w, w);
    var E = y().div.withConfig({
        displayName: "ControlGroupStyles__StyledControlsAndMessagingWrapper",
        componentId: "wjnyif-6"
    })([ "display:flex;flex-direction:column;gap:", ";min-width:0;", "" ], g.variables.spacingXSmall, (function(e) {
        var r = e.$controlsLayout;
        return r !== "none" && (0, f.css)([ "flex-grow:1;" ]);
    }));
    var P = y()(b()).withConfig({
        displayName: "ControlGroupStyles__StyledControlsWrapper",
        componentId: "wjnyif-7"
    })([ "position:relative;z-index:1;gap:", ";", "" ], g.variables.spacingMedium, (0, 
    g.pickVariant)("$controlsLayout", {
        fillJoin: (0, f.css)([ "gap:0;" ]),
        stack: (0, f.css)([ "flex-direction:column;" ])
    }));
    var k = y().span.withConfig({
        displayName: "ControlGroupStyles__StyledAsterisk",
        componentId: "wjnyif-8"
    })([ "color:", ";margin-right:2px;" ], g.variables.contentColorNegative);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const _ = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationTriangle"
    const T = require("@splunk/react-icons/ExclamationTriangle");
    var q =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/themes/mixins"
    const D = require("@splunk/themes/mixins");
    // CONCATENATED MODULE: ./src/ControlGroup/ErrorMessageBarStyles.ts
    var M = y().div.withConfig({
        displayName: "ErrorMessageBarStyles__StyledErrorMessageBarWrapper",
        componentId: "sc-1praezx-0"
    })([ "display:flex;gap:", ";word-break:break-word;border-radius:", ";padding:", ";background-color:", ";" ], g.variables.spacingXSmall, g.variables.borderRadius, g.variables.spacingXSmall, g.variables.notificationColorNegativeWeak);
    var L = y()(q()).withConfig({
        displayName: "ErrorMessageBarStyles__StyledIcon",
        componentId: "sc-1praezx-1"
    })([ "position:relative;fill:", ";" ], g.variables.interactiveColorAccentError);
    var N = y().p.withConfig({
        displayName: "ErrorMessageBarStyles__StyledErrorMessage",
        componentId: "sc-1praezx-2"
    })([ "", ";", ";" ], g.mixins.reset("block"), (0, D.typography)("smallBody", {
        color: "active"
    }));
    // CONCATENATED MODULE: ./src/ControlGroup/ErrorMessageBar.tsx
    function B(e, r) {
        if (null == e) return {};
        var t, n, o = G(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function G(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var A = (0, _._)("Error: ");
    var $ = {
        children: i().node.isRequired
    };
    function W(e) {
        var r = e.children, t = B(e, [ "children" ]);
        // @docs-props-type ErrorMessageBarPropsBase
                
        return n().createElement(M, t,  n().createElement(L, {
            variant: "filled"
        }),  n().createElement(a(), null, A),  n().createElement(N, null, r));
    }
    W.propTypes = $;
    /* harmony default export */    const F = W;
    // CONCATENATED MODULE: ./src/ControlGroup/ControlGroup.tsx
    function R(e) {
        "@babel/helpers - typeof";
        return R = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, R(e);
    }
    function X() {
        return X = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, X.apply(null, arguments);
    }
    function z(e, r) {
        if (null == e) return {};
        var t, n, o = H(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function H(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function J(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function V(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? J(Object(t), !0).forEach((function(r) {
                K(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : J(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function K(e, r, t) {
        return (r = U(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function U(e) {
        var r = Q(e, "string");
        return "symbol" == R(r) ? r : r + "";
    }
    function Q(e, r) {
        if ("object" != R(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != R(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function Y(e, r) {
        return ne(e) || te(e, r) || ee(e, r) || Z();
    }
    function Z() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ee(e, r) {
        if (e) {
            if ("string" == typeof e) return re(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? re(e, r) : void 0;
        }
    }
    function re(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function te(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, l, a = [], s = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    s = !1;
                } else for (;!(s = (n = i.call(t)).done) && (a.push(n.value), a.length !== r); s = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!s && null != t["return"] && (l = t["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return a;
        }
    }
    function ne(e) {
        if (Array.isArray(e)) return e;
    }
    // props ControlGroup passes to configure its children's props
    // props ControlGroup may access and/or override on children
        var oe = {
        children: i().node,
        controlsLayout: i().oneOf([ "fill", "fillJoin", "none", "stack" ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().oneOfType([ i().bool, i().string ]),
        help: i().node,
        hideLabel: i().bool,
        label: i().string.isRequired,
        labelFor: i().string,
        labelPosition: i().oneOf([ "left", "top" ]),
        labelWidth: i().oneOfType([ i().number, i().string ]),
        required: i().bool,
        tooltip: i().node,
        tooltipDefaultPlacement: i().oneOf([ "above", "below", "left", "right" ])
    };
    // Style cloned onto child elements when fill behavior set.
        var ie = {
        flexGrow: 1
    };
    function le(e) {
        return e.type && e.type.propTypes && Object.prototype.hasOwnProperty.call(e.type.propTypes, "inputId");
    }
    function ae(e, r) {
        var t = "".concat(e, "-").concat(r);
        var n = (0, u.createDOMID)(e);
        return {
            childKey: t,
            value: n
        };
    }
    function se(e) {
        var r = {};
        var t = e.length;
        var n = 0;
        e.forEach((function(e, t) {
            if (le(e) && e.props.inputId == null) {
                var o = ae("input-id", t), i = o.childKey, l = o.value;
                r[i] = l;
            } else if (e.props.id == null) {
                var a = ae("id", t), s = a.childKey, c = a.value;
                r[s] = c;
            } else {
                // child already has an inputId or id, so we don't need to make a childId for it
                n += 1;
            }
        }));
        return {
            newChildIds: r,
            allChildrenHaveId: n === t
        };
    }
    function ce(e, r, t) {
        var n = "".concat(e, "-").concat(r);
        var o = t[n];
        if (!o || !o.length) {
            // By this point we should have already generated ids for each child and this code shouldn't get executed
            // But leaving this as a fallback
            return undefined;
        }
        return o;
    }
    function ue(e) {
        if (e.length === 0) {
            return undefined;
        }
        var r = Y(e, 1), t = r[0];
        var n = e.find((function(e) {
            return e.type && e.type.componentType === "Text";
        }));
        var o = le(t);
        var i = t.props, l = i.id, a = i.inputId;
 // see function doc block for details
                if (e.length === 1 && (t === null || t === void 0 ? void 0 : t.props)) {
            return o ? a : l;
        }
        if (n === null || n === void 0 ? void 0 : n.props) {
            return n.props.inputId;
        }
        if (t.props) {
            return o ? a : l;
        }
        return undefined;
    }
    // Configure the Children
        function de(e, r, o) {
        var i = e.props, l = i.disabled, a = i.error, s = i.id, c = i.inputId, u = i.style, d = i.children;
        var p = o.childIds, f = o.controlsLayout, y = o.help, v = o.helpId, b = o.error, m = o.errorId, h = o.label, g = o.labelFor, w = o.labelId, O = o.numChildren, S = o.required, x = o.showErrorText;
        var I = {
            key: e.key || r,
            labelledBy: w
        };
        if (f === "fillJoin") {
            if (r > 0) {
                I.prepend = true;
            }
            if (r < O - 1) {
                I.append = true;
            }
            if (I.prepend || I.append) {
                I.inline = false;
            }
        }
        if (f === "fill" && O > 1) {
            I.inline = true;
        }
        if (f === "stack") {
            I.inline = false;
        }
        if (y && typeof b === "boolean") {
            I.describedBy = v;
        } else if (y && typeof b === "string") {
            I.describedBy = "".concat(v, " ").concat(m);
        } else if (!y && typeof b === "string") {
            I.describedBy = m;
        }
        if (S) {
            I.required = true;
        }
        // TODO: SUI-3402 refactor so that parent component doesn't have to know what it's child components are
                if (e.type.componentType === "Multiselect") {
            var C = e;
            if (!C.props.compact) {
                I.labelText = h;
            }
        } else {
            I.labelText = h;
        }
        if (O === 1 && (f === "fillJoin" || f === "fill")) {
            // Some controls like Select do not grow by default, so flexGrow is applied
            // The spread operator is used as css styles are not nested, so there is no need for a deep copy.
            I.style = u ? V(V({}, u), ie) : ie;
        }
        if (!g && e) {
            if (le(e)) {
                I.inputId = c || ce("input-id", r, p);
            } else {
                I.id = s || ce("id", r, p);
            }
        }
        if (x && a === true) {
            I["aria-errormessage"] = m;
        }
        var j = l !== null && l !== void 0 ? l : false;
        // RadioBar and RadioList need special handling in the case where the parent grouping component is not set to disabled
        // but all of the children are disabled.
                var E = e.type.componentType === "RadioBar" || e.type.componentType === "RadioList";
        if (E && !e.props.disabled) {
            j = n().Children.toArray(d).every((function(e) {
                var r;
                
                return n().isValidElement(e) && ((r = e.props) === null || r === void 0 ? void 0 : r.disabled);
            }));
        }
        return {
            child:  (0, t.cloneElement)(e, I),
            childDisabled: j
        };
    }
    function pe(e, r) {
        var t = e.length;
        var n = 0;
        var o = V({
            numChildren: t
        }, r);
        var i = e.map((function(e, r) {
            var t = de(e, r, o), i = t.child, l = t.childDisabled;
            if (l === true) {
                n += 1;
            }
            return i;
        }));
        return {
            childrenFormatted: i,
            controlGroupDisabled: n === t
        };
    }
    var fe = 120;
    /**
 * `ControlGroup` places a label and optional help text around one or more controls. The `ControlGroup`
 * will automatically add aria attributes to associate the controls with the labels and help text to
 * address accessibility requirements.
 *
 * `ControlGroup` provides layouts to assist in aligning and laying out controls, but the defaults are
 * not helpful in all cases, nor will the layout options address all cases. Consider setting
 * `controlsLayout` to none and manually positioning the controls as required.
 *
 * `ControlGroup` uses the HTML `label` tag. The rules for determining which child component is linked
 * to the label's `for` attribute are:
 * 1. If one or more `children` are `Text` components, the first one is used.
 * 2. If there aren't any `Text` components, the first child is used.
 *
 * If the linked child supports an `inputId` prop and it's set, its value is used for the label's `for`
 * attribute. If `inputId` is supported but not set a generated id is used instead. If `inputId` isn't
 * supported `id` is used. The `labelFor` prop may be used to override the `for` attribute.
 */    function ye(e) {
        var r = e.children, o = e.controlsLayout, i = o === void 0 ? "fill" : o, l = e.error, s = l === void 0 ? false : l, d = e.help, f = e.hideLabel, y = e.label, v = e.labelFor, b = e.labelPosition, m = b === void 0 ? "top" : b, h = e.labelWidth, g = h === void 0 ? fe : h, w = e.required, _ = w === void 0 ? false : w, T = e.tooltip, q = e.tooltipDefaultPlacement, D = z(e, [ "children", "controlsLayout", "error", "help", "hideLabel", "label", "labelFor", "labelPosition", "labelWidth", "required", "tooltip", "tooltipDefaultPlacement" ]);
        // @docs-props-type ControlGroupPropsBase
                var M = (0, t.useState)((function() {
            return {
                errorId: (0, u.createDOMID)("error"),
                helpId: (0, u.createDOMID)("help"),
                labelId: (0, u.createDOMID)("label")
            };
        })), L = Y(M, 1), N = L[0], B = N.errorId, G = N.helpId, A = N.labelId;
        /** Child ids need to be the same on each render because of SUI-2817 */        var $ = (0, 
        t.useState)({}), W = Y($, 2), R = W[0], H = W[1];
        var J = (0, t.useState)(false), V = Y(J, 2), K = V[0], U = V[1];
        var Q = c()(n().Children.count(r));
        var Z = (0, t.useCallback)((function(e) {
            H(e);
        }), []);
        var ee = (0, t.useCallback)((function(e) {
            U(e);
        }), []);
        var re = (0, t.useMemo)((function() {
            return t.Children.toArray(r).filter(t.isValidElement);
        }), [ r ]);
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ s, d, m ]);
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ m, g ]);
        (0, t.useEffect)((function() {
            // If it's not the first render and the number of child elements has changed, reset childIds
            if (Q != null && n().Children.count(r) !== Q) {
                H({});
            }
            // If childIds has already been populated, labelFor has a value, or if all children have ids then we don't need to make ids for the children.
                        if (Object.keys(R).length === 0 && !v && !K) {
                var e = se(re), t = e.newChildIds, o = e.allChildrenHaveId;
                H(t);
                U(o);
            }
        }), [ K, R, r, v, Q, Z, ee, re ]);
        var te = (0, t.useMemo)((function() {
            return {
                labelAttrs: {
                    text: y,
                    id: A
                }
            };
        }), [ y, A ]);
        var ne = typeof s === "string";
        var oe = ne && s !== "";
        var ie = Number.isFinite(g) ? "".concat(g, "px") : g;
        var le = m === "left" ? {
            width: ie
        } : undefined;
        var ae = pe(re, {
            childIds: R,
            controlsLayout: i,
            help: d,
            helpId: G,
            error: s,
            errorId: B,
            label: y,
            labelFor: v,
            labelId: A,
            required: _,
            showErrorText: oe
        }), ce = ae.childrenFormatted, de = ae.controlGroupDisabled;
        var ye = (0, t.useMemo)((function() {
            return v || ue(ce);
        }), [ ce, v ]);
        var ve =  n().createElement(I, {
            style: le,
            $labelPosition: m
        },  n().createElement(C, {
            $labelPosition: m
        },  n().createElement(S, {
            "data-test": "label",
            id: A,
            htmlFor: ye
        }, _ &&  n().createElement(k, {
            "aria-hidden": "true"
        }, "*"), y, !f && T &&  n().createElement(a(), null, T)), !f && T &&  n().createElement(j, {
            closeWhen: "notOnClick",
            content: T,
            defaultPlacement: q,
            "aria-hidden": "true"
        })));
        
        return n().createElement(x, X({
            "data-test": "control-group",
            "data-test-required": _,
            "data-test-children-disabled": de,
            $labelPosition: m,
            $disabled: de,
            "aria-invalid": !!s || undefined
        }, D),  n().createElement(p.Provider, {
            value: te
        }, f ?  n().createElement(a(), null, ve) : ve,  n().createElement(E, {
            $controlsLayout: i
        },  n().createElement(P, {
            $controlsLayout: i,
            "data-test": "controls",
            flex: i !== "none"
        }, ce), oe &&  n().createElement(F, {
            "data-test": "error",
            id: B
        }, s), d &&  n().createElement(O, {
            "data-test": "help",
            id: G
        }, d))));
    }
    ye.propTypes = oe;
    /* harmony default export */    const ve = ye;
    // CONCATENATED MODULE: ./src/ControlGroup/index.ts
    module.exports = r;
    /******/})();
