// @splunk/react-ui/Date.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ V,
        validateValue: () => /* reexport */ F
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "lodash/omit"
    const a = require("lodash/omit");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "moment"
    const l = require("moment");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/Calendar"
    const f = require("@splunk/react-ui/Calendar");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const d = require("@splunk/react-ui/Popover");
    var v =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const p = require("@splunk/react-ui/useControlled");
    var y =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const m = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const b = require("styled-components");
    var g =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/Calendar"
    const h = require("@splunk/react-icons/Calendar");
    var C =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const k = require("@splunk/react-ui/Text");
    var S =  e.n(k);
    // CONCATENATED MODULE: ./src/Date/DateStyles.ts
    var O = g()(S()).withConfig({
        displayName: "DateStyles__StyledText",
        componentId: "kacgdc-0"
    })([ "&[data-inline]{width:160px;}" ]);
    var w = g()(C()).withConfig({
        displayName: "DateStyles__StyledCalendarIcon",
        componentId: "kacgdc-1"
    })([ "transform:translateY(1px);" ]);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function D(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Date/Date.tsx
    function j() {
        return j = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, j.apply(null, arguments);
    }
    function x(e, n) {
        return E(e) || A(e, n) || q(e, n) || I();
    }
    function I() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function q(e, n) {
        if (e) {
            if ("string" == typeof e) return T(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? T(e, n) : void 0;
        }
    }
    function T(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function A(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, a, o, l, i = [], u = !0, c = !1;
            try {
                if (o = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = o.call(t)).done) && (i.push(r.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (l = t["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function E(e) {
        if (Array.isArray(e)) return e;
    }
    function M(e, n) {
        if (null == e) return {};
        var t, r, a = P(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (r = 0; r < o.length; r++) {
                t = o[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function P(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */
    /** @public */
    /** @public */    var _ = {
        append: c().bool,
        canClear: c().bool,
        defaultValue: c().string,
        describedBy: c().string,
        disabled: c().bool,
        elementRef: c().oneOfType([ c().func, c().object ]),
        error: c().bool,
        highlightToday: c().bool,
        inline: c().bool,
        inputId: c().string,
        inputOnly: c().bool,
        labelledBy: c().string,
        locale: c().string,
        name: c().string,
        onBlur: c().func,
        onChange: c().func,
        onClick: c().func,
        onFocus: c().func,
        onKeyDown: c().func,
        prepend: c().bool,
        value: c().string
    };
    /**
 * This static value can be used to convert a moment date to a compatible string
 * to set the `value` prop.
 * ```
 * moment().format(Date.momentFormat);
 * ```
 * @public
 * @name momentFormat
 * @memberOf Date
 * @type string
 */    var R = "YYYY-MM-DD";
    var F = function e(n) {
        if (n === "") {
            return;
        }
        if (!i()(n, R).isValid()) {
            // eslint-disable-next-line no-console
            console.error('Invalid date string value "'.concat(n, '" provided to Moment.'));
        }
    };
    function K(e) {
        var n = e.append, a = e.canClear, l = e.defaultValue, u = e.disabled, c = e.elementRef, d = e.highlightToday, p = e.inline, b = p === void 0 ? true : p, g = e.inputOnly, h = e.locale, C = h === void 0 ? "en_US" : h, k = e.name, S = e.onChange, I = e.onClick, q = e.onFocus, T = e.onKeyDown, A = e.prepend, E = e.value, P = M(e, [ "append", "canClear", "defaultValue", "disabled", "elementRef", "highlightToday", "inline", "inputOnly", "locale", "name", "onChange", "onClick", "onFocus", "onKeyDown", "prepend", "value" ]);
        // @docs-props-type DatePropsBase
                var _ = y()({
            componentName: "Date",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var F = (0, t.useState)(null), V = x(F, 2), Y = V[0], B = V[1];
        var N = (0, t.useState)(false), U = x(N, 2), z = U[0], $ = U[1];
        var G = (0, t.useState)(), H = x(G, 2), J = H[0], L = H[1];
        var Q = (0, t.useState)((function() {
            if (typeof l === "string") {
                return l;
            }
            return i()().locale(C).format(R);
        })), W = x(Q, 2), X = W[0], Z = W[1];
        var ee = (0, t.useState)(null), ne = x(ee, 2), te = ne[0], re = ne[1];
        var ae = (0, t.useState)(null), oe = x(ae, 2), le = oe[0], ie = oe[1];
        var ue = (0, t.useRef)(false);
        var ce = (0, m.createDOMID)("calendar");
        var fe = (0, t.useMemo)((function() {
            return _ ? E : X;
        }), [ _, E, X ]);
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ fe ]);
        (0, t.useEffect)((function() {
            /**
     * If `openWithFocus` and `calendarOpen` are set the user is attempting to navigate to the dropdown, but it's closed.
     * On the first update following the state change set focus to the active day within the calendar.
     */
            if (ue.current && z) {
                ue.current = false;
                (0, f.focusActiveDay)(te);
            }
        }), [ te, z ]);
        var se = (0, t.useMemo)((function() {
            if (J === undefined) {
                if (fe === undefined || fe === "") {
                    return "";
                }
                return i()(fe, R, C).format("l");
            }
            return J;
        }), [ fe, C, J ]);
        var de = (0, t.useCallback)((function(e, n, t) {
            var r = n.value;
            if (t !== "textInput") {
                le === null || le === void 0 ? void 0 : le.focus();
            }
            if (fe !== r) {
                $(false);
                L(undefined);
                Z(_ ? undefined : r);
                S === null || S === void 0 ? void 0 : S(e, {
                    value: r,
                    name: k
                });
            } else {
                $(false);
            }
        }), [ fe, le, _, k, S ]);
        var ve = (0, t.useCallback)((function(e) {
            var n = i()(J, "l", C);
            if (J === "") {
                de(e, {
                    value: ""
                }, "textInput");
            } else if (n.isValid()) {
                de(e, {
                    value: n.format(K.momentFormat)
                }, "textInput");
            } else {
                $(false);
                L(undefined);
            }
        }), [ de, C, J ]);
        var pe = (0, t.useCallback)((function(e, n) {
            if (!g) {
                $(true);
            }
            q === null || q === void 0 ? void 0 : q(e, n);
        }), [ g, q ]);
        var ye = (0, t.useCallback)((function(e) {
            if (!g && !z) {
                $(true);
            }
            I === null || I === void 0 ? void 0 : I(e);
        }), [ z, g, I ]);
        var me = (0, t.useCallback)((function(e, n) {
            var t = n.value;
            L(t);
            $(!g);
        }), [ g ]);
        var be = (0, t.useCallback)((function(e) {
            var n = e.key;
            if (n === "Enter" || n === "Tab") {
                ve(e);
            } else if (!g && n === "ArrowDown") {
                e.preventDefault();
                if (!z) {
                    ue.current = true;
                    $(true);
                } else if (te) {
                    (0, f.focusActiveDay)(te);
                }
            } else if (n === "Escape") {
                $(false);
                L(undefined);
                le === null || le === void 0 ? void 0 : le.focus();
            }
            T === null || T === void 0 ? void 0 : T(e);
        }), [ ve, te, z, le, g, T ]);
        var ge = (0, t.useCallback)((function(e) {
            B(e);
            D(c, e);
        }), [ c ]);
        var he = (0, t.useCallback)((function(e) {
            ie(e);
        }), []);
        var Ce = (0, t.useCallback)((function(e) {
            re(e);
        }), []);
        var ke = (0, t.useCallback)((function(e) {
            var n = e.event, t = e.reason;
            if ((t === "clickAway" || t === "escapeKey") && n !== undefined && n.target !== Y) {
                ve(n);
                if (t === "escapeKey") {
                    le === null || le === void 0 ? void 0 : le.focus();
                }
            }
        }), [ Y, le, ve ]);
        var Se = !g &&  r().createElement(w, null);
        var Oe;
        var we = {};
        if (!g) {
            we.role = "combobox";
            we["aria-controls"] = z ? ce : undefined;
            we["aria-expanded"] = u ? false : z;
            we["aria-haspopup"] = true;
            Oe =  r().createElement(v(), {
                anchor: Y,
                id: ce,
                open: u ? false : z,
                onRequestClose: ke
            },  r().createElement(s(), {
                elementRef: Ce,
                highlightToday: d,
                locale: C,
                onChange: de,
                value: fe === "" ? undefined : fe
            }));
        }
        
        return r().createElement(O, j({
            append: n,
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            canClear: a,
            "data-test": "date",
            "data-test-value": fe,
            "data-test-popover-id": ce,
            disabled: u,
            elementRef: ge,
            inline: b,
            inputRef: he,
            name: k,
            onChange: me,
            onClick: ye,
            onFocus: pe,
            onKeyDown: be,
            prepend: A,
            value: se,
            spellCheck: false,
            startAdornment: Se
        }, we, o()(P, "multiline", "onSelect", "spellCheck")), Oe);
    }
    K.propTypes = _;
    K.momentFormat = R;
    /* harmony default export */    const V = K;
    // only for styled components using Date
        // CONCATENATED MODULE: ./src/Date/index.ts
    module.exports = n;
    /******/})();
