// @splunk/react-ui/DefinitionList.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Description: () => /* reexport */ b,
        Term: () => /* reexport */ j,
        default: () => /* reexport */ T
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: ./src/DefinitionList/DefinitionListContext.tsx
    var s =  (0, t.createContext)({
        layout: "fixed"
    });
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/DefinitionList/DefinitionListStyles.ts
    var f = c().dt.withConfig({
        displayName: "DefinitionListStyles__StyledTerm",
        componentId: "sc-1pqfrgm-0"
    })([ "", "" ], (function(e) {
        var r = e.$separatorCharacter, t = e.$termWidth;
        return (0, d.pickVariant)("$layout", {
            fixed: (0, l.css)([ "", ";", ";float:left;overflow:hidden;white-space:nowrap;word-wrap:normal;width:", ";", "" ], d.mixins.reset("block"), d.mixins.typography("body", {
                color: "muted"
            }), t, r && (0, l.css)([ "&::after{content:'", "';}" ], " ".concat(r.repeat(100)))),
            auto: (0, l.css)([ "", ";", ";grid-template-columns:[term] auto [separator] 1fr;grid-column:term;overflow:hidden;white-space:normal;word-break:break-word;width:", ";", "" ], d.mixins.reset("grid"), d.mixins.typography("body", {
                color: "muted"
            }), t, r && (0, l.css)([ "&::after{grid-column:separator;content:'", "';margin-inline-start:", ";min-width:", ";overflow:hidden;white-space:nowrap;}" ], " ".concat(r.repeat(250)), d.variables.spacingXSmall, d.variables.spacingLarge)),
            stacked: (0, l.css)([ "", ";", ";grid-column:definitionList;white-space:normal;word-break:break-word;&:not(:first-of-type){margin-block-start:", ";}" ], d.mixins.reset("grid"), d.mixins.typography("body", {
                color: "muted"
            }), d.variables.spacingLarge)
        });
    }));
    var p = c().dd.withConfig({
        displayName: "DefinitionListStyles__StyledDescription",
        componentId: "sc-1pqfrgm-1"
    })([ "", "" ], (function(e) {
        var r = e.$descriptionWidth, t = e.$termWidth;
        return (0, d.pickVariant)("$layout", {
            fixed: (0, l.css)([ "", ";", ";padding-inline-start:", ";margin-inline-start:", ";white-space:normal;word-break:break-word;width:", ";" ], d.mixins.reset("block"), d.mixins.typography("body", {
                color: "active"
            }), d.variables.spacingSmall, t, r),
            auto: (0, l.css)([ "", ";", ";grid-column:description;white-space:normal;word-break:break-word;width:", ";" ], d.mixins.reset("grid"), d.mixins.typography("body", {
                color: "active"
            }), r),
            stacked: (0, l.css)([ "", ";", ";grid-column:definitionList;white-space:normal;word-break:break-word;" ], d.mixins.reset("grid"), d.mixins.typography("body", {
                color: "active"
            }))
        });
    }));
    var u = c().dl.withConfig({
        displayName: "DefinitionListStyles__Styled",
        componentId: "sc-1pqfrgm-2"
    })([ "", "" ], (0, d.pickVariant)("$layout", {
        fixed: (0, l.css)([ "", ";", ";" ], d.mixins.reset("block"), d.mixins.typography("body")),
        auto: (0, l.css)([ "", ";grid-template-columns:[term] fit-content(", ") [description] fit-content(", ");column-gap:", ";" ], d.mixins.reset("grid"), d.variables.lineLength, d.variables.lineLength, d.variables.spacingSmall),
        stacked: (0, l.css)([ "", ";grid-template-columns:[definitionList] fit-content(", ");" ], d.mixins.reset("grid"), d.variables.lineLength)
    }));
    // CONCATENATED MODULE: ./src/DefinitionList/Description.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function y(e, r) {
        if (null == e) return {};
        var t, n, i = h(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                t = a[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function h(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var g = {
        children: o().node.isRequired,
        elementRef: o().oneOfType([ o().func, o().object ])
    };
    /**
 * Container component for a `DefinitionList` description.
 */    function v(e) {
        var r = e.children, i = e.elementRef, a = y(e, [ "children", "elementRef" ]);
        // @docs-props-type DescriptionPropsBase
                var o = (0, t.useContext)(s), l = o.descriptionWidth, c = o.layout, d = o.termWidth;
        
        return n().createElement(p, m({
            "data-test": "description",
            ref: i,
            $descriptionWidth: l,
            $layout: c,
            $termWidth: d
        }, a), r);
    }
    v.propTypes = g;
    /* harmony default export */    const b = v;
    // CONCATENATED MODULE: ./src/DefinitionList/Term.tsx
    function w() {
        return w = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, w.apply(null, arguments);
    }
    function O(e, r) {
        if (null == e) return {};
        var t, n, i = x(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                t = a[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function x(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var k = {
        children: o().node.isRequired,
        elementRef: o().oneOfType([ o().func, o().object ])
    };
    /**
 * Container component for a `DefinitionList` term.
 */    function S(e) {
        var r = e.children, i = e.elementRef, a = O(e, [ "children", "elementRef" ]);
        // @docs-props-type TermPropsBase
                var o = (0, t.useContext)(s), l = o.layout, c = o.separatorCharacter, d = o.termWidth;
        
        return n().createElement(f, w({
            "data-test": "term",
            ref: i,
            $layout: l,
            $separatorCharacter: c,
            $termWidth: d
        }, a), r);
    }
    S.propTypes = k;
    /* harmony default export */    const j = S;
    // CONCATENATED MODULE: ./src/DefinitionList/DefinitionList.tsx
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, P.apply(null, arguments);
    }
    function W(e, r) {
        if (null == e) return {};
        var t, n, i = $(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                t = a[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function $(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var C = {
        children: o().node,
        descriptionWidth: o().string,
        elementRef: o().oneOfType([ o().func, o().object ]),
        layout: o().oneOf([ "fixed", "auto", "stacked" ]),
        separatorCharacter: o().string,
        termWidth: o().string
    };
    var L = "120px";
    function R(e) {
        var r = e.children, i = e.descriptionWidth, a = e.elementRef, o = e.layout, l = o === void 0 ? "fixed" : o, c = e.separatorCharacter, d = e.termWidth, f = d === void 0 ? L : d, p = W(e, [ "children", "descriptionWidth", "elementRef", "layout", "separatorCharacter", "termWidth" ]);
        // @docs-props-type DefinitionListPropsBase
        /* eslint-disable-next-line prefer-rest-params */        var m = arguments[0];
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ l, c, m, i ]);
        var y;
        var h;
        // termWidth and descriptionWidth are not supported in stacked layout
                if (l === "stacked") {
            y = undefined;
            h = undefined;
        } else {
            y = l !== "fixed" && f === L ? undefined : f;
            h = i;
        }
        var g = (0, t.useMemo)((function() {
            return {
                descriptionWidth: h,
                layout: l,
                separatorCharacter: c,
                termWidth: y
            };
        }), [ h, l, c, y ]);
        
        return n().createElement(u, P({
            "data-test": "definition-list",
            ref: a,
            $layout: l
        }, p),  n().createElement(s.Provider, {
            value: g
        }, r));
    }
    R.propTypes = C;
    R.Description = b;
    R.Term = j;
    /* harmony default export */    const T = R;
    // CONCATENATED MODULE: ./src/DefinitionList/index.ts
    module.exports = r;
    /******/})();
