// @splunk/react-ui/Dropdown.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = o => {
            /******/ var n = o && o.__esModule ? 
            /******/ () => o["default"]
            /******/ : () => o
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (o, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(o, r)) {
                    /******/ Object.defineProperty(o, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, o) => Object.prototype.hasOwnProperty.call(e, o)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var o = {};
    // ESM COMPAT FLAG
        e.r(o);
    // EXPORTS
        e.d(o, {
        default: () => /* reexport */ w
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const t = require("prop-types");
    var a =  e.n(t);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const l = require("@splunk/react-ui/Popover");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const u = require("@splunk/react-ui/useControlled");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const s = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function f(e, o) {
        if (e) {
            if (typeof e === "function") {
                e(o);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = o;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Dropdown/Dropdown.tsx
    function d(e, o) {
        return b(e) || g(e, o) || v(e, o) || p();
    }
    function p() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function v(e, o) {
        if (e) {
            if ("string" == typeof e) return y(e, o);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? y(e, o) : void 0;
        }
    }
    function y(e, o) {
        (null == o || o > e.length) && (o = e.length);
        for (var n = 0, r = Array(o); n < o; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function g(e, o) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, t, a, l, i = [], u = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === o) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = a.call(n)).done) && (i.push(r.value), i.length !== o); u = !0) {
                }
            } catch (e) {
                c = !0, t = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (l = n["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw t;
                }
            }
            return i;
        }
    }
    function b(e) {
        if (Array.isArray(e)) return e;
    }
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var m = [ "toggleClick", "toggleKeydown" ];
    var C = [ "clickAway", "contentClick", "escapeKey", "offScreen", "tabKey", "toggleClick" ];
    var k = [ "contentClick", "escapeKey", "tabKey", "toggleClick" ];
    var h = {
        align: a().oneOf([ "center", "edge", "end" ]),
        canCoverAnchor: a().bool,
        children: a().oneOfType([ a().node, a().func ]),
        closeReasons: a().arrayOf(a().oneOf(C)),
        defaultPlacement: a().oneOf([ "above", "below", "left", "right", "vertical", "horizontal" ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        focusToggleReasons: a().arrayOf(a().oneOf(C)),
        inputId: a().string,
        onRequestClose: a().func,
        onRequestOpen: a().func,
        open: a().bool,
        openWithArrowKeys: a().bool,
        repositionMode: a().oneOf([ "none", "flip", "any" ]),
        retainFocus: a().bool,
        takeFocus: a().bool,
        toggle: a().element.isRequired
    };
    function O(e) {
        var o = e["aria-labelledby"], t = e.align, a = t === void 0 ? "center" : t, l = e.canCoverAnchor, u = l === void 0 ? true : l, p = e.children, v = e.closeReasons, y = v === void 0 ? C : v, g = e.defaultPlacement, b = g === void 0 ? "below" : g, m = e.elementRef, h = e.focusToggleReasons, O = h === void 0 ? k : h, w = e.inputId, R = e.open, S = e.openWithArrowKeys, A = e.onRequestClose, I = e.onRequestOpen, q = e.repositionMode, P = q === void 0 ? "flip" : q, D = e.retainFocus, K = D === void 0 ? false : D, j = e.takeFocus, M = j === void 0 ? true : j, T = e.toggle;
        // @docs-props-type DropdownPropsBase
                var F = (0, n.useState)(null), E = d(F, 2), N = E[0], _ = E[1];
        var x = (0, n.useState)(false), W = d(x, 2), U = W[0], z = W[1];
        var V = (0, n.useRef)(null);
        var $ = c()({
            componentName: "Dropdown",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultOpen",
            valuePropName: "open"
        });
        var B = $ ? R : U;
        var G = (0, n.useState)((function() {
            return {
                popoverId: (0, s.createDOMID)("popover"),
                toggleId: (0, s.createDOMID)("toggle")
            };
        })), H = d(G, 1), J = H[0], L = J.popoverId, Q = J.toggleId;
        var X = w || T.props.id || Q;
        var Y = (0, n.useCallback)((function(e) {
            V.current = e;
            _(e);
            // TS: ref does not exist on React.ReactElement - technically, this is undocumented API
                        f(T.ref, e);
        }), [ T ]);
        var Z = T.props.disabled;
        var ee = Z === true || Z === "disabled" || Z === "dimmed";
        var oe = (0, n.useCallback)((function(e) {
            var o, n;
            if (ee) {
                return;
            }
            (o = (n = T.props).onKeyDown) === null || o === void 0 ? void 0 : o.call(n, e);
            if (!S) {
                return;
            }
            var r = e.key;
            if (r === "ArrowUp" || r === "ArrowDown") {
                e.preventDefault();
                I === null || I === void 0 ? void 0 : I(e, {
                    reason: "toggleKeydown"
                });
                if (!$) {
                    z(true);
                }
            }
        }), [ $, ee, S, I, T ]);
        var ne = (0, n.useCallback)((function(e) {
            var o = e.event, n = e.reason;
            if (o && n === "clickAway") {
                var r = o.target;
                while (r) {
                    // Ignore clicks on toggle.
                    if (V.current === r) {
                        return;
                    }
                    r = r.parentNode;
                }
            }
            if (B && y.includes(n)) {
                if (O.includes(n)) {
                    var t;
                    (t = V.current) === null || t === void 0 ? void 0 : t.focus();
                }
                if (!$) {
                    z(false);
                }
                A === null || A === void 0 ? void 0 : A(e);
            }
        }), [ y, O, $, B, A ]);
        var re = (0, n.useCallback)((function(e) {
            ne({
                reason: "contentClick",
                event: e
            });
        }), [ ne ]);
        var te = (0, n.useCallback)((function(e) {
            var o, n;
            if (ee) {
                e.preventDefault();
                return;
            }
            (o = (n = T.props).onClick) === null || o === void 0 ? void 0 : o.call(n, e);
            if (B) {
                ne({
                    reason: "toggleClick",
                    event: e
                });
            } else {
                I === null || I === void 0 ? void 0 : I(e, {
                    reason: "toggleClick"
                });
                if (!$) {
                    z(true);
                }
            }
        }), [ ne, $, B, ee, I, T ]);
        var ae =  (0, n.cloneElement)(T, {
            onClick: te,
            onKeyDown: oe,
            ref: Y,
            "aria-controls": B ? L : undefined,
            "aria-haspopup": T.props["aria-haspopup"] || true,
            "aria-expanded": B,
            "data-test": T.props["data-test"] || "dropdown",
            "data-test-popover-id": L,
            id: X
        });
        
        return r().createElement(r().Fragment, null, ae,  r().createElement(i(), {
            align: a,
            open: !!N && B,
            autoCloseWhenOffScreen: y.includes("offScreen"),
            anchor: N,
            canCoverAnchor: u,
            elementRef: m,
            retainFocus: K,
            defaultPlacement: b,
            onRequestClose: ne,
            repositionMode: P,
            id: L,
            "aria-labelledby": "".concat(o !== null && o !== void 0 ? o : "", " ").concat(X).trim(),
            takeFocus: M,
            toggleId: X
        }, typeof p === "function" ? function() {
            
            // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
            return r().createElement("div", {
                onClick: re
            }, p.apply(void 0, arguments));
        } : 
        
        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
        r().createElement("div", {
            onClick: re
        }, p)));
    }
    O.propTypes = h;
    O.possibleOpenReasons = m;
    O.possibleCloseReasons = C;
    /* harmony default export */    const w = O;
    // CONCATENATED MODULE: ./src/Dropdown/index.ts
    module.exports = o;
    /******/})();
