// @splunk/react-ui/DualListbox.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Option: () => /* reexport */ ne,
        default: () => /* reexport */ He
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronLeft"
    const i = require("@splunk/react-icons/ChevronLeft");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const u = require("@splunk/react-icons/ChevronRight");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/useForceUpdate"
    const s = require("@splunk/react-ui/useForceUpdate");
    var v =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const f = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const d = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/scroll"
    const p = require("@splunk/ui-utils/scroll");
    // CONCATENATED MODULE: ./src/DualListbox/DualListboxContext.tsx
    /**
 * A React context used to inform subcomponets of DualListbox of parent component values.
 * The context interface is `DualListboxContextValue`.
 * Defaults to `'{}'`.
 * @public
 */
    var b =  n().createContext({});
    /* harmony default export */    const m = b;
    // CONCATENATED MODULE: external "styled-components"
    const h = require("styled-components");
    var g =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const y = require("@splunk/react-ui/Box");
    var k =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/themes"
    const C = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/DualListbox/DualListboxStyles.ts
    var O = g()(k()).withConfig({
        displayName: "DualListboxStyles__StyledBox",
        componentId: "sc-1te6bz0-0"
    })([ "display:grid;grid-template-columns:[list0] 180px [actions] min-content [list1] 180px;grid-template-rows:[headers] min-content [lists] 160px;gap:", " ", ";", " ", "" ], C.variables.spacingXSmall, C.variables.spacingSmall, (function(e) {
        var r = e.$inline;
        return r && (0, h.css)([ "display:inline-grid;" ]);
    }), (function(e) {
        var r = e.$fill;
        return r && (0, h.css)([ "height:100%;grid-template-columns:[list0] 1fr [actions] min-content [list1] 1fr;grid-template-rows:[headers] min-content [lists] 1fr;position:relative;width:100%;" ]);
    }));
    var S = g().div.withConfig({
        displayName: "DualListboxStyles__StyledFlexColumn",
        componentId: "sc-1te6bz0-1"
    })([ "&::before{content:'';grid-column:headers;grid-row:lists;}align-items:center;display:flex;flex-direction:column;gap:", ";grid-column:actions;grid-row:lists;justify-content:center;" ], C.variables.spacingSmall);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const x = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/react-ui/Checkbox"
    const w = require("@splunk/react-ui/Checkbox");
    var R =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const E = require("@splunk/react-ui/ScreenReaderContent");
    var j =  e.n(E);
    // CONCATENATED MODULE: ./src/DualListbox/LabelStyles.ts
    var P = g().label.withConfig({
        displayName: "LabelStyles__StyledLabel",
        componentId: "unwplx-0"
    })([ "color:", ";flex:0 0 auto;" ], C.variables.contentColorActive);
    var I = g().div.withConfig({
        displayName: "LabelStyles__StyledLabelContainer",
        componentId: "unwplx-1"
    })([ "display:flex;flex-direction:column;" ]);
    var L = g().div.withConfig({
        displayName: "LabelStyles__StyledSwitchContainer",
        componentId: "unwplx-2"
    })([ "align-items:center;border:", " solid transparent;display:flex;gap:", ";grid-row:headers;padding:0 ", ";", "" ], C.variables.inputBorderWidth, C.variables.spacingSmall, C.variables.spacingSmall, (function(e) {
        var r = e.$columnNumber;
        return (0, h.css)([ "grid-column:", ";" ], "list".concat(r));
    }));
    // CONCATENATED MODULE: ./src/DualListbox/ListboxContext.tsx
    /**
 * A React context used to inform subcomponets of Listbox of parent component values.
 * The context interface is `ListboxContextValue`.
 * Defaults to `'{}'`.
 * @public
 */
    var q =  n().createContext({});
    /* harmony default export */    const N = q;
    // CONCATENATED MODULE: ./src/DualListbox/Label.tsx
    var _ = {
        id: l().string.isRequired,
        index: l().number.isRequired,
        listId: l().string.isRequired,
        name: l().string.isRequired,
        onCheckboxChange: l().func,
        value: l().string.isRequired
    };
    function V(e) {
        var r = e.id, a = e.index, l = e.listId, i = e.name, o = e.onCheckboxChange, u = e.value;
        // @docs-props-type ListboxLabelPropsBase
                var c = (0, t.useContext)(N);
        var s = v()();
        var p = (0, t.useRef)((0, x.createDOMID)("checkbox"));
        var b = c.updateLabelRefreshHookByName, m = c.getLabelState;
        (0, t.useEffect)((function() {
            b === null || b === void 0 ? void 0 : b(i, s);
            return function() {
                return b === null || b === void 0 ? void 0 : b(i);
            };
        }), [ i, b, s ]);
        if (!m) {
            return null;
        }
        var h = m(), g = h.disabled, y = h.selected, k = h.selectedValueCount, C = h.valueCount;
        
        return n().createElement(L, {
            $columnNumber: a
        },  n().createElement(j(), {
            "aria-hidden": true,
            id: p.current
        }, (0, f.sprintf)((0, d._)("Select all %(listName)s Options"), {
            listName: u
        })),  n().createElement(R(), {
            "data-test": "select-all",
            disabled: g,
            labelledBy: p.current,
            onChange: o,
            checked: y,
            value: u
        }),  n().createElement(I, null,  n().createElement(P, {
            id: r,
            htmlFor: l
        }, u),  n().createElement("span", {
            "aria-hidden": true
        }, "".concat(k, "/").concat(C, " selected"))));
    }
    V.propTypes = _;
    /* harmony default export */    const D = V;
    // CONCATENATED MODULE: ./src/DualListbox/ListboxStyles.ts
    var B = g().ul.withConfig({
        displayName: "ListboxStyles__StyledListbox",
        componentId: "sc-11oqtd-0"
    })([ "background-color:", ";border:", " solid ", ";border-radius:", ";grid-row:lists;list-style:none;margin:0;min-height:0;min-width:0;overflow:auto;padding:0;position:relative;&:focus{box-shadow:", ";outline:none;}", "" ], C.variables.transparent, C.variables.inputBorderWidth, C.variables.interactiveColorBorder, C.variables.borderRadius, C.variables.focusShadow, (function(e) {
        var r = e.$columnNumber;
        return (0, h.css)([ "grid-column:", ";" ], "list".concat(r));
    }));
    // CONCATENATED MODULE: ./src/DualListbox/Listbox.tsx
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function A(e, r) {
        if (null == e) return {};
        var t, n, a = H(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function H(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var T = {
        children: l().node,
        index: l().number.isRequired,
        label: l().string.isRequired,
        multiSelect: l().bool,
        name: l().string.isRequired,
        onClick: l().func,
        onKeyDown: l().func,
        onSelectAllClick: l().func
    };
    function K(e) {
        var r = e.children, a = e.elementRef, l = e.index, i = e.label, o = e.name, u = e.onClick, c = e.onKeyDown, s = e.onSelectAllClick, v = A(e, [ "children", "elementRef", "index", "label", "name", "onClick", "onKeyDown", "onSelectAllClick" ]);
        // @docs-props-type ListboxPropsBase
                var f = (0, t.useRef)((0, x.createDOMID)("label"));
        var d = (0, t.useRef)((0, x.createDOMID)("listbox"));
        var p = (0, t.useContext)(m), b = p.controlled, h = p.getSelectedStateForValue, g = p.getActiveValue, y = p.getSelectedValueCount, k = p.getValueCount, C = p.updateLabelRefreshHookByName, O = p.updateOptionRefreshHookByValue;
        var S = (0, t.useCallback)((function(e, r) {
            var t = r.value;
            u === null || u === void 0 ? void 0 : u(e, {
                name: o,
                value: t
            });
        }), [ o, u ]);
        var w = (0, t.useCallback)((function(e) {
            return {
                active: (g === null || g === void 0 ? void 0 : g(o)) === e,
                selected: !!(h === null || h === void 0 ? void 0 : h(e))
            };
        }), [ o, g, h ]);
        var R = (0, t.useCallback)((function() {
            var e = (k === null || k === void 0 ? void 0 : k(o)) || 0;
            var r = (y === null || y === void 0 ? void 0 : y(o)) || 0;
            var t = false;
            if (r > 0) {
                t = r === e ? true : "indeterminate";
            }
            return {
                disabled: e < 1,
                selected: t,
                selectedValueCount: r,
                valueCount: e
            };
        }), [ o, y, k ]);
        var E = function e(r) {
            c === null || c === void 0 ? void 0 : c(r, {
                name: o
            });
        };
        var j = function e(r, t) {
            var n = t.checked;
            s === null || s === void 0 ? void 0 : s(r, {
                name: o,
                selected: n
            });
        };
        var P = function e(r) {
            // this prevents default text selection when using shift + click
            if (r.shiftKey && r.target !== r.currentTarget) {
                r.preventDefault();
            }
        };
        var I = (0, t.useMemo)((function() {
            return {
                onOptionClick: S,
                updateLabelRefreshHookByName: C,
                updateOptionRefreshHookByValue: O,
                getOptionStateForValue: w,
                getLabelState: R,
                controlled: b
            };
        }), [ S, C, O, w, R, b ]);
        
        return n().createElement(N.Provider, {
            value: I
        },  n().createElement(D, {
            id: f.current,
            index: l,
            listId: d.current,
            name: o,
            onCheckboxChange: j,
            value: i
        }),  n().createElement(B, M({
            "aria-activedescendant": "",
            "aria-labelledby": f.current,
            "aria-multiselectable": true,
            "data-test": "listbox",
            id: d.current,
            onKeyDown: E,
            onMouseDown: P,
            ref: a,
            role: "listbox",
            tabIndex: 0,
            $columnNumber: l
        }, v), r));
    }
    K.propTypes = T;
    /* harmony default export */    const F = K;
    // CONCATENATED MODULE: ./src/DualListbox/listboxUtils.ts
    var $ = function e(r, t) {
        var n = r.filter((function(e) {
            return e !== t;
        }));
        if (r.length === n.length) {
            n.push(t);
        }
        return n;
    };
    var z = function e(r, t) {
        var n = new Set(t);
        return r.filter((function(e) {
            return !n.has(e);
        }));
    };
    var U = function e(r, t, n) {
        if (r < 0) {
            return undefined;
        }
        var a = t[r];
        var l = new Set(n);
        if (!l.has(a)) {
            return a;
        }
        for (var i = r + 1; i < t.length; i += 1) {
            var o = t[i];
            if (!l.has(o)) {
                return o;
            }
        }
        for (var u = r - 1; u > 0; u -= 1) {
            var c = t[u];
            if (!l.has(c)) {
                return c;
            }
        }
        return undefined;
    };
    // CONCATENATED MODULE: external "@splunk/themes/mixins"
    const W = require("@splunk/themes/mixins");
    // CONCATENATED MODULE: ./src/DualListbox/OptionStyles.ts
    var X = (0, h.css)([ "background:", ";font-weight:", ";" ], C.mixins.overlayColors(C.variables.interactiveColorOverlaySelected, C.variables.interactiveColorOverlayActive), C.variables.fontWeightBold);
    var G = g().li.withConfig({
        displayName: "OptionStyles__StyledListItem",
        componentId: "q275d9-0"
    })([ "", " align-items:center;color:", ";cursor:pointer;display:flex;min-height:", ";overflow:hidden;padding:0 ", ";word-wrap:break-word;", " &:active{", "}", "" ], (0, 
    W.typography)({
        lineHeight: "spacious"
    }), C.variables.contentColorActive, C.variables.inputHeight, C.variables.spacingSmall, (function(e) {
        var r = e.$active;
        return r && X;
    }), X, (0, C.pickVariant)("$selected", {
        true: (0, h.css)([ "background:", ";&:hover{background:", ";}" ], C.variables.interactiveColorOverlaySelected, C.mixins.overlayColors(C.variables.interactiveColorOverlaySelected, C.variables.interactiveColorOverlayHover)),
        false: (0, h.css)([ "&:hover{background:", ";}" ], C.variables.interactiveColorOverlayHover)
    }));
    var J = g()(R()).withConfig({
        displayName: "OptionStyles__StyledCheckbox",
        componentId: "q275d9-1"
    })([ "padding-inline-end:", ";" ], C.variables.spacingSmall);
    var Q = g().div.withConfig({
        displayName: "OptionStyles__StyledLabel",
        componentId: "q275d9-2"
    })([ "word-break:break-word;word-wrap:break-word;" ]);
    // CONCATENATED MODULE: ./src/DualListbox/Option.tsx
    function Y() {
        return Y = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, Y.apply(null, arguments);
    }
    function Z(e, r) {
        if (null == e) return {};
        var t, n, a = ee(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function ee(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var re = {
        id: l().string.isRequired,
        label: l().string.isRequired,
        listName: l().string,
        selected: l().bool,
        value: l().string.isRequired
    };
    function te(e) {
        var r = e.id, a = e.label, l = e.selected, i = e.value, o = Z(e, [ "id", "label", "selected", "value" ]);
        // @docs-props-type OptionPropsBase
                var u = (0, t.useContext)(N);
        var c = v()();
        var s = u.onOptionClick, f = u.getOptionStateForValue, d = u.updateOptionRefreshHookByValue, p = u.controlled;
        (0, t.useEffect)((function() {
            d === null || d === void 0 ? void 0 : d(i, c);
            return function() {
                return d === null || d === void 0 ? void 0 : d(i);
            };
        }), [ i, d, c ]);
        var b = (0, t.useCallback)((function(e) {
            s === null || s === void 0 ? void 0 : s(e, {
                value: i
            });
        }), [ s, i ]);
        if (!f) {
            return null;
        }
        var m = f(i), h = m.active, g = m.selected;
        var y = !!(p ? l : g);
        
        return n().createElement(G, Y({
            "aria-selected": y,
            "data-test": "option",
            "data-test-value": i,
            "data-test-active": h,
            id: r,
            onClick: b,
            role: "option",
            $active: h,
            $selected: y
        }, o),  n().createElement(J, {
            checked: y,
            inert: true
        }),  n().createElement(Q, null, a));
    }
    te.propTypes = re;
    /* harmony default export */    const ne = te;
    // CONCATENATED MODULE: ./src/DualListbox/ScreenReaderWrapper.tsx
    function ae() {
        return ae = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, ae.apply(null, arguments);
    }
    function le(e) {
        var r = ae({}, e);
        var a = v()();
        var l = (0, t.useContext)(m);
        var i = l.getScreenReaderMessage, o = l.updateScreenReaderRefreshHook;
        (0, t.useEffect)((function() {
            o === null || o === void 0 ? void 0 : o(a);
            return function() {
                return o === null || o === void 0 ? void 0 : o();
            };
        }), [ o, a ]);
        if (!i) {
            return null;
        }
        
        return n().createElement(j(), ae({
            "aria-live": "assertive"
        }, r), i() || "");
    }
    /* harmony default export */    const ie = le;
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const oe = require("@splunk/react-ui/Tooltip");
    var ue =  e.n(oe);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const ce = require("@splunk/react-ui/ButtonSimple");
    var se =  e.n(ce);
    // CONCATENATED MODULE: ./src/DualListbox/ToolbarButtonStyles.ts
    var ve = g()(se()).withConfig({
        displayName: "ToolbarButtonStyles__StyledButtonSimple",
        componentId: "k7zksz-0"
    })([ "height:", ";width:", ";" ], C.variables.inputHeight, C.variables.inputHeight);
    var fe = g().span.withConfig({
        displayName: "ToolbarButtonStyles__StyledIcon",
        componentId: "k7zksz-1"
    })([ "align-items:center;display:flex;flex-direction:row;flex-shrink:0;justify-content:center;max-width:100%;" ]);
    // CONCATENATED MODULE: ./src/DualListbox/ToolbarContext.tsx
    /**
 * A React context used to inform subcomponets of Toolbar of parent component values.
 * The context interface is `ToolbarContextValue`.
 * Defaults to `'{}'`.
 * @public
 */
    var de =  n().createContext({});
    /* harmony default export */    const pe = de;
    // CONCATENATED MODULE: ./src/DualListbox/ToolbarButton.tsx
    function be() {
        return be = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, be.apply(null, arguments);
    }
    function me(e, r) {
        if (null == e) return {};
        var t, n, a = he(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function he(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var ge = {
        icon: l().node.isRequired,
        name: l().string.isRequired,
        tooltipLabel: l().string.isRequired,
        tooltipPlacement: l().oneOf([ "above", "below", "left", "right" ])
    };
    function ye(e) {
        var r = e.ariaLabel, a = e.ariaKeyShortcuts, l = e.icon, i = e.name, o = e.tooltipLabel, u = e.tooltipPlacement, c = me(e, [ "ariaLabel", "ariaKeyShortcuts", "icon", "name", "tooltipLabel", "tooltipPlacement" ]);
        // @docs-props-type ToolbarButtonPropsBase
                var s = (0, t.useContext)(pe);
        var f = v()();
        var d = s.getButtonState, p = s.onButtonClick, b = s.updateButtonRefreshHookByName;
        (0, t.useEffect)((function() {
            b === null || b === void 0 ? void 0 : b(i, f);
            return function() {
                return b === null || b === void 0 ? void 0 : b(i);
            };
        }), [ i, b, f ]);
        var m = (0, t.useCallback)((function(e) {
            p === null || p === void 0 ? void 0 : p(e, {
                name: i
            });
        }), [ p, i ]);
        if (!d) {
            return null;
        }
        var h = d(i), g = h.disabled;
        
        return n().createElement(ue(), {
            defaultPlacement: u,
            content:  n().createElement("span", {
                "aria-hidden": true
            }, o)
        },  n().createElement(ve, be({
            appearance: "subtle",
            "aria-label": r,
            "aria-keyshortcuts": a,
            "aria-disabled": g || undefined,
            "data-test": "toolbar-button",
            disabled: g,
            onClick: m
        }, c),  n().createElement(fe, null, l)));
    }
    ye.propsTypes = ge;
    /* harmony default export */    const ke = ye;
    // CONCATENATED MODULE: ./src/utils/fuzzyMatch.ts
    // A utility for matching keyboard characters to list values
    var Ce = function e(r, t) {
        return r ? r.label.charAt(t).toLowerCase() : "";
    };
    var Oe = function e(r, t) {
        if (!r.length) {
            return r;
        }
        var n = null;
        var a = false;
        var l = r.filter((function(e) {
            var r = Ce(e, t.index);
            if (r === t.value) {
                a = true;
                return true;
            }
            // If we haven't found a match yet, keep track of the next closest match.
            // Secondary matching looks for the closest character of a higher value, and failing that, closest of a lower value.
                        if (!a) {
                var l = Ce(n, t.index);
                if (!l) {
                    n = e;
                } else if (r > t.value) {
                    if (l < t.value) {
                        n = e;
                    } else if (l > r) {
                        n = e;
                    }
                } else if (r > l) {
                    n = e;
                }
            }
            return false;
        }));
        return l.length === 0 && n ? [ n ] : l;
    };
    // CONCATENATED MODULE: ./src/DualListbox/DualListbox.tsx
    function Se(e) {
        "@babel/helpers - typeof";
        return Se = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Se(e);
    }
    function xe() {
        return xe = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, xe.apply(null, arguments);
    }
    function we(e, r) {
        return Ie(e) || Pe(e, r) || Ee(e, r) || Re();
    }
    function Re() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Ee(e, r) {
        if (e) {
            if ("string" == typeof e) return je(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? je(e, r) : void 0;
        }
    }
    function je(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function Pe(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, a, l, i, o = [], u = !0, c = !1;
            try {
                if (l = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (n = l.call(t)).done) && (o.push(n.value), o.length !== r); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (i = t["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return o;
        }
    }
    function Ie(e) {
        if (Array.isArray(e)) return e;
    }
    function Le(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function qe(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? Le(Object(t), !0).forEach((function(r) {
                Ne(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : Le(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function Ne(e, r, t) {
        return (r = _e(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function _e(e) {
        var r = Ve(e, "string");
        return "symbol" == Se(r) ? r : r + "";
    }
    function Ve(e, r) {
        if ("object" != Se(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != Se(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function De(e, r) {
        if (null == e) return {};
        var t, n, a = Be(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function Be(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */
    /** @public */    var Me = {
        children: l().node,
        controlled: l().bool,
        elementRef: l().oneOfType([ l().func, l().object ]),
        fill: l().bool,
        inline: l().bool,
        onChange: l().func,
        onSelect: l().func,
        lists: l().arrayOf(l().shape({
            name: l().string.isRequired,
            label: l().string.isRequired
        })).isRequired
    };
    function Ae(e) {
        var r, a, l, i;
        var u = e.children, s = e.controlled, b = s === void 0 ? false : s, h = e.elementRef, g = e.fill, y = g === void 0 ? false : g, k = e.inline, C = k === void 0 ? false : k, x = e.onChange, w = e.onSelect, R = e.lists, E = De(e, [ "children", "controlled", "elementRef", "fill", "inline", "onChange", "onSelect", "lists" ]);
        // @docs-props-type DualListboxPropsBase
                var j = v()();
        var P = (0, t.useRef)(false);
        (0, t.useEffect)((function() {
            P.current = true;
        }), []);
        // TODO: Implement shared isControlled hook (SUI-5283).
                var I = (0, t.useRef)(b);
        // List One "Available" Refs
                var L = (0, t.useRef)([]);
        var q = (0, t.useRef)([]);
        var N = (0, t.useRef)();
        var _ = (0, t.useRef)(null);
        var V = (0, t.useRef)([]);
        // List Two "Selected" Refs
                var D = (0, t.useRef)([]);
        var B = (0, t.useRef)([]);
        var M = (0, t.useRef)();
        var A = (0, t.useRef)(null);
        var H = (0, t.useRef)([]);
        // Refresh Hook Stores, used to render indivual elements when uncontrolled
                var T = (0, t.useRef)({});
        var K = (0, t.useRef)({});
        var W = (0, t.useRef)({});
        var X = (0, t.useRef)();
        // Activity Values Refs
                var G = (0, t.useRef)();
        var J = (0, t.useRef)({});
        var Q = (0, t.useRef)({});
        var Y = (0, t.useRef)({});
        var Z = (0, t.useRef)(0);
        // Key Match Refs
                var ee = (0, t.useRef)();
        var re = (0, t.useRef)();
        var te = (0, t.useRef)([]);
        if (false) {}
        if (false) {}
        if (false) {}
        // Unpack names and labels instead of relying on `lists` being a consistent reference
                var ne = (r = R[0]) === null || r === void 0 ? void 0 : r.name;
        var ae = (a = R[1]) === null || a === void 0 ? void 0 : a.name;
        var le = (l = R[0]) === null || l === void 0 ? void 0 : l.label;
        var oe = (i = R[1]) === null || i === void 0 ? void 0 : i.label;
        // Copy previous state for uncontrolled render so it's not voided by following value reset
                var ue = new Set(D.current);
        // This ensures we recapture the indexes after batch moves, but will be refactored when sorting is implemented (SUI-5270).
                L.current = [];
        D.current = [];
        V.current = [];
        H.current = [];
        // Clean the data stores before we copy the references into listData
                if (!P.current || I.current) {
            q.current = [];
            B.current = [];
            J.current = {};
            Y.current = {};
            Q.current = {};
        }
        var ce = [ {
            options: [],
            matchOptions: V.current,
            values: L.current,
            selectedValues: q.current,
            activeValue: N.current,
            activeFound: !N.current
        }, {
            options: [],
            matchOptions: H.current,
            values: D.current,
            selectedValues: B.current,
            activeValue: M.current,
            activeFound: !M.current
        } ];
        if (!P.current || I.current) {
            t.Children.toArray(u).filter(t.isValidElement).forEach((function(e) {
                var r = e.props, t = r.value, n = r.label;
                var a = e.props.listName === ae;
                var l = ce[a ? 1 : 0];
                // Ensure the active value is found on controlled renders
                                if (!l.activeFound && l.activeValue === t) {
                    l.activeFound = true;
                }
                l.values.push(t);
                l.options.push(e);
                l.matchOptions.push({
                    value: t,
                    label: n
                });
                var i = !!e.props.selected;
                if (i) {
                    l.selectedValues.push(t);
                }
                J.current[t] = i;
                Y.current[t] = l.values.length - 1;
                Q.current[t] = e.props.id;
            }));
        } else {
            t.Children.toArray(u).filter(t.isValidElement).forEach((function(e) {
                var r = e.props, t = r.value, n = r.label;
                var a = ue.has(t);
                var l = ce[a ? 1 : 0];
                l.values.push(t);
                l.options.push(e);
                l.matchOptions.push({
                    value: t,
                    label: n
                });
                Y.current[t] = l.values.length - 1;
            }));
        }
        var se = (0, t.useCallback)((function(e, r) {
            if (e === ne) {
                N.current = r;
            } else {
                M.current = r;
            }
        }), [ ne ]);
        var ve = (0, t.useCallback)((function(e) {
            return e === ne ? N.current : M.current;
        }), [ ne ]);
        var fe = (0, t.useCallback)((function(e, r) {
            if (e === ne) {
                q.current = r;
            } else {
                B.current = r;
            }
        }), [ ne ]);
        var de = (0, t.useCallback)((function(e) {
            return e === ne ? q.current : B.current;
        }), [ ne ]);
        var be = (0, t.useCallback)((function(e) {
            return de(e).length;
        }), [ de ]);
        var me = (0, t.useCallback)((function() {
            return G.current;
        }), []);
        var he = (0, t.useCallback)((function(e) {
            return J.current[e];
        }), []);
        var ge = (0, t.useCallback)((function(e, r) {
            if (e === ne) {
                L.current = r;
            } else {
                D.current = r;
            }
        }), [ ne ]);
        var ye = (0, t.useCallback)((function(e) {
            return e === ne ? L.current : D.current;
        }), [ ne ]);
        var Ce = (0, t.useCallback)((function(e) {
            return ye(e).length;
        }), [ ye ]);
        var Se = (0, t.useCallback)((function(e) {
            var r = Ce(e);
            return r > 0 && r === be(e);
        }), [ Ce, be ]);
        var Re = (0, t.useCallback)((function(e) {
            return e === ne ? V.current : H.current;
        }), [ ne ]);
        var Ee = (0, t.useCallback)((function(e) {
            var r = Ce(e);
            var t = be(e);
            return {
                disabled: r < 1 || t < 1
            };
        }), [ Ce, be ]);
        var je = (0, t.useCallback)((function() {
            if (G.current && G.current.length > 0) {
                var e;
                G.current = "";
                (e = X.current) === null || e === void 0 ? void 0 : e.call(X);
            }
        }), []);
        var Pe = (0, t.useCallback)((function() {
            ee.current = undefined;
            te.current = [];
            if (re.current) {
                clearTimeout(re.current);
            }
        }), []);
        var Ie = (0, t.useCallback)((function(e) {
            X.current = e;
        }), []);
        var Le = (0, t.useCallback)((function(e, r) {
            if (r) {
                W.current[e] = r;
            } else {
                delete W.current[e];
            }
        }), []);
        var Ne = (0, t.useCallback)((function(e, r) {
            if (r) {
                K.current[e] = r;
            } else {
                delete K.current[e];
            }
        }), []);
        var _e = (0, t.useCallback)((function(e, r) {
            if (r) {
                T.current[e] = r;
            } else {
                delete T.current[e];
            }
        }), []);
        var Ve = (0, t.useCallback)((function(e) {
            if (e) {
                var r, t;
                (r = (t = W.current)[e]) === null || r === void 0 ? void 0 : r.call(t);
            }
        }), []);
        var Be = (0, t.useCallback)((function(e) {
            if (e) {
                var r, t;
                (r = (t = K.current)[e]) === null || r === void 0 ? void 0 : r.call(t);
            }
        }), []);
        var Me = (0, t.useCallback)((function(e) {
            if (e) {
                var r, t;
                (r = (t = T.current)[e]) === null || r === void 0 ? void 0 : r.call(t);
            }
        }), []);
        var Ae = (0, t.useCallback)((function(e) {
            je();
            Be(e);
            Me(ne);
            Me(ae);
        }), [ je, Me, Be, ne, ae ]);
        var He = (0, t.useCallback)((function(e, r) {
            var t = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : false;
            var n = e === ne ? _.current : A.current;
            if (!n) {
                return;
            }
            var a = ve(e);
            se(e, r);
            if (!t) {
                // render new
                Ve(r);
            }
            if (r !== a) {
                if (!t) {
                    // render previous
                    Ve(a);
                }
                var l = r ? Q.current[r] : "";
                // update listbox via dom to avoid render + child render
                                n.setAttribute("aria-activedescendant", l);
                if (!t && l.length > 0) {
                    var i = n.querySelector("#".concat(l));
                    (0, p.scrollIntoViewIfNeeded)(i);
                }
            }
        }), [ ve, se, Ve, ne ]);
        var Te = (0, t.useCallback)((function(e, r) {
            var t = r.value, n = r.name;
            var a = Se(n) ? ye(n) : de(n);
            var l = $(a, t);
            if (!I.current) {
                J.current[t] = !J.current[t];
                fe(n, l);
                Ae(n);
            }
            He(n, t);
            w === null || w === void 0 ? void 0 : w(e, {
                values: l,
                listName: n
            });
        }), [ Se, de, ye, fe, He, Ae, w ]);
        var Ke = (0, t.useCallback)((function(e, r) {
            var t = r.initialIndex, n = r.name, a = r.targetIndex;
            var l = ye(n);
            var i = Math.min(t, a);
            var o = Math.max(t, a);
            var u = [];
            if (I.current) {
                u = l.slice(i, o + 1);
            } else {
                l.forEach((function(e, r) {
                    var t = false;
                    if (r >= i && r <= o) {
                        t = true;
                        u.push(e);
                    } else {
                        t = false;
                    }
                    if (J.current[e] !== t) {
                        J.current[e] = t;
                        Ve(e);
                    }
                }));
                fe(n, u);
                Ae(n);
            }
            He(n, l[a]);
            e.preventDefault();
            e.stopPropagation();
            w === null || w === void 0 ? void 0 : w(e, {
                values: u,
                listName: n
            });
        }), [ ye, w, fe, He, Ae, Ve ]);
        var Fe = (0, t.useCallback)((function(e, r) {
            var t = r.name, n = r.selected;
            var a = ye(t);
            var l = n === true ? [] : a;
            var i = n !== true;
            if (!I.current) {
                a === null || a === void 0 ? void 0 : a.forEach((function(e) {
                    if (J.current[e] !== i) {
                        J.current[e] = i;
                        Ve(e);
                    }
                }));
                fe(t, l);
                Ae(t);
            }
            w === null || w === void 0 ? void 0 : w(e, {
                values: l,
                listName: t
            });
        }), [ ye, w, fe, Ae, Ve ]);
        // handleMoveValues does not update the active value of the source, keyboard and mouse differ, only the target active value is updated
                var $e = (0, t.useCallback)((function(e, r, t) {
            var n = ye(r);
            var a = de(r);
            var l = Se(r);
            var i = ye(t);
            var o = ve(t);
            if (n.length === 0 || !l && a.length === 0) {
                return;
            }
            if (!o && a.length > 0) {
                // target lacks an active value, default to the first
                // skip the render it will be catch in batch
                He(t, a[0], true);
            }
            var u = (0, f.sprintf)((0, d._)("Moved %(count)d items to %(listLabel)s list"), {
                count: a.length,
                listLabel: t === ne ? le : oe
            });
            // These new values do not perserve sort position, may need to be refactored with sort logic (SUI-5270).
                        var c = l ? [] : z(n, a);
            var s = i.concat(a);
            if (!I.current) {
                a.forEach((function(e) {
                    J.current[e] = false;
                }));
                ge(t, s);
                ge(r, c);
                fe(r, []);
                G.current = u;
                // move is a batch update, force a full render, no internal update needed
                                j();
            }
            var v = {
                affectedValues: a,
                sourceListName: r || "",
                sourceValues: c,
                targetListName: t || "",
                targetValues: s,
                screenReaderMessage: u || "",
                values: a.slice()
            };
            e.preventDefault();
            x === null || x === void 0 ? void 0 : x(e, v);
        }), [ j, ve, Se, de, ye, x, fe, ge, He, ne, le, oe ]);
        var ze = (0, t.useCallback)((function(e, r) {
            var t = r.name;
            var n = Se(t);
            var a = ve(t);
            var l = t === ne ? ae : ne;
            if (a && (n || !!J.current[a])) {
                // active option was present in items to be removed, clear the active option
                He(t);
            }
            $e(e, t, l);
        }), [ ve, Se, $e, He, ne, ae ]);
        var Ue = (0, t.useCallback)((function(e, r) {
            var t = r.name, n = r.value;
            if (e.nativeEvent.shiftKey) {
                Ke(e, {
                    targetIndex: Y.current[n],
                    initialIndex: Z.current,
                    name: t
                });
            } else {
                Te(e, {
                    name: t,
                    value: n
                });
            }
        }), [ Ke, Te ]);
        var We = (0, t.useCallback)((function(e, r) {
            var t = r.selected, n = De(r, [ "selected" ]);
            // flip selected here because handleSelectAll expects "selected" to be the current state but Checkbox reports the new state
                        Fe(e, qe({
                selected: !t
            }, n));
        }), [ Fe ]);
        var Xe = (0, t.useCallback)((function(e, r) {
            var t = e.nativeEvent.key;
            var n = Re(r);
            // Checking for a single character to avoid complications from double-byte languages and emojis.
                        if (t.length === 1) {
                var a = [];
                var l = {
                    index: 0,
                    value: t
                };
                if (!ee.current) {
                    if (t === " ") {
                        Pe();
                        return;
                    }
                    a = Oe(n, l);
                } else if (te.current.length > 1) {
                    l.index = ee.current.index + 1;
                    a = Oe(te.current, l);
                }
                if (a.length) {
                    var i = 0;
                    // If the active option is a first character match, cycle to the next matching option.
                                        if (l.index === 0 && a.length > 1) {
                        var o = ve(r);
                        if (o) {
                            var u = a.findIndex((function(e) {
                                return e.value === o;
                            }));
                            if (u >= 0) {
                                i = u === a.length - 1 ? 0 : u + 1;
                            }
                        }
                    }
                    He(r, a[i].value);
                }
                te.current = a;
                ee.current = l;
                if (re.current) {
                    clearTimeout(re.current);
                }
                re.current = setTimeout(Pe, 500);
                e.preventDefault();
                e.stopPropagation();
            }
        }), [ ve, Re, Pe, He ]);
        var Ge = (0, t.useCallback)((function(e, r) {
            var t = r.name;
            var n = e.ctrlKey, a = e.key, l = e.metaKey, i = e.shiftKey;
            var o = ye(t);
            var u = de(t);
            var c = Se(t);
            var s = ve(t);
            var v = s ? Y.current[s] : -1;
            var f = Z.current >= 0 ? Z.current : 0;
            if (G.current && G.current.length > 0) {
                je();
            }
            var d = s;
            switch (a) {
              case "Shift":
                Z.current = v;
                break;

              case "Home":
                if (n && i) {
                    // handle select until start
                    Ke(e, {
                        targetIndex: 0,
                        initialIndex: f,
                        name: t
                    });
                } else {
                    var p = we(o, 1);
                    d = p[0];
                }
                break;

              case "End":
                if (n && i) {
                    // handle select until end
                    Ke(e, {
                        targetIndex: o.length - 1,
                        initialIndex: f,
                        name: t
                    });
                } else {
                    d = o[o.length - 1];
                }
                break;

              case "ArrowUp":
                if (i) {
                    // handle select until previous
                    Ke(e, {
                        targetIndex: Math.max(v - 1, 0),
                        initialIndex: f,
                        name: t
                    });
                } else {
                    d = o[Math.max(v - 1, 0)];
                }
                break;

              case "ArrowDown":
                if (i) {
                    // handle select until next
                    Ke(e, {
                        targetIndex: Math.min(v + 1, o.length - 1),
                        initialIndex: f,
                        name: t
                    });
                } else {
                    d = o[Math.min(v + 1, o.length - 1)];
                }
                break;

              case " ":
                if (s != null) {
                    Te(e, {
                        name: t,
                        value: s
                    });
                }
                break;

              case "Enter":
                if (t === ne) {
                    if (c) {
                        // all values moved, no longer an active value
                        d = undefined;
                    } else if (u.length > 0) {
                        // somes values moved, determine next active value
                        d = U(v, o, u);
                    }
                    $e(e, ne, ae);
                }
                break;

              case "Delete":
                if (t === ae) {
                    if (c) {
                        // all values moved, no longer an active value
                        d = undefined;
                    } else if (u.length > 0) {
                        // somes values moved, determine next active value
                        d = U(v, o, u);
                    }
                    $e(e, ae, ne);
                }
                break;

              default:
                if (a === "a" && (n || l)) {
                    // prevent text selection when using control/meta + A
                    e.preventDefault();
                    // handle control + A, otherwise let 'a' passthrough
                                        Fe(e, {
                        name: t,
                        selected: c
                    });
                } else {
                    // do fuzzy match
                    Xe(e, t);
                }
                break;
            }
            if (d !== s) {
                e.preventDefault();
                He(t, d);
            }
        }), [ je, ve, Se, de, ye, Xe, $e, Ke, Te, Fe, He, ne, ae ]);
        var Je = (0, t.useMemo)((function() {
            return {
                onButtonClick: ze,
                updateButtonRefreshHookByName: _e,
                getButtonState: Ee
            };
        }), [ ze, _e, Ee ]);
        var Qe = (0, t.useMemo)((function() {
            return {
                controlled: I.current,
                getActiveValue: ve,
                getScreenReaderMessage: me,
                getSelectedStateForValue: he,
                getSelectedValueCount: be,
                getValueCount: Ce,
                updateLabelRefreshHookByName: Ne,
                updateOptionRefreshHookByValue: Le,
                updateScreenReaderRefreshHook: Ie
            };
        }), [ ve, me, he, be, Ce, Ne, Le, Ie ]);
        if (I.current && !ce[0].activeFound) {
            // listOne active value wasn't found, likely the result of a controlled state reset
            // update interal values, skip the render it will be caught in batch
            He(ne, undefined, true);
        }
        if (I.current && !ce[1].activeFound) {
            // listTwo active value wasn't found, likely the result of a controlled state reset
            // update interal values, skip the render it will be caught in batch
            He(ae, undefined, true);
        }
        
        return n().createElement(O, xe({
            "data-test": "dual-listbox",
            ref: h,
            $inline: y ? false : C,
            $fill: y
        }, E),  n().createElement(m.Provider, {
            value: Qe
        },  n().createElement(F, {
            index: 0,
            elementRef: _,
            label: le,
            name: ne,
            onClick: Ue,
            onKeyDown: Ge,
            onSelectAllClick: We
        }, ce[0].options),  n().createElement(S, null,  n().createElement(pe.Provider, {
            value: Je
        },  n().createElement(ke, {
            ariaLabel: (0, f.sprintf)((0, d._)("Move selected options to %(listLabel)s list"), {
                listLabel: oe
            }),
            ariaKeyShortcuts: "Enter",
            "data-test": "move-to-secondary",
            icon:  n().createElement(c(), null),
            name: ne,
            tooltipLabel: (0, f.sprintf)((0, d._)("Move to %(listLabel)s"), {
                listLabel: oe
            }),
            tooltipPlacement: "left"
        }),  n().createElement(ke, {
            ariaLabel: (0, f.sprintf)((0, d._)("Move selected options to %(listLabel)s list"), {
                listLabel: le
            }),
            ariaKeyShortcuts: "Delete",
            "data-test": "move-to-primary",
            icon:  n().createElement(o(), null),
            name: ae,
            tooltipLabel: (0, f.sprintf)((0, d._)("Move to %(listLabel)s"), {
                listLabel: le
            }),
            tooltipPlacement: "right"
        }))),  n().createElement(F, {
            index: 1,
            elementRef: A,
            label: oe,
            name: ae,
            onClick: Ue,
            onKeyDown: Ge,
            onSelectAllClick: We
        }, ce[1].options),  n().createElement(ie, null)));
    }
    Ae.propTypes = Me;
    Ae.Option = ne;
    /* harmony default export */    const He = Ae;
    // CONCATENATED MODULE: ./src/DualListbox/index.ts
    module.exports = r;
    /******/})();
