// @splunk/react-ui/File.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Item: () => /* reexport */ Pn,
        default: () => /* reexport */ er
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/debounce"
    const a = require("lodash/debounce");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "lodash/isString"
    const o = require("lodash/isString");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "lodash/toUpper"
    const s = require("lodash/toUpper");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var d =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const p = require("@splunk/react-ui/EventListener");
    var f =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const v = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const m = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const y = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/File/FileContext.tsx
    var b =  (0, r.createContext)({
        disabled: false
    });
    b.displayName = "File";
    /* harmony default export */    const g = b;
    // CONCATENATED MODULE: external "styled-components"
    const h = require("styled-components");
    var k =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowBroadUnderbarUp"
    const S = require("@splunk/react-icons/ArrowBroadUnderbarUp");
    var q =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/themes"
    const w = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/File/FileStyles.ts
    var C = "24px";
    var x = "150px";
    var F = k().input.withConfig({
        displayName: "FileStyles__StyledInput",
        componentId: "sc-1wyh4cf-0"
    })([ "&[type='file']{width:0.1px;height:0.1px;opacity:0;overflow:hidden;position:absolute;z-index:-1;}" ]);
    /* wraps the drop target box and the list of files */    var I = k().div.withConfig({
        displayName: "FileStyles__StyledFileWrapper",
        componentId: "sc-1wyh4cf-1"
    })([ "display:flex;flex-direction:column;gap:", ";" ], w.variables.spacingSmall);
    var E = k()(q()).withConfig({
        displayName: "FileStyles__StyledUploadIcon",
        componentId: "sc-1wyh4cf-2"
    })([ "width:", ";height:", ";fill:", ";" ], C, C, w.variables.contentColorActive);
    var _ = k().div.withConfig({
        displayName: "FileStyles__StyledInstructionText",
        componentId: "sc-1wyh4cf-3"
    })([ "", ";" ], w.mixins.typography("body", {
        color: "active",
        weight: "semiBold"
    }));
    var R = k().div.withConfig({
        displayName: "FileStyles__StyledFileTypeSupportMessage",
        componentId: "sc-1wyh4cf-4"
    })([ "color:", ";text-align:center;" ], w.variables.contentColorDefault);
    var O = k().div.withConfig({
        displayName: "FileStyles__StyledDropTarget",
        componentId: "sc-1wyh4cf-5"
    })([ "", ";flex-direction:column;align-items:center;justify-content:center;min-height:", ";gap:", ";padding:", ";border-radius:", ";cursor:pointer;", " ", " ", " ", " ", " ", " ", "" ], w.mixins.reset("flex"), x, w.variables.spacingXSmall, w.variables.spacingSmall, w.variables.borderRadius, (function(e) {
        var n = e.$disabled;
        return !n && (0, h.css)([ "&:hover{background-color:", ";", "{color:", ";}", "{fill:", ";}}" ], w.variables.actionColorBackgroundSubtleHover, /* sc-sel */ _, w.variables.contentColorAccentStrong, /* sc-sel */ E, w.variables.contentColorAccentStrong);
    }), (function(e) {
        var n = e.$screenMode;
        return n === "normal" && (0, h.css)([ "border:", " dashed ", ";" ], w.variables.inputBorderWidth, w.variables.interactiveColorBorder);
    }), (function(e) {
        var n = e.$dragOver, r = e.$focused, t = e.$error;
        return (n || r && !t) && (0, h.css)([ "", "{color:", ";}", "{fill:", ";}" ], /* sc-sel */ _, w.variables.contentColorAccentStrong, /* sc-sel */ E, w.variables.contentColorAccentStrong);
    }), (function(e) {
        var n = e.$dragOver;
        return n && (0, h.css)([ "background-color:", ";" ], w.variables.actionColorBackgroundSubtleHover);
    }), (function(e) {
        var n = e.$focused;
        return n && (0, h.css)([ "box-shadow:", ";" ], w.variables.focusShadow);
    }), (function(e) {
        var n = e.$disabled;
        return n && (0, h.css)([ "border-color:", ";cursor:not-allowed;", ",", "{color:", ";}", "{fill:", ";}" ], w.variables.interactiveColorBorderDisabled, /* sc-sel */ _, /* sc-sel */ R, w.variables.contentColorDisabled, /* sc-sel */ E, w.variables.contentColorDisabled);
    }), (function(e) {
        var n = e.$error, r = e.$dragOver;
        return n && !r && (0, h.css)([ "border-color:", ";" ], w.variables.interactiveColorAccentError);
    }));
    var A = k().div.withConfig({
        displayName: "FileStyles__StyledHelp",
        componentId: "sc-1wyh4cf-6"
    })([ "", ";" ], w.mixins.typography("smallBody"));
    var T = k().div.withConfig({
        displayName: "FileStyles__StyledWindowDrop",
        componentId: "sc-1wyh4cf-7"
    })([ "position:fixed;top:0;left:0;right:0;bottom:0;z-index:", " + 10;" ], w.variables.zindexModal);
    var P = k().div.withConfig({
        displayName: "FileStyles__StyledFileList",
        componentId: "sc-1wyh4cf-8"
    })([ "display:flex;flex-direction:column;gap:", ";" ], w.variables.spacingXSmall);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowsCircularDouble"
    const j = require("@splunk/react-icons/ArrowsCircularDouble");
    var D =  e.n(j);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const N = require("@splunk/react-icons/Cross");
    var M =  e.n(N);
    // CONCATENATED MODULE: external "@splunk/react-icons/TrashCanCross"
    const $ = require("@splunk/react-icons/TrashCanCross");
    var B =  e.n($);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileAiff"
    const L = require("@splunk/react-icons/FileAiff");
    var X =  e.n(L);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileCss"
    const U = require("@splunk/react-icons/FileCss");
    var z =  e.n(U);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileCsv"
    const H = require("@splunk/react-icons/FileCsv");
    var W =  e.n(H);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileDat"
    const J = require("@splunk/react-icons/FileDat");
    var G =  e.n(J);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileDb"
    const V = require("@splunk/react-icons/FileDb");
    var Z =  e.n(V);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileDbf"
    const K = require("@splunk/react-icons/FileDbf");
    var Q =  e.n(K);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileDoc"
    const Y = require("@splunk/react-icons/FileDoc");
    var ee =  e.n(Y);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileEps"
    const ne = require("@splunk/react-icons/FileEps");
    var re =  e.n(ne);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileFig"
    const te = require("@splunk/react-icons/FileFig");
    var ae =  e.n(te);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileGif"
    const ie = require("@splunk/react-icons/FileGif");
    var oe =  e.n(ie);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileHtm"
    const le = require("@splunk/react-icons/FileHtm");
    var se =  e.n(le);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileHtml"
    const ce = require("@splunk/react-icons/FileHtml");
    var ue =  e.n(ce);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileIco"
    const de = require("@splunk/react-icons/FileIco");
    var pe =  e.n(de);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileJpg"
    const fe = require("@splunk/react-icons/FileJpg");
    var ve =  e.n(fe);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileJs"
    const me = require("@splunk/react-icons/FileJs");
    var ye =  e.n(me);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileJson"
    const be = require("@splunk/react-icons/FileJson");
    var ge =  e.n(be);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileJsx"
    const he = require("@splunk/react-icons/FileJsx");
    var ke =  e.n(he);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileLog"
    const Se = require("@splunk/react-icons/FileLog");
    var qe =  e.n(Se);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileMid"
    const we = require("@splunk/react-icons/FileMid");
    var Ce =  e.n(we);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileMp3"
    const xe = require("@splunk/react-icons/FileMp3");
    var Fe =  e.n(xe);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileMp4"
    const Ie = require("@splunk/react-icons/FileMp4");
    var Ee =  e.n(Ie);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileMpg"
    const _e = require("@splunk/react-icons/FileMpg");
    var Re =  e.n(_e);
    // CONCATENATED MODULE: external "@splunk/react-icons/FilePdf"
    const Oe = require("@splunk/react-icons/FilePdf");
    var Ae =  e.n(Oe);
    // CONCATENATED MODULE: external "@splunk/react-icons/FilePhp"
    const Te = require("@splunk/react-icons/FilePhp");
    var Pe =  e.n(Te);
    // CONCATENATED MODULE: external "@splunk/react-icons/FilePng"
    const je = require("@splunk/react-icons/FilePng");
    var De =  e.n(je);
    // CONCATENATED MODULE: external "@splunk/react-icons/FilePpt"
    const Ne = require("@splunk/react-icons/FilePpt");
    var Me =  e.n(Ne);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileRar"
    const $e = require("@splunk/react-icons/FileRar");
    var Be =  e.n($e);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileRtf"
    const Le = require("@splunk/react-icons/FileRtf");
    var Xe =  e.n(Le);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileSql"
    const Ue = require("@splunk/react-icons/FileSql");
    var ze =  e.n(Ue);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileSvg"
    const He = require("@splunk/react-icons/FileSvg");
    var We =  e.n(He);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileText"
    const Je = require("@splunk/react-icons/FileText");
    var Ge =  e.n(Je);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileTGZ"
    const Ve = require("@splunk/react-icons/FileTGZ");
    var Ze =  e.n(Ve);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileTiff"
    const Ke = require("@splunk/react-icons/FileTiff");
    var Qe =  e.n(Ke);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileWav"
    const Ye = require("@splunk/react-icons/FileWav");
    var en =  e.n(Ye);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileXls"
    const nn = require("@splunk/react-icons/FileXls");
    var rn =  e.n(nn);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileXml"
    const tn = require("@splunk/react-icons/FileXml");
    var an =  e.n(tn);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileZip"
    const on = require("@splunk/react-icons/FileZip");
    var ln =  e.n(on);
    // CONCATENATED MODULE: external "@splunk/react-icons/Paperclip"
    const sn = require("@splunk/react-icons/Paperclip");
    var cn =  e.n(sn);
    // CONCATENATED MODULE: external "@splunk/react-icons/FileRipped"
    const un = require("@splunk/react-icons/FileRipped");
    var dn =  e.n(un);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const pn = require("@splunk/react-ui/Button");
    var fn =  e.n(pn);
    // CONCATENATED MODULE: external "@splunk/react-ui/Progress"
    const vn = require("@splunk/react-ui/Progress");
    var mn =  e.n(vn);
    // CONCATENATED MODULE: ./src/File/ItemStyles.ts
    var yn = k().div.withConfig({
        displayName: "ItemStyles__StyledItemWrapper",
        componentId: "sc-1fsk3kr-0"
    })([ "", ";flex-direction:column;gap:", ";" ], w.mixins.reset("flex"), w.variables.spacingXSmall);
    var bn = k().div.withConfig({
        displayName: "ItemStyles__StyledItem",
        componentId: "sc-1fsk3kr-1"
    })([ "display:flex;align-items:center;max-width:400px;padding:", " ", ";gap:", ";color:", ";", " ", ";& *:is(button){flex:none;}" ], w.variables.spacingXSmall, w.variables.spacingSmall, w.variables.spacingXSmall, w.variables.contentColorActive, (function(e) {
        var n = e.$inProgress;
        return n && (0, h.css)([ "color:", ";" ], w.variables.contentColorDefault);
    }), (function(e) {
        var n = e.$disabled;
        return n && (0, h.css)([ "cursor:not-allowed;color:", ";" ], w.variables.contentColorDisabled);
    }));
    var gn = k()(dn()).withConfig({
        displayName: "ItemStyles__StyledErrorIcon",
        componentId: "sc-1fsk3kr-2"
    })([ "fill:", ";" ], w.variables.contentColorNegative);
    var hn = "28px";
    var kn = k().div.withConfig({
        displayName: "ItemStyles__StyledIcon",
        componentId: "sc-1fsk3kr-3"
    })([ "display:grid;justify-content:center;align-items:center;&:has(img){height:", ";width:", ";& img{max-width:", ";max-height:", ";}}" ], hn, hn, hn, hn);
    var Sn = k().div.withConfig({
        displayName: "ItemStyles__StyledLabel",
        componentId: "sc-1fsk3kr-4"
    })([ "overflow:hidden;white-space:nowrap;text-overflow:ellipsis;flex:1 0 0;" ]);
    var qn = k()(mn()).withConfig({
        displayName: "ItemStyles__StyledProgress",
        componentId: "sc-1fsk3kr-5"
    })([ "width:80px;height:6px;" ]);
    var wn = k()(fn()).withConfig({
        displayName: "ItemStyles__StyledButton",
        componentId: "sc-1fsk3kr-6"
    })([ "&:not([disabled],[aria-disabled='true']){color:", ";}" ], w.variables.contentColorActive);
    var Cn = k().div.withConfig({
        displayName: "ItemStyles__StyledErrorText",
        componentId: "sc-1fsk3kr-7"
    })([ "", ";color:", ";padding:", ";" ], w.mixins.typography("smallBody"), w.variables.contentColorNegative, w.variables.spacingXSmall);
    // CONCATENATED MODULE: ./src/File/ItemIcon.tsx
    var xn = {
        aif: X(),
        aiff: X(),
        css: z(),
        csv: W(),
        dat: G(),
        db: Z(),
        dbf: Q(),
        doc: ee(),
        eps: re(),
        fig: ae(),
        gif: oe(),
        htm: se(),
        html: ue(),
        ico: pe(),
        jpg: ve(),
        js: ye(),
        json: ge(),
        jsx: ke(),
        log: qe(),
        mid: Ce(),
        mp3: Fe(),
        mp4: Ee(),
        mpg: Re(),
        pdf: Ae(),
        php: Pe(),
        png: De(),
        ppt: Me(),
        rar: Be(),
        rtf: Xe(),
        sql: ze(),
        svg: We(),
        txt: Ge(),
        tgz: Ze(),
        tiff: Qe(),
        wav: en(),
        xls: rn(),
        xml: an(),
        zip: ln()
    };
    var Fn = function e(n) {
        var r;
        var a = n.name, i = a === void 0 ? "" : a, o = n.icon;
        /* if custom icon passed in use it */        if (o) {
            
            return t().createElement(kn, null, o);
        }
        var l = ((r = i.split(".").pop()) === null || r === void 0 ? void 0 : r.toLowerCase()) || "";
        var s = xn[l] || cn();
        
        return t().createElement(s, {
            "aria-hidden": "true"
        });
    };
    /* harmony default export */    const In = Fn;
    // CONCATENATED MODULE: ./src/File/Item.tsx
    function En() {
        return En = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, En.apply(null, arguments);
    }
    function _n(e, n) {
        if (null == e) return {};
        var r, t, a = Rn(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function Rn(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var On = {
        disabled: d().bool,
        elementRef: d().oneOfType([ d().func, d().object ]),
        error: d().bool,
        /** @private */
        icon: d().node,
        /** @private */
        index: d().number,
        itemId: d().any,
        name: d().string.isRequired,
        /** If the uploadPercentage is 0, the item is assumed to be queued. If the upload is complete or
   * not applicable, uploadPercentage must be undefined. */
        uploadPercentage: d().number
    };
    var An =  t().createElement(gn, {
        variant: "filled"
    });
    function Tn(e) {
        var n = e.disabled, a = e.elementRef, i = e.error, o = i === void 0 ? false : i, l = e.icon, s = e.itemId, c = e.name, u = e.index, d = u === void 0 ? 0 : u, p = e.uploadPercentage, f = _n(e, [ "disabled", "elementRef", "error", "icon", "itemId", "name", "index", "uploadPercentage" ]);
        // @docs-props-type ItemPropsBase
                var y = (0, r.useContext)(b), g = y.onRequestRemove;
        var h = (0, r.useCallback)((function(e) {
            e.preventDefault();
            g === null || g === void 0 ? void 0 : g({
                itemId: s,
                name: c,
                index: d
            });
        }), [ g, s, c, d ]);
        var k = (0, r.useContext)(b), S = k.onRequestRetry;
        var q = (0, r.useCallback)((function(e) {
            e.preventDefault();
            S === null || S === void 0 ? void 0 : S({
                itemId: s,
                name: c,
                index: d
            });
        }), [ S, s, c, d ]);
        var w = (0, r.useContext)(b), C = w.disabled;
        var x = (0, v.sprintf)((0, m._)('Remove "%(filename)s"'), {
            filename: c
        });
        var F = (0, v.sprintf)((0, m._)('Retry "%(filename)s"'), {
            filename: c
        });
        var I = (0, v.sprintf)((0, m._)('Uploading "%(filename)s"'), {
            filename: c
        });
        var E = (0, m._)("Remove");
        var _ = n !== undefined ? n : C;
        var R = p !== undefined && p > 0;
        var O = o ? An :  t().createElement(In, {
            name: c,
            icon: l
        });
        
        return t().createElement(yn, null,  t().createElement(bn, En({
            "data-test": "item",
            "data-test-disabled": _ ? "disabled" : undefined,
            "data-test-error": o,
            ref: a
        }, f, {
            $disabled: _,
            $inProgress: R
        }), O,  t().createElement(Sn, {
            "data-test": "label"
        }, c), R &&  t().createElement(t().Fragment, null,  t().createElement(qn, {
            percentage: p,
            "aria-label": I
        }),  t().createElement("span", null, p, "%")), S && o &&  t().createElement(wn, {
            "data-test": "retry",
            onClick: q,
            "aria-label": F,
            appearance: "subtle"
        },  t().createElement(D(), {
            "aria-label": (0, m._)("Retry")
        })), !_ && (p === undefined && !o ?  t().createElement(wn, {
            "data-test": "remove",
            onClick: h,
            "aria-label": x,
            appearance: "subtle"
        },  t().createElement(B(), {
            "aria-label": E
        })) :  t().createElement(wn, {
            "data-test": "remove",
            onClick: h,
            "aria-label": x,
            appearance: "subtle"
        },  t().createElement(M(), {
            "aria-label": E
        })))), o &&  t().createElement(Cn, null, (0, m._)("Something went wrong.")));
    }
    Tn.propTypes = On;
    /* harmony default export */    const Pn = Tn;
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var jn = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function Dn() {
        var e = typeof document !== "undefined" ? document : jn;
        return e;
    }
    var Nn = Dn();
    /* harmony default export */    const Mn = /* unused pure expression or super */ null && Nn;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var $n = {
        document: jn,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                n();
                return null;
            }
            return setTimeout(n, 0);
        },
        cancelAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(n);
        }
    };
    function Bn() {
        var e = typeof window !== "undefined" ? window : $n;
        return e;
    }
    var Ln = Bn();
    /* harmony default export */    const Xn = Ln;
    // CONCATENATED MODULE: ./src/File/File.tsx
    function Un() {
        return Un = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, Un.apply(null, arguments);
    }
    function zn(e, n) {
        return Vn(e) || Gn(e, n) || Wn(e, n) || Hn();
    }
    function Hn() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Wn(e, n) {
        if (e) {
            if ("string" == typeof e) return Jn(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? Jn(e, n) : void 0;
        }
    }
    function Jn(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function Gn(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, a, i, o, l = [], s = !0, c = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    s = !1;
                } else for (;!(s = (t = i.call(r)).done) && (l.push(t.value), l.length !== n); s = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!s && null != r["return"] && (o = r["return"](), Object(o) !== o)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return l;
        }
    }
    function Vn(e) {
        if (Array.isArray(e)) return e;
    }
    function Zn(e, n) {
        if (null == e) return {};
        var r, t, a = Kn(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function Kn(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */    var Qn = {
        accept: d().string,
        allowMultiple: d().bool,
        children: d().node,
        disabled: d().bool,
        dropAnywhere: d().bool,
        elementRef: d().oneOfType([ d().func, d().object ]),
        fullscreen: d().bool,
        error: d().bool,
        help: d().node,
        inputId: d().string,
        name: d().string,
        onRequestAdd: d().func,
        onRequestRemove: d().func,
        onRequestRetry: d().func,
        /** @private. */
        required: d().bool,
        supportsMessage: d().node
    };
    /**
 * File provides the ability to accept files and present uploaded files. It does not provide
 * file readers, only a reference to the file. This can be used to post binary content, or
 * upload using an array buffer.
 */    function Yn(e) {
        var n = e.accept, a = e.allowMultiple, o = e.children, s = e.disabled, u = e.dropAnywhere, d = e.elementRef, p = e.error, b = e.fullscreen, h = e.help, k = e.inputId, S = e.name, q = e.onRequestAdd, w = e.onRequestRemove, C = e.onRequestRetry, x = e.required, j = e.supportsMessage, D = Zn(e, [ "accept", "allowMultiple", "children", "disabled", "dropAnywhere", "elementRef", "error", "fullscreen", "help", "inputId", "name", "onRequestAdd", "onRequestRemove", "onRequestRetry", "required", "supportsMessage" ]);
        // @docs-props-type FilePropsBase
                var N = (0, r.useState)(false), M = zn(N, 2), $ = M[0], B = M[1];
        var L = (0, r.useState)(false), X = zn(L, 2), U = X[0], z = X[1];
        /* Each time a file is uploaded this is incremented and used to generate the
   * file input's key. In this way we get a new input without a value. */        var H = (0, 
        r.useState)(0), W = zn(H, 2), J = W[0], G = W[1];
        var V = (0, r.useMemo)((function() {
            return k || (0, y.createDOMID)("file");
        }), [ k ]);
        var Z = (0, r.useCallback)((function(e) {
            if (e !== undefined && e.length > 0) {
                var n = Array.from(e);
                q === null || q === void 0 ? void 0 : q(a ? n : [ n[0] ], {
                    name: S
                });
            } else {
                q === null || q === void 0 ? void 0 : q([], {
                    name: S
                });
            }
            G((function(e) {
                return e + 1;
            }));
        }), [ a, S, q ]);
        var K = (0, r.useCallback)((function(e) {
            var n;
            Z((n = e.currentTarget.files) !== null && n !== void 0 ? n : undefined);
        }), [ Z ]);
        var Q = (0, r.useCallback)((function() {
            z(true);
        }), []);
        var Y = (0, r.useCallback)((function() {
            z(false);
        }), []);
        var ee = (0, r.useCallback)((function(e) {
            if (!$) {
                B(true);
            }
            e.preventDefault();
        }), [ $ ]);
        var ne = (0, r.useMemo)((function() {
            return i()((function() {
                return B(false);
            }), 300);
        }), [ B ]);
        var re = (0, r.useCallback)((function(e) {
            var n;
            e.preventDefault();
            ne();
            Z((n = e.dataTransfer) === null || n === void 0 ? void 0 : n.files);
        }), [ Z, ne ]);
        var te = (0, r.useCallback)((function(e) {
            var n = e.name, r = e.index, t = e.itemId;
            C === null || C === void 0 ? void 0 : C({
                event: e,
                filename: n,
                index: r,
                itemId: t,
                name: S
            });
        }), [ S, C ]);
        var ae = (0, r.useCallback)((function(e) {
            var n = e.name, r = e.index, t = e.itemId;
            w === null || w === void 0 ? void 0 : w({
                event: e,
                filename: n,
                index: r,
                itemId: t,
                name: S
            });
        }), [ S, w ]);
        var ie = r.Children.toArray(o).filter(r.isValidElement).map((function(e, n) {
            
            return (0, r.cloneElement)(e, {
                index: n,
                key: e.key || e.props.itemId || "item-".concat(n)
            });
        }));
        var oe = b || u;
        var le = oe || s;
        var se = oe && !s;
        var ce = s ? false : $ || false;
        var ue = b ? "full" : "normal";
        var de = l()(n) && c()(n.split(",").join(", "));
        var pe = oe ? (0, m._)("Drop your file anywhere or click this area to upload") : (0, 
        m._)("Click or drag file to this area to upload");
        var fe = j;
        if (!fe) {
            fe = n ? (0, v.sprintf)((0, m._)("Supports %(fileTypes)s"), {
                fileTypes: de
            }) : (0, m._)("Supports all file types");
        }
        var ve = (0, r.useMemo)((function() {
            return {
                disabled: !!s,
                onRequestRemove: w ? ae : undefined,
                onRequestRetry: C ? te : undefined
            };
        }), [ s, ae, te, w, C ]);
        
        return t().createElement(I, Un({
            ref: d,
            "data-test": "file",
            "data-test-disabled": s ? "disabled" : undefined,
            "data-test-error": !!p
        }, D),  t().createElement("label", {
            htmlFor: V,
            "data-test": "file-label"
        },  t().createElement(O, {
            "data-test": "file-drop-target",
            onDragOver: le ? undefined : ee,
            onDragLeave: le ? undefined : ne,
            onDrop: le ? undefined : re,
            $disabled: !!s,
            $dragOver: ce,
            $error: !!p,
            $screenMode: ue,
            $focused: U
        },  t().createElement(E, {
            $screenMode: ue
        }),  t().createElement(F, {
            accept: n,
            "data-test": "file-input",
            disabled: s,
            id: V,
            key: "file-input-".concat(J),
            multiple: a || undefined,
            onBlur: Y,
            onChange: K,
            onFocus: Q,
            required: x,
            type: "file"
        }),  t().createElement(_, null, pe),  t().createElement(R, {
            "data-test": "file-supports"
        }, fe), se && $ &&  t().createElement(T, {
            "data-test": "file-window-drop",
            onDragLeave: ne
        }), se &&  t().createElement(t().Fragment, null,  t().createElement(f(), {
            target: Xn,
            eventType: "dragover",
            listener: ee
        }),  t().createElement(f(), {
            target: Xn,
            eventType: "drop",
            listener: re
        })))), !s && h &&  t().createElement(A, {
            "data-test": "help"
        }, h),  t().createElement(g.Provider, {
            value: ve
        },  t().createElement(P, null, ie)));
    }
    Yn.propTypes = Qn;
    Yn.componentType = "File";
    Yn.Item = Pn;
    /* harmony default export */    const er = Yn;
    // CONCATENATED MODULE: ./src/File/index.ts
    module.exports = n;
    /******/})();
