// @splunk/react-ui/FormRows.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Row: () => /* reexport */ ne,
        default: () => /* reexport */ ye
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-icons/Plus"
    const i = require("@splunk/react-icons/Plus");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const s = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/FormRows/FormRowsContext.tsx
    var u =  (0, r.createContext)({
        disabledRemoveButton: false,
        sortable: true
    });
    u.displayName = "FormRows";
    /* harmony default export */    const d = u;
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var f =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const v = require("@splunk/react-ui/Button");
    var m =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const p = require("@splunk/react-ui/ButtonSimple");
    var b =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/themes"
    const g = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/FormRows/FormRowsStyles.ts
    var y = (0, c.css)([ "width:", ";height:", ";" ], g.variables.inputHeight, g.variables.inputHeight);
    var h = f()(b()).withConfig({
        displayName: "FormRowsStyles__StyledDragButton",
        componentId: "sc-1j4j7h3-0"
    })([ "", ";grid-column:drag-handle;cursor:move;user-select:none;-webkit-user-drag:element;" ], y);
    var w = f()(b()).withConfig({
        displayName: "FormRowsStyles__StyledRemoveButton",
        componentId: "sc-1j4j7h3-1"
    })([ "", ";grid-column:remove-button;" ], y);
    var R = f().div.withConfig({
        displayName: "FormRowsStyles__StyledDiv",
        componentId: "sc-1j4j7h3-2"
    })([ "", ";justify-content:start;&:has(", "):has(", "){grid-template-columns:[drag-handle] min-content [content] minmax(auto,1fr) [remove-button] min-content;}&:has(", "){grid-template-columns:[drag-handle] min-content [content] minmax(auto,1fr);}&:has(", "){grid-template-columns:[content] minmax(auto,1fr) [remove-button] min-content;}grid-template-columns:[content] minmax(auto,1fr);column-gap:", ";row-gap:", ";" ], g.mixins.reset("grid"), h, w, h, w, g.variables.spacingSmall, g.variables.spacingXSmall);
    var O = f().div.withConfig({
        displayName: "FormRowsStyles__StyledHeader",
        componentId: "sc-1j4j7h3-3"
    })([ "", ";grid-column:content;" ], g.mixins.typography("largeBody", {
        color: "active"
    }));
    var x = (0, c.css)([ "grid-column:content;width:min-content;margin-block-start:", ";" ], g.variables.spacingXSmall);
    var S = f()(m()).withConfig({
        displayName: "FormRowsStyles__StyledAddRowButton",
        componentId: "sc-1j4j7h3-4"
    })([ "", ";" ], x);
    var j = f().div.withConfig({
        displayName: "FormRowsStyles__StyledMenuWrapper",
        componentId: "sc-1j4j7h3-5"
    })([ "", ";" ], x);
    var C = f().div.withConfig({
        displayName: "FormRowsStyles__StyledInputWrapper",
        componentId: "sc-1j4j7h3-6"
    })([ "display:flex;gap:", ";" ], g.variables.spacingXSmall);
    var E = function e(n) {
        return n ? "drag-handle" : "content";
    };
    /* Row includes drag handle, input, and remove button */    var k = f().div.withConfig({
        displayName: "FormRowsStyles__StyledRow",
        componentId: "sc-1j4j7h3-7"
    })([ "", ";grid-template-columns:subgrid;grid-column:", " / -1;padding:", " 0;border-radius:", ";&.sorting{z-index:calc(", " + 1);}&:focus{outline:0;}", "" ], g.mixins.reset("grid"), (function(e) {
        var n = e.$sortable;
        return E(n);
    }), g.variables.spacingXSmall, g.variables.borderRadius, g.variables.zindexPopover, (function(e) {
        var n = e.$sortable;
        return n && (0, c.css)([ "&:has(", ":active,", ":focus){background:", ";}&:not(:has(", ")){padding-inline-end:", ";}" ], h, h, g.variables.interactiveColorOverlayDrag, w, g.variables.spacingXSmall);
    }));
    // CONCATENATED MODULE: external "@splunk/react-icons/TrashCanCross"
    const q = require("@splunk/react-icons/TrashCanCross");
    var P =  e.n(q);
    // CONCATENATED MODULE: ./src/FormRows/RowInternal.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function _(e, n) {
        if (null == e) return {};
        var r, t, o = D(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function D(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var M = {
        children: a().node,
        index: a().number,
        onRequestRemove: a().func,
        renderDragHandle: a().func,
        style: a().object,
        value: a().node
    };
    var A = (0, s._)("Remove row");
    var T =  t().forwardRef((function(e, n) {
        var o = e.children, a = e.index, i = e.onRequestRemove, l = e.renderDragHandle, s = _(e, [ "children", "index", "onRequestRemove", "renderDragHandle" ]);
        var u = (0, r.useContext)(d), c = u.disabledRemoveButton, f = u.hasOnRequestAdd, v = u.sortable;
        var m = (0, r.useCallback)((function(e) {
            // FormRows remove functionality doesn't work without an index, but index is not a required prop. FormRows will throw an error
            // when rows are sortable and/or removable and index is not provided.
            i === null || i === void 0 ? void 0 : i(e, {
                index: a !== null && a !== void 0 ? a : 0
            });
        }), [ i, a ]);
        var p = i !== undefined;
        (0, r.useEffect)((function() {
            if (false) {}
            if (false) {}
            if (false) {}
        }), [ f, p, v, a ]);
        
        return t().createElement(k, I({
            $sortable: v,
            "data-test": "row",
            ref: n
        }, s), l && l(),  t().createElement(C, null, o), p &&  t().createElement(w, {
            appearance: "subtle",
            "aria-label": A,
            "data-test": "remove",
            disabled: c,
            onClick: m
        },  t().createElement(P(), null)));
    }));
    T.propTypes = M;
    /* harmony default export */    const N = T;
    // CONCATENATED MODULE: external "@dnd-kit/sortable"
    const B = require("@dnd-kit/sortable");
    // CONCATENATED MODULE: external "@dnd-kit/utilities"
    const F = require("@dnd-kit/utilities");
    // CONCATENATED MODULE: external "@splunk/react-icons/ParallelDots"
    const H = require("@splunk/react-icons/ParallelDots");
    var X =  e.n(H);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function L(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/FormRows/SortableRow.tsx
    function K() {
        return K = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, K.apply(null, arguments);
    }
    function $(e, n) {
        if (null == e) return {};
        var r, t, o = z(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function z(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var U = {
        children: a().node,
        index: a().number,
        onRequestRemove: a().func,
        value: a().node
    };
    var W = (0, s._)("Drag handle");
    var V =  t().forwardRef((function(e, n) {
        var o = e.children, a = e.index, i = $(e, [ "children", "index" ]);
        // @docs-props-type RowPropsBase
                var l = (0, B.useSortable)({
            // @dnd-kit requires "id" to be greater than 0 https://github.com/clauderic/dnd-kit/issues/1150
            // FormRows drag functionality doesn't work without an index, but index is not a required prop. FormRows will throw an error
            // when rows are sortable and/or removable and index is not provided.
            id: (a !== null && a !== void 0 ? a : 0) + 1
        }), s = l.attributes, u = l.listeners, c = l.setActivatorNodeRef, f = l.setNodeRef, v = l.transform, m = l.transition;
        var p = (0, r.useMemo)((function() {
            return {
                // CSS.Translate outputs only translate3d(x, y, 0), ignoring scale or rotation, so dragged items maintain their original size and shape. Using CSS.Transform here can incorrectly apply scale properties that will cause the element to scale or skew on drag.
                transform: F.CSS.Translate.toString(v),
                transition: m
            };
        }), [ v, m ]);
        var b = (0, r.useCallback)((function(e) {
            f(e);
            L(n, e);
        }), [ f, n ]);
        var g = (0, r.useContext)(d), y = g.onKeyDown;
        var w = (0, r.useCallback)((function(e) {
            // FormRows drag functionality doesn't work without an index, but index is not a required prop. FormRows will throw an error
            // when rows are sortable and/or removable and index is not provided.
            y === null || y === void 0 ? void 0 : y(e, {
                index: a !== null && a !== void 0 ? a : 0
            });
        }), [ y, a ]);
        var R = (0, r.useCallback)((function() {
            
            return t().createElement(h, K({
                appearance: "subtle",
                "aria-label": W,
                "data-test": "drag-handle",
                onKeyDown: w,
                ref: c
            }, u),  t().createElement(X(), null));
        }), [ w, u, c ]);
        
        return t().createElement(N, K({
            index: a,
            ref: b,
            renderDragHandle: R,
            style: p
        }, i, s), o);
    }));
    V.propTypes = U;
    /* harmony default export */    const G = V;
    // CONCATENATED MODULE: ./src/FormRows/Row.tsx
    function J() {
        return J = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, J.apply(null, arguments);
    }
    function Q(e, n) {
        if (null == e) return {};
        var r, t, o = Y(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function Y(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var Z = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        index: a().number,
        onRequestRemove: a().func,
        value: a().node
    };
    function ee(e) {
        var n = e.children, o = e.elementRef, a = Q(e, [ "children", "elementRef" ]);
        // @docs-props-type RowPropsBase
                var i = (0, r.useContext)(d), l = i.sortable;
        var s = (0, r.useMemo)((function() {
            return l ? G : N;
        }), [ l ]);
        
        return t().createElement(s, J({
            ref: o
        }, a), n);
    }
    ee.propTypes = Z;
    /* harmony default export */    const ne = ee;
    // CONCATENATED MODULE: external "@dnd-kit/core"
    const re = require("@dnd-kit/core");
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const te = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: ./src/FormRows/SortableList.tsx
    var oe = function e(n, r) {
        return n.indexOf(r) + 1;
    };
    var ae = {
        draggable: (0, s._)("To pick up a sortable row, press space or enter.\n        Use the up and down arrow keys to update the position of the row.")
    };
    var ie = function e(n) {
        var o = n.children, a = n.onSortEnd;
        var i = (0, re.useSensors)((0, re.useSensor)(re.PointerSensor), (0, re.useSensor)(re.MouseSensor));
        var l = (0, r.useRef)(true);
        var s = r.Children.toArray(o).filter(r.isValidElement);
        var u = s.map((function(e) {
            return (e.props.index || 0) + 1;
        }));
        var d = (0, r.useCallback)((function(e) {
            var n = e.active, r = e.over;
            if (r && n.id !== r.id) {
                var t = u.indexOf(n.id);
                var o = u.indexOf(r.id);
                a({
                    oldIndex: t,
                    newIndex: o
                });
            }
        }), [ u, a ]);
        var c = (0, r.useMemo)((function() {
            return {
                onDragStart: function e(n) {
                    var r = n.active.id;
                    return (0, te.sprintf)("Picked up sortable row in position %d of %d.", oe(u, r), u.length);
                },
                onDragOver: function e(n) {
                    var r = n.active, t = n.over;
                    // In this specific use-case, the picked up item's `id` is always the same as the first `over` id.
                    // The first `onDragOver` event therefore doesn't need to be announced, because it is called
                    // immediately after the `onDragStart` announcement and is redundant.
                                        if (l.current) {
                        l.current = false;
                        return undefined;
                    }
                    return t ? (0, te.sprintf)("Row moved from position %d to position %d of %d.", oe(u, r.id), oe(u, t.id), u.length) : undefined;
                },
                onDragEnd: function e(n) {
                    var r = n.over;
                    return r ? (0, te.sprintf)("Row dropped at position %d of %d.", oe(u, r.id), u.length) : undefined;
                },
                onDragCancel: function e(n) {
                    var r = n.active.id;
                    return (0, te.sprintf)("Row returned to it's starting position of %d.", oe(u, r));
                }
            };
        }), [ u ]);
        var f = (0, r.useMemo)((function() {
            return {
                screenReaderInstructions: ae,
                announcements: c
            };
        }), [ c ]);
        
        return t().createElement(re.DndContext, {
            accessibility: f,
            sensors: i,
            collisionDetection: re.closestCenter,
            onDragEnd: d
        },  t().createElement(B.SortableContext, {
            items: u,
            strategy: B.verticalListSortingStrategy
        }, s));
    };
    /* harmony default export */    const le = ie;
    // CONCATENATED MODULE: ./src/FormRows/FormRows.tsx
    function se() {
        return se = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, se.apply(null, arguments);
    }
    function ue(e, n) {
        if (null == e) return {};
        var r, t, o = de(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function de(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var ce = {
        addLabel: a().string,
        children: a().node,
        disabled: a().bool,
        elementRef: a().oneOfType([ a().func, a().object ]),
        header: a().node,
        menu: a().node,
        onRequestAdd: a().func,
        onRequestMove: a().func
    };
    /**
 * Static function for adding a row.
 *
 * Example Use:
 *
 *      onRequestAdd() => {
 *          this.setState(state => ({
 *              items: FormRows.addRow(
 *                  <FormRows.Row
 *                      index={state.items.length}
 *                      key={createDOMID()}
 *                      onRequestRemove={this.handleRequestRemove}
 *                  >
 *                      <Text />
 *                  </FormRows.Row>,
 *                  state.items
 *              ),
 *          }));
 *      };
 *
 * @param element Row element to add.
 * @param items Array of current FormRows to add to.
 * @return New array of FormRows with new Row added.
 */    var fe = function e(n, r) {
        return r.concat(n);
    };
    /**
 * Static function for moving a row.
 *
 * Example Use:
 *
 *      onRequestMove({ fromIndex, toIndex }) => {
 *          this.setState(state => ({
 *              items: FormRows.moveRow(fromIndex, toIndex, state.items),
 *          }));
 *      };
 *
 * @param fromIndex Current index of row to move.
 * @param toIndex New index to move row to.
 * @param items Array of current FormRows.
 * @return New array of FormRows arranged in new order.
 */    var ve = function e(n, t, o) {
        var a = o.filter((function(e, r) {
            return r !== n;
        }));
        a.splice(t, 0, o[n]);
        return a.map((function(e, n) {
            
            return (0, r.cloneElement)(e, {
                index: n
            });
        }));
    };
    /**
 * Static function for removing a row.
 *
 * Example Use:
 *
 *      onRequestRemove(e, { index }) => {
 *          this.setState(state => ({
 *              items: FormRows.removeRow(index, state.items),
 *          }));
 *      };
 *
 * @param index Index of Row to delete.
 * @param items Array of current FormRows.
 * @return New array of FormRows with Row of given index deleted.
 */    var me = function e(n, t) {
        return t.filter((function(e, r) {
            return r !== n;
        })).map((function(e, n) {
            
            return (0, r.cloneElement)(e, {
                index: n
            });
        }));
    };
    var pe =  t().createElement(l(), null);
    var be = (0, s._)("Add row");
    function ge(e) {
        var n = e.addLabel, o = n === void 0 ? be : n, a = e.children, i = e.disabled, l = e.elementRef, s = e.header, u = e.menu, c = e.onRequestAdd, f = e.onRequestMove, v = ue(e, [ "addLabel", "children", "disabled", "elementRef", "header", "menu", "onRequestAdd", "onRequestMove" ]);
        // @docs-props-type FormRowsPropsBase
                var m = (0, r.useCallback)((function(e, n) {
            var r = n.index;
            var o = e.key;
            if (e.currentTarget !== e.target) {
                return;
            }
            if (o === "ArrowUp") {
                e.preventDefault();
                if (r > 0) {
                    f === null || f === void 0 ? void 0 : f({
                        fromIndex: r,
                        toIndex: r - 1
                    });
                }
            } else if (o === "ArrowDown") {
                e.preventDefault();
                if (r < t().Children.toArray(a).length - 1) {
                    f === null || f === void 0 ? void 0 : f({
                        fromIndex: r,
                        toIndex: r + 1
                    });
                }
            }
        }), [ a, f ]);
        var p = (0, r.useCallback)((function(e) {
            var n = e.oldIndex, r = e.newIndex;
            f === null || f === void 0 ? void 0 : f({
                fromIndex: n,
                toIndex: r
            });
        }), [ f ]);
        var b = f !== undefined && !i;
        var g = c !== undefined;
        var y = (0, r.useMemo)((function() {
            return {
                disabledRemoveButton: i,
                onKeyDown: m,
                hasOnRequestAdd: g,
                sortable: b
            };
        }), [ i, m, g, b ]);
        
        return t().createElement(R, se({
            ref: l,
            "data-test": "form-rows"
        }, v),  t().createElement(d.Provider, {
            value: y
        }, s &&  t().createElement(O, null, s),  t().createElement(le, {
            onSortEnd: p
        }, a), u &&  t().createElement(j, null, u) || g &&  t().createElement(S, {
            disabled: i,
            appearance: "secondary",
            "data-test": "add-row",
            icon: pe,
            label: o,
            onClick: c
        })));
    }
    ge.propTypes = ce;
    ge.Row = ne;
    ge.addRow = fe;
    ge.moveRow = ve;
    ge.removeRow = me;
    /* harmony default export */    const ye = ge;
    // CONCATENATED MODULE: ./src/FormRows/index.ts
    module.exports = n;
    /******/})();
