// @splunk/react-ui/Heading.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        StyledHeading: () => /* reexport */ f,
        default: () => /* reexport */ g,
        levelToElement: () => /* reexport */ h
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Typography"
    const u = require("@splunk/react-ui/Typography");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const c = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Heading/HeadingStyles.ts
    var f = o()(s()).withConfig({
        displayName: "HeadingStyles__StyledTypography",
        componentId: "sc-15s3yx3-0"
    })([ "", " ", "" ], c.mixins.reset("block"), (function(e) {
        var t = e.$variant;
        return t && (0, a.css)([ "", "" ], c.mixins.typography(t));
    }));
    // CONCATENATED MODULE: ./src/Heading/Heading.tsx
    function d() {
        return d = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, d.apply(null, arguments);
    }
    function p(e, t) {
        if (null == e) return {};
        var r, n, i = v(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function v(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var y = {
        children: l().node.isRequired,
        elementRef: l().oneOfType([ l().func, l().object ]),
        level: l().oneOf([ 1, 2, 3, 4, 5, 6 ]).isRequired,
        variant: l().oneOf([ "title1", "title2", "title3", "title4", "title5", "title6", "title7" ])
    };
    // h1-h6 use the same intrinsic element type
        var h = {
        1: "h1",
        2: "h2",
        3: "h3",
        4: "h4",
        5: "h5",
        6: "h6"
    };
    var m = {
        h1: "title1",
        h2: "title2",
        h3: "title3",
        h4: "title4",
        h5: "title5",
        h6: "title6"
    };
    function b(e) {
        var t = e.level, r = e.elementRef, i = e.children, l = e.variant, a = p(e, [ "level", "elementRef", "children", "variant" ]);
        // @docs-props-type HeadingPropsBase
                var o = h[t];
        var u = l || m[o];
        
        return n().createElement(f, d({
            forwardedAs: o,
            "data-test": "heading",
            elementRef: r,
            $variant: u
        }, a), i);
    }
    b.propTypes = y;
    /* harmony default export */    const g = b;
    // CONCATENATED MODULE: ./src/Heading/index.ts
    module.exports = t;
    /******/})();
