// @splunk/react-ui/Image.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ x,
        isAllowedFilename: () => /* reexport */ R
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/endsWith"
    const a = require("lodash/endsWith");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "lodash/toLower"
    const l = require("lodash/toLower");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var f =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/File"
    const s = require("@splunk/react-ui/File");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Message"
    const d = require("@splunk/react-ui/Message");
    var v =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const m = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/Image/Image.tsx
    function p() {
        return p = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, p.apply(null, arguments);
    }
    function y(e, r) {
        return w(e) || O(e, r) || b(e, r) || g();
    }
    function g() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function b(e, r) {
        if (e) {
            if ("string" == typeof e) return h(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? h(e, r) : void 0;
        }
    }
    function h(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function O(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, a, o, l, i = [], u = !0, f = !1;
            try {
                if (o = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (n = o.call(t)).done) && (i.push(n.value), i.length !== r); u = !0) {
                }
            } catch (e) {
                f = !0, a = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (l = t["return"](), Object(l) !== l)) return;
                } finally {
                    if (f) throw a;
                }
            }
            return i;
        }
    }
    function w(e) {
        if (Array.isArray(e)) return e;
    }
    function j(e, r) {
        if (null == e) return {};
        var t, n, a = I(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                t = o[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (a[t] = e[t]);
            }
        }
        return a;
    }
    function I(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */    var S = {
        allowExtensions: f().arrayOf(f().string),
        defaultFilename: f().string,
        defaultImageDataURI: f().string,
        disabled: f().bool,
        dropAnywhere: f().bool,
        elementRef: f().oneOfType([ f().func, f().object ]),
        error: f().bool,
        onImageChange: f().func
    };
    var R = function e(r, t) {
        return !!r && t.some((function(e) {
            return o()(i()(r), ".".concat(i()(e)));
        }));
    };
    var E = new FileReader;
    var A = [ "gif", "jpeg", "jpg", "png" ];
    /**
 * Image provides the ability to accept image files and present a preview of the image.
 */    function q(e) {
        var r = e.allowExtensions, a = r === void 0 ? A : r, o = e.defaultFilename, l = e.defaultImageDataURI, i = e.disabled, u = e.error, f = e.onImageChange, s = j(e, [ "allowExtensions", "defaultFilename", "defaultImageDataURI", "disabled", "error", "onImageChange" ]);
        // @docs-props-type ImagePropsBase
                var d = (0, t.useState)(o), g = y(d, 2), b = g[0], h = g[1];
        var O = (0, t.useState)(l), w = y(O, 2), I = w[0], S = w[1];
        var q = (0, t.useRef)(o), x = q.current;
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ x, o ]);
        var P = (0, t.useCallback)((function(e) {
            if (e.length > 0) {
                var r = e[0];
                if (E.readyState === 1) {
                    E.abort();
                }
                E.onload = function() {
                    h(r.name);
                    S(E.result);
 // TS: guaranteed by readAsDataURL call below (see spec)
                                        if (R(r.name, a)) {
                        f === null || f === void 0 ? void 0 : f({
                            filename: r.name,
                            imageDataURI: E.result
                        });
                    }
                };
                E.readAsDataURL(r);
            }
        }), [ a, f ]);
        var U = (0, t.useCallback)((function() {
            if (E.readyState === 1) {
                E.abort();
            }
            h(undefined);
            S(undefined);
            f === null || f === void 0 ? void 0 : f({
                filename: null,
                imageDataURI: null
            });
        }), [ f ]);
        var C = a.map((function(e) {
            return ".".concat(e);
        })).join(", ");
        // TODO: Error text should be customizable, error should support a string prop for customization SUI-5414
                var D = b && !R(b, a) || u;
        var F = (0, m._)("This file extension is unsupported.");
        
        return n().createElement("div", {
            "data-test": "image"
        },  n().createElement(c(), p({}, s, {
            accept: C,
            allowMultiple: false,
            disabled: i,
            error: D,
            onRequestAdd: P,
            onRequestRemove: U,
            "data-test": "file"
        }), b && I && !D &&  n().createElement(c().Item, {
            icon:  n().createElement("img", {
                alt: b,
                "data-test": "preview",
                src: I
            }),
            name: b
        })), D &&  n().createElement(v(), {
            appearance: "fill",
            type: "error"
        }, F));
    }
    q.propTypes = S;
    /* harmony default export */    const x = q;
    // CONCATENATED MODULE: ./src/Image/index.ts
    module.exports = r;
    /******/})();
