// @splunk/react-ui/JSONTree.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ Ae
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/isString"
    const a = require("lodash/isString");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const u = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tree"
    const s = require("@splunk/react-ui/Tree");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/JSONTree/JSONTreeStyles.ts
    var v = (0, c.css)([ "", ";", ";color:", ";padding:4px;" ], p.mixins.reset("block"), p.mixins.typography({
        family: "monospace"
    }), p.variables.syntaxGray);
    var y = d().code.withConfig({
        displayName: "JSONTreeStyles__StyledScrollCode",
        componentId: "vrdt21-0"
    })([ "", ";overflow-x:auto;white-space:pre;" ], v);
    var b = d()(f()).withConfig({
        displayName: "JSONTreeStyles__StyledTree",
        componentId: "vrdt21-1"
    })([ "font-family:inherit;ul{font-family:inherit;}" ]);
    var m = d().code.withConfig({
        displayName: "JSONTreeStyles__StyledWrapCode",
        componentId: "vrdt21-2"
    })([ "", ";white-space:pre-wrap;" ], v);
    // CONCATENATED MODULE: external "lodash/fill"
    const h = require("lodash/fill");
    var g =  e.n(h);
    // CONCATENATED MODULE: external "lodash/isArray"
    const S = require("lodash/isArray");
    var k =  e.n(S);
    // CONCATENATED MODULE: external "lodash/isEmpty"
    const x = require("lodash/isEmpty");
    var O =  e.n(x);
    // CONCATENATED MODULE: external "lodash/isNumber"
    const w = require("lodash/isNumber");
    var C =  e.n(w);
    // CONCATENATED MODULE: external "lodash/isObject"
    const E = require("lodash/isObject");
    var I =  e.n(E);
    // CONCATENATED MODULE: external "lodash/keys"
    const j = require("lodash/keys");
    var R =  e.n(j);
    // CONCATENATED MODULE: external "lodash/repeat"
    const T = require("lodash/repeat");
    var A =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const P = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const _ = require("@splunk/react-ui/ButtonSimple");
    var K =  e.n(_);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const N = require("@splunk/react-ui/Clickable");
    var V =  e.n(N);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const q = require("@splunk/react-ui/Tooltip");
    var L =  e.n(q);
    // CONCATENATED MODULE: ./src/JSONTree/JSONTreeItemStyles.ts
    var D = {
        string: p.variables.syntaxTeal,
        number: p.variables.syntaxBlue,
        boolean: p.variables.syntaxPurple,
        null: p.variables.syntaxBrown,
        object: p.variables.syntaxGray,
        array: p.variables.syntaxGray
    };
    var J = (0, c.css)([ "word-break:break-word;word-wrap:break-word;" ]);
    var F = (0, c.css)([ "white-space:pre;" ]);
    /* Using this magic number to set the height to be at least the minimum target size of 24 by 24 */    var M = (0, 
    c.css)([ "padding-block:2px;" ]);
    var B = (0, c.css)([ "font-family:inherit;&:focus{box-shadow:", ";outline:0;&:active{box-shadow:none;}}&:hover{background-color:", ";}" ], p.variables.focusShadow, p.variables.neutral50);
    var G = (0, c.css)([ "color:", ";", ";", "" ], (function(e) {
        var n = e.valueType;
        return D[n];
    }), (function(e) {
        var n = e.overflowType;
        return n === "wrap" ? J : F;
    }), M);
    var U = d().span.withConfig({
        displayName: "JSONTreeItemStyles__StyledContentWrapper",
        componentId: "o558h3-0"
    })([ "color:", ";" ], p.variables.syntaxGray);
    var $ = d().span.withConfig({
        displayName: "JSONTreeItemStyles__StyledValue",
        componentId: "o558h3-1"
    })([ "", "" ], G);
    var W = d()(V()).withConfig({
        displayName: "JSONTreeItemStyles__StyledValueInteractiveClickable",
        componentId: "o558h3-2"
    })([ "", " ", " min-width:24px;text-align:center;" ], G, B);
    var X = (0, c.css)([ "", " color:", ";" ], M, p.variables.syntaxRed);
    var H = d().span.withConfig({
        displayName: "JSONTreeItemStyles__StyledProperty",
        componentId: "o558h3-3"
    })([ "", "" ], X);
    var z = d()(V()).withConfig({
        displayName: "JSONTreeItemStyles__StyledPropertyInteractiveClickable",
        componentId: "o558h3-4"
    })([ "", " ", "" ], X, B);
    var Q = d()(K()).withConfig({
        displayName: "JSONTreeItemStyles__StyledExpandLinkClickable",
        componentId: "o558h3-5"
    })([ "", ";border:none;border-radius:0;font-family:inherit;padding:1px ", " 3px ", ";&:hover{background-color:", ";text-decoration:none;}&:focus{box-shadow:", ";&:active{box-shadow:none;}}" ], M, p.variables.spacingXSmall, p.variables.spacingXSmall, p.variables.actionColorBackgroundSubtleHover, p.variables.focusShadowInset);
    var Y = d()(L()).withConfig({
        displayName: "JSONTreeItemStyles__StyledTooltip",
        componentId: "o558h3-6"
    })([ "font-family:", ";" ], p.variables.monoFontFamily);
    // CONCATENATED MODULE: ./src/JSONTree/JSONTreeItem.tsx
    function Z() {
        return Z = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, Z.apply(null, arguments);
    }
    function ee(e, n) {
        return oe(e) || ae(e, n) || re(e, n) || ne();
    }
    function ne() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function re(e, n) {
        if (e) {
            if ("string" == typeof e) return te(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? te(e, n) : void 0;
        }
    }
    function te(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function ae(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, a, o, l, i = [], u = !0, c = !1;
            try {
                if (o = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = o.call(r)).done) && (i.push(t.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (l = r["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function oe(e) {
        if (Array.isArray(e)) return e;
    }
    function le(e, n) {
        if (null == e) return {};
        var r, t, a = ie(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function ie(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var ue = {
        clickableKeyRef: i().object,
        clickableValRef: i().object,
        hasChildren: i().bool,
        indentArray: i().arrayOf(i().string),
        index: i().number,
        initialOpenState: i().bool,
        isRoot: i().bool,
        properties: i().arrayOf(i().string),
        propertyDataPath: i().string,
        propertyElement: i().node,
        renderExpandLink: i().func,
        representation: i().node,
        value: i().oneOfType([ i().string, i().object, i().array, i().number, i().bool ])
    };
    function ce(e) {
        var n = e.clickableKeyRef, a = e.clickableValRef, o = e.hasChildren, l = e.indentArray, i = e.index, u = e.initialOpenState, c = e.isRoot, d = e.onFocus, s = e.properties, p = e.propertyDataPath, v = e.propertyElement, y = e.renderExpandLink, b = e.representation, m = e.value, h = le(e, [ "clickableKeyRef", "clickableValRef", "hasChildren", "indentArray", "index", "initialOpenState", "isRoot", "onFocus", "properties", "propertyDataPath", "propertyElement", "renderExpandLink", "representation", "value" ]);
        // @docs-props-type JSONTreeItemPropsBase
                var g = (0, r.useState)(u), S = ee(g, 2), x = S[0], O = S[1];
        var w = (0, r.useCallback)((function() {
            O((function(e) {
                return !e;
            }));
        }), []);
        var C = (0, r.useRef)(null);
        var E = y ? y({
            onToggleExpansion: w,
            open: x,
            withTooltip: true,
            expandLinkRef: C
        }) : {
            expandLink: undefined
        }, I = E.expandLink;
        var j = (s === null || s === void 0 ? void 0 : s.length) && i + 1 < (s === null || s === void 0 ? void 0 : s.length) ? "," : null;
        
        return t().createElement(f().Item, Z({
            content:  t().createElement(U, null, v, o && (k()(m) ? "[" : "{"), !o &&  t().createElement(t().Fragment, null, b, j), o && I),
            customIndent: l,
            "data-test": o ? "node" : null,
            "data-test-path": p,
            endAdornment: o &&  t().createElement(U, null, x && l, k()(m) ? "]" : "}", j),
            expanded: x,
            onFocus: function e(r) {
                d === null || d === void 0 ? void 0 : d(r);
                // check if onFocus was triggered by a click event or expand link
                                var t = r.relatedTarget === (n === null || n === void 0 ? void 0 : n.current) || r.relatedTarget === (a === null || a === void 0 ? void 0 : a.current) || r.relatedTarget === (C === null || C === void 0 ? void 0 : C.current) ? r.relatedTarget : null;
                if (c) {
                    var o;
                    C === null || C === void 0 ? void 0 : (o = C.current) === null || o === void 0 ? void 0 : o.focus();
                }
                if (!t) {
                    // if JSONTreeItem contains clickable keys/vals, focus should go there
                    if (v && (n === null || n === void 0 ? void 0 : n.current)) {
                        var l;
                        (l = n.current) === null || l === void 0 ? void 0 : l.focus();
                    } else if (a === null || a === void 0 ? void 0 : a.current) {
                        var i;
                        (i = a.current) === null || i === void 0 ? void 0 : i.focus();
                    }
                } else {
                    t.focus();
                }
            },
            onToggleExpansion: w,
            tabIndex: 0
        }, h), o && b);
    }
    ce.propTypes = ue;
    // CONCATENATED MODULE: ./src/JSONTree/renderTreeItems.tsx
    function de(e) {
        "@babel/helpers - typeof";
        return de = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, de(e);
    }
    function se(e) {
        return ye(e) || ve(e) || pe(e) || fe();
    }
    function fe() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function pe(e, n) {
        if (e) {
            if ("string" == typeof e) return be(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? be(e, n) : void 0;
        }
    }
    function ve(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function ye(e) {
        if (Array.isArray(e)) return be(e);
    }
    function be(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function me(e) {
        var n = e.id, a = e.childKey, o = e.index, l = e.value, i = e.properties, u = e.propertyDataPath, c = e.obj, d = e.renderValue, s = e.renderKey, f = e.getInitialOpenState, p = e.indentArray, v = e.renderExpandLink, y = e.onFocus;
        var b = (0, r.useRef)(null);
        var m = (0, r.useRef)(null);
        var h = d({
            key: a,
            value: l,
            clickValueRef: b
        }), g = h.representation, S = h.clickableValRef;
        var x = !k()(c) ? s(a, m) : {
            propertyElement: undefined,
            clickableKeyRef: undefined
        }, w = x.propertyElement, C = x.clickableKeyRef;
        var E = I()(l) && !O()(l) || k()(l) && l.length > 0;
        
        return t().createElement(ce, {
            id: n,
            propertyDataPath: u,
            propertyElement: w,
            hasChildren: E,
            initialOpenState: f(),
            representation: g,
            indentArray: p,
            value: l,
            index: o,
            properties: i,
            renderExpandLink: v,
            clickableKeyRef: C,
            clickableValRef: S,
            onFocus: function e() {
                y(C, S);
            }
        });
    }
    function he(e) {
        var n = e.baseId, a = e.renderValue, o = e.obj;
        var l = (0, r.useRef)(null);
        var i = a({
            value: o,
            clickValueRef: l
        }), u = i.representation;
        
        return t().createElement(f().Item, {
            id: "".concat(n, "-0"),
            "data-test-path": ".",
            content: u
        });
    }
    function ge(e) {
        var n = e.baseId, r = e.defaultOpen, a = r === void 0 ? false : r, l = e.defaultExpanded, i = l === void 0 ? false : l, u = e.expandChildrenOnShiftKey, c = u === void 0 ? false : u, d = e.indent, s = d === void 0 ? 4 : d, f = e.indentLevel, p = f === void 0 ? 1 : f, v = e.obj, y = e.onClickKey, b = e.onClickValue, m = e.overflow, h = m === void 0 ? "scroll" : m, S = e.path, x = S === void 0 ? [] : S, w = e.shift, E = e.stringRenderer, j = E === void 0 ? "literal" : E, T = e.updateShift;
        var _ = g()(Array(p), A()(" ", s));
        var K;
        var N;
        var V = function e() {
            if (i && w == null) {
                return true;
            }
            if (c) {
                return !!w;
            }
            return a;
        };
        var q = function e(n) {
            if (c && T != null) {
                T(n.shiftKey);
            }
        };
        var L = function e(n, r) {
            var t, a;
            // When rendering a single value key is not present, here we prevent `[undefined]` in path
                        var o = r !== undefined ? [].concat(se(x), [ r ]) : se(x);
            var l = ".".concat(o.join("."));
            (t = N) === null || t === void 0 ? void 0 : (a = t.current) === null || a === void 0 ? void 0 : a.focus();
            y === null || y === void 0 ? void 0 : y(n, {
                key: l,
                keyPath: o
            });
        };
        var D = function e(n, r) {
            var t, a;
            // When rendering a single value key is not present, here we prevent `[undefined]` in path
                        var o = r !== undefined ? [].concat(se(x), [ r ]) : se(x);
            var l = ".".concat(o.join("."));
            (t = K) === null || t === void 0 ? void 0 : (a = t.current) === null || a === void 0 ? void 0 : a.focus();
            b === null || b === void 0 ? void 0 : b(n, {
                key: l,
                keyPath: o,
                value: n.target.textContent
            });
        };
        var J = function e(n, r) {
            var t, a;
            var o = n.key;
            if (o === "ArrowLeft" && y && ((t = N) === null || t === void 0 ? void 0 : t.current) && !r) {
                var l;
                n.stopPropagation();
                (l = N.current) === null || l === void 0 ? void 0 : l.focus();
            }
            if (o === "ArrowRight" && b && ((a = K) === null || a === void 0 ? void 0 : a.current) && r) {
                var i;
                n.stopPropagation();
                (i = K.current) === null || i === void 0 ? void 0 : i.focus();
            }
        };
        var F = function e(n) {
            var r = n.onToggleExpansion, a = n.open, o = n.withTooltip, l = n.expandLinkRef;
            var i = a ? "[-]" : "[+]";
            var u = a ? (0, P._)("Collapse all") : (0, P._)("Shift click to expand all");
            var d =  t().createElement(Q, {
                appearance: "subtle",
                elementRef: l,
                tabIndex: -1,
                onClick: function e(n) {
                    q(n);
                    r === null || r === void 0 ? void 0 : r(n);
                },
                "aria-expanded": a,
                "data-test": "toggle"
            }, i);
            if (c && o) {
                return {
                    expandLink:  t().createElement(Y, {
                        contentRelationship: "description",
                        openDelay: "secondary",
                        defaultPlacement: "above",
                        content: (0, P._)(u)
                    }, d),
                    expandLinkRef: l
                };
            }
            return {
                expandLink: d,
                expandLinkRef: l
            };
        };
        var M = function e(r) {
            var a = r.value, l = r.key, u = r.clickValueRef;
            var d;
            if (!I()(a)) {
                if (typeof a === "string") {
                    d = j === "stringify" ? JSON.stringify(a) : '"'.concat(a, '"');
                } else if (typeof a === "boolean") {
                    d = a.toString();
                } else if (a === null) {
                    d = "null";
                } else {
                    d = a;
                }
            } else if (O()(a)) {
                // isObject is true for arrays
                d = k()(a) ? "[]" : "{}";
            } else {
                d = ge({
                    baseId: n,
                    defaultOpen: i === true,
                    defaultExpanded: i,
                    expandChildrenOnShiftKey: c,
                    onClickKey: y,
                    onClickValue: b,
                    overflow: h,
                    obj: a,
                    path: l !== undefined ? [].concat(se(x), [ l ]) : se(x),
                    indentLevel: p + 1,
                    shift: w,
                    updateShift: T
                });
            }
            if (o()(d) || C()(d)) {
                var s = a === null ? "null" : // TS: bigint, function, symbol are prevented by JSONElement type
                de(a);
                if (b) {
                    return {
                        representation:  t().createElement(W, {
                            elementRef: u,
                            valueType: s,
                            overflowType: h,
                            onClick: function e(n) {
                                K = u;
                                D(n, l);
                            },
                            onKeyDown: function e(n) {
                                J(n, false);
                            },
                            tabIndex: -1
                        }, d),
                        clickableValRef: u
                    };
                }
                return {
                    representation:  t().createElement($, {
                        valueType: s,
                        overflowType: h
                    }, d)
                };
            }
            return {
                representation: d
            };
        };
        var B = function e(n, r) {
            if (y) {
                return {
                    propertyElement:  t().createElement(t().Fragment, null,  t().createElement(z, {
                        elementRef: r,
                        tabIndex: -1,
                        onClick: function e(t) {
                            N = r;
                            L(t, n);
                        },
                        onKeyDown: function e(n) {
                            J(n, true);
                        }
                    }, n), ":", " "),
                    clickableKeyRef: r
                };
            }
            return {
                propertyElement:  t().createElement(t().Fragment, null,  t().createElement(H, null, n), ":", " ")
            };
        };
        if (I()(v) && !O()(v) || k()(v) && v.length > 0) {
            var G = R()(v).map((function(e, r, a) {
                var o = v[e];
 // TS: should be safe based on (recursive) JSONElement type
                                var l = k()(v) ? parseInt(e, 10) : e;
                var i = x.length === 0 ? ".".concat(l) : ".".concat(x.join("."), ".").concat(l);
                
                return t().createElement(me, {
                    id: "".concat(n, "-").concat(i),
                    key: i,
                    propertyDataPath: i,
                    childKey: l,
                    index: r,
                    value: o,
                    properties: a,
                    obj: v,
                    renderValue: M,
                    renderKey: B,
                    getInitialOpenState: V,
                    indentArray: _,
                    renderExpandLink: F,
                    onFocus: function e(n, r) {
                        N = n;
                        K = r;
                    }
                });
            }));
            if (x.length === 0) {
                var U = I()(v) && !O()(v) || k()(v) && v.length > 0;
                
                return t().createElement(ce, {
                    id: "".concat(n, "-root"),
                    isRoot: true,
                    propertyDataPath: ".",
                    key: ".",
                    hasChildren: U,
                    initialOpenState: a || i,
                    value: v,
                    index: 0,
                    renderExpandLink: F,
                    representation: G
                });
            }
            return G;
        }
        // edge case: a single number/string/boolean being rendered (still valid JSON)
                
        return t().createElement(he, {
            baseId: n,
            obj: v,
            renderValue: M
        });
    }
    // CONCATENATED MODULE: ./src/JSONTree/JSONTree.tsx
    function Se() {
        return Se = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, Se.apply(null, arguments);
    }
    function ke(e, n) {
        return Ee(e) || Ce(e, n) || Oe(e, n) || xe();
    }
    function xe() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Oe(e, n) {
        if (e) {
            if ("string" == typeof e) return we(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? we(e, n) : void 0;
        }
    }
    function we(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function Ce(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, a, o, l, i = [], u = !0, c = !1;
            try {
                if (o = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = o.call(r)).done) && (i.push(t.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (l = r["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function Ee(e) {
        if (Array.isArray(e)) return e;
    }
    function Ie(e, n) {
        if (null == e) return {};
        var r, t, a = je(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function je(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */    var Re = {
        elementRef: i().oneOfType([ i().func, i().object ]),
        defaultExpanded: i().bool,
        indent: i().number,
        json: i().oneOfType([ i().string, i().object, i().array, i().number, i().bool ]).isRequired,
        onClickKey: i().func,
        onClickValue: i().func,
        overflow: i().oneOf([ "wrap", "scroll" ]),
        expandChildrenOnShiftKey: i().bool
    };
    /**
 * Used to visualize a JSON string.
 */    function Te(e) {
        var n = e.elementRef, a = e.defaultExpanded, l = a === void 0 ? false : a, i = e.indent, c = i === void 0 ? 4 : i, d = e.json, s = e.onClickKey, f = e.onClickValue, p = e.overflow, v = p === void 0 ? "scroll" : p, h = e.expandChildrenOnShiftKey, g = h === void 0 ? true : h, S = e.stringRenderer, k = S === void 0 ? "literal" : S, x = Ie(e, [ "elementRef", "defaultExpanded", "indent", "json", "onClickKey", "onClickValue", "overflow", "expandChildrenOnShiftKey", "stringRenderer" ]);
        // @docs-props-type JSONTreePropsBase
                var O = (0, r.useMemo)((function() {
            return x.id || (0, u.createDOMID)("json-tree");
        }), [ x.id ]);
        var w = o()(d) ? JSON.parse(d) : d;
        var C = v === "wrap" ? m : y;
        var E = (0, r.useState)(undefined), I = ke(E, 2), j = I[0], R = I[1];
        var T = (0, r.useCallback)((function(e) {
            R(e);
        }), []);
        var A = (0, r.useMemo)((function() {
            return ge({
                baseId: O,
                defaultOpen: true,
                defaultExpanded: l,
                expandChildrenOnShiftKey: g,
                indent: c,
                obj: w,
                onClickKey: s,
                onClickValue: f,
                overflow: v,
                shift: j,
                stringRenderer: k,
                updateShift: T
            });
        }), [ O, l, g, c, w, s, f, v, j, k, T ]);
        
        return t().createElement(C, Se({
            "data-language": "language-json",
            "data-test": "json-tree",
            ref: n,
            role: "tree"
        }, x),  t().createElement(b, {
            defaultIndent: false
        }, A));
    }
    Te.propTypes = Re;
    /* harmony default export */    const Ae = Te;
    // CONCATENATED MODULE: ./src/JSONTree/index.ts
    module.exports = n;
    /******/})();
