// @splunk/react-ui/Layer.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        LayerContext: () => /* reexport */ b,
        LayerStackContext: () => /* reexport */ g,
        LayerStackGlobalProvider: () => /* reexport */ E,
        default: () => /* reexport */ G
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "lodash/includes"
    const o = require("lodash/includes");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/last"
    const i = require("lodash/last");
    var u =  e.n(i);
    // CONCATENATED MODULE: external "lodash/pull"
    const l = require("lodash/pull");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "prop-types"
    const s = require("prop-types");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "react-dom"
    const d = require("react-dom");
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const v = require("@splunk/react-ui/EventListener");
    var y =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const p = require("@splunk/react-ui/usePrevious");
    var m =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const h = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/Layer/LayerStack.tsx
    /* global globalThis */
    /**
 * `LayerStackContext` is used to manage the array that `Layer` (and components that
 * depend on it, such as `Modal`) uses to determine the order of open layers.
 * @public
 */
    var g =  r().createContext([]);
    var b =  r().createContext({});
    var C = {
        children: f().node,
        name: f().string,
        scope: f().object,
        separateStackingContexts: f().bool
    };
    /**
 * A `LayerStackContext` provider that stores a shared layer stack using a global variable.
 * Applications should only use this provider if there's a known need to support multiple
 * instances of this library on the same page.
 */    function E(e) {
        var t = e.children, o = e.name, a = o === void 0 ? "__splunkui_layer_instances__" : o, i = e.scope, u = i === void 0 ? typeof window !== "undefined" ? window : globalThis : i, l = e.separateStackingContexts, c = l === void 0 ? false : l;
        if (!u[a]) {
            u[a] = [];
 // eslint-disable-line no-param-reassign
                }
        var s = (0, n.useMemo)((function() {
            return {
                separateStackingContexts: c
            };
        }), [ c ]);
        
        return r().createElement(g.Provider, {
            value: u[a]
        },  r().createElement(b.Provider, {
            value: s
        }, t));
    }
    E.propTypes = C;
    // CONCATENATED MODULE: external "styled-components"
    const S = require("styled-components");
    var k =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/themes"
    const w = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Layer/LayerStyles.ts
    var T = k().div.withConfig({
        displayName: "LayerStyles__StyledLayer",
        componentId: "ii6psl-0"
    })([ "", "" ], (function(e) {
        var t = e.$separateStackingContexts;
        return t && (0, S.css)([ "isolation:isolate;position:relative;z-index:", ";" ], w.variables.zindexLayer);
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var q = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function x() {
        var e = typeof document !== "undefined" ? document : q;
        return e;
    }
    var A = x();
    /* harmony default export */    const L = /* unused pure expression or super */ null && A;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var _ = {
        document: q,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function O() {
        var e = typeof window !== "undefined" ? window : _;
        return e;
    }
    var P = O();
    /* harmony default export */    const j = P;
    // CONCATENATED MODULE: ./src/Layer/Layer.tsx
    function M(e, t) {
        return F(e) || D(e, t) || N(e, t) || I();
    }
    function I() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function N(e, t) {
        if (e) {
            if ("string" == typeof e) return R(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? R(e, t) : void 0;
        }
    }
    function R(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function D(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, o, a, i, u = [], l = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    l = !1;
                } else for (;!(l = (r = a.call(n)).done) && (u.push(r.value), u.length !== t); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function F(e) {
        if (Array.isArray(e)) return e;
    }
    /** @public */
    /** @public */    var K = [ "clickAway", "escapeKey" ];
    var $ = {
        children: f().node,
        closeReasons: f().arrayOf(f().oneOf(K)),
        onRequestClose: f().func,
        open: f().bool
    };
    var z = {
        passive: true
    };
    // we need a custom type here in order to set layerContainer as a property
        var B = function e(t) {
        var o = t.children, i = t.closeReasons, l = i === void 0 ? K : i, s = t.open, f = t.onRequestClose;
        var v = (0, n.useContext)(b), p = v.separateStackingContexts;
        var C = (0, n.useContext)(g);
        var E = (0, n.useRef)(null);
        var S = (0, n.useState)((0, h.createGUID)()), k = M(S, 1), w = k[0];
        var q = m()(s);
        (0, n.useEffect)((function() {
            if (s && !q) {
                C.push(w);
            } else if (q && !s) {
                c()(C, w);
            }
        }), [ w, C, s, q ]);
        // it is not part of the previous useEffect because it needs to be equivalent to componentWillUnmount
        // the previous useEffect will get re-run every time open and prevOpen change which is not what we want
                (0, n.useEffect)((function() {
            return function() {
                c()(C, w);
            };
        }), [ w, C ]);
        if (!e.layerContainer) {
            var A = x();
            var L = A.createElement("div");
            L.setAttribute("data-test", "layer-container");
            A.body.appendChild(L);
            e.layerContainer = L;
        }
        // the fullscreen API does not render elements outside the fullscreen element
        // Layer renders in document.body, which means that anything using Layer will not render in fullscreen
        // this code moves the layer container into the fullscreen element, ensuring that it will render
                var _ = (0, n.useCallback)((function() {
            if (e.layerContainer) {
                var t = x();
                var n = e.layerContainer.contains(t.fullscreenElement);
                if (t.fullscreenElement != null) {
                    if (!n) {
                        t.fullscreenElement.appendChild(e.layerContainer);
                    }
                } else {
                    t.body.appendChild(e.layerContainer);
                }
            }
        }), []);
        var O = (0, n.useCallback)((function(e) {
            var t = e.nativeEvent;
            E.current = t;
        }), []);
        var P = (0, n.useCallback)((function(e) {
            // clicks inside the layer should not be considered clickAways
            if (!s || !a()(l, "clickAway") || E.current === e) {
                return;
            }
            f === null || f === void 0 ? void 0 : f({
                event: e,
                reason: "clickAway"
            });
        }), [ l, s, f ]);
        var I = (0, n.useCallback)((function(e) {
            if (s && e.key === "Escape" && u()(C) === w && a()(l, "escapeKey")) {
                f === null || f === void 0 ? void 0 : f({
                    event: e,
                    reason: "escapeKey"
                });
            }
        }), [ l, w, C, s, f ]);
        var N =  r().createElement(y(), {
            target: j,
            eventType: "fullscreenchange",
            listener: _,
            key: "fullScreenChange"
        });
        if (s) {
            var R =  (0, d.createPortal)( r().createElement(T, {
                $separateStackingContexts: p,
                "data-test": "layer",
                onMouseDown: O,
                onTouchStart: O
            }, o), e.layerContainer);
            
            return r().createElement(r().Fragment, null,  r().createElement(y(), {
                target: j,
                eventType: "keydown",
                listener: I,
                key: "eventListenerKeydown"
            }),  r().createElement(y(), {
                target: j,
                eventType: "mousedown",
                listener: P,
                key: "eventListenerMouseDown"
            }),  r().createElement(y(), {
                target: j,
                eventType: "touchstart",
                listener: P,
                key: "eventListenerTouchStart",
                options: z
            }), N, R);
        }
        // the fragment has to be present to avoid type issues with React 18
        // eslint-disable-next-line react/jsx-no-useless-fragment
                
        return r().createElement(r().Fragment, null, N);
    };
    B.propTypes = $;
    B.layerContainer = null;
    B.possibleCloseReasons = K;
    /* harmony default export */    const G = B;
    // CONCATENATED MODULE: ./src/Layer/index.ts
    module.exports = t;
    /******/})();
