// @splunk/react-ui/Link.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        LinkContext: () => /* reexport */ y,
        LinkProvider: () => /* reexport */ m,
        default: () => /* reexport */ j
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const o = require("@splunk/react-ui/ScreenReaderContent");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const s = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var c =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRightInset"
    const u = require("@splunk/react-icons/ArrowSquareTopRightInset");
    var f =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Link/icons/External.tsx
    var b = c()(f()).withConfig({
        displayName: "External",
        componentId: "sc-13bdx45-0"
    })([ "vertical-align:text-bottom;margin-left:", ";" ], v.variables.spacingXSmall);
    /* harmony default export */    const p = b;
    // CONCATENATED MODULE: ./src/Link/LinkContext.tsx
    var g = {
        enableVisitedStyling: false
    };
    var y =  (0, t.createContext)(g);
    /**
 * Control behavior for all children Link components.
 */    var m = function e(n) {
        var i = n.enableVisitedStyling, a = i === void 0 ? false : i, o = n.children;
        // @docs-props-type LinkProviderProps
                var l = (0, t.useMemo)((function() {
            return {
                enableVisitedStyling: a
            };
        }), [ a ]);
        
        return r().createElement(y.Provider, {
            value: l
        }, o);
    };
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const h = require("@splunk/react-ui/Clickable");
    var S =  e.n(h);
    // CONCATENATED MODULE: ./src/Link/LinkStyles.ts
    var w = c()(S()).withConfig({
        displayName: "LinkStyles__StyledClickable",
        componentId: "sc-1hhltcf-0"
    })([ "", ";text-decoration:none;font-size:inherit;font-weight:inherit;line-height:inherit;color:", ";", " &:focus{box-shadow:", ";outline:0;&:active{box-shadow:none;}}&:not([disabled],[aria-disabled='true']){cursor:pointer;&:hover,&:active,&:focus{color:", ";text-decoration:underline;text-decoration-thickness:2px;}}&[disabled],&[aria-disabled='true']{color:", ";}", "" ], v.mixins.reset("inline"), v.variables.contentColorLink, (0, 
    v.pickVariant)("$appearance", {
        inline: (0, d.css)([ "text-decoration:underline;" ]),
        standalone: (0, d.css)([ "font-weight:", ";" ], v.variables.fontWeightSemiBold)
    }), v.variables.focusShadow, v.variables.contentColorLink, v.variables.contentColorLinkDisabled, (function(e) {
        var n = e.$enableVisitedStyling;
        return n && (0, d.css)([ "&:visited{color:", ";}&:where([disabled],[aria-disabled='true']):visited{color:", ";}" ], v.variables.contentColorLinkVisited, v.variables.contentColorLinkVisitedDisabled);
    }));
    // CONCATENATED MODULE: ./src/Link/Link.tsx
    function x() {
        return x = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, x.apply(null, arguments);
    }
    function O(e, n) {
        if (null == e) return {};
        var t, r, i = k(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (r = 0; r < a.length; r++) {
                t = a[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function k(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var C = {
        appearance: a().oneOf([ "inline", "standalone" ]),
        children: a().node,
        disabled: a().bool,
        elementRef: a().oneOfType([ a().func, a().object ]),
        enableVisitedStyling: a().bool,
        openInNewContext: a().oneOfType([ a().bool, a().string ]),
        to: a().string
    };
    var V = (0, s._)("(Opens new window)");
    /**
 * `Link` is a simple method for configuring `Button` for inline links. For more complex behaviors,
 * see the `Button` documentation.
 */    var P =  r().forwardRef((function(e, 
    // SUI-7623 - needs to be specified for react-docgen versions less than 7.1.0
    n) {
        var i = e.appearance, a = i === void 0 ? "inline" : i, o = e.children, s = e.disabled, d = s === void 0 ? false : s, c = e.elementRef, u = e.enableVisitedStyling, f = e.openInNewContext, v = f === void 0 ? false : f, b = e.tag, g = b === void 0 ? "a" : b, m = e.to, h = O(e, [ "appearance", "children", "disabled", "elementRef", "enableVisitedStyling", "openInNewContext", "tag", "to" ]);
        // @docs-props-type LinkPropsBase
                var S = typeof v === "string" ? v : V;
        // Ensure `Link` always renders as an `a` (unless overriden by `tag`) by passing a value to Clickable,
        // using '' if disabled or the to is not passed.
                var k = g !== "button" ? d && "" || m && m || "" : undefined;
        // Allow the prop to override context
                var C = (0, t.useContext)(y), P = C.enableVisitedStyling;
        var j = u !== null && u !== void 0 ? u : P;
        
        return r().createElement(w, x({
            "data-test": "link",
            disabled: d ? "disabled" : undefined,
            elementRef: c,
            openInNewContext: !!v,
            $appearance: a,
            $enableVisitedStyling: j,
            ref: n,
            to: k
        }, h, {
            tag: g
        }), o, v &&  r().createElement(r().Fragment, null,  r().createElement(p, null),  r().createElement(l(), null, S)));
    }));
    P.propTypes = C;
    /* harmony default export */    const j = P;
    // forwardRef appears to prevent TS from inferring LinkProps in components that import Link
    // so we need to export it separately to prevent the TS build from failing with:
    // "... has or is using name 'LinkPropsBase' from external module "...packages/react-ui/src/Link/Link" but cannot be named."
        // CONCATENATED MODULE: ./src/Link/index.ts
    module.exports = n;
    /******/})();
