// @splunk/react-ui/List.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Item: () => /* reexport */ b,
        StyledList: () => /* reexport */ c,
        default: () => /* reexport */ S
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/List/ListStyles.ts
    var c = a().ul.withConfig({
        displayName: "ListStyles__Styled",
        componentId: "sc-1yyh8l9-0"
    })([ "", ";", ";counter-reset:list-item;padding-inline-start:1em;" ], s.mixins.reset("block"), s.mixins.typography("body"));
    var u = a().li.withConfig({
        displayName: "ListStyles__StyledItem",
        componentId: "sc-1yyh8l9-1"
    })([ "", ";list-style-type:inherit;" ], s.mixins.reset("list-item"));
    // CONCATENATED MODULE: ./src/List/Item.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function d(e, r) {
        if (null == e) return {};
        var t, n, o = y(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function y(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var p = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ])
    };
    /**
 * A container for items of a `List`.
 */    function m(e) {
        var r = e.children, t = e.elementRef, o = d(e, [ "children", "elementRef" ]);
        // @docs-props-type ItemPropsBase
                
        return n().createElement(u, f({
            "data-test": "item",
            ref: t
        }, o), r);
    }
    m.propTypes = p;
    /* harmony default export */    const b = m;
    // CONCATENATED MODULE: ./src/List/List.tsx
    function v() {
        return v = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, v.apply(null, arguments);
    }
    function O(e, r) {
        if (null == e) return {};
        var t, n, o = h(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                t = l[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function h(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var g = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ]),
        ordered: l().bool
    };
    function j(e) {
        var r = e.children, t = e.elementRef, o = e.ordered, l = O(e, [ "children", "elementRef", "ordered" ]);
        // @docs-props-type ListPropsBase
                
        return n().createElement(c, v({
            as: o ? "ol" : "ul",
            "data-test": "list",
            ref: t
        }, l), r);
    }
    j.propTypes = g;
    j.Item = b;
    /* harmony default export */    const S = j;
    // CONCATENATED MODULE: ./src/List/index.ts
    module.exports = r;
    /******/})();
