// @splunk/react-ui/Markdown.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        MarkdownAnchorHeading: () => /* reexport */ j,
        MarkdownBlockquote: () => /* reexport */ M,
        MarkdownCode: () => /* reexport */ A,
        MarkdownCodeBlock: () => /* reexport */ F,
        MarkdownHeading: () => /* reexport */ V,
        MarkdownImage: () => /* reexport */ ee,
        MarkdownItem: () => /* reexport */ ie,
        MarkdownLink: () => /* reexport */ pe,
        MarkdownList: () => /* reexport */ Oe,
        MarkdownParagraph: () => /* reexport */ ke,
        default: () => /* reexport */ We
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "react-markdown"
    const i = require("react-markdown");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "remark-gfm"
    const c = require("remark-gfm");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Prose"
    const d = require("@splunk/react-ui/Prose");
    var p =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-ui/Anchor"
    const v = require("@splunk/react-ui/Anchor");
    var y =  e.n(v);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownAnchorHeading.tsx
    function b() {
        return b = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, b.apply(null, arguments);
    }
    function m(e, r) {
        if (null == e) return {};
        var n, t, o = O(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function O(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function g(e) {
        "@babel/helpers - typeof";
        return g = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, g(e);
    }
    // eslint-disable-next-line no-relative-imports-of-published -- because MarkdownHeading and MarkdownAnchorHeading come from the same @splunk/react-ui/Markdown top level export so it would be a circular dependency if using the external/alias (ok here because thin/no context)
    // When this is used in MDX, the node isn't guaranteed to be text,
    // so this method extracts the actual text from the node.
        function h(e) {
        var r = g(e);
        var n = "";
        if (e != null) {
            if (r === "string" || r === "number") {
                n = e.toString();
            } else if (r === "boolean") {
                n = e ? "true" : "false";
            } else if (g(e) === "object") {
                if (e instanceof Array) {
                    n = e.map(h).join(" ");
                } else if ("props" in e) {
                    n = h(e.props.children);
                }
            }
        }
        return n;
    }
    function w(e) {
        return n.Children.toArray(e).map(h).join(" ").toLowerCase().replace(/[^a-z0-9-. ]/g, "").replace(/[-. ]+/g, "-");
    }
    function P(e) {
        var r = e.level, n = e.children, o = m(e, [ "level", "children" ]);
        // @docs-props-type MarkdownAnchorHeadingPropsBase
                
        return t().createElement(V, b({
            level: r
        }, o),  t().createElement(y(), {
            name: w(n)
        }, n));
    }
    /* harmony default export */    const j = P;
    // CONCATENATED MODULE: external "@splunk/themes"
    const k = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Markdown/MarkdownStyles.ts
    var S = s().div.withConfig({
        displayName: "MarkdownStyles__StyledCodeBlock",
        componentId: "sc-1xadih-0"
    })([ "", "" ], k.mixins.reset("block"));
    var E = s().code.withConfig({
        displayName: "MarkdownStyles__StyledCodeInline",
        componentId: "sc-1xadih-1"
    })([ "", ";font-size:inherit;font-family:", ";background-color:", ";" ], k.mixins.reset("inline"), k.variables.monoFontFamily, k.variables.neutral100);
    var x = s().blockquote.withConfig({
        displayName: "MarkdownStyles__StyledHint",
        componentId: "sc-1xadih-2"
    })([ "", " padding:", ";border-left:10px solid ", ";>:last-child{margin-bottom:0;}" ], k.mixins.reset("block"), k.variables.spacingSmall, (0, 
    k.pick)({
        dark: k.variables.white,
        light: k.variables.neutral500
    }));
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownBlockquote.tsx
    function R(e, r) {
        if (null == e) return {};
        var n, t, o = q(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function q(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function I(e) {
        var r = e.children, n = R(e, [ "children" ]);
        // @docs-props-type MarkdownBlockquotePropsBase
                
        return t().createElement(x, n, r);
    }
    /* harmony default export */    const M = I;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownCode.tsx
    function T(e, r) {
        if (null == e) return {};
        var n, t, o = C(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function C(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function _(e) {
        var r = e.children, n = T(e, [ "children" ]);
        // @docs-props-type MarkdownCodePropsBase
                
        return t().createElement(E, n, r);
    }
    /* harmony default export */    const A = _;
    // CONCATENATED MODULE: external "@splunk/react-ui/Code"
    const B = require("@splunk/react-ui/Code");
    var H =  e.n(B);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const U = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownCodeBlock.tsx
    function L(e, r) {
        if (null == e) return {};
        var n, t, o = N(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function N(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function D(e) {
        var r = e.children, n = e.containerAppearance, o = n === void 0 ? "none" : n, l = e.language, i = L(e, [ "children", "containerAppearance", "language" ]);
        // @docs-props-type MarkdownCodeBlockPropsBase
                
        return t().createElement(S, i,  t().createElement(H(), {
            "aria-label": (0, U._)("Code Block"),
            containerAppearance: o,
            language: l,
            value: String(r).replace(/\n$/, "")
        }));
    }
    /* harmony default export */    const F = D;
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const z = require("@splunk/react-ui/Heading");
    var $ =  e.n(z);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownHeading.tsx
    function G() {
        return G = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, G.apply(null, arguments);
    }
    function J(e, r) {
        if (null == e) return {};
        var n, t, o = K(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function K(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function Q(e) {
        var r = e.level, n = e.children, o = J(e, [ "level", "children" ]);
        // @docs-props-type MarkdownHeadingPropsBase
                var l = Math.min(r, 6);
        
        return t().createElement($(), G({
            level: l
        }, o), n);
    }
    /* harmony default export */    const V = Q;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownImage.tsx
    function W() {
        return W = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, W.apply(null, arguments);
    }
    function X(e, r) {
        if (null == e) return {};
        var n, t, o = Y(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Y(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function Z(e) {
        var r = e.src, n = e.title, o = e.alt, l = X(e, [ "src", "title", "alt" ]);
        // @docs-props-type MarkdownImagePropsBase
                
        return t().createElement("img", W({
            src: r,
            title: n,
            alt: o
        }, l));
    }
    /* harmony default export */    const ee = Z;
    // CONCATENATED MODULE: external "@splunk/react-ui/List"
    const re = require("@splunk/react-ui/List");
    var ne =  e.n(re);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownItem.tsx
    function te(e, r) {
        if (null == e) return {};
        var n, t, o = oe(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function oe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function le(e) {
        var r = e.children, n = te(e, [ "children" ]);
        // @docs-props-type MarkdownItemPropsBase
                
        return t().createElement(re.Item, n, r);
    }
    /* harmony default export */    const ie = le;
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const ae = require("@splunk/react-ui/Link");
    var ce =  e.n(ae);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownLink.tsx
    function ue() {
        return ue = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, ue.apply(null, arguments);
    }
    function fe(e, r) {
        if (null == e) return {};
        var n, t, o = se(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function se(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function de(e) {
        var r = e.children, n = e.href, o = fe(e, [ "children", "href" ]);
        // @docs-props-type MarkdownLinkPropsBase
                
        return t().createElement(ce(), ue({
            to: n
        }, o), r);
    }
    /* harmony default export */    const pe = de;
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownList.tsx
    function ve() {
        return ve = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, ve.apply(null, arguments);
    }
    function ye(e, r) {
        if (null == e) return {};
        var n, t, o = be(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function be(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function me(e) {
        var r = e.children, n = e.ordered, o = ye(e, [ "children", "ordered" ]);
        // @docs-props-type MarkdownListPropsBase
                
        return t().createElement(ne(), ve({}, o, {
            ordered: n
        }), r);
    }
    /* harmony default export */    const Oe = me;
    // CONCATENATED MODULE: external "@splunk/react-ui/Paragraph"
    const ge = require("@splunk/react-ui/Paragraph");
    var he =  e.n(ge);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownParagraph.tsx
    function we(e, r) {
        if (null == e) return {};
        var n, t, o = Pe(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Pe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function je(e) {
        var r = e.children, n = we(e, [ "children" ]);
        // @docs-props-type MarkdownParagraphPropsBase
                
        return t().createElement(he(), n, r);
    }
    /* harmony default export */    const ke = je;
    // CONCATENATED MODULE: external "@splunk/react-ui/Table"
    const Se = require("@splunk/react-ui/Table");
    var Ee =  e.n(Se);
    // CONCATENATED MODULE: ./src/Markdown/renderers/MarkdownTable.tsx
    function xe() {
        return xe = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, xe.apply(null, arguments);
    }
    function Re(e, r) {
        if (null == e) return {};
        var n, t, o = qe(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function qe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function Ie(e) {
        var r = e.children, n = Re(e, [ "children" ]);
        
        return t().createElement(Ee(), n, r);
    }
    Ie.splunkUiType = "Table";
    function Me(e) {
        var r = e.children, n = Re(e, [ "children" ]);
        
        return t().createElement(Ee().Body, n, r);
    }
    Me.splunkUiType = "Table.Body";
    function Te(e) {
        var r = e.children, n = e.style, o = Re(e, [ "children", "style" ]);
        
        return t().createElement(Ee().Cell, xe({}, o, {
            align: n === null || n === void 0 ? void 0 : n.textAlign
        }), r);
    }
    Te.splunkUiType = "Table.Cell";
    function Ce(e) {
        var r;
        return (e === null || e === void 0 ? void 0 : (r = e.props) === null || r === void 0 ? void 0 : r.children) != null;
    }
    // `Table.Head` automatically adds a `tr`, so the one generated by `react-markdown` needs to be thrown out
        function _e(e) {
        var r = e.children, n = Re(e, [ "children" ]);
        // We can assume there is only one tr inside of thead as multiple trs are not supported in Markdown tables
                var o = r[0];
        // The types from react-markdown are in conflict here.
        // react-markdown expects the `thead` component to take React.ReactNode as children to be a valid custom component,
        // but will actually be passed react components from our custom `th` components, which also have children props
                var l;
        if (Ce(o)) {
            l = o.props.children;
        }
        
        return t().createElement(Ee().Head, n, l);
    }
    _e.splunkUiType = "Table.Head";
    function Ae(e) {
        var r = e.children, n = e.style, o = Re(e, [ "children", "style" ]);
        
        return t().createElement(Ee().HeadCell, xe({}, o, {
            align: n === null || n === void 0 ? void 0 : n.textAlign
        }), r);
    }
    Ae.splunkUiType = "Table.HeadCell";
    function Be(e) {
        var r = e.children, n = Re(e, [ "children" ]);
        
        return t().createElement(Ee().Row, n, r);
    }
    Be.splunkUiType = "Table.Row";
    // CONCATENATED MODULE: ./src/Markdown/Markdown.tsx
    function He(e) {
        "@babel/helpers - typeof";
        return He = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, He(e);
    }
    function Ue() {
        return Ue = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Ue.apply(null, arguments);
    }
    function Le(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function Ne(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? Le(Object(n), !0).forEach((function(r) {
                De(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : Le(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function De(e, r, n) {
        return (r = Fe(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function Fe(e) {
        var r = ze(e, "string");
        return "symbol" == He(r) ? r : r + "";
    }
    function ze(e, r) {
        if ("object" != He(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != He(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function $e(e, r) {
        if (null == e) return {};
        var n, t, o = Ge(e, r);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                n = l[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Ge(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var Je = {
        blockquoteRenderer: l().func,
        codeBlockRenderer: l().func,
        codeRenderer: l().func,
        elementRef: l().oneOfType([ l().func, l().object ]),
        headingRenderer: l().func,
        imageRenderer: l().func,
        itemRenderer: l().func,
        linkRenderer: l().func,
        listRenderer: l().func,
        paragraphRenderer: l().func,
        text: l().string.isRequired,
        rendererOptions: l().object
    };
    // <pre /> is used to wrap <code /> blocks, but Code already adds a pre tag
    // This custom renderer, removes the default tag from react-markdown to avoid nested pre tags
        var Ke = function e(r) {
        // eslint-disable-next-line react/jsx-no-useless-fragment, react/prop-types
        
        return t().createElement(t().Fragment, null, r.children);
    };
    var Qe = [ u() ];
    /**
 * The `Markdown` component renders the given Markdown text as a React component.
 * The component prefers @splunk/react-ui components over plain HTML components. For example
 * links are rendered as the `@splunk/react-ui/Link` component instead of plain `<a>` tag.
 */    function Ve(e) {
        var r = e.blockquoteRenderer, o = e.codeBlockRenderer, l = e.codeRenderer, i = e.elementRef, c = e.headingRenderer, u = e.imageRenderer, s = e.itemRenderer, d = e.linkRenderer, v = e.listRenderer, y = e.paragraphRenderer, b = e.text, m = e.rendererOptions, O = $e(e, [ "blockquoteRenderer", "codeBlockRenderer", "codeRenderer", "elementRef", "headingRenderer", "imageRenderer", "itemRenderer", "linkRenderer", "listRenderer", "paragraphRenderer", "text", "rendererOptions" ]);
 // @docs-props-type MarkdownPropsBase
                var g = (0, n.useCallback)((function(e) {
            var r, n;
            var i = e.className, a = i === void 0 ? "" : i, c = e.node;
 // react-markdown does not indicate if the element is inline or a codeblock
            // we have to infer this by checking if the node's position
            // is on the same line or across multiple lines
                        var u = c === null || c === void 0 ? void 0 : (r = c.position) === null || r === void 0 ? void 0 : r.start.line;
            var f = c === null || c === void 0 ? void 0 : (n = c.position) === null || n === void 0 ? void 0 : n.end.line;
            var s = u !== undefined && f !== undefined && u === f;
            if (s) {
                return l ? l(Ne({}, e)) :  t().createElement(A, e);
            }
            // Languages are passed from `react-markdown` via `classname`
            // Follows react-markdown's README recommendation for handling syntax highlighting
            // https://github.com/remarkjs/react-markdown?tab=readme-ov-file#use-custom-components-syntax-highlight
                        var d = (a === null || a === void 0 ? void 0 : a.replace("language-", "")) || undefined;
            return o ? o(Ne({
                language: d
            }, e)) :  t().createElement(F, Ue({
                language: d
            }, e));
        }), [ o, l ]);
        var h = (0, n.useMemo)((function() {
            return {
                a: d || pe,
                blockquote: r || M,
                code: g,
                h1: c || V,
                h2: c || V,
                h3: c || V,
                h4: c || V,
                h5: c || V,
                h6: c || V,
                img: u || ee,
                li: s || ie,
                ol: v || Oe,
                p: y || ke,
                pre: Ke,
                ul: v || Oe,
                table: Ie,
                thead: _e,
                th: Ae,
                td: Te,
                tr: Be,
                tbody: Me
            };
        }), [ r, g, c, u, s, d, v, y ]);
        var w = (0, n.useMemo)((function() {
            // prettier-ignore
            return {
                pre: (0, f.css)([ ":is(", ")" ], S),
                blockquote: (0, f.css)([ ":is(", ")" ], x)
            };
        }), []);
        
        return t().createElement(p(), Ue({
            "data-test": "markdown",
            elementRef: i,
            elementSelectors: w
        }, O),  t().createElement(a(), Ue({
            remarkPlugins: Qe,
            components: h
        }, m), b));
    }
    Ve.propTypes = Je;
    /* harmony default export */    const We = Ve;
    // CONCATENATED MODULE: ./src/Markdown/index.ts
    module.exports = r;
    /******/})();
