// @splunk/react-ui/Menu.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Divider: () => /* reexport */ $,
        Heading: () => /* reexport */ Z,
        Item: () => /* reexport */ _e,
        MenuContext: () => /* reexport */ T,
        default: () => /* reexport */ Ve
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "lodash/omit"
    const i = require("lodash/omit");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const s = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const d = require("@splunk/react-ui/Divider");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/react-ui/Checkbox"
    const p = require("@splunk/react-ui/Checkbox");
    var b =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const m = require("@splunk/react-ui/Clickable");
    var y =  e.n(m);
    // CONCATENATED MODULE: ./src/Menu/ItemStyles.ts
    var g = (0, c.css)([ "", " overflow:inherit;white-space:inherit;text-overflow:inherit;" ], v.mixins.typography("smallBody"));
    var h = u().div.withConfig({
        displayName: "ItemStyles__StyledItemDescriptionBottom",
        componentId: "sc-4kc053-0"
    })([ "", ";" ], g);
    var S = u().span.withConfig({
        displayName: "ItemStyles__StyledItemDescriptionRight",
        componentId: "sc-4kc053-1"
    })([ "", ";display:flex;align-self:flex-start;align-items:center;float:right;padding-left:", ";max-width:50%;min-height:20px;text-align:right;" ], g, v.variables.spacingLarge);
    var O = (0, c.css)([ "", " flex:0 0 auto;& > svg{height:20px;}" ], v.mixins.reset("inline-flex"));
    var w = u().div.withConfig({
        displayName: "ItemStyles__StyledItemSelectedAdornment",
        componentId: "sc-4kc053-2"
    })([ "", ";padding-left:", ";align-items:flex-start;color:", ";" ], O, v.variables.spacingLarge, v.variables.contentColorActive);
    var k = u().span.withConfig({
        displayName: "ItemStyles__StyledItemAdornment",
        componentId: "sc-4kc053-3"
    })([ "", ";", " padding-right:", ";align-items:center;color:", ";" ], O, (function(e) {
        var t = e.endAdornment;
        return t && (0, c.css)([ "padding-left:", ";" ], v.variables.spacingSmall);
    }), v.variables.spacingSmall, v.variables.contentColorMuted);
    var x = u()(b()).withConfig({
        displayName: "ItemStyles__StyledCheckbox",
        componentId: "sc-4kc053-4"
    })([ "min-height:0;padding-right:", ";" ], v.variables.spacingSmall);
    var C = u()(y()).withConfig({
        displayName: "ItemStyles__StyledClickable",
        componentId: "sc-4kc053-5"
    })([ "line-height:", ";display:block;position:relative;cursor:pointer;color:", ";word-wrap:break-word;max-width:100%;width:100%;min-height:", ";padding:", " ", ";", " ", " ", " &:not([disabled],[aria-disabled='true']){&:hover{background:", ";", "}&:active{background:", ";}", "}&:focus{outline:0;box-shadow:", ";background:", ";}&[disabled],&[aria-disabled='true']{color:", ";", ",", ",", ",", "{color:", ";}}" ], v.variables.lineHeightSpacious, v.variables.contentColorActive, v.variables.inputHeight, v.variables.spacingXSmall, v.variables.spacingLarge, (function(e) {
        var t = e.$active, r = e.$preventFocus;
        return t && !r && (0, c.css)([ "box-shadow:", ";" ], v.variables.focusShadowInset);
    }), (function(e) {
        var t = e.$active;
        return t && (0, c.css)([ "background:", ";font-weight:", ";" ], v.variables.interactiveColorOverlayActive, v.variables.fontWeightBold);
    }), (function(e) {
        var t = e.$isSelectable, r = e.$selectableAppearance, n = e.$selected;
        return t && !n && r === "checkmark" && // Adds whitespace to right to reserve space for checkmark for selectable Items
        // spacingLarge x 3 = padding-left + icon size + padding-right
        (0, c.css)([ "padding-right:calc(", " * 3);" ], v.variables.spacingLarge);
    }), v.variables.interactiveColorOverlayHover, (function(e) {
        var t = e.$active;
        return t && (0, c.css)([ "background:", ";" ], v.mixins.overlayColors(v.variables.interactiveColorOverlayActive, v.variables.interactiveColorOverlayHover));
    }), v.variables.interactiveColorOverlayActive, (function(e) {
        var t = e.$selected, r = e.$active;
        return t && (0, c.css)([ "font-weight:", ";", "" ], v.variables.fontWeightSemiBold, r && (0, 
        c.css)([ "font-weight:", ";" ], v.variables.fontWeightBold));
    }), v.variables.focusShadowInset, v.variables.interactiveColorOverlayHover, v.variables.contentColorDisabled, /* sc-sel */ h, /* sc-sel */ S, /* sc-sel */ w, /* sc-sel */ k, v.variables.contentColorDisabled);
    var j = u().span.withConfig({
        displayName: "ItemStyles__StyledInnerWrapper",
        componentId: "sc-4kc053-6"
    })([ "display:flex;align-items:flex-start;" ]);
    var I = u().div.withConfig({
        displayName: "ItemStyles__StyledTitleAndDescriptionWrapper",
        componentId: "sc-4kc053-7"
    })([ "display:flex;flex:1;width:100%;align-self:flex-start;word-break:break-word;white-space:normal;", " ", "" ], (function(e) {
        var t = e.$descriptionBottom;
        return t && (0, c.css)([ "flex-direction:column;" ]);
    }), (function(e) {
        var t = e.$truncate;
        return t && (0, c.css)([ "white-space:nowrap;overflow:hidden;text-overflow:ellipsis;" ]);
    }));
    var P = u().div.withConfig({
        displayName: "ItemStyles__StyledLabel",
        componentId: "sc-4kc053-8"
    })([ "flex:1;overflow:inherit;white-space:inherit;text-overflow:inherit;max-width:100%;min-height:20px;", "" ], (function(e) {
        var t = e.$truncate;
        return t && (0, c.css)([ "display:block;clear:both;" ]);
    }));
    var E = u().span.withConfig({
        displayName: "ItemStyles__StyledMatch",
        componentId: "sc-4kc053-9"
    })([ "border-bottom:1px solid ", ";" ], v.variables.contentColorActive);
    var _ = u().span.withConfig({
        displayName: "ItemStyles__StyledSubmenu",
        componentId: "sc-4kc053-10"
    })([ "float:right;padding-left:", ";color:", ";" ], v.variables.spacingSmall, v.variables.contentColorDefault);
    // CONCATENATED MODULE: ./src/Menu/DividerStyles.ts
    // Adds margin if direct sibling is a Menu.Item to provide spacing between Divider and Item
    var D = u()(f()).withConfig({
        displayName: "DividerStyles__Styled",
        componentId: "l3zfh3-0"
    })([ "", " + &,& + ", "{margin-block-start:", ";}" ], C, C, v.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/Menu/MenuContext.tsx
    var A =  (0, r.createContext)({
        role: "menu",
        preventFocus: false
    });
    /* harmony default export */    const T = A;
    // CONCATENATED MODULE: ./src/Menu/Divider.tsx
    function N() {
        return N = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, N.apply(null, arguments);
    }
    var R = {};
    /**
 * A non-interactive menu item used to visually separate groups of items in the menu.
 */    function q(e) {
        var t = N({}, e);
        var i = (0, r.useContext)(T), a = i.role;
        
        return n().createElement(D, N({
            "aria-hidden": a === "listbox",
            "data-test": "divider",
            role: "separator"
        }, t));
    }
    q.propTypes = R;
    /* Remove the item if it is the first item after filtering */    q.filterFirst = true;
    /* Remove consecutive items with filterConsecutive = true (Dividers and Headings) */    q.filterConsecutive = true;
    /* Remove the item if it is the last item after filtering. */    q.filterLast = true;
    /* Tag the item as Divider */    q.as = "Divider";
    /* harmony default export */    const $ = q;
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const H = require("@splunk/react-ui/Heading");
    var M =  e.n(H);
    // CONCATENATED MODULE: ./src/Menu/HeadingStyles.ts
    var F = u().div.withConfig({
        displayName: "HeadingStyles__Styled",
        componentId: "mcd2ws-0"
    })([ "", " border-top:1px solid transparent;padding:", " ", ";&:not(:first-child){border-top:1px solid ", ";margin-top:", ";}" ], v.mixins.reset("block"), v.variables.spacingSmall, v.variables.spacingLarge, v.variables.borderColor, v.variables.spacingSmall);
    var L = u()(M()).withConfig({
        displayName: "HeadingStyles__StyledHeading",
        componentId: "mcd2ws-1"
    })([ "color:", ";" ], v.variables.contentColorDefault);
    // CONCATENATED MODULE: ./src/Menu/Heading.tsx
    function B(e) {
        "@babel/helpers - typeof";
        return B = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, B(e);
    }
    function K() {
        return K = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, K.apply(null, arguments);
    }
    function V(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function W(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? V(Object(r), !0).forEach((function(t) {
                z(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : V(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function z(e, t, r) {
        return (t = X(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function X(e) {
        var t = G(e, "string");
        return "symbol" == B(t) ? t : t + "";
    }
    function G(e, t) {
        if ("object" != B(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != B(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function J(e, t) {
        if (null == e) return {};
        var r, n, i = Q(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function Q(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var U = {
        children: l().node,
        title: l().bool,
        outerStyle: l().object
    };
    /**
 * A non-interactive `Menu` item used to separate and label groups of `Menu` items.
 */    function Y(e) {
        var t = e.children, i = e.outerStyle, a = e.title, o = J(e, [ "children", "outerStyle", "title" ]);
        // @docs-props-type HeadingPropsBase
                var l = (0, r.useContext)(T), s = l.role;
        var c = (0, r.useMemo)((function() {
            return W(W({}, i), {}, {
                margin: 0
            });
        }), [ i ]);
        // headings are not supported in role="listbox"
        // this causes Firefox to incorrectly read the total number
        // of items in the list if it contains a heading or divider
        // so we set them to aria-hidden to prevent that
                
        return n().createElement(F, {
            "aria-hidden": s === "listbox" ? true : undefined
        },  n().createElement(L, K({
            style: c,
            level: a ? 4 : 5,
            variant: a ? "title5" : "title6",
            "data-test": "heading"
        }, o), t));
    }
    Y.propTypes = U;
    /* Remove consecutive items with filterConsecutive = true (Dividers and Headings) */    Y.filterConsecutive = true;
    /* Remove the item if it is the last item after filtering. */    Y.filterLast = true;
    /* Tag the item as a Heading */    Y.as = "Heading";
    /* harmony default export */    const Z = Y;
    // CONCATENATED MODULE: external "lodash/isString"
    const ee = require("lodash/isString");
    var te =  e.n(ee);
    // CONCATENATED MODULE: external "@splunk/react-icons/Checkmark"
    const re = require("@splunk/react-icons/Checkmark");
    var ne =  e.n(re);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const ie = require("@splunk/react-icons/ChevronRight");
    var ae =  e.n(ie);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const oe = require("@splunk/react-ui/ScreenReaderContent");
    var le =  e.n(oe);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const se = require("@splunk/react-ui/Tooltip");
    var ce =  e.n(se);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const ue = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowSquareTopRightInset"
    const de = require("@splunk/react-icons/ArrowSquareTopRightInset");
    var fe =  e.n(de);
    // CONCATENATED MODULE: ./src/Menu/icons/External.tsx
    var ve = u()(fe()).withConfig({
        displayName: "External__Styled",
        componentId: "sc-1turs2s-0"
    })([ "vertical-align:text-top;margin:0 0 0 4px;" ]);
    function pe() {
        
        return n().createElement(ve, null);
    }
    /* harmony default export */    const be = pe;
    // CONCATENATED MODULE: ./src/utils/disabled.ts
    /**
 * Normalizes the disabled prop and returns both the boolean state and normalized string value.
 * Use this function to ensure consistent handling of disabled prop across components.
 *
 * @param disabled - The disabled prop value
 * @param defaultWhenBooleanTrue - Optional normalization override for disabled={true}
 * @returns Object with isDisabled boolean|undefined and disabledValue string
 *
 * TODO (SUI-8600): `disabled` prop should default to 'dimmed' for data entry components
 */
    function me(e) {
        var t = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "dimmed";
        if (e === "dimmed") {
            return {
                isDisabled: true,
                disabledValue: "dimmed"
            };
        }
        if (e === true) {
            return {
                isDisabled: true,
                disabledValue: t
            };
        }
        if (e === "disabled") {
            return {
                isDisabled: true,
                disabledValue: "disabled"
            };
        }
        if (e === false) {
            return {
                isDisabled: false,
                disabledValue: undefined
            };
        }
        return {
            isDisabled: undefined,
            disabledValue: undefined
        };
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function ye(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Menu/Item.tsx
    function ge(e) {
        "@babel/helpers - typeof";
        return ge = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, ge(e);
    }
    function he(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function Se(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? he(Object(r), !0).forEach((function(t) {
                Oe(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : he(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function Oe(e, t, r) {
        return (t = we(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function we(e) {
        var t = ke(e, "string");
        return "symbol" == ge(t) ? t : t + "";
    }
    function ke(e, t) {
        if ("object" != ge(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != ge(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function xe() {
        return xe = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, xe.apply(null, arguments);
    }
    function Ce(e, t) {
        if (null == e) return {};
        var r, n, i = je(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function je(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var Ie = {
        active: l().bool,
        children: l().node,
        description: l().string,
        descriptionPosition: l().oneOf([ "right", "bottom" ]),
        disabled: l().oneOfType([ l().bool, l().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: l().oneOfType([ l().func, l().object ]),
        endAdornment: l().node,
        hasSubmenu: l().bool,
        /** @private. The internal key of the tab */
        itemKey: l().number,
        matchRanges: l().arrayOf(l().shape({
            start: l().number.isRequired,
            end: l().number.isRequired
        })),
        onClick: l().func,
        /** @private */
        onFocus: l().func,
        openInNewContext: l().oneOfType([ l().bool, l().string ]),
        /** @private */
        preventFocus: l().bool,
        role: l().oneOf([ "menuitem", "menuitemradio", "menuitemcheckbox", "listboxitem", "link", "option" ]),
        selectable: l().bool,
        selectableAppearance: l().oneOf([ "checkmark", "checkbox" ]),
        selected: l().oneOfType([ l().bool, l().oneOf([ "some" ]) ]),
        startAdornment: l().node,
        to: l().string,
        truncate: l().bool
    };
    var Pe = (0, ue._)("(Opens new window)");
    function Ee(e) {
        var t;
        var i = e.active, o = e.children, l = e.description, s = e.descriptionPosition, c = s === void 0 ? "bottom" : s, u = e.disabled, d = e.endAdornment, f = e.elementRef, v = e.hasSubmenu, p = e.itemKey, b = e.matchRanges, m = e.onClick, y = e.onFocus, g = e.openInNewContext, O = e.role, D = e.selectable, A = e.selectableAppearance, N = A === void 0 ? "checkmark" : A, R = e.selected, q = e.startAdornment, $ = e.to, H = e.truncate, M = Ce(e, [ "active", "children", "description", "descriptionPosition", "disabled", "endAdornment", "elementRef", "hasSubmenu", "itemKey", "matchRanges", "onClick", "onFocus", "openInNewContext", "role", "selectable", "selectableAppearance", "selected", "startAdornment", "to", "truncate" ]);
        // @docs-props-type ItemPropsBase
                var F = (0, r.useContext)(T), L = F.preventFocus;
 // Use arguments[0] (raw props) to check if `selectableAppearance` was explicitly passed by the caller.
        // This prevents the default value from incorrectly triggering the dev warning.
        // eslint-disable-next-line prefer-rest-params
                var B = (t = arguments[0]) === null || t === void 0 ? void 0 : t.selectableAppearance;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ H, c, b, o, R, D, N, B ]);
        var K = (0, r.useCallback)((function(e) {
            y === null || y === void 0 ? void 0 : y(e, {
                itemKey: p
            });
        }), [ p, y ]);
        var V = (0, r.useCallback)((function(e) {
            if (L) {
                e.preventDefault();
            }
        }), [ L ]);
        var W = (0, r.useMemo)((function() {
            if (!b || !te()(o)) {
                return o;
            }
            var e = [];
            // before first match. May be empty string.
                        e.push(o.substring(0, b[0].start));
            b.forEach((function(t, r) {
                e.push(
                
                // eslint-disable-next-line react/no-array-index-key
                n().createElement(E, {
                    key: r,
                    "data-test": "match"
                }, o.substring(t.start, t.end)));
                if (r < b.length - 1) {
                    e.push(o.substring(t.end, b[r + 1].start));
                } else {
                    e.push(o.substring(t.end, o.length));
                }
            }));
            return e;
        }), [ o, b ]);
        var z = R === true || R === "some";
        var X = D || z;
        var G = me(u), J = G.isDisabled, Q = G.disabledValue;
        var U = O || {
            nonselectable: "menuitem",
            checkmark: "menuitemradio",
            checkbox: "menuitemcheckbox"
        }[X ? N : "nonselectable"];
        var Y = (0, r.useMemo)((function() {
            return {
                "aria-haspopup": v ? true : undefined
            };
        }), [ v ]);
        if (U === "menuitemradio" || U === "menuitemcheckbox") {
            Y["aria-checked"] = R === "some" ? "mixed" : z;
        } else if (U === "option") {
            Y["aria-selected"] = z;
        }
        var Z = l && c === "right";
        var ee = l && !Z;
        var re = false;
        var ie;
        if (g) {
            re = true;
            ie = typeof g === "string" ? g : Pe;
        }
        var oe = (0, r.useMemo)((function() {
            
            return n().createElement(C, xe({
                $active: i,
                $isSelectable: X,
                $preventFocus: L || false,
                $selectableAppearance: N !== null && N !== void 0 ? N : false,
                $selected: z,
                "data-has-icon": !!q || !!d,
                "data-test": "item",
                "data-test-disabled": Q,
                "data-test-selected": X ? z : null,
                disabled: u,
                elementRef: f,
                onClick: m,
                onFocus: K,
                onMouseDown: V,
                openInNewContext: re,
                role: U,
                tabIndex: L ? -1 : undefined,
                to: $
            }, Y, a()(M, "onFocus")), v &&  n().createElement(_, null,  n().createElement(ae(), null)),  n().createElement(j, null, D && N === "checkbox" &&  n().createElement(x, {
                checked: R === "some" ? "indeterminate" : R,
                disabled: J,
                inert: true
            }), q &&  n().createElement(k, null, q),  n().createElement(I, {
                $descriptionBottom: !!ee,
                $truncate: H !== null && H !== void 0 ? H : false
            },  n().createElement(P, {
                $truncate: H !== null && H !== void 0 ? H : false,
                "data-test": "label"
            }, W, g &&  n().createElement(n().Fragment, null,  n().createElement(be, null),  n().createElement(le(), null, ie))), ee &&  n().createElement(h, {
                "data-test": "description"
            }, l), Z &&  n().createElement(S, {
                "data-test": "description"
            }, l)), d &&  n().createElement(k, {
                endAdornment: d
            }, d), R && N === "checkmark" &&  n().createElement(w, null,  n().createElement(ne(), {
                "data-test": "selected-checkmark",
                inline: true
            }))));
        }), [ i, X, L, N, z, q, d, Q, u, f, m, K, V, re, U, $, Y, M, v, D, R, J, ee, H, W, g, ie, l, Z ]);
        var se = H && te()(o);
        var ue = (0, r.useCallback)((function(e) {
            var t = function t(r) {
                ye(e.elementRef, r);
                ye(oe.props.elementRef, r);
            };
            // Merge the anchorProps from Tooltip with the props of the clickable element to ensure consistent behavior for screen readers
                        var r =  n().cloneElement(oe, Se(Se(Se({}, "aria-describedby" in e ? {
                "aria-describedby": e["aria-describedby"]
            } : {}), "aria-labelledby" in e ? {
                "aria-labelledby": e["aria-labelledby"]
            } : {}), {}, {
                onFocus: function t(r) {
                    var n, i, a;
                    (n = e.onFocus) === null || n === void 0 ? void 0 : n.call(e, r);
                    (i = (a = oe.props).onFocus) === null || i === void 0 ? void 0 : i.call(a, r);
                },
                onBlur: function t(r) {
                    var n, i, a;
                    (n = e.onBlur) === null || n === void 0 ? void 0 : n.call(e, r);
                    (i = (a = oe.props).onBlur) === null || i === void 0 ? void 0 : i.call(a, r);
                },
                onClick: function t(r) {
                    var n, i, a;
                    (n = e.onClick) === null || n === void 0 ? void 0 : n.call(e, r);
                    (i = (a = oe.props).onClick) === null || i === void 0 ? void 0 : i.call(a, r);
                },
                elementRef: t
            }));
            return r;
        }), [ oe ]);
        if (se) {
            
            return n().createElement(ce(), {
                content: o,
                defaultPlacement: "right",
                inline: false,
                contentRelationship: "label",
                renderAnchor: ue
            });
        }
        return oe;
    }
    Ee.propTypes = Ie;
    Ee.as = "Item";
    /* harmony default export */    const _e = Ee;
    // only for styled components using Item
        // CONCATENATED MODULE: external "@splunk/react-ui/Scroll"
    const De = require("@splunk/react-ui/Scroll");
    var Ae =  e.n(De);
    // CONCATENATED MODULE: ./src/Menu/MenuStyles.ts
    var Te = (0, c.css)([ "border-top:1px solid ", ";" ], v.variables.borderColor);
    var Ne = (0, c.css)([ "", ";background-color:", ";border-radius:", ";min-width:60px;overflow:auto;position:relative;padding:", " 0;&:focus{outline:0;box-shadow:", ";background:", ";}& + &{", "}" ], v.mixins.reset("block"), v.variables.backgroundColorPopup, v.variables.borderRadius, v.variables.spacingSmall, v.variables.focusShadowInset, v.variables.interactiveColorOverlayHover, Te);
    var Re = u().div.withConfig({
        displayName: "MenuStyles__Styled",
        componentId: "sc-1olffp9-0"
    })([ "", "" ], Ne);
    var qe = u()(Ae()).withConfig({
        displayName: "MenuStyles__StyledScroll",
        componentId: "sc-1olffp9-1"
    })([ "", " ", " + &,& + ", "{", "}" ], Ne, /* sc-sel */ Re, /* sc-sel */ Re, Te);
    // CONCATENATED MODULE: ./src/Menu/Menu.tsx
    function $e() {
        return $e = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, $e.apply(null, arguments);
    }
    function He(e, t) {
        if (null == e) return {};
        var r, n, i = Me(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function Me(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var Fe = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ]),
        focusMode: l().oneOf([ "roving", "normal", "never" ]),
        stopScrollPropagation: l().bool
    };
    // the default focus control for Menu: loop, verticalArrows(up/down keys), home/end keys
        var Le = function e() {
        return {
            enableLoop: true,
            orientation: "vertical",
            enableTab: false,
            enableHomeEnd: true
        };
    };
    var Be = function e(t) {
        var r = t.key;
 // Handle special case for tab navigation
                if (r === "Tab" && (0, s.isTabKey)(t.nativeEvent)) {
            return t.shiftKey ? "tabShift" : "Tab";
        }
        return r;
    };
    function Ke(e) {
        var t = e.children, i = e.elementRef, o = e.focusMode, l = o === void 0 ? "roving" : o, c = e.stopScrollPropagation, u = He(e, [ "children", "elementRef", "focusMode", "stopScrollPropagation" ]);
        // @docs-props-type MenuPropsBase
                var d = (0, r.useContext)(T), f = d.role, v = f === void 0 ? "menu" : f;
        var p = (0, r.useRef)(null);
        var b = r.Children.toArray(t).filter(r.isValidElement).reduce((function(e, t, r, n) {
            /* Filter out initial Dividers
     * Requires reduce() over filter() because a Heading may have been
     * before the Divider.
     */
            if (t.type.as === "Divider" && e.length === 0) {
                return e;
            }
            // Filter out consecutive Dividers and Headings
            // If Divider is next to Heading then it will remove both Heading and Divider unless Heading has title prop set to it
                        if ((t.type.as === "Heading" || t.type.as === "Divider") && !t.props.title && t.type.filterConsecutive && n.length > r + 1) {
                var i = n[r + 1];
                if (i.type.as === "Heading" || i.type.as === "Divider") {
                    return e;
                }
            }
            // Filter out last Dividers and Headings
                        if ((t.type.as === "Divider" || t.type.as === "Heading") && r === n.length - 1) {
                return e;
            }
            e.push(t);
            return e;
        }), []);
        var m = (0, r.useCallback)((function(e) {
            p.current = e;
            ye(i, e);
        }), [ i ]);
        var y = Le();
        var g = (0, r.useCallback)((function(e) {
            // if command key on Mac was pressed ignore
            if (e.nativeEvent.metaKey) {
                return;
            }
            var t = Be(e);
            var r = t === "Tab" || t === "tabShift";
            var n = t === "Enter";
            // Allow default behavior for the enter key.
            // If tab navigation is enabled, prevent default behavior for all other keys.
            // If tab navigation is disabled, allow default behavior for tab and tab+shift key
                        if (!n && (y.enableTab || !r)) {
                e.preventDefault();
            }
            var i = p.current;
            var a = [];
            if (i) {
                a = (0, s.getSortedTabbableElements)(i, {
                    ignoreTabIndex: true
                });
            }
            var o = e.target instanceof HTMLElement ? e.target : i === null || i === void 0 ? void 0 : i.querySelector(":focus");
            var l = o instanceof HTMLElement ? a.indexOf(o) : -1;
            (0, s.handleFocus)(t, a, l, y);
        }), [ y ]);
        var h = function e() {
            return {
                role: "menu",
                preventFocus: l === "never"
            };
        };
        var S = l === "normal" || l === "never" ? undefined : g;
        var O = {
            "data-test": "menu",
            onKeyDown: S,
            role: v
        };
        return c ?  n().createElement(qe, $e({}, O, {
            elementRef: m,
            stopScrollPropagation: true
        }, a()(u, "tagName")), b) :  n().createElement(Re, $e({}, O, {
            ref: m
        }, u),  n().createElement(T.Provider, {
            value: h()
        }, b));
    }
    Ke.propTypes = Fe;
    Ke.Item = _e;
    Ke.Divider = $;
    Ke.Heading = Z;
    /* harmony default export */    const Ve = Ke;
    // CONCATENATED MODULE: ./src/Menu/index.ts
    module.exports = t;
    /******/})();
