// @splunk/react-ui/MessageBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ B
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckCircle"
    const o = require("@splunk/react-icons/CheckCircle");
    var s =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const l = require("@splunk/react-icons/Cross");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationTriangle"
    const u = require("@splunk/react-icons/ExclamationTriangle");
    var p =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const d = require("@splunk/react-icons/InformationCircle");
    var f =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationDiamond"
    const v = require("@splunk/react-icons/InformationDiamond");
    var g =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const m = require("@splunk/react-ui/ScreenReaderContent");
    var y =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const b = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const C = require("styled-components");
    var h =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const w = require("@splunk/react-ui/Button");
    var S =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/themes"
    const k = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/MessageBar/MessageBarStyles.ts
    var O = h().div.withConfig({
        displayName: "MessageBarStyles__MessageBarContent",
        componentId: "sc-1uyhzza-0"
    })([ "", ";", ";grid-column:content;grid-row:1 / -1;max-width:", ";word-wrap:break-word;" ], k.mixins.reset("inline"), k.mixins.typography("body"), k.variables.lineLength);
    var q = h()(S()).withConfig({
        displayName: "MessageBarStyles__StyledCloseButton",
        componentId: "sc-1uyhzza-1"
    })([ "grid-column:close;" ]);
    var _ = h().span.withConfig({
        displayName: "MessageBarStyles__StyledIcon",
        componentId: "sc-1uyhzza-2"
    })([ "width:20px;height:20px;margin-left:auto;fill:", ";grid-row:1;grid-column:icon;" ], (0, 
    k.pickVariant)("$type", {
        info: k.variables.notificationColorInfo,
        success: k.variables.notificationColorPositive,
        warning: k.variables.notificationColorCaution,
        error: k.variables.notificationColorNegative
    }));
    var x = h().div.withConfig({
        displayName: "MessageBarStyles__MessageBarWrapper",
        componentId: "sc-1uyhzza-3"
    })([ "", ";grid-template-columns:[icon] 20px ", " [content] fit-content(", ") minmax(", ",1fr) [close] min-content;grid-template-rows:1lh auto;align-items:center;border:1px solid ", ";border-radius:", ";background-color:", ";padding:", " ", ";width:100%;" ], k.mixins.reset("grid"), k.variables.spacingSmall, k.variables.lineLength, k.variables.spacingSmall, (0, 
    k.pickVariant)("$type", {
        info: k.variables.notificationColorInfoStrong,
        success: k.variables.notificationColorPositiveStrong,
        warning: k.variables.notificationColorCautionStrong,
        error: k.variables.notificationColorNegativeStrong
    }), k.variables.borderRadius, (0, k.pickVariant)("$type", {
        info: k.variables.notificationColorInfoWeak,
        success: k.variables.notificationColorPositiveWeak,
        warning: k.variables.notificationColorCautionWeak,
        error: k.variables.notificationColorNegativeWeak
    }), k.variables.spacingSmall, k.variables.spacingMedium);
    // CONCATENATED MODULE: ./src/MessageBar/MessageBar.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function P(e, r) {
        if (null == e) return {};
        var n, t, a = j(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                n = i[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function j(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var z = {
        children: i().node.isRequired,
        onRequestClose: i().func,
        type: i().oneOf([ "info", "success", "warning", "error" ]).isRequired
    };
    var M = Object.freeze({
        info: (0, b._)("Info"),
        warning: (0, b._)("Warning"),
        error: (0, b._)("Alert"),
        success: (0, b._)("Success")
    });
    var E = Object.freeze({
        info: f(),
        warning: g(),
        error: p(),
        success: s()
    });
    function R(e) {
        var r = e.children, n = e.elementRef, a = e.onRequestClose, i = e.type, o = P(e, [ "children", "elementRef", "onRequestClose", "type" ]);
        // @docs-props-type MessageBarPropsBase
                var s = E[i];
        var l = M[i];
        
        return t().createElement(x, I({
            ref: n,
            $type: i,
            "data-test": "message-bar",
            "data-test-type": i
        }, o, {
            role: "region"
        }),  t().createElement(_, {
            as: s,
            $type: i,
            variant: "filled",
            "aria-hidden": true,
            "data-test": "icon"
        }),  t().createElement(y(), null, l),  t().createElement(O, {
            "data-test": "content"
        }, r), a &&  t().createElement(q, {
            appearance: "subtle",
            onClick: a
        },  t().createElement(c(), null),  t().createElement(y(), null, (0, b._)("Close"))));
    }
    R.propTypes = z;
    /* harmony default export */    const B = R;
    // CONCATENATED MODULE: ./src/MessageBar/index.ts
    module.exports = r;
    /******/})();
