// @splunk/react-ui/Message.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ V
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckCircle"
    const i = require("@splunk/react-icons/CheckCircle");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const s = require("@splunk/react-icons/Cross");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationTriangle"
    const u = require("@splunk/react-icons/ExclamationTriangle");
    var p =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const d = require("@splunk/react-icons/InformationCircle");
    var v =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationDiamond"
    const f = require("@splunk/react-icons/InformationDiamond");
    var m =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const g = require("@splunk/react-ui/ScreenReaderContent");
    var y =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const b = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const h = require("styled-components");
    var w =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const S = require("@splunk/react-ui/Button");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/themes"
    const x = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Message/MessageStyles.ts
    var C = {
        info: x.variables.notificationColorInfo,
        success: x.variables.notificationColorPositive,
        warning: x.variables.notificationColorCaution,
        error: x.variables.notificationColorNegative
    };
    var _ = w().div.withConfig({
        displayName: "MessageStyles__StyledTitle",
        componentId: "eg66af-0"
    })([ "", ";" ], x.mixins.typography("title4"));
    var k = w().div.withConfig({
        displayName: "MessageStyles__StyledContent",
        componentId: "eg66af-1"
    })([ "", ";grid-row:1 / -1;grid-column:content;max-width:", ";word-wrap:break-word;&:not(:has(> ", ")){", ";}" ], x.mixins.typography("body"), x.variables.lineLength, _, x.mixins.typography("body", {
        color: "active"
    }));
    var R = w().span.withConfig({
        displayName: "MessageStyles__StyledIcon",
        componentId: "eg66af-2"
    })([ "fill:", ";grid-row:1;height:20px;width:20px;" ], (0, x.pickVariant)("$type", C));
    var q = w()(O()).withConfig({
        displayName: "MessageStyles__StyledRemove",
        componentId: "eg66af-3"
    })([ "grid-column:remove;" ]);
    var I = w().div.withConfig({
        displayName: "MessageStyles__Styled",
        componentId: "eg66af-4"
    })([ "", ";grid-template-rows:1lh auto;align-items:center;padding:", ";", ";", "" ], x.mixins.reset("grid"), x.variables.spacingSmall, (function(e) {
        var n = e.$hasRemoveIcon;
        return (0, h.css)([ "grid-template-columns:[icon] 20px ", " [content] auto ", " [remove] min-content;" ], x.variables.spacingSmall, n ? x.variables.spacingSmall : "0");
    }), (function(e) {
        var n = e.$appearance, r = e.$hasRemoveIcon, t = e.$type;
        var a = C[t];
        return n === "fill" && (0, h.css)([ "background-color:", ";border-left:8px solid ", ";border-radius:", ";box-shadow:", ";grid-template-columns:[icon] 20px ", " [content] fit-content(", ");padding-inline:", ";", "" ], x.variables.backgroundColorPopup, a, x.variables.borderRadius, x.variables.embossShadow, x.variables.spacingSmall, x.variables.lineLength, x.variables.spacingMedium, r && (0, 
        h.css)([ "grid-template-columns:[icon] 20px ", " [content] fit-content(", ") minmax(", ",1fr) [remove] min-content;" ], x.variables.spacingSmall, x.variables.lineLength, x.variables.spacingSmall));
    }));
    // CONCATENATED MODULE: ./src/Message/Title.tsx
    function j(e, n) {
        if (null == e) return {};
        var r, t, a = P(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function P(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var E = {
        children: o().node
    };
    /**
 * A title component for use in `Message`.
 */    function M(e) {
        var n = e.children, r = j(e, [ "children" ]);
        // @docs-props-type TitleProps
                
        return t().createElement(_, r, n);
    }
    M.propTypes = E;
    /* harmony default export */    const $ = M;
    // CONCATENATED MODULE: ./src/Message/Message.tsx
    function T() {
        return T = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, T.apply(null, arguments);
    }
    function N(e, n) {
        if (null == e) return {};
        var r, t, a = L(e, n);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (t = 0; t < o.length; t++) {
                r = o[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function L(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var z = {
        appearance: o().oneOf([ "default", "fill" ]),
        children: o().node,
        elementRef: o().oneOfType([ o().func, o().object ]),
        onRequestRemove: o().func,
        type: o().oneOf([ "info", "warning", "error", "success" ])
    };
    var A = Object.freeze({
        info: v(),
        warning: m(),
        error: p(),
        success: l()
    });
    var B = Object.freeze({
        info: (0, b._)("Info"),
        warning: (0, b._)("Warning"),
        error: (0, b._)("Alert"),
        success: (0, b._)("Success")
    });
    var D =  t().createElement(c(), {
        height: "20px",
        width: "20px"
    });
    function F(e) {
        var n = e.appearance, a = n === void 0 ? "default" : n, o = e.children, i = e.elementRef, l = e.onRequestRemove, s = e.type, c = s === void 0 ? "warning" : s, u = N(e, [ "appearance", "children", "elementRef", "onRequestRemove", "type" ]);
        // @docs-props-type MessagePropsBase
                var p = (0, r.useCallback)((function(e) {
            l === null || l === void 0 ? void 0 : l(e);
        }), [ l ]);
        var d = A[c];
        var v = B[c];
        
        return t().createElement(I, T({
            ref: i,
            $appearance: a,
            $hasRemoveIcon: !!l,
            $type: c,
            "data-test-type": c,
            "data-test": "message"
        }, u),  t().createElement(t().Fragment, null,  t().createElement(R, {
            as: d,
            $type: c,
            variant: "filled"
        }),  t().createElement(y(), null, v)),  t().createElement(k, {
            "data-test": "content"
        }, o), l &&  t().createElement(q, {
            "aria-label": "Remove",
            "data-test": "remove",
            appearance: "subtle",
            onClick: p,
            icon: D
        }));
    }
    F.propTypes = z;
    F.Title = $;
    /* harmony default export */    const V = F;
    // CONCATENATED MODULE: ./src/Message/index.ts
    module.exports = n;
    /******/})();
