// @splunk/react-ui/Modal.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Body: () => /* reexport */ D,
        Footer: () => /* reexport */ V,
        Header: () => /* reexport */ ue,
        default: () => /* reexport */ _e
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "lodash/defer"
    const o = require("lodash/defer");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "lodash/memoize"
    const a = require("lodash/memoize");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/ModalLayer"
    const s = require("@splunk/react-ui/ModalLayer");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const d = require("@splunk/react-ui/usePrevious");
    var p =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const y = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const v = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/react-ui/ScrollContainerContext"
    const b = require("@splunk/react-ui/ScrollContainerContext");
    // CONCATENATED MODULE: external "styled-components"
    const m = require("styled-components");
    var g =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const h = require("@splunk/react-ui/Box");
    var O =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/themes"
    const S = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Modal/BodyStyles.ts
    var w = g()(O()).withConfig({
        displayName: "BodyStyles__StyledBox",
        componentId: "lv54z7-0"
    })([ "background-color:", ";padding:", " ", ";flex:0 1 auto;overflow:auto;" ], S.variables.backgroundColorDialog, S.variables.spacingMedium, S.variables.spacingXLarge);
    // CONCATENATED MODULE: ./src/Modal/Body.tsx
    function x() {
        return x = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, x.apply(null, arguments);
    }
    function j(e, t) {
        return I(e) || E(e, t) || k(e, t) || C();
    }
    function C() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function k(e, t) {
        if (e) {
            if ("string" == typeof e) return P(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? P(e, t) : void 0;
        }
    }
    function P(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function E(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (n = i.call(r)).done) && (l.push(n.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function I(e) {
        if (Array.isArray(e)) return e;
    }
    function q(e, t) {
        if (null == e) return {};
        var r, n, o = _(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function _(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var T = {
        children: c().node
    };
    /**
 * A styled container for `Modal` body content.
 */    function A(e) {
        var t = e.children, o = q(e, [ "children" ]);
        // @docs-props-type BodyPropsBase
                var i = (0, r.useState)(), a = j(i, 2), l = a[0], u = a[1];
        var c = (0, r.useCallback)((function(e) {
            u(e);
        }), []);
        
        return n().createElement(w, x({
            "data-test": "body"
        }, o, {
            ref: c
        }),  n().createElement(b.ScrollContainerProvider, {
            value: l
        }, t));
    }
    A.propTypes = T;
    /* harmony default export */    const D = A;
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const M = require("@splunk/react-ui/Divider");
    var R =  e.n(M);
    // CONCATENATED MODULE: ./src/Modal/FooterStyles.ts
    var B = g().div.withConfig({
        displayName: "FooterStyles__StyledDiv",
        componentId: "yszcmv-0"
    })([ "flex:0 0 auto;padding:", " ", ";background-color:", ";& >:is(button,a){min-width:80px;}", "" ], S.variables.spacingLarge, S.variables.spacingXLarge, S.variables.backgroundColorDialog, (0, 
    S.pickVariant)("$layout", {
        auto: (0, m.css)([ "display:flex;align-items:center;justify-content:flex-end;gap:", ";& > *:is(button,a,*){flex:0 0 auto;}& >:not(button,a){margin-right:auto;}&:has(:is(button,a) +:is(button,a) +:is(button,a)) >:is(button,a):first-child{margin-right:auto;}" ], S.variables.spacingSmall),
        none: (0, m.css)([ "text-align:right;" ])
    }));
    // CONCATENATED MODULE: ./src/Modal/ModalContext.tsx
    var F =  (0, r.createContext)({});
    /* harmony default export */    const N = F;
    // CONCATENATED MODULE: ./src/Modal/Footer.tsx
    function L() {
        return L = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, L.apply(null, arguments);
    }
    function H(e, t) {
        if (null == e) return {};
        var r, n, o = X(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function X(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var z = {
        children: c().node,
        layout: c().oneOf([ "auto", "none" ])
    };
    /**
 * A styled container for `Modal` footer content.
 */    function $(e) {
        var t = e.children, o = e.layout, i = o === void 0 ? "auto" : o, a = H(e, [ "children", "layout" ]);
        // @docs-props-type FooterPropsBase
                var l = (0, r.useContext)(N), u = l.divider;
        var c = u === "both" || u === "footer";
        
        return n().createElement(n().Fragment, null, c &&  n().createElement(R(), {
            appearance: "weak",
            decorative: true
        }),  n().createElement(B, L({
            "data-test": "footer",
            $layout: i
        }, a), t));
    }
    $.propTypes = z;
    /* harmony default export */    const V = $;
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const U = require("@splunk/react-icons/Cross");
    var K =  e.n(U);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const W = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const G = require("@splunk/react-ui/Button");
    var J =  e.n(G);
    // CONCATENATED MODULE: ./src/Modal/HeaderStyles.ts
    var Q = g()(O()).withConfig({
        displayName: "HeaderStyles__Styled",
        componentId: "sc-1y722ut-0"
    })([ "", ";flex:0 0 auto;position:relative;min-height:", ";background-color:", ";padding:", " ", ";gap:", ";" ], S.mixins.reset("flex"), (0, 
    S.pick)({
        enterprise: "30px",
        prisma: {
            comfortable: "60px",
            compact: "52px"
        }
    }), S.variables.backgroundColorDialog, S.variables.spacingLarge, S.variables.spacingXLarge, S.variables.spacingSmall);
    var Y = g().div.withConfig({
        displayName: "HeaderStyles__StyledTitleWrapper",
        componentId: "sc-1y722ut-1"
    })([ "", ";flex-direction:column;align-self:center;margin-right:auto;" ], S.mixins.reset("flex"));
    var Z = g().h2.withConfig({
        displayName: "HeaderStyles__StyledTitle",
        componentId: "sc-1y722ut-2"
    })([ "", ";", " margin:0;overflow-wrap:break-word;" ], S.mixins.reset("block"), S.mixins.typography("title3"));
    var ee = g().p.withConfig({
        displayName: "HeaderStyles__StyledSubtitle",
        componentId: "sc-1y722ut-3"
    })([ "", ";overflow-wrap:break-word;" ], S.mixins.reset("block"));
    var te = g()(J()).withConfig({
        displayName: "HeaderStyles__StyledButton",
        componentId: "sc-1y722ut-4"
    })([ "flex-shrink:0;flex-grow:0;align-self:flex-start;margin-left:auto;" ]);
    // CONCATENATED MODULE: ./src/Modal/Header.tsx
    function re() {
        return re = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, re.apply(null, arguments);
    }
    function ne(e, t) {
        if (null == e) return {};
        var r, n, o = oe(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function oe(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var ie = {
        children: c().node,
        icon: c().node,
        hideCloseButton: c().bool,
        title: c().string,
        subtitle: c().node
    };
    var ae =  n().createElement(K(), {
        width: "20px",
        height: "20px"
    });
    /**
 * A styled container for `Modal` header content.
 */    function le(e) {
        var t = e.children, o = e.icon, i = e.hideCloseButton, a = i === void 0 ? false : i, l = e.title, u = e.subtitle, c = ne(e, [ "children", "icon", "hideCloseButton", "title", "subtitle" ]);
        // @docs-props-type HeaderPropsBase
                var s = (0, r.useContext)(N), f = s.titleId, d = s.subtitleId, p = s.divider, y = s.onRequestClose;
        var v = p === "both" || p === "header";
        var b =  (0, r.isValidElement)(o) &&  (0, r.cloneElement)(o, {
            width: o.props.width || "48px",
            height: o.props.height || "48px"
        });
        var m = (0, r.useCallback)((function(e) {
            y === null || y === void 0 ? void 0 : y({
                event: e,
                reason: "clickCloseButton"
            });
        }), [ y ]);
        
        return n().createElement(n().Fragment, null,  n().createElement(Q, re({
            "data-test": "header"
        }, c), b, l ?  n().createElement(Y, null, l &&  n().createElement(Z, {
            id: f,
            "data-test": "title"
        }, l), u &&  n().createElement(ee, {
            id: d,
            "data-test": "subtitle"
        }, u)) : t, !a && y &&  n().createElement(te, {
            appearance: "subtle",
            "aria-label": (0, W._)("Close"),
            "data-test": "close",
            icon: ae,
            onClick: m
        })), v &&  n().createElement(R(), {
            appearance: "weak",
            decorative: true
        }));
    }
    le.propTypes = ie;
    /* harmony default export */    const ue = le;
    // CONCATENATED MODULE: external "@react-spring/web"
    const ce = require("@react-spring/web");
    // CONCATENATED MODULE: ./src/Modal/ModalStyles.ts
    var se = g()(ce.animated.div).withConfig({
        displayName: "ModalStyles__Styled",
        componentId: "sc-5fn8ds-0"
    })([ "", ";flex-direction:column;position:fixed;left:50%;transform:translateX(-50%);z-index:", ";box-shadow:", ";max-height:calc(100vh - ", " * 4);max-width:calc(100vw - ", " * 4);border-radius:", ";overflow:hidden;" ], S.mixins.reset("flex"), S.variables.zindexModal, S.variables.modalShadow, S.variables.spacingXLarge, S.variables.spacingXLarge, S.variables.borderRadius);
    // CONCATENATED MODULE: ./src/utils/resolveInitialFocusTarget.ts
    function fe(e) {
        "@babel/helpers - typeof";
        return fe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, fe(e)
        /**
 * Resolves the initial focus target based on the initialFocus prop value.
 *
 * @param initialFocus - The initialFocus prop value ('first', 'container', HTMLElement, or React.Component with focus method)
 * @param containerEl - The container element to use for 'first' and 'container' options
 * @returns The element or component to focus, or null if none found
 */;
    }
    function de(e, t) {
        if (e === "first" && t) {
            var r;
            return (r = (0, y.getSortedTabbableElements)(t)[0]) !== null && r !== void 0 ? r : null;
        }
        if (e === "container") {
            return t;
        }
        if (fe(e) === "object") {
            return e;
        }
        return null;
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function pe(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Modal/Modal.tsx
    function ye(e) {
        "@babel/helpers - typeof";
        return ye = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, ye(e);
    }
    function ve() {
        return ve = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, ve.apply(null, arguments);
    }
    function be(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function me(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? be(Object(r), !0).forEach((function(t) {
                ge(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : be(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function ge(e, t, r) {
        return (t = he(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function he(e) {
        var t = Oe(e, "string");
        return "symbol" == ye(t) ? t : t + "";
    }
    function Oe(e, t) {
        if ("object" != ye(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != ye(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function Se(e, t) {
        return ke(e) || Ce(e, t) || xe(e, t) || we();
    }
    function we() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function xe(e, t) {
        if (e) {
            if ("string" == typeof e) return je(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? je(e, t) : void 0;
        }
    }
    function je(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function Ce(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (n = i.call(r)).done) && (l.push(n.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function ke(e) {
        if (Array.isArray(e)) return e;
    }
    function Pe(e, t) {
        if (null == e) return {};
        var r, n, o = Ee(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function Ee(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */    var Ie = {
        children: c().node,
        closeOnClickAway: c().bool,
        divider: c().oneOf([ "header", "footer", "both", "none" ]),
        elementRef: c().oneOfType([ c().func, c().object ]),
        initialFocus: c().oneOfType([ c().object, c().oneOf([ "first", "container" ]) ]),
        onRequestClose: c().func,
        open: c().bool,
        returnFocus: c().oneOfType([ c().func, c().object ]).isRequired
    };
    function qe(e) {
        var t = e.children, o = e.closeOnClickAway, a = e.elementRef, u = e.divider, c = u === void 0 ? "both" : u, s = e.initialFocus, d = s === void 0 ? "first" : s, b = e.onRequestClose, m = e.open, g = m === void 0 ? false : m, h = e.returnFocus, O = e.style, S = Pe(e, [ "children", "closeOnClickAway", "elementRef", "divider", "initialFocus", "onRequestClose", "open", "returnFocus", "style" ]);
        // @docs-props-type ModalPropsBase
                var w = (0, r.useRef)(null);
        // this timeout ensures that the Modal is finished rendering before setting the focus
                var x;
        var j = (0, r.useRef)(undefined);
        var C = (0, r.useRef)(null);
        var k = (0, r.useState)((function() {
            return {
                headerTitleId: (0, v.createDOMID)("titleId"),
                headerSubtitleId: (0, v.createDOMID)("subtitleId")
            };
        })), P = Se(k, 1), E = P[0], I = E.headerTitleId, q = E.headerSubtitleId;
        var _ = p()(g);
        var T = (0, r.useCallback)((function() {
            return {
                top: 0
            };
        }), []);
        var A = (0, r.useCallback)((function() {
            if (g) {
                return {
                    top: 40
                };
            }
            return {
                top: 0
            };
        }), [ g ]);
        var D = (0, r.useCallback)((function(e) {
            if (g) {
                j.current = i()((function() {
                    // not using the takeFocus() util here because it
                    // doesn't work for all values of "initialFocus"
                    // and it has a nested defer() call, which we want to avoid
                    if (C.current !== d) {
                        var e = de(d, w.current);
                        // Typescript isn't happy with optional chaining here
                                                if (e != null && "focus" in e) {
                            e.focus();
                        }
                        C.current = d;
                    }
                }));
            } else {
                C.current = null;
                clearTimeout(j.current);
                if (e) {
                    /**
         * We return focus to the invoking element here instead of in `handleRequestClose` because users can
         * close the modal by updating the value of the `open` prop without calling `onRequestClose`
         */
                    if (h != null) {
                        if (typeof h === "function") {
                            h();
                        } else {
                            i()((function() {
                                var e, t;
                                (e = h.current) === null || e === void 0 ? void 0 : (t = e.focus) === null || t === void 0 ? void 0 : t.call(e);
                            }));
                        }
                    }
                }
            }
        }), [ d, g, h ]);
        (0, r.useEffect)((function() {
            if (C.current !== d) {
                // cancel the defer() set up in handleInitialFocus()
                // so that focus won't jump around if "initialFocus" changes
                clearTimeout(j.current);
            }
            D(_);
        }), [ D, d, g, x, _ ]);
        var M = (0, r.useCallback)((function(e) {
            w.current = e;
            pe(a, e);
        }), [ a, w ]);
        var R = function e(t) {
            if (w.current !== null) {
                (0, y.handleTab)(w.current, t.nativeEvent);
            }
        };
        var B = (0, r.useCallback)((function(e) {
            C.current = null;
            b === null || b === void 0 ? void 0 : b(e);
        }), [ b ]);
        var F = (0, r.useMemo)((function() {
            return {
                divider: c,
                onRequestClose: b,
                subtitleId: q,
                titleId: I
            };
        }), [ c, b, I, q ]);
        var L = l()((function(e) {
            var t = e.opacity, r = e.modalStyle, n = e.top;
            return me(me({}, r), {}, {
                opacity: t,
                top: n
            });
        }));
        var H = (0, r.useCallback)((function(e) {
            var o = e.top, i = e.opacity;
            var a = r.Children.toArray(t).filter(r.isValidElement);
            
            return n().createElement(se, ve({
                ref: M,
                "data-test": "modal",
                style: L({
                    opacity: i,
                    modalStyle: O,
                    top: o
                }),
                tabIndex: -1,
                onKeyDown: R,
                role: "dialog",
                "aria-labelledby": I,
                "aria-describedby": q
            }, S),  n().createElement(N.Provider, {
                value: F
            }, a));
        }), [ t, L, M, I, q, S, O, F ]);
        var X = o ? "interactive" : "visible";
        
        return n().createElement(f(), {
            getDefaultMotionStyle: T,
            getMotionStyle: A,
            onRequestClose: B,
            open: g,
            renderModal: H,
            scrim: X
        });
    }
    qe.propTypes = Ie;
    qe.Header = ue;
    qe.Footer = V;
    qe.Body = D;
    /* harmony default export */    const _e = qe;
    // CONCATENATED MODULE: ./src/Modal/index.ts
    module.exports = t;
    /******/})();
