// @splunk/react-ui/ModalLayer.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ C
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const a = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "@splunk/react-ui/Layer"
    const u = require("@splunk/react-ui/Layer");
    var l =  e.n(u);
    // CONCATENATED MODULE: external "@react-spring/web"
    const c = require("@react-spring/web");
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ModalLayer/ModalLayerStyles.ts
    var v = (0, s.css)([ "", ";position:fixed;top:0;right:0;bottom:0;left:0;" ], y.mixins.reset("block"));
    var p = f()(c.animated.div).withConfig({
        displayName: "ModalLayerStyles__StyledClickAwayOverlay",
        componentId: "sc-1sa3n1j-0"
    })([ "", ";background-color:", ";z-index:", ";" ], v, y.variables.backgroundColorScrim, y.variables.zindexModalBackdrop);
    // CONCATENATED MODULE: ./src/ModalLayer/ModalLayer.tsx
    function d(e) {
        "@babel/helpers - typeof";
        return d = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, d(e);
    }
    function b(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function m(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? b(Object(t), !0).forEach((function(r) {
                g(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : b(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function g(e, r, t) {
        return (r = O(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function O(e) {
        var r = S(e, "string");
        return "symbol" == d(r) ? r : r + "";
    }
    function S(e, r) {
        if ("object" != d(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != d(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function h(e, r) {
        return A(e) || M(e, r) || w(e, r) || j();
    }
    function j() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function w(e, r) {
        if (e) {
            if ("string" == typeof e) return P(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? P(e, r) : void 0;
        }
    }
    function P(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function M(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, a, u = [], l = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    l = !1;
                } else for (;!(l = (n = i.call(t)).done) && (u.push(n.value), u.length !== r); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function A(e) {
        if (Array.isArray(e)) return e;
    }
    var k = {
        childrenInAnimation: i().bool,
        getDefaultMotionStyle: i().func,
        getMotionStyle: i().func,
        onRequestClose: i().func,
        open: i().bool,
        renderModal: i().func.isRequired,
        scrim: i().oneOf([ "visible", "hidden", "interactive" ])
    };
    var q = [ "escapeKey" ];
    function E(e) {
        var r, o;
        var i = e.childrenInAnimation, u = e.getDefaultMotionStyle, c = e.getMotionStyle, s = e.onRequestClose, f = e.open, y = e.renderModal, v = e.scrim, d = v === void 0 ? "interactive" : v;
        var b = (0, t.useState)(false), g = h(b, 2), O = g[0], S = g[1];
        var j = (0, t.useRef)(true);
        var w = (0, t.useState)(f), P = h(w, 2), M = P[0], A = P[1];
        if (f !== M) {
            // we can't use prevOpen here since that's a ref behind the scenes
            // and won't get updated until the component re-renders successfully
            // whereas setAnimating will interrupt the re-render and we'll end up in an infinite loop
            // so we need to update the value of wasOpen at the same time
            S(true);
            A(f);
        }
        // equivalent to componentDidMount and componentWillUnmount
                (0, t.useEffect)((function() {
            // we need to set this to true on every mount because React.StrictMode will call the effect twice
            // and we need to ensure that allowAnimationUpdates is true while the component is mounted
            j.current = true;
            return function() {
                j.current = false;
            };
        }), []);
        var k = function e(r) {
            s === null || s === void 0 ? void 0 : s({
                reason: "clickAway",
                event: r
            });
        };
        var E = function e() {
            if (j.current) {
                S(false);
            }
        };
        var C = m(m({}, (r = u === null || u === void 0 ? void 0 : u()) !== null && r !== void 0 ? r : {}), {}, {
            overlayOpacity: 0,
            opacity: 0
        });
        var D = m(m({}, (o = c === null || c === void 0 ? void 0 : c()) !== null && o !== void 0 ? o : {}), {}, {
            overlayOpacity: f ? 1 : 0,
            opacity: f ? 1 : 0
        });
        var R = (0, a.useAnimation)({
            from: C,
            to: D,
            onRest: E
        });
        var x = (0, t.useMemo)((function() {
            return {
                opacity: R.overlayOpacity
            };
        }), [ R ]);
        var I = function e() {
            var r = f || O;
            var t = r && d !== "hidden";
            
            return n().createElement("div", null, y(R), t &&  n().createElement(p, {
                "data-test": "modal-overlay",
                onMouseDown: d === "interactive" ? k : undefined,
                key: "clickAway",
                style: x
            }));
        };
        var _ = f || O || i;
        
        return n().createElement(l(), {
            open: _,
            closeReasons: q,
            onRequestClose: s
        }, _ && I());
    }
    E.propTypes = k;
    /* harmony default export */    const C = E;
    // CONCATENATED MODULE: ./src/ModalLayer/index.ts
    module.exports = r;
    /******/})();
