// @splunk/react-ui/Monogram.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ G,
        getInitials: () => /* reexport */ U
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/omit"
    const i = require("lodash/omit");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var s =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const u = require("@splunk/react-ui/Clickable");
    var f =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Monogram/MonogramStyles.ts
    // Thanks: https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0
    var d = function e(r) {
        var t = 0;
        for (var n = 0; n < r.length; ) {
            t = (t << 5) - t + r.charCodeAt(n++) | 0;
 // eslint-disable-line no-bitwise, no-plusplus
                }
        return t;
    };
    var m = [ "#006d9c", "#ec9960", "#af575a", "#62b3b2", "#4fa484", "#f8be34", "#5a4575", "#708794", "#294e70", "#b6c75a", "#5cc05c", "#007abd", "#dc4e41", "#f1813f", "#f8be34", "#53a051", "#006d9c", "#77d6d8", "#f589ad", "#6a2c5d" ];
    var b = (0, c.css)([ "", ";align-items:center;justify-content:center;border-radius:50%;font-weight:normal;", " background-color:", ";", ";& + &{margin-left:4px;}" ], p.mixins.reset("inline-flex"), (0, 
    p.pick)({
        prisma: (0, c.css)([ "color:", ";" ], p.variables.contentColorActive)
    }), (function(e) {
        var r = e.$backgroundColor, t = e.$initials;
        if (r === "theme") {
            return (0, p.pick)({
                enterprise: {
                    dark: p.variables.neutral300,
                    light: p.variables.neutral100
                },
                prisma: p.variables.neutral100
            });
        }
        if (r === "auto") {
            return m[d(t) % m.length];
        }
        return r;
    }), (function(e) {
        var r = e.$size;
        if (typeof r === "number") {
            return (0, c.css)([ "height:", "px;width:", "px;" ], r, r);
        }
        if (typeof r === "string") {
            return (0, p.pickVariant)("$size", {
                small: (0, c.css)([ "font-size:", ";", "" ], p.variables.fontSize, (0, p.pick)({
                    enterprise: (0, c.css)([ "height:32px;width:32px;" ]),
                    prisma: (0, c.css)([ "height:24px;width:24px;" ])
                })),
                medium: (0, c.css)([ "height:48px;width:48px;font-size:", ";" ], p.variables.fontSize),
                large: (0, c.css)([ "height:80px;width:80px;font-size:", ";" ], p.variables.fontSizeXXLarge),
                xlarge: (0, c.css)([ "width:144px;height:144px;font-size:36px;font-weight:700;" ])
            });
        }
        return (0, c.css)([ "height:48px;width:48px;font-size:", ";" ], p.variables.fontSize);
    }));
    var v = s().div.withConfig({
        displayName: "MonogramStyles__Styled",
        componentId: "r5ud6k-0"
    })([ "", "" ], b);
    var y = s()(f()).withConfig({
        displayName: "MonogramStyles__StyledClickableComponent",
        componentId: "r5ud6k-1"
    })([ "", "" ], b);
    var g = s()(y).withConfig({
        displayName: "MonogramStyles__StyledClickable",
        componentId: "r5ud6k-2"
    })([ "cursor:pointer;position:relative;:focus,:hover{", "}:focus{box-shadow:", ";z-index:1;}:active{", "}" ], (0, 
    p.pick)({
        enterprise: (0, c.css)([ "box-shadow:", ";" ], p.variables.focusShadowInset),
        prisma: (0, c.css)([ "border:1px solid transparent;background-color:", ";" ], p.variables.neutral200)
    }), (0, p.pick)({
        enterprise: p.variables.focusShadowInset,
        prisma: p.variables.focusShadow
    }), (0, p.pick)({
        enterprise: (0, c.css)([ "opacity:0.85;" ]),
        prisma: (0, c.css)([ "border:1px solid transparent;background-color:", ";opacity:1;" ], p.variables.neutral200)
    }));
    var h = s().span.withConfig({
        displayName: "MonogramStyles__Initials",
        componentId: "r5ud6k-3"
    })([ "opacity:0.8;text-transform:capitalize;" ]);
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var O = "is deprecated and will be removed in the next major version.";
    var k = function e(r) {
        var n = r.additionalMessage, i = n === void 0 ? "" : n, o = r.componentName;
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ i, o ]);
    };
    var w = function e(r) {
        var t = r.additionalMessage, n = t === void 0 ? "" : t, i = r.componentName, o = r.propName, a = r.propValue;
        useEffect((function() {
            if (false) {}
        }), [ n, i, o, a ]);
    };
    var S = function e(r) {
        var t = r.additionalMessage, n = t === void 0 ? "" : t, i = r.componentName, o = r.deprecatedPropValue, a = r.propName, l = r.propValue;
        useEffect((function() {
            if (false) {}
        }), [ n, i, a, l, o ]);
    };
    // CONCATENATED MODULE: ./src/Monogram/Monogram.tsx
    function j(e) {
        "@babel/helpers - typeof";
        return j = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, j(e);
    }
    function x(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function C(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? x(Object(t), !0).forEach((function(r) {
                z(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : x(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function z(e, r, t) {
        return (r = P(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function P(e) {
        var r = E(e, "string");
        return "symbol" == j(r) ? r : r + "";
    }
    function E(e, r) {
        if ("object" != j(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != j(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function I(e, r) {
        return q(e) || $(e, r) || N(e, r) || _();
    }
    function _() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function N(e, r) {
        if (e) {
            if ("string" == typeof e) return A(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? A(e, r) : void 0;
        }
    }
    function A(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function $(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, i, o, a, l = [], c = !0, s = !1;
            try {
                if (o = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    c = !1;
                } else for (;!(c = (n = o.call(t)).done) && (l.push(n.value), l.length !== r); c = !0) {
                }
            } catch (e) {
                s = !0, i = e;
            } finally {
                try {
                    if (!c && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw i;
                }
            }
            return l;
        }
    }
    function q(e) {
        if (Array.isArray(e)) return e;
    }
    function R(e, r) {
        if (null == e) return {};
        var t, n, i = T(e, r);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (n = 0; n < o.length; n++) {
                t = o[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (i[t] = e[t]);
            }
        }
        return i;
    }
    function T(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */    var D = {
        backgroundColor: l().string,
        elementRef: l().oneOfType([ l().func, l().object ]),
        initials: l().string.isRequired,
        name: l().string,
        onClick: l().func,
        size: l().oneOfType([ l().oneOf([ "small", "medium", "large" ]), l().number ])
    };
    function V(e) {
        return !!e.onClick;
    }
    /**
 * Returns a suitable set of initials for a name. Uses the first character of each
 * name segment and omits middle segments if the segment count is greater than three.
 * @param {string} name - The full name.
 * @returns {string} Limited to three characters. Empty if `name` is empty.
 * @public
 */    function U(e) {
        var r = (e.match(/[^\s-]+/g) || []).map((function(e) {
            return e[0];
        })).join("");
        return r.length > 3 ? "".concat(r[0]).concat(r[r.length - 1]) : r;
    }
    function X(e) {
        var r = e.backgroundColor, t = e.initials, n = e.size, i = R(e, [ "backgroundColor", "initials", "size" ]);
        var o = {
            "data-test": "monogram",
            $backgroundColor: r,
            $initials: t,
            $size: n
        };
        return [ o, i ];
    }
    function L(e) {
        var r = X(e), i = I(r, 2), a = i[0], l = i[1];
        var c = l.children, s = l.elementRef, u = l.name, f = l.onClick, p = R(l, [ "children", "elementRef", "name", "onClick" ]);
        var d = (0, t.useCallback)((function(e) {
            f(e, {
                name: u
            });
        }), [ f, u ]);
        
        return n().createElement(g, M({
            elementRef: s,
            onClick: d
        }, a, o()(p, Object.keys(D))), c);
    }
    function B(e) {
        var r = X(e), t = I(r, 2), i = t[0], a = t[1];
        var l = a.children, c = a.elementRef, s = R(a, [ "children", "elementRef" ]);
        
        return n().createElement(v, M({
            ref: c
        }, i, o()(s, Object.keys(D))), l);
    }
    /**
 * @deprecated
 * Monogram has been deprecated and will be removed in a future major version. Use Avatar instead.
 */    function F(e) {
        var r = e.backgroundColor, t = r === void 0 ? "theme" : r, i = e.initials, o = e.size, a = o === void 0 ? "medium" : o, l = R(e, [ "backgroundColor", "initials", "size" ]);
        // @docs-props-type MonogramPropsBase
                k({
            additionalMessage: "Use Avatar instead.",
            componentName: "Monogram"
        });
        var c =  n().createElement(h, {
            "data-test": "initials"
        }, i);
        var s = C({
            backgroundColor: t,
            size: a,
            initials: i
        }, l);
        return V(s) ?  n().createElement(L, s, c) :  n().createElement(B, s, c);
    }
    F.propTypes = D;
    /* harmony default export */    const G = F;
    // CONCATENATED MODULE: ./src/Monogram/index.ts
    module.exports = r;
    /******/})();
