// @splunk/react-ui/Multiselect.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Divider: () => /* reexport */ u.Divider,
        Heading: () => /* reexport */ u.Heading,
        Option: () => /* reexport */ h,
        default: () => /* reexport */ cn
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/has"
    const o = require("lodash/has");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const u = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const s = require("@splunk/react-ui/useControlled");
    var c =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const d = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/react-ui/SelectBase"
    const f = require("@splunk/react-ui/SelectBase");
    var p =  e.n(f);
    // CONCATENATED MODULE: ./src/Multiselect/Option.tsx
    function v() {
        return v = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, v.apply(null, arguments);
    }
    function b(e, n) {
        if (null == e) return {};
        var r, t, o = m(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function m(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var g = {
        /** @private */
        active: i().bool,
        children: i().node,
        /**
   * @private this is passed down from Multiselect.
   */
        compact: i().bool,
        description: i().string,
        descriptionPosition: i().oneOf([ "right", "bottom" ]),
        disabled: i().oneOfType([ i().bool, i().oneOf([ "dimmed", "disabled" ]) ]),
        /**
   * @private Undocumented due to limitations in supporting in compact `Multiselect` (see SUI-8286 notes for more info).
   */
        elementRef: i().oneOfType([ i().func, i().object ]),
        hidden: i().bool,
        icon: i().node,
        label: i().string.isRequired,
        matchRanges: i().arrayOf(i().shape({
            start: i().number.isRequired,
            end: i().number.isRequired
        })),
        /** @private */
        onClick: i().func,
        /** @private */
        role: i().oneOf([ "menuitemcheckbox", "option" ]),
        /** @private */
        selected: i().bool,
        selectedAppearance: i().oneOf([ "info", "success", "warning", "error" ]),
        selectedBackgroundColor: i().string,
        selectedForegroundColor: i().string,
        truncate: i().bool,
        value: i().oneOfType([ i().string, i().number, i().bool ]).isRequired
    };
    /**
 * An option within a `Multiselect`.
 */    function y(e) {
        var n = e.compact, r = e.children, o = e.description, a = e.descriptionPosition, l = a === void 0 ? "bottom" : a, i = e.disabled, u = e.elementRef, s = e.hidden, c = e.icon, d = e.label, p = e.matchRanges, m = e.role, g = e.selected, y = e.truncate, h = e.value, O = b(e, [ "compact", "children", "description", "descriptionPosition", "disabled", "elementRef", "hidden", "icon", "label", "matchRanges", "role", "selected", "truncate", "value" ]);
        // @docs-props-type OptionPropsBase
        // selectedAppearance, selectedBackgroundColor, and selectedForegroundColor are not used directly by the Option component.
        // They are consumed by the parent component (Multiselect-Normal) to render the Chip, so they are not included in the above prop list.
                
        return t().createElement(f.Option, v({
            description: o,
            descriptionPosition: l,
            disabled: i,
            elementRef: u,
            hidden: s,
            icon: c,
            label: d,
            matchRanges: p,
            role: m,
            selected: g,
            truncate: y,
            value: h
        }, O, {
            multiple: n
        }), r || d);
    }
    y.propTypes = g;
    /* harmony default export */    const h = y;
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var O = "is deprecated and will be removed in the next major version.";
    var w = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, o = n.componentName;
        useEffect((function() {
            if (false) {}
        }), [ t, o ]);
    };
    var S = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, o = n.componentName, a = n.propName, l = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ t, o, a, l ]);
    };
    var C = function e(n) {
        var t = n.additionalMessage, o = t === void 0 ? "" : t, a = n.componentName, l = n.deprecatedPropValue, i = n.propName, u = n.propValue;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ o, a, i, u, l ]);
    };
    // CONCATENATED MODULE: ./src/Multiselect/Compact.tsx
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function P(e, n) {
        if (null == e) return {};
        var r, t, o = k(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function k(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var j = {
        allowNewValues: i().bool,
        animateLoading: i().bool,
        append: i().bool,
        children: i().node,
        controlledFilter: i().bool,
        defaultPlacement: i().oneOf([ "above", "below", "vertical" ]),
        defaultValues: i().array,
        describedBy: i().string,
        disabled: i().oneOfType([ i().bool, i().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        filter: i().oneOf([ false, true, "controlled" ]),
        footerMessage: i().node,
        inline: i().bool,
        inputId: i().string,
        inputRef: i().oneOfType([ i().func, i().object ]),
        isLoadingOptions: i().bool,
        labelledBy: i().string,
        loadingMessage: i().node,
        /** @private. */
        maxLabelItems: i().number,
        menuStyle: i().object,
        name: i().string,
        noOptionsMessage: i().node,
        onChange: i().func,
        onClose: i().func,
        onFilterChange: i().func,
        onOpen: i().func,
        onScroll: i().func,
        onScrollBottom: i().func,
        placeholder: i().string,
        prepend: i().bool,
        repositionMode: i().oneOf([ "none", "flip" ]),
        selectAllAppearance: i().oneOf([ "buttongroup", "checkbox", "none" ]),
        showSelectedValuesFirst: i().oneOf([ "nextOpen", "immediately", "never" ]),
        tabConfirmsNewValue: i().bool,
        values: i().array,
        /** @private */
        virtualization: i().number
    };
    // TS: not typed controlled/uncontrolled, handled by Multiselect wrapper component
        var V = {};
    var E = (0, d._)("No matches");
    var N = (0, d._)("Select...");
    function x(e) {
        var n = e.allowNewValues, o = e.animateLoading, a = e.append, l = e.children, i = e.controlledFilter, u = e.defaultPlacement, s = u === void 0 ? "vertical" : u, c = e.defaultValues, d = e.describedBy, v = e.disabled, b = e.elementRef, m = e.error, g = e.filter, y = e.footerMessage, h = e.inline, O = e.inputId, w = e.inputRef, S = e.isLoadingOptions, k = e.labelledBy, j = e.loadingMessage, x = e.menuStyle, B = x === void 0 ? V : x, R = e.name, I = e.noOptionsMessage, F = I === void 0 ? E : I, q = e.onChange, A = e.onClose, D = e.onFilterChange, L = e.onOpen, T = e.onScroll, _ = e.onScrollBottom, H = e.placeholder, $ = H === void 0 ? N : H, z = e.prepend, K = e.repositionMode, W = e.selectAllAppearance, U = e.showSelectedValuesFirst, G = U === void 0 ? "nextOpen" : U, X = e.tabConfirmsNewValue, J = e.values, Q = e.virtualization, Y = P(e, [ "allowNewValues", "animateLoading", "append", "children", "controlledFilter", "defaultPlacement", "defaultValues", "describedBy", "disabled", "elementRef", "error", "filter", "footerMessage", "inline", "inputId", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onChange", "onClose", "onFilterChange", "onOpen", "onScroll", "onScrollBottom", "placeholder", "prepend", "repositionMode", "selectAllAppearance", "showSelectedValuesFirst", "tabConfirmsNewValue", "values", "virtualization" ]);
        // @docs-props-type CompactPropsBase
                C({
            componentName: "Multiselect",
            deprecatedPropValue: "buttongroup",
            propName: "selectAllAppearance",
            propValue: W
        });
        var Z = true;
        if (g != null) {
            Z = g;
            if (false) {}
        } else {
            Z = i ? "controlled" : true;
        }
        var ee = (0, r.useMemo)((function() {
            return {
                children: r.Children.toArray(l).map((function(e) {
                    if ((0, f.isOption)(e)) {
                        var n;
                        var r = (n = e.key) !== null && n !== void 0 ? n : "".concat(e.props.label, "-").concat(e.props.value);
                        
                        return t().createElement(f.Option, M({
                            key: r
                        }, e.props, {
                            multiple: true
                        }));
                    }
                    return e;
                })),
                filter: Z
            };
        }), [ l, Z ]);
        if (c != null) {
            ee.defaultValues = c;
        }
        if (J != null) {
            ee.values = J;
        }
        
        return t().createElement(p(), M({
            allowNewValues: n,
            animateLoading: o,
            append: a,
            describedBy: d,
            disabled: v,
            defaultPlacement: s,
            elementRef: b,
            error: m,
            footerMessage: y,
            inline: h,
            inputId: O,
            inputRef: w,
            isLoadingOptions: S,
            labelledBy: k,
            loadingMessage: j,
            menuStyle: B,
            name: R,
            noOptionsMessage: F,
            onChange: q,
            onClose: A,
            onFilterChange: D,
            onOpen: L,
            onScroll: T,
            onScrollBottom: _,
            placeholder: $,
            prepend: z,
            repositionMode: K,
            selectAllAppearance: W,
            showSelectedValuesFirst: G,
            tabConfirmsNewValue: X,
            virtualization: Q
        }, Y, ee, {
            multiple: true
        }));
    }
    x.propTypes = j;
    x.Option = h;
    x.Divider = u.Divider;
    x.Heading = u.Heading;
    /* harmony default export */    const B = x;
    // CONCATENATED MODULE: external "lodash/defer"
    const R = require("lodash/defer");
    var I =  e.n(R);
    // CONCATENATED MODULE: external "lodash/get"
    const F = require("lodash/get");
    var q =  e.n(F);
    // CONCATENATED MODULE: external "lodash/isString"
    const A = require("lodash/isString");
    var D =  e.n(A);
    // CONCATENATED MODULE: external "lodash/keys"
    const L = require("lodash/keys");
    var T =  e.n(L);
    // CONCATENATED MODULE: external "lodash/memoize"
    const _ = require("lodash/memoize");
    var H =  e.n(_);
    // CONCATENATED MODULE: external "lodash/omit"
    const $ = require("lodash/omit");
    var z =  e.n($);
    // CONCATENATED MODULE: external "lodash/pickBy"
    const K = require("lodash/pickBy");
    var W =  e.n(K);
    // CONCATENATED MODULE: external "lodash/without"
    const U = require("lodash/without");
    var G =  e.n(U);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const X = require("@splunk/react-ui/Popover");
    var J =  e.n(X);
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const Q = require("@splunk/react-ui/ResultsMenu");
    var Y =  e.n(Q);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const Z = require("@splunk/react-ui/ScreenReaderContent");
    var ee =  e.n(Z);
    // CONCATENATED MODULE: external "@splunk/ui-utils/filter"
    const ne = require("@splunk/ui-utils/filter");
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const re = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const te = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/scroll"
    const oe = require("@splunk/ui-utils/scroll");
    // CONCATENATED MODULE: external "styled-components"
    const ae = require("styled-components");
    var le =  e.n(ae);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const ie = require("@splunk/react-ui/Box");
    var ue =  e.n(ie);
    // CONCATENATED MODULE: external "@splunk/react-ui/Chip"
    const se = require("@splunk/react-ui/Chip");
    var ce =  e.n(se);
    // CONCATENATED MODULE: external "@splunk/themes"
    const de = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Multiselect/NormalStyles.ts
    var fe = le()(ue()).withConfig({
        displayName: "NormalStyles__StyledBox",
        componentId: "sc-1uwwpco-0"
    })([ "display:flex;align-items:center;flex-wrap:wrap;gap:", ";border-radius:", ";min-width:200px;min-height:", ";max-height:300px;overflow-y:auto;border:", " solid ", ";padding-inline:", ";background-color:", ";", " ", " ", " ", " ", " ", " ", " &[data-inline]{width:400px;}" ], de.variables.spacingXSmall, de.variables.borderRadius, de.variables.inputHeight, de.variables.inputBorderWidth, de.variables.interactiveColorBorder, de.variables.spacingXSmall, de.variables.interactiveColorBackground, (function(e) {
        var n = e.$append;
        return n && (0, ae.css)([ "border-start-end-radius:0;border-end-end-radius:0;border-inline-end:none;" ]);
    }), (function(e) {
        var n = e.$prepend;
        return n && (0, ae.css)([ "border-start-start-radius:0;border-end-start-radius:0;" ]);
    }), (function(e) {
        var n = e.$hasFocus;
        return n && (0, ae.css)([ "box-shadow:", ";" ], de.variables.focusShadow);
    }), (function(e) {
        var n = e.$error;
        return n && (0, ae.css)([ "border-color:", ";&:hover:not([disabled]){border-color:", ";}" ], de.variables.interactiveColorAccentError, de.variables.interactiveColorAccentErrorStrong);
    }), (function(e) {
        var n = e.$disabled;
        return n && (0, ae.css)([ "border-color:", ";background-color:", ";cursor:not-allowed;" ], de.variables.interactiveColorBorderDisabled, de.variables.interactiveColorBackgroundDisabled);
    }), (function(e) {
        var n = e.$disabled, r = e.$hasFocus;
        return !n && !r && (0, ae.css)([ "&:hover{border-color:", ";}" ], de.variables.interactiveColorBorderHover);
    }), (function(e) {
        var n = e.$popoverOpen;
        return n && (0, ae.css)([ "position:relative;z-index:calc(", " - 2);" ], de.variables.zindexFixedNavbar);
    }));
    var pe = le().span.withConfig({
        displayName: "NormalStyles__StyledButtonsWrapper",
        componentId: "sc-1uwwpco-1"
    })([ "display:contents;" ]);
    var ve = le().input.withConfig({
        displayName: "NormalStyles__StyledInput",
        componentId: "sc-1uwwpco-2"
    })([ "", ";flex:1 0 auto;max-width:100%;" ], de.mixins.reset("block"));
    var be = le()(ce()).withConfig({
        displayName: "NormalStyles__StyledChip",
        componentId: "sc-1uwwpco-3"
    })([ "line-height:1.1429;height:auto;", ";&:not([disabled]):focus{box-shadow:", ";}" ], (0, 
    de.pick)({
        compact: (0, ae.css)([ "padding-block:2px;" ])
    }), de.variables.focusShadowInset);
    // CONCATENATED MODULE: ./src/utils/disabled.ts
    /**
 * Normalizes the disabled prop and returns both the boolean state and normalized string value.
 * Use this function to ensure consistent handling of disabled prop across components.
 *
 * @param disabled - The disabled prop value
 * @param defaultWhenBooleanTrue - Optional normalization override for disabled={true}
 * @returns Object with isDisabled boolean|undefined and disabledValue string
 *
 * TODO (SUI-8600): `disabled` prop should default to 'dimmed' for data entry components
 */
    function me(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "dimmed";
        if (e === "dimmed") {
            return {
                isDisabled: true,
                disabledValue: "dimmed"
            };
        }
        if (e === true) {
            return {
                isDisabled: true,
                disabledValue: n
            };
        }
        if (e === "disabled") {
            return {
                isDisabled: true,
                disabledValue: "disabled"
            };
        }
        if (e === false) {
            return {
                isDisabled: false,
                disabledValue: undefined
            };
        }
        return {
            isDisabled: undefined,
            disabledValue: undefined
        };
    }
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var ge = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function ye() {
        var e = typeof document !== "undefined" ? document : ge;
        return e;
    }
    var he = ye();
    /* harmony default export */    const Oe = /* unused pure expression or super */ null && he;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function we(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Multiselect/Normal.tsx
    function Se(e) {
        "@babel/helpers - typeof";
        return Se = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Se(e);
    }
    function Ce() {
        return Ce = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, Ce.apply(null, arguments);
    }
    function Me(e) {
        return je(e) || ke(e) || Ne(e) || Pe();
    }
    function Pe() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ke(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function je(e) {
        if (Array.isArray(e)) return xe(e);
    }
    function Ve(e, n) {
        return Re(e) || Be(e, n) || Ne(e, n) || Ee();
    }
    function Ee() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Ne(e, n) {
        if (e) {
            if ("string" == typeof e) return xe(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? xe(e, n) : void 0;
        }
    }
    function xe(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function Be(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, o, a, l, i = [], u = !0, s = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = a.call(r)).done) && (i.push(t.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (l = r["return"](), Object(l) !== l)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return i;
        }
    }
    function Re(e) {
        if (Array.isArray(e)) return e;
    }
    function Ie(e, n) {
        if (null == e) return {};
        var r, t, o = Fe(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function Fe(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    function qe(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function Ae(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? qe(Object(r), !0).forEach((function(n) {
                De(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : qe(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function De(e, n, r) {
        return (n = Le(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function Le(e) {
        var n = Te(e, "string");
        return "symbol" == Se(n) ? n : n + "";
    }
    function Te(e, n) {
        if ("object" != Se(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != Se(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    var _e = {
        allowNewValues: i().bool,
        animateLoading: i().bool,
        append: i().bool,
        children: i().node,
        controlledFilter: i().bool,
        defaultPlacement: i().oneOf([ "above", "below", "vertical" ]),
        defaultValues: i().array,
        describedBy: i().string,
        disabled: i().oneOfType([ i().bool, i().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        footerMessage: i().node,
        inline: i().bool,
        inputId: i().string,
        inputRef: i().oneOfType([ i().func, i().object ]),
        isLoadingOptions: i().bool,
        labelledBy: i().string,
        loadingMessage: i().node,
        menuStyle: i().object,
        name: i().string,
        noOptionsMessage: i().node,
        onChange: i().func,
        onClose: i().func,
        onFilterChange: i().func,
        onOpen: i().func,
        onScroll: i().func,
        onScrollBottom: i().func,
        placeholder: i().string,
        prepend: i().bool,
        /** @private. */
        required: i().bool,
        repositionMode: i().oneOf([ "none", "flip" ]),
        tabConfirmsNewValue: i().bool,
        values: i().array
    };
    var He = H()((function(e) {
        return {
            flexBasis: e,
            width: e
        };
    }));
    // export for testing purpose
        var $e = H()((function(e) {
        var n = e.anchorWidth, r = n === void 0 ? 0 : n, t = e.maxHeight, o = e.menuStyle;
        return Ae({
            maxHeight: t,
            overflow: "auto",
            width: Math.max(r, 200)
        }, o || {});
    }));
    // TS: not typed controlled/uncontrolled, handled by Multiselect wrapper component
        var ze = {};
    var Ke = (0, d._)("No matches");
    var We = (0, d._)("Select...");
    var Ue = function e(n) {
        var r = me(n, "disabled"), t = r.isDisabled;
        return !!t;
    };
    function Ge(e) {
        var n = e.allowNewValues, o = e.animateLoading, l = e.append, i = e.children, u = e.controlledFilter, s = e.defaultPlacement, f = s === void 0 ? "vertical" : s, p = e.defaultValues, v = e.describedBy, b = e.disabled, m = e.elementRef, g = e.error, y = e.footerMessage, O = e.inline, w = e.inputId, S = e.inputRef, C = e.isLoadingOptions, M = e.labelledBy, P = e.loadingMessage, k = e.menuStyle, j = k === void 0 ? ze : k, V = e.name, E = e.noOptionsMessage, N = E === void 0 ? Ke : E, x = e.onChange, B = e.onClose, R = e.onFilterChange, F = e.onScroll, A = e.onScrollBottom, L = e.onOpen, _ = e.placeholder, H = _ === void 0 ? We : _, $ = e.prepend, K = e.required, U = e.repositionMode, X = e.tabConfirmsNewValue, Q = e.values, Z = Ie(e, [ "allowNewValues", "animateLoading", "append", "children", "controlledFilter", "defaultPlacement", "defaultValues", "describedBy", "disabled", "elementRef", "error", "footerMessage", "inline", "inputId", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onChange", "onClose", "onFilterChange", "onScroll", "onScrollBottom", "onOpen", "placeholder", "prepend", "required", "repositionMode", "tabConfirmsNewValue", "values" ]);
        // @docs-props-type NormalPropsBase
                var ae = c()({
            componentName: "Multiselect",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultValues",
            valuePropName: "values"
        });
        if (false) {}
        var le = (0, r.useState)(0), ie = Ve(le, 2), ue = ie[0], se = ie[1];
        var ce = (0, r.useState)(null), de = Ve(ce, 2), ge = de[0], he = de[1];
        var Oe = (0, r.useState)(""), Se = Ve(Oe, 2), Pe = Se[0], ke = Se[1];
        var je = (0, r.useState)(false), Ee = Ve(je, 2), Ne = Ee[0], xe = Ee[1];
        var Be = (0, r.useState)(false), Re = Ve(Be, 2), Fe = Re[0], qe = Re[1];
        var De = (0, r.useState)(p || []), Le = Ve(De, 2), Te = Le[0], _e = Le[1];
        var Ge = (0, r.useState)(""), Xe = Ve(Ge, 2), Je = Xe[0], Qe = Xe[1];
        var Ye = ae && Q ? Q : Te;
        var Ze = Ye.indexOf(Pe) >= 0;
        var en = me(b, "disabled"), nn = en.isDisabled, rn = en.disabledValue;
        var tn = (0, r.useMemo)((function() {
            return r.Children.toArray(i);
        }), [ i ]);
        var on = (0, r.useMemo)((function() {
            return tn.filter(r.isValidElement).filter((function(e) {
                return e.type === h;
            }));
        }), [ tn ]);
        var an = (0, r.useMemo)((function() {
            var e = new Map;
            on.forEach((function(n) {
                e.set(n.props.value, n);
            }));
            return [ e ];
        }), [ on ]), ln = Ve(an, 1), un = ln[0];
        var sn = (0, r.useMemo)((function() {
            var e = Ye.slice(0);
            var n = new Set;
            // Map Options to selected values
                        if (Ye.length) {
                on.forEach((function(r) {
                    var t = Ye.indexOf(r.props.value);
                    if (t !== -1) {
                        e[t] = r;
                        if (Ue(r.props.disabled)) {
                            n.add(r.props.value);
                        }
                    }
                }));
            }
            return {
                selectedItems: e,
                disabledSelectedItems: n
            };
        }), [ on, Ye ]), cn = sn.selectedItems, dn = sn.disabledSelectedItems;
        var fn = (0, r.useRef)(null);
        var pn = (0, r.useRef)(null);
        var vn = (0, r.useRef)(undefined);
        var bn = (0, r.useRef)(0);
        var mn = (0, r.useState)((function() {
            return {
                activeItemId: (0, te.createDOMID)("active-item"),
                popoverId: (0, te.createDOMID)("popover")
            };
        })), gn = Ve(mn, 1), yn = gn[0], hn = yn.activeItemId, On = yn.popoverId;
        (0, r.useEffect)((function() {
            if (ae) {
                ke("");
                se(0);
            }
        }), [ Q, ae ]);
        // Tracks the prev activeIndex so its known if scrolling is needed in handleActiveOptionMount - can't use usePrevious b/c the value needs wiped on Menu close
                (0, r.useEffect)((function() {
            pn.current = ue;
        }), [ ue ]);
        var wn = (0, r.useCallback)((function(e) {
            var n = e.resetFilterKeyword;
            if (!Fe) {
                ke((function(e) {
                    return n ? "" : e;
                }));
                xe(true);
                qe(true);
                L === null || L === void 0 ? void 0 : L();
            }
        }), [ Fe, L ]);
        var Sn = (0, r.useCallback)((function() {
            if (Ne) {
                wn({
                    resetFilterKeyword: false
                });
            } else {
                var e;
                (e = fn.current) === null || e === void 0 ? void 0 : e.focus();
            }
        }), [ Ne, wn ]);
        var Cn = (0, r.useCallback)((function(e, n) {
            if (!Fe || nn) {
                return;
            }
            var r = (Ye || []).concat([ n ]);
            if (!ae) {
                _e(r);
                se(0);
                qe(true);
                ke("");
            }
            x === null || x === void 0 ? void 0 : x(e, {
                values: r,
                name: V
            });
        }), [ Fe, nn, Ye, ae, x, V ]);
        var Mn = (0, r.useCallback)((function(e, n) {
            var r = G()(Ye, n);
            if (!ae) {
                _e(r);
            }
            x === null || x === void 0 ? void 0 : x(e, {
                values: r,
                name: V
            });
        }), [ Ye, ae, V, x ]);
        var Pn = (0, r.useCallback)((function(e) {
            if (Fe && !C) {
                A === null || A === void 0 ? void 0 : A(e);
                Qe("".concat(bn.current, " options, loading more options"));
            }
        }), [ Fe, C, A ]);
        var kn = (0, r.useCallback)((function(e, n) {
            var r = n.value;
            I()((function() {
                return Mn(e, r);
            }));
        }), [ Mn ]);
        var jn = (0, r.useCallback)((function(e) {
            xe(true);
            if (nn) {
                return;
            }
            if (Pe !== "") {
                R === null || R === void 0 ? void 0 : R(e, {
                    keyword: ""
                });
            }
            wn({
                resetFilterKeyword: true
            });
        }), [ nn, Pe, R, wn ]);
        var Vn = (0, r.useCallback)((function(e) {
            var n;
            var r = bn.current;
            var t = e.key;
            if (nn) {
                return;
            }
            var o = vn.current === undefined ? undefined : un.get(vn.current);
            var a = Ue(o === null || o === void 0 ? void 0 : o.props.disabled);
            if (t === "Tab" && Fe) {
                if (!a && X && vn.current !== undefined && bn.current <= 1) {
                    var l;
                    e.preventDefault();
                    Cn(e, vn.current);
                    (l = fn.current) === null || l === void 0 ? void 0 : l.focus();
                } else {
                    qe(false);
                }
            }
            if (e.shiftKey || e.metaKey || e.ctrlKey) {
                return;
            }
            if (t === "End" && Fe) {
                e.preventDefault();
                if (r > 0) {
                    se(r - 1);
                }
            }
            if (t === "Home" && Fe) {
                e.preventDefault();
                se(0);
            }
            if (t === "ArrowDown") {
                e.preventDefault();
                if (Fe) {
                    if (r > 0) {
                        se((function(e) {
                            return Math.min(e + 1, r - 1);
                        }));
                    }
                } else {
                    se(0);
                    qe(true);
                }
                if (i && A) {
                    var u = tn.length - (2 + ((Ye === null || Ye === void 0 ? void 0 : Ye.length) || 0));
                    if (ue === u) {
                        Pn(e);
                    }
                }
            }
            if (t === "ArrowUp") {
                e.preventDefault();
                if (Fe) {
                    se((function(e) {
                        return Math.max(e - 1, 0);
                    }));
                } else {
                    se(0);
                    qe(true);
                }
            }
            if (t === "Enter" && !a && vn.current !== undefined && Fe) {
                e.preventDefault();
                Cn(e, vn.current);
            }
            if (t === "Backspace" && ((n = fn.current) === null || n === void 0 ? void 0 : n.value) === "" && Ye.length) {
                var s;
                for (var c = Ye.length - 1; c >= 0; c -= 1) {
                    var d = Ye[c];
                    if (!dn.has(d)) {
                        s = d;
                        break;
                    }
                }
                if (s !== undefined) {
                    Mn(e, s);
                }
            }
        }), [ nn, Fe, Ye, X, Cn, i, un, tn, A, ue, Pn, dn, Mn ]);
        var En = (0, r.useCallback)((function(e) {
            ke(e.target.value);
            qe(true);
            se(0);
            R === null || R === void 0 ? void 0 : R(e, {
                keyword: e.target.value
            });
        }), [ R ]);
        var Nn = (0, r.useCallback)((function(e, n) {
            var r;
            var t = n.value;
            e.stopPropagation();
            Cn(e, t);
            (r = fn.current) === null || r === void 0 ? void 0 : r.focus();
        }), [ Cn ]);
        // Close when component becomes disabled/dimmed during an open state to prevent further selection
                (0, r.useEffect)((function() {
            if (nn && Fe) {
                qe(false);
                pn.current = null;
                B === null || B === void 0 ? void 0 : B();
            }
        }), [ nn, Fe, B ]);
        var xn = (0, r.useCallback)((function(e) {
            var n = ye();
            var r = n.getElementById(On);
            var t = e.relatedTarget || n.activeElement;
 // IE11 doesn't support relatedTarget but sets activeElement
                        var o = r && t && r.contains(t);
 // TS: target as Node is an unsafe assumption
                        ke(o ? Pe : "");
            xe(false);
            if (!nn && !o) {
                R === null || R === void 0 ? void 0 : R(e, {
                    keyword: ""
                });
            }
        }), [ nn, Pe, R, On ]);
        var Bn = (0, r.useCallback)((function(e) {
            var n = e.reason, r = e.event;
            if (n === "escapeKey" || n === "offScreen" || n === "clickAway" && r !== undefined && !(ge === null || ge === void 0 ? void 0 : ge.contains(r.target))) {
                qe(false);
                pn.current = null;
                B === null || B === void 0 ? void 0 : B();
            }
            if (n === "escapeKey") {
                var t;
                (t = fn.current) === null || t === void 0 ? void 0 : t.focus();
            }
        }), [ ge, B ]);
        var Rn = (0, r.useCallback)((function(e) {
            he(e);
            we(m, e);
        }), [ m ]);
        var In = (0, r.useCallback)((function(e) {
            fn.current = e;
            we(S, e);
        }), [ S ]);
        var Fn = (0, r.useCallback)((function(e) {
            if (pn.current !== ue) {
                (0, oe.scrollIntoViewIfNeeded)(e);
            }
        }), [ ue ]);
        var qn = (0, r.useCallback)((function(e) {
            // selectedOptions may contain items or unmatched values at this point
            return e.map((function(e) {
                var n = e;
 // TS: faking props here
                                var r = n.props;
                if (r != null) {
                    var o = r.children, a = r.icon, l = r.label, i = r.selectedAppearance, u = r.selectedBackgroundColor, s = r.selectedForegroundColor, c = r.value;
                    
                    return t().createElement(be, {
                        "aria-selected": true,
                        disabled: !!nn || Ue(r.disabled),
                        icon: a,
                        key: c.toString(),
                        "data-test": "selected-option",
                        onRequestRemove: kn,
                        role: "option",
                        value: c,
                        appearance: i,
                        backgroundColor: u,
                        foregroundColor: s
                    }, o || l);
                }
                
                return t().createElement(be, {
                    "aria-selected": true,
                    disabled: nn,
                    key: (0, te.createGUID)(),
                    "data-test": "selected-option",
                    onRequestRemove: kn,
                    role: "option",
                    value: n
                }, n);
            }));
        }), [ nn, kn ]);
        var An = function e(n) {
            return n.map((function(e) {
                var n = e;
 // TS: faking props here
                                var r = n.props;
                if (r != null) {
                    
                    return t().createElement("span", {
                        key: r.value.toString()
                    }, r.children || r.label, (0, re.sprintf)((0, d._)(" selected")));
                }
                
                return t().createElement("span", {
                    key: (0, te.createGUID)()
                },  t().createElement(t().Fragment, null, n, (0, re.sprintf)((0, d._)(" selected"))));
            }));
        };
        // Filter the items
                var Dn = (0, ne.stringToKeywords)(Pe);
        var Ln = u ? tn : tn.filter((function(e) {
            if (q()(e, [ "props", "label" ], false)) {
                return (0, ne.testPhrase)(e.props.label, Dn);
            }
            return true;
 // Keep all headers and non-interactive options
                }));
        bn.current = 0;
        vn.current = undefined;
        var Tn = r.Children.map(Ln, (function(e, n) {
            var t = e.props;
            if (!t || !a()(t, "value")) {
                // ignore Headings and Dividers
                return e;
            }
            var o = e;
            var l = o.props, i = l.children, s = l.label, c = l.matchRanges, d = l.value;
 // find out if the search string exactly matches a value
                        if (d === Pe) {
                Ze = true;
            }
            // remove items that are already selected
                        var f = Ye.indexOf(d);
            if (f >= 0) {
                return null;
            }
            // highlight matched text
                        var p = i === undefined || D()(i);
            var v = p && !u && !c && Dn && (0, ne.keywordLocations)(s, Dn);
            // clone item
                        var b =  (0, r.cloneElement)(o, {
                // eslint-disable-next-line react/no-array-index-key
                key: n,
                onClick: Nn,
                matchRanges: p && (c || v) || undefined
            });
            return b;
        })).filter((function(e) {
            return e !== null;
        })) || [];
        // Add the option to add the new value
                if (n && !Ze && Pe) {
            Tn.unshift( t().createElement(h, {
                label: "".concat(Pe),
                description: (0, d._)("(New value)"),
                descriptionPosition: "right",
                value: Pe,
                key: "newValue",
                onClick: Nn
            }));
        }
        // highlight the selected Item
                var _n = r.Children.map(Tn, (function(e) {
            var n = e.props;
            if (!n || !a()(n, "value")) {
                // ignore Headings and Dividers
                return e;
            }
            var t = e;
            var o = t.props.value;
            var l = me(t.props.disabled, "disabled"), i = l.disabledValue;
            if (i === "disabled") {
                return t;
            }
            var u = bn.current === ue;
            bn.current += 1;
            if (!u) {
                return t;
            }
            vn.current = o;
            var s =  (0, r.cloneElement)(t, {
                active: u,
                id: hn,
                elementRef: function e(n) {
                    Fn(n);
                    we(t.props.elementRef, n);
                }
            });
            return s;
        }));
        var Hn = function e(n) {
            var r = n.anchorWidth, a = n.maxHeight, l = n.placement;
            var i = $e({
                anchorWidth: r,
                maxHeight: a,
                menuStyle: j
            });
            
            return t().createElement(Y(), {
                focusMode: "never",
                placement: l !== null && l !== void 0 ? l : undefined,
                isLoading: C,
                onScrollBottom: A ? Pn : undefined,
                noOptionsMessage: N,
                footerMessage: y,
                animateLoading: o,
                loadingMessage: P,
                onScroll: F,
                style: i
            }, _n);
        };
        var $n = He("".concat(Pe.length * .8, "em"));
        var zn = Ae(Ae({}, W()(Z, (function(e, n) {
            return n === "role" || n.indexOf("aria-") === 0;
        }))), {}, {
            "aria-describedby": v,
            "aria-labelledby": M,
            "aria-invalid": g || undefined
        });
        var Kn = Ae({}, W()(Z, (function(e, n) {
            return n === "aria-label" || n === "aria-describedby";
        })));
        
        return t().createElement(t().Fragment, null,  t().createElement(fe, Ce({
            key: "control",
            $append: l,
            "data-test-values": JSON.stringify(Ye),
            "data-test-disabled": rn,
            inline: O,
            $hasFocus: Ne
        }, z().apply(void 0, [ Z ].concat(Me(T()(zn)))), {
            onClick: nn ? undefined : Sn,
            $disabled: nn,
            $error: g,
            "data-test-popover-id": On,
            $popoverOpen: Fe,
            $prepend: $,
            flex: true,
            ref: Rn,
            role: "group",
            "aria-disabled": nn || undefined,
            "aria-labelledby": M
        }, Kn),  t().createElement(pe, {
            role: "listbox",
            "aria-invalid": g,
            "aria-multiselectable": "true"
        }, qn(cn)),  t().createElement(ve, Ce({
            role: "combobox",
            "data-test": "textbox",
            id: w,
            ref: In,
            onBlur: xn,
            onFocus: jn,
            onChange: nn ? undefined : En,
            onKeyDown: nn ? undefined : Vn,
            value: Pe,
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            spellCheck: false,
            readOnly: !!nn,
            style: $n,
            placeholder: Ye.length ? "" : H,
            required: K,
            disabled: rn === "disabled",
            "aria-disabled": rn === "dimmed" || undefined,
            "aria-activedescendant": Fe && bn.current > 0 ? hn : undefined,
            "aria-autocomplete": "list",
            "aria-controls": Fe ? On : undefined,
            "aria-owns": Fe ? On : undefined,
            "aria-expanded": Fe,
            "aria-haspopup": true
        }, zn)),  t().createElement(J(), {
            open: Fe && !!ge,
            autoCloseWhenOffScreen: true,
            anchor: ge,
            onRequestClose: Bn,
            canCoverAnchor: false,
            defaultPlacement: f,
            repositionMode: U,
            id: On
        }, Hn)),  t().createElement(ee(), {
            role: "status",
            "aria-relevant": "text"
        }, Je),  t().createElement(ee(), {
            role: "alert",
            "aria-relevant": "additions"
        }, An(cn)));
    }
    Ge.propTypes = _e;
    Ge.Option = h;
    Ge.Divider = u.Divider;
    Ge.Heading = u.Heading;
    /* harmony default export */    const Xe = Ge;
    // CONCATENATED MODULE: ./src/Multiselect/Multiselect.tsx
    function Je(e) {
        "@babel/helpers - typeof";
        return Je = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Je(e);
    }
    function Qe() {
        return Qe = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, Qe.apply(null, arguments);
    }
    function Ye(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function Ze(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? Ye(Object(r), !0).forEach((function(n) {
                en(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : Ye(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function en(e, n, r) {
        return (n = nn(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function nn(e) {
        var n = rn(e, "string");
        return "symbol" == Je(n) ? n : n + "";
    }
    function rn(e, n) {
        if ("object" != Je(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != Je(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function tn(e, n) {
        if (null == e) return {};
        var r, t, o = on(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function on(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */    var an = {
        allowNewValues: i().bool,
        animateLoading: i().bool,
        append: i().bool,
        children: i().node,
        compact: i().bool,
        controlledFilter: i().bool,
        defaultPlacement: i().oneOf([ "above", "below", "vertical" ]),
        defaultValues: i().array,
        describedBy: i().string,
        disabled: i().oneOfType([ i().bool, i().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        filter: i().oneOf([ false, true, "controlled" ]),
        footerMessage: i().node,
        inline: i().bool,
        inputId: i().string,
        inputRef: i().oneOfType([ i().func, i().object ]),
        isLoadingOptions: i().bool,
        labelledBy: i().string,
        loadingMessage: i().node,
        /** @private. */
        maxLabelItems: i().number,
        menuStyle: i().object,
        name: i().string,
        noOptionsMessage: i().node,
        onChange: i().func,
        onClose: i().func,
        onFilterChange: i().func,
        onOpen: i().func,
        onScroll: i().func,
        onScrollBottom: i().func,
        placeholder: i().string,
        prepend: i().bool,
        repositionMode: i().oneOf([ "none", "flip" ]),
        selectAllAppearance: i().oneOf([ "buttongroup", "checkbox", "none" ]),
        showSelectedValuesFirst: i().oneOf([ "nextOpen", "immediately", "never" ]),
        tabConfirmsNewValue: i().bool,
        values: i().array,
        /** @private */
        virtualization: i().number
    };
    var ln = (0, d._)("No matches");
    var un = (0, d._)("Select...");
    function sn(e) {
        var n = e.allowNewValues, o = e.animateLoading, a = e.append, l = e.compact, i = e.children, u = e.controlledFilter, s = e.defaultPlacement, d = s === void 0 ? "vertical" : s, f = e.defaultValues, p = e.describedBy, v = e.disabled, b = e.elementRef, m = e.error, g = e.filter, y = e.footerMessage, h = e.inline, O = e.inputId, w = e.inputRef, S = e.isLoadingOptions, C = e.labelledBy, M = e.loadingMessage, P = e.menuStyle, k = e.name, j = e.noOptionsMessage, V = j === void 0 ? ln : j, E = e.onChange, N = e.onClose, x = e.onFilterChange, R = e.onOpen, I = e.onScroll, F = e.onScrollBottom, q = e.placeholder, A = q === void 0 ? un : q, D = e.prepend, L = e.repositionMode, T = L === void 0 ? "flip" : L, _ = e.selectAllAppearance, H = e.showSelectedValuesFirst, $ = e.tabConfirmsNewValue, z = e.values, K = e.virtualization, W = tn(e, [ "allowNewValues", "animateLoading", "append", "compact", "children", "controlledFilter", "defaultPlacement", "defaultValues", "describedBy", "disabled", "elementRef", "error", "filter", "footerMessage", "inline", "inputId", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onChange", "onClose", "onFilterChange", "onOpen", "onScroll", "onScrollBottom", "placeholder", "prepend", "repositionMode", "selectAllAppearance", "showSelectedValuesFirst", "tabConfirmsNewValue", "values", "virtualization" ]);
        // @docs-props-type MultiselectPropsBase
                var U = c()({
            componentName: "Multiselect",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultValues",
            valuePropName: "values"
        });
        /* eslint-disable-next-line prefer-rest-params */        var G = arguments[0];
        (0, r.useEffect)((function() {
            if (l !== true) {
                if (false) {}
                if (false) {}
            }
        }), [ l, G ]);
        // Compact uses Button, which defaults `inline` to true, explicitly pass a boolean to prevent the underlying Button from applying its default.
                var X = !!h;
        var J = Ze({
            allowNewValues: n,
            animateLoading: o,
            append: a,
            children: i,
            controlledFilter: u,
            defaultPlacement: d,
            describedBy: p,
            disabled: v,
            elementRef: b,
            error: m,
            footerMessage: y,
            inline: X,
            inputId: O,
            inputRef: w,
            isLoadingOptions: S,
            labelledBy: C,
            loadingMessage: M,
            menuStyle: P,
            name: k,
            noOptionsMessage: V,
            onClose: N,
            onChange: E,
            onFilterChange: x,
            onOpen: R,
            onScroll: I,
            onScrollBottom: F,
            placeholder: A,
            prepend: D,
            repositionMode: T,
            showSelectedValuesFirst: H,
            tabConfirmsNewValue: $,
            virtualization: K
        }, W);
        var Q = (0, r.useMemo)((function() {
            return {
                defaultValues: f
            };
        }), [ f ]);
        var Y = (0, r.useMemo)((function() {
            return {
                values: z
            };
        }), [ z ]);
        var Z = (0, r.useMemo)((function() {
            return U ? Y : Q;
        }), [ Q, U, Y ]);
        var ee = {
            filter: g,
            selectAllAppearance: _,
            showSelectedValuesFirst: H,
            virtualization: K
        };
        if (l) {
            
            return t().createElement(B, Qe({
                "data-test": "multiselect"
            }, Z, ee, J));
        }
        
        return t().createElement(Xe, Qe({
            "data-test": "multiselect"
        }, Z, J));
    }
    // TODO: SUI-3402, this can be removed when ControlGroup no longer needs to inspect componentType
        sn.componentType = "Multiselect";
    sn.Option = h;
    sn.Heading = u.Heading;
    sn.Divider = u.Divider;
    sn.propTypes = an;
    /* harmony default export */    const cn = sn;
    // CONCATENATED MODULE: ./src/Multiselect/index.ts
    module.exports = n;
    /******/})();
