// @splunk/react-ui/Number.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ ce
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "lodash/isFinite"
    const o = require("lodash/isFinite");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const u = require("@splunk/react-ui/useControlled");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const s = require("@splunk/react-ui/usePrevious");
    var d =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const f = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const v = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "styled-components"
    const p = require("styled-components");
    var m =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronDown"
    const b = require("@splunk/react-icons/ChevronDown");
    var y =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronUp"
    const h = require("@splunk/react-icons/ChevronUp");
    var g =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const S = require("@splunk/react-ui/Clickable");
    var C =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const w = require("@splunk/react-ui/Divider");
    var O =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const k = require("@splunk/react-ui/Text");
    var j =  e.n(k);
    // CONCATENATED MODULE: external "@splunk/themes"
    const x = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Number/NumberStyles.ts
    var A = "24px";
    var B = m()(j()).withConfig({
        displayName: "NumberStyles__StyledText",
        componentId: "sc-2ekj9s-0"
    })([ "&[data-inline]{width:180px;}", "" ], (function(e) {
        var r = e.$startAdornment;
        return r ? (0, p.css)([ "padding-inline-start:0;" ]) : (0, p.css)([ "padding-inline-end:0;" ]);
    }));
    var D = m().div.withConfig({
        displayName: "NumberStyles__StyledControls",
        componentId: "sc-2ekj9s-1"
    })([ "width:", ";height:100%;", "" ], A, (function(e) {
        var r = e.$isAppend, n = e.$disabled, t = e.$error;
        var o;
        if (n) {
            o = x.variables.interactiveColorBorderDisabled;
        } else if (t) {
            o = x.variables.interactiveColorAccentError;
        } else {
            o = x.variables.interactiveColorBorder;
        }
        return (0, p.css)([ "", "" ], r ? (0, p.css)([ "border-inline-end:", " solid ", ";" ], x.variables.inputBorderWidth, o) : (0, 
        p.css)([ "border-inline-start:", " solid ", ";" ], x.variables.inputBorderWidth, o));
    }));
    var N = (0, p.css)([ "height:12px;width:12px;" ]);
    var P = m()(y()).withConfig({
        displayName: "NumberStyles__StyledChevronDown",
        componentId: "sc-2ekj9s-2"
    })([ "", ";" ], N);
    var T = m()(g()).withConfig({
        displayName: "NumberStyles__StyledChevronUp",
        componentId: "sc-2ekj9s-3"
    })([ "", ";" ], N);
    var _ = (0, p.css)([ "display:flex;align-items:center;justify-content:center;font-size:inherit;min-width:0;min-height:0;color:", ";height:calc(50% - 0.5px);width:100%;&:hover:not([disabled]){background-color:", ";}&:active:not([disabled]){background-color:", ";box-shadow:", ";}" ], x.variables.contentColorActive, x.variables.actionColorBackgroundSecondaryHover, x.variables.actionColorBackgroundSecondary, x.variables.focusShadowInset);
    var E = m()(C()).withConfig({
        displayName: "NumberStyles__StyledDecrementButton",
        componentId: "sc-2ekj9s-4"
    })([ "", ";border-top-left-radius:0;border-top-right-radius:0;" ], _);
    var I = m()(C()).withConfig({
        displayName: "NumberStyles__StyledIncrementButton",
        componentId: "sc-2ekj9s-5"
    })([ "", ";border-bottom-left-radius:0;border-bottom-right-radius:0;" ], _);
    var q = m()(O()).withConfig({
        displayName: "NumberStyles__StyledDivider",
        componentId: "sc-2ekj9s-6"
    })([ "border-color:", ";" ], (function(e) {
        var r = e.$disabled, n = e.$error;
        if (r) {
            return x.variables.interactiveColorBorderDisabled;
        }
        if (n) {
            return x.variables.interactiveColorAccentError;
        }
        return x.variables.interactiveColorBorder;
    }));
    // CONCATENATED MODULE: external "lodash/isString"
    const $ = require("lodash/isString");
    var K =  e.n($);
    // CONCATENATED MODULE: ./src/Number/utils.ts
    var M = function e(r) {
        // A more robust and expandable solution would be to use formatToParts() once IE requirements are dropped.
        var n = new Intl.NumberFormat(r.replace("_", "-")).format(1.2);
        return n.indexOf(",") > -1 ? "," : ".";
    };
    var R = function e(r) {
        var n = r.val, t = r.min, o = r.max, i = r.roundTo;
        var l = n;
        if (!a()(n)) {
            return l;
        }
        if (t !== undefined && a()(t)) {
            l = Math.max(t, l);
        }
        if (o !== undefined && a()(o)) {
            l = Math.min(o, l);
        }
        var u = Math.pow(10, i);
        return Math.round(l * u) / u;
    };
    var U = function e(r) {
        var n = r.str, t = r.locale, o = r.min, i = r.max, l = r.roundTo;
        var u = n;
        if (!K()(u) || u === "") {
            return undefined;
        }
        if (M(t) === ".") {
            // remove leading non-numeric characters.
            u = u.replace(/^[^\d.-]/, "");
        } else {
            // remove leading non-numeric characters and replace comma with period
            u = u.replace(/^[^\d,-]/, "").replace(",", ".");
        }
        var c = R({
            val: parseFloat(u),
            min: o,
            max: i,
            roundTo: l
        });
        if (a()(c)) {
            return c;
        }
        return undefined;
    };
    var F = function e(r, n) {
        if (r === undefined || !a()(r)) {
            return "";
        }
        if (M(n) === ".") {
            return r.toString();
        }
        // comma separated case
                return r.toString().replace(".", ",");
    };
    var V = function e(r) {
        var n = r.value, t = r.max;
        return n !== undefined && a()(n) && t !== undefined && a()(t) && n >= t;
    };
    var z = function e(r) {
        var n = r.value, t = r.min;
        return n !== undefined && a()(n) && t !== undefined && a()(t) && n <= t;
    };
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function W(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Number/Number.tsx
    function H(e) {
        "@babel/helpers - typeof";
        return H = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, H(e);
    }
    function G() {
        return G = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, G.apply(null, arguments);
    }
    function J(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function L(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? J(Object(n), !0).forEach((function(r) {
                Q(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : J(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function Q(e, r, n) {
        return (r = X(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function X(e) {
        var r = Y(e, "string");
        return "symbol" == H(r) ? r : r + "";
    }
    function Y(e, r) {
        if ("object" != H(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != H(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function Z(e, r) {
        return oe(e) || te(e, r) || re(e, r) || ee();
    }
    function ee() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function re(e, r) {
        if (e) {
            if ("string" == typeof e) return ne(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? ne(e, r) : void 0;
        }
    }
    function ne(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, t = Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function te(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var t, o, a, i, l = [], u = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (t = a.call(n)).done) && (l.push(t.value), l.length !== r); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function oe(e) {
        if (Array.isArray(e)) return e;
    }
    function ae(e, r) {
        if (null == e) return {};
        var n, t, o = ie(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function ie(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */
    /** @public */
    /** @public */    var le = {
        append: l().bool,
        children: l().node,
        defaultValue: l().number,
        describedBy: l().string,
        disabled: l().bool,
        elementRef: l().oneOfType([ l().func, l().object ]),
        inputRef: l().oneOfType([ l().func, l().object ]),
        error: l().bool,
        hideStepButtons: l().bool,
        inline: l().bool,
        inputId: l().string,
        labelledBy: l().string,
        locale: l().string,
        max: l().number,
        min: l().number,
        name: l().string,
        onBlur: l().func,
        onChange: l().func,
        onClick: l().func,
        onFocus: l().func,
        endAdornment: l().node,
        startAdornment: l().node,
        onKeyDown: l().func,
        onKeyUp: l().func,
        onSelect: l().func,
        prepend: l().bool,
        roundTo: l().number,
        step: l().number,
        value: l().number
    };
    function ue(e) {
        var r = e.append, o = e.defaultValue, i = e.describedBy, l = e.disabled, u = e.elementRef, s = e.error, p = e.hideStepButtons, m = e.inline, b = e.inputRef, y = e.locale, h = y === void 0 ? "en-US" : y, g = e.max, S = e.min, C = e.name, w = e.onBlur, O = e.onChange, k = e.onKeyDown, j = e.onKeyUp, x = e.prepend, A = e.roundTo, N = A === void 0 ? 5 : A, _ = e.step, $ = _ === void 0 ? 1 : _, K = e.endAdornment, H = e.startAdornment, J = e.value, Q = ae(e, [ "append", "defaultValue", "describedBy", "disabled", "elementRef", "error", "hideStepButtons", "inline", "inputRef", "locale", "max", "min", "name", "onBlur", "onChange", "onKeyDown", "onKeyUp", "prepend", "roundTo", "step", "endAdornment", "startAdornment", "value" ]);
        // @docs-props-type NumberPropsBase
                var X = c()({
            componentName: "Number",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var Y = X ? J : o;
        // stringValue stores the current contents of the input text box - it might be out of range, invalid, ...
                var ee = (0, n.useState)(F(Y, h)), re = Z(ee, 2), ne = re[0], te = re[1];
        var oe = (0, n.useState)(Y), ie = Z(oe, 2), le = ie[0], ue = ie[1];
        var ce = d()(J);
        var se = (0, n.useRef)(null);
        var de = (0, n.useCallback)((function(e) {
            W(se, e);
            W(b, e);
        }), [ b ]);
        var fe = (0, n.useCallback)((function(e, r) {
            if (r !== le || !X) {
                O === null || O === void 0 ? void 0 : O(e, {
                    value: r,
                    name: C,
                    reason: "stepButton"
                });
            }
            if (!X) {
                ue(r);
                te(F(r, h));
            }
        }), [ X, h, C, O, le ]);
        var ve = (0, n.useCallback)((function(e, r) {
            var n = U({
                str: ne,
                locale: h,
                min: S,
                max: g,
                roundTo: N
            });
            var t;
            // increment appropriate number
                        if (n !== undefined && a()(n)) {
                // increment the entered value
                t = n + r;
            } else if (o !== undefined && a()(o)) {
                // increment the defaultValue
                t = o + r;
            } else if (r > 0) {
                // increment up to max or step value
                t = g || $;
            } else {
                // increment down to min or 0
                t = S || 0;
 // set to min or 0
                        }
            fe(e, R({
                val: t,
                min: S,
                max: g,
                roundTo: N
            }));
        }), [ o, fe, h, g, S, N, $, ne ]);
        var pe = (0, n.useCallback)((function(e, r) {
            var n = R({
                val: parseFloat(r.value),
                min: S,
                max: g,
                roundTo: N
            });
            var t = F(n, h);
            te(t);
            w === null || w === void 0 ? void 0 : w(e, L(L({}, r), {}, {
                value: t
            }));
        }), [ h, g, S, w, N ]);
        var me = (0, n.useCallback)((function(e, r) {
            var n = r.value;
            var t = U({
                str: n,
                locale: h,
                min: S,
                max: g,
                roundTo: N
            });
            if (le !== t || !X) {
                O === null || O === void 0 ? void 0 : O(e, {
                    value: t,
                    name: C,
                    reason: "input"
                });
            }
            te(n);
            if (!X) {
                ue(t);
            }
        }), [ X, h, g, S, C, O, N, le ]);
        var be = (0, n.useCallback)((function(e) {
            var r = e.altKey, n = e.ctrlKey, t = e.key, o = e.nativeEvent, i = e.metaKey;
            if (!i && !n && !r && (0, v.addsCharacter)(o) !== false) {
                // only allow numbers, negative and decimals
                if (!(0, v.isNumeric)(o, {
                    locale: h
                })) {
                    e.preventDefault();
                }
                var l = S !== undefined && a()(S) && S >= 0;
                // if minus is not allowed or there is already one, prevent input
                                if ((0, v.isMinus)(o) && (l || ne.indexOf("-") >= 0)) {
                    e.preventDefault();
                }
                // if decimal is not allowed or there is already one, prevent input
                                if ((0, v.isDecimal)(o, {
                    locale: h
                }) && (N <= 0 || ne.indexOf(M(h)) >= 0)) {
                    e.preventDefault();
                }
            }
            // Arrow up and down will increment
                        if (t === "ArrowUp") {
                ve(e, $);
                e.preventDefault();
            } else if (t === "ArrowDown") {
                ve(e, -$);
                e.preventDefault();
            }
            k === null || k === void 0 ? void 0 : k(e);
        }), [ ve, h, S, k, N, $, ne ]);
        var ye = (0, n.useCallback)((function(e) {
            var r;
            if (e.key === "Enter" && (se === null || se === void 0 ? void 0 : (r = se.current) === null || r === void 0 ? void 0 : r.value)) {
                var n = se.current.value;
                var t = R({
                    val: parseFloat(n),
                    min: S,
                    max: g,
                    roundTo: N
                });
                te(F(t, h));
            }
            j === null || j === void 0 ? void 0 : j(e);
        }), [ h, g, S, j, N ]);
        var he = (0, n.useCallback)((function(e) {
            ve(e, $);
        }), [ ve, $ ]);
        var ge = (0, n.useCallback)((function(e) {
            ve(e, -$);
        }), [ ve, $ ]);
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ r, x ]);
        (0, n.useEffect)((function() {
            // if isControlled (value prop passed), keep local valueNumeric up to date with value prop
            if (X) {
                ue(J);
            }
        }), [ X, J ]);
        (0, n.useEffect)((function() {
            // This useEffect has multiple dependencies, but we only want to run this code if the `value` prop has changed.
            if (J !== ce) {
                // on value (number) prop change compare the new value to the number version of stringValue (state)
                // This is because we pass in the number version of stringValue to the onChange handler
                // Thus if these values match we want to ignore the update as that is the current value based on stringValue
                var e = U({
                    str: ne,
                    locale: h,
                    min: S,
                    max: g,
                    roundTo: N
                });
                if (e !== J && X) {
                    te(F(J, h));
                }
            }
        }), [ X, h, g, S, ce, N, J, ne ]);
        var Se = {
            append: r,
            prepend: !r,
            error: s,
            inline: false,
            tabIndex: -1
        };
        var Ce = V({
            value: le,
            max: g
        });
        var we = z({
            value: le,
            min: S
        });
        var Oe = r && !x;
        var ke =  t().createElement(D, {
            $isAppend: Oe,
            $disabled: l,
            $error: s
        },  t().createElement(I, G({
            "aria-label": (0, f._)("Increment"),
            "data-test": "increment",
            disabled: l || Ce,
            onClick: he
        }, Se),  t().createElement(T, null)),  t().createElement(q, {
            $disabled: l,
            $error: s,
            decorative: true
        }),  t().createElement(E, G({
            "aria-label": (0, f._)("Decrement"),
            "data-test": "decrement",
            disabled: l || we,
            onClick: ge
        }, Se),  t().createElement(P, null)));
        var je = Oe && !p && ke;
        var xe = !Oe && !p && ke;
        var Ae = H || je;
        
        return t().createElement(B, G({
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            "data-test": "number",
            elementRef: u,
            inputRef: de,
            disabled: l,
            name: C,
            describedBy: i,
            append: r,
            prepend: x
        }, Q, {
            spellCheck: false,
            inline: m,
            onBlur: pe,
            onChange: me,
            onKeyDown: be,
            onKeyUp: ye,
            value: ne,
            "aria-describedby": i,
            error: s,
            "data-test-value": le,
            startAdornment: Ae,
            endAdornment: K || xe,
            $startAdornment: !!Ae,
            role: "spinbutton",
            "aria-valuenow": le,
            "aria-valuemin": S,
            "aria-valuemax": g,
            "aria-invalid": s
        }));
    }
    ue.displayName = "Number";
    ue.propTypes = le;
    /* harmony default export */    const ce = ue;
    // only for styled components using Number
        // CONCATENATED MODULE: ./src/Number/index.ts
    module.exports = r;
    /******/})();
