// @splunk/react-ui/Paginator.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = a => {
            /******/ var n = a && a.__esModule ? 
            /******/ () => a["default"]
            /******/ : () => a
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (a, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(a, r)) {
                    /******/ Object.defineProperty(a, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, a) => Object.prototype.hasOwnProperty.call(e, a)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var a = {};
    // ESM COMPAT FLAG
        e.r(a);
    // EXPORTS
        e.d(a, {
        Compact: () => /* reexport */ L,
        PageControl: () => /* reexport */ Y,
        default: () => /* reexport */ ie
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const t = require("prop-types");
    var l =  e.n(t);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronLeft"
    const o = require("@splunk/react-icons/ChevronLeft");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const u = require("@splunk/react-icons/ChevronRight");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const s = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const p = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var g =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const d = require("@splunk/react-ui/Button");
    var v =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Paginator/ButtonStyles.ts
    var m = g()(v()).withConfig({
        displayName: "ButtonStyles__StyledButton",
        componentId: "sc-1neztq7-0"
    })([ "flex-shrink:0;font-weight:", ";&:not([disabled],[aria-disabled='true']){&:not([aria-current]){color:", ";}&[aria-current='false']{color:", ";}&[aria-current='page']{color:", ";border-color:", ";font-weight:", ";}&:hover{color:", ";}}&:is([disabled]),&:is([aria-disabled='true']){color:", ";}" ], b.variables.fontWeightNormal, b.variables.contentColorActive, b.variables.contentColorDefault, b.variables.contentColorActive, b.variables.contentColorAccent, b.variables.fontWeightBold, b.variables.contentColorAccentStrong, b.variables.contentColorDisabled);
    // CONCATENATED MODULE: ./src/Paginator/Button.tsx
    function y() {
        return y = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var n = arguments[a];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, y.apply(null, arguments);
    }
    function h(e, a) {
        if (null == e) return {};
        var n, r, t = P(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                n = l[r], -1 === a.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function P(e, a) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== a.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var O = {
        children: l().node,
        disabled: l().bool,
        onClick: l().func,
        page: l().number
    };
    function C(e) {
        var a = e.children, t = e.disabled, l = e.onClick, o = e.page, i = h(e, [ "children", "disabled", "onClick", "page" ]);
        // @docs-props-type PaginatorButtonPropsBase
                var u = (0, n.useCallback)((function(e) {
            // Button can be a <button> or <a> tag depending on props but its callback always claims to be either
            // this causes type problems, and we know we're always getting a <button> since we don't pass the "to" prop
            // so override the type here to prevent type issues in components using this one
            l === null || l === void 0 ? void 0 : l(e, {
                page: o
            });
        }), [ l, o ]);
        
        return r().createElement(m, y({
            "data-test": "page",
            "data-test-page": o,
            appearance: "subtle",
            disabled: t && "dimmed",
            onClick: u
        }, i), a);
    }
    C.propTypes = O;
    /* harmony default export */    const S = C;
    // CONCATENATED MODULE: ./src/Paginator/CompactStyles.ts
    var w = g().nav.withConfig({
        displayName: "CompactStyles__StyledNav",
        componentId: "dwgzkj-0"
    })([ "", " flex:0 1 auto;flex-wrap:nowrap;margin-left:auto;gap:", ";" ], b.mixins.reset("inline-flex"), b.variables.spacingXSmall);
    var k = g().span.withConfig({
        displayName: "CompactStyles__StyledLabel",
        componentId: "dwgzkj-1"
    })([ "align-items:center;display:flex;flex-wrap:wrap;margin-right:", ";" ], b.variables.spacingXSmall);
    // CONCATENATED MODULE: ./src/Paginator/Compact.tsx
    function E() {
        return E = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var n = arguments[a];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, E.apply(null, arguments);
    }
    function _(e, a) {
        if (null == e) return {};
        var n, r, t = x(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                n = l[r], -1 === a.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function x(e, a) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== a.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */
    /** @public */    var j = {
        current: l().number,
        elementRef: l().oneOfType([ l().func, l().object ]),
        onChange: l().func,
        renderLabel: l().oneOfType([ l().bool, l().func ]),
        totalPages: l().number
    };
    var N = function e(a) {
        var n = a.current, r = a.totalPages;
        var t = null;
        if (n != null && r != null) {
            t = (0, s.sprintf)((0, p._)("%d of %d pages"), n, r);
        } else if (n != null) {
            t = (0, s.sprintf)((0, p._)("Page %d"), n);
        }
        return t;
    };
    function I(e) {
        var a = e.onChange, t = e.current, l = e.elementRef, o = e.renderLabel, u = e.totalPages, s = _(e, [ "onChange", "current", "elementRef", "renderLabel", "totalPages" ]);
        // @docs-props-type PaginatorCompactPropsBase
                var f = (0, n.useCallback)((function(e, n) {
            var r = n.page;
            a === null || a === void 0 ? void 0 : a(e, {
                direction: "prev",
                page: r
            });
        }), [ a ]);
        var g = (0, n.useCallback)((function(e, n) {
            var r = n.page;
            a === null || a === void 0 ? void 0 : a(e, {
                direction: "next",
                page: r
            });
        }), [ a ]);
        var d = null;
        if (o) {
            var v = typeof o === "function" ? o : N;
            d = v({
                current: t,
                totalPages: u
            });
        }
        
        return r().createElement(w, E({
            "aria-label": (0, p._)("Pagination"),
            "data-test": "paginator",
            "data-test-current": t,
            ref: l
        }, s), d &&  r().createElement(k, {
            "data-test": "label"
        }, d),  r().createElement(S, {
            "aria-label": (0, p._)("Go to previous page"),
            "data-test": "prev",
            disabled: t != null && t <= 1,
            onClick: f,
            page: t != null ? t - 1 : undefined
        },  r().createElement(i(), null)),  r().createElement(S, {
            "aria-label": (0, p._)("Go to next page"),
            "data-test": "next",
            disabled: t != null && u != null && t >= u,
            onClick: g,
            page: t != null ? t + 1 : undefined
        },  r().createElement(c(), null)));
    }
    I.propTypes = j;
    /* harmony default export */    const L = I;
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const q = require("@splunk/react-ui/ScreenReaderContent");
    var R =  e.n(q);
    // CONCATENATED MODULE: external "@splunk/react-ui/Select"
    const T = require("@splunk/react-ui/Select");
    var M =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const D = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/Paginator/PageSelectStyles.ts
    var z = g().span.withConfig({
        displayName: "PageSelectStyles__StyledWrapper",
        componentId: "haycz4-0"
    })([ "display:flex;" ]);
    var G = g().span.withConfig({
        displayName: "PageSelectStyles__StyledCount",
        componentId: "haycz4-1"
    })([ "align-items:center;display:flex;flex-wrap:wrap;margin:0 ", ";" ], b.variables.spacingSmall);
    var A = g()(M()).withConfig({
        displayName: "PageSelectStyles__StyledSelect",
        componentId: "haycz4-2"
    })([ "flex:0 0 auto;" ]);
    // CONCATENATED MODULE: ./src/Paginator/PageSelect.tsx
    function B(e, a) {
        if (null == e) return {};
        var n, r, t = W(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                n = l[r], -1 === a.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function W(e, a) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== a.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */    var X = {
        onChange: l().func,
        current: l().number,
        totalPages: l().number
    };
    function V(e) {
        var a = e.onChange, t = e.current, l = t === void 0 ? 1 : t, o = e.totalPages, i = B(e, [ "onChange", "current", "totalPages" ]);
        // @docs-props-type PaginatorPageSelectPropsBase
                var u = (0, n.useRef)((0, D.createDOMID)("pages"));
        var c = (0, n.useRef)((0, D.createDOMID)("totalPages"));
        var f = function e(n, r) {
            var t = r.value;
            a === null || a === void 0 ? void 0 : a(n, {
                page: t
            });
        };
        var g = [];
        for (var d = 1; d <= o; d += 1) {
            g.push( r().createElement(M().Option, {
                label: "".concat(d),
                key: "".concat(d),
                value: d
            }));
        }
        var v = (0, s.sprintf)((0, p._)("of %d pages"), o);
        
        return r().createElement(z, i,  r().createElement(A, {
            value: l,
            onChange: f,
            "aria-labelledby": "".concat(u.current, " ").concat(c.current)
        }, g),  r().createElement(R(), {
            id: u.current,
            "aria-hidden": true
        }, l),  r().createElement(G, {
            id: c.current,
            "aria-hidden": true
        }, " ", v));
    }
    V.propTypes = X;
    /* harmony default export */    const F = V;
    // CONCATENATED MODULE: ./src/Paginator/PageControl.tsx
    function H() {
        return H = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var n = arguments[a];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, H.apply(null, arguments);
    }
    function J(e, a) {
        if (null == e) return {};
        var n, r, t = K(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                n = l[r], -1 === a.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function K(e, a) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== a.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */    var Q = {
        elementRef: l().oneOfType([ l().func, l().object ]),
        onChange: l().func,
        current: l().number,
        totalPages: l().number
    };
    function U(e) {
        var a = e.onChange, t = e.current, l = t === void 0 ? 1 : t, o = e.totalPages, i = J(e, [ "onChange", "current", "totalPages" ]);
        // @docs-props-type PaginatorPageControlPropsBase
                var u = (0, n.useCallback)((function(e, n) {
            var r = n.page;
            // page will always be defined because we're passing "current" to Compact
            // getting TypeScript to understand this seems more trouble than it's worth
            // so we'll validate this behavior through tests
                        a === null || a === void 0 ? void 0 : a(e, {
                page: r
            });
        }), [ a ]);
        var c = (0, n.useCallback)((function(e) {
            var n = e.current, t = e.totalPages;
            if (n != null && t != null) {
                
                return r().createElement(F, {
                    current: n,
                    onChange: a,
                    totalPages: t
                });
            }
            return null;
        }), [ a ]);
        
        return r().createElement(L, H({
            current: l,
            renderLabel: c,
            onChange: u,
            totalPages: o
        }, i));
    }
    U.propTypes = Q;
    /* harmony default export */    const Y = U;
    // CONCATENATED MODULE: ./src/Paginator/PageSeparatorStyles.ts
    var Z = g().span.withConfig({
        displayName: "PageSeparatorStyles__StyledEllipsisWrapper",
        componentId: "j50fqf-0"
    })([ "display:flex;align-items:center;cursor:default;padding:", " ", ";" ], b.variables.spacingXSmall, b.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/Paginator/PageSeparator.tsx
    function $() {
        
        return r().createElement(Z, {
            role: "separator"
        }, "…");
    }
    /* harmony default export */    const ee = $;
    // CONCATENATED MODULE: ./src/Paginator/PaginatorStyles.ts
    var ae = g().nav.withConfig({
        displayName: "PaginatorStyles__StyledNav",
        componentId: "pmabsp-0"
    })([ "", " gap:", ";" ], b.mixins.reset("inline-flex"), b.variables.spacingXSmall);
    // CONCATENATED MODULE: ./src/Paginator/Paginator.tsx
    function ne() {
        return ne = Object.assign ? Object.assign.bind() : function(e) {
            for (var a = 1; a < arguments.length; a++) {
                var n = arguments[a];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, ne.apply(null, arguments);
    }
    function re(e, a) {
        if (null == e) return {};
        var n, r, t = te(e, a);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (r = 0; r < l.length; r++) {
                n = l[r], -1 === a.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (t[n] = e[n]);
            }
        }
        return t;
    }
    function te(e, a) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== a.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */
    /** @public */    var le = {
        alwaysShowLastPageLink: l().bool,
        generatePageProps: l().func,
        current: l().number,
        elementRef: l().oneOfType([ l().func, l().object ]),
        numPageLinks: l().number,
        onChange: l().func,
        totalPages: l().number
    };
    function oe(e) {
        var a = e.alwaysShowLastPageLink, t = a === void 0 ? false : a, l = e.generatePageProps, o = e.current, u = o === void 0 ? 1 : o, f = e.numPageLinks, g = f === void 0 ? 5 : f, d = e.onChange, v = e.totalPages, b = e.elementRef, m = re(e, [ "alwaysShowLastPageLink", "generatePageProps", "current", "numPageLinks", "onChange", "totalPages", "elementRef" ]);
        // @docs-props-type PaginatorPropsBase
                var y = (0, n.useCallback)((function(e, a) {
            var n = a.page;
            // page will always be defined because we're passing "page" to PaginatorButton
            // getting TypeScript to understand this seems more trouble than it's worth
            // so we'll validate this behavior through tests
                        d === null || d === void 0 ? void 0 : d(e, {
                page: n
            });
        }), [ d ]);
        // Can't show more links than total number of pages.
                var h = Math.min(g, v);
        var P = h;
        var O = v - h + 1;
        var C = [];
        if (v <= 1) {
            return null;
        }
        if ((u > v || u < 1) && process.env.NODE_ENV !== "production") {}
        /**
   * Paginator shows numPageLinks page buttons that are nearby the current page.
   * firstPageNearbyCurrent and lastPageNearbyCurrent refer to the page number of the first and last of these buttons.
   */        var w;
        var k;
        if (u < P) {
            w = 1;
            k = h;
        } else if (u >= P && u <= O) {
            w = u - Math.floor((h - 2) / 2);
            k = u + 1 === v ? v : u + Math.ceil((h - 2) / 2);
        } else {
            w = O;
            k = v;
        }
        // bump up the last page to avoid ellipsis
                k = t && k + 1 === v ? v : k;
        // check to avoid ellipsis between 1 and 2
                if (w === 2) {
            w = 1;
        }
        var E = u - 1;
        C.push( r().createElement(S, ne({}, l === null || l === void 0 ? void 0 : l({
            page: E
        }), {
            "aria-label": (0, p._)("Go to previous page"),
            "data-test": "prev",
            disabled: u === 1,
            key: "prev",
            onClick: y,
            page: E
        }),  r().createElement(i(), null), (0, p._)("Prev")));
        if (u >= P && v > h && w !== 1) {
            var _ = 1;
            C.push( r().createElement(S, ne({}, l === null || l === void 0 ? void 0 : l({
                page: _
            }), {
                "aria-label": (0, p._)("Go to first page"),
                key: "first",
                onClick: y,
                page: _
            }), "1"));
            C.push( r().createElement(ee, {
                "data-test": "page",
                key: "prevEllipsis"
            }));
        }
        for (var x = w; x <= k; x += 1) {
            var j = x === u;
            C.push( r().createElement(S, ne({}, l === null || l === void 0 ? void 0 : l({
                page: x
            }), {
                "aria-current": j ? "page" : false,
                "aria-label": (0, s.sprintf)((0, p._)("Page %(pageNumber)d"), {
                    pageNumber: x
                }),
                key: x,
                onClick: y,
                selected: j,
                page: x
            }), String(x)));
        }
        if (v > h && k !== v) {
            C.push( r().createElement(ee, {
                "data-test": "page",
                key: "nextEllipsis"
            }));
            if (t) {
                C.push( r().createElement(S, ne({}, l === null || l === void 0 ? void 0 : l({
                    page: v
                }), {
                    "aria-label": (0, p._)("Go to last page"),
                    "data-test": "last",
                    key: "last",
                    onClick: y,
                    page: v
                }), String(v)));
            }
        }
        var N = u + 1;
        C.push( r().createElement(S, ne({}, l === null || l === void 0 ? void 0 : l({
            page: N
        }), {
            "aria-label": (0, p._)("Go to next page"),
            "data-test": "next",
            disabled: u === v,
            key: "next",
            onClick: y,
            page: N
        }), (0, p._)("Next"),  r().createElement(c(), null)));
        
        return r().createElement(ae, ne({
            "aria-label": (0, p._)("Pagination"),
            "data-test": "paginator",
            "data-test-current": u,
            ref: b
        }, m), C);
    }
    oe.Compact = L;
    oe.PageControl = Y;
    oe.propTypes = le;
    /* harmony default export */    const ie = oe;
    // CONCATENATED MODULE: ./src/Paginator/index.ts
    module.exports = a;
    /******/})();
