// @splunk/react-ui/Popover.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        PopoverContext: () => /* reexport */ de,
        PopoverMenuContext: () => /* reexport */ pt,
        PopoverProvider: () => /* reexport */ me,
        default: () => /* reexport */ st
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "lodash/clamp"
    const o = require("lodash/clamp");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/defer"
    const i = require("lodash/defer");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/every"
    const c = require("lodash/every");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "lodash/has"
    const f = require("lodash/has");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "lodash/intersection"
    const p = require("lodash/intersection");
    var v =  e.n(p);
    // CONCATENATED MODULE: external "lodash/isFinite"
    const d = require("lodash/isFinite");
    var h =  e.n(d);
    // CONCATENATED MODULE: external "lodash/isFunction"
    const m = require("lodash/isFunction");
    var b =  e.n(m);
    // CONCATENATED MODULE: external "lodash/omit"
    const g = require("lodash/omit");
    var y =  e.n(g);
    // CONCATENATED MODULE: external "lodash/throttle"
    const w = require("lodash/throttle");
    var P =  e.n(w);
    // CONCATENATED MODULE: external "prop-types"
    const S = require("prop-types");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const E = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const C = require("@splunk/react-ui/EventListener");
    var j =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/Layer"
    const x = require("@splunk/react-ui/Layer");
    var k =  e.n(x);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScrollContainerContext"
    const A = require("@splunk/react-ui/ScrollContainerContext");
    var T =  e.n(A);
    // CONCATENATED MODULE: external "@splunk/themes/useSplunkTheme"
    const M = require("@splunk/themes/useSplunkTheme");
    var R =  e.n(M);
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const W = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: ./src/Popover/getPlacement.ts
    function H(e) {
        "@babel/helpers - typeof";
        return H = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, H(e);
    }
    function q(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function I(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? q(Object(r), !0).forEach((function(t) {
                N(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : q(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function N(e, t, r) {
        return (t = L(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function L(e) {
        var t = D(e, "string");
        return "symbol" == H(t) ? t : t + "";
    }
    function D(e, t) {
        if ("object" != H(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != H(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function F(e) {
        var t = e.align, r = e.anchorPos, n = e.outerContainerEl, o = e.padding, a = e.placement;
        switch (a) {
          case "above":
            return {
                top: r.top - n.offsetHeight,
                left: t === "edge" ? r.left - o : r.middle - n.offsetWidth / 2
            };

          case "below":
            // Align 'end' is only supported by below
            // TODO: SUI-5101 Enable users to align Popover and Dropdown
            if (t === "end") {
                return {
                    top: r.bottom,
                    left: r.right - n.offsetWidth + o
                };
            }
            return {
                top: r.bottom,
                left: t === "edge" ? r.left - o : r.middle - n.offsetWidth / 2
            };

          case "left":
            return {
                top: t === "edge" ? r.top - o : r.center - n.offsetHeight / 2,
                left: r.left - n.offsetWidth
            };

          case "right":
            return {
                top: t === "edge" ? r.top - o : r.center - n.offsetHeight / 2,
                left: r.right
            };

          default:
            throw new Error("".concat(a, " is not a valid placement value. Valid options are: 'above', 'below', 'left', or 'right'"));
        }
    }
    var $ = function e(t) {
        var r = t.align, n = t.anchorPos, o = t.outerContainerEl, a = t.padding;
        if (r === "edge") {
            return n.left + o.offsetWidth - a;
        }
        if (r === "end") {
            return n.right + a;
        }
        return n.middle + o.offsetWidth / 2;
    };
    var _ = function e(t) {
        var r = t.align, n = t.anchorPos, o = t.outerContainerEl, a = t.padding;
        if (r === "edge") return n.left - a;
        if (r === "end") return n.right - o.offsetWidth + a;
        return n.middle - o.offsetWidth / 2;
    };
    var V = function e(t) {
        var r = t.align, n = t.anchorPos, o = t.outerContainerEl, a = t.padding;
        if (r === "edge") return n.top - a;
        return n.top - o.offsetHeight / 2;
    };
    var z = function e(t) {
        var r = t.align, n = t.anchorPos, o = t.outerContainerEl, a = t.padding;
        if (r === "edge") return n.top + o.offsetHeight - a;
        return n.bottom + o.offsetHeight / 2;
    };
    function K(e) {
        var t = e.align, r = e.anchorPos, n = e.scrollContainerPos, o = e.canCoverAnchor, a = e.defaultPlacement, i = e.outerContainerEl, l = e.padding, c = e.repositionMode, u = e.windowHeight, f = e.windowWidth;
        var s = c === "flip";
        var p = c === "any";
        var v = s || p;
        var d = e.placement || a;
        // Translate vertical/horizontal to above/below/left/right
                if (a === "vertical") {
            d = r.top > u - r.bottom ? "above" : "below";
        } else if (a === "horizontal") {
            d = r.left > f - r.right ? "left" : "right";
        }
        var h = d;
        // Initialize the result variables. These will be mutated as needed and returned.
                var m = F({
            align: t,
            anchorPos: r,
            outerContainerEl: i,
            padding: l,
            placement: h
        }), b = m.top, g = m.left;
        var y = "auto";
        var w = f;
        var P = u;
        var S = h;
        // Boolean convenience variables to simplify positioning logic.
                var O = r.top - i.offsetHeight > 0;
        var E = r.bottom + i.offsetHeight < u;
        var C = r.left - i.offsetWidth > 0;
        var j = r.right + i.offsetWidth < f;
        var x = r.top * .7 > u - r.bottom;
        var k = $({
            align: t,
            anchorPos: r,
            outerContainerEl: i,
            padding: l
        }) > f;
        var A = _({
            align: t,
            anchorPos: r,
            outerContainerEl: i,
            padding: l
        }) < 0;
        var T = V({
            align: t,
            anchorPos: r,
            outerContainerEl: i,
            padding: l
        }) < 0;
        var M = z({
            align: t,
            anchorPos: r,
            outerContainerEl: i,
            padding: l
        }) > u;
        // Handle each of the four placement options individually.
                if (S === "above") {
            if (!O && v) {
                if (E) {
                    return K(I(I({}, e), {}, {
                        placement: "below"
                    }));
                }
                if (p && j) {
                    return K(I(I({}, e), {}, {
                        placement: "right"
                    }));
                }
                if (p && C) {
                    return K(I(I({}, e), {}, {
                        placement: "left"
                    }));
                }
                if (o) {
                    S = "misaligned";
                    b = 0;
                }
            }
            if (S !== "misaligned") {
                y = u - b - i.offsetHeight;
                if (n) {
                    y = Math.min(y, u - n.top);
                }
                b = "auto";
            }
            if (k) {
                g = Math.max(f - i.offsetWidth, 0);
            } else if (A) {
                g = 0;
            }
            if (!o) {
                P = r.top;
            }
        }
        if (S === "below") {
            if (!E && v) {
                if (O && x) {
                    return K(I(I({}, e), {}, {
                        placement: "above"
                    }));
                }
                if (p && j) {
                    return K(I(I({}, e), {}, {
                        placement: "right"
                    }));
                }
                if (p && C) {
                    return K(I(I({}, e), {}, {
                        placement: "left"
                    }));
                }
                if (o) {
                    S = "misaligned";
                    b = 0;
                }
            }
            if (n) {
                b = Math.min(b, n.bottom);
            }
            if (k) {
                g = Math.max(f - i.offsetWidth, 0);
            } else if (A) {
                g = 0;
            }
            if (!o) {
                P = u - r.bottom;
            }
        }
        if (S === "left") {
            if (!C && v) {
                if (j) {
                    return K(I(I({}, e), {}, {
                        placement: "right"
                    }));
                }
                if (p && E) {
                    return K(I(I({}, e), {}, {
                        placement: "below"
                    }));
                }
                if (p && O) {
                    return K(I(I({}, e), {}, {
                        placement: "above"
                    }));
                }
                if (o) {
                    S = "misaligned";
                    b = 0;
                }
            }
            if (T) {
                b = 0;
            } else if (M) {
                b = Math.max(u - i.offsetHeight, 0);
            }
            if (!o) {
                w = r.left;
            }
        }
        if (S === "right") {
            if (!j && v) {
                if (C) {
                    return K(I(I({}, e), {}, {
                        placement: "left"
                    }));
                }
                if (p && E) {
                    return K(I(I({}, e), {}, {
                        placement: "below"
                    }));
                }
                if (p && O) {
                    return K(I(I({}, e), {}, {
                        placement: "above"
                    }));
                }
                if (o) {
                    S = "misaligned";
                    b = 0;
                }
            }
            if (T) {
                b = 0;
            } else if (M) {
                b = Math.max(u - i.offsetHeight, 0);
            }
            if (!o) {
                w = f - r.left;
            }
        }
        // SUI-8136: Set container width as min-width if it fits within viewport, otherwise allow it to shrink to fit
                var R = (S === "above" || S === "below") && i.offsetWidth <= w ? i.offsetWidth : 0;
        return {
            placement: S,
            maxHeight: P,
            maxWidth: w,
            outerContainerStyle: {
                top: b,
                left: g,
                bottom: y,
                minWidth: R
            }
        };
    }
    // CONCATENATED MODULE: external "@react-spring/web"
    const X = require("@react-spring/web");
    // CONCATENATED MODULE: external "styled-components"
    const B = require("styled-components");
    var Y =  e.n(B);
    // CONCATENATED MODULE: external "@splunk/themes"
    const U = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Popover/PopoverStyles.ts
    var G = 8;
    // this works around SUI-7977 and SUI-8062 by insetting the arrow into the box
    // such that a 1px gap does not appear between the arrow and the box when the Popover is resized or zoomed
        var J = 1;
    var Q = Y()(X.animated.div).withConfig({
        displayName: "PopoverStyles__Styled",
        componentId: "sc-1nahsvw-0"
    })([ "position:fixed;z-index:", ";left:-300%;top:-300%;" ], U.variables.zindexPopover);
    // These hex codes are the inverted values of variables.backgroundColorPopup which are not exposed as tokens
        var Z = (0, B.css)([ "", "" ], (0, U.pick)({
        light: "#27292e",
        dark: U.variables.white
    }));
    var ee = Y().div.withConfig({
        displayName: "PopoverStyles__StyledBox",
        componentId: "sc-1nahsvw-1"
    })([ "", ";", ";", ";" ], U.mixins.reset("block"), (function(e) {
        var t = e.$open;
        return !t && (0, B.css)([ "pointer-events:none;" ]);
    }), (function(e) {
        var t = e.$appearance, r = e.$boxPadding;
        return t !== "none" && (0, B.css)([ "padding:", "px;" ], r);
    }));
    var te = Y().div.withConfig({
        displayName: "PopoverStyles__StyledContent",
        componentId: "sc-1nahsvw-2"
    })([ "border-radius:", ";background-color:", ";color:", ";", "" ], U.variables.borderRadius, (0, 
    U.pickVariant)("$appearance", {
        normal: U.variables.backgroundColorPopup,
        inverted: Z
    }), (0, U.pickVariant)("$appearance", {
        normal: U.variables.contentColorDefault,
        inverted: U.variables.contentColorInverted
    }), (0, U.pickVariant)("$appearance", {
        normal: (0, B.css)([ "box-shadow:", ";" ], U.variables.overlayShadow)
    }));
    var re = Y().div.withConfig({
        displayName: "PopoverStyles__StyledArrow",
        componentId: "sc-1nahsvw-3"
    })([ "width:0;height:0;border-left:calc(", "px + ", "px) solid transparent;border-right:calc(", "px + ", "px) solid transparent;position:absolute;border-bottom-width:calc(", "px + ", "px);border-bottom-style:solid;", ";" ], G, J, G, J, G, J, (function(e) {
        var t = e.$appearance;
        return t !== "none" && (0, B.css)([ "border-bottom-color:", ";" ], (0, U.pickVariant)("$appearance", {
            normal: U.variables.backgroundColorPopup,
            inverted: Z
        }));
    }));
    var ne = Y().svg.withConfig({
        displayName: "PopoverStyles__StyledHitArea",
        componentId: "sc-1nahsvw-4"
    })([ "position:fixed;pointer-events:none;path{pointer-events:auto;stroke-width:0;stroke:transparent;fill:transparent;}" ]);
    // CONCATENATED MODULE: ./src/Popover/PopoverHitArea.tsx
    function oe(e, t) {
        return ue(e) || ce(e, t) || ie(e, t) || ae();
    }
    function ae() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ie(e, t) {
        if (e) {
            if ("string" == typeof e) return le(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? le(e, t) : void 0;
        }
    }
    function le(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function ce(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, a, i, l = [], c = !0, u = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (n = a.call(r)).done) && (l.push(n.value), l.length !== t); c = !0) {
                }
            } catch (e) {
                u = !0, o = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (u) throw o;
                }
            }
            return l;
        }
    }
    function ue(e) {
        if (Array.isArray(e)) return e;
    }
    var fe = {
        anchorRect: O().object,
        popoverEl: O().object,
        popoverPlacement: O().oneOf([ "above", "below", "left", "right", "misaligned" ])
    };
    var se = function e(t) {
        var r = t.anchorRect, n = t.popoverEl, o = t.popoverPlacement;
        if (r != null && n != null && o != null) {
            var a = n.getBoundingClientRect();
            if (o === "above" || o === "below") {
                var i = r.left - a.left;
                var l = r.height;
                var c = 0;
                if (o === "above") {
                    l = 0;
                    c = r.height;
                }
                var u = "0 ".concat(l);
                var f = "".concat(a.width, " ").concat(l);
                var s = "".concat(i, " ").concat(l);
                var p = "".concat(i, " ").concat(c);
                var v = "".concat(i + r.width, " ").concat(l);
                var d = "".concat(i + r.width, " ").concat(c);
                var h = "\n                        M ".concat(u, "\n                        S ").concat(s, " , ").concat(p, "\n                        L ").concat(s, "\n                        L ").concat(v, "\n                        L ").concat(d, "\n                        S ").concat(v, " , ").concat(f, "\n                    ");
                return {
                    hitAreaPath: h,
                    hitAreaStyle: {
                        left: a.left,
                        top: r.top,
                        width: a.width,
                        height: r.height
                    }
                };
            }
            if (o === "left" || o === "right") {
                var m = r.top - a.top;
                var b = 0;
                var g = r.width;
                if (o === "right") {
                    b = r.width;
                    g = 0;
                }
                var y = "".concat(b, " 0");
                var w = "".concat(b, " ").concat(a.height);
                var P = "".concat(b, " ").concat(m);
                var S = "".concat(g, " ").concat(m);
                var O = "".concat(b, " ").concat(m + r.height);
                var E = "".concat(g, " ").concat(m + r.height);
                var C = "\n                        M ".concat(y, "\n                        S ").concat(P, " , ").concat(S, "\n                        L ").concat(P, "\n                        L ").concat(O, "\n                        L ").concat(E, "\n                        S ").concat(O, " , ").concat(w, "\n                    ");
                return {
                    hitAreaPath: C,
                    hitAreaStyle: {
                        left: r.left,
                        top: a.top,
                        width: r.width,
                        height: a.height
                    }
                };
            }
        }
        return {
            hitAreaPath: null,
            hitAreaStyle: {}
        };
    };
    var pe =  n().forwardRef((function(e, t) {
        var o = e.anchorRect, a = e.popoverEl, i = e.popoverPlacement;
        var l = (0, r.useState)(null), c = oe(l, 2), u = c[0], f = c[1];
        var s = (0, r.useState)({}), p = oe(s, 2), v = p[0], d = p[1];
        (0, r.useEffect)((function() {
            var e = se({
                anchorRect: o,
                popoverEl: a,
                popoverPlacement: i
            }), t = e.hitAreaPath, r = e.hitAreaStyle;
            f(t);
            d(r);
        }), [ o, a, i ]);
        if (u != null) {
            
            return n().createElement(ne, {
                "data-test": "popover-hit-area",
                style: v
            },  n().createElement("path", {
                d: u,
                ref: t
            }));
        }
        return null;
    }));
    pe.propTypes = fe;
    /* harmony default export */    const ve = pe;
    // CONCATENATED MODULE: ./src/Popover/PopoverProvider.tsx
    var de =  (0, r.createContext)({});
    var he = {
        /** @private */
        allowPropagation: O().bool,
        children: O().node,
        hideArrow: O().bool
    };
    /**
 * Provides a method for controlling certain `Popover` props in components that use `Popover`.
 */    var me = function e(t) {
        var o = t.allowPropagation, a = t.children, i = t.hideArrow;
        var l = (0, r.useMemo)((function() {
            return {
                allowPropagation: o,
                hideArrow: i
            };
        }), [ o, i ]);
        
        return n().createElement(de.Provider, {
            value: l
        }, a);
    };
    me.propTypes = he;
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var be = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function ge() {
        var e = typeof document !== "undefined" ? document : be;
        return e;
    }
    var ye = ge();
    /* harmony default export */    const we = /* unused pure expression or super */ null && ye;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var Pe = {
        document: be,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function Se() {
        var e = typeof window !== "undefined" ? window : Pe;
        return e;
    }
    var Oe = Se();
    /* harmony default export */    const Ee = /* unused pure expression or super */ null && Oe;
    // CONCATENATED MODULE: ./src/utils/getBoundingClientRect.ts
    function Ce(e) {
        "@babel/helpers - typeof";
        return Ce = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Ce(e);
    }
    function je(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function xe(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? je(Object(r), !0).forEach((function(t) {
                ke(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : je(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function ke(e, t, r) {
        return (t = Ae(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function Ae(e) {
        var t = Te(e, "string");
        return "symbol" == Ce(t) ? t : t + "";
    }
    function Te(e, t) {
        if ("object" != Ce(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != Ce(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    // Checks if browser is Safari
        function Me() {
        return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
    }
    // Formats return value to DOMRect
        function Re(e) {
        return xe(xe({}, e), {}, {
            top: e.y,
            left: e.x,
            right: e.x + e.width,
            bottom: e.y + e.height
        });
    }
    /**
 * Provides DOMRect information about the size of an element and its position relative to the viewport.
 * Accounts for Safari's use of visual viewport and adds offsets to set as layout viewport.
 * @param {Element} element - The element to be evaluated
 * @param {boolean} offsets - Use when position of element relative to the layout viewport offsets are needed for Safari.
 * @returns {Omit<DOMRect, 'toJSON'>}
 * @public
 */    function We(e, t) {
        var r, n;
        var o = e.getBoundingClientRect();
        var a = Me();
        if (!t || !a) {
            return o;
        }
        var i = Se();
        var l = o.left + (((r = i.visualViewport) === null || r === void 0 ? void 0 : r.offsetLeft) || 0);
        var c = o.top + (((n = i.visualViewport) === null || n === void 0 ? void 0 : n.offsetTop) || 0);
        var u = o.width, f = o.height;
        return Re({
            width: u,
            height: f,
            x: l,
            y: c
        });
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function He(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var qe = "is deprecated and will be removed in the next major version.";
    var Ie = function e(t) {
        var r = t.additionalMessage, n = r === void 0 ? "" : r, o = t.componentName;
        useEffect((function() {
            if (false) {}
        }), [ n, o ]);
    };
    var Ne = function e(t) {
        var r = t.additionalMessage, n = r === void 0 ? "" : r, o = t.componentName, a = t.propName, i = t.propValue;
        useEffect((function() {
            if (false) {}
        }), [ n, o, a, i ]);
    };
    var Le = function e(t) {
        var n = t.additionalMessage, o = n === void 0 ? "" : n, a = t.componentName, i = t.deprecatedPropValue, l = t.propName, c = t.propValue;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ o, a, l, c, i ]);
    };
    // CONCATENATED MODULE: ./src/Popover/Popover.tsx
    function De(e) {
        "@babel/helpers - typeof";
        return De = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, De(e);
    }
    function Fe() {
        return Fe = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, Fe.apply(null, arguments);
    }
    function $e(e, t) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            t && (n = n.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), r.push.apply(r, n);
        }
        return r;
    }
    function _e(e) {
        for (var t = 1; t < arguments.length; t++) {
            var r = null != arguments[t] ? arguments[t] : {};
            t % 2 ? $e(Object(r), !0).forEach((function(t) {
                Ve(e, t, r[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : $e(Object(r)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(r, t));
            }));
        }
        return e;
    }
    function Ve(e, t, r) {
        return (t = ze(t)) in e ? Object.defineProperty(e, t, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = r, e;
    }
    function ze(e) {
        var t = Ke(e, "string");
        return "symbol" == De(t) ? t : t + "";
    }
    function Ke(e, t) {
        if ("object" != De(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var n = r.call(e, t || "default");
            if ("object" != De(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function Xe(e, t) {
        return Je(e) || Ge(e, t) || Ye(e, t) || Be();
    }
    function Be() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Ye(e, t) {
        if (e) {
            if ("string" == typeof e) return Ue(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? Ue(e, t) : void 0;
        }
    }
    function Ue(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function Ge(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, a, i, l = [], c = !0, u = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (n = a.call(r)).done) && (l.push(n.value), l.length !== t); c = !0) {
                }
            } catch (e) {
                u = !0, o = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (u) throw o;
                }
            }
            return l;
        }
    }
    function Je(e) {
        if (Array.isArray(e)) return e;
    }
    function Qe(e, t) {
        if (null == e) return {};
        var r, n, o = Ze(e, t);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                r = a[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function Ze(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var et = [ "clickAway", "escapeKey", "offScreen", "tabKey" ];
    var tt = {
        align: O().oneOf([ "center", "edge", "end" ]),
        anchor: O().object,
        animation: O().bool,
        animationConfig: O().object,
        appearance: O().oneOf([ "normal", "inverted", "none" ]),
        autoCloseWhenOffScreen: O().bool,
        canCoverAnchor: O().bool,
        children: O().oneOfType([ O().node, O().func ]),
        closeReasons: O().arrayOf(O().oneOf(et)),
        defaultPlacement: O().oneOf([ "above", "below", "left", "right", "vertical", "horizontal" ]),
        elementRef: O().oneOfType([ O().func, O().object ]),
        id: O().string,
        hideArrow: O().bool,
        hitAreaRef: O().oneOfType([ O().func, O().object ]),
        onRequestClose: O().func,
        open: O().bool,
        outerRef: O().oneOfType([ O().func, O().object ]),
        pointTo: O().shape({
            x: O().number,
            y: O().number
        }),
        repositionMode: O().oneOf([ "none", "flip", "any" ]),
        retainFocus: O().bool,
        takeFocus: O().bool,
        splunkTheme: O().object
    };
    var rt = {
        passive: false,
        capture: true
    };
    function nt(e, t) {
        var r = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 1;
        return !!e && !!t && u()(e, (function(e, n) {
            if (h()(e)) {
                return Math.abs(t[n] - e) <= r;
            }
            return t[n] === e;
        }));
    }
    function ot(e) {
        var t = e.anchorPos, r = e.borderRadius, n = e.placement, o = e.outerContainerStyle, i = e.outerContainerEl;
        if (n === "misaligned") {
            return [ {
                display: "none"
            } ];
        }
        var l = {
            display: "block"
        };
        var c = G * 2;
        // bottom edge
                var u = i.offsetHeight / 2 - (G * 2 + r + G / 2);
        // top edge
                var f = -(i.offsetHeight / 2 - (G + r + G / 2));
        var s = t.center - (o.top + i.offsetHeight / 2) - G / 2;
        var p = a()(s, f, u);
        // right edge
                var v = i.offsetWidth / 2 - (c + G + r);
        // left edge
                var d = -(i.offsetWidth / 2 - (G + r));
        var h = t.middle - (o.left + i.offsetWidth / 2) - G;
        var m = a()(h, d, v);
        var b = {
            left: {
                translateX: (G + J) / 2,
                translateY: p,
                rotate: 90
            },
            right: {
                translateX: -(G + J) / 2,
                translateY: p,
                rotate: -90
            },
            above: {
                translateX: m - J / 2,
                translateY: 0,
                rotate: 180
            },
            below: {
                translateX: m - J / 2,
                translateY: 0,
                rotate: 0
            }
        };
        var g = b[n];
        l.transform = "translate(".concat(g.translateX, "px, ").concat(g.translateY, "px) rotate(").concat(g.rotate, "deg)");
        // set new positions
                var y = {
            left: "right",
            right: "left",
            above: "bottom",
            below: "top"
        };
        l[y[n]] = "0";
        var w = {
            left: "top",
            right: "top",
            above: "left",
            below: "left"
        };
        l[w[n]] = "calc(50% - ".concat(J, "px/2)");
        return [ l, g ];
    }
    function at(e) {
        var t = e.anchorEl, r = e.offsets, n = r === void 0 ? false : r, o = e.pointTo;
        var a = We(t, n);
        var i = {
            // these are rounded to avoid causing visual regressions
            // they were previously clientWidth and clientHeight which were already rounded
            width: Math.round(a.width),
            height: Math.round(a.height)
        };
        var l = o && s()(o, "x");
        var c = o && s()(o, "y");
        i.left = l ? a.left + ((o === null || o === void 0 ? void 0 : o.x) || 0) : a.left;
        i.top = c ? a.top + ((o === null || o === void 0 ? void 0 : o.y) || 0) : a.top;
        i.right = l ? i.left + i.width : a.left + i.width || a.right;
        i.bottom = c ? i.top + i.height : a.top + i.height || a.bottom;
        // when pointTo is set, the middle (horizontal) and center (vertical)
        // these values are set to the upper left corner of the anchor
        // which causes the whole Popover to be aligned to the upper left corner
        // (ignoring the "align" prop) and then shifted by the value of "pointTo"
                i.middle = l ? i.left : a.left + (i.right - a.left) / 2;
        i.center = c ? i.top : a.top + (i.bottom - a.top) / 2;
        return i;
    }
    function it(e) {
        return e !== window && e !== window.document;
    }
    function lt(e, t) {
        var r = Se();
        if (e.top < 0 || e.top > r.innerHeight || e.left < 0 || e.left > r.innerWidth) {
            return true;
        }
        if (t) {
            if (e.height + e.top < t.top || e.top > t.bottom || e.width + e.left < t.left || e.left > t.right) {
                return true;
            }
        }
        return false;
    }
    function ct(e, t) {
        // this accomodates the arrow height
        // and provides some spacing between the end of the Popover and the edge of the window
        var r = G * 2;
        return t !== "none" && e != null && h()(e) ? e - r : e || null;
    }
    var ut = {};
    /**
 * `Popover` is used to create layovers such as dropdowns, contextual menus, or tooltips. Use
 * this only when the other components don't provide sufficient functionality or control. A controlled
 * `Dropdown` covers use cases where you might consider using `Popover` directly.
 */    function ft(e) {
        var t, o;
        var a = e.align, i = a === void 0 ? "center" : a, c = e.anchor, u = e.animation, f = u === void 0 ? true : u, s = e.animationConfig, p = s === void 0 ? ut : s, d = e.appearance, h = d === void 0 ? "normal" : d, m = e.autoCloseWhenOffScreen, g = m === void 0 ? true : m, w = e.canCoverAnchor, S = e.children, O = e.closeReasons, C = O === void 0 ? et : O, x = e.defaultPlacement, A = x === void 0 ? "below" : x, M = e.elementRef, H = e.hideArrow, q = e.hitAreaRef, I = e.id, N = e.onRequestClose, L = e.open, D = e.outerRef, F = e.pointTo, $ = e.repositionMode, _ = $ === void 0 ? "flip" : $, V = e.retainFocus, z = V === void 0 ? true : V, X = e.takeFocus, B = e.toggleId, Y = Qe(e, [ "align", "anchor", "animation", "animationConfig", "appearance", "autoCloseWhenOffScreen", "canCoverAnchor", "children", "closeReasons", "defaultPlacement", "elementRef", "hideArrow", "hitAreaRef", "id", "onRequestClose", "open", "outerRef", "pointTo", "repositionMode", "retainFocus", "takeFocus", "toggleId" ]);
        // @docs-props-type PopoverPropsBase
                Le({
            componentName: "Popover",
            deprecatedPropValue: "inverted",
            propName: "appearance",
            propValue: h
        });
        var U = (0, r.useState)(L), J = Xe(U, 2), Z = J[0], ne = J[1];
        var oe = (0, r.useState)(false), ae = Xe(oe, 2), ie = ae[0], le = ae[1];
        if (L !== ie) {
            // formerly implemented by componentWillReceiveProps
            // we can't use prevOpen here since that's a ref behind the scenes
            // and won't get updated until the component re-renders successfully
            // whereas setAnimating will interrupt the re-render and we'll end up in an infinite loop
            // so we need to update the value of wasOpen at the same time
            ne(f);
            le(L);
        }
        // this is a single state object because all these properties
        // are updated at the same time by setPlacement
                var ce = (0, r.useState)({}), ue = Xe(ce, 2), fe = ue[0], se = ue[1];
        var pe = (0, r.useRef)(true);
        var he = (0, r.useRef)(false);
        var me = (0, r.useRef)(null);
        var be = (0, r.useRef)(null);
        var ge = (0, r.useRef)(null);
        var ye = {
            anchorHeight: ((t = fe.anchorPos) === null || t === void 0 ? void 0 : t.height) || null,
            anchorWidth: ((o = fe.anchorPos) === null || o === void 0 ? void 0 : o.width) || null,
            placement: fe.placement || null,
            maxHeight: ct(fe.maxHeight, h),
            maxWidth: ct(fe.maxWidth, h),
            toggleId: B
        };
        var we = (0, r.useMemo)((function() {
            return f ? {
                opacity: L ? 1 : 0
            } : {
                opacity: 1
            };
        }), [ f, L ]);
        var Pe = Se();
        var Oe = L || Z;
        var Ee = (0, r.useContext)(T()) || Pe;
        var Ce = Ee !== Pe;
        var je = (0, r.useContext)(de), xe = je.allowPropagation, ke = je.hideArrow;
        var Ae = H !== null && H !== void 0 ? H : ke;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ c ]);
        var Te = (0, r.useCallback)((function(e) {
            if (L && (C === null || C === void 0 ? void 0 : C.includes(e.reason))) {
                N === null || N === void 0 ? void 0 : N(e);
            }
        }), [ C, L, N ]);
        var Me = (0, r.useCallback)((function(e, t) {
            var r = lt(e, t);
            if (r) {
                Te({
                    reason: "offScreen"
                });
            }
            return r;
        }), [ Te ]);
        var Re = R()(), We = Re.borderRadius, qe = Re.spacingXSmall;
        var Ie = parseInt(We, 10);
        var Ne = parseInt(qe, 10);
        // this is intentionally smaller when the arrow is hidden to address SUI-5452
                var De = Ae ? Ne : G;
        // do not call this directly - use setPlacement (the throttled version) instead
                var $e = (0, r.useCallback)((function(e) {
            se((function(t) {
                var r, n;
                // If these conditions are not met, we cannot set the popover.
                                if (!L || !ge.current || !c) {
                    // if (__DEV__ && !anchor && open) {
                    //     throw new Error(`Invalid anchor ${anchor} provided to Popover.`);
                    // }
                    return t;
                }
                var o = at({
                    anchorEl: c,
                    pointTo: F
                });
                var a = at({
                    anchorEl: c,
                    offsets: true,
                    pointTo: F
                });
                var l = it(Ee) ? at({
                    anchorEl: Ee,
                    pointTo: F
                }) : undefined;
                if (e && g) {
                    if (Me(o, l)) {
                        return t;
                    }
                }
                var u = K({
                    align: i,
                    anchorPos: a,
                    scrollContainerPos: l,
                    canCoverAnchor: w,
                    defaultPlacement: A,
                    repositionMode: _,
                    outerContainerEl: ge.current,
                    padding: De,
                    windowWidth: Pe.innerWidth,
                    windowHeight: Pe.innerHeight
                }), f = u.placement, s = u.outerContainerStyle, p = u.maxHeight, v = u.maxWidth;
                var d = nt(a, t.anchorPos) && nt(s, t.outerContainerStyle) && f === t.placement && p === t.maxHeight && v === t.maxWidth;
                var h = true;
                var m = {};
                var b;
                if (me.current) {
                    var P = ot({
                        anchorPos: a,
                        borderRadius: Ie,
                        outerContainerStyle: s,
                        placement: f,
                        outerContainerEl: ge.current
                    });
                    var S = Xe(P, 2);
                    m = S[0];
                    b = S[1];
                }
                h = nt(y()(m, "transform"), y()(t.arrowStyle, "transform")) && nt((r = b) !== null && r !== void 0 ? r : {}, (n = t.arrowStyleTransformMeta) !== null && n !== void 0 ? n : {});
                // If none of the position data has changed, do not set state.
                                if (d && h) {
                    return t;
                }
                return {
                    anchorPos: a,
                    arrowStyle: m,
                    arrowStyleTransformMeta: b,
                    outerContainerStyle: s,
                    placement: f,
                    maxHeight: p,
                    maxWidth: v
                };
            }));
        }), [ i, c, g, Ie, De, w, Me, A, L, F, _, Ee, Pe.innerHeight, Pe.innerWidth ]);
        // these are useMemo rather than useCallback so that ESLint can correctly analyze the dependencies
        // see https://github.com/facebook/react/issues/19240#issuecomment-652945246
                var Ve = (0, r.useMemo)((function() {
            return P()($e.bind(null, true), 0);
        }), [ $e ]);
        var ze = (0, r.useMemo)((function() {
            return P()($e.bind(null, true), 0);
        }), [ $e ]);
        var Ke = (0, r.useMemo)((function() {
            return P()($e, 0, {
                leading: false
            });
        }), [ $e ]);
        // this needs to run on every render - equivalent to componentDidUpdate
                (0, r.useEffect)((function() {
            if (!be.current) {
                return;
            }
            if (L || Z) {
                Ke();
            }
        }));
        (0, r.useEffect)((function() {
            if (X && L && be.current && fe.placement != null && !he.current) {
                he.current = true;
                (0, W.takeFocus)(be.current);
            }
        }), [ L, fe.placement, X ]);
        (0, r.useEffect)((function() {
            // we need to set this to true on every mount because React.StrictMode will call the effect twice
            // and we need to ensure that allowAnimationUpdates is true while the component is mounted
            pe.current = true;
            return function() {
                Ke.cancel();
                Ve.cancel();
                ze.cancel();
                pe.current = false;
            };
            // no dependencies here since this is equivalent to componentDidMount and componentWillUnmount
            // eslint-disable-next-line react-hooks/exhaustive-deps
                }), []);
        // EventListeners only need to be rendered when open,
        // otherwise leads to negative performance impacts
                var Be = Oe ?  n().createElement(n().Fragment, null, Ce &&  n().createElement(j(), {
            target: Ee,
            eventType: "scroll",
            listener: Ve,
            options: rt,
            key: "eventListener"
        }),  n().createElement(j(), {
            target: Pe,
            eventType: "resize",
            listener: Ke,
            key: "eventListenerOnWindowResize"
        }),  n().createElement(j(), {
            target: Pe,
            eventType: "scroll",
            listener: ze,
            options: rt,
            key: "eventListenerOnWindowScroll"
        })) : null;
        var Ye = (0, r.useCallback)((function() {
            // checking allowAnimationUpdates avoids trying to update state after the component has been unmounted - see SUI-3666
            if (pe.current) {
                ne(false);
            }
        }), []);
        var Ue = (0, r.useCallback)((function(e) {
            be.current = e;
            if (e && X) {
                l()(W.takeFocus, e);
            }
            He(M, e);
        }), [ M, X ]);
        var Ge = (0, r.useCallback)((function(e) {
            ge.current = e;
            He(D, e);
        }), [ D ]);
        var Je = (0, r.useCallback)((function(e) {
            if (be.current) {
                if (z) {
                    (0, W.handleTab)(be.current, e.nativeEvent);
                } else if ((0, W.isTabKey)(e.nativeEvent)) {
                    var t = (0, W.getSortedTabbableElements)(be.current);
                    if (t.length > 0) {
                        var r = e.shiftKey ? t[0] : t[t.length - 1];
                        if (e.target instanceof HTMLElement && r === e.target) {
                            Te({
                                event: e.nativeEvent,
                                reason: "tabKey"
                            });
                        }
                    }
                }
            }
        }), [ Te, z ]);
        var Ze = (0, r.useCallback)((function(e) {
            if (!xe) {
                e.stopPropagation();
            }
        }), [ xe ]);
        var tt = (0, r.useMemo)((function() {
            return {
                opacity: f ? 0 : 1
            };
        }), [ f ]);
        var ft = (0, r.useMemo)((function() {
            return _e({
                tension: 300,
                friction: 20,
                clamp: true
            }, p);
        }), [ p ]);
        var st = (0, E.useAnimation)({
            from: tt,
            to: we,
            // Setting clamp to true will stop animating at the point it hit's it's goal value.
            // This will prevent click through while animating out (SUI-6618).
            config: ft,
            onRest: Ye
        });
        var pt = (0, r.useMemo)((function() {
            return _e(_e({}, fe.outerContainerStyle), st);
        }), [ fe.outerContainerStyle, st ]);
        var vt = (0, r.useMemo)((function() {
            return v()(C === null || C === void 0 ? void 0 : C.filter((function(e) {
                return e !== "offScreen";
            })), k().possibleCloseReasons);
        }), [ C ]);
        var dt = b()(S) ? S(ye) : S;
        
        return n().createElement(n().Fragment, null, Be,  n().createElement(k(), {
            closeReasons: vt,
            open: Oe,
            onRequestClose: Te,
            key: "Layer"
        }, Oe && be.current && q &&  n().createElement(ve, {
            anchorRect: fe.anchorPos,
            popoverEl: be.current,
            popoverPlacement: fe.placement,
            ref: q
        }), Oe &&  n().createElement(Q, {
            "data-test": "outer-popover",
            style: pt,
            ref: Ge
        },  n().createElement(ee, Fe({
            $appearance: h,
            $boxPadding: De,
            $open: L,
            "data-test": "popover",
            ref: Ue,
            id: I,
            onKeyDown: Je
        }, Y),  n().createElement(n().Fragment, null, h === "none" && dt, h !== "none" && !Ae &&  n().createElement(re, {
            "data-test": "popover-arrow",
            $appearance: h,
            onClick: Ze,
            ref: me,
            style: fe.arrowStyle
        }), h !== "none" &&  n().createElement(te, {
            $appearance: h,
            onClick: Ze
        }, dt))))));
    }
    ft.propTypes = tt;
    ft.possibleCloseReasons = et;
    /* harmony default export */    const st = ft;
    // CONCATENATED MODULE: ./src/Popover/PopoverMenuContext.tsx
    var pt =  (0, r.createContext)({});
    // CONCATENATED MODULE: ./src/Popover/index.ts
    module.exports = t;
    /******/})();
