// @splunk/react-ui/Progress.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ _
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/isNumber"
    const o = require("lodash/isNumber");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var s =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const l = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const f = require("@splunk/react-ui/Tooltip");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const d = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Progress/ProgressStyles.ts
    var v = "8px";
    var b = (0, c.keyframes)([ "from{background-position:130%;}to{background-position:-30%;}" ]);
    var y = u().div.attrs((function(e) {
        var r = e.$percent;
        /**
   * Prevents styled-components from generating new classes for each percent value (SUI-7625).
   * Pseudo-elements can't have inline styles, so width can't be set directly.
   */        return {
            style: {
                "--percent": "".concat(r, "%")
            }
        };
    })).withConfig({
        displayName: "ProgressStyles__StyledProgressWrapper",
        componentId: "csowex-0"
    })([ "", " height:", ";position:relative;border:1px solid ", ";border-radius:", ";background:", ";&::before{display:block;position:absolute;top:-1px;left:-1px;content:'';border-radius:", ";height:100%;", " width:var(--percent);", "}" ], d.mixins.reset("block"), v, d.variables.interactiveColorBorder, d.variables.borderRadius, d.variables.neutral100, d.variables.borderRadius, (function(e) {
        var r = e.$animated, t = e.$type;
        // Animated is only supported for the info type.
        // This is enforced in the component and duplicating here
                return r && t === "info" ? (0, c.css)([ "background:radial-gradient( circle at center,", ",", " 30% );background-size:200% 100%;animation:", " 2500ms infinite cubic-bezier(0.33,0,0.67,1);" ], d.variables.notificationColorInfoStrong, d.variables.notificationColorInfo, b) : (0, 
        c.css)([ "background:", ";" ], (0, d.pickVariant)("$type", {
            info: d.variables.notificationColorInfo,
            success: d.variables.notificationColorPositive,
            error: d.variables.notificationColorNegative
        }));
    }), (function(e) {
        var r = e.$percent;
        return (0, c.css)([ "", " ", "" ], r > 0 && (0, c.css)([ "border:1px solid ", ";" ], (0, 
        d.pickVariant)("$type", {
            info: d.variables.notificationColorInfoStrong,
            success: d.variables.notificationColorPositiveStrong,
            error: d.variables.notificationColorNegativeStrong
        })), r !== 100 && (0, c.css)([ "border-end-end-radius:0;border-start-end-radius:0;border-inline-end:none;" ]));
    }));
    var g = u()(p()).attrs((function(e) {
        var r = e.$percent;
        /* Prevents styled-components from generating new classes for each percent value (SUI-7625). */        return {
            style: {
                width: "".concat(r, "%")
            }
        };
    })).withConfig({
        displayName: "ProgressStyles__StyledTooltip",
        componentId: "csowex-1"
    })([ "height:", ";" ], v);
    var m = u().progress.withConfig({
        displayName: "ProgressStyles__StyledProgress",
        componentId: "csowex-2"
    })([ "opacity:0;position:absolute;inset:0;" ]);
    // CONCATENATED MODULE: ./src/Progress/Progress.tsx
    function h() {
        return h = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, h.apply(null, arguments);
    }
    function S(e, r) {
        return P(e) || C(e, r) || k(e, r) || O();
    }
    function O() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function k(e, r) {
        if (e) {
            if ("string" == typeof e) return w(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? w(e, r) : void 0;
        }
    }
    function w(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function C(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, i, a, s = [], l = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    l = !1;
                } else for (;!(l = (n = i.call(t)).done) && (s.push(n.value), s.length !== r); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return s;
        }
    }
    function P(e) {
        if (Array.isArray(e)) return e;
    }
    function x(e, r) {
        if (null == e) return {};
        var t, n, o = j(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function j(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var $ = {
        elementRef: s().oneOfType([ s().func, s().object ]),
        percentage: s().number,
        tooltip: s().node,
        type: s().oneOf([ "info", "success", "error" ])
    };
    function I(e) {
        var r = e.elementRef, o = e.percentage, a = e.tooltip, s = e.type, c = s === void 0 ? "info" : s, u = x(e, [ "elementRef", "percentage", "tooltip", "type" ]);
        // @docs-props-type ProgressPropsBase
                if (false) {}
        var f = (0, t.useState)(false), p = S(f, 2), d = p[0], v = p[1];
        var b = (0, l.useAnimationToggle)() === "on" && c === "info";
        var O = (0, t.useCallback)((function() {
            v(true);
        }), []);
        var k = (0, t.useCallback)((function() {
            v(false);
        }), []);
        var w = i()(o) ? o : 0;
        var C = a || "".concat(w, "%");
        
        return n().createElement(y, h({
            onMouseEnter: O,
            onMouseLeave: k,
            ref: r,
            "data-test": "progress",
            $percent: w,
            $type: c,
            $animated: b
        }, u),  n().createElement(g, {
            content: C,
            inline: false,
            $percent: w,
            open: w > 0 && d
        },  n().createElement(m, {
            value: w,
            max: 100
        })));
    }
    I.propTypes = $;
    /* harmony default export */    const _ = I;
    // CONCATENATED MODULE: ./src/Progress/index.ts
    module.exports = r;
    /******/})();
