// @splunk/react-ui/Prose.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Prose: () => /* reexport */ q,
        default: () => /* reexport */ H
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/Heading"
    const a = require("@splunk/react-ui/Heading");
    // CONCATENATED MODULE: external "@splunk/react-ui/List"
    const u = require("@splunk/react-ui/List");
    // CONCATENATED MODULE: external "@splunk/react-ui/Paragraph"
    const s = require("@splunk/react-ui/Paragraph");
    // CONCATENATED MODULE: external "@splunk/themes/mixins"
    const f = require("@splunk/themes/mixins");
    var p =  e.n(f);
    // CONCATENATED MODULE: ./src/Prose/Prose.tsx
    function y(e) {
        "@babel/helpers - typeof";
        return y = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, y(e);
    }
    function b() {
        return b = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, b.apply(null, arguments);
    }
    function d(e, r) {
        if (null == e) return {};
        var t, n, o = m(e, r);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                t = i[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function m(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    function v(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function O(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? v(Object(t), !0).forEach((function(r) {
                g(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : v(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function g(e, r, t) {
        return (r = S(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function S(e) {
        var r = h(e, "string");
        return "symbol" == y(r) ? r : r + "";
    }
    function h(e, r) {
        if ("object" != y(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != y(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    var j = {
        p: s.StyledParagraph,
        h1: a.StyledHeading,
        h2: a.StyledHeading,
        h3: a.StyledHeading,
        h4: a.StyledHeading,
        h5: a.StyledHeading,
        h6: a.StyledHeading,
        ul: u.StyledList,
        ol: u.StyledList
    };
    var P = c().section.withConfig({
        displayName: "Prose__StyledProse",
        componentId: "sc-9sfct5-0"
    })([ "", "" ], (function(e) {
        var r = e.$elementSelectors;
        return p().prose({
            elementSelectors: O(O({}, j), r)
        });
    }));
    var w = {
        as: i().string,
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        elementSelectors: i().object
    };
    /**
 * Prose styles `@splunk/react-ui` content components for readability with long-form text content.
 */    function q(e) {
        var r = e.elementRef, t = e.as, o = e.children, i = e.elementSelectors, l = d(e, [ "elementRef", "as", "children", "elementSelectors" ]);
        // @docs-props-type ProsePropsBase
                
        return n().createElement(P, b({
            "data-test": "prose",
            as: t,
            ref: r,
            $elementSelectors: i
        }, l), o);
    }
    q.propTypes = w;
    /* harmony default export */    const H = q;
    // CONCATENATED MODULE: ./src/Prose/index.ts
    module.exports = r;
    /******/})();
