// @splunk/react-ui/RadioList.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var a = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, a) => {
            /******/ for (var n in a) {
                /******/ if (e.o(a, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: a[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Option: () => /* reexport */ k,
        default: () => /* reexport */ F
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var n =  e.n(a);
    // CONCATENATED MODULE: external "lodash/omit"
    const o = require("lodash/omit");
    var t =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const d = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var s =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Paragraph"
    const u = require("@splunk/react-ui/Paragraph");
    var v =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/RadioList/OptionStyles.ts
    var b = s().div.withConfig({
        displayName: "OptionStyles__StyledWrapper",
        componentId: "imaqjt-0"
    })([ "display:grid;grid-column:1 / -1;grid-template-columns:subgrid;" ]);
    var p = s().label.withConfig({
        displayName: "OptionStyles__StyledLabel",
        componentId: "imaqjt-1"
    })([ "grid-column:label;color:", ";cursor:pointer;" ], (function(e) {
        var r = e.$disabled;
        return r ? f.variables.contentColorDisabled : f.variables.contentColorActive;
    }));
    var m = s()(v()).withConfig({
        displayName: "OptionStyles__StyledDescription",
        componentId: "imaqjt-2"
    })([ "grid-column:label;" ]);
    var y = "20px";
    var g = s().input.withConfig({
        displayName: "OptionStyles__StyledRadioInput",
        componentId: "imaqjt-3"
    })([ "", ";grid-column:input;appearance:none;align-items:center;justify-items:center;height:", ";width:", ";border-radius:50%;border:2px solid ", ";background-color:", ";&::after{display:block;content:'';border-radius:50%;height:0;width:0;opacity:0;border:5px solid transparent;}&:checked::after{opacity:1;}&:hover{border-color:", ";background-color:", ";}&:checked{&,&::after{border-color:", ";}&:hover{border-color:", ";}}&:disabled{cursor:not-allowed;&,&:hover,&:checked,&:checked:hover,&:checked::after{border-color:", ";}&,&:hover{background-color:transparent;}}&:focus{box-shadow:", ";}", "" ], f.mixins.reset("grid"), y, y, f.variables.interactiveColorBorder, f.variables.interactiveColorBackground, f.variables.interactiveColorBorderHover, f.mixins.overlayColors(f.variables.interactiveColorBackground, f.variables.interactiveColorOverlayHover), f.variables.interactiveColorAccent, f.mixins.overlayColors(f.variables.interactiveColorAccent, f.variables.interactiveColorBorderHover), f.variables.interactiveColorBorderDisabled, f.variables.focusShadow, (function(e) {
        var r = e.$error;
        return r && (0, c.css)([ "&,&:checked,&:checked::after{border-color:", ";}&:checked:hover{border-color:", ";}&:hover{background-color:", ";border-color:", ";}" ], f.variables.interactiveColorAccentError, f.mixins.overlayColors(f.variables.interactiveColorAccentError, f.variables.interactiveColorBorderHover), f.variables.interactiveColorOverlayHover, f.mixins.overlayColors(f.variables.interactiveColorAccentError, f.variables.interactiveColorBorderHover));
    }));
    // CONCATENATED MODULE: ./src/RadioList/RadioListContext.ts
    var h =  (0, a.createContext)({});
    h.displayName = "RadioList";
    /* harmony default export */    const O = h;
    // CONCATENATED MODULE: ./src/RadioList/Option.tsx
    function C() {
        return C = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var a = arguments[r];
                for (var n in a) {
                    ({}).hasOwnProperty.call(a, n) && (e[n] = a[n]);
                }
            }
            return e;
        }, C.apply(null, arguments);
    }
    function S(e, r) {
        if (null == e) return {};
        var a, n, o = w(e, r);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            for (n = 0; n < t.length; n++) {
                a = t[n], -1 === r.indexOf(a) && {}.propertyIsEnumerable.call(e, a) && (o[a] = e[a]);
            }
        }
        return o;
    }
    function w(e, r) {
        if (null == e) return {};
        var a = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                a[n] = e[n];
            }
        }
        return a;
    }
    var j = {
        children: l().node,
        description: l().string,
        disabled: l().bool,
        error: l().bool,
        id: l().string,
        onChange: l().func,
        value: l().any.isRequired
    };
    function x(e) {
        var r = e.children, o = e.description, t = e.disabled, i = t === void 0 ? false : t, l = e.error, c = l === void 0 ? false : l, s = e.id, u = e.onChange, v = e.value, f = S(e, [ "children", "description", "disabled", "error", "id", "onChange", "value" ]);
        // @docs-props-type OptionPropsBase
                var y = (0, a.useRef)(s || (0, d.createDOMID)("radio-option")), h = y.current;
        var w = (0, a.useContext)(O);
        var j = function e(r) {
            var a;
            r.stopPropagation();
            u === null || u === void 0 ? void 0 : u(r, {
                value: v,
                name: w.name
            });
            (a = w.onChange) === null || a === void 0 ? void 0 : a.call(w, r, {
                value: v,
                name: w.name
            });
        };
        var x = (0, a.useMemo)((function() {
            return "".concat(h, "-description");
        }), [ h ]);
        var k = false;
        if ((w === null || w === void 0 ? void 0 : w.value) !== undefined) {
            k = w.value === v;
        }
        var q = i;
        if (w === null || w === void 0 ? void 0 : w.disabled) {
            q = w.disabled;
        }
        var E = c;
        if (w === null || w === void 0 ? void 0 : w.error) {
            E = w.error;
        }
        
        return n().createElement(b, C({
            "data-test": "option",
            "data-test-value": v,
            "data-test-disabled": q ? "disabled" : undefined
        }, f, {
            "data-checked": k
        }),  n().createElement(g, {
            id: h,
            type: "radio",
            name: w.name,
            $error: E,
            disabled: q,
            value: v.toString(),
            checked: k,
            onChange: j,
            "data-test": "input",
            required: w.required,
            "aria-describedby": o ? x : undefined
        }),  n().createElement(p, {
            htmlFor: h,
            "data-test": "label",
            $disabled: q
        }, r), o &&  n().createElement(m, {
            "data-test": "description",
            id: x
        }, o));
    }
    x.propTypes = j;
    /* harmony default export */    const k = x;
    // CONCATENATED MODULE: ./src/RadioList/RadioListStyles.ts
    var q = s().div.withConfig({
        displayName: "RadioListStyles__StyledRadioList",
        componentId: "sc-1qg10hx-0"
    })([ "", ";grid-template-columns:[input] min-content [label] auto;gap:", " ", ";", "" ], f.mixins.reset("grid"), f.variables.spacingMedium, f.variables.spacingSmall, (function(e) {
        return e.direction === "row" && (0, c.css)([ "", ";align-items:start;gap:", ";", "{grid-template-columns:[input] min-content [label] auto;column-gap:", ";}" ], f.mixins.reset("flex"), f.variables.spacingXLarge, /* sc-sel */ b, f.variables.spacingSmall);
    }));
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var E = "is deprecated and will be removed in the next major version.";
    var P = function e(r) {
        var a = r.additionalMessage, n = a === void 0 ? "" : a, o = r.componentName;
        useEffect((function() {
            if (false) {}
        }), [ n, o ]);
    };
    var B = function e(r) {
        var n = r.additionalMessage, o = n === void 0 ? "" : n, t = r.componentName, i = r.propName, l = r.propValue;
        (0, a.useEffect)((function() {
            if (false) {}
        }), [ o, t, i, l ]);
    };
    var R = function e(r) {
        var a = r.additionalMessage, n = a === void 0 ? "" : a, o = r.componentName, t = r.deprecatedPropValue, i = r.propName, l = r.propValue;
        useEffect((function() {
            if (false) {}
        }), [ n, o, i, l, t ]);
    };
    // CONCATENATED MODULE: ./src/RadioList/RadioList.tsx
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var a = arguments[r];
                for (var n in a) {
                    ({}).hasOwnProperty.call(a, n) && (e[n] = a[n]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function I(e, r) {
        return L(e) || D(e, r) || A(e, r) || N();
    }
    function N() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function A(e, r) {
        if (e) {
            if ("string" == typeof e) return M(e, r);
            var a = {}.toString.call(e).slice(8, -1);
            return "Object" === a && e.constructor && (a = e.constructor.name), "Map" === a || "Set" === a ? Array.from(e) : "Arguments" === a || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(a) ? M(e, r) : void 0;
        }
    }
    function M(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var a = 0, n = Array(r); a < r; a++) {
            n[a] = e[a];
        }
        return n;
    }
    function D(e, r) {
        var a = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != a) {
            var n, o, t, i, l = [], d = !0, c = !1;
            try {
                if (t = (a = a.call(e)).next, 0 === r) {
                    if (Object(a) !== a) return;
                    d = !1;
                } else for (;!(d = (n = t.call(a)).done) && (l.push(n.value), l.length !== r); d = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!d && null != a["return"] && (i = a["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function L(e) {
        if (Array.isArray(e)) return e;
    }
    function T(e, r) {
        if (null == e) return {};
        var a, n, o = V(e, r);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            for (n = 0; n < t.length; n++) {
                a = t[n], -1 === r.indexOf(a) && {}.propertyIsEnumerable.call(e, a) && (o[a] = e[a]);
            }
        }
        return o;
    }
    function V(e, r) {
        if (null == e) return {};
        var a = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                a[n] = e[n];
            }
        }
        return a;
    }
    /** @public */    var H = {
        children: l().node,
        defaultValue: l().any,
        describedBy: l().string,
        direction: l().oneOf([ "row", "column" ]),
        disabled: l().bool,
        elementRef: l().oneOfType([ l().func, l().object ]),
        error: l().bool,
        labelledBy: l().string,
        name: l().string,
        onChange: l().func,
        /** @private. */
        required: l().bool,
        value: l().any
    };
    function $(e) {
        var r = e.children, o = e.defaultValue, i = e.describedBy, l = e.direction, c = e.disabled, s = c === void 0 ? false : c, u = e.elementRef, v = e.error, f = v === void 0 ? false : v, b = e.labelledBy, p = e.name, m = e.onChange, y = e.required, g = e.value, h = T(e, [ "children", "defaultValue", "describedBy", "direction", "disabled", "elementRef", "error", "labelledBy", "name", "onChange", "required", "value" ]);
        // @docs-props-type RadioListPropsBase
                B({
            componentName: "RadioList",
            propName: "direction",
            propValue: l
        });
        // TODO: Implement shared isControlled hook (SUI-5283).
        // See material-ui for similar pattern to handle controlled/un-controlled in functional components w/ hooks
        // https://github.com/mui-org/material-ui/blob/60d99a39836fb82f4da1477a717f642c216fb0b9/packages/material-ui/src/utils/useControlled.js
                var C = (0, a.useRef)(g !== undefined), S = C.current;
        var w = (0, a.useState)(o), j = I(w, 2), x = j[0], k = j[1];
        var E = S ? g : x;
        var P = (0, a.useRef)(p || (0, d.createDOMID)("radio-name")), R = P.current;
        var N = (0, a.useCallback)((function(e, r) {
            if (!S) {
                k(r.value);
            }
            m === null || m === void 0 ? void 0 : m(e, r);
        }), [ S, m ]);
        // Check correct props are passed
                if (false) {}
        if (false) {}
        // Check if changing from controlled to un-controlled
                (0, a.useEffect)((function() {
            if (false) {}
        }), [ g, S ]);
        var A = (0, a.useRef)(o), M = A.current;
        (0, a.useEffect)((function() {
            if (false) {}
        }), [ M, o, S ]);
        var D = (0, a.useMemo)((function() {
            return {
                disabled: s,
                error: f,
                onChange: N,
                name: R,
                required: y,
                value: E
            };
        }), [ s, f, N, R, y, E ]);
        var L = h["aria-labelledby"];
        var V = h["aria-describedby"];
        
        return n().createElement(O.Provider, {
            value: D
        },  n().createElement(q, _({
            direction: l || "column",
            role: "radiogroup",
            "data-test": "radio-list",
            "data-test-value": E,
            "data-test-disabled": s ? "disabled" : undefined,
            ref: u
        }, t()(h, [ "aria-labelledby", "aria-describedby" ]), {
            "aria-labelledby": L ? "".concat(L, " ").concat(b) : b,
            "aria-describedby": V ? "".concat(V, " ").concat(i) : i,
            "aria-invalid": f
        }), r));
    }
    $.propTypes = H;
    $.Option = k;
    $.componentType = "RadioList";
    /* harmony default export */    const F = $;
    // only for styled components using RadioList
        // CONCATENATED MODULE: ./src/RadioList/index.ts
    module.exports = r;
    /******/})();
