// @splunk/react-ui/RadioBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Option: () => /* reexport */ A,
        default: () => /* reexport */ G
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "lodash/omit"
    const o = require("lodash/omit");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const d = require("@splunk/react-ui/useControlled");
    var u =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const c = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var b =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const v = require("@splunk/react-ui/Button");
    var f =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/RadioBar/OptionStyles.ts
    var y = 10;
    var m = b()(f()).withConfig({
        displayName: "OptionStyles__StyledButton",
        componentId: "fyq77p-0"
    })([ "border:", " solid ", ";border-radius:0;min-width:0;padding:", " ", ";z-index:", ";&&{color:", ";}&:first-child{border-top-left-radius:", ";border-bottom-left-radius:", ";}&:last-child{border-top-right-radius:", ";border-bottom-right-radius:", ";}&:not(:last-child){margin-right:-", ";}&&:focus{background-color:", ";box-shadow:", ";color:", ";z-index:", ";border:0;span{padding:0 ", ";}}&:not([disabled],[aria-disabled='true']){", "}&[disabled],&[aria-disabled='true']{background-color:", ";color:", ";z-index:0;}" ], p.variables.inputBorderWidth, p.variables.borderColor, p.variables.spacingXSmall, p.variables.spacingSmall, y, p.variables.contentColorDefault, p.variables.borderRadius, p.variables.borderRadius, p.variables.borderRadius, p.variables.borderRadius, p.variables.inputBorderWidth, p.variables.actionColorBackgroundSecondaryActive, p.variables.focusShadowInset, p.variables.contentColorActive, y * 5, p.variables.inputBorderWidth, (0, 
    p.pickVariant)("$optionSelected", {
        true: (0, s.css)([ "", ";background-color:", ";border-color:", ";border-width:", ";z-index:", ";" ], p.mixins.typography({
            weight: "bold",
            color: "active"
        }), p.variables.actionColorBackgroundSecondaryActive, p.variables.actionColorBorderSecondaryActive, p.variables.inputBorderWidth, y * 4),
        false: (0, s.css)([ "&:hover{background-color:", ";border-color:", ";color:", ";z-index:", ";}" ], p.variables.actionColorBackgroundSecondaryHover, p.variables.borderColor, p.variables.contentColorActive, y * 3)
    }), p.variables.actionColorBackgroundSecondaryDisabled, p.variables.contentColorDisabled);
    var g = b().div.withConfig({
        displayName: "OptionStyles__StyledLabel",
        componentId: "fyq77p-1"
    })([ "", " flex:0 1 auto;" ], p.mixins.ellipsis());
    var h = b().div.withConfig({
        displayName: "OptionStyles__StyledAdornment",
        componentId: "fyq77p-2"
    })([ "display:flex;" ]);
    // CONCATENATED MODULE: ./src/RadioBar/RadioBarContext.tsx
    var O =  (0, n.createContext)({});
    O.displayName = "RadioBar";
    /* harmony default export */    const S = O;
    // CONCATENATED MODULE: ./src/RadioBar/Option.tsx
    function w() {
        return w = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, w.apply(null, arguments);
    }
    function C(e, r) {
        if (null == e) return {};
        var n, t, o = x(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function x(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var j = {
        disabled: l().bool,
        startAdornment: l().node,
        endAdornment: l().node,
        label: l().string,
        /** @private Set by `RadioBar`. */
        role: l().oneOf([ "radio", "menuitemradio" ]),
        /** @private Set by `RadioBar`. */
        selected: l().bool,
        /** The value of the `Option`. */
        value: l().any.isRequired
    };
    function k(e) {
        var r = e.disabled, o = e.label, a = e.selected, i = e.startAdornment, l = e.endAdornment, d = e.value, u = e.role, c = u === void 0 ? "radio" : u, s = C(e, [ "disabled", "label", "selected", "startAdornment", "endAdornment", "value", "role" ]);
        // @docs-props-type OptionPropsBase
                var b = (0, n.useContext)(S), v = b.disabled, f = b.error, p = b.firstValue, y = b.name, O = b.onClick, x = b.onKeyDown, j = b.selectedValue;
        var k = v || r;
        var A = -1;
        if (j) {
            if (d === j) {
                A = 0;
            }
        } else if (d === p) {
            A = 0;
        }
        var B = A;
        var P = (0, n.useCallback)((function(e) {
            if (!a) {
                O === null || O === void 0 ? void 0 : O(e, {
                    value: d,
                    label: o
                });
            }
        }), [ o, O, a, d ]);
        var D = (0, n.useCallback)((function(e) {
            return x === null || x === void 0 ? void 0 : x(e, {
                value: d,
                label: o
            });
        }), [ o, x, d ]);
        
        return t().createElement(m, w({
            "aria-invalid": f,
            "data-test": "option",
            "data-test-value": d,
            "data-test-disabled": k ? "disabled" : undefined,
            appearance: "default",
            disabled: k ? "disabled" : undefined,
            $optionSelected: a,
            value: d,
            name: y
        }, s, {
            onClick: P,
            role: c,
            "aria-checked": a,
            tabIndex: B,
            onKeyDown: D
        }), i &&  t().createElement(h, null, i), o &&  t().createElement(g, null, o), l &&  t().createElement(h, null, l));
    }
    k.propTypes = j;
    /* harmony default export */    const A = k;
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const B = require("@splunk/react-ui/Box");
    var P =  e.n(B);
    // CONCATENATED MODULE: ./src/RadioBar/RadioBarStyles.ts
    var D = b()(P()).withConfig({
        displayName: "RadioBarStyles__StyledRadioBar",
        componentId: "gg1b79-0"
    })([ "height:", ";", "" ], p.variables.inputHeight, (function(e) {
        var r = e.$error;
        return r && (0, s.css)([ "& > ", "{border-color:", ";:not([disabled],[aria-disabled='true']){&:hover{border-color:", ";}&[aria-checked='true']{background-color:", ";border-color:", ";}}}" ], m, p.variables.actionColorBorderDestructiveSecondary, p.variables.actionColorBorderDestructiveSecondary, p.variables.actionColorBackgroundDestructiveSecondaryActive, p.variables.actionColorBorderDestructiveSecondaryActive);
    }));
    // CONCATENATED MODULE: ./src/RadioBar/RadioBar.tsx
    function R(e) {
        "@babel/helpers - typeof";
        return R = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, R(e);
    }
    function E() {
        return E = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, E.apply(null, arguments);
    }
    function q(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function I(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? q(Object(n), !0).forEach((function(r) {
                T(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : q(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function T(e, r, n) {
        return (r = _(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function _(e) {
        var r = V(e, "string");
        return "symbol" == R(r) ? r : r + "";
    }
    function V(e, r) {
        if ("object" != R(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != R(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function M(e, r) {
        return K(e) || W(e, r) || $(e, r) || N();
    }
    function N() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function $(e, r) {
        if (e) {
            if ("string" == typeof e) return z(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? z(e, r) : void 0;
        }
    }
    function z(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, t = Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function W(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var t, o, a, i, l = [], d = !0, u = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    d = !1;
                } else for (;!(d = (t = a.call(n)).done) && (l.push(t.value), l.length !== r); d = !0) {
                }
            } catch (e) {
                u = !0, o = e;
            } finally {
                try {
                    if (!d && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (u) throw o;
                }
            }
            return l;
        }
    }
    function K(e) {
        if (Array.isArray(e)) return e;
    }
    function F(e, r) {
        if (null == e) return {};
        var n, t, o = H(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function H(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    /** @public */    var L = {
        children: l().node,
        defaultValue: l().oneOfType([ l().string, l().number, l().bool ]),
        describedBy: l().string,
        disabled: l().bool,
        elementRef: l().oneOfType([ l().func, l().object ]),
        error: l().bool,
        inline: l().bool,
        labelledBy: l().string,
        name: l().string,
        onChange: l().func,
        /** @private. */
        required: l().bool,
        role: l().oneOf([ "radiogroup", "menubar" ]),
        value: l().oneOfType([ l().string, l().number, l().bool ])
    };
    // TODO(SUI-7012): Once converted to a functional component, this function can be removed and RadioBar should use useRovingFocus hook
        var U = function e(r, n, t, o) {
        var a = r === "forward" ? 1 : -1;
        var i = r === "forward" ? "nextSibling" : "previousSibling";
        var l = r === "forward" ? "firstElementChild" : "lastElementChild";
        var d = r === "forward" ? 0 : n - 1;
        var u = t[i];
        var c = null;
        var s = o + a;
        while (u !== t && c === null) {
            if (u === null) {
                var b;
                u = (b = t.parentElement) === null || b === void 0 ? void 0 : b[l];
                s = d;
            }
            if (u.disabled) {
                var v = u[i];
                u = v;
                s += a;
            } else {
                c = u;
            }
        }
        return {
            nextTargetRes: c,
            nextIndexRes: s
        };
    };
    /**
 * RadioBar is a form control that provides the ability to select one option out of a group.
 */    function X(e) {
        var r = e.children, o = e.defaultValue, i = e.describedBy, l = e.disabled, d = l === void 0 ? false : l, s = e.elementRef, b = e.error, v = e.labelledBy, f = e.name, p = e.onChange, y = e.required, m = e.role, g = m === void 0 ? "radiogroup" : m, h = e.value, O = F(e, [ "children", "defaultValue", "describedBy", "disabled", "elementRef", "error", "labelledBy", "name", "onChange", "required", "role", "value" ]);
        // @docs-props-type RadioBarPropsBase
                var w = u()({
            componentName: "RadioBar",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var C = (0, n.useState)(o), x = M(C, 2), j = x[0], k = x[1];
        var A = w ? h : j;
        var B = (0, n.useMemo)((function() {
            return f || (0, c.createDOMID)("radio-name");
        }), [ f ]);
        var P = (0, n.useMemo)((function() {
            // Everything in here is grouped because together they are based on children changing
            var e = n.Children.toArray(r).filter(n.isValidElement);
            var t = e.length > 0 && e.every((function(e) {
                var r;
                return (e === null || e === void 0 ? void 0 : (r = e.props) === null || r === void 0 ? void 0 : r.disabled) === true;
            }));
            var o = e.map((function(e) {
                return e.props.value;
            }));
            var a = e.map((function(e, r) {
                
                return (0, n.cloneElement)(e, {
                    selected: e.props.value === A,
                    key: e.key || r,
                    role: g === "radiogroup" ? "radio" : "menuitemradio"
                });
            }));
            return {
                childrenFormatted: a,
                childrenAreDisabled: t,
                filteredValues: o
            };
        }), [ r, g, A ]), R = P.childrenFormatted, q = P.childrenAreDisabled, T = P.filteredValues;
        var _ = T[0];
        var V = (0, n.useMemo)((function() {
            return h == null || T.includes(A);
        }), [ T, A, h ]);
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ T, A, V ]);
        var N = (0, n.useCallback)((function(e, r) {
            if (h !== r.value) {
                p === null || p === void 0 ? void 0 : p(e, I({
                    name: B
                }, r));
                if (!w) {
                    k(r.value);
                }
            }
        }), [ w, B, p, h ]);
        var $ = (0, n.useCallback)((function(e, r) {
            var n = r.value, t = r.label;
            var o = e.key;
            var a = e.currentTarget;
            var i = null;
            var l = 0;
            var d = T.indexOf(n);
            if (o === "ArrowDown" || o === "ArrowRight") {
                e.preventDefault();
                var u = U("forward", T.length, a, d), c = u.nextTargetRes, s = u.nextIndexRes;
                i = c;
                l = s;
            } else if (o === "ArrowUp" || o === "ArrowLeft") {
                e.preventDefault();
                var b = U("backward", T.length, a, d), v = b.nextTargetRes, f = b.nextIndexRes;
                i = v;
                l = f;
            }
            if (i) {
                var p;
                (p = i) === null || p === void 0 ? void 0 : p.focus();
                var y = T[l];
                N(e, {
                    value: y,
                    label: t
                });
            }
        }), [ T, N ]);
        var z = (0, n.useMemo)((function() {
            return {
                disabled: d,
                error: b,
                firstValue: _,
                name: B,
                onClick: N,
                onKeyDown: $,
                selectedValue: V ? A : undefined
            };
        }), [ d, b, _, N, $, B, V, A ]);
        var W = d || q;
        
        return t().createElement(D, E({
            flex: true,
            $disabled: W,
            $error: b,
            ref: s,
            "data-test": "radio-bar",
            "data-test-value": A,
            "data-test-disabled": W ? "disabled" : undefined,
            role: g,
            "aria-labelledby": v,
            "aria-describedby": i,
            "aria-disabled": W || undefined,
            "aria-required": y,
            "aria-invalid": g === "radiogroup" ? b : undefined
        }, a()(O, "onChange")),  t().createElement(S.Provider, {
            value: z
        }, R));
    }
    X.Option = A;
    X.propTypes = L;
    X.componentType = "RadioBar";
    /* harmony default export */    const G = X;
    // only for styled components using RadioBar
        // CONCATENATED MODULE: ./src/RadioBar/index.ts
    module.exports = r;
    /******/})();
