// @splunk/react-ui/Resize.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ X
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const a = require("@splunk/react-ui/EventListener");
    var s =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const c = require("@splunk/react-ui/ScreenReaderContent");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const l = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const f = require("styled-components");
    var d =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const v = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Resize/ResizeStyles.ts
    var h = "12px";
    var p = d().div.withConfig({
        displayName: "ResizeStyles__Styled",
        componentId: "sc-181mwes-0"
    })([ "", ";position:relative;overflow-x:hidden;overflow-y:hidden;" ], v.mixins.reset("block"));
    var m = d().svg.withConfig({
        displayName: "ResizeStyles__StyledCornerSvg",
        componentId: "sc-181mwes-1"
    })([ "display:block;stroke:currentColor;" ]);
    var w = d().svg.withConfig({
        displayName: "ResizeStyles__StyledSideSvg",
        componentId: "sc-181mwes-2"
    })([ "display:block;fill:currentColor;" ]);
    var y = d().div.withConfig({
        displayName: "ResizeStyles__StyledBorder",
        componentId: "sc-181mwes-3"
    })([ "", ";height:100%;width:100%;border:0 solid ", ";", " ", " ", " ", "" ], v.mixins.reset("block"), v.variables.borderColorWeak, (function(e) {
        var t = e.$borderSides;
        return t.n && (0, f.css)([ "border-top-width:", ";" ], h);
    }), (function(e) {
        var t = e.$borderSides;
        return t.e && (0, f.css)([ "border-right-width:", ";" ], h);
    }), (function(e) {
        var t = e.$borderSides;
        return t.s && (0, f.css)([ "border-bottom-width:", ";" ], h);
    }), (function(e) {
        var t = e.$borderSides;
        return t.w && (0, f.css)([ "border-left-width:", ";" ], h);
    }));
    var b = (0, f.css)([ "", ";color:", ";", " &:focus{box-shadow:inset 0 0 0 1px ", ";}", "" ], v.mixins.reset("block"), v.variables.contentColorDefault, (function(e) {
        var t = e.$isSeparator, r = e.$cardinalDirection;
        var n = [ "n", "e", "w", "s" ].includes(r);
        return t && n ? (0, f.css)([ "&:hover,&:focus,&:active{color:", ";background-color:", ";}" ], v.variables.contentColorAccentStrong, v.variables.borderColorWeak) : (0, 
        f.css)([ "&:hover,&:focus,&:active{color:", ";background-color:", ";}" ], v.variables.contentColorAccentStrong, v.variables.actionColorBackgroundSubtleHover);
    }), v.variables.focusColor, (function(e) {
        var t = e.$showHandles;
        return t === "on-hover" && (0, f.css)([ "opacity:0;transition:opacity 200ms;", ":hover > div > &,&:focus{opacity:1;}" ], /* sc-sel */ p);
    }));
    var g = (0, f.css)([ "", ";width:", ";height:", ";position:absolute;background-clip:content-box;z-index:2;" ], b, h, h);
    var S = (0, f.css)([ "height:100%;" ]);
    var x = (0, f.css)([ "width:100%;display:flex;justify-content:center;" ]);
    var E = (0, f.css)([ "", ";position:absolute;z-index:1;", "" ], b, (function(e) {
        var t = e.$isSeparator;
        return t && (0, v.pickVariant)("$cardinalDirection", {
            n: x,
            s: x,
            e: S,
            w: S
        });
    }));
    var z = function e(t) {
        var r = t.$isSeparator;
        return r ? "0" : "calc(50% - ".concat(h, " / 2)");
    };
    var k = d().button.withConfig({
        displayName: "ResizeStyles__StyledResize",
        componentId: "sc-181mwes-4"
    })([ "", "" ], (0, v.pickVariant)("$cardinalDirection", {
        nw: (0, f.css)([ "", ";top:0;left:0;cursor:nwse-resize;" ], g),
        n: (0, f.css)([ "", ";cursor:ns-resize;top:0;left:", ";" ], E, z),
        ne: (0, f.css)([ "", ";right:0;top:0;cursor:nesw-resize;> ", "{transform:rotate(90deg);}" ], g, /* sc-sel */ m),
        w: (0, f.css)([ "", ";cursor:ew-resize;left:0;top:", ";" ], E, z),
        e: (0, f.css)([ "", ";cursor:ew-resize;right:0;top:", ";" ], E, z),
        sw: (0, f.css)([ "", ";left:0;bottom:0;cursor:nesw-resize;> ", "{transform:rotate(-90deg);}" ], g, /* sc-sel */ m),
        s: (0, f.css)([ "", ";cursor:ns-resize;bottom:0;left:", ";" ], E, z),
        se: (0, f.css)([ "", ";right:0;bottom:0;cursor:nwse-resize;> ", "{transform:rotate(180deg);}" ], g, /* sc-sel */ m)
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var R = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function A() {
        var e = typeof document !== "undefined" ? document : R;
        return e;
    }
    var C = A();
    /* harmony default export */    const O = /* unused pure expression or super */ null && C;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var _ = {
        document: R,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function $() {
        var e = typeof window !== "undefined" ? window : _;
        return e;
    }
    var D = $();
    /* harmony default export */    const j = D;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function q(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Resize/Resize.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function T(e, t) {
        return P(e) || N(e, t) || H(e, t) || L();
    }
    function L() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function H(e, t) {
        if (e) {
            if ("string" == typeof e) return M(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? M(e, t) : void 0;
        }
    }
    function M(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function N(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, i, a, s = [], c = !0, u = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    c = !1;
                } else for (;!(c = (n = i.call(r)).done) && (s.push(n.value), s.length !== t); c = !0) {
                }
            } catch (e) {
                u = !0, o = e;
            } finally {
                try {
                    if (!c && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (u) throw o;
                }
            }
            return s;
        }
    }
    function P(e) {
        if (Array.isArray(e)) return e;
    }
    function B(e, t) {
        if (null == e) return {};
        var r, n, o = U(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function U(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */
    /** @public */    var W = {
        appearance: i().oneOf([ "border", "overlay", "separator" ]),
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        keyIncrement: i().number,
        onRequestResize: i().func.isRequired,
        resizeHandles: i().arrayOf(i().oneOf([ "nw", "n", "ne", "w", "e", "sw", "s", "se" ])).isRequired,
        showHandles: i().oneOf([ "always", "on-hover" ])
    };
    var F = [ "nw", "n", "ne", "w", "e", "sw", "s", "se" ];
    var K = {
        nw: (0, l._)("Resize northwest"),
        n: (0, l._)("Resize north"),
        ne: (0, l._)("Resize northeast"),
        w: (0, l._)("Resize west"),
        e: (0, l._)("Resize east"),
        sw: (0, l._)("Resize southwest"),
        s: (0, l._)("Resize south"),
        se: (0, l._)("Resize southeast")
    };
    /**
 * Resize is a utility container with drag handles for resizing.
 */    function V(e) {
        var t = e.appearance, o = t === void 0 ? "overlay" : t, i = e.children, a = e.elementRef, c = e.keyIncrement, l = c === void 0 ? 10 : c, f = e.onRequestResize, d = e.resizeHandles, v = e.showHandles, h = v === void 0 ? "always" : v, b = B(e, [ "appearance", "children", "elementRef", "keyIncrement", "onRequestResize", "resizeHandles", "showHandles" ]);
        // @docs-props-type ResizePropsBase
                (0, r.useEffect)((function() {
            if (false) {}
        }), [ o, d ]);
        var g = (0, r.useState)(), S = T(g, 2), x = S[0], E = S[1];
        var z = (0, r.useState)(null), R = T(z, 2), A = R[0], C = R[1];
        var O = (0, r.useState)(), _ = T(O, 2), $ = _[0], D = _[1];
        var L = (0, r.useState)(), H = T(L, 2), M = H[0], N = H[1];
        var P = (0, r.useState)(), U = T(P, 2), W = U[0], V = U[1];
        var X = (0, r.useState)(), Y = T(X, 2), G = Y[0], J = Y[1];
        var Q = (0, r.useCallback)((function(e) {
            C(e);
            q(a, e);
        }), [ a ]);
        var Z = (0, r.useCallback)((function() {
            E(undefined);
        }), [ E ]);
        var ee = (0, r.useCallback)((function(e, t) {
            if (e.button > 0) {
                return;
            }
            e.preventDefault();
            e.persist();
            E(t);
            N(A === null || A === void 0 ? void 0 : A.offsetWidth);
            D(A === null || A === void 0 ? void 0 : A.offsetHeight);
            V(e.clientX);
            J(e.clientY);
        }), [ A ]);
        var te = (0, r.useMemo)((function() {
            var e = {};
            F.forEach((function(t) {
                e["handleDragStart".concat(t)] = function(e) {
                    ee(e, t);
                };
            }));
            return e;
        }), [ ee ]);
        var re = (0, r.useCallback)((function(e) {
            var t = (x === null || x === void 0 ? void 0 : x.match(/(w|e)$/)) ? e.clientX - (W || 0) : 0;
            var r = (x === null || x === void 0 ? void 0 : x.match(/^(s|n)/)) ? e.clientY - (G || 0) : 0;
            var n = {
                width: (x === null || x === void 0 ? void 0 : x.match(/w$/)) ? (M || 0) - t : (M || 0) + t,
                height: (x === null || x === void 0 ? void 0 : x.match(/^n/)) ? ($ || 0) - r : ($ || 0) + r
            };
            f(e, n);
        }), [ x, f, $, M, W, G ]);
        var ne = (0, r.useCallback)((function(e, t) {
            var r = e.key;
            var n = t.slice(-1);
            var o = {
                e: {
                    ArrowRight: 1,
                    ArrowLeft: -1
                },
                w: {
                    ArrowRight: -1,
                    ArrowLeft: 1
                }
            };
            var i = (n === "e" || n === "w") && (r === "ArrowRight" || r === "ArrowLeft") ? o[n][r] : 0;
            var a = t.charAt(0);
            var s = {
                n: {
                    ArrowDown: -1,
                    ArrowUp: 1
                },
                s: {
                    ArrowDown: 1,
                    ArrowUp: -1
                }
            };
            var c = (a === "n" || a === "s") && (r === "ArrowUp" || r === "ArrowDown") ? s[a][r] : 0;
            if ((i || c) && A) {
                e.preventDefault();
                var u = {
                    width: A.offsetWidth + i * l,
                    height: A.offsetHeight + c * l
                };
                f(e, u);
            }
        }), [ A, l, f ]);
        var oe = (0, r.useMemo)((function() {
            var e = {};
            F.forEach((function(t) {
                e["handleKeyDown".concat(t)] = function(e) {
                    ne(e, t);
                };
            }));
            return e;
        }), [ ne ]);
        var ie = (0, r.useCallback)((function(e) {
            var t = e.length > 1;
            var r;
            if (t) {
                r =  n().createElement(m, {
                    viewBox: "0 0 12 12",
                    width: "12px",
                    height: "12px"
                },  n().createElement("path", {
                    d: "M10.7783 4L7.24279 7.53553L3.70725 11.0711"
                }),  n().createElement("path", {
                    d: "M7.94971 3.5L5.47483 5.97487L2.99996 8.44975"
                }));
            } else if (e === "n" || e === "s") {
                r =  n().createElement(w, {
                    viewBox: "0 0 12 12",
                    width: "12px",
                    height: "12px"
                },  n().createElement("rect", {
                    width: "12",
                    height: "1",
                    x: "0",
                    y: "4",
                    rx: "0.5",
                    ry: "0.5"
                }),  n().createElement("rect", {
                    width: "12",
                    height: "1",
                    x: "0",
                    y: "7",
                    rx: "0.5",
                    ry: "0.5"
                }));
            } else if (e === "e" || e === "w") {
                r =  n().createElement(w, {
                    viewBox: "0 0 12 12",
                    width: "12px",
                    height: "12px"
                },  n().createElement("rect", {
                    width: "1",
                    height: "12",
                    x: "4",
                    y: "0",
                    rx: "0.5",
                    ry: "0.5"
                }),  n().createElement("rect", {
                    width: "1",
                    height: "12",
                    x: "7",
                    y: "0",
                    rx: "0.5",
                    ry: "0.5"
                }));
            }
            
            return n().createElement(k, {
                key: e,
                "data-test": "resize-handle-".concat(e),
                onKeyDown: oe["handleKeyDown".concat(e)],
                onMouseDown: te["handleDragStart".concat(e)],
                $cardinalDirection: e,
                $isSeparator: o === "separator",
                $showHandles: h
            }, r,  n().createElement(u(), null, K[e]));
        }), [ o, te, oe, h ]);
        /* When appearance is border, determine which sides need margin.
   * Converts ['se', 'n'] to { e: true, s: true, n: true}
   */        var ae = (0, r.useMemo)((function() {
            var e = {};
            if (o === "border") {
                d.join("").split("").filter((function(e, t, r) {
                    return r.indexOf(e) === t;
                })).forEach((function(t) {
                    e[t] = true;
                }));
            }
            return e;
        }), [ o, d ]);
        // Handles are inserted before and after children to optimize tab order
                var se = F.slice(0, 4).filter((function(e) {
            return d.indexOf(e) > -1;
        })).map(ie);
        var ce = F.slice(4).filter((function(e) {
            return d.indexOf(e) > -1;
        })).map(ie);
        
        return n().createElement(p, I({
            "data-test": "resize",
            ref: Q
        }, b),  n().createElement(y, {
            $borderSides: ae,
            $isSeparator: o === "separator"
        }, A && se, i, A && ce, x &&  n().createElement(n().Fragment, null,  n().createElement(s(), {
            target: j,
            eventType: "mousemove",
            listener: re
        }),  n().createElement(s(), {
            target: j,
            eventType: "mouseup",
            listener: Z
        }))));
    }
    V.propTypes = W;
    /* harmony default export */    const X = V;
    // CONCATENATED MODULE: ./src/Resize/index.ts
    module.exports = t;
    /******/})();
