// @splunk/react-ui/ResultsMenu.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        VirtualizedResultsMenu: () => /* reexport */ Re,
        default: () => /* reexport */ B
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const i = require("@splunk/react-ui/Divider");
    var u =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const l = require("@splunk/react-ui/Menu");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const s = require("@splunk/react-ui/ScreenReaderContent");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const d = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const v = require("styled-components");
    var m =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/WaitSpinner"
    const b = require("@splunk/react-ui/WaitSpinner");
    var p =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/ResultsMenu/ResultsMenuStyles.ts
    var g = m()(c()).withConfig({
        displayName: "ResultsMenuStyles__StyledMenu",
        componentId: "avbhl8-0"
    })([ "overflow:auto;flex-direction:column;" ]);
    var h = m().div.withConfig({
        displayName: "ResultsMenuStyles__Styled",
        componentId: "avbhl8-1"
    })([ "", ";flex-direction:column;max-height:calc(100vh - 20px);border-radius:", ";background-color:", ";& > ", ":not(:first-child){border-start-start-radius:0;border-start-end-radius:0;}& > ", ":not(:last-child){border-end-start-radius:0;border-end-end-radius:0;}" ], y.mixins.reset("flex"), y.variables.borderRadius, y.variables.backgroundColorPopup, g, g);
    var S = m().div.withConfig({
        displayName: "ResultsMenuStyles__StyledFooter",
        componentId: "avbhl8-2"
    })([ "padding:", " ", ";color:", ";" ], y.variables.spacingSmall, y.variables.spacingLarge, y.variables.contentColorDefault);
    var E = m().li.withConfig({
        displayName: "ResultsMenuStyles__StyledLoading",
        componentId: "avbhl8-3"
    })([ "", ";padding:", " ", ";gap:", ";" ], y.mixins.reset("flex"), y.variables.spacingSmall, y.variables.spacingLarge, y.variables.spacingXSmall);
    var O = m()(c().Item).withConfig({
        displayName: "ResultsMenuStyles__StyledNoOptions",
        componentId: "avbhl8-4"
    })([ "&[disabled],&[aria-disabled='true']{color:", ";}" ], y.variables.contentColorDefault);
    var M = m()(p()).withConfig({
        displayName: "ResultsMenuStyles__StyledWait",
        componentId: "avbhl8-5"
    })([ "flex:0 0 auto;" ]);
    var w = m().div.withConfig({
        displayName: "ResultsMenuStyles__StyledLoadingMessage",
        componentId: "avbhl8-6"
    })([ "flex:1 0 0;color:", ";" ], y.variables.contentColorDefault);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function C(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/ResultsMenu/ResultsMenu.tsx
    function k() {
        return k = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, k.apply(null, arguments);
    }
    function j(e, r) {
        return x(e) || T(e, r) || R(e, r) || I();
    }
    function I() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function R(e, r) {
        if (e) {
            if ("string" == typeof e) return P(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? P(e, r) : void 0;
        }
    }
    function P(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, t = Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function T(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var t, o, a, i, u = [], l = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    l = !1;
                } else for (;!(l = (t = a.call(n)).done) && (u.push(t.value), u.length !== r); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function x(e) {
        if (Array.isArray(e)) return e;
    }
    function A(e, r) {
        if (null == e) return {};
        var n, t, o = L(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function L(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var _ = {
        animateLoading: a().bool,
        children: a().node,
        childrenStart: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        focusMode: a().oneOf([ "roving", "normal", "never" ]),
        footerMessage: a().node,
        isLoading: a().bool,
        loadingMessage: a().node,
        menuId: a().string,
        noOptionsMessage: a().node,
        onScroll: a().func,
        onScrollBottom: a().func,
        placement: a().string
    };
    var q = (0, d._)("Loading...");
    var D = (0, d._)("No matches");
    // The remaining amount of pixels when scrolled from bottom of menu to trigger onScrollBottom().
        var N = 400;
    var H = 28;
    function F(e) {
        var r = e.animateLoading, o = e["aria-multiselectable"], a = e.children, i = e.childrenStart, c = e.elementRef, s = e.focusMode, d = e.footerMessage, v = e.isLoading, m = e.labelledBy, b = e.loadingMessage, p = b === void 0 ? q : b, y = e.menuId, I = e.noOptionsMessage, R = I === void 0 ? D : I, P = e.onScroll, T = e.onScrollBottom, x = e.placement, L = e.style, _ = e.tabIndex, F = A(e, [ "animateLoading", "aria-multiselectable", "children", "childrenStart", "elementRef", "focusMode", "footerMessage", "isLoading", "labelledBy", "loadingMessage", "menuId", "noOptionsMessage", "onScroll", "onScrollBottom", "placement", "style", "tabIndex" ]);
        // @docs-props-type ResultsMenuPropsBase
                var B = (0, n.useState)(undefined), U = j(B, 2), z = U[0], V = U[1];
        var K = (0, n.useState)(null), W = j(K, 2), $ = W[0], G = W[1];
        var X = (0, n.useState)(null), J = j(X, 2), Q = J[0], Y = J[1];
        var Z = (0, n.useState)(0), ee = j(Z, 2), re = ee[0], ne = ee[1];
        var te = (0, n.useState)(0), oe = j(te, 2), ae = oe[0], ie = oe[1];
        var ue = (0, n.useState)(false), le = j(ue, 2), ce = le[0], se = le[1];
        var fe = (0, n.useCallback)((function(e) {
            Y(e);
        }), []);
        var de = (0, n.useCallback)((function(e) {
            G(e);
            C(c, e);
        }), [ c ]);
        var ve = (0, n.useCallback)((function() {
            ie(document.documentElement.scrollTop);
        }), []);
        var me = (0, n.useCallback)((function() {
            ie(document.documentElement.scrollTop);
        }), []);
        var be = (0, n.useCallback)((function(e) {
            // Safety net to ensure window doesn't scroll if menu is scrolled pass the numberOfItemsLoaded at high velocity.
            e.stopPropagation();
            document.documentElement.scrollTop = ae;
        }), [ ae ]);
        var pe = (0, n.useCallback)((function() {
            return !!Q && Q.scrollHeight === Q.offsetHeight;
        }), [ Q ]);
        var ye = (0, n.useCallback)((function(e) {
            if (!ce) {
                se(true);
                T === null || T === void 0 ? void 0 : T(e);
            }
        }), [ ce, T ]);
        var ge = (0, n.useCallback)((function(e) {
            if (T && e.target && Q) {
                var r = Q.scrollHeight - Q.offsetHeight - N;
                // Adding 1 due to border of menu item.
                                if (Q.scrollTop + 1 >= r) {
                    ye(e);
                }
            }
            P === null || P === void 0 ? void 0 : P(e);
        }), [ ye, Q, P, T ]);
        var he = (0, n.useCallback)((function() {
            var e = n.Children.count(a);
            // If menu is full length, load more items.
                        if (pe()) {
                ye(null);
            }
            // Avoid triggering this logic if this.state.childrenCount is null
            // because that's not a real change in the number of children
            // and thus we want to avoid resetting the value of scrollBottomTriggered
                        if (z != null && e !== z) {
                ne(e - (z !== null && z !== void 0 ? z : 0));
                se(false);
            }
            if (e !== z) {
                V(e);
            }
        }), [ pe, a, z, ye ]);
        (0, n.useEffect)((function() {
            if (!$ || !Q) {
                return;
            }
            // If onScrollBottom is defined, determine if it should be triggered.
                        if (T && a) {
                he();
            }
        }), [ $, Q, a, T, he ]);
        // Assumption: that you cannot be filtered if you are a result
                var Se = n.Children.toArray(a).filter(n.isValidElement).some((function(e) {
            var r = e.type;
            return !(r.as === "Divider" && r.filterFirst || (r.as === "Divider" || r.as === "Heading") && (r.filterLast || r.filterConsecutive));
        }));
        var Ee = pe();
        var Oe = (0, n.useMemo)((function() {
            return {
                height: ce ? re * H : 0
            };
        }), [ ce, re ]);
        var Me = !Se && !!R && !v;
        var we = (0, n.useCallback)((function() {
            return {
                role: "listbox",
                preventFocus: s === "never"
            };
        }), [ s ]);
        var Ce = (0, n.useCallback)((function() {
            var e = !!n.Children.toArray(a).length;
            return d && e &&  t().createElement(t().Fragment, null, x !== "above" &&  t().createElement(u(), null),  t().createElement(S, {
                "data-test": "footer-message",
                key: "footer"
            }, d), x === "above" &&  t().createElement(u(), null));
        }), [ d, a, x ]);
        
        return t().createElement(h, k({
            "data-test": "results-menu",
            key: "wrapper",
            ref: de,
            onWheel: T ? be : undefined,
            onMouseEnter: T ? ve : undefined,
            onMouseLeave: T ? me : undefined
        }, F, {
            style: L
        }), x !== "above" && i, x === "above" && Ce(),  t().createElement(l.MenuContext.Provider, {
            value: we()
        },  t().createElement(g, {
            "aria-multiselectable": o,
            "aria-labelledby": m,
            elementRef: fe,
            focusMode: s,
            id: y,
            key: "menu",
            onScroll: ge,
            stopScrollPropagation: true,
            tabIndex: _
        }, Me &&  t().createElement(O, {
            "data-test": "no-results-message",
            disabled: true
        }, R), a, T && !Ee && 
        
        // Bottom spacer fills in the space of new items being loaded by using the minimum possible height x menuItems.
        t().createElement("div", {
            "data-test": "results-menu-bottom-spacer",
            style: Oe
        }), v &&  t().createElement(E, null, r &&  t().createElement(M, null),  t().createElement(w, null, p)))),  t().createElement(f(), {
            "aria-live": "polite"
        }, Me && R), x !== "above" && Ce(), x === "above" && i);
    }
    F.propTypes = _;
    /* harmony default export */    const B = F;
    // CONCATENATED MODULE: external "lodash/omit"
    const U = require("lodash/omit");
    var z =  e.n(U);
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const V = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: ./src/ResultsMenu/VirtualizedResultsMenu/groupChildren.ts
    // Cheap(er than processing potentially thousands of array elements) memoization
    var K = {
        result: []
    };
    /**
 * Given component children, restructure them into an array of arrays, each of length `groupSize`.
 *
 * The last array is not guaranteed to be full, it will have a length in the range `[1, groupSize]`
 *
 * @example
 * ```js
 * groupChildren([1, 2, 3, 4, 5, 6, 7], 2);
 * // > [[1, 2], [3, 4], [5, 6], [7]]
 * ```
 * @param {Array} children One node, or an array of nodes, to be divided into groups
 * @param {number} groupSize The size of each group to be created
 * @returns A 2d array where each inner-array is a group of size `groupSize` of children
 * and the order of the `children` received equals the order of the flattened result
 */    var W = function e(r, t) {
        if (K.children === r && K.groupSize === t) {
            return K.result;
        }
        K.children = r;
        K.groupSize = t;
        if (t <= 1) {
            if (false) {}
            // don't break in non-__DEV__ and return one big group
                        K.result = [ n.Children.toArray(r) ];
        } else {
            K.result = n.Children.toArray(r).reduce((function(e, r, n) {
                var o = Math.floor(n / t);
                if (!Array.isArray(e[o])) {
                    // eslint-disable-next-line no-param-reassign
                    e[o] = [];
                }
                e[o].push(r);
                return e;
            }), []);
        }
        return K.result;
    };
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const $ = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/ResultsMenu/VirtualizedResultsMenu/VirtualizedItem.ts
    // This wrapper should only be used with Menu.Item/derivatives thereof, so omitting Heading/Divider is fine
    var G = {
        children: a().node,
        onMount: a().func.isRequired,
        onUnmount: a().func.isRequired
    };
    /** @private Used by `VirtualizedResultsMenu` */    var X =  t().forwardRef((function(e, r) {
        var t = e.children, o = e.onMount, a = e.onUnmount;
        // @docs-props-type VirtualizedItemPropsBase
                var i = (0, n.useRef)(null);
        (0, n.useEffect)((function() {
            var e = i.current;
            if (e) {
                o(e);
            }
            return function() {
                if (e) {
                    a(e);
                }
            };
        }), [ o, a ]);
        var u = n.Children.only(t);
        if (!u) {
            if (false) {}
            return null;
        }
        
        return (0, n.cloneElement)(u, {
            elementRef: function e(n) {
                i.current = n;
                C(r, n);
            }
        });
    }));
    X.propTypes = G;
    X.as = "Item";
    // CONCATENATED MODULE: ./src/ResultsMenu/VirtualizedResultsMenu/injectVirtualizedItem.tsx
    /**
 * Find the index of an array element which is nearest to targetIndex and is a Menu.Item component
 * @param {Array} elements An array to search
 * @param {Number} targetIndex The target index. This is the ideal value to return.
 * @returns The index in the array of the nearest `Menu.Item` component to `targetIndex`, or -1 if no element is a `Menu.Item` component
 */
    var J = function e(r, n) {
        var t = -1;
        var o = -1;
        r.forEach((function(e, r) {
            var a = Math.abs(n - r);
            if (t >= 0 && t < a) {
                return;
            }
            if (e.type !== l.Divider && e.type !== l.Heading) {
                t = a;
                o = r;
            }
        }));
        return o;
    };
    /**
 * @private Keep the injection of `VirtualizedItem` DRY. Will always replace the middle-most non-divider/heading
 * with a VirtualizedItem instance.
 * - If `trackFirstElement` the **FIRST** non-divider/heading item will also be replaced.
 * - If `trackLastElement` the **LAST** non-divider/heading item will also be replaced.
 */    var Q = function e(r, n, o) {
        var a = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : {}, i = a.trackFirstElement, u = a.trackLastElement;
        if (!Array.isArray(r)) {
            return [];
        }
        var l = J(r, Math.floor(r.length / 2));
        var c = i && J(r, 0);
        var s = u && J(r, r.length - 1);
        return r.map((function(e, r) {
            if (r === l || r === c || r === s) {
                
                return t().createElement(X, {
                    key: (0, $.createGUID)(),
                    onMount: n,
                    onUnmount: o
                }, e);
            }
            return e;
        }));
    };
    // CONCATENATED MODULE: ./src/ResultsMenu/VirtualizedResultsMenu/VirtualizedResultsMenu.tsx
    function Y(e) {
        "@babel/helpers - typeof";
        return Y = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Y(e);
    }
    function Z() {
        return Z = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Z.apply(null, arguments);
    }
    function ee(e) {
        return te(e) || ne(e) || ie(e) || re();
    }
    function re() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ne(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function te(e) {
        if (Array.isArray(e)) return ue(e);
    }
    function oe(e, r) {
        return ce(e) || le(e, r) || ie(e, r) || ae();
    }
    function ae() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ie(e, r) {
        if (e) {
            if ("string" == typeof e) return ue(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? ue(e, r) : void 0;
        }
    }
    function ue(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, t = Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function le(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var t, o, a, i, u = [], l = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    l = !1;
                } else for (;!(l = (t = a.call(n)).done) && (u.push(t.value), u.length !== r); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function ce(e) {
        if (Array.isArray(e)) return e;
    }
    function se(e, r) {
        if (null == e) return {};
        var n, t, o = fe(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function fe(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function de(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function ve(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? de(Object(n), !0).forEach((function(r) {
                me(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : de(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function me(e, r, n) {
        return (r = be(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function be(e) {
        var r = pe(e, "string");
        return "symbol" == Y(r) ? r : r + "";
    }
    function pe(e, r) {
        if ("object" != Y(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != Y(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    // eslint-disable-next-line no-relative-imports-of-published -- TODO(SUI-8190): VirtualizedResultsMenu wraps ResultsMenu, using external import creates circular dependency
    // TODO(SUI-5920): add transitions to this component in React 18
        var ye = ve(ve({}, z()(_, "onDownKeyPress", "onEndKeyPress", "onHomeKeyPress", "onUpKeyPress")), {}, {
        virtualization: a().number
    });
    var ge = [];
    /**
 * A wrapper for `ResultsMenu` which virtualizes the `children`. The received `children` array will be split into **"panes"** each with `virtualization` entries
 * and, at any given time, 3 or fewer of these panes will be rendered in the DOM.
 *
 * Given the children `[1, 2, 3, 4, 5, 6, 7, 8, 9]` and `virtualization=4`, then at first only elements `[1, 2, 3, 4, 5, 6, 7, 8]` will be added to the DOM.
 * Once either child `6` or `8` intersect the `Menu` rendered by `ResultsMenu` the next pane of children will also be included and, if needed, the oldest pane
 * will be removed.
 *
 * In effect this means the menu will show the "central" content at pane index _`n`_, a previous-content-buffer comprised of pane index _`n - 1`_,
 * and an upcoming-content-buffer comprised of pane index _`n + 1`_.
 *
 * An `IntersectionObserver` will observe the first node in the _`n - 1`_ pane, the last node in the _`n + 1`_ pane, and the central node in both _`n ± 1`_ panes
 * for intersections with the rendered `Menu` and trigger a *pane change* upon that intersection. As such, in SSR or other environments which don't support `IntersectionObserver`
 * this component should not be used and will be replaced by the barrel file with a vanilla `ResultsMenu`.
 *
 * @throws in `__DEV__` when `virtualization` is too small as compared to the menu height (`virtualization` must be greater than the number of visible items in the rendered menu)
 * @throws in `__DEV__` when `virtualization` is less than or equal to 1 (`virtualization` must be ≥ 2)
 * @throws in `__DEV__` when `virtualization` is changed during the lifecycle of the `VirtualizedResultsMenu` component
 */    function he(e) {
        var r = e.virtualization, o = r === void 0 ? 2 : r, a = e.elementRef, i = e.children, u = e.focusMode, l = u === void 0 ? "normal" : u, c = se(e, [ "virtualization", "elementRef", "children", "focusMode" ]);
        // @docs-props-type VirtualizedResultsMenuPropsBase
                var s = (0, n.useRef)(o);
        // If a user tries to set virtualization to a value smaller than the number of options
        // visible at a given moment then the menu will flicker endlessly. This state is set in
        // the IntersectionObserver to fix the issue, but in `__DEV__` an error is thrown.
                var f = (0, n.useState)(o), d = oe(f, 2), v = d[0], m = d[1];
 // these refs keep track of whether we need to focus the first/last
        // element after the next render
                var b = (0, n.useRef)(false);
        var p = (0, n.useRef)(false);
        var y = (0, n.useRef)(i);
        // this code makes the "Home" and "End" keys work by injecting
        // an elementRef callback function into the first and last item
        // that will attempt to focus the first/last item if appropriate
        // we need to do this because if we call focus() right after changePane()
        // the new pane won't be in the DOM yet, so the wrong element will end up focused
                (0, n.useMemo)((function() {
            var e = n.Children.toArray(i);
            var r = J(e, 0);
            var t = J(e, e.length - 1);
            y.current = e.map((function(e, o) {
                var a = e;
                if (o === r) {
                    a =  (0, n.isValidElement)(e) ?  (0, n.cloneElement)(e, {
                        elementRef: function e(r) {
                            if (b === null || b === void 0 ? void 0 : b.current) {
                                b.current = false;
                                r === null || r === void 0 ? void 0 : r.focus();
                            }
                        }
                    }) : e;
                } else if (o === t) {
                    a =  (0, n.isValidElement)(e) ?  (0, n.cloneElement)(e, {
                        elementRef: function e(r) {
                            if (p === null || p === void 0 ? void 0 : p.current) {
                                p.current = false;
                                r === null || r === void 0 ? void 0 : r.focus();
                            }
                        }
                    }) : e;
                }
                return a;
            }));
        }), [ i ]);
        var g = y.current;
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ o ]);
        // `useRef` will recompute its initial value every render cycle.
        // To avoid the array processing we can provide an initialization function to useState and init the ref with the result
                var h = (0, n.useState)((function() {
            return W(g, v);
        })), S = oe(h, 1), E = S[0];
 // NOSONAR
        // `windowPanes` is an array of arrays, each of length `virtualization`. Each "pane" is just a subset of
        // `children` which will be rendered in a chunk, and at most 3 of these panes will render at any given time.
                var O = (0, n.useRef)(E);
        var M = (0, n.useRef)(0);
        var w = (0, n.useRef)(null);
        // An IntersectionObserver is used to detect when pane changes are needed.
        // When loadPrevPaneRef or loadNextPaneRef entries intersect the menu, the currentPaneId will be updated.
                var k = (0, n.useRef)();
        var j = (0, n.useRef)(ge);
        var I = (0, n.useRef)(ge);
        var R = (0, n.useCallback)((function() {
            var e;
            var r = O.current, n = r === void 0 ? [] : r;
            var t = M.current;
            /**
     * create onMount for an child of the menu which will be used to trigger pane
     * changes upon intersection with the menu's visible bounds
     */            var o = function e(r) {
                return function(e) {
                    var n;
                    if (!e) {
                        return;
                    }
                    r.current.push(e);
                    (n = k.current) === null || n === void 0 ? void 0 : n.observe(e);
                };
            };
            /**
     * onUnmount to cleanup the operations of `onVirtualizeMount`
     */            var a = function e(r) {
                var n;
                if (!r) {
                    return;
                }
                var t = I.current.indexOf(r);
                if (t >= 0) {
                    I.current.splice(t, 1);
                }
                var o = j.current.indexOf(r);
                if (o) {
                    j.current.splice(o, 1);
                }
                (n = k.current) === null || n === void 0 ? void 0 : n.unobserve(r);
            };
            return [].concat(ee(Q(n[t - 1], o(I), a, {
                trackFirstElement: true
            })), ee((e = n[t]) !== null && e !== void 0 ? e : []), ee(Q(n[t + 1], o(j), a, {
                trackLastElement: true
            })));
        }), []);
        var P = (0, n.useState)(R), T = oe(P, 2), x = T[0], A = T[1];
        var L = (0, n.useCallback)((function(e) {
            var r = M.current;
            M.current = Math.max(0, Math.min(e, O.current.length - 1));
            if (r === M.current) {
                // no change in current pane
                return;
            }
            A(R);
        }), [ R ]);
        // Link an IntersectionObserver instance with the menu children to be observed
                var _ = (0, n.useCallback)((function(e) {
            var r;
            C(a, e);
            w.current = e;
            // This is overly defensive and could probably be removed
                        (r = k.current) === null || r === void 0 ? void 0 : r.disconnect();
            if (!e) {
                return;
            }
            I.current = [];
            j.current = [];
            // Run the observation flow
                        k.current = new IntersectionObserver((function(r) {
                var n;
                var t = M.current;
                var o = e.firstElementChild;
                if (!!o && o.scrollHeight < e.clientHeight) {
                    var a;
                    // If the first element doesn't require a scrollbar then it's likely a filter input
                                        o = (a = o) === null || a === void 0 ? void 0 : a.nextElementSibling;
                }
                if (((n = o) === null || n === void 0 ? void 0 : n.scrollTop) === 0 && t === 0 && O.current.length > 1 && r.some((function(e) {
                    return e.isIntersecting;
                }))) {
                    // The menu hasn't scrolled, is on pane 0, and already has an intersection, the `virtualization` prop is too small
                    m((function(e) {
                        return e + 10;
                    }));
                    if (false) {}
                    return;
                }
                var i = false;
                r.forEach((function(e) {
                    if (i || !e.isIntersecting) {
                        // don't process if the pane changed or the entry isn't intersecting
                        return;
                    }
                    if (I.current.includes(e.target)) {
                        i = true;
                        L(t - 1);
                    } else if (j.current.includes(e.target)) {
                        i = true;
                        L(t + 1);
                    }
                }));
            }), {
                root: e
            });
        }), [ a, L ]);
        var q = (0, n.useCallback)((function(e) {
            var r = w.current;
            if (!r) {
                return;
            }
            e.preventDefault();
            var n = e.key;
            if (n === "Home") {
                var t = M.current === 0;
                if (t) {
                    var o;
                    // if the first pane is already rendered, find the first element and focus it
                                        var a = (0, V.getSortedTabbableElements)(r, {
                        ignoreTabIndex: true
                    });
                    var i = a[0];
                    i === null || i === void 0 ? void 0 : (o = i.focus) === null || o === void 0 ? void 0 : o.call(i);
                } else {
                    // if it's not rendered, set up focusing it after the next render
                    b.current = true;
                    // then call changePane() to trigger the render
                                        L(0);
                }
                return;
            }
            var u = O.current.length - 1;
            if (n === "End") {
                var l = M.current === u;
                if (l) {
                    var c;
                    // if the last pane is already rendered, find the first element and focus it
                                        var s = (0, V.getSortedTabbableElements)(r, {
                        ignoreTabIndex: true
                    });
                    var f = s[s.length - 1];
                    f === null || f === void 0 ? void 0 : (c = f.focus) === null || c === void 0 ? void 0 : c.call(f);
                } else {
                    // if it's not rendered, set up focusing it after the next render
                    p.current = true;
                    // then call changePane() to trigger the render
                                        L(u);
                }
                return;
            }
            var d = (0, V.getSortedTabbableElements)(r, {
                ignoreTabIndex: true
            });
            var v = e.target instanceof HTMLElement ? e.target : r === null || r === void 0 ? void 0 : r.querySelector(":focus");
            var m = v instanceof HTMLElement ? d.indexOf(v) : -1;
            (0, V.handleFocus)(n, d, m, {
                enableLoop: false,
                // VirtualizedResultsMenu does not support looping focus
                orientation: "vertical",
                enableTab: false,
                enableHomeEnd: false
            });
        }), [ L ]);
        (0, n.useEffect)((function() {
            m((function(e) {
                if (false) {}
                return o;
            }));
        }), [ o ]);
        (0, n.useEffect)((function() {
            O.current = W(g, v);
            A(R);
        }), [ g, v, R ]);
        (0, n.useEffect)((function() {
            return function() {
                var e;
                return (e = k.current) === null || e === void 0 ? void 0 : e.disconnect();
            };
        }), []);
        (0, n.useEffect)((function() {
            if (l === "never") {
                return undefined;
            }
            var e = w.current;
            if (e) {
                e.addEventListener("keydown", q);
            }
            return function() {
                e === null || e === void 0 ? void 0 : e.removeEventListener("keydown", q);
            };
        }), [ q, l ]);
        
        return t().createElement(B, Z({
            elementRef: _,
            focusMode: l
        }, c), x);
    }
    he.propTypes = ye;
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var Se = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function Ee() {
        var e = typeof document !== "undefined" ? document : Se;
        return e;
    }
    var Oe = Ee();
    /* harmony default export */    const Me = /* unused pure expression or super */ null && Oe;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var we = {
        document: Se,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(r) {
            if (typeof setTimeout === "undefined") {
                r();
                return null;
            }
            return setTimeout(r, 0);
        },
        cancelAnimationFrame: function e(r) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(r);
        }
    };
    function Ce() {
        var e = typeof window !== "undefined" ? window : we;
        return e;
    }
    var ke = Ce();
    /* harmony default export */    const je = /* unused pure expression or super */ null && ke;
    // CONCATENATED MODULE: ./src/ResultsMenu/VirtualizedResultsMenu/index.ts
    // eslint-disable-next-line no-relative-imports-of-published --  TODO(SUI-8190): VirtualizedResultsMenu wraps ResultsMenu, using external import creates circular dependency
    // In environments without IntersectionObserver support this should just export ResultsMenu
    var Ie = "IntersectionObserver" in Ce() ? he : B;
    /* harmony default export */    const Re = Ie;
    // CONCATENATED MODULE: ./src/ResultsMenu/index.ts
    module.exports = r;
    /******/})();
