// @splunk/react-ui/Scroll.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ G
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "lodash/isFinite"
    const o = require("lodash/isFinite");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const u = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "styled-components"
    const c = require("styled-components");
    var f =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: external "@react-spring/web"
    const d = require("@react-spring/web");
    // CONCATENATED MODULE: external "lodash/keys"
    const p = require("lodash/keys");
    var v =  e.n(p);
    // CONCATENATED MODULE: external "lodash/omit"
    const y = require("lodash/omit");
    var m =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const b = require("@splunk/react-ui/EventListener");
    var g =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScrollContainerContext"
    const h = require("@splunk/react-ui/ScrollContainerContext");
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const S = require("@splunk/react-ui/usePrevious");
    var w =  e.n(S);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function O(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Scroll/Inner.tsx
    function P() {
        return P = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, P.apply(null, arguments);
    }
    function j(e, t) {
        return L(e) || A(e, t) || C(e, t) || T();
    }
    function T() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function C(e, t) {
        if (e) {
            if ("string" == typeof e) return k(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? k(e, t) : void 0;
        }
    }
    function k(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function A(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, l, a, i = [], u = !0, c = !1;
            try {
                if (l = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (n = l.call(r)).done) && (i.push(n.value), i.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return i;
        }
    }
    function L(e) {
        if (Array.isArray(e)) return e;
    }
    function q(e, t) {
        if (null == e) return {};
        var r, n, o = E(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function E(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var x = {
        children: i().node,
        defaultLeft: i().number,
        defaultTop: i().number,
        elementRef: i().oneOfType([ i().func, i().object ]),
        left: i().object,
        onScroll: i().func,
        stopScrollPropagation: i().oneOf([ true, false, "window" ]),
        tagName: i().string,
        top: i().object
    };
    var I = {
        passive: false
    };
    function M(e) {
        var t = e.children, o = e.defaultLeft, a = o === void 0 ? 0 : o, i = e.defaultTop, u = i === void 0 ? 0 : i, c = e.elementRef, f = e.left, s = e.onScroll, p = e.stopScrollPropagation, y = e.tagName, b = y === void 0 ? "div" : y, S = e.top, T = q(e, [ "children", "defaultLeft", "defaultTop", "elementRef", "left", "onScroll", "stopScrollPropagation", "tagName", "top" ]);
        // @docs-props-type InnerPropsBase
                var C = (0, r.useState)("visible"), k = j(C, 2), A = k[0], L = k[1];
        var E = (0, r.useState)("visible"), M = j(E, 2), N = M[0], R = M[1];
        var X = (0, r.useState)(null), Y = j(X, 2), H = Y[0], W = Y[1];
        var _ = w()(H);
        (0, r.useEffect)((function() {
            if (H) {
                if (!_) {
                    H.scrollTop = u;
                    H.scrollLeft = a;
                }
                if (l()(S)) {
                    H.scrollTop = (S === null || S === void 0 ? void 0 : S.get()) || 0;
                }
                if (l()(f)) {
                    H.scrollLeft = (f === null || f === void 0 ? void 0 : f.get()) || 0;
                }
            }
        }), [ H, _, u, a, S, f ]);
        var D = (0, r.useCallback)((function(e) {
            var t = H;
            if (!t) {
                return;
            }
            // if the element is scrollable in both directions, it's not safe to stop propagation.
                        if (t.clientWidth !== t.scrollWidth && t.clientHeight !== t.scrollHeight) {
                return;
            }
            if (t.clientWidth !== t.scrollWidth) {
                if (e.deltaX < 0 && t.scrollLeft < Math.abs(e.deltaX)) {
                    t.scrollLeft = 0;
                    e.stopPropagation();
                    e.preventDefault();
                    return;
                }
                var r = t.scrollWidth - t.clientWidth;
                if (e.deltaX && e.deltaX + t.scrollLeft > r) {
                    t.scrollLeft = r;
                    e.stopPropagation();
                    e.preventDefault();
                }
            }
            if (t.clientHeight !== t.scrollHeight) {
                if (e.deltaY < 0 && t.scrollTop < Math.abs(e.deltaY)) {
                    t.scrollTop = 0;
                    e.stopPropagation();
                    e.preventDefault();
                    return;
                }
                var n = t.scrollHeight - t.clientHeight;
                if (e.deltaY && e.deltaY + t.scrollTop > n) {
                    t.scrollTop = n;
                    e.stopPropagation();
                    e.preventDefault();
                }
            }
        }), [ H ]);
        var F = (0, r.useCallback)((function(e) {
            W(e);
            O(c, e);
        }), [ c ]);
        var U = (0, r.useCallback)((function() {
            L(document.body.style.overflowX);
            R(document.body.style.overflowY);
            document.body.style.overflowX = "hidden";
            document.body.style.overflowY = "hidden";
        }), []);
        var $ = (0, r.useCallback)((function() {
            document.body.style.overflowX = A;
            document.body.style.overflowY = N;
        }), [ A, N ]);
        var z = d.animated[b];
        
        return n().createElement(n().Fragment, null,  n().createElement(z, P({}, m()(T, v()(x)), {
            ref: F,
            onMouseEnter: p === "window" ? U : undefined,
            onMouseLeave: p === "window" ? $ : undefined,
            onScroll: s,
            scrollTop: S,
            scrollLeft: f
        }),  n().createElement(h.ScrollContainerProvider, {
            value: H
        }, t)), p === true && H &&  n().createElement(g(), {
            target: H,
            eventType: "wheel",
            listener: D,
            options: I
        }));
    }
    M.propTypes = x;
    /* harmony default export */    const N = M;
    // CONCATENATED MODULE: ./src/Scroll/ScrollStyles.ts
    var R = f()(N).withConfig({
        displayName: "ScrollStyles__StyledInner",
        componentId: "sc-1xspttf-0"
    })([ "", ";overflow:auto;overflow-y:auto;overflow-x:hidden;" ], s.mixins.reset("block"));
    // CONCATENATED MODULE: ./src/Scroll/Scroll.tsx
    function X() {
        return X = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, X.apply(null, arguments);
    }
    function Y(e, t) {
        return F(e) || D(e, t) || W(e, t) || H();
    }
    function H() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function W(e, t) {
        if (e) {
            if ("string" == typeof e) return _(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? _(e, t) : void 0;
        }
    }
    function _(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function D(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, o, l, a, i = [], u = !0, c = !1;
            try {
                if (l = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (n = l.call(r)).done) && (i.push(n.value), i.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (a = r["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return i;
        }
    }
    function F(e) {
        if (Array.isArray(e)) return e;
    }
    function U(e, t) {
        if (null == e) return {};
        var r, n, o = $(e, t);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (n = 0; n < l.length; n++) {
                r = l[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function $(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var z = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        left: i().number,
        onScroll: i().func,
        onScrollComplete: i().func,
        stopScrollPropagation: i().oneOf([ true, false, "window" ]),
        tagName: i().string,
        top: i().number
    };
    function B(e) {
        var t = e.children, o = e.elementRef, a = e.left, i = e.onScroll, c = e.onScrollComplete, f = e.stopScrollPropagation, s = e.tagName, d = s === void 0 ? "div" : s, p = e.top, v = U(e, [ "children", "elementRef", "left", "onScroll", "onScrollComplete", "stopScrollPropagation", "tagName", "top" ]);
        // @docs-props-type ScrollPropsBase
                var y = (0, r.useState)(0), m = Y(y, 2), b = m[0], g = m[1];
        var h = (0, r.useState)(0), S = Y(h, 2), w = S[0], O = S[1];
        var P = (0, r.useCallback)((function() {
            c === null || c === void 0 ? void 0 : c();
        }), [ c ]);
        var j = (0, r.useCallback)((function(e) {
            g(e.currentTarget.scrollLeft);
            O(e.currentTarget.scrollTop);
            i === null || i === void 0 ? void 0 : i(e);
        }), [ i ]);
        var T = {};
        var C = (0, u.useAnimation)({
            config: {
                precision: 10
            },
            from: {
                left: 0,
                top: 0
            },
            to: T,
            onRest: P
        });
        // When isFinite, the prop is defined and therefore animating.
                if (l()(a)) {
            T.left = a;
        } else {
            T.left = b;
        }
        if (l()(p)) {
            T.top = p;
        } else {
            T.top = w;
        }
        
        return n().createElement(R, X({
            "data-test": "scroll",
            key: "inner",
            top: l()(p) ? C.top : undefined,
            left: l()(a) ? C.left : undefined,
            elementRef: o,
            onScroll: j,
            stopScrollPropagation: f,
            tagName: d
        }, v), t);
    }
    B.propTypes = z;
    /* harmony default export */    const G = B;
    // CONCATENATED MODULE: ./src/Scroll/index.ts
    module.exports = t;
    /******/})();
