// @splunk/react-ui/Search.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Divider: () => /* reexport */ p.Divider,
        Heading: () => /* reexport */ p.Heading,
        Option: () => /* reexport */ x,
        default: () => /* reexport */ ie
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/extend"
    const o = require("lodash/extend");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/memoize"
    const i = require("lodash/memoize");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/omit"
    const u = require("lodash/omit");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/pick"
    const s = require("lodash/pick");
    var d =  e.n(s);
    // CONCATENATED MODULE: external "prop-types"
    const f = require("prop-types");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const p = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const b = require("@splunk/react-ui/Popover");
    var m =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const y = require("@splunk/react-ui/ResultsMenu");
    var g =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const h = require("@splunk/react-ui/Text");
    var O =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const S = require("@splunk/react-ui/usePrevious");
    var w =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const C = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const k = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/keyboard"
    const j = require("@splunk/ui-utils/keyboard");
    // CONCATENATED MODULE: external "@splunk/ui-utils/scroll"
    const P = require("@splunk/ui-utils/scroll");
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function R(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Search/Option.tsx
    function I() {
        return I = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, I.apply(null, arguments);
    }
    function M(e, n) {
        if (null == e) return {};
        var r, t, o = A(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function A(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var q = {
        /** @private */
        active: v().bool,
        description: v().string,
        descriptionPosition: v().oneOf([ "right", "bottom" ]),
        disabled: v().bool,
        elementRef: v().oneOfType([ v().func, v().object ]),
        endAdornment: v().node,
        label: v().string,
        matchRanges: v().arrayOf(v().shape({
            start: v().number.isRequired,
            end: v().number.isRequired
        })),
        onClick: v().func,
        openInNewContext: v().bool,
        startAdornment: v().node,
        to: v().string,
        truncate: v().bool,
        value: v().string.isRequired
    };
    /**
 * An option within a `Search`.
 */    function D(e) {
        var n = e.active, o = e.description, a = e.descriptionPosition, i = a === void 0 ? "bottom" : a, l = e.disabled, u = e.elementRef, c = e.endAdornment, s = e.label, d = e.matchRanges, f = e.onClick, v = e.openInNewContext, b = e.startAdornment, m = e.to, y = e.truncate, g = e.value, h = M(e, [ "active", "description", "descriptionPosition", "disabled", "elementRef", "endAdornment", "label", "matchRanges", "onClick", "openInNewContext", "startAdornment", "to", "truncate", "value" ]);
        // @docs-props-type OptionPropsBase
                var O = (0, r.useCallback)((function(e) {
            if (!l) {
                f === null || f === void 0 ? void 0 : f(e, {
                    to: m,
                    value: g
                });
            }
        }), [ l, f, m, g ]);
        var S = (0, r.useCallback)((function(e) {
            R(u, e);
        }), [ u ]);
        var w = s === undefined ? g : s;
        
        return t().createElement(p.Item, I({
            active: n,
            "data-test": "search-option",
            "data-test-value": g,
            description: o,
            descriptionPosition: i,
            disabled: l ? "disabled" : undefined,
            elementRef: S,
            endAdornment: c,
            matchRanges: d,
            onClick: O,
            openInNewContext: v,
            startAdornment: b,
            to: m,
            truncate: y
        }, h, {
            role: "option"
        }), w);
    }
    D.propTypes = q;
    D.as = "Item";
    /* harmony default export */    const x = D;
    // CONCATENATED MODULE: external "styled-components"
    const E = require("styled-components");
    var B =  e.n(E);
    // CONCATENATED MODULE: external "@splunk/react-icons/Magnifier"
    const T = require("@splunk/react-icons/Magnifier");
    var H =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/themes"
    const L = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Search/SearchStyles.ts
    var _ = B()(H()).withConfig({
        displayName: "SearchStyles__StyledSearchIcon",
        componentId: "sc-1bohk7c-0"
    })([ "color:", ";pointer-events:none;", "" ], L.variables.contentColorMuted, (function(e) {
        var n = e.$disabled;
        return n && (0, E.css)([ "color:", ";" ], L.variables.contentColorDisabled);
    }));
    // CONCATENATED MODULE: ./src/Search/Search.tsx
    function N(e) {
        "@babel/helpers - typeof";
        return N = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, N(e);
    }
    function V() {
        return V = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, V.apply(null, arguments);
    }
    function K(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function F(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? K(Object(r), !0).forEach((function(n) {
                W(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : K(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function W(e, n, r) {
        return (n = U(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function U(e) {
        var n = $(e, "string");
        return "symbol" == N(n) ? n : n + "";
    }
    function $(e, n) {
        if ("object" != N(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != N(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function z(e, n) {
        return Q(e) || J(e, n) || Y(e, n) || X();
    }
    function X() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Y(e, n) {
        if (e) {
            if ("string" == typeof e) return G(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? G(e, n) : void 0;
        }
    }
    function G(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function J(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, o, a, i, l = [], u = !0, c = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = a.call(r)).done) && (l.push(t.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function Q(e) {
        if (Array.isArray(e)) return e;
    }
    function Z(e, n) {
        if (null == e) return {};
        var r, t, o = ee(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function ee(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */    var ne = {
        animateLoading: v().bool,
        append: v().bool,
        children: v().node,
        defaultPlacement: v().oneOf([ "above", "below", "vertical" ]),
        defaultValue: v().string,
        describedBy: v().string,
        disabled: v().bool,
        elementRef: v().oneOfType([ v().func, v().object ]),
        error: v().bool,
        footerMessage: v().node,
        inline: v().bool,
        inputRef: v().oneOfType([ v().func, v().object ]),
        isLoadingOptions: v().bool,
        labelledBy: v().string,
        loadingMessage: v().node,
        menuStyle: v().object,
        name: v().string,
        noOptionsMessage: v().node,
        onBlur: v().func,
        onChange: v().func,
        onClose: v().func,
        onFocus: v().func,
        onKeyDown: v().func,
        onOpen: v().func,
        onScroll: v().func,
        onSelect: v().func,
        placeholder: v().string,
        prepend: v().bool,
        value: v().string
    };
    function re(e, n) {
        var r = n.clientX, t = n.clientY;
        var o = e.getBoundingClientRect(), a = o.top, i = o.left, l = o.bottom, u = o.right;
        return r > i && r < u && t > a && t < l;
    }
    var te = l()((function(e) {
        var n = e.anchorWidth, r = e.maxHeight, t = e.menuStyle;
        return a()({
            maxHeight: r,
            overflow: "auto",
            width: Math.max(n !== null && n !== void 0 ? n : 0, 200)
        }, t);
    }));
    var oe = (0, C._)("Search");
    function ae(e) {
        var n = e.animateLoading, o = n === void 0 ? false : n, a = e.append, i = e.children, l = e.defaultPlacement, u = e.defaultValue, s = e.describedBy, f = e.disabled, v = f === void 0 ? false : f, p = e.elementRef, b = e.error, y = b === void 0 ? false : b, h = e.footerMessage, S = e.inline, I = S === void 0 ? false : S, M = e.inputRef, A = e.isLoadingOptions, q = A === void 0 ? false : A, D = e.labelledBy, x = e.loadingMessage, E = e.menuStyle, B = E === void 0 ? {} : E, T = e.name, H = e.noOptionsMessage, L = e.onChange, N = e.onClose, K = e.onFocus, W = e.onKeyDown, U = e.onOpen, $ = e.placeholder, X = $ === void 0 ? (0, 
        C._)("Search...") : $, Y = e.prepend, G = e.value, J = Z(e, [ "animateLoading", "append", "children", "defaultPlacement", "defaultValue", "describedBy", "disabled", "elementRef", "error", "footerMessage", "inline", "inputRef", "isLoadingOptions", "labelledBy", "loadingMessage", "menuStyle", "name", "noOptionsMessage", "onChange", "onClose", "onFocus", "onKeyDown", "onOpen", "placeholder", "prepend", "value" ]);
        // @docs-props-type SearchPropsBase
                var Q = !!i || q;
        // state
                var ee = (0, r.useState)(0), ne = z(ee, 2), ae = ne[0], ie = ne[1];
        var le = (0, r.useState)(false), ue = z(le, 2), ce = ue[0], se = ue[1];
        var de = (0, r.useState)(undefined), fe = z(de, 2), ve = fe[0], pe = fe[1];
        var be = (0, r.useState)(u || ""), me = z(be, 2), ye = me[0], ge = me[1];
        var he = (0, r.useState)((function() {
            return {
                popoverId: (0, k.createDOMID)("popover"),
                menuId: (0, k.createDOMID)("menu"),
                activeItemId: (0, k.createDOMID)("active-item")
            };
        })), Oe = z(he, 1), Se = Oe[0], we = Se.popoverId, Ce = Se.menuId, ke = Se.activeItemId;
 // previous state
                var je = w()(ae);
        // refs
                var Pe = (0, r.useRef)(null);
        var Re = (0, r.useRef)(null);
        var Ie = (0, r.useRef)(G !== undefined && G !== null);
        var Me = (0, r.useRef)([]);
        var Ae = (0, r.useRef)(je);
        var qe = (0, r.useRef)(0);
        var De = (0, r.useRef)(undefined);
        var xe = function e(n) {
            R(Pe, n);
            R(p, n);
        };
        var Ee = function e(n) {
            R(Re, n);
            R(M, n);
        };
        var Be = function e() {
            return Ie.current ? G : ye;
        };
        var Te = function e() {
            var n = Be();
            var t = r.Children.toArray(i).filter(r.isValidElement);
            var o = !ve && t.find((function(e) {
                return e.props.label !== undefined && e.props.value === n;
            }));
            if (o) {
                return o === null || o === void 0 ? void 0 : o.props.label;
            }
            return n;
        };
        var He = function e() {
            se(true);
            ie(-1);
            U === null || U === void 0 ? void 0 : U();
        };
        var Le = function e() {
            se(false);
            Ae.current = null;
            N === null || N === void 0 ? void 0 : N();
        };
        var _e = function e() {
            var n;
            (n = Re.current) === null || n === void 0 ? void 0 : n.focus();
        };
        var Ne = function e(n, r) {
            He();
            K === null || K === void 0 ? void 0 : K(n, r);
        };
        var Ve = function e(n, r) {
            var t = r.value;
            if (!Ie.current) {
                ge(t);
            }
            ie(-1);
            pe(t);
            L === null || L === void 0 ? void 0 : L(n, {
                value: t,
                name: T
            });
        };
        var Ke = function e(n, r) {
            Ve(n, F({}, r));
            _e();
            Le();
        };
        var Fe = function e(n) {
            var r = qe.current;
            var t = n.key;
            if (ce) {
                switch (t) {
                  case "Enter":
                    {
                        var o;
                        n.preventDefault();
                        // triggering onClick so that links can be activated with the keyboard
                        // since the menu items themselves never receive focus
                        // this will implicitly call handleSelectOption()
                                                (o = document.getElementById(ke)) === null || o === void 0 ? void 0 : o.click();
                        break;
                    }

                  case "Tab":
                    Le();
                    break;

                  case "ArrowDown":
                    if (r <= 0) {
                        break;
                    }
                    ie((function(e) {
                        return Math.min(e + 1, r - 1);
                    }));
                    break;

                  case "ArrowUp":
                    if (r <= 0) {
                        break;
                    }
                    ie((function(e) {
                        return Math.max(e - 1, 0);
                    }));
                    break;

                  default:
 // do nothing
                                }
            } else if ((0, j.addsCharacter)(n.nativeEvent) !== false || 
            // Safari 9.0 returns undefined
            t === "Enter" || t === "Backspace" || t === "ArrowDown" || t === "ArrowUp") {
                He();
            }
            W === null || W === void 0 ? void 0 : W(n);
        };
        var We = function e() {
            if (!ce && !v) {
                He();
            }
        };
        var Ue = function e(n) {
            var r = n.event, t = n.reason;
            var o = t === "offScreen" || t === "escapeKey" || t === "clickAway" && Pe.current && !re(Pe.current, r);
            if (o) {
                Le();
            }
        };
        var $e = function e(n) {
            if (Ae.current !== ae) {
                if (n) {
                    (0, P.scrollIntoViewIfNeeded)(n);
                }
            }
        };
        var ze = function e(n) {
            var a = n.anchorWidth, l = n.maxHeight;
            var u = r.Children.toArray(i).filter(r.isValidElement);
            // Highlight Active
                        De.current = undefined;
            qe.current = 0;
            // Only update the menu options if they are valid and the dropdown is open.
            // This is necessary to handle the outro animation of the dropdown.
                        if (Q && ce) {
                Me.current = u.map((function(e) {
                    if (e.type.as !== "Item") {
                        // ignore non-Option items such as Headings and Dividers
                        return e;
                    }
                    var n = function n(r, t) {
                        var o, a;
                        var i = (o = (a = e.props).onClick) === null || o === void 0 ? void 0 : o.call(a, r, t);
                        if (i !== false) {
                            Ke(r, t);
                        }
                    };
                    if (e.props.disabled) {
                        return e;
                    }
                    var t = qe.current === ae;
                    qe.current += 1;
                    if (t) {
                        De.current = e.props.value;
                        
                        return (0, r.cloneElement)(e, {
                            elementRef: $e,
                            id: ke,
                            onClick: n,
                            active: true
                        });
                    }
                    
                    return (0, r.cloneElement)(e, {
                        onClick: n
                    });
                }));
            }
            var c = te({
                anchorWidth: a,
                maxHeight: l,
                menuStyle: B
            });
            
            return t().createElement(g(), V({
                style: c,
                focusMode: "never",
                isLoading: q
            }, d()(J, "onScroll"), {
                noOptionsMessage: H,
                footerMessage: h,
                animateLoading: o,
                loadingMessage: x,
                menuId: Ce
            }), Me.current);
        };
        var Xe = Be();
        var Ye = Te();
        var Ge = ce && !!Pe.current && Q;
        var Je = Q ? "combobox" : "searchbox";
        var Qe = {
            "aria-activedescendant": Ge && ae > -1 ? ke : undefined,
            "aria-expanded": Q ? ce && !!Pe.current : undefined,
            "aria-controls": Ge ? Ce : undefined
        };
        var Ze =  t().createElement(_, {
            $disabled: v,
            "aria-label": oe,
            "data-test": "search-icon"
        });
        
        return t().createElement(O(), V({
            "data-test": "search"
        }, c()(J, "onScroll"), {
            append: a,
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            canClear: true,
            "data-test-popover-id": we,
            "data-test-label": Ye,
            "data-test-value": Xe,
            "data-test-open": Ge,
            describedBy: s,
            disabled: v,
            elementRef: xe,
            error: y,
            inline: I,
            inputRef: Ee,
            labelledBy: D,
            name: T,
            onFocus: Ne,
            onClick: We,
            onChange: Ve,
            onKeyDown: Fe,
            placeholder: X,
            prepend: Y,
            spellCheck: false,
            startAdornment: Ze,
            value: Ye,
            role: Je
        }, Qe),  t().createElement(m(), {
            anchor: function() {
                return Pe.current;
            }(),
            autoCloseWhenOffScreen: true,
            canCoverAnchor: false,
            defaultPlacement: l,
            id: we,
            onRequestClose: Ue,
            open: Ge,
            repositionMode: "flip"
        }, ze));
    }
    ae.propTypes = ne;
    ae.Option = x;
    ae.Divider = p.Divider;
    ae.Heading = p.Heading;
    /* harmony default export */    const ie = ae;
    // CONCATENATED MODULE: ./src/Search/index.ts
    module.exports = n;
    /******/})();
