// @splunk/react-ui/SelectBase.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var n = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, n) => {
            /******/ for (var t in n) {
                /******/ if (e.o(n, t) && !e.o(r, t)) {
                    /******/ Object.defineProperty(r, t, {
                        enumerable: true,
                        get: n[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Controls: () => /* reexport */ dr,
        Divider: () => /* reexport */ w.Divider,
        Heading: () => /* reexport */ w.Heading,
        Option: () => /* reexport */ J,
        SelectAllOption: () => /* reexport */ oe,
        default: () => /* reexport */ gr,
        isOption: () => /* reexport */ lr
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var t =  e.n(n);
    // CONCATENATED MODULE: external "lodash/castArray"
    const l = require("lodash/castArray");
    var a =  e.n(l);
    // CONCATENATED MODULE: external "lodash/has"
    const i = require("lodash/has");
    var o =  e.n(i);
    // CONCATENATED MODULE: external "lodash/memoize"
    const u = require("lodash/memoize");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/pick"
    const s = require("lodash/pick");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "prop-types"
    const d = require("prop-types");
    var p =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-icons/CaretSmallDown"
    const v = require("@splunk/react-icons/CaretSmallDown");
    var b =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-icons/Magnifier"
    const m = require("@splunk/react-icons/Magnifier");
    var y =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const g = require("@splunk/react-ui/Dropdown");
    var h =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Link"
    const S = require("@splunk/react-ui/Link");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const w = require("@splunk/react-ui/Menu");
    // CONCATENATED MODULE: external "@splunk/react-ui/ResultsMenu"
    const C = require("@splunk/react-ui/ResultsMenu");
    var k =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const x = require("@splunk/react-ui/ScreenReaderContent");
    var j =  e.n(x);
    // CONCATENATED MODULE: external "@splunk/react-ui/Text"
    const E = require("@splunk/react-ui/Text");
    var P =  e.n(E);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const A = require("@splunk/react-ui/useControlled");
    var L =  e.n(A);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const I = require("@splunk/react-ui/usePrevious");
    var D =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/ui-utils/filter"
    const R = require("@splunk/ui-utils/filter");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const M = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const T = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "@splunk/ui-utils/scroll"
    const q = require("@splunk/ui-utils/scroll");
    // CONCATENATED MODULE: ./src/utils/disabled.ts
    /**
 * Normalizes the disabled prop and returns both the boolean state and normalized string value.
 * Use this function to ensure consistent handling of disabled prop across components.
 *
 * @param disabled - The disabled prop value
 * @param defaultWhenBooleanTrue - Optional normalization override for disabled={true}
 * @returns Object with isDisabled boolean|undefined and disabledValue string
 *
 * TODO (SUI-8600): `disabled` prop should default to 'dimmed' for data entry components
 */
    function _(e) {
        var r = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "dimmed";
        if (e === "dimmed") {
            return {
                isDisabled: true,
                disabledValue: "dimmed"
            };
        }
        if (e === true) {
            return {
                isDisabled: true,
                disabledValue: r
            };
        }
        if (e === "disabled") {
            return {
                isDisabled: true,
                disabledValue: "disabled"
            };
        }
        if (e === false) {
            return {
                isDisabled: false,
                disabledValue: undefined
            };
        }
        return {
            isDisabled: undefined,
            disabledValue: undefined
        };
    }
    // CONCATENATED MODULE: ./src/SelectBase/OptionBase.tsx
    function B(e) {
        "@babel/helpers - typeof";
        return B = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, B(e);
    }
    function V() {
        return V = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, V.apply(null, arguments);
    }
    function N(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function F(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? N(Object(n), !0).forEach((function(r) {
                K(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : N(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function K(e, r, n) {
        return (r = H(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function H(e) {
        var r = z(e, "string");
        return "symbol" == B(r) ? r : r + "";
    }
    function z(e, r) {
        if ("object" != B(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != B(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function W(e, r) {
        if (null == e) return {};
        var n, t, l = $(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (l[n] = e[n]);
            }
        }
        return l;
    }
    function $(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    var X = {
        /** @private */
        active: p().bool,
        children: p().node,
        description: p().string,
        descriptionPosition: p().oneOf([ "right", "bottom" ]),
        disabled: p().oneOfType([ p().bool, p().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: p().oneOfType([ p().func, p().object ]),
        endAdornment: p().node,
        hidden: p().bool,
        icon: p().node,
        label: p().string.isRequired,
        /**
   * @private Passed down from <BaseSelect>
   */
        multiple: p().bool,
        matchRanges: p().arrayOf(p().shape({
            start: p().number.isRequired,
            end: p().number.isRequired
        })),
        /** @private */
        onClick: p().func,
        /** @private */
        role: p().oneOf([ "menuitemcheckbox", "option" ]),
        /** @private */
        selected: p().oneOfType([ p().bool, p().oneOf([ "some" ]) ]),
        truncate: p().bool,
        value: p().oneOfType([ p().string, p().number, p().bool ]).isRequired
    };
    /**
 * An option within a `Multiselect`.
 */    function U(e) {
        var r = e.children, l = e.descriptionPosition, a = l === void 0 ? "bottom" : l, i = e.disabled, o = e.elementRef, u = e.endAdornment, c = e.icon, s = e.label, f = e.multiple, d = e.onClick, p = e.role, v = p === void 0 ? "option" : p, b = e.value, m = W(e, [ "children", "descriptionPosition", "disabled", "elementRef", "endAdornment", "icon", "label", "multiple", "onClick", "role", "value" ]);
        // @docs-props-type OptionPropsBase
                var y = _(i, "disabled"), g = y.isDisabled, h = y.disabledValue;
        var S = (0, n.useCallback)((function(e) {
            if (!g) {
                d === null || d === void 0 ? void 0 : d(e, {
                    value: b
                });
            }
        }), [ g, d, b ]);
        var O = b.toString();
        var C = F({
            descriptionPosition: a,
            disabled: h,
            "data-test-disabled": h
        }, m);
        
        return t().createElement(w.Item, V({
            "data-test-value": b,
            "data-test": "option",
            elementRef: o,
            endAdornment: u
        }, C, {
            selectable: true,
            selectableAppearance: f ? "checkbox" : "checkmark",
            startAdornment: c,
            onClick: S,
            role: v,
            value: O
        }), r || s);
    }
    U.propTypes = X;
    U.type = w.Item;
 // For components to distinguish if their children are Options or Headings/Dividers
    /* harmony default export */    const J = U;
    // CONCATENATED MODULE: ./src/SelectBase/SelectAllOption.tsx
    function G(e) {
        "@babel/helpers - typeof";
        return G = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, G(e);
    }
    function Q() {
        return Q = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Q.apply(null, arguments);
    }
    function Y(e, r) {
        if (null == e) return {};
        var n, t, l = Z(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (l[n] = e[n]);
            }
        }
        return l;
    }
    function Z(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function ee(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function re(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? ee(Object(n), !0).forEach((function(r) {
                ne(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : ee(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function ne(e, r, n) {
        return (r = te(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function te(e) {
        var r = le(e, "string");
        return "symbol" == G(r) ? r : r + "";
    }
    function le(e, r) {
        if ("object" != G(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != G(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    var ae = re(re({}, J.propTypes), {}, {
        totalCount: p().number,
        // unlike OptionBase, there's a default value for this prop
        value: p().string
    });
    function ie(e) {
        var r = e.active, n = e.elementRef, l = e.id, a = e.label, i = e.onClick, o = e.selected, u = e.totalCount, c = e.value, s = c === void 0 ? "selectAll" : c, f = Y(e, [ "active", "elementRef", "id", "label", "onClick", "selected", "totalCount", "value" ]);
        // @docs-props-type SelectAllOptionPropsBase
        // When toggle is changed the total count is not displayed anymore so the count is now shown in the Select all menu item
                var d = u != null ? " (".concat(u === null || u === void 0 ? void 0 : u.toString(), ")") : "";
        
        return t().createElement(J, Q({
            active: r,
            elementRef: n,
            "aria-keyshortcuts": "Control+A",
            "data-test": "select-all",
            description: "".concat(d, " Ctrl-a"),
            descriptionPosition: "right",
            id: l,
            key: "selectAll",
            multiple: true,
            label: a,
            onClick: i,
            selected: o,
            role: "menuitemcheckbox",
            value: s
        }, f));
    }
    ie.propTypes = ae;
    /* harmony default export */    const oe = ie;
    // CONCATENATED MODULE: external "styled-components"
    const ue = require("styled-components");
    var ce =  e.n(ue);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const se = require("@splunk/react-ui/Button");
    var fe =  e.n(se);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const de = require("@splunk/react-ui/Divider");
    var pe =  e.n(de);
    // CONCATENATED MODULE: external "@splunk/themes"
    const ve = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SelectBase/SelectBaseStyles.ts
    var be = ce()(fe()).withConfig({
        displayName: "SelectBaseStyles__StyledButton",
        componentId: "sc-1lmonqb-0"
    })([ "&[data-inline]{width:", ";}", "" ], (function(e) {
        var r = e.$multiple;
        return r ? "400px" : "auto";
    }), (function(e) {
        var r = e.$multiple;
        return !r && "flex-grow: 0;";
    }));
    var me = ce().span.withConfig({
        displayName: "SelectBaseStyles__StyledLinkIcon",
        componentId: "sc-1lmonqb-1"
    })([ "padding-right:", ";" ], ve.variables.spacingXSmall);
    var ye = ce().span.withConfig({
        displayName: "SelectBaseStyles__StyledLinkCaret",
        componentId: "sc-1lmonqb-2"
    })([ "padding-left:", ";" ], ve.variables.spacingXSmall);
    var ge = ce().div.withConfig({
        displayName: "SelectBaseStyles__StyledFilter",
        componentId: "sc-1lmonqb-3"
    })([ "padding:", " ", " ", ";min-width:160px;" ], ve.variables.spacingLarge, ve.variables.spacingLarge, ve.variables.spacingSmall);
    var he = ce().span.withConfig({
        displayName: "SelectBaseStyles__StyledCount",
        componentId: "sc-1lmonqb-4"
    })([ "padding-right:", ";" ], ve.variables.spacingXSmall);
    var Se = ce()(O()).withConfig({
        displayName: "SelectBaseStyles__StyledControlsLink",
        componentId: "sc-1lmonqb-5"
    })([ "", ";" ], (function(e) {
        var r = e.$disabled;
        return r && (0, ue.css)([ "color:", ";" ], ve.variables.contentColorDisabled);
    }));
    var Oe = ce().div.withConfig({
        displayName: "SelectBaseStyles__StyledToggleAllControls",
        componentId: "sc-1lmonqb-6"
    })([ "", ";gap:", ";padding:", " ", " ", ";" ], ve.mixins.reset("flex"), ve.variables.spacingMedium, ve.variables.spacingXSmall, ve.variables.spacingLarge, ve.variables.spacingSmall);
    var we = ce()(pe()).withConfig({
        displayName: "SelectBaseStyles__StyledControlsDivider",
        componentId: "sc-1lmonqb-7"
    })([ "border-color:", ";" ], ve.variables.borderColor);
    // CONCATENATED MODULE: ./src/utils/fuzzyMatch.ts
    // A utility for matching keyboard characters to list values
    var Ce = function e(r, n) {
        return r ? r.label.charAt(n).toLowerCase() : "";
    };
    var ke = function e(r, n) {
        if (!r.length) {
            return r;
        }
        var t = null;
        var l = false;
        var a = r.filter((function(e) {
            var r = Ce(e, n.index);
            if (r === n.value) {
                l = true;
                return true;
            }
            // If we haven't found a match yet, keep track of the next closest match.
            // Secondary matching looks for the closest character of a higher value, and failing that, closest of a lower value.
                        if (!l) {
                var a = Ce(t, n.index);
                if (!a) {
                    t = e;
                } else if (r > n.value) {
                    if (a < n.value) {
                        t = e;
                    } else if (a > r) {
                        t = e;
                    }
                } else if (r > a) {
                    t = e;
                }
            }
            return false;
        }));
        return a.length === 0 && t ? [ t ] : a;
    };
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var xe = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function je() {
        var e = typeof document !== "undefined" ? document : xe;
        return e;
    }
    var Ee = je();
    /* harmony default export */    const Pe = /* unused pure expression or super */ null && Ee;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var Ae = {
        document: xe,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(r) {
            if (typeof setTimeout === "undefined") {
                r();
                return null;
            }
            return setTimeout(r, 0);
        },
        cancelAnimationFrame: function e(r) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(r);
        }
    };
    function Le() {
        var e = typeof window !== "undefined" ? window : Ae;
        return e;
    }
    var Ie = Le();
    /* harmony default export */    const De = /* unused pure expression or super */ null && Ie;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function Re(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/SelectBase/SelectBase.tsx
    function Me(e) {
        return _e(e) || qe(e) || Fe(e) || Te();
    }
    function Te() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function qe(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function _e(e) {
        if (Array.isArray(e)) return Ke(e);
    }
    function Be() {
        return Be = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var n = arguments[r];
                for (var t in n) {
                    ({}).hasOwnProperty.call(n, t) && (e[t] = n[t]);
                }
            }
            return e;
        }, Be.apply(null, arguments);
    }
    function Ve(e, r) {
        return ze(e) || He(e, r) || Fe(e, r) || Ne();
    }
    function Ne() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Fe(e, r) {
        if (e) {
            if ("string" == typeof e) return Ke(e, r);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? Ke(e, r) : void 0;
        }
    }
    function Ke(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var n = 0, t = Array(r); n < r; n++) {
            t[n] = e[n];
        }
        return t;
    }
    function He(e, r) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var t, l, a, i, o = [], u = !0, c = !1;
            try {
                if (a = (n = n.call(e)).next, 0 === r) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (t = a.call(n)).done) && (o.push(t.value), o.length !== r); u = !0) {
                }
            } catch (e) {
                c = !0, l = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (i = n["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw l;
                }
            }
            return o;
        }
    }
    function ze(e) {
        if (Array.isArray(e)) return e;
    }
    function We(e, r) {
        if (null == e) return {};
        var n, t, l = $e(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                n = a[t], -1 === r.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (l[n] = e[n]);
            }
        }
        return l;
    }
    function $e(e, r) {
        if (null == e) return {};
        var n = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== r.indexOf(t)) continue;
                n[t] = e[t];
            }
        }
        return n;
    }
    function Xe(e) {
        "@babel/helpers - typeof";
        return Xe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, Xe(e);
    }
    function Ue(e, r) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            r && (t = t.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), n.push.apply(n, t);
        }
        return n;
    }
    function Je(e) {
        for (var r = 1; r < arguments.length; r++) {
            var n = null != arguments[r] ? arguments[r] : {};
            r % 2 ? Ue(Object(n), !0).forEach((function(r) {
                Ge(e, r, n[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : Ue(Object(n)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(n, r));
            }));
        }
        return e;
    }
    function Ge(e, r, n) {
        return (r = Qe(r)) in e ? Object.defineProperty(e, r, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = n, e;
    }
    function Qe(e) {
        var r = Ye(e, "string");
        return "symbol" == Xe(r) ? r : r + "";
    }
    function Ye(e, r) {
        if ("object" != Xe(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var t = n.call(e, r || "default");
            if ("object" != Xe(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    var Ze = function e(r) {
        var n = _(r, "disabled"), t = n.isDisabled;
        return !!t;
    };
    var er = {
        allowKeyMatching: p().bool,
        allowNewValues: p().bool,
        animateLoading: p().bool,
        appearance: p().oneOf([ "default", "link", "subtle" ]),
        append: p().bool,
        children: p().node,
        defaultPlacement: p().oneOf([ "above", "below", "vertical" ]),
        defaultValues: p().array,
        describedBy: p().string,
        disabled: p().oneOfType([ p().bool, p().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: p().oneOfType([ p().func, p().object ]),
        error: p().bool,
        filter: p().oneOf([ false, true, "controlled" ]),
        footerMessage: p().node,
        inline: p().bool,
        inputId: p().string,
        inputRef: p().oneOfType([ p().func, p().object ]),
        isLoadingOptions: p().bool,
        labelledBy: p().string,
        labelText: p().string,
        loadingMessage: p().node,
        /** @private. */
        maxLabelItems: p().number,
        menuStyle: p().object,
        multiple: p().bool,
        name: p().string,
        noOptionsMessage: p().node,
        onChange: p().func,
        onClick: p().func,
        onClose: p().func,
        onFilterChange: p().func,
        onOpen: p().func,
        onScroll: p().func,
        onScrollBottom: p().func,
        /** @private. */
        required: p().bool,
        placeholder: p().string,
        prefixLabel: p().string,
        prepend: p().bool,
        repositionMode: p().oneOf([ "none", "flip" ]),
        selectAllAppearance: p().oneOf([ "buttongroup", "checkbox", "none" ]),
        showSelectedValuesFirst: p().oneOf([ "nextOpen", "immediately", "never" ]),
        suffixLabel: p().string,
        tabConfirmsNewValue: p().bool,
        toggle: p().node,
        toggleContent: p().oneOf([ "optionChildren", "optionLabel" ]),
        values: p().array,
        /** @private. */
        virtualization: p().number
    };
    var rr = c()((function(e) {
        return [ e ];
    }));
    // preserve separate widths for single vs. multi mode
        var nr = c()((function(e) {
        var r = e.anchorWidth, n = e.isMultiple, t = e.maxHeight, l = e.menuStyle;
        return n ? Je({
            width: Math.max(r !== null && r !== void 0 ? r : 0, 200),
            maxHeight: t
        }, l) : Je({
            minWidth: r !== null && r !== void 0 ? r : undefined,
            maxWidth: Math.max(r !== null && r !== void 0 ? r : 0, 300),
            maxHeight: t
        }, l);
    }));
    var tr = [ "clickAway", "escapeKey", "offScreen", "tabKey", "toggleClick" ];
    function lr(e) {
        return e && o()(e.props, "value");
    }
    function ar(e, r) {
        return "".concat(Xe(e), "-").concat(e, "-").concat(r);
    }
    var ir = [ "append", "error", "prepend" ];
    var or = (0, M._)("No matches");
    var ur = (0, M._)("Select...");
    var cr = 30;
    var sr = {};
    var fr =  t().createElement(y(), null);
    var dr = function e(r) {
        var n = r.activeItemId, l = r.filterA11yId, a = r.filterKeyword, i = r.hasChildren, o = r.inputId, u = r.inputRef, c = r.menuListboxId, s = r.multiple, f = r.onClearAll, d = r.onSelectAll, p = r.onTextBlur, v = r.onTextChange, b = r.onTextFocus, m = r.onTextKeyDown, y = r.optionSelection, g = r.placement, h = r.selectAllAppearance, S = r.textHasFocus;
        var O = (0, M._)("Select all options".concat(y.current === "all" ? " disabled" : ""));
        var w = (0, M._)("Clear all options".concat(y.current === "none" ? " disabled" : ""));
        // only visually disable these as opposed to fully disabling to avoid losing keyboard focus - see SUI-2731
                var C =  t().createElement(Oe, {
            key: "selectAll"
        },  t().createElement(Se, {
            disabled: y.current === "all",
            appearance: "standalone",
            "aria-label": O,
            onClick: d,
            "data-test": "select-all",
            tag: "button"
        }, a ? (0, M._)("Select all Matches") : (0, M._)("Select all")),  t().createElement(Se, {
            disabled: y.current === "none",
            appearance: "standalone",
            "aria-label": w,
            onClick: f,
            "data-test": "clear-all",
            tag: "button"
        }, a ? (0, M._)("Clear all Matches") : (0, M._)("Clear all")));
        
        return t().createElement("div", {
            key: "controls"
        }, l &&  t().createElement(j(), {
            id: l
        }, (0, M._)("Type to filter")), g === "above" &&  t().createElement(we, null),  t().createElement(ge, {
            key: "filter",
            "data-test": "filter"
        },  t().createElement(P(), {
            value: a,
            autoCapitalize: "off",
            autoComplete: "off",
            autoCorrect: "off",
            spellCheck: false,
            onChange: v,
            onKeyDown: m,
            onFocus: b,
            onBlur: p,
            placeholder: (0, M._)("Filter"),
            role: "combobox",
            "aria-expanded": "true",
            "aria-controls": c,
            "aria-owns": S && i ? n : undefined,
            "aria-label": (0, M._)("Filter"),
            "aria-autocomplete": "list",
            "aria-activedescendant": S && i ? n : undefined,
            inputRef: u,
            inputId: o,
            canClear: true,
            startAdornment: fr
        })), s && i && h === "buttongroup" && C, g === "below" &&  t().createElement(we, null));
    };
    var pr = function e(r) {
        var n = r.prefixLabel, t = r.label, l = r.suffixLabel;
        var i = t;
        if (n) {
            i = [ "".concat(n, ": ") ].concat(i);
        }
        if (l) {
            i = a()(i).concat(" ".concat(l));
        }
        return i;
    };
    var vr = [];
    var br =  t().forwardRef((function(e, r) {
        var l = e.appearance, a = l === void 0 ? "default" : l, i = e.append, o = e.children, u = e.currentValues, c = u === void 0 ? vr : u, s = e.describedBy, d = e.disabled, p = e.elementRef, v = e.error, m = e.inline, y = e.labelText, g = e.labelledBy, h = e.multiple, S = e.onClick, w = e.placeholder, C = e.prefixLabel, k = e.prepend, x = e.required, j = e.suffixLabel, E = e.toggle, P = e.toggleContent, A = e.maxLabelItems, L = A === void 0 ? cr : A, I = We(e, [ "appearance", "append", "children", "currentValues", "describedBy", "disabled", "elementRef", "error", "inline", "labelText", "labelledBy", "multiple", "onClick", "placeholder", "prefixLabel", "prepend", "required", "suffixLabel", "toggle", "toggleContent", "maxLabelItems" ]);
        var D = (0, n.useMemo)((function() {
            var e = n.Children.toArray(o);
            var r = new Map;
            for (var t = 0; t < e.length; t += 1) {
                var l = e[t];
                if (lr(l)) {
                    r.set(l.props.value, l);
                }
            }
            return r;
        }), [ o ]);
        var R = (0, n.useMemo)((function() {
            var e = {
                icon: null,
                label: [],
                ariaLabel: []
            };
            var r = h && c.length > L ? Math.max(1, L) : c.length;
            for (var n = 0; n < r; n += 1) {
                var t = c[n];
                var l = D.get(t);
                if (l) {
                    var a = l.props, i = a.children, o = a.icon, u = a.label;
                    var s = P !== "optionLabel" && i ? i : u;
                    e.label.push(s);
                    e.ariaLabel.push(u);
                    // if not in multiple mode, add the icon
                                        if (!h && c.length === 1) {
                        e.icon = o;
                    }
                } else if (h) {
                    // only add values that don't match an option in "multiple" mode to preserve old behaviour
                    e.label.push(t);
                    e.ariaLabel.push(t.toString());
                }
                if (n < r - 1) {
                    e.label.push((0, M._)(", "));
                    e.ariaLabel.push((0, M._)(", "));
                }
            }
            if (h && c.length > r && e.label.length) {
                // add visual ellipsis to ensure truncation is shown
                e.label.push((0, M._)("..."));
            }
            // only apply prefix / suffix if the label is not empty
                        if (e.label.length > 0) {
                // If there's more than one item selected, read out the selected total
                // rather than reading out each selected item
                e.ariaLabel = pr({
                    prefixLabel: C,
                    label: e.label.length > 1 ? [ "".concat(c.length, " items selected") ] : e.ariaLabel,
                    suffixLabel: j
                });
                e.label = pr({
                    prefixLabel: C,
                    label: e.label,
                    suffixLabel: j
                });
            }
            // single <Select> behaviour is to show the placeholder if all parts of the label
            // are empty strings so we replicate this behaviour here
                        if (!e.label.length || !h && e.label.every((function(e) {
                return e === "";
            }))) {
                e.label = rr(w);
                e.ariaLabel = e.label;
            }
            return e;
        }), [ c, L, h, D, w, C, j, P ]), T = R.label, q = R.ariaLabel, B = R.icon;
        var V = "aria-labelledby" in I;
        var N = _(d, "disabled"), F = N.disabledValue;
        var K = Je({
            "aria-describedby": s,
            "aria-disabled": d === "dimmed" || undefined,
            "aria-label": g || V ? undefined : "".concat(y ? "".concat(y, ", ") : "").concat(q.join("")),
            // aria-labelledby takes precedence over aria-label, so existence of both is redundant
            "aria-labelledby": g,
            "aria-required": x,
            "data-select-appearance": a,
            append: i,
            prepend: k,
            onClick: S,
            role: "combobox",
            disabled: a === "link" ? d === true || d === "disabled" : F,
            elementRef: p,
            error: v,
            ref: r
        }, I);
        if (h) {
            K["data-test-values"] = JSON.stringify(c);
        } else {
            var H = Ve(c, 1), z = H[0];
            K["data-test-value"] = z;
        }
        K["data-test-disabled"] = F;
        if (E) {
            
            return (0, n.cloneElement)(E, K);
        }
        if (a === "link") {
            
            return t().createElement(O(), Be({
                tag: "button",
                appearance: "standalone"
            }, K, {
                "data-select-appearance": "link"
            }), !!B &&  t().createElement(me, null, B), T || w,  t().createElement(ye, null,  t().createElement(b(), null)));
        }
        // Using Button's secondary appearance as Select's default appearance.
                var W = a === "default" ? "secondary" : a;
        
        return t().createElement(be, Be({}, K, {
            $multiple: h,
            appearance: W,
            label: T,
            error: v,
            icon: B,
            inline: m,
            isMenu: true,
            onClick: S
        }, f()(I, ir)), !!c.length && h &&  t().createElement(he, {
            "data-role": "count"
        }, "(", c.length, ")"));
    }));
    function mr(e) {
        var r = e.currentValues, n = e.multiple;
        if (n || !(r === null || r === void 0 ? void 0 : r.length)) {
            return r !== null && r !== void 0 ? r : [];
        }
        return [ r[0] ];
    }
    function yr(e) {
        var r = e.allowKeyMatching, l = r === void 0 ? true : r, a = e.animateLoading, i = e.appearance, o = i === void 0 ? "default" : i, u = e.append, c = e.allowNewValues, s = e.children, f = e.defaultPlacement, d = f === void 0 ? "vertical" : f, p = e.defaultValues, v = e.describedBy, b = e.disabled, m = e.elementRef, y = e.error, g = e.filter, S = e.footerMessage, O = e.inline, x = e.inputId, j = e.inputRef, E = e.isLoadingOptions, P = e.labelledBy, A = e.labelText, I = e.loadingMessage, B = e.menuStyle, V = B === void 0 ? sr : B, N = e.multiple, F = e.name, K = e.noOptionsMessage, H = K === void 0 ? or : K, z = e.onChange, W = e.onScroll, $ = e.onScrollBottom, X = e.onFilterChange, U = e.onClick, G = e.onClose, Q = e.onOpen, Y = e.required, Z = e.placeholder, ee = Z === void 0 ? ur : Z, re = e.prefixLabel, ne = e.prepend, te = e.repositionMode, le = te === void 0 ? "flip" : te, ae = e.selectAllAppearance, ie = ae === void 0 ? "buttongroup" : ae, ue = e.showSelectedValuesFirst, ce = e.suffixLabel, se = e.tabConfirmsNewValue, fe = e.values, de = e.virtualization, pe = e.toggle, ve = e.toggleContent, be = ve === void 0 ? "optionChildren" : ve, me = e.maxLabelItems, ye = me === void 0 ? cr : me, ge = We(e, [ "allowKeyMatching", "animateLoading", "appearance", "append", "allowNewValues", "children", "defaultPlacement", "defaultValues", "describedBy", "disabled", "elementRef", "error", "filter", "footerMessage", "inline", "inputId", "inputRef", "isLoadingOptions", "labelledBy", "labelText", "loadingMessage", "menuStyle", "multiple", "name", "noOptionsMessage", "onChange", "onScroll", "onScrollBottom", "onFilterChange", "onClick", "onClose", "onOpen", "required", "placeholder", "prefixLabel", "prepend", "repositionMode", "selectAllAppearance", "showSelectedValuesFirst", "suffixLabel", "tabConfirmsNewValue", "values", "virtualization", "toggle", "toggleContent", "maxLabelItems" ]);
        // @docs-props-type SelectBasePropsBase
                var he = L()({
            componentName: "SelectBase",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultValues",
            valuePropName: "values"
        });
        // states
                var Se = (0, n.useState)((function() {
            return {
                activeItemId: (0, T.createDOMID)("active-item"),
                menuListboxId: (0, T.createDOMID)("menu-listbox")
            };
        })), Oe = Ve(Se, 1), we = Oe[0], Ce = we.activeItemId, xe = we.menuListboxId;
        var je = (0, n.useState)(0), Ee = Ve(je, 2), Pe = Ee[0], Ae = Ee[1];
        var Ie = (0, n.useState)(""), De = Ve(Ie, 2), Te = De[0], qe = De[1];
        var _e = (0, n.useState)(false), Ne = Ve(_e, 2), Fe = Ne[0], Ke = Ne[1];
        var He = (0, n.useState)(false), ze = Ve(He, 2), $e = ze[0], Xe = ze[1];
        var Ue = (0, n.useState)([]), Je = Ve(Ue, 2), Ge = Je[0], Qe = Je[1];
        var Ye = (0, n.useState)(p || []), er = Ve(Ye, 2), rr = er[0], ir = er[1];
 // previous state
                var fr = D()(Pe);
        // refs
                var pr = (0, n.useRef)(null);
        var vr = (0, n.useRef)(null);
        var yr = (0, n.useRef)(null);
        var gr = (0, n.useRef)([]);
        var hr = (0, n.useRef)([]);
        var Sr = (0, n.useRef)(fr);
        var Or = (0, n.useRef)();
        var wr = (0, n.useRef)({});
        var Cr = (0, n.useRef)(0);
        var kr = (0, n.useRef)([]);
        var xr = (0, n.useRef)(null);
        var jr = (0, n.useRef)();
        var Er = (0, n.useRef)(0);
        var Pr = (0, n.useRef)("none");
        var Ar = (0, n.useState)(g ? (0, T.createDOMID)("filter") : undefined), Lr = Ve(Ar, 1), Ir = Lr[0];
        var Dr = _(b, "disabled"), Rr = Dr.isDisabled;
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ o, u, y, ne ]);
        var Mr = (0, n.useMemo)((function() {
            var e = t().Children.toArray(s);
            var r = new Map;
            for (var n = 0; n < e.length; n += 1) {
                var l = e[n];
                if (lr(l)) {
                    r.set(l.props.value, l);
                }
            }
            return [ e, r ];
        }), [ s ]), Tr = Ve(Mr, 2), qr = Tr[0], _r = Tr[1];
        var Br = mr({
            currentValues: he ? fe : rr,
            multiple: N
        });
        // mutable object reference to currentValues to permit idempotent callbacks
                var Vr = (0, n.useRef)(null);
        Vr.current = Br;
        var Nr = (0, n.useCallback)((function() {
            // in non-multiple mode, don't move values to the top of the list
            return N && ue !== "never" ? Vr.current : [];
        }), [ N, ue ]);
        var Fr = (0, n.useCallback)((function(e) {
            if (Rr) {
                return;
            }
            var r = Te;
            Ke(true);
            Qe(Nr());
            // SUI-3234: preserve existing behavior difference between <Select> and <Multiselect.Compact>
                        if (N) {
                qe("");
            }
            if (r !== Te) {
                X === null || X === void 0 ? void 0 : X(e, {
                    keyword: Te
                });
            }
        }), [ Rr, Te, Nr, N, X ]);
        (0, n.useEffect)((function() {
            if (Fe) {
                Q === null || Q === void 0 ? void 0 : Q();
                if (xr.current && !g) {
                    xr.current.focus();
                } else {
                    var e;
                    Ae((e = jr.current) !== null && e !== void 0 ? e : 0);
                }
            }
        }), [ Q, Fe, Ge, Te, g ]);
        var Kr = (0, n.useCallback)((function() {
            Ke(false);
            Ae(0);
            Sr.current = null;
            G === null || G === void 0 ? void 0 : G();
        }), [ G ]);
        var Hr = (0, n.useCallback)((function() {
            Kr();
        }), [ Kr ]);
        var zr = (0, n.useCallback)((function(e, r) {
            if (Rr) {
                return;
            }
            var n = new Set(Vr.current);
            var t;
            if (N) {
                if (n.has(r)) {
                    n["delete"](r);
                } else {
                    n.add(r);
                }
                t = Array.from(n);
            } else {
                // non-multiple mode must always have a value
                t = [ r ];
            }
            var l = !he;
            if (l) {
                ir(t);
            }
            if (N) {
                // in uncontrolled multiple mode, keep the menu open
                if (l) {
                    Ke(true);
                }
            } else {
                var a;
                // non-multiple mode only supports a single selection
                // so close the menu once a selection is made
                                Hr({
                    reason: "contentClick"
                });
                (a = pr.current) === null || a === void 0 ? void 0 : a.focus();
            }
            z === null || z === void 0 ? void 0 : z(e, {
                values: t,
                name: F,
                reason: "valueToggle"
            });
        }), [ Hr, he, Rr, N, F, z ]);
        var Wr = function e() {
            vr.current = null;
            gr.current = [];
            if (yr.current) {
                clearTimeout(yr.current);
            }
        };
        var $r = (0, n.useCallback)((function(e) {
            if (Rr) {
                return;
            }
            // this doesn't make sense if we can't select multiple values
                        if (!N) {
                return;
            }
            var r = Vr.current;
            var n = new Set(r);
            var t = new Set([].concat(Me(r), Me(kr.current)));
            // to keep track of custom values we create a copy of selected values and remove
            // items as they are accounted for in children, leaving only custom values
                        var l = new Set(t);
            var a = new Set;
            qr.forEach((function(e) {
                if (lr(e)) {
                    // this value is being represented in children so it is not custom
                    l["delete"](e.props.value);
                    var r = Ze(e.props.disabled);
                    if (t.has(e.props.value) && (!r || n.has(e.props.value))) {
                        a.add(e.props.value);
                    }
                }
            }));
            var i = [].concat(Me(Array.from(l)), Me(Array.from(a)));
            if (!he) {
                ir(i);
            }
            z === null || z === void 0 ? void 0 : z(e, {
                values: i,
                name: F,
                reason: "selectAll"
            });
        }), [ qr, he, Rr, N, F, z ]);
        var Xr = (0, n.useCallback)((function(e) {
            if (Rr) {
                return;
            }
            // this doesn't make sense if we can't select multiple values
                        if (!N) {
                return;
            }
            var r = new Set(Vr.current);
            // Preserve hidden values only when "Clear matches" semantics apply
                        var n = Te !== "";
            var t = new Set;
            if (n) {
                t = new Set(Vr.current);
                kr.current.forEach((function(e) {
                    t["delete"](e);
                }));
            }
            var l = new Set;
            for (var a = 0; a < qr.length; a += 1) {
                var i = qr[a];
                // this will unselect all selected values unless those values are disabled or hidden by the filter
                                if (lr(i)) {
                    var o = Ze(i.props.disabled);
                    if (r.has(i.props.value) && o || t.has(i.props.value)) {
                        l.add(i.props.value);
                    }
                }
            }
            var u = Array.from(l);
            if (!he) {
                ir(u);
            }
            z === null || z === void 0 ? void 0 : z(e, {
                values: u,
                name: F,
                reason: "clearAll"
            });
        }), [ qr, Te, he, Rr, N, F, z ]);
        var Ur = (0, n.useCallback)((function(e) {
            if (Fe && !E) {
                $ === null || $ === void 0 ? void 0 : $(e);
            }
        }), [ Fe, E, $ ]);
        var Jr = (0, n.useCallback)((function(e) {
            if (Rr) {
                return;
            }
            var r = Cr.current;
            var t = e.key;
            var l = Or.current !== null && Or.current !== undefined ? _r.get(Or.current) : undefined;
            var a = Ze(l === null || l === void 0 ? void 0 : l.props.disabled);
            if (t === "Tab") {
                if (se && Or.current && Cr.current <= 1) {
                    if (!a) {
                        e.preventDefault();
                        zr(e, Or.current);
                    }
                }
            }
            if (e.shiftKey || e.metaKey || e.ctrlKey) {
                if (t === "a" && (e.ctrlKey || e.metaKey)) {
                    // handle control + A
                    if (Pr.current === "all") {
                        Xr(e);
                    } else {
                        $r(e);
                    }
                }
                return;
            }
            if (t === "ArrowDown") {
                e.preventDefault();
                var i = ie === "checkbox" ? r : r - 1;
                if (i >= 0) {
                    Ae(Math.min(Pe + 1, i));
                }
                if (s && $) {
                    var o = n.Children.toArray(s).length - (2 + Vr.current.length);
                    if (Pe === o) {
                        Ur(e);
                    }
                }
            }
            if (t === "ArrowUp") {
                e.preventDefault();
                Ae(Math.max(Pe - 1, 0));
            }
            if (t === "Enter" && Or.current && Fe) {
                if (Or.current === "selectAll") {
                    if (Pr.current === "all") {
                        Xr(e);
                    } else {
                        $r(e);
                    }
                } else if (!a) {
                    e.preventDefault();
                    zr(e, Or.current);
                }
            }
        }), [ Pe, s, Xr, Ur, $r, Rr, _r, $, Fe, ie, se, zr ]);
        var Gr = (0, n.useCallback)((function(e, r) {
            var n = r.value;
            if (Rr) {
                return;
            }
            e.preventDefault();
            if (!Fe) {
                return;
            }
            zr(e, n);
        }), [ Rr, Fe, zr ]);
        var Qr = (0, n.useCallback)((function(e, r) {
            var n = e.nativeEvent.key;
            // Checking for a single character to avoid complications from double-byte languages and emojis.
                        if (n.length === 1) {
                var t = [];
                var l = {
                    index: 0,
                    value: n
                };
                if (!vr.current) {
                    if (n === " ") {
                        Wr();
                        return;
                    }
                    t = ke(hr.current, l);
                } else if (gr.current.length > 1) {
                    l.index = vr.current.index + 1;
                    t = ke(gr.current, l);
                }
                if (t.length) {
                    var a;
                    var i = 0;
                    // If the active option is a first character match, cycle to the next matching option.
                                        if (l.index === 0 && t.length > 1) {
                        var o = t.indexOf(hr.current[r]);
                        if (o >= 0) {
                            i = o === t.length - 1 ? 0 : o + 1;
                        }
                    }
                    var u = t[i];
                    var c = u.value, s = u.label;
                    var f = wr.current[ar(c, s)];
                    f === null || f === void 0 ? void 0 : (a = f.focus) === null || a === void 0 ? void 0 : a.call(f);
                }
                gr.current = t;
                vr.current = l;
                if (yr.current) {
                    clearTimeout(yr.current);
                }
                yr.current = setTimeout(Wr, 500);
                e.preventDefault();
                e.stopPropagation();
            }
        }), []);
        var Yr = (0, n.useCallback)((function(e, r) {
            var n = r.value;
            if (Rr) {
                return;
            }
            qe(n);
            Ke(true);
            Ae(0);
            X === null || X === void 0 ? void 0 : X(e, {
                keyword: n
            });
        }), [ Rr, X ]);
        var Zr = (0, n.useCallback)((function() {
            Xe(true);
        }), []);
        var en = (0, n.useCallback)((function() {
            Xe(false);
        }), []);
        // Close when component becomes disabled/dimmed during an expanded state to prevent further selection
                (0, n.useEffect)((function() {
            if (Rr && Fe) {
                Kr();
            }
        }), [ Rr, Fe, Kr ]);
        var rn = (0, n.useCallback)((function(e) {
            if (Sr.current !== Pe) {
                (0, q.scrollIntoViewIfNeeded)(e);
            }
        }), [ Pe ]);
        var nn = (0, n.useCallback)((function(e, r, n) {
            if (n) {
                xr.current = e;
            }
            if (e == null) {
                delete wr.current[r];
            } else {
                wr.current[r] = e;
            }
        }), [ wr ]);
        var tn = (0, n.useCallback)((function(e) {
            pr.current = e;
            Re(m, e);
        }), [ m, pr ]);
        var ln = Br.some((function(e) {
            var r = _r.get(e);
            var n = Ze(r === null || r === void 0 ? void 0 : r.props.disabled);
            return r && !n;
        }));
        var an = ue === "immediately" ? Nr() : Ge;
        var on = (0, n.useMemo)((function() {
            Cr.current = 0;
            jr.current = undefined;
            Er.current = 0;
            Or.current = undefined;
            hr.current = [];
            var e = function e(r, n) {
                return function(e) {
                    return nn(e, r, n);
                };
            };
            var r;
            var a = 0;
            var i = false;
            // used to avoid overwriting the selected item ref in multiple mode
                        var o;
            var u = new Set(Br);
            var s = new Set(an !== null && an !== void 0 ? an : []);
            var f = qr.reduce((function(c, f, d) {
                // ignore Headings and Dividers
                if (!lr(f)) {
                    c.push(f);
                    return c;
                }
                var p = f.props, v = p.disabled, b = p.hidden, m = p.label, y = p.value;
 // Find out if the search string exactly matches a value
                                if (y === Te) {
                    r = true;
                }
                var h = u.has(y);
                var S = !!h && v !== true && v !== "disabled" && !o;
                var O = l && !N && !g && !E && !$;
                var C = ar(y, m);
                var k = -1;
                if (O && v !== true && v !== "disabled" && !b) {
                    hr.current.push({
                        label: m,
                        value: y
                    });
                    k = hr.current.length - 1;
                }
                // Format the Menu.Item
                                var x =  (0, n.cloneElement)(f, {
                    elementRef: e(C, S),
                    key: f.key || d,
                    onClick: Gr,
                    onKeyDown: O ? function(e) {
                        return Qr(e, k);
                    } : undefined,
                    selected: h,
                    multiple: N,
                    role: "option"
                });
                if (S) {
                    o = true;
                }
                if (s.has(y)) {
                    if (a === 0) {
                        c.splice(a, 0,  t().createElement(w.Divider, {
                            key: "topDivider"
                        }));
                        i = true;
                    }
                    c.splice(a, 0, x);
                    a += 1;
                } else {
                    c.push(x);
                }
                return c;
            }), []);
            // In multiple mode, add missing items
                        if (N) {
                for (var d = Br.length - 1; d >= 0; d -= 1) {
                    var p = Br[d];
                    var v = _r.get(p);
                    if (!v) {
                        if (p === Te) {
                            r = true;
                        }
                        var b = s.has(p);
                        var m = an.length;
                        if (a === 0) {
                            f.splice(0, 0,  t().createElement(w.Divider, {
                                key: "topDivider"
                            }));
                            a += 1;
                            i = true;
                        }
                        var y = String(p);
                        var h = ar(p, y);
                        f.splice(b ? 0 : m + 1, 0,  t().createElement(J, {
                            elementRef: e(h),
                            label: y,
                            value: p,
                            key: "missing-value-".concat(p),
                            onClick: Gr,
                            multiple: N,
                            selected: true
                        }));
                        if (b) {
                            a += 1;
                        }
                    }
                }
            }
            var S = g === "controlled";
            // Filter the items
                        var O = (0, R.stringToKeywords)(Te);
            f = S ? f : f.filter((function(e) {
                if (lr(e)) {
                    return (0, R.testPhrase)(e.props.label, O);
                }
                return true;
 // Keep all headers and non-interactive options
                        })).map((function(e) {
                if (!lr(e)) {
                    return e;
                }
                // highlight matched text
                                var r = O && (0, R.keywordLocations)(e.props.label, O);
                
                return (0, n.cloneElement)(e, {
                    matchRanges: r || undefined
                });
            }));
            // Add the option to add the new value
                        if (c && !r && Te) {
                var C = i ? a + 1 : a;
                var k = "".concat(Te, " (new value)");
                var x = ar(Te, k);
                f.splice(C, 0,  t().createElement(J, {
                    elementRef: e(x),
                    label: k,
                    value: Te,
                    key: "newValue",
                    multiple: N,
                    onClick: Gr
                }));
            }
            // When selectAll is a menu item, add an offset so that activeItem accounts for the extra selectAll option.
                        var j = ie === "checkbox" && N && f.length > 1 ? 1 : 0;
            var P = true;
            // Highlight the selected Items and remove hidden
                        f = f.reduce((function(e, r) {
                // ignore Dividers & Headings
                if (!lr(r)) {
                    e.push(r);
                    return e;
                }
                // Ignore any hidden items
                                if (r.props && r.props.hidden) {
                    return e;
                }
                if (r.props.selected && r.props.disabled !== true && r.props.disabled !== "disabled" && jr.current == null) {
                    jr.current = Cr.current;
                }
                var t = r.props.disabled;
                if (t === true || t === "disabled") {
                    e.push(r);
                    return e;
                }
                var l = j === Pe;
                j += 1;
                Cr.current += 1;
                // Dimmed Options can receive focus but are not selectable
                                Er.current += r.props.selected && t !== "dimmed" ? 1 : 0;
                if (r.key !== "newValue" && t !== "dimmed" && !r.props.selected) {
                    P = false;
                }
                if (!l || !$e) {
                    e.push(r);
                    return e;
                }
                Or.current = r.props.value;
                var a =  (0, n.cloneElement)(r, {
                    active: l,
                    elementRef: rn,
                    id: Ce
                });
                e.push(a);
                return e;
            }), []);
            Pr.current = Er.current === 0 && "none" || P && "all" || "some";
            // add select all option
                        if (ie === "checkbox" && N && Cr.current > 0) {
                var A = Pe === 0 && !!g;
                var L;
                if (Pr.current === "all") {
                    if (Te === "") {
                        L = (0, M._)("Clear all");
                    } else {
                        L = (0, M._)("Clear all matches");
                    }
                } else if (Te === "") {
                    L = (0, M._)("Select all");
                } else {
                    L = (0, M._)("Select all matches");
                }
                if (A) {
                    Or.current = "selectAll";
                }
                f.unshift( t().createElement(oe, {
                    key: "select-all-option",
                    active: A,
                    elementRef: rn,
                    id: A ? Ce : undefined,
                    onClick: Pr.current === "all" ? Xr : $r,
                    label: L,
                    selected: Pr.current === "all" || Pr.current !== "none" && "some",
                    totalCount: !!pe && Br.length || undefined,
                    tabIndex: g ? -1 : undefined
                }),  t().createElement(w.Divider, {
                    key: "selectAllDivider"
                }));
            }
            kr.current = f.reduce((function(e, r) {
                if (lr(r)) {
                    e.push(r.props.value);
                }
                return e;
            }), []);
            return f;
        }), [ Pe, Ce, l, c, qr, _r, Br, g, Te, rn, Xr, Gr, Qr, nn, $r, E, N, an, $, ie, $e, pe ]);
        var un = function e(r) {
            var n = r.anchorWidth, l = r.maxHeight, i = r.placement, o = r.toggleId;
            var u = nr({
                anchorWidth: n,
                isMultiple: N,
                maxHeight: l,
                menuStyle: V
            });
            var c = Br.length > 0 && !ln && !g ? 0 : undefined;
            var s = {
                "aria-multiselectable": N || undefined,
                childrenStart: !!g &&  t().createElement(dr, {
                    activeItemId: Ce,
                    filterA11yId: Ir,
                    filterKeyword: Te,
                    hasChildren: on.some((function(e) {
                        return lr(e);
                    })),
                    inputId: x,
                    inputRef: j,
                    menuListboxId: xe,
                    multiple: N,
                    onClearAll: Xr,
                    onSelectAll: $r,
                    onTextBlur: en,
                    onTextChange: Yr,
                    onTextFocus: Zr,
                    onTextKeyDown: Jr,
                    optionSelection: Pr,
                    placement: i,
                    selectAllAppearance: ie,
                    textHasFocus: $e
                }),
                focusMode: g ? "never" : undefined,
                isLoading: E,
                labelledBy: "".concat(Ir !== null && Ir !== void 0 ? Ir : "", " ").concat(o !== null && o !== void 0 ? o : "").trim(),
                // NVDA ignores aria-labelledby attribute on popover, but reads it on menu
                menuId: xe,
                onScrollBottom: $ ? Ur : undefined,
                placement: i !== null && i !== void 0 ? i : undefined,
                noOptionsMessage: H,
                footerMessage: S,
                animateLoading: a,
                loadingMessage: I,
                onScroll: W,
                style: u,
                tabIndex: c
            };
            if (de) {
                
                return t().createElement(C.VirtualizedResultsMenu, Be({
                    virtualization: de
                }, s), on);
            }
            
            return t().createElement(k(), s, on);
        };
        var cn = (0, n.useMemo)((function() {
            
            return t().createElement(br, Be({
                appearance: o,
                append: u,
                currentValues: Br,
                "data-test": N ? "multiselect" : "select",
                describedBy: v,
                disabled: b,
                elementRef: tn,
                error: y,
                inline: O,
                labelText: A,
                labelledBy: P,
                multiple: N,
                onClick: Rr ? undefined : U,
                placeholder: ee,
                prefixLabel: re,
                prepend: ne,
                required: Y,
                suffixLabel: ce,
                toggle: pe,
                toggleContent: be,
                maxLabelItems: ye
            }, ge), s);
        }), [ o, u, Br, N, v, b, tn, y, O, A, P, Rr, U, ee, re, ne, Y, ce, pe, be, ye, ge, s ]);
        
        return t().createElement(h(), {
            closeReasons: tr,
            inputId: x,
            "aria-labelledby": Ir,
            toggle: cn,
            onRequestOpen: Fr,
            onRequestClose: Hr,
            open: Fe,
            openWithArrowKeys: !Rr,
            repositionMode: le,
            defaultPlacement: d,
            canCoverAnchor: Le().innerHeight < 500,
            retainFocus: false,
            takeFocus: Br.length === 0 || Br.length > 0 && !ln || !!g
        }, un);
    }
    yr.propTypes = er;
    yr.componentType = "SelectBase";
    yr.Option = J;
    yr.Divider = w.Divider;
    yr.Heading = w.Heading;
    /* harmony default export */    const gr = yr;
    // CONCATENATED MODULE: ./src/SelectBase/index.ts
    module.exports = r;
    /******/})();
