// @splunk/react-ui/Select.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var o = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(o, {
                a: o
            });
            /******/            return o;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, o) => {
            /******/ for (var t in o) {
                /******/ if (e.o(o, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: o[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Divider: () => /* reexport */ i.Divider,
        Heading: () => /* reexport */ i.Heading,
        Option: () => /* reexport */ m,
        default: () => /* reexport */ x
    });
    // CONCATENATED MODULE: external "react"
    const o = require("react");
    var t =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const r = require("prop-types");
    var a =  e.n(r);
    // CONCATENATED MODULE: external "@splunk/react-ui/SelectBase"
    const i = require("@splunk/react-ui/SelectBase");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const u = require("@splunk/react-ui/useControlled");
    var d =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const s = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: ./src/Select/Option.tsx
    function c() {
        return c = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var o = arguments[n];
                for (var t in o) {
                    ({}).hasOwnProperty.call(o, t) && (e[t] = o[t]);
                }
            }
            return e;
        }, c.apply(null, arguments);
    }
    function f(e, n) {
        if (null == e) return {};
        var o, t, r = p(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                o = a[t], -1 === n.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (r[o] = e[o]);
            }
        }
        return r;
    }
    function p(e, n) {
        if (null == e) return {};
        var o = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                o[t] = e[t];
            }
        }
        return o;
    }
    var b = {
        /** @private */
        active: a().bool,
        children: a().node,
        description: a().string,
        descriptionPosition: a().oneOf([ "right", "bottom" ]),
        disabled: a().oneOfType([ a().bool, a().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        /** @private */
        endAdornment: a().node,
        hidden: a().bool,
        icon: a().node,
        label: a().string.isRequired,
        matchRanges: a().arrayOf(a().shape({
            start: a().number.isRequired,
            end: a().number.isRequired
        })),
        /** @private */
        onClick: a().func,
        /** @private */
        role: a().oneOf([ "menuitemcheckbox", "option" ]),
        /** @private */
        selected: a().bool,
        truncate: a().bool,
        value: a().oneOfType([ a().string, a().number, a().bool ]).isRequired
    };
    /**
 * An option within a `Select`. Any elements passed to it should be memoized.
 */    function v(e) {
        var n = e.children, o = e.descriptionPosition, r = o === void 0 ? "bottom" : o, a = e.disabled, l = e.elementRef, u = e.label, d = f(e, [ "children", "descriptionPosition", "disabled", "elementRef", "label" ]);
        // @docs-props-type OptionPropsBase
                
        return t().createElement(i.Option, c({
            descriptionPosition: r,
            disabled: a,
            label: u
        }, d, {
            elementRef: l,
            multiple: false
        }), n || u);
    }
    var g =  (0, o.memo)(v);
    // @ts-expect-error Need to set propTypes on OptionMemoized so that javascript propType validation will work
        g.propTypes = b;
    /* harmony default export */    const m = g;
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var O = "is deprecated and will be removed in the next major version.";
    var y = function e(n) {
        var o = n.additionalMessage, t = o === void 0 ? "" : o, r = n.componentName;
        useEffect((function() {
            if (false) {}
        }), [ t, r ]);
    };
    var h = function e(n) {
        var o = n.additionalMessage, t = o === void 0 ? "" : o, r = n.componentName, a = n.propName, i = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ t, r, a, i ]);
    };
    var P = function e(n) {
        var t = n.additionalMessage, r = t === void 0 ? "" : t, a = n.componentName, i = n.deprecatedPropValue, l = n.propName, u = n.propValue;
        (0, o.useEffect)((function() {
            if (false) {}
        }), [ r, a, l, u, i ]);
    };
    // CONCATENATED MODULE: ./src/Select/Select.tsx
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var o = arguments[n];
                for (var t in o) {
                    ({}).hasOwnProperty.call(o, t) && (e[t] = o[t]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function j(e, n) {
        if (null == e) return {};
        var o, t, r = S(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                o = a[t], -1 === n.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (r[o] = e[o]);
            }
        }
        return r;
    }
    function S(e, n) {
        if (null == e) return {};
        var o = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                o[t] = e[t];
            }
        }
        return o;
    }
    /** @public */
    /** @public */
    /** @public */    var w = {
        allowKeyMatching: a().bool,
        animateLoading: a().bool,
        appearance: a().oneOf([ "default", "link", "subtle" ]),
        append: a().bool,
        children: a().node,
        defaultPlacement: a().oneOf([ "above", "below", "vertical" ]),
        defaultValue: a().oneOfType([ a().string, a().number, a().bool ]),
        describedBy: a().string,
        disabled: a().oneOfType([ a().bool, a().oneOf([ "dimmed", "disabled" ]) ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        error: a().bool,
        filter: a().oneOf([ false, true, "controlled" ]),
        footerMessage: a().node,
        inline: a().bool,
        inputId: a().string,
        inputRef: a().oneOfType([ a().func, a().object ]),
        isLoadingOptions: a().bool,
        labelledBy: a().string,
        labelText: a().string,
        loadingMessage: a().node,
        menuStyle: a().object,
        name: a().string,
        noOptionsMessage: a().node,
        onChange: a().func,
        onClose: a().func,
        onFilterChange: a().func,
        onOpen: a().func,
        onScroll: a().func,
        onScrollBottom: a().func,
        placeholder: a().string,
        prefixLabel: a().string,
        prepend: a().bool,
        repositionMode: a().oneOf([ "none", "flip" ]),
        suffixLabel: a().string,
        toggleContent: a().oneOf([ "optionChildren", "optionLabel" ]),
        /** @private. */
        toggle: a().node,
        value: a().oneOfType([ a().string, a().number, a().bool ]),
        /** @private. */
        virtualization: a().number
    };
    var C = (0, s._)("No matches");
    var T = (0, s._)("Select...");
    function R(e) {
        var n = e.allowKeyMatching, r = n === void 0 ? true : n, a = e.appearance, i = a === void 0 ? "default" : a, u = e.defaultPlacement, s = u === void 0 ? "vertical" : u, c = e.defaultValue, f = e.disabled, p = e.inline, b = p === void 0 ? true : p, v = e.noOptionsMessage, g = v === void 0 ? C : v, m = e.onChange, O = e.placeholder, y = O === void 0 ? T : O, h = e.toggleContent, S = h === void 0 ? "optionChildren" : h, w = e.value, R = j(e, [ "allowKeyMatching", "appearance", "defaultPlacement", "defaultValue", "disabled", "inline", "noOptionsMessage", "onChange", "placeholder", "toggleContent", "value" ]);
        // @docs-props-type SelectPropsBase
                P({
            componentName: "Select",
            deprecatedPropValue: "link",
            propName: "appearance",
            propValue: i
        });
        var x = d()({
            componentName: "Select",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var q = (0, o.useMemo)((function() {
            // wrap defaultValue in an array once to avoid failing <SelectBase>'s defaultValues comparison check
            if (c != null) {
                return [ c ];
            }
            return [];
        }), [ c ]);
        // Converts the provided props into a structure suitable for SelectBase.
                var N = (0, o.useMemo)((function() {
            var e = x ? undefined : q;
            var n = function e(n, o) {
                var t = o.name, r = o.values;
                m === null || m === void 0 ? void 0 : m(n, {
                    name: t,
                    value: r[0]
                });
            };
            var o = {
                onChange: n
            };
            if (x) {
                o.values = w != null ? [ w ] : [];
            } else if (e != null) {
                o.defaultValues = e;
            }
            return o;
        }), [ q, m, x, w ]);
        
        return t().createElement(l(), M({
            allowKeyMatching: r,
            appearance: i,
            defaultPlacement: s,
            disabled: f,
            inline: b,
            noOptionsMessage: g,
            placeholder: y,
            toggleContent: S
        }, R, N, {
            multiple: false
        }));
    }
    R.propTypes = w;
    R.Option = m;
    R.Heading = i.Heading;
    R.Divider = i.Divider;
    /* harmony default export */    const x = R;
    // CONCATENATED MODULE: ./src/Select/index.ts
    module.exports = n;
    /******/})();
