// @splunk/react-ui/SidePanel.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ Q
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "lodash/defer"
    const o = require("lodash/defer");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const u = require("prop-types");
    var a =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/ModalLayer"
    const l = require("@splunk/react-ui/ModalLayer");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const f = require("@splunk/react-ui/usePrevious");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "styled-components"
    const y = require("styled-components");
    var d =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/TransitionOpen"
    const v = require("@splunk/react-ui/TransitionOpen");
    var p =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SidePanel/SidePanelStyles.ts
    var m = {
        inline: "auto",
        container: "100%",
        page: "100vw"
    };
    var g = {
        inline: "auto",
        container: "100%",
        page: "100vh"
    };
    var S = {
        inline: "static",
        container: "absolute",
        page: "fixed"
    };
    var h = {
        inline: "auto",
        container: b.variables.zindexLayer,
        page: b.variables.zindexModal
    };
    var O = (0, y.css)([ "", " position:", ";flex-direction:column;", " background-color:", ";z-index:", ";" ], b.mixins.reset("flex"), (function(e) {
        var t = e.$dockLayout;
        return S[t];
    }), (function(e) {
        var t = e.$dockLayout;
        return t !== "inline" && (0, y.css)([ "box-shadow:", ";" ], b.variables.overlayShadow);
    }), b.variables.backgroundColorSidebar, (function(e) {
        var t = e.$dockLayout;
        return h[t];
    }));
    var P = d()(p()).withConfig({
        displayName: "SidePanelStyles__StyledLeftPanelTransitionOpen",
        componentId: "mtbcuz-0"
    })([ "", ";top:0;left:0;height:", ";" ], O, (function(e) {
        var t = e.$dockLayout;
        return g[t];
    }));
    var j = d()(p()).withConfig({
        displayName: "SidePanelStyles__StyledRightPanelTransitionOpen",
        componentId: "mtbcuz-1"
    })([ "", ";right:0;top:0;height:", ";" ], O, (function(e) {
        var t = e.$dockLayout;
        return g[t];
    }));
    var k = d()(p()).withConfig({
        displayName: "SidePanelStyles__StyledTopPanelTransitionOpen",
        componentId: "mtbcuz-2"
    })([ "", ";top:0;left:0;width:", ";" ], O, (function(e) {
        var t = e.$dockLayout;
        return m[t];
    }));
    var w = d()(p()).withConfig({
        displayName: "SidePanelStyles__StyledBottomPanelTransitionOpen",
        componentId: "mtbcuz-3"
    })([ "", ";left:0;bottom:0;width:", ";" ], O, (function(e) {
        var t = e.$dockLayout;
        return m[t];
    }));
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const C = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: ./src/utils/resolveInitialFocusTarget.ts
    function T(e) {
        "@babel/helpers - typeof";
        return T = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, T(e)
        /**
 * Resolves the initial focus target based on the initialFocus prop value.
 *
 * @param initialFocus - The initialFocus prop value ('first', 'container', HTMLElement, or React.Component with focus method)
 * @param containerEl - The container element to use for 'first' and 'container' options
 * @returns The element or component to focus, or null if none found
 */;
    }
    function L(e, t) {
        if (e === "first" && t) {
            var n;
            return (n = (0, C.getSortedTabbableElements)(t)[0]) !== null && n !== void 0 ? n : null;
        }
        if (e === "container") {
            return t;
        }
        if (T(e) === "object") {
            return e;
        }
        return null;
    }
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function E(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/SidePanel/SidePanel.tsx
    function A(e) {
        "@babel/helpers - typeof";
        return A = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, A(e);
    }
    function F() {
        return F = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, F.apply(null, arguments);
    }
    function _(e, t) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            t && (r = r.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), n.push.apply(n, r);
        }
        return n;
    }
    function M(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = null != arguments[t] ? arguments[t] : {};
            t % 2 ? _(Object(n), !0).forEach((function(t) {
                N(e, t, n[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : _(Object(n)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(n, t));
            }));
        }
        return e;
    }
    function N(e, t, n) {
        return (t = R(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function R(e) {
        var t = q(e, "string");
        return "symbol" == A(t) ? t : t + "";
    }
    function q(e, t) {
        if ("object" != A(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != A(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function x(e, t) {
        return B(e) || D(e, t) || $(e, t) || I();
    }
    function I() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function $(e, t) {
        if (e) {
            if ("string" == typeof e) return z(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? z(e, t) : void 0;
        }
    }
    function z(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function D(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, o, i, u, a = [], l = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    l = !1;
                } else for (;!(l = (r = i.call(n)).done) && (a.push(r.value), a.length !== t); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != n["return"] && (u = n["return"](), Object(u) !== u)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return a;
        }
    }
    function B(e) {
        if (Array.isArray(e)) return e;
    }
    function U(e, t) {
        if (null == e) return {};
        var n, r, o = Y(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function Y(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */    var G = {
        children: a().node,
        dockLayout: a().oneOf([ "page", "container", "inline" ]),
        dockPosition: a().oneOf([ "top", "bottom", "left", "right" ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        initialFocus: a().oneOfType([ a().object, a().oneOf([ "first", "container" ]) ]),
        innerClassName: a().string,
        innerStyle: a().object,
        onAnimationEnd: a().func,
        onRequestClose: a().func,
        open: a().bool,
        outerClassName: a().string,
        outerStyle: a().object,
        returnFocus: a().oneOfType([ a().func, a().object ]),
        useLayerForClickAway: a().bool
    };
    var H = {
        left: "slideFromLeft",
        right: "slideFromRight",
        top: "slideFromTop",
        bottom: "slideFromBottom"
    };
    var J = {
        left: P,
        right: j,
        top: k,
        bottom: w
    };
    function K(e) {
        var t = e.children, o = e.dockLayout, u = o === void 0 ? "page" : o, a = e.dockPosition, l = a === void 0 ? "right" : a, f = e.elementRef, y = e.initialFocus, d = e.innerClassName, v = e.innerStyle, p = e.onAnimationEnd, b = e.onRequestClose, m = e.open, g = e.outerClassName, S = e.outerStyle, h = e.returnFocus, O = e.useLayerForClickAway, P = O === void 0 ? true : O, j = U(e, [ "children", "dockLayout", "dockPosition", "elementRef", "initialFocus", "innerClassName", "innerStyle", "onAnimationEnd", "onRequestClose", "open", "outerClassName", "outerStyle", "returnFocus", "useLayerForClickAway" ]);
        // @docs-props-type SidePanelPropsBase
                var k = (0, n.useRef)(null);
        var w = (0, n.useRef)(undefined);
        // Use symbol to distinguish "never applied" from "applied with undefined/null"
                var C = (0, n.useRef)(Symbol("FOCUS_NOT_YET_APPLIED")).current;
        var T = (0, n.useRef)(C);
        var A = (0, n.useState)(false), _ = x(A, 2), N = _[0], R = _[1];
        var q = s()(m);
        (0, n.useEffect)((function() {
            // prevOpen starts undefined, and we don't want to make the comparison in that situation
            if (q != null && m !== q) {
                R(true);
            }
        }), [ m, q ]);
        // Focus handling strategy: consistent across all dockLayouts
        // - If initialFocus is passed, use it
        // - Else use 'first' (first focusable element)
        // - Else use container (fallback when no focusable elements)
                var I = (0, n.useCallback)((function() {
            // Clear pending focus if initialFocus target changed
            var e = y !== null && y !== void 0 ? y : null;
            if (T.current !== e) {
                clearTimeout(w.current);
            }
            // PANEL OPENING: Apply initial focus if needed
                        if (m) {
                w.current = i()((function() {
                    var e = y !== null && y !== void 0 ? y : "first";
                    // Only apply focus if the target changed from last time
                                        if (T.current !== (e !== null && e !== void 0 ? e : null)) {
                        var t, n;
                        var r = L(e, k.current);
                        // Fall back to container if target has no focus method
                                                if (!((t = r) === null || t === void 0 ? void 0 : t.focus)) {
                            r = k.current;
                        }
                        // TODO: bug fix https://splunk.atlassian.net/browse/SUI-8621
                                                (n = r) === null || n === void 0 ? void 0 : n.focus({
                            preventScroll: true
                        });
                        T.current = e !== null && e !== void 0 ? e : null;
                    }
                }));
            }
        }), [ m, y ]);
        var $ = (0, n.useCallback)((function() {
            // PANEL CLOSING: Reset state and handle return focus
            if (!m) {
                // Reset focus tracking state
                T.current = C;
                clearTimeout(w.current);
                // Return focus to invoking element if panel was previously open
                                if (q && h) {
                    if (typeof h === "function") {
                        h();
                    } else {
                        i()((function() {
                            var e, t;
                            (e = h.current) === null || e === void 0 ? void 0 : (t = e.focus) === null || t === void 0 ? void 0 : t.call(e);
                        }));
                    }
                }
            }
        }), [ m, q, h, C ]);
        // Handle return focus when panel closes, runs on change of handleReturnFocus deps
                (0, n.useEffect)((function() {
            $();
        }), [ $ ]);
        // Handle initial focus when panel opens - runs on change of handleInitialFocus deps
                (0, n.useEffect)((function() {
            I();
        }), [ I ]);
        var z = (0, n.useMemo)((function() {
            if (P) {
                return "interactive";
            }
            return "hidden";
        }), [ P ]);
        var D = (0, n.useMemo)((function() {
            if (l === "left" || l === "right") {
                return {
                    height: "100%"
                };
            }
            return {
                width: "100%"
            };
        }), [ l ]);
        var B = (0, n.useMemo)((function() {
            return M(M({}, D), v);
        }), [ D, v ]);
        var Y = (0, n.useCallback)((function() {
            R(false);
            p === null || p === void 0 ? void 0 : p();
        }), [ p ]);
        var G = (0, n.useCallback)((function(e) {
            E(k, e);
            E(f, e);
        }), [ k, f ]);
        var K = (0, n.useCallback)((function(e) {
            T.current = null;
            b === null || b === void 0 ? void 0 : b(e);
        }), [ b ]);
        var Q = (0, n.useMemo)((function() {
            return J[l];
        }), [ l ]);
        var V = (0, n.useMemo)((function() {
            return {
                retainFocus: true,
                role: "dialog",
                tabIndex: -1
            };
        }), []);
        var W = (0, n.useMemo)((function() {
            
            return r().createElement(Q, F({
                $dockLayout: u,
                animateOnMount: true,
                animation: H[l],
                "data-test": "side-panel",
                elementRef: G,
                innerClassName: d,
                innerStyle: B,
                onAnimationEnd: Y,
                open: m,
                outerClassName: g,
                outerStyle: S
            }, u === "page" && V, j), t);
        }), [ t, u, l, B, Y, G, d, m, g, S, V, Q, j ]);
        var X = (0, n.useCallback)((function() {
            return W;
        }), [ W ]);
        // If dockLayout is container or inline we want to return without the ModalLayer below.
                if (u !== "page") {
            return W;
        }
        
        return r().createElement(c(), {
            childrenInAnimation: N,
            onRequestClose: K,
            open: m,
            renderModal: X,
            scrim: z
        });
    }
    K.propTypes = G;
    /* harmony default export */    const Q = K;
    // CONCATENATED MODULE: ./src/SidePanel/index.ts
    module.exports = t;
    /******/})();
