// @splunk/react-ui/Slider.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ re
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "decimal.js-light"
    const a = require("decimal.js-light");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const u = require("@splunk/react-ui/EventListener");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const c = require("@splunk/react-ui/Popover");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const f = require("@splunk/react-ui/useControlled");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "styled-components"
    const p = require("styled-components");
    var m =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Slider/getStepMarksBackground.ts
    /* this file is being ignored by stylelint via .stylelintignore */
    /* the radial-gradient returned in the map below needs `css` before backticks
 * for variable interpolation but is not valid css on its own which throws
 * a stylelint `Unknown word  CssSyntaxError` */
    /* getStepMarksBackground generates the multiple radial gradient backgrounds for step marks */
    function h(e) {
        var t = e.stepWidth, r = e.thumbPosition, n = r === void 0 ? 0 : r, a = e.beforeThumbStepColor, i = e.afterThumbStepColor;
        /* need to subtract 1 to avoid extra step at the end */        var o = Math.floor(100 / t) - 1;
        var l = new Array(o).fill(false);
        return (0, p.css)([ "background:", ";" ], l.map((function(e, r) {
            var o = (r + 1) * t;
            var u = o <= n ? a : i;
            /* css doesn't allow trailing comma */            var s = r < l.length - 1 ? "," : "";
            return (0, p.css)([ "radial-gradient(circle,", " 0%,", " 3px,transparent 3px) calc(", "% - 1.5px) center no-repeat ", "" ], u, u, o, s);
        })));
    }
    /* harmony default export */    const y = h;
    // CONCATENATED MODULE: ./src/Slider/SliderStyles.ts
    var g = 8;
    var w = 1;
    var C = 24;
    var S = m().div.withConfig({
        displayName: "SliderStyles__StyledSliderWrapper",
        componentId: "sc-16ooit4-0"
    })([ "", ";gap:", ";width:100%;", "" ], b.mixins.reset("flex"), b.variables.spacingLarge, (function(e) {
        var t = e.$inline;
        return t && (0, p.css)([ "width:300px;" ]);
    }));
    // Use .attrs() to set CSS variable via style attribute to avoid styled-components class regeneration.
    // The ::before pseudo-element reads this variable, which is the only way to style pseudo-elements dynamically
    // without regenerating CSS classes on every position change.
        var x = m().div.attrs((function(e) {
        var t = e.$thumbPosition;
        return {
            style: t !== undefined ? {
                "--slider-fill-width": "".concat(t, "%")
            } : undefined
        };
    })).withConfig({
        displayName: "SliderStyles__StyledSliderBar",
        componentId: "sc-16ooit4-1"
    })([ "position:relative;left:0;height:", "px;width:100%;border-radius:10px;background-color:", ";border:", "px solid;border-color:", ";", " ", "" ], g, (0, 
    b.pickVariant)("$state", {
        default: b.variables.neutral300,
        error: b.variables.neutral300,
        disabled: b.variables.neutral200
    }), w, (0, b.pickVariant)("$state", {
        default: b.variables.interactiveColorBorder,
        error: b.variables.interactiveColorBorder,
        disabled: "transparent"
    }), (function(e) {
        var t = e.$thumbPosition;
        return (t || t === 0) && (0, p.css)([ "&::before{position:absolute;border-radius:", ";content:'';height:calc( ", "px + ", "px );top:-", "px;left:-", "px;width:var(--slider-fill-width,0%);background-color:", ";}" ], b.variables.borderRadius, g, w * 2, w, w, (0, 
        b.pickVariant)("$state", {
            default: b.variables.interactiveColorAccent,
            error: b.variables.interactiveColorAccentError,
            disabled: b.variables.neutral300
        }));
    }), (function(e) {
        var t = e.$stepWidth, r = e.$thumbPosition;
        return t && (0, p.css)([ "&::after{position:absolute;top:0;left:0;width:100%;height:100%;", ";background-size:8px 8px;}" ], y({
            beforeThumbStepColor: (0, b.pickVariant)("$state", {
                default: b.variables.contentColorAccentWeak,
                error: b.variables.contentColorNegativeWeak,
                disabled: b.variables.contentColorDisabled
            }),
            afterThumbStepColor: (0, b.pickVariant)("$state", {
                default: b.variables.contentColorDefault,
                error: b.variables.contentColorDefault,
                disabled: b.variables.contentColorDisabled
            }),
            stepWidth: t,
            thumbPosition: r
        }));
    }));
    // Use .attrs() to compute positioning styles and set them via style attribute.
    // This avoids styled-components class regeneration during drag operations, significantly
    // improving performance when the thumb moves continuously.
        var k = m().button.attrs((function(e) {
        var t = e.$thumbPosition, r = e.$stepWidth;
        if (t === undefined) {
            return {
                style: undefined
            };
        }
        if (r) {
            // With step marks
            var n = -C / 2;
 /* used to align thumb center with position */            if (t === 0) {
                /* used to align thumb at start of bar (without going out of bar bounds) */
                n = 0;
            } else if (t === 100) {
                /* used to align thumb at end of bar (without going out of bar bounds) */
                n = -C;
            }
            return {
                style: {
                    left: "".concat(t, "%"),
                    marginLeft: "".concat(n, "px")
                }
            };
        }
        /* Smoothly position thumb center at the min/max step position while keeping it
     within bounds. This calculation doesn't match exactly with step marks so it is
     only used when stepMarks are too small to be rendered. */        return {
            style: {
                left: "calc(".concat(t, "% - ").concat(t / 100 * C, "px)")
            }
        };
    })).withConfig({
        displayName: "SliderStyles__StyledSliderThumb",
        componentId: "sc-16ooit4-2"
    })([ "display:block;position:absolute;cursor:pointer;width:", "px;height:", "px;border-radius:50%;background-color:", ";border:2px solid ", ";", " ", "" ], C, C, (0, 
    b.pickVariant)("$state", {
        default: b.variables.white,
        error: b.variables.interactiveColorAccentError,
        disabled: b.variables.neutral300
    }), (0, b.pickVariant)("$state", {
        default: b.variables.borderColorStrong,
        error: b.variables.interactiveColorAccentErrorStrong,
        disabled: b.variables.neutral300
    }), (function(e) {
        var t = e.$state;
        return t !== "disabled" && (0, p.css)([ "&:hover,&:focus,&:active{box-shadow:", ";}" ], b.variables.focusShadow);
    }), (function(e) {
        var t = e.$state;
        return t === "disabled" && (0, p.css)([ "cursor:not-allowed;" ]);
    }));
    var E = m().div.withConfig({
        displayName: "SliderStyles__StyledPopoverContent",
        componentId: "sc-16ooit4-3"
    })([ "padding:", ";" ], b.variables.spacingMedium);
    var $ = m().span.withConfig({
        displayName: "SliderStyles__StyledLabel",
        componentId: "sc-16ooit4-4"
    })([ "align-self:center;color:", ";" ], (function(e) {
        var t = e.$disabled;
        return t ? b.variables.contentColorDisabled : b.variables.contentColorActive;
    }));
    /* using content: '' to control if step marks should be shown
 * to avoid further issues with variable interpolation given the
 * complexity of the multiple radial gradient backgrounds. This also
 * allows us not to pass in StyledSliderTypes to StyledInput */    var A = (0, p.css)([ "", "{&::after{content:'';}}" ], /* sc-sel */ x);
    var O = m().div.withConfig({
        displayName: "SliderStyles__StyledInput",
        componentId: "sc-16ooit4-5"
    })([ "width:100%;min-width:200px;display:flex;align-items:center;flex:1 0 0;position:relative;cursor:default;height:", ";max-width:100%;", " &:hover{", "{border-color:", ";}}", "" ], b.variables.inputHeight, (function(e) {
        var t = e.$state;
        return t === "disabled" && (0, p.css)([ "cursor:not-allowed;" ]);
    }), /* sc-sel */ x, (0, b.pickVariant)("$state", {
        default: b.variables.interactiveColorBorderHover,
        error: b.variables.interactiveColorBorderHover,
        disabled: "transparent"
    }), (0, b.pickVariant)("$stepAppearance", {
        focus: (0, p.css)([ "&:focus-within{", "}" ], A),
        always: (0, p.css)([ "", "" ], A)
    }));
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var P = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function T() {
        var e = typeof document !== "undefined" ? document : P;
        return e;
    }
    var V = T();
    /* harmony default export */    const B = /* unused pure expression or super */ null && V;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var I = {
        document: P,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                t();
                return null;
            }
            return setTimeout(t, 0);
        },
        cancelAnimationFrame: function e(t) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(t);
        }
    };
    function L() {
        var e = typeof window !== "undefined" ? window : I;
        return e;
    }
    var _ = L();
    /* harmony default export */    const j = _;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function q(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Slider/Slider.tsx
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function N(e, t) {
        return H(e) || F(e, t) || R(e, t) || D();
    }
    function D() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function R(e, t) {
        if (e) {
            if ("string" == typeof e) return W(e, t);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? W(e, t) : void 0;
        }
    }
    function W(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var r = 0, n = Array(t); r < t; r++) {
            n[r] = e[r];
        }
        return n;
    }
    function F(e, t) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var n, a, i, o, l = [], u = !0, s = !1;
            try {
                if (i = (r = r.call(e)).next, 0 === t) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (n = i.call(r)).done) && (l.push(n.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                s = !0, a = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (o = r["return"](), Object(o) !== o)) return;
                } finally {
                    if (s) throw a;
                }
            }
            return l;
        }
    }
    function H(e) {
        if (Array.isArray(e)) return e;
    }
    function U(e, t) {
        if (null == e) return {};
        var r, n, a = z(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function z(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    /** @public */    var K = {
        defaultValue: l().number,
        describedBy: l().string,
        disabled: l().bool,
        displayValue: l().string,
        elementRef: l().oneOfType([ l().func, l().object ]),
        error: l().bool,
        inline: l().bool,
        inputId: l().string,
        labelledBy: l().string,
        max: l().number,
        maxLabel: l().node,
        min: l().number,
        minLabel: l().node,
        name: l().string,
        onChange: l().func,
        /** @private. */
        required: l().bool,
        step: l().number,
        stepMarks: l().oneOf([ "focus", "always", "never" ]),
        thumbRef: l().oneOfType([ l().func, l().object ]),
        value: l().number
    };
    var X = function e(t) {
        var r = t.max, n = t.min, a = t.value;
        if (a > r) {
            return r;
        }
        if (a < n) {
            return n;
        }
        return a;
    };
    // If the current value is in between a step, i.e. defaultValue is not at a
    // step mark, this function handles navigating to the next correct step
        var G = function e(t) {
        var r = t.direction, n = t.min, a = t.step, o = t.value;
        var l = new (i())(o).minus(n).modulo(a).toNumber();
        if (l > 0) {
            return r === "forward" ? a - l : l;
        }
        return a;
    };
    var J = function e(t, r) {
        if (r === null) {
            return undefined;
        }
        var n = r.getBoundingClientRect();
        var a = t.clientX - n.left;
        return X({
            max: 100,
            min: 0,
            value: a / n.width * 100
        });
    };
    var Q = function e(t) {
        var r = t.max, n = t.min, a = t.step, o = t.value;
        // Handles if the max is in between steps but val is larger than the halfway point between last step mark and max
                if (o > r - (r - n) % a / 2) {
            return r;
        }
        return new (i())(o).minus(n).div(a).todp(0).mul(a).add(n).toNumber();
    };
    var Y = function e(t) {
        var r = t.position, n = t.max, a = t.min, i = t.step;
        var o = n - a;
        var l = r / 100 * o + a;
        return Q({
            max: n,
            min: a,
            step: i,
            value: l
        });
    };
    var Z = function e(t) {
        var r = t.max, n = t.min, a = t.sliderBar, i = t.step;
        // don't render stepMarks if they are too small i.e. stepWidth is less than SliderThumb divided by 2
        // in that case, undefined is returned
                if (a === null) {
            return undefined;
        }
        var o = r - n;
        var l = Math.round(o / i);
        var u = a.getBoundingClientRect();
        var s = i * 100 / o;
        return u.width / l > C / 2 ? s : undefined;
    };
    var ee = function e(t) {
        var r = t.max, n = t.min, a = t.value;
        if (a === undefined) {
            return undefined;
        }
        var i = r - n;
        return (a - n) / i * 100;
    };
    function te(e) {
        var t = e.defaultValue, a = e.describedBy, i = e.disabled, o = e.displayValue, l = e.elementRef, u = e.error, c = e.inline, f = e.inputId, p = e.labelledBy, m = e.max, b = m === void 0 ? 5 : m, h = e.maxLabel, y = e.min, g = y === void 0 ? 1 : y, w = e.minLabel, C = e.name, A = e.onChange, P = e.required, T = e.step, V = T === void 0 ? 1 : T, B = e.stepMarks, I = B === void 0 ? "focus" : B, L = e.thumbRef, _ = e.value, D = U(e, [ "defaultValue", "describedBy", "disabled", "displayValue", "elementRef", "error", "inline", "inputId", "labelledBy", "max", "maxLabel", "min", "minLabel", "name", "onChange", "required", "step", "stepMarks", "thumbRef", "value" ]);
        // @docs-props-type SliderPropsBase
                var R = v()({
            componentName: "Slider",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var W = (0, r.useState)(false), F = N(W, 2), H = F[0], z = F[1];
        var K = (0, r.useState)(false), te = N(K, 2), re = te[0], ne = te[1];
        var ae = (0, r.useState)(null), ie = N(ae, 2), oe = ie[0], le = ie[1];
        var ue = (0, r.useState)(null), se = N(ue, 2), ce = se[0], de = se[1];
        var fe = (0, r.useMemo)((function() {
            return R ? Q({
                value: (b - g) / 2,
                min: g,
                max: b,
                step: V
            }) : t;
        }), [ R, t, b, g, V ]);
        var ve = (0, r.useState)(R ? _ : fe), pe = N(ve, 2), me = pe[0], be = pe[1];
        var he = (0, r.useCallback)((function(e, t) {
            var r = t.newValue;
            if (me !== r) {
                A === null || A === void 0 ? void 0 : A(e, {
                    value: r,
                    name: C
                });
                if (!R) {
                    be(r);
                }
            }
        }), [ R, C, A, me ]);
        // handlers
                var ye = (0, r.useCallback)((function() {
            ne(false);
        }), []);
        var ge = (0, r.useCallback)((function() {
            ne(true);
        }), []);
        var we = (0, r.useCallback)((function(e) {
            le(e);
        }), []);
        var Ce = (0, r.useCallback)((function(e) {
            de(e);
            q(L, e);
        }), [ L ]);
        var Se = (0, r.useCallback)((function(e) {
            var t = J(e, oe);
            if (t !== undefined) {
                var r = Y({
                    position: t,
                    max: b,
                    min: g,
                    step: V
                });
                he(e, {
                    newValue: r
                });
            }
        }), [ b, g, he, oe, V ]);
        var xe = (0, r.useCallback)((function(e) {
            var t = me;
            if (t !== undefined) {
                var r = e.key;
                var n;
                var a = G({
                    direction: "forward",
                    min: g,
                    step: V,
                    value: t
                });
                var i = G({
                    direction: "backward",
                    min: g,
                    step: V,
                    value: t
                });
                if (r === "ArrowRight") {
                    n = t + a;
                } else if (r === "ArrowLeft") {
                    n = t - i;
                } else if (r === "ArrowUp") {
                    e.preventDefault();
                    n = t + a;
                } else if (r === "ArrowDown") {
                    e.preventDefault();
                    n = t - i;
                } else {
                    return;
                }
                t = X({
                    max: b,
                    min: g,
                    value: n
                });
                t = Q({
                    max: b,
                    min: g,
                    step: V,
                    value: t
                });
                he(e, {
                    newValue: t
                });
            }
        }), [ b, g, he, V, me ]);
        var ke = (0, r.useCallback)((function(e) {
            e.preventDefault();
            ce === null || ce === void 0 ? void 0 : ce.focus();
            z(true);
            ne(true);
        }), [ ce ]);
        var Ee = (0, r.useCallback)((function() {
            ne(true);
        }), []);
        var $e = (0, r.useCallback)((function() {
            if (!H) {
                ne(false);
            }
        }), [ H ]);
        var Ae = (0, r.useCallback)((function(e) {
            if (H) {
                var t = J(e, oe);
                if (t !== undefined) {
                    var r = Y({
                        position: t,
                        min: g,
                        max: b,
                        step: V
                    });
                    he(e, {
                        newValue: r
                    });
                }
            }
        }), [ H, b, g, he, oe, V ]);
        var Oe = (0, r.useCallback)((function() {
            z(false);
            ne(false);
        }), []);
        var Pe = (0, r.useCallback)((function(e) {
            // When a label is clicked, it triggers a synthetic click event on the associated control.
            // For Slider, we only want to focus the thumb, not trigger any other behavior.
            // We can detect label-initiated clicks because they have a detail of 0 (synthetic)
            // and the button wasn't the actual click target.
            if (e.detail === 0) {
                e.preventDefault();
                e.stopPropagation();
                ce === null || ce === void 0 ? void 0 : ce.focus();
            }
        }), [ ce ]);
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ b, g ]);
        (0, r.useEffect)((function() {
            // keep valueState in sync with value prop when controlled
            if (R) {
                be(_);
            }
        }), [ R, _ ]);
        var Te = i && "disabled" || u && "error" || "default";
        var Ve = ee({
            value: me,
            min: g,
            max: b
        });
        var Be = V ? Z({
            max: b,
            min: g,
            sliderBar: oe,
            step: V
        }) : undefined;
         /* eslint-disable jsx-a11y/no-static-element-interactions */
        return n().createElement(S, M({
            "data-test": "slider",
            "data-test-disabled": i ? "disabled" : undefined,
            "data-test-value": me,
            ref: l,
            $inline: c
        }, D), H &&  n().createElement(n().Fragment, null,  n().createElement(s(), {
            target: j,
            eventType: "mouseup",
            listener: Oe
        }),  n().createElement(s(), {
            target: j,
            eventType: "mousemove",
            listener: Ae
        })), w !== null &&  n().createElement($, {
            "data-test": "min-label",
            $disabled: i
        }, w || g),  n().createElement(O, {
            onClick: i ? undefined : Se,
            $state: Te,
            $stepAppearance: I
        },  n().createElement(x, {
            "aria-required": P,
            "data-test": "bar",
            ref: we,
            $thumbPosition: Ve,
            $stepWidth: Be,
            $state: Te
        }),  n().createElement(k, {
            id: f,
            "aria-describedby": a,
            "aria-labelledby": p,
            "aria-invalid": u,
            "aria-valuemax": b,
            "aria-valuemin": g,
            "aria-valuenow": me,
            "data-test": "thumb",
            onBlur: ye,
            onClick: i ? undefined : Pe,
            onMouseEnter: Ee,
            onMouseLeave: $e,
            onKeyDown: i ? undefined : xe,
            onMouseDown: i ? undefined : ke,
            onFocus: ge,
            ref: Ce,
            role: "slider",
            type: "button",
            disabled: i,
            $thumbPosition: Ve,
            $stepWidth: Be,
            $state: Te
        }),  n().createElement(d(), {
            open: re,
            anchor: ce,
            align: "center"
        },  n().createElement(E, null, o || me))), h !== null &&  n().createElement($, {
            "data-test": "max-label",
            $disabled: i
        }, h || b));
    }
    te.propTypes = K;
    /* harmony default export */    const re = te;
    // CONCATENATED MODULE: ./src/Slider/index.ts
    module.exports = t;
    /******/})();
