// @splunk/react-ui/SlidingPanels.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Panel: () => /* reexport */ A,
        default: () => /* reexport */ K
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "@react-spring/web"
    const o = require("@react-spring/web");
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var a =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const u = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const f = require("@splunk/react-ui/Box");
    var s =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SlidingPanels/SlidingPanelsStyles.ts
    var p = c()(s()).withConfig({
        displayName: "SlidingPanelsStyles__StyledBox",
        componentId: "su6isq-0"
    })([ "overflow:hidden;position:relative;" ]);
    var d = c()(o.animated.div).withConfig({
        displayName: "SlidingPanelsStyles__StyledAnimatedDiv",
        componentId: "su6isq-1"
    })([ "", ";", ";" ], y.mixins.reset("block"), y.mixins.clearfix());
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function v(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/SlidingPanels/Panel.tsx
    function m() {
        return m = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, m.apply(null, arguments);
    }
    function b(e, n) {
        return j(e) || h(e, n) || g(e, n) || O();
    }
    function O() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function g(e, n) {
        if (e) {
            if ("string" == typeof e) return S(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? S(e, n) : void 0;
        }
    }
    function S(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function h(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, u = [], l = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    l = !1;
                } else for (;!(l = (r = i.call(t)).done) && (u.push(r.value), u.length !== n); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function j(e) {
        if (Array.isArray(e)) return e;
    }
    function w(e, n) {
        if (null == e) return {};
        var t, r, o = P(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function P(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var x = {
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        /** @private */
        onMount: a().func,
        /** @private */
        onUnmount: a().func,
        panelId: a().oneOfType([ a().string, a().number ]).isRequired
    };
    /**
 * Container for arbitrary content.
 */    function E(e) {
        var n = e.children, o = e.elementRef, i = e.onMount, a = e.onUnmount, u = e.panelId, l = w(e, [ "children", "elementRef", "onMount", "onUnmount", "panelId" ]);
        // @docs-props-type PanelPropsBase
                var c = (0, t.useState)(null), f = b(c, 2), s = f[0], y = f[1];
        var p = (0, t.useCallback)((function(e) {
            y(e);
            v(o, e);
        }), [ o ]);
        (0, t.useEffect)((function() {
            i === null || i === void 0 ? void 0 : i(s, u);
            return function() {
                a === null || a === void 0 ? void 0 : a(u);
            };
        }), [ i, a, s, u ]);
        
        return r().createElement(d, m({
            "data-test": "panel",
            "data-test-panel-id": u
        }, l, {
            ref: p
        }), n);
    }
    E.propTypes = x;
    /* harmony default export */    const A = E;
    // CONCATENATED MODULE: ./src/SlidingPanels/SlidingPanels.tsx
    function I(e) {
        "@babel/helpers - typeof";
        return I = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, I(e);
    }
    function M() {
        return M = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, M.apply(null, arguments);
    }
    function C(e, n) {
        return k(e) || R(e, n) || T(e, n) || N();
    }
    function N() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function T(e, n) {
        if (e) {
            if ("string" == typeof e) return q(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? q(e, n) : void 0;
        }
    }
    function q(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function R(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, u = [], l = !0, c = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    l = !1;
                } else for (;!(l = (r = i.call(t)).done) && (u.push(r.value), u.length !== n); l = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!l && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return u;
        }
    }
    function k(e) {
        if (Array.isArray(e)) return e;
    }
    function U(e, n) {
        if (null == e) return {};
        var t, r, o = _(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function _(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    function D(e, n) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            n && (r = r.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), t.push.apply(t, r);
        }
        return t;
    }
    function B(e) {
        for (var n = 1; n < arguments.length; n++) {
            var t = null != arguments[n] ? arguments[n] : {};
            n % 2 ? D(Object(t), !0).forEach((function(n) {
                V(e, n, t[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : D(Object(t)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(t, n));
            }));
        }
        return e;
    }
    function V(e, n, t) {
        return (n = $(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function $(e) {
        var n = L(e, "string");
        return "symbol" == I(n) ? n : n + "";
    }
    function L(e, n) {
        if ("object" != I(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != I(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    // can't use PanelId as key type, because an index signature parameter type must be 'string', 'number', 'symbol', or a template literal type.
        var W = {
        activePanelId: a().oneOfType([ a().string, a().number ]).isRequired,
        children: a().node,
        elementRef: a().oneOfType([ a().func, a().object ]),
        innerClassName: a().string,
        innerStyle: a().object,
        onAnimationEnd: a().func,
        outerClassName: a().string,
        outerStyle: a().object,
        transition: a().oneOf([ "forward", "backward" ])
    };
    var z = {
        enter: {
            x: "0px"
        },
        initial: {
            x: "0px"
        }
    };
    var F = function e(n, t) {
        var r = "".concat(n, "px");
        var o = "".concat(n * -1, "px");
        var i = t === "forward" ? r : o;
        var a = t === "forward" ? o : r;
        return {
            from: {
                x: i
            },
            /**
     * Use absolute positioning so the arriving and leaving panel take up the same space.
     * Without this, the arriving panel and leaving panel will be visible at the same time, because
     * the arriving panel would insert itself underneath the still visible leaving panel.
     */
            leave: {
                position: "absolute",
                x: a
            }
        };
    };
    var G = function e(n) {
        return Object.values(n).reduce((function(e, n) {
            if ((n === null || n === void 0 ? void 0 : n.tagName) === "DIV") {
                return Math.max(e, n.clientWidth);
            }
            return e;
        }), 0);
    };
    /* This component is used to wrap each child panel in animated.div via StyledAnimatedDiv
   apply any custom inner styles or classnames that have been passed in
   apply styles coming from the transitions call that is rendering this component */    var H =  r().memo((function(e) {
        var n = e.innerClassName, o = e.innerStyle, i = e.onMount, a = e.onUnmount, u = e.panel, l = e.style;
        var c = (0, t.useMemo)((function() {
            
            return (0, t.cloneElement)(u, {
                onMount: i,
                onUnmount: a
            });
        }), [ u, i, a ]);
        var f = (0, t.useMemo)((function() {
            return B(B({}, o), l);
        }), [ o, l ]);
        
        return r().createElement(d, {
            style: f,
            className: n
        }, c);
    }));
    function J(e) {
        var n = e.activePanelId, i = e.children, a = e.elementRef, l = e.innerClassName, c = e.innerStyle, f = e.onAnimationEnd, s = e.outerClassName, y = e.outerStyle, d = e.transition, v = d === void 0 ? "forward" : d, m = U(e, [ "activePanelId", "children", "elementRef", "innerClassName", "innerStyle", "onAnimationEnd", "outerClassName", "outerStyle", "transition" ]);
        // @docs-props-type SlidingPanelsPropsBase
                var b = (0, t.useState)(0), O = C(b, 2), g = O[0], S = O[1];
        var h = (0, t.useState)([]), j = C(h, 2), w = j[0], P = j[1];
        var x = (0, t.useState)({}), E = C(x, 2), A = E[0], I = E[1];
        var N = (0, o.useSpringRef)();
        var T = (0, t.useMemo)((function() {
            return t.Children.toArray(i).filter(t.isValidElement);
        }), [ i ]);
        /**
   * maxWidth is used to determine how far to translate the x value of the animated panels.
   * Without maxWidth, we run into a problem if the leaving panel width > the arriving panel width.
   * In this case, useAnimationTransition will only know to shift the leaving panel by the width of the arriving panel,
   * which would result in seeing part of the leaving panel's content on screen at the same time as the arriving panel's content.
   */        var q = (0, t.useMemo)((function() {
            return G(A);
        }), [ A ]);
        /* determines what our transform translateX will look like based on is it transitioning forward or backward */        var R = (0, 
        t.useMemo)((function() {
            return F(q, v);
        }), [ q, v ]);
        var k = (0, u.useAnimationTransition)(g, B(B(B({
            ref: N
        }, z), R), {}, {
            onRest: function e() {
                f === null || f === void 0 ? void 0 : f();
            }
        }));
        (0, t.useEffect)((function() {
            /* if it can't find activePanelId it will set currentPanelIndex to -1 */
            if (false) {}
        }), [ g ]);
        (0, t.useEffect)((function() {
            /* starts animation */
            N.start();
        }), [ N, g ]);
        (0, t.useEffect)((function() {
            /* sets the currentPanelIndex in state whenever activePanelId or children changes */
            var e = T.findIndex((function(e) {
                var t = e.props;
                return t.panelId === n;
            }));
            S(e);
        }), [ n, T ]);
        var _ = (0, t.useCallback)((function(e, n) {
            if (e != null && n != null) {
                I((function(t) {
                    return B(B({}, t), {}, V({}, n, e));
                }));
            }
        }), []);
        var D = (0, t.useCallback)((function(e) {
            I((function(n) {
                var t = B({}, n);
                delete t[e];
                return t;
            }));
        }), []);
        (0, t.useLayoutEffect)((function() {
            /**
     * use useLayoutEffect instead of useEffect to setPanels because we need panels to be populated before mount
     * so SlidingPanels will appear with a panel already inside the StyledBox.  You can recreate issue by going
     * to Storybook and quickly switching between basic and dropdown examples.  The basic example will eventually
     * show a small square on initial render, vs what we want to be the entire initial panel.
     */
            P(T);
        }), [ T ]);
        
        return r().createElement(p, M({
            className: s,
            "data-test-active-panel-id": n,
            "data-test": "sliding-panels",
            ref: a,
            style: y
        }, m), k((function(e, n) {
            var t = w[n];
            return t ?  r().createElement(H, {
                innerClassName: l,
                innerStyle: c,
                onMount: _,
                onUnmount: D,
                panel: t,
                style: e
            }) : null;
        })));
    }
    J.propTypes = W;
    J.Panel = A;
    /* harmony default export */    const K = J;
    // CONCATENATED MODULE: ./src/SlidingPanels/index.ts
    module.exports = n;
    /******/})();
