// @splunk/react-ui/SplitButton.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Item: () => /* reexport */ k,
        default: () => /* reexport */ B
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-icons/CaretSmallDown"
    const o = require("@splunk/react-icons/CaretSmallDown");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const c = require("@splunk/react-ui/Dropdown");
    var u =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/Menu"
    const p = require("@splunk/react-ui/Menu");
    var s =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const d = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "lodash/omit"
    const f = require("lodash/omit");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const m = require("@splunk/react-ui/Button");
    var b =  e.n(m);
    // CONCATENATED MODULE: ./src/SplitButton/Item.tsx
    function y() {
        return y = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, y.apply(null, arguments);
    }
    function g(e, n) {
        if (null == e) return {};
        var r, t, a = O(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function O(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */    var h = {
        appearance: i().oneOf([ "default", "secondary", "primary", "destructive", "destructiveSecondary" ]),
        children: i().node,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        icon: i().node,
        isMain: i().bool,
        onClick: i().func
    };
    /**
 * An item within a `SplitButton`.
 */    function S(e) {
        var n = e.appearance, a = n === void 0 ? "secondary" : n, i = e.disabled, o = e.elementRef, l = e.icon, c = e.isMain, u = e.children, p = e.onClick, d = g(e, [ "appearance", "disabled", "elementRef", "icon", "isMain", "children", "onClick" ]);
        // @docs-props-type ItemPropsBase
                var f = (0, r.useCallback)((function(e) {
            if (!i) {
                p === null || p === void 0 ? void 0 : p(e, {});
            }
        }), [ i, p ]);
        var m = (0, r.useCallback)((function(e, n) {
            var r = n.action, t = n.icon, a = n.label, o = n.value;
            if (!i) {
                p === null || p === void 0 ? void 0 : p(e, {
                    action: r,
                    icon: t,
                    label: a,
                    value: o
                });
            }
        }), [ i, p ]);
        if (c) {
            
            return t().createElement(b(), y({
                appearance: a,
                append: true,
                disabled: i,
                elementRef: o,
                icon: l,
                onClick: m
            }, d), u);
        }
        
        return t().createElement(s().Item, y({
            disabled: i,
            elementRef: o,
            onClick: f,
            startAdornment: l
        }, v()(d, [ "role" ])), u);
    }
    S.propTypes = h;
    /* harmony default export */    const k = S;
    // CONCATENATED MODULE: external "styled-components"
    const w = require("styled-components");
    var j =  e.n(w);
    // CONCATENATED MODULE: external "@splunk/themes"
    const E = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/SplitButton/SplitButtonStyles.ts
    var C = j()(b()).withConfig({
        displayName: "SplitButtonStyles__StyledToggle",
        componentId: "sc-9igsyh-0"
    })([ "min-width:0;width:24px;" ]);
    var P = j().div.withConfig({
        displayName: "SplitButtonStyles__StyledDiv",
        componentId: "sc-9igsyh-1"
    })([ "", " ", " ", "" ], E.mixins.reset("inline-flex"), (function(e) {
        var n = e.$appearance;
        return (n === "primary" || n === "destructive") && (0, w.css)([ "gap:1px;" ]);
    }), (function(e) {
        var n = e.$inline;
        return !n && (0, w.css)([ "width:100%;" ]);
    }));
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var M = "is deprecated and will be removed in the next major version.";
    var R = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, a = n.componentName;
        useEffect((function() {
            if (false) {}
        }), [ t, a ]);
    };
    var q = function e(n) {
        var r = n.additionalMessage, t = r === void 0 ? "" : r, a = n.componentName, i = n.propName, o = n.propValue;
        useEffect((function() {
            if (false) {}
        }), [ t, a, i, o ]);
    };
    var x = function e(n) {
        var t = n.additionalMessage, a = t === void 0 ? "" : t, i = n.componentName, o = n.deprecatedPropValue, l = n.propName, c = n.propValue;
        (0, r.useEffect)((function() {
            if (false) {}
        }), [ a, i, l, c, o ]);
    };
    // CONCATENATED MODULE: ./src/SplitButton/SplitButton.tsx
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function N(e, n) {
        if (null == e) return {};
        var r, t, a = T(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (t = 0; t < i.length; t++) {
                r = i[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function T(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var I = {
        appearance: i().oneOf([ "default", "secondary", "primary", "destructive", "destructiveSecondary" ]),
        children: i().node,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        onClick: i().func,
        inline: i().bool,
        toggleRef: i().oneOfType([ i().func, i().object ])
    };
    function V(e) {
        var n = e.appearance, a = n === void 0 ? "secondary" : n, i = e.children, o = e.disabled, c = e.elementRef, p = e.inline, f = p === void 0 ? true : p, v = e.onClick, m = e.toggleRef, b = N(e, [ "appearance", "children", "disabled", "elementRef", "inline", "onClick", "toggleRef" ]);
        // @docs-props-type SplitButtonPropsBase
                x({
            componentName: "SplitButton",
            deprecatedPropValue: "default",
            propName: "appearance",
            propValue: a
        });
        var y = r.Children.toArray(i).filter(r.isValidElement);
        var g = 0;
        var O = 0;
        y.forEach((function(e, n) {
            var r = e.props, t = r.disabled, a = r.isMain;
            if (a) {
                g = n;
            }
            if (t) {
                O += 1;
            }
        }));
        var h =  (0, r.cloneElement)(y[g], {
            isMain: true,
            appearance: a,
            disabled: o || y[g].props.disabled
        });
        // Remove mainButton from array of dropdown items
                y.splice(g, 1);
        var S = y;
        var k =  t().createElement(C, {
            prepend: true,
            disabled: o || O === S.length,
            elementRef: m,
            appearance: a,
            "data-test": "split-button-toggle",
            "aria-label": (0, d._)("Open more options"),
            icon:  t().createElement(l(), null)
        });
        
        return t().createElement(P, _({
            "aria-label": (0, d._)("Split button"),
            "data-inline": f || null,
            "data-test": "split-button-container",
            $inline: f,
            ref: c,
            role: "group",
            onClick: o ? undefined : v,
            $appearance: a
        }, b), h,  t().createElement("div", null,  t().createElement(u(), {
            toggle: k
        },  t().createElement(s(), null, S))));
    }
    V.propTypes = I;
    V.Item = k;
    /* harmony default export */    const B = V;
    // CONCATENATED MODULE: ./src/SplitButton/index.ts
    module.exports = n;
    /******/})();
