// @splunk/react-ui/StaticContent.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ b
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const i = require("prop-types");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const s = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/StaticContent/StaticContentStyles.ts
    var u = o().div.withConfig({
        displayName: "StaticContentStyles__StyledDiv",
        componentId: "sxqyrp-0"
    })([ "", " position:relative;margin:0;font-weight:", ";word-wrap:break-word;min-height:", ";padding:0 ", ";align-items:center;" ], (function(e) {
        var n = e.$inline;
        return n ? s.mixins.reset("inline-flex") : s.mixins.reset("flex");
    }), s.variables.fontWeightSemiBold, s.variables.inputHeight, s.variables.spacingXSmall);
    // CONCATENATED MODULE: ./src/StaticContent/StaticContent.tsx
    function c() {
        return c = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, c.apply(null, arguments);
    }
    function f(e, n) {
        if (null == e) return {};
        var r, t, i = d(e, n);
        if (Object.getOwnPropertySymbols) {
            var l = Object.getOwnPropertySymbols(e);
            for (t = 0; t < l.length; t++) {
                r = l[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (i[r] = e[r]);
            }
        }
        return i;
    }
    function d(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    var p = {
        children: l().node,
        elementRef: l().oneOfType([ l().func, l().object ]),
        inline: l().bool,
        labelledBy: l().string
    };
    /**
 * This component is intended for use in a control group, either to display a static value or
 * between two controls.
 */    function y(e) {
        var n = e.children, r = e.elementRef, i = e.inline, l = i === void 0 ? false : i, a = e.labelledBy, o = f(e, [ "children", "elementRef", "inline", "labelledBy" ]);
        // @docs-props-type StaticContentPropsBase
                
        return t().createElement(u, c({
            "aria-labelledby": a,
            "data-test": "static-content",
            $inline: l,
            ref: r
        }, o),  t().createElement("span", null, n));
    }
    y.propTypes = p;
    /* harmony default export */    const b = y;
    // CONCATENATED MODULE: ./src/StaticContent/index.ts
    module.exports = n;
    /******/})();
