// @splunk/react-ui/StepBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Step: () => /* reexport */ N,
        default: () => /* reexport */ D
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const o = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/StepBar/StepBarContext.tsx
    var l =  (0, r.createContext)({
        activeStepId: 0,
        orientation: "horizontal"
    });
    l.displayName = "StepBar";
    /* harmony default export */    const c = l;
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var p =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckCircle"
    const d = require("@splunk/react-icons/CheckCircle");
    var u =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/react-icons/ExclamationCircle"
    const v = require("@splunk/react-icons/ExclamationCircle");
    var f =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/StepBar/StepStyles.ts
    var m = "2px";
    var y = "24px";
    var g = "10px";
 /* half of the actual rendered svg size (20px) */    var h = (0, 
    s.css)([ "height:", ";width:", ";" ], y, y);
    var C = p().div.withConfig({
        displayName: "StepStyles__StyledBar",
        componentId: "sc-756fxp-0"
    })([ "position:absolute;top:10px;height:", ";width:100%;&::before,&::after{content:'';display:block;height:", ";}" ], m, m);
    var S = p().li.withConfig({
        displayName: "StepStyles__Styled",
        componentId: "sc-756fxp-1"
    })([ "", ";display:grid;", " min-width:min-content;position:relative;overflow:hidden;color:", ";counter-increment:", ";&:not(:first-of-type){", "{&::before{box-shadow:inset 500px 0 ", ";width:calc(50% - ", ");}}}&:not(:last-of-type){", "{&::after{width:50%;position:absolute;top:0;box-shadow:inset 500px 0 ", ";inset-inline-start:calc(50% + ", ");}}}" ], b.mixins.typography("body", {
        weight: "semiBold"
    }), (0, b.pickVariant)("$orientation", {
        horizontal: (0, s.css)([ "grid-template-rows:auto 1fr;row-gap:", ";justify-items:center;" ], b.variables.spacingXSmall),
        vertical: (0, s.css)([ "grid-template-rows:1lh auto;grid-template-columns:auto 1fr;column-gap:", ";align-items:center;" ], b.variables.spacingMedium)
    }), (0, b.pickVariant)("$status", {
        error: b.variables.contentColorActive,
        active: b.variables.contentColorAccent,
        next: b.variables.contentColorDefault,
        prev: b.variables.contentColorActive
    }), (function(e) {
        var t = e.$idCounter;
        return t;
    }), /* sc-sel */ C, (0, b.pickVariant)("$status", {
        error: b.variables.contentColorAccent,
        active: b.variables.contentColorAccent,
        next: b.variables.borderColor,
        prev: b.variables.contentColorAccent
    }), g, /* sc-sel */ C, (0, b.pickVariant)("$status", {
        error: b.variables.borderColor,
        active: b.variables.borderColor,
        next: b.variables.borderColor,
        prev: b.variables.contentColorAccent
    }), g);
    var x = p().div.withConfig({
        displayName: "StepStyles__StyledStepIndicatorWrapper",
        componentId: "sc-756fxp-2"
    })([ "", "" ], (0, b.pickVariant)("$orientation", {
        vertical: (0, s.css)([ "grid-row:1;" ])
    }));
    var w = "16px";
    var O = p().span.withConfig({
        displayName: "StepStyles__StyledCircle",
        componentId: "sc-756fxp-3"
    })([ "display:grid;height:", ";width:", ";border-radius:50%;margin:2px;", " &::before{content:counter(", ");line-height:", ";text-align:center;}" ], w, w, (0, 
    b.pickVariant)("$status", {
        active: (0, s.css)([ "background-color:", ";border:2px solid ", ";color:", ";" ], b.variables.contentColorAccent, b.variables.contentColorAccent, b.variables.contentColorInverted),
        next: (0, s.css)([ "border:2px solid ", ";color:", ";" ], b.variables.contentColorDefault, b.variables.contentColorDefault)
    }), (function(e) {
        var t = e.$idCounter;
        return t;
    }), w);
    var I = p()(u()).withConfig({
        displayName: "StepStyles__StyledCheckCircleIcon",
        componentId: "sc-756fxp-4"
    })([ "", ";color:", ";" ], h, b.variables.contentColorAccent);
    var $ = p()(f()).withConfig({
        displayName: "StepStyles__StyledExclamationCircleIcon",
        componentId: "sc-756fxp-5"
    })([ "", ";color:", ";" ], h, b.variables.contentColorNegative);
    var k = p().span.withConfig({
        displayName: "StepStyles__StyledStepLabel",
        componentId: "sc-756fxp-6"
    })([ "", "" ], (0, b.pickVariant)("$orientation", {
        horizontal: (0, s.css)([ "padding-inline:calc(", " / 2);text-align:center;" ], b.variables.spacingXXLarge),
        vertical: (0, s.css)([ "grid-row:1 / -1;overflow-wrap:break-word;word-break:break-word;" ])
    }));
    // CONCATENATED MODULE: ./src/StepBar/Step.tsx
    function _() {
        return _ = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, _.apply(null, arguments);
    }
    function E(e, t) {
        if (null == e) return {};
        var r, n, a = j(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function j(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var P = {
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        error: i().bool,
        stepId: i().any
    };
    function A(e) {
        var t = e.children, a = e.elementRef, i = e.error, o = i === void 0 ? false : i, c = e.stepId, s = E(e, [ "children", "elementRef", "error", "stepId" ]);
        // @docs-props-type StepPropsBase
                var p = (0, r.useContext)(l), d = p.activeStepId, u = p.idCounter, v = p.orientation;
        var f = "next";
        var b = d === c;
        if (c != null && c < d) {
            f = "prev";
        }
        if (b && !o) {
            f = "active";
        }
        if (b && o) {
            f = "error";
        }
        var m = {
            error:  n().createElement($, null),
            prev:  n().createElement(I, null),
            next:  n().createElement(O, {
                $status: f,
                $idCounter: u
            }),
            active:  n().createElement(O, {
                $status: f,
                $idCounter: u
            })
        };
        
        return n().createElement(S, _({
            $status: f,
            "data-test-status": f,
            "data-test": "step",
            "data-test-step-id": c,
            ref: a
        }, s, {
            $idCounter: u,
            $orientation: v
        }), v === "horizontal" &&  n().createElement(C, {
            $status: f
        }),  n().createElement(x, {
            $orientation: v
        }, m[f]),  n().createElement(k, {
            $orientation: v
        }, t));
    }
    A.propTypes = P;
    /* harmony default export */    const N = A;
    // CONCATENATED MODULE: ./src/StepBar/StepBarStyles.ts
    var V = p().ul.withConfig({
        displayName: "StepBarStyles__Styled",
        componentId: "sc-13bf2x4-0"
    })([ "", ";", " counter-reset:", ";", ";" ], b.mixins.reset("grid"), (0, b.pickVariant)("$orientation", {
        horizontal: (0, s.css)([ "grid-auto-flow:column;padding:", ";" ], b.variables.spacingXLarge),
        vertical: (0, s.css)([ "gap:", ";" ], b.variables.spacingXLarge)
    }), (function(e) {
        var t = e.$idCounter;
        return t;
    }), (function(e) {
        var t = e.$inline;
        return t && (0, s.css)([ "display:inline-flex;", "" ], (0, b.pickVariant)("$orientation", {
            vertical: (0, s.css)([ "display:inline-grid;" ])
        }));
    }));
    // CONCATENATED MODULE: ./src/StepBar/StepBar.tsx
    function q() {
        return q = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, q.apply(null, arguments);
    }
    function R(e, t) {
        if (null == e) return {};
        var r, n, a = z(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function z(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var M = {
        activeStepId: i().any.isRequired,
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        inline: i().bool,
        orientation: i().oneOf([ "horizontal", "vertical" ])
    };
    function T(e) {
        var t = e.activeStepId, a = e.children, i = e.elementRef, l = e.inline, s = l === void 0 ? false : l, p = e.orientation, d = p === void 0 ? "horizontal" : p, u = R(e, [ "activeStepId", "children", "elementRef", "inline", "orientation" ]);
        // @docs-props-type StepBarPropsBase
                var v = (0, r.useRef)((0, o.createDOMID)("stepbar")), f = v.current;
        var b = 0;
        var m = r.Children.toArray(a).filter(r.isValidElement);
        var y = m.length - 1;
        var g = m.map((function(e, n) {
            var a = e.props.stepId || n;
            var i = e.props.error;
            if (a === t) {
                b = n;
            }
            
            return (0, r.cloneElement)(e, {
                stepId: a,
                error: i
            });
        }));
        var h = (0, r.useMemo)((function() {
            return {
                idCounter: f,
                lastChildIndex: y,
                activeStepId: t,
                orientation: d
            };
        }), [ t, f, y, d ]);
        
        return n().createElement(V, q({
            $inline: s,
            "data-test": "step-bar",
            "data-test-active-step-id": t,
            role: "progressbar",
            "aria-valuemin": 1,
            "aria-valuemax": y + 1,
            "aria-valuenow": b + 1,
            $idCounter: f,
            $orientation: d,
            ref: i
        }, u),  n().createElement(c.Provider, {
            value: h
        }, g));
    }
    T.propTypes = M;
    T.Step = N;
    /* harmony default export */    const D = T;
    // CONCATENATED MODULE: ./src/StepBar/index.ts
    module.exports = t;
    /******/})();
