// @splunk/react-ui/Switch.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var a = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(a, {
                a
            });
            /******/            return a;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, a) => {
            /******/ for (var i in a) {
                /******/ if (e.o(a, i) && !e.o(r, i)) {
                    /******/ Object.defineProperty(r, i, {
                        enumerable: true,
                        get: a[i]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ G
    });
    // CONCATENATED MODULE: external "react"
    const a = require("react");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var t =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const l = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const n = require("@splunk/react-ui/ScreenReaderContent");
    var c =  e.n(n);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const d = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const s = require("styled-components");
    var u =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckBoxCompleted"
    const b = require("@splunk/react-icons/CheckBoxCompleted");
    var v =  e.n(b);
    // CONCATENATED MODULE: external "@splunk/react-icons/CheckBoxIndeterminate"
    const f = require("@splunk/react-icons/CheckBoxIndeterminate");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const g = require("@splunk/react-ui/Box");
    var m =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const h = require("@splunk/react-ui/Clickable");
    var y =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/themes"
    const k = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Switch/SwitchStyles.ts
    var C = "20px";
    var S = "12px";
    var w = "4px";
 // diameter - toggleIndicatorDiameter / 2
        var x = u()(m()).withConfig({
        displayName: "SwitchStyles__StyledBox",
        componentId: "sc-844ieu-0"
    })([ "display:inline;position:relative;flex-shrink:0;gap:", ";min-height:", ";align-items:center;color:", ";", "" ], k.variables.spacingSmall, k.variables.inputHeight, k.variables.contentColorActive, (function(e) {
        var r = e.$disabled;
        return r && (0, s.css)([ "color:", ";" ], k.variables.contentColorDisabled);
    }));
    var B = (0, s.css)([ "", " position:relative;width:", ";height:", ";line-height:", ";border-radius:", ";border:", " solid ", ";background-color:", ";cursor:pointer;flex:0 0 auto;&:not([disabled]){&:focus{box-shadow:", ";}color:", ";", "}&[disabled]{border-color:", ";background-color:", ";", ";}" ], k.mixins.reset("inline"), C, C, C, k.variables.borderRadius, k.variables.inputBorderWidth, k.variables.interactiveColorBorder, k.variables.interactiveColorBackground, k.variables.focusShadow, (0, 
    k.pick)({
        light: k.variables.white,
        dark: k.variables.black
    }), (function(e) {
        var r = e.$error, a = e.$selected;
        if (a && !r) {
            return (0, s.css)([ "background-color:", ";border:none;" ], k.variables.interactiveColorAccent);
        }
        if (!a && r) {
            return (0, s.css)([ "border-color:", ";&:hover{border-color:", ";background-color:", ";}" ], k.variables.interactiveColorAccentError, k.mixins.overlayColors(k.variables.interactiveColorAccentError, k.variables.interactiveColorBorderHover), k.mixins.overlayColors(k.variables.interactiveColorBackground, k.variables.interactiveColorOverlayHover));
        }
        if (a && r) {
            return (0, s.css)([ "background-color:", ";border:none;" ], k.variables.interactiveColorAccentError);
        }
        return (0, s.css)([ "&:hover{border-color:", ";background-color:", ";}" ], k.variables.interactiveColorBorderHover, k.mixins.overlayColors(k.variables.interactiveColorBackground, k.variables.interactiveColorOverlayHover));
    }), k.variables.interactiveColorBorderDisabled, k.variables.interactiveColorBackgroundDisabled, (function(e) {
        var r = e.$selected;
        return r && (0, s.css)([ "border:none;background-color:", ";" ], k.variables.neutral100);
    }));
    var O = u()(y()).withConfig({
        displayName: "SwitchStyles__StyledCheckboxClickable",
        componentId: "sc-844ieu-1"
    })([ "", "" ], B);
    var $ = (0, s.css)([ "display:block;height:", ";width:", ";" ], C, C);
    var I = u()(v()).withConfig({
        displayName: "SwitchStyles__StyledCheck",
        componentId: "sc-844ieu-2"
    })([ "", "" ], $);
    var N = u()(p()).withConfig({
        displayName: "SwitchStyles__StyledSome",
        componentId: "sc-844ieu-3"
    })([ "", "" ], $);
    var q = u().div.withConfig({
        displayName: "SwitchStyles__StyledToggleIndicator",
        componentId: "sc-844ieu-4"
    })([ "--toggle-indicator-margin:calc( ", " - ", " );background-color:", ";border-radius:50%;box-sizing:border-box;width:", ";height:", ";position:absolute;left:var(--toggle-indicator-margin);top:var(--toggle-indicator-margin);", ";", " ", " ", ";" ], w, k.variables.inputBorderWidth, (0, 
    k.pick)({
        light: k.variables.black,
        dark: k.variables.white
    }), S, S, (function(e) {
        var r = e.$delay;
        return r && (0, s.css)([ "transition:left ", ";" ], r);
    }), (0, k.pick)({
        prisma: (0, s.css)([ "box-shadow:", ";" ], k.variables.embossShadow)
    }), (function(e) {
        var r = e.$selected, a = e.$disabled;
        return (0, s.css)([ "", " ", "" ], r && (0, s.css)([ "left:calc(100% - (", " + var(--toggle-indicator-margin)));" ], S), !a && r && (0, 
        s.css)([ "background-color:", ";" ], (0, k.pick)({
            light: k.variables.white,
            dark: k.variables.black
        })));
    }), (function(e) {
        var r = e.$disabled;
        return r && (0, s.css)([ "background-color:", ";" ], k.variables.neutral400);
    }));
    var E = (0, s.css)([ "position:relative;width:calc(", " * 2);height:", ";background-color:", ";transition:background-color ", ";flex:0 0 auto;border-radius:calc(", " / 2);border:", " solid ", ";&:not([disabled]){&:focus{box-shadow:", ";}", "}&[disabled]{border:", " solid ", ";background-color:", ";", "}" ], C, C, k.variables.interactiveColorBackground, (function(e) {
        return e.$delay;
    }), C, k.variables.inputBorderWidth, k.variables.interactiveColorBorder, k.variables.focusShadow, (function(e) {
        var r = e.$selected;
        if (!r) {
            return (0, s.css)([ "&:hover{border-color:", ";background-color:", ";}" ], k.variables.interactiveColorBorderHover, k.mixins.overlayColors(k.variables.interactiveColorBackground, k.variables.interactiveColorOverlayHover));
        }
        return (0, s.css)([ "border-color:transparent;background-color:", ";" ], k.variables.interactiveColorAccent);
    }), k.variables.inputBorderWidth, k.variables.interactiveColorBorderDisabled, k.variables.interactiveColorBackgroundDisabled, (function(e) {
        var r = e.$selected;
        return r && (0, s.css)([ "background-color:", ";" ], k.variables.neutral100);
    }));
    var _ = u()(y()).withConfig({
        displayName: "SwitchStyles__StyledToggleClickable",
        componentId: "sc-844ieu-5"
    })([ "", "" ], E);
    var L = u().label.withConfig({
        displayName: "SwitchStyles__StyledLabel",
        componentId: "sc-844ieu-6"
    })([ "", " flex:1 1 auto;color:inherit;line-height:", ";", ";" ], k.mixins.reset("inline-block"), k.variables.lineHeight, (function(e) {
        var r = e.$disabled;
        return !r && (0, s.css)([ "cursor:pointer;" ]);
    }));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function R(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var T = "is deprecated and will be removed in the next major version.";
    var j = function e(r) {
        var a = r.additionalMessage, i = a === void 0 ? "" : a, o = r.componentName;
        useEffect((function() {
            if (false) {}
        }), [ i, o ]);
    };
    var P = function e(r) {
        var i = r.additionalMessage, o = i === void 0 ? "" : i, t = r.componentName, l = r.propName, n = r.propValue;
        (0, a.useEffect)((function() {
            if (false) {}
        }), [ o, t, l, n ]);
    };
    var D = function e(r) {
        var i = r.additionalMessage, o = i === void 0 ? "" : i, t = r.componentName, l = r.deprecatedPropValue, n = r.propName, c = r.propValue;
        (0, a.useEffect)((function() {
            if (false) {}
        }), [ o, t, n, c, l ]);
    };
    // CONCATENATED MODULE: ./src/Switch/Switch.tsx
    function A() {
        return A = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var a = arguments[r];
                for (var i in a) {
                    ({}).hasOwnProperty.call(a, i) && (e[i] = a[i]);
                }
            }
            return e;
        }, A.apply(null, arguments);
    }
    function H(e, r) {
        if (null == e) return {};
        var a, i, o = M(e, r);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            for (i = 0; i < t.length; i++) {
                a = t[i], -1 === r.indexOf(a) && {}.propertyIsEnumerable.call(e, a) && (o[a] = e[a]);
            }
        }
        return o;
    }
    function M(e, r) {
        if (null == e) return {};
        var a = {};
        for (var i in e) {
            if ({}.hasOwnProperty.call(e, i)) {
                if (-1 !== r.indexOf(i)) continue;
                a[i] = e[i];
            }
        }
        return a;
    }
    /** @public */
    /** @public */    var V = {
        appearance: t().oneOf([ "checkbox", "toggle" ]),
        children: t().node,
        describedBy: t().string,
        disabled: t().bool,
        elementRef: t().oneOfType([ t().func, t().object ]),
        error: t().bool,
        id: t().string,
        inline: t().bool,
        labelledBy: t().string,
        onClick: t().func,
        /** @private. */
        required: t().bool,
        selected: t().oneOf([ true, false, "some" ]),
        selectedLabel: t().string,
        toggleRef: t().oneOfType([ t().func, t().object ]),
        someSelectedLabel: t().string,
        unselectedLabel: t().string,
        value: t().any
    };
    // eslint-disable-next-line docs-props-type-match
    // eslint-disable-next-line docs-props-type-match
        var W = function e(r) {
        var a = r.ariaChecked, o = r.ariaInvalid, t = r.ariaLabelledBy, n = r.ariaRequired, c = r.dataTest, d = r.dataSelected, s = r.describedBy, u = r.disabled, b = r.elementRef, v = r.id, f = r.onClick, p = r.selected;
        var g = u === true ? "disabled" : undefined;
        var m = (0, l.useAnimationToggle)();
        var h = m === "on" ? "200ms" : undefined;
        var y =  i().createElement(q, {
            $delay: h,
            $disabled: !!g,
            $selected: p
        });
        
        return i().createElement(_, {
            role: "switch",
            "aria-checked": a,
            "aria-describedby": s,
            "aria-invalid": o,
            "aria-labelledby": t,
            "aria-required": n,
            "data-selected": d,
            "data-test": c,
            disabled: g,
            elementRef: b,
            id: v,
            onClick: f,
            $delay: h,
            $selected: p || false
        }, y);
    };
    var F = function e(r) {
        var a = r.ariaChecked, o = r.ariaInvalid, t = r.ariaLabelledBy, l = r.ariaRequired, n = r.dataTest, c = r.dataSelected, d = r.describedBy, s = r.disabled, u = r.error, b = r.id, v = r.elementRef, f = r.onClick, p = r.role, g = r.selected, m = r.tabIndex;
        /* Setting viewbox to rescale the icons to appear larger. Cannot directly adjust svg
   * width and height without the svg being larger than the checkbox and box-model.
   */        var h = "4 4 16 16";
        var y =  i().createElement(i().Fragment, null, g === true &&  i().createElement(I, {
            viewBox: h
        }), g === "some" &&  i().createElement(N, {
            viewBox: h
        }));
        var k = s === true ? "disabled" : undefined;
        
        return i().createElement(O, {
            "aria-checked": a,
            "aria-describedby": d,
            "aria-invalid": o,
            "aria-labelledby": t,
            "aria-required": l,
            "data-selected": c,
            "data-test": n,
            disabled: k,
            elementRef: v,
            id: b,
            onClick: f,
            role: p || "checkbox",
            tabIndex: m,
            $error: u || false,
            $selected: g === "some" || g || false
        }, y);
    };
    /**
 * `Switch` is a basic form control with an on/off state.
 */    function z(e) {
        var r = e.appearance, o = r === void 0 ? "checkbox" : r, t = e.children, l = e.describedBy, n = e.disabled, s = e.elementRef, u = e.error, b = e.id, v = e.inline, f = e.labelledBy, p = e.onClick, g = e.required, m = e.selected, h = e.selectedLabel, y = e.someSelectedLabel, k = e.role, C = e.tabIndex, S = e.toggleRef, w = e.unselectedLabel, B = e.value, O = H(e, [ "appearance", "children", "describedBy", "disabled", "elementRef", "error", "id", "inline", "labelledBy", "onClick", "required", "selected", "selectedLabel", "someSelectedLabel", "role", "tabIndex", "toggleRef", "unselectedLabel", "value" ]);
        // @docs-props-type SwitchPropsBase
                D({
            componentName: "Switch",
            deprecatedPropValue: "checkbox",
            propName: "appearance",
            propValue: o
        });
        P({
            componentName: "Switch",
            propName: "selectedLabel",
            propValue: h
        });
        P({
            componentName: "Switch",
            propName: "someSelectedLabel",
            propValue: y
        });
        P({
            componentName: "Switch",
            propName: "unselectedLabel",
            propValue: w
        });
        /* If has internal label defined with children, must have an id and must use generated labelId.
   * Otherwise, use customer defined id and external label.
   */        var $ = t && b == null ? (0, d.createDOMID)("clickable") : b;
        var I = t ? (0, d.createDOMID)("label") : f;
        (0, a.useEffect)((function() {
            if (false) {}
            if (false) {}
        }), [ m, o, u ]);
        var N = (0, a.useCallback)((function(e) {
            p === null || p === void 0 ? void 0 : p(e, {
                value: B,
                // @ts-expect-error - safe due to individual prop interfaces
                selected: m
            });
        }), [ p, B, m ]);
        var q = (0, a.useCallback)((function(e) {
            R(S, e);
        }), [ S ]);
        var E = m === "some" ? "mixed" : m;
        var _ = {
            true: h,
            false: w,
            some: y
        };
        var T = {
            id: $,
            onClick: n ? undefined : N,
            elementRef: q,
            ariaLabelledBy: I,
            ariaChecked: E,
            ariaInvalid: u ? true : undefined,
            ariaRequired: g,
            dataTest: "toggle",
            dataSelected: m,
            describedBy: l,
            role: k,
            tabIndex: C
        };
        var j = !!h || !!y || !!w;
        
        return i().createElement(x, A({
            flex: true,
            inline: v,
            "data-test": "switch",
            "data-test-selected": m,
            "data-test-value": B,
            "data-test-error": u ? true : undefined,
            "data-error": u ? true : undefined,
            "data-test-disabled": n ? "disabled" : undefined,
            ref: s,
            $disabled: n || false
        }, O), o === "toggle" ?  i().createElement(W, A({
            disabled: n || false,
            selected: !!m
        }, T)) :  i().createElement(F, A({
            error: u || false,
            disabled: n || false,
            selected: m
        }, T)), j &&  i().createElement(c(), null, _["".concat(m)]), t &&  i().createElement(L, {
            $disabled: n || false,
            "data-test": "label",
            id: I,
            htmlFor: $,
            "data-disabled": n || null
        }, t));
    }
    z.propTypes = V;
    /* harmony default export */    const G = z;
    // CONCATENATED MODULE: ./src/Switch/index.ts
    module.exports = r;
    /******/})();
