// @splunk/react-ui/TabBar.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Tab: () => /* reexport */ ae,
        default: () => /* reexport */ Ee
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const o = require("@splunk/react-ui/usePrevious");
    var l =  e.n(o);
    // CONCATENATED MODULE: external "lodash/omit"
    const u = require("lodash/omit");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-icons/DotsThreeVertical"
    const s = require("@splunk/react-icons/DotsThreeVertical");
    var d =  e.n(s);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const f = require("@splunk/react-ui/Dropdown");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const p = require("@splunk/react-ui/Popover");
    var b =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const m = require("@splunk/react-ui/ScreenReaderContent");
    var g =  e.n(m);
    // CONCATENATED MODULE: external "@splunk/themes"
    const y = require("@splunk/themes");
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const h = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const S = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const w = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: ./src/TabBar/TabBarContext.tsx
    var O = function e() {
        return undefined;
    };
    var C =  (0, n.createContext)({
        isMenuOpen: false,
        onMenuOpen: O,
        onMenuClose: O,
        registerTab: O
    });
    C.displayName = "TabBar";
    /* harmony default export */    const x = C;
    // CONCATENATED MODULE: external "styled-components"
    const I = require("styled-components");
    var k =  e.n(I);
    // CONCATENATED MODULE: external "@splunk/react-ui/Badge"
    const T = require("@splunk/react-ui/Badge");
    var j =  e.n(T);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const P = require("@splunk/react-ui/Clickable");
    var E =  e.n(P);
    // CONCATENATED MODULE: ./src/TabBar/TabStyles.ts
    var M = (0, I.css)([ "color:", ";" ], y.variables.contentColorActive);
    var $ = "2px";
    var A = (0, y.pickVariant)("$layout", {
        horizontal: (0, I.css)([ "background:", ";height:", ";" ], y.variables.contentColorAccentWeak, $),
        vertical: (0, I.css)([ "background:", ";width:", ";" ], y.variables.contentColorAccentWeak, $)
    });
    var _ = function e(t) {
        var n = t.$hasIcon, r = t.$hasMenu, a = t.$iconPosition;
        if (a === "above" && n && r) {
            return (0, I.css)([ "grid-template-areas:'icon icon icon' 'menu-counter-neg-space content menu' 'gutter gutter gutter' 'underline underline underline';grid-template-columns:", " 1fr min-content;grid-template-rows:repeat(2,auto) ", " auto;" ], y.variables.spacingXSmall, y.variables.spacingXSmall);
        }
        if (a === "above" && n) {
            return (0, I.css)([ "grid-template-areas:'icon' 'content' 'gutter' 'underline';grid-template-columns:1fr;grid-template-rows:repeat(2,auto) ", " auto;" ], y.variables.spacingXSmall);
        }
        if (a === "left" && n && r) {
            return (0, I.css)([ "grid-template-areas:'menu-counter-neg-space icon content menu' 'underline underline underline underline';grid-template-columns:", " min-content 1fr min-content;" ], y.variables.spacingXSmall);
        }
        if (a === "left" && n) {
            return (0, I.css)([ "grid-template-areas:'icon content' 'underline underline';grid-template-columns:min-content 1fr;" ]);
        }
        if (r) {
            return (0, I.css)([ "grid-template-areas:'menu-counter-neg-space content menu' 'underline underline underline';grid-template-columns:", " minmax(auto,min-content) min-content;" ], y.variables.spacingXSmall);
        }
        return (0, I.css)([ "grid-template-areas:'content' 'underline';grid-template-columns:minmax(auto,min-content);" ]);
    };
    var z = function e(t) {
        var n = t.$hasIcon, r = t.$iconPosition;
        if (r === "above" && n) {
            return (0, I.css)([ "grid-template-areas:'icon gutter underline' 'content gutter underline';grid-template-columns:1fr ", " ", ";" ], y.variables.spacingXSmall, $);
        }
        if (r === "left" && n) {
            return (0, I.css)([ "grid-template-areas:'icon content gutter underline';grid-template-columns:min-content 1fr ", " ", ";" ], y.variables.spacingXSmall, $);
        }
        return (0, I.css)([ "grid-template-areas:'content gutter underline';grid-template-columns:1fr ", " ", ";" ], y.variables.spacingXSmall, $);
    };
    var D = k()(E()).withConfig({
        displayName: "TabStyles__StyledClickable",
        componentId: "sc-1ry8mzj-0"
    })([ "color:", ";column-gap:", ";display:grid;justify-content:center;align-items:baseline;min-height:", ";position:relative;", " ", " ", ";&[aria-selected='true']{cursor:default;color:", ";font-weight:", ";}&[aria-selected='false']{box-shadow:none;font-weight:", ";&:hover:not([disabled]){", "}&[aria-haspopup='true']{", "}}&:focus{box-shadow:", ";color:", ";}" ], y.variables.contentColorDefault, y.variables.spacingXSmall, y.variables.inputHeight, (function(e) {
        var t = e.$maxTabWidth;
        return t && (0, I.css)([ "max-width:", "px;" ], t);
    }), (function(e) {
        var t = e.$hasIcon, n = e.$hasMenu, r = e.$iconPosition, a = e.$layout;
        if (a === "vertical") {
            return z({
                $hasIcon: t,
                $iconPosition: r
            });
        }
        return _({
            $hasIcon: t,
            $hasMenu: n,
            $iconPosition: r
        });
    }), (0, y.pickVariant)("$layout", {
        horizontal: (0, I.css)([ "margin-bottom:1px;" ]),
        vertical: (0, I.css)([ "align-items:center;right:1px;" ])
    }), y.variables.contentColorActive, y.variables.fontWeightBold, y.variables.fontWeightSemiBold, M, M, y.variables.focusShadow, y.variables.contentColorActive);
    var N = k().div.withConfig({
        displayName: "TabStyles__StyledUnderline",
        componentId: "sc-1ry8mzj-1"
    })([ "grid-area:underline;position:absolute;", ";[aria-selected='true'] > &&{background-color:", ";", ";}", ":hover:not([disabled]):not([aria-selected='true']) > &&{", ";}[aria-selected='false'][aria-haspopup='true'] > &&{", "}" ], (0, 
    y.pickVariant)("$layout", {
        horizontal: (0, I.css)([ "bottom:-1px;height:0;width:100%;transition:height 0.2s;" ]),
        vertical: (0, I.css)([ "width:1px;height:100%;right:-1px;transition:width 0.2s;" ])
    }), y.variables.contentColorAccent, (0, y.pickVariant)("$layout", {
        horizontal: (0, I.css)([ "height:", ";" ], $),
        vertical: (0, I.css)([ "width:", ";" ], $)
    }), /* sc-sel */ D, A, A);
    var R = k().span.withConfig({
        displayName: "TabStyles__StyledMenu",
        componentId: "sc-1ry8mzj-2"
    })([ "display:inline-block;grid-area:menu;margin-left:4px;> svg{transform:translateY(-1px);}border-radius:", ";cursor:pointer;", " &:hover{background-color:", ";border-color:", ";}", "[aria-haspopup='true'] &,", ":hover &,", ":focus &{visibility:visible;}" ], y.variables.borderRadius, (0, 
    y.pickVariant)("$active", {
        false: (0, I.css)([ "visibility:hidden;" ])
    }), y.variables.interactiveColorOverlayHover, y.variables.interactiveColorBorderHover, /* sc-sel */ D, /* sc-sel */ D, /* sc-sel */ D);
    var q = k().span.withConfig({
        displayName: "TabStyles__StyledIcon",
        componentId: "sc-1ry8mzj-3"
    })([ "grid-area:icon;", "" ], (0, y.pickVariant)("$iconPosition", {
        above: (0, I.css)([ "display:flex;justify-content:center;" ]),
        left: (0, I.css)([ "> svg{transform:translateY(-1px);}" ])
    }));
    var V = k().div.withConfig({
        displayName: "TabStyles__StyledContent",
        componentId: "sc-1ry8mzj-4"
    })([ "display:inline-flex;align-items:baseline;grid-area:content;min-width:10px;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;", "" ], (function(e) {
        var t = e.$iconPosition;
        return t === "above" && (0, I.css)([ "justify-content:center;text-align:center;" ]);
    }));
    var B = k().div.withConfig({
        displayName: "TabStyles__StyledLabel",
        componentId: "sc-1ry8mzj-5"
    })([ "&::after{display:block;content:attr(data-title);font-weight:", ";height:0;color:transparent;overflow:hidden;visibility:hidden;white-space:nowrap;}" ], y.variables.fontWeightBold);
    var W = k()(j()).withConfig({
        displayName: "TabStyles__StyledBadge",
        componentId: "sc-1ry8mzj-6"
    })([ "line-height:10px;margin-inline-start:0.3em;padding:0.4em 0.6em;" ]);
    var X = k().div.withConfig({
        displayName: "TabStyles__StyledTooltipContent",
        componentId: "sc-1ry8mzj-7"
    })([ "padding:8px;font-size:", ";" ], y.variables.fontSizeSmall);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function L(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/TabBar/Tab.tsx
    function F() {
        return F = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, F.apply(null, arguments);
    }
    function H(e, t) {
        return J(e) || G(e, t) || K(e, t) || U();
    }
    function U() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function K(e, t) {
        if (e) {
            if ("string" == typeof e) return Y(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? Y(e, t) : void 0;
        }
    }
    function Y(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function G(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, a, i, o, l = [], u = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = i.call(n)).done) && (l.push(r.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (o = n["return"](), Object(o) !== o)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return l;
        }
    }
    function J(e) {
        if (Array.isArray(e)) return e;
    }
    function Q(e, t) {
        if (null == e) return {};
        var n, r, a = Z(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function Z(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var ee = {
        ariaControls: i().string,
        count: i().number,
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        icon: i().node,
        id: i().string,
        label: i().oneOfType([ i().string, i().element ]),
        /** @private Maximum count number in tab label */
        maxCount: i().number,
        tabId: i().string,
        to: i().string,
        tooltip: i().node,
        /**
   * @private
   * Accepts a Menu component that is rendered for that Tab.
   * Is not compatible with `count`.
   * The `menu` prop is only supported in horizontal layout due to both design constraints as well as keyboard support.
   */
        menu: i().node,
        /**
   * @private
   * Internal prop injected by TabBar to track order.
   */
        index: i().number
    };
    var te = {
        tension: 400
    };
    var ne = [ "offScreen", "escapeKey" ];
    var re = function e(t) {
        var a = t.ariaControls, i = t.count, o = t.disabled, l = t.elementRef, u = t.icon, s = t.label, f = t.maxCount, p = t.tabId, m = t.to, O = t.tooltip, C = t.menu, I = t.index, k = Q(t, [ "ariaControls", "count", "disabled", "elementRef", "icon", "label", "maxCount", "tabId", "to", "tooltip", "menu", "index" ]);
        // @docs-props-type TabPropsBase
                var T = (0, n.useState)(false), j = H(T, 2), P = j[0], E = j[1];
        var M = (0, n.useState)(null), $ = H(M, 2), A = $[0], _ = $[1];
        var z = (0, n.useContext)(x), U = z.activeTabId, K = z.disabled, Y = z.focusedTabId, G = z.iconPosition, J = G === void 0 ? "above" : G, Z = z.isMenuOpen, ee = z.layout, re = ee === void 0 ? "horizontal" : ee, ae = z.maxTabWidth, ie = z.onClick, oe = z.onFocus, le = z.onMenuOpen, ue = z.onMenuClose, ce = z.registerTab;
        var se = (0, n.useState)((0, w.createDOMID)("aria-id")), de = H(se, 1), fe = de[0];
        var ve = (0, n.useState)((0, w.createDOMID)("menu-description")), pe = H(ve, 1), be = pe[0];
        var me = (0, n.useState)((0, w.createDOMID)("popover")), ge = H(me, 1), ye = ge[0];
        var he = (0, n.useMemo)((function() {
            return p || (0, w.createGUID)();
        }), [ p ]);
        var Se = K || o ? "disabled" : false;
        var we = s ? s.toString() : "";
        var Oe = U === he;
        // title is used for CSS styling in TabStyles. labelValue and countValue needs to be included in the title.
                var Ce = (0, n.useMemo)((function() {
            // SUI-6877: if the label isn't a string, don't attempt to do this since it'll just be wrong
            return s && typeof s === "string" ? we : "";
        }), [ s, we ]);
        var xe = (0, n.useCallback)((function() {
            A === null || A === void 0 ? void 0 : A.focus();
        }), [ A ]);
        var Ie = (0, n.useCallback)((function(e) {
            _(e);
            L(l, e);
            // Register with index, assert index is defined as it's injected by TabBar
                        ce({
                tabId: he,
                element: e,
                index: I
            });
        }), [ l, ce, he, I ]);
        var ke = (0, n.useCallback)((function() {
            E(true);
        }), []);
        var Te = (0, n.useCallback)((function() {
            E(false);
        }), []);
        var je = (0, n.useCallback)((function(e) {
            // preserve the previous behaviour from 2ad8e7eaf47 to avoid firing unnecessary onChange events
            // while preventing the "to" prop from incorrectly triggering <Clickable>'s providedOnClick on the active tab
            if (!Oe) {
                ie === null || ie === void 0 ? void 0 : ie(e, {
                    tabId: he
                });
            }
            if (m == null || m != null && Oe) {
                e.preventDefault();
            }
        }), [ Oe, ie, he, m ]);
        var Pe = (0, n.useCallback)((function(e) {
            ke();
            oe === null || oe === void 0 ? void 0 : oe(e, {
                tabId: he
            });
        }), [ ke, oe, he ]);
        var Ee = (0, n.useCallback)((function() {
            xe();
        }), [ xe ]);
        var Me = (0, n.useCallback)((function() {
            Te();
        }), [ Te ]);
        var $e = (0, n.useCallback)((function(e) {
            // prevent clicking on the menu icon from changing the active tab
            e.stopPropagation();
            le(he);
        }), [ le, he ]);
        var Ae = Y === he && Z && !!C;
        var _e =  r().createElement(R, {
            $active: Oe,
            "data-test": "menu-toggle",
            tabIndex: -1,
            onFocus: Ee
        },  r().createElement(d(), null));
        var ze = function e(t) {
            
            return r().createElement(r().Fragment, null, t,  r().createElement(v(), {
                toggle: _e,
                align: "center",
                onRequestOpen: $e,
                onRequestClose: ue,
                open: Ae
            }, C));
        };
        var De = k["data-title"];
        var Ne =  r().createElement(V, {
            "data-test": "label",
            $iconPosition: J
        },  r().createElement(B, {
            "data-title": De || Ce
        }, s), !C && (i === 0 || i) &&  r().createElement(W, {
            "data-test": "count",
            foregroundColor: Se ? y.variables.contentColorDisabled : undefined,
            label: f ? (0, h.truncateNumber)({
                number: i,
                max: f
            }) : "".concat(i)
        }), !!C &&  r().createElement(g(), {
            "aria-hidden": true,
            id: be
        }, (0, S._)("Press Shift + F10 to open the tab’s menu")));
        
        // TODO: Fix the ts error: https://splunk.atlassian.net/browse/SUI-5569
        // eslint-disable-next-line
        // @ts-ignore-next-line
        return r().createElement(D, F({
            $hasIcon: !!u,
            $hasMenu: !!C,
            $iconPosition: J,
            $layout: re,
            $maxTabWidth: ae,
            "aria-controls": a,
            "aria-selected": Oe,
            "aria-labelledby": fe,
            "aria-describedby": C ? be : undefined,
            "aria-haspopup": Ae,
            "data-test": "tab",
            "data-test-tab-id": he,
            "data-test-popover-id": O ? ye : undefined,
            "data-test-disabled": Se || undefined,
            disabled: Se,
            elementRef: Ie,
            onClick: je,
            onFocus: Pe,
            onMouseEnter: ke,
            onBlur: Me,
            onMouseLeave: Te,
            role: "tab",
            tabIndex: Oe ? undefined : -1,
            to: m
        }, c()(k, "elementRef")), u &&  r().createElement(q, {
            $iconPosition: J
        }, u), C && re === "horizontal" ? ze(Ne) : Ne,  r().createElement(N, {
            $layout: re
        }), !o && O &&  r().createElement(b(), {
            role: "tooltip",
            anchor: A,
            animationConfig: te,
            closeReasons: ne,
            defaultPlacement: re === "vertical" ? "right" : "above",
            id: ye,
            open: !!A && P,
            align: "center"
        },  r().createElement(X, null, O)), O &&  r().createElement(g(), {
            "aria-hidden": "true",
            id: fe
        }, O));
    };
    re.propTypes = ee;
    re.as = "Tab";
    /* harmony default export */    const ae = re;
    // CONCATENATED MODULE: ./src/TabBar/TabBarStyles.ts
    var ie = k().div.withConfig({
        displayName: "TabBarStyles__Styled",
        componentId: "sc-1t85fen-0"
    })([ "", " position:relative;&::before{content:'';display:block;position:absolute;left:0;top:0;right:0;bottom:0;border:0 solid ", ";}", ";" ], y.mixins.reset("flex"), y.variables.neutral300, (0, 
    y.pickVariant)("$layout", {
        horizontal: (0, I.css)([ "column-gap:", ";&::before{border-bottom-width:1px;}" ], y.variables.spacingXLarge),
        vertical: (0, I.css)([ "display:inline-flex;flex-direction:column;row-gap:", ";&::before{border-right-width:1px;}" ], y.variables.spacingXLarge)
    }));
    // CONCATENATED MODULE: ./src/TabBar/tabBarUtils.tsx
    // A utility for keyboard navigation for TabBar
    function oe(e, t, n) {
        for (var r = 0; r < e.length; r += 1) {
            var a = (r + n) % e.length;
            var i = e[a].element;
            if (i instanceof HTMLButtonElement && i.disabled !== true) {
                return e[a];
            }
        }
        return e[t];
    }
    function le(e, t, n) {
        for (var r = e.length; r > 0; r -= 1) {
            var a = (r + n) % e.length;
            var i = e[a].element;
            if (i instanceof HTMLButtonElement && i.disabled !== true) {
                return e[a];
            }
        }
        return e[t];
    }
    // CONCATENATED MODULE: ./src/utils/useDeprecate.tsx
    var ue = "is deprecated and will be removed in the next major version.";
    var ce = function e(t) {
        var n = t.additionalMessage, r = n === void 0 ? "" : n, a = t.componentName;
        useEffect((function() {
            if (false) {}
        }), [ r, a ]);
    };
    var se = function e(t) {
        var r = t.additionalMessage, a = r === void 0 ? "" : r, i = t.componentName, o = t.propName, l = t.propValue;
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ a, i, o, l ]);
    };
    var de = function e(t) {
        var n = t.additionalMessage, r = n === void 0 ? "" : n, a = t.componentName, i = t.deprecatedPropValue, o = t.propName, l = t.propValue;
        useEffect((function() {
            if (false) {}
        }), [ r, a, o, l, i ]);
    };
    // CONCATENATED MODULE: ./src/TabBar/TabBar.tsx
    function fe(e) {
        "@babel/helpers - typeof";
        return fe = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, fe(e);
    }
    function ve() {
        return ve = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, ve.apply(null, arguments);
    }
    function pe(e, t) {
        if (null == e) return {};
        var n, r, a = be(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function be(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    function me(e, t) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            t && (r = r.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), n.push.apply(n, r);
        }
        return n;
    }
    function ge(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = null != arguments[t] ? arguments[t] : {};
            t % 2 ? me(Object(n), !0).forEach((function(t) {
                ye(e, t, n[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : me(Object(n)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(n, t));
            }));
        }
        return e;
    }
    function ye(e, t, n) {
        return (t = he(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function he(e) {
        var t = Se(e, "string");
        return "symbol" == fe(t) ? t : t + "";
    }
    function Se(e, t) {
        if ("object" != fe(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != fe(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function we(e, t) {
        return ke(e) || Ie(e, t) || Ce(e, t) || Oe();
    }
    function Oe() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Ce(e, t) {
        if (e) {
            if ("string" == typeof e) return xe(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? xe(e, t) : void 0;
        }
    }
    function xe(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function Ie(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, a, i, o, l = [], u = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = i.call(n)).done) && (l.push(r.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (o = n["return"](), Object(o) !== o)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return l;
        }
    }
    function ke(e) {
        if (Array.isArray(e)) return e;
    }
    /** @public */
    /** @public */    var Te = {
        autoActivate: i().bool,
        activeTabId: i().string,
        children: i().node,
        /** @private */
        disabled: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ]),
        iconPosition: i().oneOf([ "above", "left" ]),
        layout: i().oneOf([ "horizontal", "vertical" ]),
        maxTabWidth: i().number,
        onChange: i().func
    };
    var je = function e(t) {
        return Array.from(t.entries()).sort((function(e, t) {
            var n = we(e, 2), r = n[1].index;
            var a = we(t, 2), i = a[1].index;
            return r - i;
        })).map((function(e) {
            var t = we(e, 2), n = t[0], r = t[1];
            return ge({
                tabId: n
            }, r);
        }));
    };
    function Pe(e) {
        var t;
        var a = e.activeTabId, i = e.autoActivate, o = e.children, u = e.disabled, c = u === void 0 ? false : u, s = e.elementRef, d = e.iconPosition, f = d === void 0 ? "left" : d, v = e.layout, p = v === void 0 ? "horizontal" : v, b = e.maxTabWidth, m = e.onChange, g = pe(e, [ "activeTabId", "autoActivate", "children", "disabled", "elementRef", "iconPosition", "layout", "maxTabWidth", "onChange" ]);
        // @docs-props-type TabBarPropsBase
        // eslint-disable-next-line prefer-rest-params
                var y = (t = arguments[0]) === null || t === void 0 ? void 0 : t.iconPosition;
        se({
            componentName: "TabBar and TabLayout",
            propName: "iconPosition",
            propValue: y
        });
        var h = (0, n.useState)(false), S = we(h, 2), w = S[0], O = S[1];
        var C = (0, n.useState)(a), I = we(C, 2), k = I[0], T = I[1];
 // need to keep track of the focused tabId in a ref to avoid re-generating registerTab
        // because every time `registerTab` changes, it calls `ref` on all tabs
                var j = (0, n.useRef)(a);
        j.current = k;
        // If the tab that was focused is removed, we need to keep track of what index it was last at
                var P = (0, n.useRef)();
        var E = (0, n.useRef)(null);
        // map of tabId to tab element and index
                var M = (0, n.useRef)();
        if (!M.current) {
            M.current = new Map;
        }
        var $ = p === "vertical" ? "ArrowDown" : "ArrowRight";
        var A = p === "vertical" ? "ArrowUp" : "ArrowLeft";
        // focus the tab belonging to the given tabId
                var _ = (0, n.useCallback)((function(e) {
            var t;
            if (!e) {
                return;
            }
            var n = (t = M.current) === null || t === void 0 ? void 0 : t.get(e);
            if (!n) {
                var r = je(M.current);
                var a = r[0];
                if (a) {
                    var i;
                    T(e);
                    (i = a.element) === null || i === void 0 ? void 0 : i.focus();
                }
                return;
            }
            T(e);
            n.element.focus();
        }), []);
        var z = (0, n.useCallback)((function(e) {
            var t = e.tabId, n = e.element, r = e.index;
            if (n) {
                var a;
                (a = M.current) === null || a === void 0 ? void 0 : a.set(t, {
                    element: n,
                    index: r
                });
            } else {
                var i;
                (i = M.current) === null || i === void 0 ? void 0 : i["delete"](t);
                // if the tab being removed is the focused tab, keep track of its last known index
                                if (j.current === t) {
                    P.current = r;
                }
            }
        }), []);
        var D = (0, n.useCallback)((function(e) {
            E.current = e;
            L(s, e);
        }), [ s, E ]);
        // callback to handle focus event sent from a Tab
                var N = (0, n.useCallback)((function(e, t) {
            var n = t.tabId;
            if (document.activeElement === e.target && k === n) {
                // tab already focused, no need to continue
                return;
            }
            _(n);
            if (i && a !== n) {
                m === null || m === void 0 ? void 0 : m(e, {
                    selectedTabId: n
                });
            }
        }), [ a, i, k, _, m ]);
        var R = (0, n.useCallback)((function(e, t) {
            var n = t.tabId;
            _(n);
            m === null || m === void 0 ? void 0 : m(e, {
                selectedTabId: n
            });
        }), [ _, m ]);
        var q = (0, n.useCallback)((function(e) {
            // if command key on Mac was pressed ignore
            if (e.nativeEvent.metaKey) {
                return;
            }
            var t = je(M.current);
            var n = k ? t.findIndex((function(e) {
                return e.tabId === k;
            })) : -1;
            var r = t.map((function(e) {
                return e.element;
            }));
            var a = n === -1 ? 0 : n;
            var i = e.key;
            var o;
            if (i === $) {
                o = oe(t, a, a + 1);
            } else if (i === A) {
                o = le(t, a, a - 1);
            } else if (i === "Home") {
                o = oe(t, a, 0);
            } else if (i === "End") {
                o = le(t, a, r.length - 1);
            } else if (p === "horizontal" && e.shiftKey && i === "F10") {
                O((function(e) {
                    if (e) {
                        var t = r[a];
                        t === null || t === void 0 ? void 0 : t.focus();
                    }
                    return !e;
                }));
            } else if (p === "horizontal" && i === "ArrowDown") {
                O(true);
            }
            if (o) {
                _(o.tabId);
                e.preventDefault();
            }
        }), [ k, _, p, $, A ]);
        var V = (0, n.useCallback)((function(e) {
            _(e);
 // in case they clicked on a menu of tab they were not even focused on yet
                        O(true);
        }), [ _ ]);
        var B = (0, n.useCallback)((function(e) {
            var t = e.event;
            // prevent content clicks/enter key from changing the active tab
                        t === null || t === void 0 ? void 0 : t.stopPropagation();
            O(false);
        }), []);
        var W = (0, n.useMemo)((function() {
            return {
                disabled: c,
                focusedTabId: k,
                iconPosition: f,
                layout: p,
                maxTabWidth: b,
                onClick: R,
                onFocus: N,
                onMenuOpen: V,
                onMenuClose: B,
                isMenuOpen: w,
                registerTab: z,
                activeTabId: a
            };
        }), [ c, k, f, p, R, N, b, V, B, w, z, a ]);
        var X = (0, n.useMemo)((function() {
            return n.Children.toArray(o).filter(n.isValidElement).map((function(e, t) {
                if (e.type.as === "Tab") {
                    
                    return (0, n.cloneElement)(e, {
                        index: t
                    });
                }
                return e;
            }));
        }), [ o ]);
        var F = l()(X);
        (0, n.useEffect)((function() {
            // if the focused tab is removed, focus the next tab
            if (F !== X) {
                var e;
                var t = je(M.current);
                if (t.find((function(e) {
                    return e.tabId === j.current;
                }))) {
                    // focused tab is still present
                    return;
                }
                var n = Math.min((e = P.current) !== null && e !== void 0 ? e : 0, t.length - 1);
                var r = t[n];
                if (r) {
                    _(r.tabId);
                }
            }
        }), [ X, _, F ]);
        
        return r().createElement(ie, ve({
            "data-tab-layout": p,
            "data-test-active-tab-id": a,
            "data-test": "tab-bar",
            role: "tablist",
            ref: D,
            $layout: p,
            onKeyDown: q
        }, g),  r().createElement(x.Provider, {
            value: W
        }, X));
    }
    Pe.propTypes = Te;
    Pe.Tab = ae;
    /* harmony default export */    const Ee = Pe;
    // CONCATENATED MODULE: ./src/TabBar/index.ts
    module.exports = t;
    /******/})();
