// @splunk/react-ui/Table.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        Body: () => /* reexport */ J,
        Caption: () => /* reexport */ de,
        Cell: () => /* reexport */ Oe,
        Head: () => /* reexport */ er,
        HeadCell: () => /* reexport */ $n,
        HeadDropdownCell: () => /* reexport */ xr,
        Row: () => /* reexport */ Ht,
        default: () => /* reexport */ Qo
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "@dnd-kit/core"
    const o = require("@dnd-kit/core");
    // CONCATENATED MODULE: external "@dnd-kit/modifiers"
    const i = require("@dnd-kit/modifiers");
    // CONCATENATED MODULE: external "@dnd-kit/sortable"
    const a = require("@dnd-kit/sortable");
    // CONCATENATED MODULE: external "lodash/forEach"
    const l = require("lodash/forEach");
    var u =  e.n(l);
    // CONCATENATED MODULE: external "lodash/has"
    const s = require("lodash/has");
    // CONCATENATED MODULE: external "lodash/indexOf"
    const c = require("lodash/indexOf");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "lodash/isEqual"
    const f = require("lodash/isEqual");
    var v =  e.n(f);
    // CONCATENATED MODULE: external "lodash/isNumber"
    const p = require("lodash/isNumber");
    // CONCATENATED MODULE: external "lodash/omit"
    const m = require("lodash/omit");
    var b =  e.n(m);
    // CONCATENATED MODULE: external "lodash/throttle"
    const h = require("lodash/throttle");
    var y =  e.n(h);
    // CONCATENATED MODULE: external "prop-types"
    const g = require("prop-types");
    var w =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const C = require("@splunk/react-ui/EventListener");
    var x =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScrollContainerContext"
    const S = require("@splunk/react-ui/ScrollContainerContext");
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const O = require("@splunk/react-ui/usePrevious");
    var k =  e.n(O);
    // CONCATENATED MODULE: external "@splunk/react-ui/useResizeObserver"
    const R = require("@splunk/react-ui/useResizeObserver");
    var E =  e.n(R);
    // CONCATENATED MODULE: external "@splunk/ui-utils/format"
    const I = require("@splunk/ui-utils/format");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const T = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const j = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "lodash/includes"
    const D = require("lodash/includes");
    var P =  e.n(D);
    // CONCATENATED MODULE: external "lodash/without"
    const A = require("lodash/without");
    var q =  e.n(A);
    // CONCATENATED MODULE: external "styled-components"
    const _ = require("styled-components");
    var z =  e.n(_);
    // CONCATENATED MODULE: external "@splunk/themes"
    const M = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Table/BodyStyles.ts
    var $ = z().tbody.withConfig({
        displayName: "BodyStyles__Styled",
        componentId: "sc-1k5p6pq-0"
    })([ "", " z-index:0;" ], M.mixins.reset("table-row-group"));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function N(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Table/Body.tsx
    function L() {
        return L = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, L.apply(null, arguments);
    }
    function W(e, n) {
        return F(e) || B(e, n) || K(e, n) || H();
    }
    function H() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function K(e, n) {
        if (e) {
            if ("string" == typeof e) return V(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? V(e, n) : void 0;
        }
    }
    function V(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function B(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, l = [], u = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = i.call(t)).done) && (l.push(r.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function F(e) {
        if (Array.isArray(e)) return e;
    }
    function U(e, n) {
        if (null == e) return {};
        var t, r, o = X(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function X(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var Y = {
        /** @private. Generally passed by Table rather than added directly. */
        actions: w().bool,
        children: w().node,
        elementRef: w().oneOfType([ w().func, w().object ]),
        /** @private. Generally passed by Table rather than added directly. */
        movableColumns: w().bool,
        /** @private. Generally passed by Table rather than added directly. */
        rowExpansion: w().oneOf([ "single", "multi", "none", "controlled" ]),
        /** @private. Generally passed by Table rather than added directly. */
        onRequestMoveRow: w().func,
        /** @private. */
        pinnedColumns: w().object,
        /** @private. Generally passed by Table rather than added directly. */
        primaryColumnIndex: w().number,
        /** @private. Generally passed by Table rather than added directly. */
        stripeRows: w().bool
    };
    function G(e) {
        var n = e.actions, o = e.children, i = e.elementRef, a = e.pinnedColumns, l = e.movableColumns, u = e.onRequestMoveRow, s = e.rowExpansion, c = s === void 0 ? "none" : s, d = e.primaryColumnIndex, f = e.stripeRows, v = U(e, [ "actions", "children", "elementRef", "pinnedColumns", "movableColumns", "onRequestMoveRow", "rowExpansion", "primaryColumnIndex", "stripeRows" ]);
 // @docs-props-type BodyPropsBase
                var p = [];
        var m = (0, t.useState)([]), b = W(m, 2), h = b[0], y = b[1];
        var g = (0, t.useRef)(null);
        var w = (0, t.useCallback)((function(e) {
            g.current = e;
            N(i, e);
        }), [ i, g ]);
        var C = function e(n, t, r) {
            var o = h;
            if (c === "single") {
                if (P()(o, t)) {
                    y([]);
                } else {
                    y([ t ]);
                }
            } else if (c === "multi") {
                if (P()(o, t)) {
                    y(q()(o, t));
                } else {
                    y(o === null || o === void 0 ? void 0 : o.concat(t));
                }
            }
            r === null || r === void 0 ? void 0 : r(n);
        };
        var x = t.Children.toArray(o).filter(t.isValidElement);
        var S = (0, t.useState)(new Map), O = W(S, 2), k = O[0], R = O[1];
        r().useEffect((function() {
            R((function(e) {
                var n = new Map(e);
                var r = false;
                x.forEach((function(o) {
                    var i;
                    if (!o.props.expansionRow) return;
                    var a = (i = o.key) !== null && i !== void 0 ? i : x.indexOf(o);
                    var l = t.Children.count(o.props.expansionRow);
                    var u = e.get(a);
                    if (!u || u.length !== l) {
                        n.set(a, Array.from({
                            length: l
                        }, (function() {
                            return (0, j.createDOMID)();
                        })));
                        r = true;
                    }
                }));
                return r ? n : e;
            }));
        }), [ x ]);
        x.forEach((function(e, r) {
            if (e) {
                var o, i;
                if (false) {}
                var s = (o = e.key) !== null && o !== void 0 ? o : r;
                var v = r % 2 ? "even" : "odd";
                var m = f ? v : "none";
                var b = c === "controlled" ? e.props.expanded : P()(h, s);
                var y = function n(t) {
                    C(t, s, e.props.onExpansion);
                };
                var g = e.props.expansionRow ? (i = k.get(s)) === null || i === void 0 ? void 0 : i.join(" ") : undefined;
                p.push( (0, t.cloneElement)(e, {
                    stripe: m,
                    actions: n,
                    expanded: b,
                    expansionRowIds: g,
                    expandable: c !== "none",
                    pinnedColumns: a,
                    isDraggable: !!u,
                    movableColumns: l,
                    primaryColumnIndex: d,
                    onExpansion: y,
                    key: e.key || r
                }));
                if (b && e.props.expansionRow) {
                    t.Children.forEach(e.props.expansionRow, (function(e, n) {
                        var r;
                        var o = (r = k.get(s)) === null || r === void 0 ? void 0 : r[n];
                        p.push( (0, t.cloneElement)(e, {
                            key: o,
                            id: o,
                            stripe: m,
                            movableColumns: l,
                            "data-expansion-row": "true"
                        }));
                    }));
                }
            }
        }));
        if (p.length === 0) {
            return null;
        }
        
        return r().createElement($, L({
            "data-test": "body"
        }, v, {
            ref: w
        }), p);
    }
    G.propTypes = Y;
    G.splunkUiType = "Table.Body";
    /* harmony default export */    const J = G;
    // CONCATENATED MODULE: ./src/Table/TableStyles.ts
    var Q = z().div.withConfig({
        displayName: "TableStyles__Styled",
        componentId: "sc-1cmfss7-0"
    })([ "", ";max-width:100%;position:relative;z-index:1;@media print{max-height:none !important;}" ], M.mixins.reset("block"));
    var Z = z().div.withConfig({
        displayName: "TableStyles__StyledTableContainer",
        componentId: "sc-1cmfss7-1"
    })([ "overflow:auto;", " &::before{content:'';width:100%;height:0;top:0;position:absolute;}" ], (function(e) {
        var n = e.$dragging;
        return n && (0, _.css)([ "overflow:hidden;" ]);
    }));
    var ee = z().table.withConfig({
        displayName: "TableStyles__StyledTable",
        componentId: "sc-1cmfss7-2"
    })([ "", ";position:relative;border-collapse:collapse;border-spacing:0;min-width:100%;z-index:0;height:1px;&[data-fixed-column='true']{table-layout:fixed;min-width:0;width:0;}", " @media print{width:100%;max-width:100%;table-layout:auto;}" ], M.mixins.reset("table"), (function(e) {
        var n = e.$resizableFillLayout;
        return n && (0, _.css)([ "width:100%;min-width:100%;table-layout:fixed;" ]);
    }));
    var ne = z().div.withConfig({
        displayName: "TableStyles__StyledDockedScrollbar",
        componentId: "sc-1cmfss7-3"
    })([ "position:fixed;bottom:0;overflow:auto;z-index:calc(", " + 1);" ], M.variables.zindexFixedNavbar);
    var te = z().div.withConfig({
        displayName: "TableStyles__StyledDockedScrollbarContent",
        componentId: "sc-1cmfss7-4"
    })([ "height:1px;" ]);
    var re = z().caption.withConfig({
        displayName: "TableStyles__StyledTableCaption",
        componentId: "sc-1cmfss7-5"
    })([ "padding:", " 0;font-size:", ";caption-side:", ";" ], M.variables.spacingXSmall, M.variables.fontSizeLarge, (function(e) {
        var n = e.side;
        return n;
    }));
    var oe = z().table.withConfig({
        displayName: "TableStyles__StyledDragShadowTable",
        componentId: "sc-1cmfss7-6"
    })([ "display:block;background:", ";border-collapse:collapse;height:", ";width:", ";" ], M.variables.interactiveColorOverlayDrag, (function(e) {
        var n = e.$height;
        return "".concat(n, "px");
    }), (function(e) {
        var n = e.$width;
        return "".concat(n, "px");
    }));
    var ie = z().div.withConfig({
        displayName: "TableStyles__StyledGuideLine",
        componentId: "sc-1cmfss7-7"
    })([ "position:absolute;width:1px;left:", ";top:0;height:", ";pointer-events:none;background-color:", ";" ], (function(e) {
        var n = e.$left;
        return "".concat(n, "px");
    }), (function(e) {
        var n = e.$height;
        return "".concat(n, "px");
    }), M.variables.interactiveColorAccent);
    // CONCATENATED MODULE: ./src/Table/Caption.tsx
    function ae() {
        return ae = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, ae.apply(null, arguments);
    }
    function le(e, n) {
        if (null == e) return {};
        var t, r, o = ue(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function ue(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var se = {
        children: w().node.isRequired,
        side: w().oneOf([ "top", "bottom" ])
    };
    /**
 * Tables that use a docked header must place the caption on the bottom side.
 * Tables that use a fixed header cannot use captions.
 */    function ce(e) {
        var n = e.children, t = e.side, o = t === void 0 ? "top" : t, i = le(e, [ "children", "side" ]);
        // @docs-props-type CaptionPropsBase
                
        return r().createElement(re, ae({
            "data-test": "caption",
            side: o
        }, i), n);
    }
    ce.propTypes = se;
    ce.splunkUiType = "Table.Caption";
    /* harmony default export */    const de = ce;
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const fe = require("@splunk/react-ui/ButtonSimple");
    // CONCATENATED MODULE: ./src/Table/CellStyles.ts
    var ve = "2px";
    var pe = (0, _.css)([ "box-shadow:", ";" ], M.variables.focusShadowInset);
    var me = z().td.withConfig({
        displayName: "CellStyles__Styled",
        componentId: "sc-1955xk2-0"
    })([ "", " line-height:", ";padding:", " ", ";border-left:0 solid transparent;border-right:0 solid transparent;color:", ";word-wrap:break-word;vertical-align:top;*:where(button):focus,&& *:is(:has(> input:focus)):not([disabled]){box-shadow:", ";}padding-block:", ";padding-inline:", ";", " ", " ", " [data-expansion-row='true'] > &:first-child{padding-top:8px;}[data-expansion-row='true'] > &:last-child{padding-bottom:14px;}", "" ], M.mixins.reset("table-cell"), M.variables.lineHeightSpacious, ve, M.variables.spacingSmall, M.variables.contentColorDefault, M.variables.focusShadowInset, (0, 
    M.pick)({
        comfortable: "8px",
        compact: "6px"
    }), M.variables.spacingSmall, (0, M.pickVariant)("$variant", {
        expand: (0, _.css)([ "height:100%;padding:0;" ]),
        toggle: (0, _.css)([ "height:100%;padding:0;" ])
    }), (function(e) {
        var n = e.$variant, t = e.$isPrimaryCell;
        return (0, M.pickVariant)("$appearance", {
            data: (0, _.css)([ "", "" ], n === "actions" && (0, _.css)([ "cursor:default;padding-block:", ";padding-inline:", ";*:where(button){", " &:focus{box-shadow:", ";}}" ], ve, M.variables.spacingSmall, (0, 
            fe.buttonMixin)("subtle", {
                color: M.variables.contentColorDefault
            }), M.variables.focusShadowInset)),
            rowLink: (0, _.css)([ "color:", ";*:hover > &&{cursor:pointer;", "}" ], M.variables.contentColorActive, t && (0, 
            _.css)([ "text-decoration:underline;" ])),
            link: (0, _.css)([ "color:", ";&:hover{cursor:pointer;text-decoration:underline;}&:focus{text-decoration:underline;box-shadow:", ";}" ], M.variables.contentColorLink, M.variables.focusShadowInset)
        });
    }), (function(e) {
        var n = e.$clickable, t = e.$focusWithin;
        return n && (0, _.css)([ "cursor:pointer;&:hover{background-color:", ";}&:focus{", " background-color:", ";}", "" ], M.variables.neutral100, pe, M.variables.neutral100, t && (0, 
        _.css)([ "&:focus-within{", " background-color:", ";}" ], pe, M.variables.neutral100));
    }), (function(e) {
        var n = e.$isPinned;
        return n && (0, _.css)([ "background-color:", ";position:sticky;inset-inline-end:-1px;&::before{content:'';position:absolute;left:0;top:0;bottom:0;border-left:1px solid ", ";}" ], M.variables.backgroundColorPage, M.variables.borderColor);
    }));
    /*
 * A wrapper ensures that the Table Cell contents align to the top while staying within the height of inputHeight.
 * This allows text to align properly with expand and select icons when other cell contents are larger.
 */    var be = z().div.withConfig({
        displayName: "CellStyles__StyledCellContent",
        componentId: "sc-1955xk2-1"
    })([ "display:flex;align-items:flex-start;", " ", " ", "" ], (0, M.pickVariant)("$variant", {
        expand: (0, _.css)([ "height:100%;" ]),
        toggle: (0, _.css)([ "height:100%;" ])
    }), (function(e) {
        var n = e.$variant, t = e.$align;
        return n ? (0, _.css)([ "display:block;text-align:", ";" ], t) : (0, _.css)([ "justify-content:", ";text-align:", ";" ], t, t);
    }), (0, M.pickVariant)("$horizontalOverflow", {
        auto: (0, _.css)([ "word-break:break-word;" ]),
        scroll: (0, _.css)([ "overflow-wrap:wrap;" ])
    }));
    // CONCATENATED MODULE: ./src/Table/TableContext.ts
    var he =  (0, t.createContext)({});
    he.displayName = "Table";
    /* harmony default export */    const ye = he;
    // CONCATENATED MODULE: ./src/Table/Cell.tsx
    function ge() {
        return ge = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, ge.apply(null, arguments);
    }
    function we(e, n) {
        if (null == e) return {};
        var t, r, o = Ce(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function Ce(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */    var xe = {
        align: w().oneOf([ "left", "center", "right" ]),
        /** @private */
        appearance: w().oneOf([ "data", "link", "rowLink" ]),
        /** @private Used to set the $clickable appearance without an onClick */
        appearClickable: w().bool,
        children: w().node,
        data: w().any,
        elementRef: w().oneOfType([ w().func, w().object ]),
        /** @private */
        isPrimaryCell: w().bool,
        /** @private */
        isPinned: w().bool,
        onClick: w().func,
        /** @private */
        onKeyDown: w().func,
        /** @private */
        disabled: w().bool,
        /** @private. */
        expand: w().bool,
        /** @private. */
        stripe: w().oneOf([ "odd", "even", "none" ]),
        /** @private. */
        variant: w().oneOf([ "toggle", "expand", "drag", "actions" ])
    };
    function Se(e) {
        var n = e.align, o = n === void 0 ? "left" : n, i = e.appearance, a = e.appearClickable, l = e.children, u = e.data, s = e.disabled, c = e.elementRef, d = e.isPinned, f = e.isPrimaryCell, v = e.onClick, p = e.onKeyDown, m = e.stripe, b = e.variant, h = we(e, [ "align", "appearance", "appearClickable", "children", "data", "disabled", "elementRef", "isPinned", "isPrimaryCell", "onClick", "onKeyDown", "stripe", "variant" ]);
        // @docs-props-type CellPropsBase
                var y = v !== undefined;
        var g = a || y;
        var w = g ? "link" : "data";
        var C = (0, t.useContext)(ye), x = C.horizontalOverflow;
        var S = (0, t.useCallback)((function(e) {
            N(c, e);
        }), [ c ]);
        var O = (0, t.useCallback)((function(e) {
            if (b === "actions") {
                // Action buttons have their own onClicks inside the cell so we don't want Row onClick to fire
                e.stopPropagation();
            }
            v === null || v === void 0 ? void 0 : v(e, u);
        }), [ u, v, b ]);
        var k = (0, t.useCallback)((function(e) {
            if (e.key === "Enter") {
                v === null || v === void 0 ? void 0 : v(e, u);
            }
            p === null || p === void 0 ? void 0 : p(e);
        }), [ u, v, p ]);
        
        return r().createElement(me, ge({
            "data-test": "cell",
            $appearance: i || w,
            $clickable: g ? true : undefined,
            $disabled: !!s,
            $focusWithin: a,
            $isPinned: d,
            $isPrimaryCell: f,
            $stripe: m,
            $variant: b,
            onClick: O,
            onKeyDown: k,
            ref: S,
            tabIndex: y ? 0 : undefined
        }, h),  r().createElement(be, {
            $align: o,
            $horizontalOverflow: x,
            $variant: b
        }, l));
    }
    Se.propTypes = xe;
    Se.splunkUiType = "Table.Cell";
    /* harmony default export */    const Oe = Se;
    // CONCATENATED MODULE: external "lodash/isString"
    const ke = require("lodash/isString");
    var Re =  e.n(ke);
    // CONCATENATED MODULE: external "lodash/merge"
    const Ee = require("lodash/merge");
    var Ie =  e.n(Ee);
    // CONCATENATED MODULE: external "@splunk/react-icons/ParallelDots"
    const Te = require("@splunk/react-icons/ParallelDots");
    var je =  e.n(Te);
    // CONCATENATED MODULE: external "@splunk/react-ui/Tooltip"
    const De = require("@splunk/react-ui/Tooltip");
    var Pe =  e.n(De);
    // CONCATENATED MODULE: external "@splunk/react-ui/Divider"
    const Ae = require("@splunk/react-ui/Divider");
    var qe =  e.n(Ae);
    // CONCATENATED MODULE: ./src/Table/HeadInnerStyles.ts
    var _e = 4;
    var ze = _e;
 // While this could theoretically be different, it helps for cohesivity in look and feel when some columns are truncated and others are not.
        var Me = (0, _.css)([ "display:inline-flex;align-items:center;grid-row:1;height:1lh;" ]);
    var $e = z().span.withConfig({
        displayName: "HeadInnerStyles__StyledIcon",
        componentId: "sc-1sa0wng-0"
    })([ "", " grid-column:icon;" ], Me);
    var Ne = z().span.withConfig({
        displayName: "HeadInnerStyles__StyledTooltipIcon",
        componentId: "sc-1sa0wng-1"
    })([ "", " grid-column:tooltip;" ], Me);
    var Le = z().span.withConfig({
        displayName: "HeadInnerStyles__StyledLabel",
        componentId: "sc-1sa0wng-2"
    })([ "grid-column:label;grid-row:1 / -1;position:relative;height:100%;", " ", " ", " ", "" ], (function(e) {
        var n = e.$variant;
        return n === "toggleAll" && (0, _.css)([ "grid-column:1 / -1;" ]);
    }), (function(e) {
        var n = e.$labelLength;
        return n && n > ze && (0, _.css)([ "word-wrap:break-word;word-break:break-word;" ]);
    }), (function(e) {
        var n = e.$truncate, t = e.$labelLength;
        return (0, M.pickVariant)("$horizontalOverflow", {
            auto: n && t && t > _e && (0, _.css)([ "", " min-width:4ch;" ], M.mixins.ellipsis()),
            scroll: (0, _.css)([ "white-space:nowrap;" ])
        });
    }), (0, M.pickVariant)("$align", {
        right: (0, _.css)([ "& > &{text-align:right;}" ]),
        left: (0, _.css)([ "& > &{text-align:left;}" ]),
        center: (0, _.css)([ "& > &{text-align:center;}" ])
    }));
    var We = (0, _.css)([ "box-shadow:", ";outline:none;color:", ";text-decoration:underline;" ], M.variables.focusShadowInset, M.variables.contentColorActive);
    var He = z().div.withConfig({
        displayName: "HeadInnerStyles__Styled",
        componentId: "sc-1sa0wng-3"
    })([ "", ";", ";", ";position:relative;", " grid-template-rows:1lh 1fr;align-items:start;flex-grow:1;height:100%;padding-block:", ";padding-inline:", ";justify-content:", ";", " ", " @media print{background-image:none;}", "" ], M.mixins.reset("grid"), M.mixins.clearfix(), M.mixins.typography("body", {
        weight: "bold"
    }), (function(e) {
        var n = e.$hasTooltip, t = e.$hasIcon;
        return (0, _.css)([ "grid-template-columns:[label] auto ", " ", ";column-gap:", ";" ], n ? "[tooltip] min-content" : "", t ? "[icon] min-content" : "", n || t ? M.variables.spacingXSmall : 0);
    }), (0, M.pick)({
        comfortable: "8px",
        compact: "6px"
    }), M.variables.spacingSmall, (0, M.pickVariant)("$align", {
        center: "center",
        left: "flex-start",
        right: "flex-end"
    }), (function(e) {
        var n = e.$variant;
        return n && (0, _.css)([ "grid-template-columns:1fr;grid-template-rows:1fr;" ]);
    }), (0, M.pickVariant)("$variant", {
        toggleAll: (0, _.css)([ "padding:0;" ]),
        actions: (0, _.css)([ "padding-block:2px;" ])
    }), (function(e) {
        var n = e.$clickable, t = e.$focusWithin;
        return n && (0, _.css)([ "cursor:pointer;&:focus{", "}", " &:hover{outline:none;text-decoration:underline;color:", ";}" ], We, t && (0, 
        _.css)([ "&:focus-within{", "}" ], We), M.variables.contentColorActive);
    }));
    var Ke = z()(qe()).withConfig({
        displayName: "HeadInnerStyles__StyledResize",
        componentId: "sc-1sa0wng-4"
    })([ "", ";position:absolute;right:-5px;width:9px;top:0;bottom:0;z-index:1;cursor:col-resize;th:last-child > div > ", " > &{right:1px;width:4px;}&:hover{outline:none;&::before{content:'';background:", ";position:absolute;left:4px;width:1px;top:0;bottom:0;}}&:focus{outline:none;&::before{content:'';position:absolute;background:", ";box-shadow:", ";left:4px;width:1px;top:0;bottom:0;}}" ], M.mixins.reset("block"), /* sc-sel */ He, M.variables.borderColorStrong, M.variables.focusColor, M.variables.focusShadow);
    // CONCATENATED MODULE: ./src/Table/HeadCellStyles.ts
    var Ve = z().div.withConfig({
        displayName: "HeadCellStyles__StyledInnerWrapper",
        componentId: "sc-1hj81sa-0"
    })([ "display:flex;height:100%;align-items:flex-start;" ]);
    var Be = z().th.withConfig({
        displayName: "HeadCellStyles__Styled",
        componentId: "sc-1hj81sa-1"
    })([ "", " line-height:", ";background-color:", ";box-sizing:content-box;text-align:left;vertical-align:", ";position:relative;height:100%;&:not(:first-child,:has(", ")){", "{border-left:1px solid ", ";}}&:has(", "){", "{border-right:1px solid ", ";}}&:nth-last-child(2):has(", ") + &:last-child{", "{border-left:none;}}*:where(button):focus{box-shadow:", ";}", " ", " ", " ", "" ], M.mixins.reset("table-cell"), M.variables.lineHeightSpacious, M.variables.neutral100, (function(e) {
        var n = e.$variant;
        return n ? "middle" : "top";
    }), Ke, /* sc-sel */ Ve, M.variables.borderColorWeak, Ke, /* sc-sel */ Ve, M.variables.borderColor, Ke, /* sc-sel */ Ve, M.variables.focusShadowInset, (function(e) {
        var n = e.$isDragging;
        return n && (0, _.css)([ "background-color:", ";&:focus{box-shadow:none;}" ], M.variables.backgroundColorPage);
    }), (function(e) {
        var n = e.$variant;
        return n && (0, _.css)([ "&:first-child + &{", "{border-left:1px solid ", ";}}" ], Ve, M.variables.borderColorWeak);
    }), (0, M.pickVariant)("$variant", {
        actions: (0, _.css)([ "*:where(button){", " &:focus{box-shadow:", ";}}" ], (0, fe.buttonMixin)("subtle", {
            color: M.variables.contentColorDefault
        }), M.variables.focusShadowInset)
    }), (function(e) {
        var n = e.$isPinned;
        return n && (0, _.css)([ "position:sticky;inset-inline-end:-1px;&::before{content:'';position:absolute;left:0;top:0;bottom:0;border-left:1px solid ", ";}" ], M.variables.borderColor);
    }));
    var Fe = z().div.withConfig({
        displayName: "HeadCellStyles__StyledIcon",
        componentId: "sc-1hj81sa-2"
    })([ "display:inline-flex;align-items:center;height:1lh;" ]);
    // TODO(SUI-7400): Refactor the drag handler of re-order column
        var Ue = z().div.withConfig({
        displayName: "HeadCellStyles__StyledDragHandler",
        componentId: "sc-1hj81sa-3"
    })([ "display:inline-flex;align-items:start;align-self:stretch;justify-content:center;min-width:20px;cursor:move;padding-block:", ";-webkit-user-select:none;-moz-user-select:none;user-select:none;&:hover{background:", ";color:", ";}&:focus{background:", ";box-shadow:", ";color:", ";}" ], (0, 
    M.pick)({
        comfortable: "8px",
        compact: "6px"
    }), M.variables.interactiveColorOverlayHover, M.variables.contentColorActive, M.variables.interactiveColorOverlayDrag, M.variables.focusShadowInset, M.variables.contentColorActive);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowDown"
    const Xe = require("@splunk/react-icons/ArrowDown");
    var Ye =  e.n(Xe);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowUp"
    const Ge = require("@splunk/react-icons/ArrowUp");
    var Je =  e.n(Ge);
    // CONCATENATED MODULE: external "@splunk/react-icons/ArrowUpDown"
    const Qe = require("@splunk/react-icons/ArrowUpDown");
    var Ze =  e.n(Qe);
    // CONCATENATED MODULE: external "@splunk/react-icons/CaretSmallDown"
    const en = require("@splunk/react-icons/CaretSmallDown");
    var nn =  e.n(en);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const tn = require("@splunk/react-icons/InformationCircle");
    var rn =  e.n(tn);
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var on = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function an() {
        var e = typeof document !== "undefined" ? document : on;
        return e;
    }
    var ln = an();
    /* harmony default export */    const un = /* unused pure expression or super */ null && ln;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var sn = {
        document: on,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                n();
                return null;
            }
            return setTimeout(n, 0);
        },
        cancelAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(n);
        }
    };
    function cn() {
        var e = typeof window !== "undefined" ? window : sn;
        return e;
    }
    var dn = cn();
    /* harmony default export */    const fn = /* unused pure expression or super */ null && dn;
    // CONCATENATED MODULE: ./src/Table/HeadInner.tsx
    function vn() {
        return vn = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, vn.apply(null, arguments);
    }
    function pn(e) {
        return hn(e) || bn(e) || wn(e) || mn();
    }
    function mn() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function bn(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function hn(e) {
        if (Array.isArray(e)) return Cn(e);
    }
    function yn(e, n) {
        return Sn(e) || xn(e, n) || wn(e, n) || gn();
    }
    function gn() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function wn(e, n) {
        if (e) {
            if ("string" == typeof e) return Cn(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? Cn(e, n) : void 0;
        }
    }
    function Cn(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function xn(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, l = [], u = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = i.call(t)).done) && (l.push(r.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function Sn(e) {
        if (Array.isArray(e)) return e;
    }
    function On(e, n) {
        if (null == e) return {};
        var t, r, o = kn(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function kn(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var Rn = {
        align: w().oneOf([ "left", "center", "right" ]),
        cellId: w().string,
        clickable: w().bool,
        columnId: w().string,
        elementRef: w().oneOfType([ w().func, w().object ]),
        focusWithin: w().bool,
        hasTooltip: w().bool,
        index: w().number,
        isMenu: w().bool,
        label: w().node,
        onAutosizeColumn: w().func,
        onRequestResize: w().func,
        resizable: w().bool,
        sortDir: w().oneOf([ "none", "asc", "desc" ]),
        truncate: w().bool,
        width: w().oneOfType([ w().number, w().oneOf([ "auto" ]) ]),
        variant: w().oneOf([ "toggleAll", "info", "actions" ])
    };
    var En = new Set([ "none", "asc", "desc" ]);
    var In = Object.freeze({
        none: (0, T._)("Click to sort"),
        asc: (0, T._)("Press to sort descending"),
        desc: (0, T._)("Press to sort ascending")
    });
    var Tn = Object.freeze({
        none: Ze(),
        asc: Je(),
        desc: Ye()
    });
    function jn(e) {
        var n = e.align, o = n === void 0 ? "left" : n, i = e.clickable, a = e.cellId, l = e.columnId, u = e.elementRef, s = e.focusWithin, c = e.hasTooltip, d = e.index, f = e.isMenu, v = e.label, p = e.onAutosizeColumn, m = e.onRequestResize, h = e.resizable, y = h === void 0 ? true : h, g = e.sortDir, w = e.truncate, C = w === void 0 ? true : w, S = e.variant, O = e.width, k = On(e, [ "align", "clickable", "cellId", "columnId", "elementRef", "focusWithin", "hasTooltip", "index", "isMenu", "label", "onAutosizeColumn", "onRequestResize", "resizable", "sortDir", "truncate", "variant", "width" ]);
        // @docs-props-type HeadInnerPropsBase
                var R = (0, T._)("Click to open menu");
        var E = (0, t.useMemo)((function() {
            return typeof v === "string" ? v.length : 0;
        }), [ v ]);
        var j = (0, t.useContext)(ye), D = j.horizontalOverflow;
        var P = (0, t.useState)(false), A = yn(P, 2), q = A[0], _ = A[1];
        var z = (0, t.useState)(0), M = yn(z, 2), $ = M[0], L = M[1];
        var W = (0, t.useState)(0), H = yn(W, 2), K = H[0], V = H[1];
        var B = (0, t.useRef)(null);
        var F = (0, t.useRef)(O === "auto" ? undefined : O);
        var U;
        var X;
        if (g !== undefined && En.has(g)) {
            U = Tn[g];
            X = In[g];
        }
        if (O === "auto") {
            var Y;
            var G = B.current;
            var J = G === null || G === void 0 ? void 0 : (Y = G.parentElement) === null || Y === void 0 ? void 0 : Y.getBoundingClientRect();
            F.current = J === null || J === void 0 ? void 0 : J.width;
        } else {
            F.current = O;
        }
        var Q = (0, t.useCallback)((function(e) {
            B.current = e;
            N(u, e);
        }), [ u, B ]);
        var Z = (0, t.useCallback)((function(e) {
            var n = e.key;
            if (m === undefined || d === undefined) {
                return;
            }
            if (O === "auto") {
                var t;
                var r = B.current;
                var o = r === null || r === void 0 ? void 0 : (t = r.parentElement) === null || t === void 0 ? void 0 : t.getBoundingClientRect();
                F.current = o === null || o === void 0 ? void 0 : o.width;
            } else {
                F.current = O;
            }
            if (F.current === undefined) {
                return;
            }
            if (n === "ArrowLeft") {
                e.preventDefault();
 // prevent text selection
                                var i = Math.max(F.current - 10, 20);
                m(e, {
                    index: d,
                    columnId: l,
                    id: a,
                    width: i
                });
            }
            if (n === "ArrowRight") {
                e.preventDefault();
 // prevent text selection
                                var u = F.current + 10;
                m(e, {
                    index: d,
                    columnId: l,
                    id: a,
                    width: u
                });
            }
            if (n === "Enter") {
                e.preventDefault();
 // SUI-7320: prevent sorting and dropdown opening
                        }
        }), [ l, a, d, m, O ]);
        var ee = (0, t.useCallback)((function(e) {
            if (O === "auto") {
                var n;
                var t = B.current;
                var r = t === null || t === void 0 ? void 0 : (n = t.parentElement) === null || n === void 0 ? void 0 : n.getBoundingClientRect();
                F.current = r === null || r === void 0 ? void 0 : r.width;
            } else {
                F.current = O;
            }
            if (F.current === undefined) {
                return;
            }
            e.preventDefault();
 // prevent text selection
                        L(e.clientX);
            V(F.current || 0);
            _(true);
        }), [ O ]);
        var ne = function e(n) {
            var t = $ - n.clientX;
            var r = Math.max(K - t, 16);
            if (d !== undefined) {
                m === null || m === void 0 ? void 0 : m(n, {
                    index: d,
                    columnId: l,
                    id: a,
                    width: r
                });
            }
        };
        var te = function e() {
            _(false);
        };
        var re = (0, t.useCallback)((function(e) {
            if (d !== undefined) {
                p === null || p === void 0 ? void 0 : p(e, {
                    index: d,
                    columnId: l,
                    id: a
                });
            }
        }), [ a, l, d, p ]);
        var oe = cn();
        var ie = F.current ? (0, I.sprintf)((0, T._)("%(width)d pixels"), {
            width: F.current
        }) : undefined;
        
        return r().createElement(He, vn({
            as: i ? "button" : "div",
            ref: Q,
            $align: o,
            $clickable: i,
            $focusWithin: s,
            $hasIcon: !!f || !!g,
            $hasTooltip: !!c,
            $variant: S,
            $labelLength: E
        }, b()(k, [].concat(pn(Object.keys(jn.propTypes)), [ "id" ]))), q &&  r().createElement(r().Fragment, null,  r().createElement(x(), {
            target: oe,
            eventType: "mouseup",
            listener: te
        }),  r().createElement(x(), {
            target: oe,
            eventType: "mousemove",
            listener: ne
        })), m && y &&  r().createElement(Ke, {
            "aria-label": (0, I.sprintf)((0, T._)("Resize %(label)s"), {
                label: v
            }),
            "aria-controls": a,
            "aria-valuetext": ie,
            "data-test": "resize",
            onMouseDown: ee,
            onDoubleClick: re,
            onKeyDown: Z,
            orientation: "vertical",
            tabIndex: 0
        }),  r().createElement(Le, {
            $align: o,
            $horizontalOverflow: D,
            $labelLength: E,
            $truncate: C,
            $variant: S
        }, v), c &&  r().createElement(Ne, null,  r().createElement(rn(), {
            "data-test": "tooltip-icon"
        })), f &&  r().createElement($e, null,  r().createElement(nn(), {
            "data-test": "caret-down",
            "aria-label": R
        })), g &&  r().createElement($e, null, U !== undefined &&  r().createElement(U, {
            "aria-hidden": false,
            "aria-label": X
        })));
    }
    jn.propTypes = Rn;
    /* harmony default export */    const Dn = jn;
    // CONCATENATED MODULE: ./src/Table/HeadCell.tsx
    function Pn() {
        return Pn = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, Pn.apply(null, arguments);
    }
    function An(e, n) {
        if (null == e) return {};
        var t, r, o = qn(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function qn(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */
    /** @public */    var _n = {
        align: w().oneOf([ "left", "center", "right" ]),
        /** @private Used to set the $clickable appearance without an onClick */
        appearClickable: w().bool,
        children: w().node,
        columnId: w().string,
        /** @private. */
        dragId: w().oneOfType([ w().string, w().number ]),
        elementRef: w().oneOfType([ w().func, w().object ]),
        headCellScreenReaderText: w().string,
        /** @private The index of the cell, skipping the info column. */
        index: w().number,
        /** @private. */
        isPinned: w().bool,
        /** @private. */
        onAutosizeColumn: w().func,
        /** @private. */
        onClick: w().func,
        /**
   * @private. Passed through and works as expected.
   */
        onKeyDown: w().func,
        /** @private. */
        onRequestMoveColumn: w().func,
        /** @private. */
        onRequestResize: w().func,
        onSort: w().func,
        resizable: w().bool,
        sortDir: w().oneOf([ "asc", "desc", "none" ]),
        sortKey: w().string,
        tooltip: w().node,
        truncate: w().bool,
        /** @private. */
        variant: w().oneOf([ "toggleAll", "info", "actions" ]),
        /**
   * Used internally to suppress focus and id when this `HeadCell` is superseded by one in an
   * overlaid `HeadTable` for user interactions.
   * @private
   */
        visible: w().bool,
        width: w().oneOfType([ w().number, w().oneOf([ "auto" ]) ])
    };
    var zn = {
        asc: "ascending",
        desc: "descending",
        none: "none"
    };
    function Mn(e) {
        var n;
        var o = e.align, i = o === void 0 ? "left" : o, l = e.appearClickable, u = e.children, s = e.columnId, c = e.elementRef, d = e.dragId, f = e.headCellScreenReaderText, v = e.id, p = e.index, m = e.isPinned, b = e.onAutosizeColumn, h = e.onClick, y = e.onKeyDown, g = e.onRequestMoveColumn, w = e.onRequestResize, C = e.onSort, x = e.resizable, S = x === void 0 ? true : x, O = e.sortDir, k = O === void 0 ? "none" : O, R = e.sortKey, E = e.style, D = e.tooltip, P = e.truncate, A = P === void 0 ? true : P, q = e.variant, _ = e.visible, z = _ === void 0 ? true : _, M = e.width, $ = An(e, [ "align", "appearClickable", "children", "columnId", "elementRef", "dragId", "headCellScreenReaderText", "id", "index", "isPinned", "onAutosizeColumn", "onClick", "onKeyDown", "onRequestMoveColumn", "onRequestResize", "onSort", "resizable", "sortDir", "sortKey", "style", "tooltip", "truncate", "variant", "visible", "width" ]);
        // @docs-props-type HeadCellPropsBase
                var L = !!g;
        var W = (0, a.useSortable)({
            // dragId is optional in HeadCell but required by useSortable
            // since it is always set in Table, use a type assertion here to avoid TypeScript error
            id: d
        }), H = W.attributes, K = W.isDragging, V = W.listeners, B = W.setNodeRef;
        var F = (0, t.useCallback)((function(e) {
            N(c, e);
            if (L) {
                B(e);
            }
        }), [ L, c, B ]);
        var U = (0, t.useCallback)((function(e) {
            // ignore clicks on the resize handle
            if (e.target.getAttribute("data-test") !== "resize" && C && p !== undefined) {
                C(e, {
                    sortKey: R,
                    sortDir: k,
                    id: v,
                    columnId: s,
                    index: p
                });
            }
            h === null || h === void 0 ? void 0 : h();
        }), [ s, v, p, h, C, k, R ]);
        var X = (0, t.useCallback)((function(e) {
            if (p !== undefined) {
                y === null || y === void 0 ? void 0 : y(e, {
                    index: p,
                    columnId: s
                });
            }
        }), [ s, p, y ]);
        var Y = !!C && !!k;
        var G = h !== undefined;
        var J = q === "actions" ? false : A;
        var Q = (0, t.useMemo)((function() {
            return Ie()({}, E, {
                width: M
            });
        }), [ E, M ]);
        var Z;
        if (f) {
            Z = f;
        } else if (Re()(u)) {
            Z = u;
        }
        var ee = (n = v !== null && v !== void 0 ? v : s) !== null && n !== void 0 ? n : (0, 
        j.createDOMID)("headCellId-");
        var ne = (0, t.useMemo)((function() {
            return {
                align: i,
                clickable: l || Y || G,
                columnId: s,
                focusWithin: l,
                label: u,
                cellId: ee,
                index: p,
                onAutosizeColumn: b,
                onClick: C || G ? U : undefined,
                onKeyDown: C || G ? X : undefined,
                onRequestResize: w,
                resizable: z && S,
                sortDir: C && k,
                truncate: J,
                variant: q,
                width: M
            };
        }), [ i, l, Y, G, s, u, ee, p, b, C, U, X, w, z, S, k, J, q, M ]);
        var te = (0, t.useCallback)((function(e) {
            
            return r().createElement(Dn, Pn({}, e, ne, {
                hasTooltip: true
            }));
        }), [ ne ]);
        
        return r().createElement(Be, Pn({
            style: Q,
            ref: F,
            "aria-sort": C && (zn[k] || "none"),
            "data-test": "head-cell",
            "data-test-label": Re()(u) ? u : undefined,
            "data-test-sort-dir": C && k,
            id: z ? ee : undefined,
            $isDragging: L && K,
            $isPinned: m,
            $variant: q
        }, $),  r().createElement(Ve, null, L &&  r().createElement(Ue, Pn({
            "aria-label": (0, I.sprintf)((0, T._)("Reorder %(labelString)s"), {
                labelString: Z
            }),
            "data-test": "drag",
            draggable: true
        }, H, V),  r().createElement(Fe, null,  r().createElement(je(), null))), D ?  r().createElement(Pe(), {
            content: D,
            renderAnchor: te
        }) :  r().createElement(Dn, ne)));
    }
    Mn.propTypes = _n;
    Mn.splunkUiType = "Table.HeadCell";
    /* harmony default export */    const $n = Mn;
    // CONCATENATED MODULE: ./src/Table/HeadStyles.ts
    var Nn = z().thead.withConfig({
        displayName: "HeadStyles__Styled",
        componentId: "iap21t-0"
    })([ "", ";", "" ], M.mixins.reset("table-header-group"), (function(e) {
        var n = e.$isFixed, t = e.$zIndex;
        return n && (0, _.css)([ "position:sticky;top:0;z-index:", ";" ], t);
    }));
    var Ln = z()($n).withConfig({
        displayName: "HeadStyles__StyledToggleAll",
        componentId: "iap21t-1"
    })([ "width:32px;min-width:32px;" ]);
    var Wn = z()($n).withConfig({
        displayName: "HeadStyles__StyledInfo",
        componentId: "iap21t-2"
    })([ "width:32px;min-width:32px;" ]);
    var Hn = z()($n).withConfig({
        displayName: "HeadStyles__StyledDragRow",
        componentId: "iap21t-3"
    })([ "width:32px;" ]);
    // CONCATENATED MODULE: external "@dnd-kit/utilities"
    const Kn = require("@dnd-kit/utilities");
    // CONCATENATED MODULE: external "@splunk/react-icons/DotsThreeVertical"
    const Vn = require("@splunk/react-icons/DotsThreeVertical");
    var Bn =  e.n(Vn);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const Fn = require("@splunk/react-ui/Button");
    var Un =  e.n(Fn);
    // CONCATENATED MODULE: external "@splunk/react-ui/Dropdown"
    const Xn = require("@splunk/react-ui/Dropdown");
    var Yn =  e.n(Xn);
    // CONCATENATED MODULE: external "@splunk/react-icons/ChevronRight"
    const Gn = require("@splunk/react-icons/ChevronRight");
    var Jn =  e.n(Gn);
    // CONCATENATED MODULE: ./src/Table/ExpandButtonStyles.ts
    var Qn = z().button.withConfig({
        displayName: "ExpandButtonStyles__StyledButton",
        componentId: "sc-7nlw5q-0"
    })([ "", ";height:100%;width:100%;cursor:pointer;justify-content:center;color:", ";&:hover{color:", ";}&:focus{color:", ";box-shadow:", ";}" ], M.mixins.reset("flex"), M.variables.contentColorDefault, M.variables.contentColorActive, M.variables.contentColorActive, M.variables.focusShadowInset);
    var Zn = z()(Jn()).withConfig({
        displayName: "ExpandButtonStyles__StyledChevron",
        componentId: "sc-7nlw5q-1"
    })([ "padding-block:", ";transition:transform 200ms;", "" ], (0, M.pick)({
        comfortable: "10px",
        compact: "8px"
    }), (function(e) {
        var n = e.$expanded;
        return n && (0, _.css)([ "transform:rotate(90deg);" ]);
    }));
    // CONCATENATED MODULE: ./src/Table/ExpandButton.tsx
    function et() {
        return et = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, et.apply(null, arguments);
    }
    function nt(e, n) {
        if (null == e) return {};
        var t, r, o = tt(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function tt(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var rt = {
        expanded: w().bool
    };
    var ot = function e(n) {
        var t = n.expanded, o = nt(n, [ "expanded" ]);
        // @docs-props-type ExpandButtonPropsBase
                
        return r().createElement(Qn, et({
            title: t ? (0, T._)("Collapse row") : (0, T._)("Expand row")
        }, o),  r().createElement(Zn, {
            $expanded: t
        }));
    };
    ot.propTypes = rt;
    /* harmony default export */    const it = ot;
    // CONCATENATED MODULE: ./src/Table/RowDragCellStyles.ts
    var at = z().td.withConfig({
        displayName: "RowDragCellStyles__StyledCell",
        componentId: "sc-9kbaj2-0"
    })([ "", " cursor:move;touch-action:none;position:relative;vertical-align:middle;" ], M.mixins.reset("table-cell"));
    var lt = z().div.withConfig({
        displayName: "RowDragCellStyles__StyledDrag",
        componentId: "sc-9kbaj2-1"
    })([ "", " ", ";justify-content:center;cursor:move;-webkit-user-drag:element;user-select:none;align-items:center;position:absolute;width:100%;top:0;bottom:0;&:focus{outline:0;box-shadow:", ";}&:hover{background-color:", ";outline:none;}&:active{background-color:", ";box-shadow:", ";}& > div{margin:0;}@media print{background-image:none;}" ], M.mixins.reset("flex"), M.mixins.clearfix(), M.variables.focusShadowInset, M.variables.interactiveColorOverlayDrag, M.variables.interactiveColorOverlayActive, M.variables.focusShadowInset);
    // CONCATENATED MODULE: ./src/Table/RowDragCell.tsx
    function ut() {
        return ut = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, ut.apply(null, arguments);
    }
    function st(e, n) {
        if (null == e) return {};
        var t, r, o = ct(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function ct(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var dt = {
        setActivatorNodeRef: w().func.isRequired,
        listeners: w().objectOf(w().func)
    };
    function ft(e) {
        var n = e.listeners, t = e.setActivatorNodeRef, o = st(e, [ "listeners", "setActivatorNodeRef" ]);
        // @docs-props-type RowDragCellPropsBase
                
        return r().createElement(at, {
            "data-helper": true
        },  r().createElement(lt, ut({
            "data-test": "drag",
            draggable: true,
            ref: t,
            tabIndex: 0
        }, n, o),  r().createElement(je(), null)));
    }
    ft.propTypes = dt;
    /* harmony default export */    const vt = ft;
    // CONCATENATED MODULE: ./src/Table/RowStyles.ts
    var pt = z().tr.withConfig({
        displayName: "RowStyles__StyledStripeNone",
        componentId: "f0igqq-0"
    })([ "", ";", ";", " @media print{background-color:none;}" ], M.mixins.reset("table-row"), (function(e) {
        var n = e.$clickable, t = e.$isActionColumnPinned;
        return n && (0, _.css)([ "cursor:pointer;&:hover{background-color:", ";", "}&:focus{background-color:", ";position:relative;&::after{content:'';display:block;position:absolute;left:0;right:0;top:0;bottom:0;box-shadow:", ";pointer-events:none;}", "}" ], M.variables.neutral100, t && (0, 
        _.css)([ "td[data-test='row-actions']{background-color:inherit;}" ]), M.variables.neutral100, M.variables.focusShadowInset, t && (0, 
        _.css)([ "td[data-test='row-actions']{background-color:inherit;}" ]));
    }), (function(e) {
        var n = e.disabled, t = e.$expandable, r = e.$isDraggable;
        return !n && !t && r && (0, _.css)([ "&:hover{background-color:", ";}" ], M.variables.interactiveColorOverlayHover);
    }));
    var mt = z()(pt).withConfig({
        displayName: "RowStyles__StyledStripeEven",
        componentId: "f0igqq-1"
    })([ "background-color:", ";", "" ], M.variables.neutral50, (function(e) {
        var n = e.$isActionColumnPinned;
        return n && (0, _.css)([ "td[data-test='row-actions']{background-color:inherit;}" ]);
    }));
    // CONCATENATED MODULE: external "@splunk/react-ui/Checkbox"
    const bt = require("@splunk/react-ui/Checkbox");
    var ht =  e.n(bt);
    // CONCATENATED MODULE: external "@splunk/react-ui/Clickable"
    const yt = require("@splunk/react-ui/Clickable");
    var gt =  e.n(yt);
    // CONCATENATED MODULE: ./src/Table/ToggleStyles.ts
    var wt = z()(gt()).withConfig({
        displayName: "ToggleStyles__StyledClickable",
        componentId: "sc-1x3ikch-0"
    })([ "height:100%;width:100%;display:flex;justify-content:center;&[disabled]{cursor:not-allowed;}" ]);
    var Ct = z()(ht()).withConfig({
        displayName: "ToggleStyles__StyledCheckbox",
        componentId: "sc-1x3ikch-1"
    })([ "align-items:flex-start;padding-block:", ";" ], (0, M.pick)({
        comfortable: "8px",
        compact: "6px"
    }));
    // CONCATENATED MODULE: ./src/Table/Toggle.tsx
    function xt() {
        return xt = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, xt.apply(null, arguments);
    }
    function St(e, n) {
        if (null == e) return {};
        var t, r, o = Ot(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function Ot(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var kt = {
        disabled: w().bool,
        onClick: w().func,
        selected: w().oneOf([ true, false, "some" ])
    };
    function Rt(e) {
        var n = e.disabled, t = e.onClick, o = e.selected, i = St(e, [ "disabled", "onClick", "selected" ]);
        // @docs-props-type TogglePropsBase
                var a = n === true ? "disabled" : undefined;
        
        // we use a Clickable around the inert Checkbox (which can seem ironic) so that the whole table cell is clickable
        // we cannot use the Cell's onClick itself because of a11y issues we have faced in the past (see SUI-5958).
        return r().createElement(wt, xt({
            "aria-checked": o === "some" ? "mixed" : o,
            disabled: a,
            onClick: t,
            role: "checkbox"
        }, i),  r().createElement(Ct, {
            checked: o === "some" ? "indeterminate" : o,
            disabled: !!a,
            inert: true
        }));
    }
    Rt.propTypes = kt;
    /* harmony default export */    const Et = Rt;
    // CONCATENATED MODULE: ./src/Table/Row.tsx
    function It(e) {
        "@babel/helpers - typeof";
        return It = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, It(e);
    }
    function Tt(e, n) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            n && (r = r.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), t.push.apply(t, r);
        }
        return t;
    }
    function jt(e) {
        for (var n = 1; n < arguments.length; n++) {
            var t = null != arguments[n] ? arguments[n] : {};
            n % 2 ? Tt(Object(t), !0).forEach((function(n) {
                Dt(e, n, t[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : Tt(Object(t)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(t, n));
            }));
        }
        return e;
    }
    function Dt(e, n, t) {
        return (n = Pt(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function Pt(e) {
        var n = At(e, "string");
        return "symbol" == It(n) ? n : n + "";
    }
    function At(e, n) {
        if ("object" != It(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != It(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function qt(e, n) {
        if (null == e) return {};
        var t, r, o = _t(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function _t(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    function zt() {
        return zt = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, zt.apply(null, arguments);
    }
    var Mt = {
        odd: pt,
        even: mt,
        none: pt
    };
    /** @public */
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var $t = {
        /** @private. Generally passed by Table rather than added directly. */
        activeElementId: w().string,
        actionPrimary: w().element,
        actionsSecondary: w().element,
        children: w().node,
        data: w().any,
        disabled: w().bool,
        /** @private. Generally passed by Table rather than added directly. */
        isDraggable: w().bool,
        /** @private. Generally passed by Table rather than added directly. */
        dragId: w().oneOfType([ w().string, w().number ]),
        elementRef: w().oneOfType([ w().func, w().object ]),
        /** @private. */
        expandable: w().bool,
        expanded: w().bool,
        /** @private. Identifies the unique id (or ids) for an expanded row (or rows). In expandable rows `ExpansionRowIcon` uses `expansionRowId` for accessibility aria attributes. */
        expansionRowIds: w().string,
        expansionRow: w().oneOfType([ w().element, w().arrayOf(w().element) ]),
        /** @private. Indicates whether the table has an actions column. */
        actions: w().bool,
        /** @private. Generally passed by `Table` rather than added directly. */
        index: w().number,
        /** @private. Generally passed by `Table` rather than added directly. */
        movableColumns: w().bool,
        onClick: w().func,
        onExpansion: w().func,
        /** @private. This is passed through and works as expected. */
        onKeyDown: w().func,
        /** @private. Generally passed by `Table` rather than added directly. */
        onRequestMoveRow: w().func,
        onRequestToggle: w().func,
        /** @private. */
        pinnedColumns: w().object,
        /** @private. Generally passed by `Table` rather than added directly. */
        primaryColumnIndex: w().number,
        rowScreenReaderText: w().string,
        selected: w().bool,
        /** @private. */
        stripe: w().oneOf([ "odd", "even", "none" ])
    };
    var Nt =  r().createElement(Bn(), null);
    var Lt =  r().forwardRef((function(e, n) {
        
        return r().createElement(Pe(), zt({
            content: (0, T._)("Actions"),
            contentRelationship: "label"
        }, e),  r().createElement(Un(), {
            appearance: "subtle",
            "data-test": "actions-secondary-toggle",
            icon: Nt,
            elementRef: n
        }));
    }));
    function Wt(e) {
        var n;
        var o = e.actions, i = e.actionPrimary, l = e.actionsSecondary, u = e.children, s = e.data, c = e.dragId, d = e.disabled, f = e.pinnedColumns, v = e.isDraggable, p = e.elementRef, m = e.expandable, h = e.expanded, y = e.expansionRowIds, g = e.expansionRow, w = e.index, C = e.movableColumns, x = e.onClick, S = e.onExpansion, O = e.onKeyDown, k = e.onRequestToggle, R = e.primaryColumnIndex, E = R === void 0 ? 0 : R, D = e.rowScreenReaderText, P = e.selected, A = e.stripe, q = A === void 0 ? "none" : A, _ = e.style, z = qt(e, [ "actions", "actionPrimary", "actionsSecondary", "children", "data", "dragId", "disabled", "pinnedColumns", "isDraggable", "elementRef", "expandable", "expanded", "expansionRowIds", "expansionRow", "index", "movableColumns", "onClick", "onExpansion", "onKeyDown", "onRequestToggle", "primaryColumnIndex", "rowScreenReaderText", "selected", "stripe", "style" ]);
 // @docs-props-type RowPropsBase
                var M = (0, t.useMemo)((function() {
            return {
                id: v ? c : (0, j.createGUID)(),
                disabled: !v
            };
        }), [ v, c ]);
        var $ = (0, a.useSortable)(M), L = $.attributes, W = $.listeners, H = $.transform, K = $.transition, V = $.setNodeRef, B = $.setActivatorNodeRef;
        var F = (0, t.useMemo)((function() {
            return v ? jt(jt({}, _), {}, {
                transform: Kn.CSS.Translate.toString(H),
                transition: K
            }) : _;
        }), [ v, H, K, _ ]);
        var U = (0, t.useRef)(null);
        var X = (0, t.useCallback)((function(e) {
            var n = e.target;
            while (n) {
                if (U.current === n) {
                    break;
                }
                // Ignore clicks on row-actions
                                if (n.tagName === "BUTTON" || n.tagName === "A") {
                    return;
                }
                n = n.parentNode;
            }
            if (!e.defaultPrevented) {
                x === null || x === void 0 ? void 0 : x(e, s);
            }
        }), [ s, x ]);
        var Y = (0, t.useCallback)((function(e) {
            if (e.key === "Enter" && x) {
                x(e, s);
            }
            O === null || O === void 0 ? void 0 : O(e);
        }), [ s, x, O ]);
        var G = (0, t.useCallback)((function(e) {
            if (!d) {
                e.preventDefault();
                k === null || k === void 0 ? void 0 : k(e, s);
            }
        }), [ s, d, k ]);
        var J = (0, t.useCallback)((function(e) {
            e.preventDefault();
            S === null || S === void 0 ? void 0 : S(e, s);
        }), [ s, S ]);
        var Q = function n() {
            if (i && i.props.onClick) {
                
                return (0, t.cloneElement)(i, {
                    onClick: function n(t) {
                        var r, o;
                        t.preventDefault();
                        return (r = (o = i.props).onClick) === null || r === void 0 ? void 0 : r.call(o, t, e.data);
                    }
                });
            }
            return i;
        };
        var Z = function n() {
            if (l && l.props.children) {
                var r = t.Children.toArray(l.props.children).filter(t.isValidElement);
                var o = r.map((function(n) {
                    if (n.props.onClick) {
                        
                        return (0, t.cloneElement)(n, {
                            onClick: function t(r) {
                                var o, i;
                                r.preventDefault();
                                return (o = (i = n.props).onClick) === null || o === void 0 ? void 0 : o.call(i, r, e.data);
                            }
                        });
                    }
                    return n;
                }));
                
                return (0, t.cloneElement)(l, {
                    children: o
                });
            }
            return l;
        };
        function ee(e) {
            U.current = e;
            N(p, e);
            if (v) {
                V(e);
            }
        }
        var ne = t.Children.toArray(u).map((function(e, n) {
            
            return (0, t.cloneElement)(e, {
                appearance: x ? "rowLink" : e.props.appearance,
                disabled: d,
                isPrimaryCell: E === n,
                stripe: q
            });
        }));
        var te;
        if (D) {
            te = D;
        } else if ((n = ne[E]) === null || n === void 0 ? void 0 : n.props.children) {
            var re;
            te = "".concat((re = ne[E]) === null || re === void 0 ? void 0 : re.props.children);
        } else {
            var oe;
            te = "".concat((oe = ne[0]) === null || oe === void 0 ? void 0 : oe.props.children);
            if (false) {}
        }
        var ie = (0, I.sprintf)((0, T._)("%(selected)s"), {
            selected: P ? "selected" : "unselected"
        });
        var ae = Mt[q];
        var le = h ? t.Children.count(g) + 1 : undefined;
        var ue = C ? "false" : undefined;
        var se = function e() {
            
            return r().createElement(Lt, null);
        };
        var ce = (0, I.sprintf)((0, T._)("Collapse row - %(rowLabelValue)s"), {
            rowLabelValue: te
        });
        var de = (0, I.sprintf)((0, T._)("Expand row - %(rowLabelValue)s"), {
            rowLabelValue: te
        });
        
        return r().createElement(ae, zt({
            "data-test": "row",
            "data-test-selected": k ? P : undefined,
            $clickable: !!x,
            "data-has-movable-columns": C ? "true" : undefined,
            disabled: !!d,
            $expandable: !!m,
            $isActionColumnPinned: f === null || f === void 0 ? void 0 : f.actions,
            $isDraggable: !!v,
            ref: ee,
            tabIndex: x ? 0 : undefined
        }, b()(z, "data"), {
            onClick: x ? X : undefined,
            onKeyDown: Y,
            $rowSelected: !!(k && P),
            style: F
        }), v && w != null &&  r().createElement(vt, zt({
            "aria-label": (0, I.sprintf)((0, T._)("Reorder %(rowLabelValue)s"), {
                rowLabelValue: te
            }),
            setActivatorNodeRef: B,
            listeners: W
        }, L)), k &&  r().createElement(Oe, {
            align: "center",
            appearance: x ? "rowLink" : "data",
            disabled: d,
            variant: "toggle",
            "data-movable-column": ue,
            rowSpan: le,
            "data-test": "toggle"
        },  r().createElement(Et, {
            "aria-label": (0, I.sprintf)((0, T._)("%(rowLabelValue)s %(selectedString)s"), {
                rowLabelValue: te,
                selectedString: ie
            }),
            disabled: d,
            onClick: G,
            selected: P
        })), m && g &&  r().createElement(Oe, {
            align: "center",
            appearance: "data",
            "data-movable-column": ue,
            "data-test": "expand",
            expand: h,
            rowSpan: le,
            variant: "expand"
        },  r().createElement(it, {
            expanded: h,
            onClick: J,
            title: h ? ce : de,
            "aria-controls": h ? y : undefined,
            "aria-expanded": h
        })), m && !g &&  r().createElement(Oe, {
            variant: "expand",
            "data-movable-column": ue
        }), ne, o && !i && !l &&  r().createElement(Oe, {
            align: "right",
            appearance: x ? "rowLink" : "data",
            "data-test": "row-actions",
            variant: "actions",
            "data-movable-column": ue,
            isPinned: f === null || f === void 0 ? void 0 : f.actions
        }), (i || l) &&  r().createElement(Oe, {
            align: "right",
            appearance: "data",
            "data-test": "row-actions",
            variant: "actions",
            "data-movable-column": ue,
            isPinned: f === null || f === void 0 ? void 0 : f.actions
        }, i && Q(), l &&  r().createElement(Yn(), {
            toggle: se()
        }, Z())));
    }
    Wt.propTypes = $t;
    Wt.splunkUiType = "Table.Row";
    /* harmony default export */    const Ht = Wt;
    // CONCATENATED MODULE: ./src/Table/Head.tsx
    function Kt() {
        return Kt = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, Kt.apply(null, arguments);
    }
    function Vt(e, n) {
        if (null == e) return {};
        var t, r, o = Bt(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function Bt(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var Ft = {
        /** @private. */
        actions: w().arrayOf(w().node),
        /** @private. */
        actionsColumnWidth: w().number,
        /** @private */
        cellMinWidth: w().number,
        children: w().node,
        elementRef: w().oneOfType([ w().func, w().object ]),
        /** @private. */
        hasDragColumn: w().bool,
        /** @private. */
        hasInfoColumn: w().bool,
        /** @private. */
        onAutosizeColumn: w().func,
        /** @private. */
        onRequestMoveColumn: w().func,
        /** @private. */
        onRequestResizeColumn: w().func,
        /** @private. */
        onRequestToggleAllRows: w().func,
        /** @private. */
        pinnedColumns: w().object,
        /** @private. */
        rowSelection: w().oneOf([ "all", "some", "none" ])
    };
    var Ut = 1;
    var Xt = {
        all: true,
        none: false,
        some: "some"
    };
    var Yt = (0, T._)("No rows selected");
    var Gt = (0, T._)("Some rows selected");
    var Jt = (0, T._)("All rows selected");
    var Qt = function e(n) {
        var t, r;
        var o = cn();
        if (!n) {
            return [];
        }
        return Array.from((t = (r = n.firstElementChild) === null || r === void 0 ? void 0 : r.children) !== null && t !== void 0 ? t : []).map((function(e) {
            var n = o.getComputedStyle(e);
            return e.clientWidth - parseFloat(n.getPropertyValue("padding-right")) - parseFloat(n.getPropertyValue("padding-left"));
        }));
    };
    function Zt(e) {
        var n = e.actions, o = n === void 0 ? [] : n, i = e.actionsColumnWidth, a = e.cellMinWidth, l = a === void 0 ? 60 : a, u = e.children, s = e.elementRef, c = e.hasDragColumn, d = e.hasInfoColumn, f = e.pinnedColumns, v = e.onAutosizeColumn, p = e.onRequestMoveColumn, m = e.onRequestResizeColumn, b = e.onRequestToggleAllRows, h = e.rowSelection, y = Vt(e, [ "actions", "actionsColumnWidth", "cellMinWidth", "children", "elementRef", "hasDragColumn", "hasInfoColumn", "pinnedColumns", "onAutosizeColumn", "onRequestMoveColumn", "onRequestResizeColumn", "onRequestToggleAllRows", "rowSelection" ]);
        // @docs-props-type HeadPropsBase
                var g = (0, t.useContext)(ye), w = g.headType, C = g.tableContainerWidthWithoutBorders, x = g.resizableFillLayout;
        var S = (0, t.useRef)(null);
        var O = (0, t.useRef)([]);
        (0, t.useEffect)((function() {
            // Validate children width
            if (false) {
                var e;
            }
        }), [ u, m, x ]);
        var k = (0, t.useCallback)((function(e) {
            var n = e.fromIndex, r = e.toIndex;
            if (r < t.Children.toArray(u).length) {
                // HeadCell never requests less than zero
                p === null || p === void 0 ? void 0 : p({
                    fromIndex: n,
                    toIndex: r
                });
            }
        }), [ u, p ]);
        var R = (0, t.useCallback)((function(e, n) {
            var t = n.columnId, r = n.id, o = n.index, i = n.width;
            var a = Qt(S.current);
            if (o >= a.length) {
                m === null || m === void 0 ? void 0 : m(e, n);
                return;
            }
            var u = a[o];
            var s = i - u;
            if (s <= 0) {
                // handle shrinking
                if (i >= l) {
                    m === null || m === void 0 ? void 0 : m(e, n);
                } else {
                    // shrink the cell as small as we can
                    m === null || m === void 0 ? void 0 : m(e, {
                        columnId: t,
                        id: r,
                        index: o,
                        width: l
                    });
                }
            } else {
                // handle expansion
                var c = [];
                var d = 0;
                var f = true;
                if (O.current.length !== a.length) {
                    m === null || m === void 0 ? void 0 : m(e, n);
                    return;
                }
                O.current.forEach((function(e, n) {
                    if (e === "auto") {
                        d += 1;
                        c.push(n);
                    }
                }));
                if (d > 0) {
                    // if there are auto width cells, check if shrinking them by changePerEachAutoCell make them smaller than cellMinWidth
                    var v = s / d;
                    c.every((function(e) {
                        if (e < a.length && a[e] - v < l) {
                            f = false;
                            return false;
                        }
                        return true;
                    }));
                    if (f) {
                        m === null || m === void 0 ? void 0 : m(e, n);
                    } else {
                        // make the cell as big as we can without letting any cells go below cellMinWidth
                        // that means all auto cells are at cellMinWidth and all the set width cells stay the same size
                        var p = 0;
                        c.forEach((function(e) {
                            if (e < a.length) {
                                p += Math.max(a[e] - l, 0);
                            }
                        }));
                        var b = a[o] + p;
                        m === null || m === void 0 ? void 0 : m(e, {
                            columnId: t,
                            id: r,
                            index: o,
                            width: b
                        });
                    }
                } else if (C != null) {
                    // if there are no auto width cells, check if shrinking the last cell by requested change makes it smaller than cellMinWidth
                    /**
         * We need to calculate the percieved width of the last cell, because if the user modifies the size of the last head cell,
         * the width of the last cell will technically be whatever they made it, but the last cell will get cut off if it extends past
         * the table container's width.
         */
                    var h = a.slice(0, a.length - 1);
                    var y = h.reduce((function(e, n) {
                        return e + n;
                    }));
                    var g = C - y;
                    if (g - s >= l) {
                        m === null || m === void 0 ? void 0 : m(e, n);
                    } else {
                        // make the cell as big as we can without letting perceivedWidthOfLastCell go below cellMinWidth
                        var w = g - l;
                        var x = u + w;
                        m === null || m === void 0 ? void 0 : m(e, {
                            columnId: t,
                            id: r,
                            index: o,
                            width: x
                        });
                    }
                } else {
                    /**
         * If tableContainerWidthWithoutBorders == null, there isn't a way to correctly calculate
         * the perceived width so this case will result in columns possibly going offscreen. In testing,
         * tableContainerWidthWithoutBorders attains a value quickly enough that it isn't a problem.
         */
                    m === null || m === void 0 ? void 0 : m(e, n);
                }
            }
        }), [ l, m, C ]);
        var E = (0, t.useCallback)((function(e) {
            N(s, e);
            S.current = e;
        }), [ s ]);
        var I = (0, t.useMemo)((function() {
            return t.Children.toArray(u).filter(t.isValidElement);
        }), [ u ]);
        var T = I.length + (o.length || i ? 1 : 0);
        var j = (0, t.useMemo)((function() {
            var e = [];
            var n = I.map((function(n, r) {
                e.push(n.props.width);
                
                return (0, t.cloneElement)(n, {
                    index: r,
                    onRequestResize: x ? R : m,
                    onRequestMoveColumn: p ? k : undefined,
                    onAutosizeColumn: v,
                    key: n.key || n.props.columnId || r
                });
            }));
            O.current = e;
            return n;
        }), [ I, x, R, m, p, k, v ]);
        var D = h === undefined ? undefined : Xt[h];
        var P = i || 48;
        var A = w === "fixed" ? "fixed-head" : "head";
        var q;
        if (D === "some") {
            q = Gt;
        } else if (D) {
            q = Jt;
        } else {
            q = Yt;
        }
        var _ = (0, t.useMemo)((function() {
            return {
                minWidth: P
            };
        }), [ P ]);
        var z = (0, t.useMemo)((function() {
            return {
                minWidth: i
            };
        }), [ i ]);
        
        return r().createElement(Nn, Kt({
            ref: E,
            "data-test": A,
            $isFixed: w === "fixed",
            $zIndex: T + Ut
        }, y),  r().createElement(Ht, null, c &&  r().createElement(Hn, {
            key: "drag_rows_head_cell",
            "data-test": "drag-rows-head-cell",
            align: "center",
            resizable: false
        }), b &&  r().createElement(Ln, {
            key: "toggle_all",
            resizable: false,
            align: "center",
            variant: "toggleAll",
            truncate: false,
            "data-test": "toggle-all"
        },  r().createElement(Et, {
            "aria-label": q,
            onClick: b,
            selected: D
        })), d &&  r().createElement(Wn, {
            key: "more_info_head_cell",
            "data-test": "more-info-head-cell",
            resizable: false,
            variant: "info"
        }), j, o.length > 0 &&  r().createElement($n, {
            key: "actions_head_cell",
            "data-test": "actions-head-cell",
            align: "right",
            resizable: false,
            width: P,
            style: _,
            variant: "actions",
            isPinned: f === null || f === void 0 ? void 0 : f.actions
        }, o), o.length === 0 && i &&  r().createElement($n, {
            key: "actions_head_cell",
            "data-test": "actions-head-cell",
            align: "right",
            resizable: false,
            width: i,
            style: z,
            variant: "actions",
            isPinned: f === null || f === void 0 ? void 0 : f.actions
        }, " ")));
    }
    Zt.propTypes = Ft;
    Zt.splunkUiType = "Table.Head";
    /* harmony default export */    const er = Zt;
    // CONCATENATED MODULE: external "lodash/isFinite"
    const nr = require("lodash/isFinite");
    var tr =  e.n(nr);
    // CONCATENATED MODULE: external "lodash/isFunction"
    const rr = require("lodash/isFunction");
    var or =  e.n(rr);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const ir = require("@splunk/react-ui/Popover");
    var ar =  e.n(ir);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const lr = require("@splunk/react-ui/useControlled");
    var ur =  e.n(lr);
    // CONCATENATED MODULE: ./src/Table/HeadDropdownCell.tsx
    function sr() {
        return sr = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, sr.apply(null, arguments);
    }
    function cr(e, n) {
        return mr(e) || pr(e, n) || fr(e, n) || dr();
    }
    function dr() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function fr(e, n) {
        if (e) {
            if ("string" == typeof e) return vr(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? vr(e, n) : void 0;
        }
    }
    function vr(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function pr(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, l = [], u = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = i.call(t)).done) && (l.push(r.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function mr(e) {
        if (Array.isArray(e)) return e;
    }
    function br(e, n) {
        if (null == e) return {};
        var t, r, o = hr(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function hr(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    /** @public */
    /** @public */
    /** @public */    var yr = [ "clickAway", "contentClick", "escapeKey", "offScreen", "tabKey", "toggleClick" ];
    var gr = {
        align: w().oneOf([ "left", "center", "right" ]),
        buttonRef: w().oneOfType([ w().func, w().object ]),
        canCoverHead: w().bool,
        children: w().node.isRequired,
        closeReasons: w().arrayOf(w().oneOf(yr)),
        columnId: w().string,
        /** @private. */
        dragId: w().oneOfType([ w().string, w().number ]),
        elementRef: w().oneOfType([ w().func, w().object ]),
        headCellScreenReaderText: w().string,
        focusToggleReasons: w().arrayOf(w().oneOf(yr)),
        /** @private The index of the cell, skipping the info column. */
        index: w().number,
        label: w().node,
        /** @private. */
        onAutosizeColumn: w().func,
        /**
   * @private. This will be passed through, and will work as expected.
   */
        onKeyDown: w().func,
        onRequestClose: w().func,
        /** @private. */
        onRequestMoveColumn: w().func,
        onRequestOpen: w().func,
        /** @private. */
        onRequestResize: w().func,
        open: w().bool,
        repositionMode: w().oneOf([ "none", "flip", "any" ]),
        resizable: w().bool,
        retainFocus: w().bool,
        takeFocus: w().bool,
        truncate: w().bool,
        /**
   * @private
   * Used internally to suppress focus when this `HeadDropdownCell` is superseded by one in an
   * overlaid `HeadTable` for user interactions.
   */
        visible: w().bool,
        width: w().number,
        /** @private. */
        variant: w().oneOf([ "toggleAll", "info", "actions" ])
    };
    var wr = [ "contentClick", "escapeKey", "toggleClick" ];
    function Cr(e) {
        var n;
        var o = e.align, i = o === void 0 ? "left" : o, l = e.buttonRef, u = e.canCoverHead, s = u === void 0 ? true : u, c = e.children, d = e.closeReasons, f = d === void 0 ? yr : d, v = e.columnId, p = e.dragId, m = e.elementRef, b = e.headCellScreenReaderText, h = e.focusToggleReasons, y = h === void 0 ? wr : h, g = e.id, w = e.index, C = e.label, x = e.onAutosizeColumn, S = e.onKeyDown, O = e.onRequestClose, R = e.onRequestMoveColumn, E = e.onRequestOpen, D = e.onRequestResize, A = e.open, q = e.repositionMode, _ = q === void 0 ? "flip" : q, z = e.resizable, M = z === void 0 ? true : z, $ = e.retainFocus, L = e.style, W = e.takeFocus, H = W === void 0 ? true : W, K = e.truncate, V = K === void 0 ? true : K, B = e.variant, F = e.visible, U = F === void 0 ? true : F, X = e.width, Y = br(e, [ "align", "buttonRef", "canCoverHead", "children", "closeReasons", "columnId", "dragId", "elementRef", "headCellScreenReaderText", "focusToggleReasons", "id", "index", "label", "onAutosizeColumn", "onKeyDown", "onRequestClose", "onRequestMoveColumn", "onRequestOpen", "onRequestResize", "open", "repositionMode", "resizable", "retainFocus", "style", "takeFocus", "truncate", "variant", "visible", "width" ]);
        // @docs-props-type HeadDropdownCellPropsBase
                var G = ur()({
            componentName: "HeadDropdownCell",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            valuePropName: "open"
        });
        var J = (0, t.useState)(), Q = cr(J, 2), Z = Q[0], ee = Q[1];
        var ne = (0, t.useState)(null), te = cr(ne, 2), re = te[0], oe = te[1];
        var ie = (0, t.useState)(null), ae = cr(ie, 2), le = ae[0], ue = ae[1];
        var se = (0, t.useState)(false), ce = cr(se, 2), de = ce[0], fe = ce[1];
        var ve = (0, j.createDOMID)("popover");
        var pe = !!R;
        var me = (0, a.useSortable)({
            // dragId is optional in HeadCell but required by useSortable
            // since it is always set in Table, use a type assertion here to avoid TypeScript error
            id: p
        }), be = me.attributes, he = me.listeners, ye = me.setNodeRef, ge = me.isDragging;
        var we = (0, t.useMemo)((function() {
            return G ? A : de;
        }), [ G, de, A ]);
        var Ce = k()(U);
        var xe = (n = g !== null && g !== void 0 ? g : v) !== null && n !== void 0 ? n : (0, 
        j.createDOMID)("headCellId-");
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ D, V ]);
        var Se = (0, t.useCallback)((function(e) {
            var n = e.reason, t = e.event;
            if (n === "clickAway" && t !== undefined) {
                var r = t.target;
                while (r) {
                    // Ignore clicks on toggle.
                    if (r === le) {
                        return;
                    }
                    r = r.parentNode;
                }
            }
            if (we && P()(f, n) && w !== undefined) {
                if (P()(y, n)) {
                    le === null || le === void 0 ? void 0 : le.focus();
                }
                if (!G) {
                    fe(false);
                }
                O === null || O === void 0 ? void 0 : O(t, {
                    index: w,
                    reason: n,
                    columnId: v
                });
            }
        }), [ f, v, y, w, le, G, we, O ]);
        (0, t.useEffect)((function() {
            if (Ce && !U) {
                Se({
                    reason: "offScreen"
                });
            }
        }), [ Se, Ce, U ]);
        var Oe = (0, t.useCallback)((function(e) {
            N(m, e);
            if (pe) {
                ye(e);
            }
        }), [ pe, m, ye ]);
        var ke = (0, t.useCallback)((function(e) {
            ue(e);
            oe(e);
            N(l, e);
        }), [ l ]);
        var Re = (0, t.useCallback)((function(e) {
            e.stopPropagation();
            // ignore clicks on the resize handle
                        if (e.target.getAttribute("data-test") === "resize") {
                return;
            }
            if (!we) {
                ee(e.clientX || undefined);
            }
            if (we) {
                Se === null || Se === void 0 ? void 0 : Se({
                    reason: "toggleClick",
                    event: e
                });
            } else if (w !== undefined) {
                E === null || E === void 0 ? void 0 : E(e, {
                    reason: "toggleClick",
                    columnId: v,
                    index: w
                });
                if (!G) {
                    fe(true);
                }
            }
        }), [ v, Se, w, G, we, E ]);
        var Ee = (0, t.useCallback)((function(e) {
            if (w !== undefined) {
                S === null || S === void 0 ? void 0 : S(e, {
                    index: w,
                    columnId: v
                });
            }
        }), [ v, w, S ]);
        var Te = (0, t.useCallback)((function(e) {
            Se({
                reason: "contentClick",
                event: e
            });
        }), [ Se ]);
        var De = (0, t.useMemo)((function() {
            if (Z !== undefined && tr()(Z)) {
                var e;
                return {
                    x: Z - ((e = re === null || re === void 0 ? void 0 : re.getBoundingClientRect().left) !== null && e !== void 0 ? e : 0)
                };
            }
            return undefined;
        }), [ Z, re ]);
        var Pe = (0, t.useMemo)((function() {
            return Ie()({}, L, {
                width: X
            });
        }), [ L, X ]);
        var Ae;
        if (b) {
            Ae = b;
        } else if (typeof C === "string") {
            Ae = C;
        }
        
        return r().createElement(Be, sr({
            style: Pe,
            "data-test": "head-cell",
            $isDragging: pe && ge,
            $variant: B,
            id: U ? xe : undefined,
            ref: Oe
        }, Y),  r().createElement(Ve, null, pe &&  r().createElement(Ue, sr({
            "aria-label": (0, I.sprintf)((0, T._)("Reorder %(labelString)s"), {
                labelString: Ae
            }),
            "data-test": "drag"
        }, be, he),  r().createElement(Fe, null,  r().createElement(je(), null))),  r().createElement(Dn, {
            align: i,
            "aria-expanded": we,
            "aria-haspopup": true,
            "aria-owns": ve,
            clickable: true,
            columnId: v,
            elementRef: ke,
            cellId: xe,
            index: w,
            isMenu: true,
            label: C,
            resizable: U && M,
            onAutosizeColumn: x,
            onRequestResize: D,
            onClick: Re,
            onKeyDown: Ee,
            truncate: V,
            width: X,
            "data-test": "inner"
        }),  r().createElement(ar(), {
            align: "center",
            anchor: re,
            "aria-labelledby": xe,
            autoCloseWhenOffScreen: P()(f, "offScreen"),
            canCoverAnchor: s,
            defaultPlacement: "below",
            id: ve,
            onRequestClose: Se,
            open: !!re && we,
            pointTo: De,
            repositionMode: _,
            retainFocus: $,
            takeFocus: H
        }, or()(c) ? function() {
            
            // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
            return r().createElement("div", {
                onClick: Te
            }, c.apply(void 0, arguments));
        } : 
        
        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
        r().createElement("div", {
            onClick: Te
        }, c))));
    }
    Cr.propTypes = gr;
    Cr.splunkUiType = "Table.HeadDropdownCell";
    Cr.possibleCloseReasons = yr;
    /* harmony default export */    const xr = Cr;
    // CONCATENATED MODULE: external "lodash/extend"
    const Sr = require("lodash/extend");
    var Or =  e.n(Sr);
    // CONCATENATED MODULE: ./src/Table/HeadTableStyles.ts
    var kr = z()(ee).withConfig({
        displayName: "HeadTableStyles__Styled",
        componentId: "g85wgf-0"
    })([ "min-width:0;table-layout:fixed;" ]);
    var Rr = z().div.withConfig({
        displayName: "HeadTableStyles__StyledDockedContainer",
        componentId: "g85wgf-1"
    })([ "overflow:hidden;position:fixed;z-index:", ";@media print{display:none;}" ], M.variables.zindexFixedNavbar);
    var Er = z()(Rr).withConfig({
        displayName: "HeadTableStyles__StyledFixedContainer",
        componentId: "g85wgf-2"
    })([ "position:absolute;box-shadow:none;z-index:1;" ]);
    // CONCATENATED MODULE: ./src/Table/HeadTable.tsx
    function Ir() {
        return Ir = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, Ir.apply(null, arguments);
    }
    function Tr(e, n) {
        if (null == e) return {};
        var t, r, o = jr(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function jr(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    var Dr = {
        cellWidths: w().arrayOf(w().number).isRequired,
        dragIndex: w().number,
        elementRef: w().oneOfType([ w().func, w().object ]),
        hasDragColumn: w().bool,
        hasRowExpansion: w().bool,
        hasRowSelection: w().bool,
        headType: w().oneOf([ "docked", "fixed" ]).isRequired,
        horizontalOffset: w().number,
        isFixedColumn: w().bool,
        rowSelection: w().oneOf([ "all", "some", "none" ]),
        tableStyle: w().object,
        tableWidth: w().number.isRequired,
        tHead: w().element.isRequired,
        top: w().number.isRequired,
        width: w().number.isRequired
    };
    /**
 * @private
 * HeadTable is an internal component that provides a detached table head that can
 * then be positioned in different ways outside of the main table.
 */    function Pr(e) {
        var n = e.cellWidths, o = e.elementRef, i = e.hasDragColumn, a = e.hasRowExpansion, l = e.hasRowSelection, u = e.headType, s = e.horizontalOffset, c = s === void 0 ? 0 : s, d = e.isFixedColumn, f = e.tableStyle, v = e.tableWidth, p = e.tHead, m = e.top, h = e.width, y = Tr(e, [ "cellWidths", "elementRef", "hasDragColumn", "hasRowExpansion", "hasRowSelection", "headType", "horizontalOffset", "isFixedColumn", "tableStyle", "tableWidth", "tHead", "top", "width" ]);
        // @docs-props-type HeadTablePropsBase
                var g = u === "docked";
        var w = g ? Rr : Er;
        var C = (0, t.useRef)(null);
        var x = (0, t.useCallback)((function(e) {
            N(C, e);
        }), [ C ]);
        var S = 0;
        if (a) {
            S += 1;
        }
        if (l) {
            S += 1;
        }
        if (i) {
            S += 1;
        }
        // Set the widths of the head cells if not fixed column
                var O = (0, t.useMemo)((function() {
            return t.Children.toArray(p.props.children);
        }), [ p.props.children ]);
        var k = (0, t.useMemo)((function() {
            if (d) {
                // ensure consistent keys
                return O;
            }
            return O.filter(t.isValidElement).map((function(e, r) {
                
                return (0, t.cloneElement)(e, {
                    style: Or()({}, e.props.style, {
                        width: n[r + S]
                    })
                });
            }));
        }), [ n, S, d, O ]);
        var R =  (0, t.cloneElement)(p, {
            elementRef: x,
            "data-test": "".concat(u, "-head")
        }, k);
        var E = (0, t.useMemo)((function() {
            return {
                top: m,
                width: h
            };
        }), [ m, h ]);
        
        return r().createElement(w, {
            style: E,
            ref: o
        },  r().createElement(kr, Ir({
            "data-test": "".concat(u, "-head-table"),
            "data-fixed-column": d ? "true" : undefined,
            style: Or()({}, f, {
                marginLeft: c,
                width: d ? undefined : v
            })
        }, b()(y, "dragIndex")), R));
    }
    Pr.propTypes = Dr;
    /* harmony default export */    const Ar = Pr;
    // CONCATENATED MODULE: ./src/Table/KeyboardSensor.ts
    function qr(e) {
        "@babel/helpers - typeof";
        return qr = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, qr(e);
    }
    function _r(e, n) {
        return $r(e) || Mr(e, n) || Wr(e, n) || zr();
    }
    function zr() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Mr(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, l = [], u = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = i.call(t)).done) && (l.push(r.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function $r(e) {
        if (Array.isArray(e)) return e;
    }
    function Nr(e) {
        return Kr(e) || Hr(e) || Wr(e) || Lr();
    }
    function Lr() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Wr(e, n) {
        if (e) {
            if ("string" == typeof e) return Vr(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? Vr(e, n) : void 0;
        }
    }
    function Hr(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function Kr(e) {
        if (Array.isArray(e)) return Vr(e);
    }
    function Vr(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function Br(e, n) {
        if (!(e instanceof n)) throw new TypeError("Cannot call a class as a function");
    }
    function Fr(e, n) {
        for (var t = 0; t < n.length; t++) {
            var r = n[t];
            r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), 
            Object.defineProperty(e, Yr(r.key), r);
        }
    }
    function Ur(e, n, t) {
        return n && Fr(e.prototype, n), t && Fr(e, t), Object.defineProperty(e, "prototype", {
            writable: !1
        }), e;
    }
    function Xr(e, n, t) {
        return (n = Yr(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function Yr(e) {
        var n = Gr(e, "string");
        return "symbol" == qr(n) ? n : n + "";
    }
    function Gr(e, n) {
        if ("object" != qr(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != qr(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    // based on https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/sensors/keyboard/KeyboardSensor.ts
    // eslint-disable-next-line max-classes-per-file
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/sensors/utilities/Listeners.ts
        var Jr =  function() {
        function e(n) {
            var t = this;
            Br(this, e);
            Xr(this, "listeners", []);
            Xr(this, "removeAll", (function() {
                t.listeners.forEach((function(e) {
                    var n;
                    return (n = t.target) === null || n === void 0 ? void 0 : n.removeEventListener.apply(n, Nr(e));
                }));
            }));
            this.target = n;
            this.target = n;
        }
        Ur(e, [ {
            key: "add",
            value: function e(n, t, r) {
                var o;
                (o = this.target) === null || o === void 0 ? void 0 : o.addEventListener(n, t, r);
                this.listeners.push([ n, t, r ]);
            }
        } ]);
        return e;
    }();
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/transform/parseTransform.ts
        function Qr(e) {
        if (e.startsWith("matrix3d(")) {
            var n = e.slice(9, -1).split(/, /);
            return {
                x: +n[12],
                y: +n[13],
                scaleX: +n[0],
                scaleY: +n[5]
            };
        }
        if (e.startsWith("matrix(")) {
            var t = e.slice(7, -1).split(/, /);
            return {
                x: +t[4],
                y: +t[5],
                scaleX: +t[0],
                scaleY: +t[3]
            };
        }
        return null;
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/transform/inverseTransform.ts
        function Zr(e, n, t) {
        var r = Qr(n);
        if (!r) {
            return e;
        }
        var o = r.scaleX, i = r.scaleY, a = r.x, l = r.y;
        var u = e.left - a - (1 - o) * parseFloat(t);
        var s = e.top - l - (1 - i) * parseFloat(t.slice(t.indexOf(" ") + 1));
        var c = o ? e.width / o : e.width;
        var d = i ? e.height / i : e.height;
        return {
            width: c,
            height: d,
            top: s,
            right: u + c,
            bottom: s + d,
            left: u
        };
    }
    var eo = {
        ignoreTransform: false
    };
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/rect/getRect.ts
        function no(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : eo;
        var t = e.getBoundingClientRect();
        if (n.ignoreTransform) {
            var r = (0, Kn.getWindow)(e).getComputedStyle(e), o = r.transform, i = r.transformOrigin;
            if (o) {
                t = Zr(t, o, i);
            }
        }
        var a = t, l = a.top, u = a.left, s = a.width, c = a.height, d = a.bottom, f = a.right;
        return {
            top: l,
            left: u,
            width: s,
            height: c,
            bottom: d,
            right: f
        };
    }
    function to(e) {
        if (e === document.scrollingElement) {
            var n = window, t = n.innerWidth, r = n.innerHeight;
            return {
                top: 0,
                left: 0,
                right: t,
                bottom: r,
                width: t,
                height: r
            };
        }
        var o = e.getBoundingClientRect(), i = o.top, a = o.left, l = o.right, u = o.bottom;
        return {
            top: i,
            left: a,
            right: l,
            bottom: u,
            width: e.clientWidth,
            height: e.clientHeight
        };
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/documentScrollingElement.ts
        function ro(e) {
        if (!Kn.canUseDOM || !e) {
            return false;
        }
        return e === document.scrollingElement;
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/getScrollPosition.ts
        function oo(e) {
        var n = {
            x: 0,
            y: 0
        };
        var t = ro(e) ? {
            height: window.innerHeight,
            width: window.innerWidth
        } : {
            height: e.clientHeight,
            width: e.clientWidth
        };
        var r = {
            x: e.scrollWidth - t.width,
            y: e.scrollHeight - t.height
        };
        var o = e.scrollTop <= n.y;
        var i = e.scrollLeft <= n.x;
        var a = e.scrollTop >= r.y;
        var l = e.scrollLeft >= r.x;
        return {
            isTop: o,
            isLeft: i,
            isBottom: a,
            isRight: l,
            maxScroll: r,
            minScroll: n
        };
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/coordinates/constants.ts
        var io = {
        x: 0,
        y: 0
    };
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/isFixed.ts
        function ao(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : (0, 
        Kn.getWindow)(e).getComputedStyle(e);
        return n.position === "fixed";
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/isScrollable.ts
        function lo(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : (0, 
        Kn.getWindow)(e).getComputedStyle(e);
        var t = /(auto|scroll|overlay)/;
        var r = [ "overflow", "overflowX", "overflowY" ];
        return r.some((function(e) {
            var r = n[e];
            return typeof r === "string" ? t.test(r) : false;
        }));
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/getScrollableAncestors.ts
        function uo(e, n) {
        var t = [];
        function r(o) {
            if (n != null && t.length >= n) {
                return t;
            }
            if (!o) {
                return t;
            }
            if ((0, Kn.isDocument)(o) && o.scrollingElement != null && !t.includes(o.scrollingElement)) {
                t.push(o.scrollingElement);
                return t;
            }
            if (!(0, Kn.isHTMLElement)(o) || (0, Kn.isSVGElement)(o)) {
                return t;
            }
            if (t.includes(o)) {
                return t;
            }
            var i = (0, Kn.getWindow)(e).getComputedStyle(o);
            if (o !== e) {
                if (lo(o, i)) {
                    t.push(o);
                }
            }
            if (ao(o, i)) {
                return t;
            }
            return r(o.parentNode);
        }
        if (!e) {
            return t;
        }
        return r(e);
    }
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/getScrollableAncestors.ts
        function so(e) {
        var n = uo(e, 1), t = _r(n, 1), r = t[0];
        return r !== null && r !== void 0 ? r : null;
    }
    var co = {
        start: [ o.KeyboardCode.Space, o.KeyboardCode.Enter ],
        cancel: [ o.KeyboardCode.Esc ],
        end: [ o.KeyboardCode.Space, o.KeyboardCode.Enter ]
    };
    // https://github.com/clauderic/dnd-kit/blob/694dcc2f62e5269541fc941fa6c9af46ccd682ad/packages/core/src/utilities/scroll/scrollIntoViewIfNeeded.ts
        function fo(e) {
        var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : no;
        if (!e) {
            return;
        }
        var t = n(e), r = t.top, o = t.left, i = t.bottom, a = t.right;
        var l = so(e);
        if (!l) {
            return;
        }
        if (i <= 0 || a <= 0 || r >= window.innerHeight || o >= window.innerWidth) {
            e.scrollIntoView({
                block: "center",
                inline: "center"
            });
        }
    }
    var vo =  function() {
        function e(n) {
            Br(this, e);
            Xr(this, "autoScrollEnabled", false);
            Xr(this, "referenceCoordinates", void 0);
            Xr(this, "listeners", void 0);
            Xr(this, "windowListeners", void 0);
            this.props = n;
            var t = n.event.target;
            this.props = n;
            this.listeners = new Jr((0, Kn.getOwnerDocument)(t));
            this.windowListeners = new Jr((0, Kn.getWindow)(t));
            this.handleKeyDown = this.handleKeyDown.bind(this);
            this.handleCancel = this.handleCancel.bind(this);
            this.attach();
        }
        Ur(e, [ {
            key: "attach",
            value: function e() {
                var n = this;
                this.handleStart();
                this.windowListeners.add("resize", this.handleCancel);
                this.windowListeners.add("visibilitychange", this.handleCancel);
                setTimeout((function() {
                    return n.listeners.add("keydown", n.handleKeyDown);
                }));
            }
        }, {
            key: "handleStart",
            value: function e() {
                var n = this.props, t = n.activeNode, r = n.onStart;
                var o = t.node.current;
                if (o) {
                    fo(o);
                }
                r(io);
            }
        }, {
            key: "handleKeyDown",
            value: function e(n) {
                if ((0, Kn.isKeyboardEvent)(n)) {
                    var t = this.props, r = t.active, i = t.context, a = t.options;
                    var l = a.keyboardCodes, u = l === void 0 ? co : l, s = a.coordinateGetter, c = a.scrollBehavior, d = c === void 0 ? "smooth" : c;
                    var f = n.code;
                    if (u.end.includes(f)) {
                        this.handleEnd(n);
                        return;
                    }
                    if (u.cancel.includes(f)) {
                        this.handleCancel(n);
                        return;
                    }
                    var v = i.current.collisionRect;
                    var p = v ? {
                        x: v.left,
                        y: v.top
                    } : io;
                    if (!this.referenceCoordinates) {
                        this.referenceCoordinates = p;
                    }
                    var m = s(n, {
                        active: r,
                        context: i.current,
                        currentCoordinates: p
                    });
                    if (m) {
                        var b = (0, Kn.subtract)(m, p);
                        var h = {
                            x: 0,
                            y: 0
                        };
                        var y = i.current.scrollableAncestors;
                        for (var g = 0; g < y.length; g += 1) {
                            var w = y[g];
                            var C = n.code;
                            var x = oo(w), S = x.isTop, O = x.isRight, k = x.isLeft, R = x.isBottom, E = x.maxScroll, I = x.minScroll;
                            var T = to(w);
                            // isDraggableNodeVerticallyInView and isDraggableNodeHorizontallyInView are custom
                            // we want to scroll only when a dragging node will be out of view,
                            // originally it tries to always place a dragging node in the middle of the container
                                                        var j = T.top <= m.y && T.bottom >= m.y + ((v === null || v === void 0 ? void 0 : v.height) || 0);
                            var D = T.left <= m.x && T.right >= m.x + ((v === null || v === void 0 ? void 0 : v.width) || 0);
                            var P = (C === o.KeyboardCode.Right && !O || C === o.KeyboardCode.Left && !k) && !D;
                            var A = (C === o.KeyboardCode.Down && !R || C === o.KeyboardCode.Up && !S) && !j;
                            if (P) {
                                var q = w.scrollLeft + b.x;
                                var _ = C === o.KeyboardCode.Right && q <= E.x || C === o.KeyboardCode.Left && q >= I.x;
                                if (_ && !b.y) {
                                    // We don't need to update coordinates, the scroll adjustment alone will trigger
                                    // logic to auto-detect the new container we are over
                                    w.scrollTo({
                                        left: q,
                                        behavior: d
                                    });
                                    return;
                                }
                                if (_) {
                                    h.x = w.scrollLeft - q;
                                } else {
                                    h.x = C === o.KeyboardCode.Right ? w.scrollLeft - E.x : w.scrollLeft - I.x;
                                }
                                if (h.x) {
                                    w.scrollBy({
                                        left: -h.x,
                                        behavior: d
                                    });
                                }
                                break;
                            } else if (A) {
                                var z = w.scrollTop + b.y;
                                var M = C === o.KeyboardCode.Down && z <= E.y || C === o.KeyboardCode.Up && z >= I.y;
                                if (M && !b.x) {
                                    // We don't need to update coordinates, the scroll adjustment alone will trigger
                                    // logic to auto-detect the new container we are over
                                    w.scrollTo({
                                        top: z,
                                        behavior: d
                                    });
                                    return;
                                }
                                if (M) {
                                    h.y = w.scrollTop - z;
                                } else {
                                    h.y = C === o.KeyboardCode.Down ? w.scrollTop - E.y : w.scrollTop - I.y;
                                }
                                if (h.y) {
                                    w.scrollBy({
                                        top: -h.y,
                                        behavior: d
                                    });
                                }
                                break;
                            }
                        }
                        this.handleMove(n, (0, Kn.add)((0, Kn.subtract)(m, this.referenceCoordinates), h));
                    }
                }
            }
        }, {
            key: "handleMove",
            value: function e(n, t) {
                var r = this.props.onMove;
                n.preventDefault();
                r(t);
            }
        }, {
            key: "handleEnd",
            value: function e(n) {
                var t = this.props.onEnd;
                n.preventDefault();
                this.detach();
                t();
            }
        }, {
            key: "handleCancel",
            value: function e(n) {
                var t = this.props.onCancel;
                n.preventDefault();
                this.detach();
                t();
            }
        }, {
            key: "detach",
            value: function e() {
                this.listeners.removeAll();
                this.windowListeners.removeAll();
            }
        } ]);
        return e;
    }();
    Xr(vo, "activators", [ {
        eventName: "onKeyDown",
        handler: function e(n, t, r) {
            var o = t.keyboardCodes, i = o === void 0 ? co : o, a = t.onActivation;
            var l = r.active;
            var u = n.nativeEvent.code;
            if (i.start.includes(u)) {
                var s = l.activatorNode.current;
                if (s && n.target !== s) {
                    return false;
                }
                n.preventDefault();
                a === null || a === void 0 ? void 0 : a({
                    event: n.nativeEvent
                });
                return true;
            }
            return false;
        }
    } ]);
    // CONCATENATED MODULE: ./src/utils/getBoundingClientRect.ts
    function po(e) {
        "@babel/helpers - typeof";
        return po = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, po(e);
    }
    function mo(e, n) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            n && (r = r.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), t.push.apply(t, r);
        }
        return t;
    }
    function bo(e) {
        for (var n = 1; n < arguments.length; n++) {
            var t = null != arguments[n] ? arguments[n] : {};
            n % 2 ? mo(Object(t), !0).forEach((function(n) {
                ho(e, n, t[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : mo(Object(t)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(t, n));
            }));
        }
        return e;
    }
    function ho(e, n, t) {
        return (n = yo(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function yo(e) {
        var n = go(e, "string");
        return "symbol" == po(n) ? n : n + "";
    }
    function go(e, n) {
        if ("object" != po(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != po(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    // Checks if browser is Safari
        function wo() {
        return /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
    }
    // Formats return value to DOMRect
        function Co(e) {
        return bo(bo({}, e), {}, {
            top: e.y,
            left: e.x,
            right: e.x + e.width,
            bottom: e.y + e.height
        });
    }
    /**
 * Provides DOMRect information about the size of an element and its position relative to the viewport.
 * Accounts for Safari's use of visual viewport and adds offsets to set as layout viewport.
 * @param {Element} element - The element to be evaluated
 * @param {boolean} offsets - Use when position of element relative to the layout viewport offsets are needed for Safari.
 * @returns {Omit<DOMRect, 'toJSON'>}
 * @public
 */    function xo(e, n) {
        var t, r;
        var o = e.getBoundingClientRect();
        var i = wo();
        if (!n || !i) {
            return o;
        }
        var a = cn();
        var l = o.left + (((t = a.visualViewport) === null || t === void 0 ? void 0 : t.offsetLeft) || 0);
        var u = o.top + (((r = a.visualViewport) === null || r === void 0 ? void 0 : r.offsetTop) || 0);
        var s = o.width, c = o.height;
        return Co({
            width: s,
            height: c,
            x: l,
            y: u
        });
    }
    // CONCATENATED MODULE: ./src/utils/types.ts
    // see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31245 for details
    // regarding the css prop. certain (versions of) libraries included in the compilation
    // - such as storybook->emotion - introduce the css prop using a global augmentation,
    // which causes the prop to end up in our declaration files (via JSX.IntrinsicElements).
    // it's difficult to control all nested dependencies and their types, which is why the
    // css prop is omitted in the two prop helpers above - this ensures the css prop isn't
    // erroneously included in react-ui's public api.
    // see https://github.com/microsoft/TypeScript/pull/12253
    // see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html
    function So(e, n) {
        if (e === undefined || e === null) {
            throw new Error(n !== null && n !== void 0 ? n : "Unexpected undefined or null");
        }
    }
    // CONCATENATED MODULE: ./src/Table/Table.tsx
    function Oo() {
        return Oo = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, Oo.apply(null, arguments);
    }
    function ko(e) {
        return Io(e) || Eo(e) || Do(e) || Ro();
    }
    function Ro() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Eo(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function Io(e) {
        if (Array.isArray(e)) return Po(e);
    }
    function To(e, n) {
        return qo(e) || Ao(e, n) || Do(e, n) || jo();
    }
    function jo() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function Do(e, n) {
        if (e) {
            if ("string" == typeof e) return Po(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? Po(e, n) : void 0;
        }
    }
    function Po(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function Ao(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, l = [], u = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (r = i.call(t)).done) && (l.push(r.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function qo(e) {
        if (Array.isArray(e)) return e;
    }
    function _o(e, n) {
        if (null == e) return {};
        var t, r, o = zo(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function zo(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    // Exposed as static members of the Table class
    // Internal Helper Modules
    /** @public */
    /** @public */
    /** @public */
    /** @public */    var Mo = {
        actions: w().arrayOf(w().element),
        actionsColumnWidth: w().number,
        children: w().node,
        dockOffset: w().number,
        dockScrollBar: w().bool,
        elementRef: w().oneOfType([ w().func, w().object ]),
        headType: w().oneOf([ "docked", "fixed", "inline" ]),
        horizontalOverflow: w().oneOf([ "auto", "scroll" ]),
        innerStyle: w().object,
        onRequestToggleAllRows: w().func,
        onScroll: w().func,
        outerStyle: w().object,
        pinnedColumns: w().object,
        primaryColumnIndex: w().number,
        rowExpansion: w().oneOf([ "single", "multi", "controlled", "none" ]),
        rowSelection: w().oneOf([ "all", "some", "none" ]),
        stripeRows: w().bool,
        tableStyle: w().object,
        onRequestMoveColumn: w().func,
        onRequestMoveRow: w().func,
        onRequestResizeColumn: w().func,
        resizableFillLayout: w().bool
    };
    var $o = function e(n) {
        return n === "inline";
    };
    var No = function e(n) {
        return n === "docked";
    };
    var Lo = function e(n) {
        return n === "fixed";
    };
    var Wo = function e(n) {
        var t = cn();
        var r = xo(n);
        return {
            top: r.top + t.pageYOffset,
            left: r.left + t.pageXOffset
        };
    };
    var Ho = function e(n) {
        var t = an();
        if (n === undefined || n === null) {
            return undefined;
        }
        var r = n.children[0].children;
        var o = d()(r, t.activeElement);
        if (o > -1) {
            return {
                target: "headCell",
                index: o
            };
        }
        var i = n.querySelectorAll("[data-test=resize]");
        var a = d()(i, t.activeElement);
        if (a > -1) {
            return {
                target: "resizeButton",
                index: a
            };
        }
        return undefined;
    };
    var Ko = function e(n, t) {
        return n.indexOf(t) + 1;
    };
    var Vo = function e(n, t) {
        return n.indexOf(t) + 1;
    };
    var Bo = {
        coordinateGetter: a.sortableKeyboardCoordinates
    };
    var Fo = {
        draggable: (0, T._)("To pick up a sortable row, press space or enter.\n            Use the up and down arrow keys to update the position of the row.\n            Press space or enter again to drop the row in its new position, or press escape to cancel.")
    };
    var Uo = {
        draggable: (0, T._)("To pick up a sortable column, press space or enter.\n            Use the left and right arrow keys to update the position of the column.\n            Press space or enter again to drop the column in its new position, or press escape to cancel.")
    };
    /**
 * @private
 * @private
 * Applies the provided `headFocusState` to the provided `head`.
 * @param head - A reference to a mounted `Head` component.
 * @param {Object} headFocusState
 * @param {String} headFocusState.target - Focus can be applied to a 'headCell' or a
 * 'resizeButton'.
 * @param {Number} headFocusState.index - The index of the element to set focus on.
 */    function Xo(e, n) {
        var t, r;
        var o = n.target, i = n.index;
        if (false) {}
        var a;
        if (o === "headCell") {
            a = e === null || e === void 0 ? void 0 : e.children[0].children[i];
        } else if (o === "resizeButton") {
            a = e === null || e === void 0 ? void 0 : e.querySelectorAll("[data-test=resize]")[i];
        }
        if (false) {}
        (t = a) === null || t === void 0 ? void 0 : (r = t.focus) === null || r === void 0 ? void 0 : r.call(t);
    }
    var Yo = {
        passive: false,
        capture: true
    };
    var Go = [];
    function Jo(e) {
        var n = e.actions, l = n === void 0 ? Go : n, s = e.actionsColumnWidth, c = e.children, d = e.dockOffset, f = d === void 0 ? 0 : d, p = e.dockScrollBar, m = e.elementRef, h = e.headType, g = h === void 0 ? "inline" : h, w = e.horizontalOverflow, C = w === void 0 ? "auto" : w, O = e.innerStyle, R = e.onRequestMoveColumn, D = e.onRequestMoveRow, P = e.onRequestResizeColumn, A = e.onRequestToggleAllRows, q = e.onScroll, _ = e.outerStyle, z = e.pinnedColumns, M = e.primaryColumnIndex, $ = M === void 0 ? 0 : M, L = e.resizableFillLayout, W = e.rowExpansion, H = W === void 0 ? "none" : W, K = e.rowSelection, V = K === void 0 ? "none" : K, B = e.stripeRows, F = e.tableStyle, U = _o(e, [ "actions", "actionsColumnWidth", "children", "dockOffset", "dockScrollBar", "elementRef", "headType", "horizontalOverflow", "innerStyle", "onRequestMoveColumn", "onRequestMoveRow", "onRequestResizeColumn", "onRequestToggleAllRows", "onScroll", "outerStyle", "pinnedColumns", "primaryColumnIndex", "resizableFillLayout", "rowExpansion", "rowSelection", "stripeRows", "tableStyle" ]);
 // @docs-props-type TablePropsBase
                var X = (0, t.useState)([]), Y = To(X, 2), G = Y[0], J = Y[1];
        var re = (0, t.useState)(), ae = To(re, 2), le = ae[0], ue = ae[1];
        var se = (0, t.useState)(), ce = To(se, 2), de = ce[0], fe = ce[1];
        var ve = (0, t.useState)(), pe = To(ve, 2), me = pe[0], be = pe[1];
        var he = (0, t.useState)(), ge = To(he, 2), we = ge[0], Ce = ge[1];
        var xe = (0, t.useRef)(true);
        var Se = (0, t.useContext)(S.ScrollContainerContext);
        var Oe = (0, o.useSensors)((0, o.useSensor)(o.MouseSensor, {}), (0, o.useSensor)(o.TouchSensor, {}), (0, 
        o.useSensor)(vo, Bo));
        var ke = (0, t.useState)([]), Re = To(ke, 2), Ee = Re[0], Ie = Re[1];
        var Te = (0, t.useState)(-1), je = To(Te, 2), De = je[0], Pe = je[1];
        var Ae = (0, t.useRef)();
        var qe = (0, t.useRef)();
        var _e = (0, t.useState)(), ze = To(_e, 2), Me = ze[0], $e = ze[1];
        var Ne = (0, t.useRef)(null);
        var Le = (0, t.useRef)(null);
        var We = (0, t.useRef)();
        var He = (0, t.useRef)(null);
        var Ke = (0, t.useRef)();
        var Ve = (0, t.useRef)(null);
        var Be = (0, t.useRef)(null);
        var Fe = k()(g);
        var Ue = (0, t.useRef)([]);
        var Xe = (0, t.useRef)([]);
        var Ye = (0, t.useRef)([]);
        var Ge = (0, t.useRef)([]);
        var Je = Xe.current;
        var Qe = Ue.current;
        var Ze = Ge.current;
        var en = (0, t.useCallback)((function() {
            var e;
            var n = cn();
            if ($o(g)) {
                return;
            }
            var t = Be.current;
            var r = Le.current;
            var o = Ve.current;
            if (!o || !r) {
                return;
            }
            var i = Qt(Le.current);
            var a = (e = Le.current) === null || e === void 0 ? void 0 : e.clientHeight;
            var l = Wo(o).top - n.pageYOffset - (a || 0) + o.offsetHeight;
            var u = Math.min(l, Number(f));
            Ce(t === null || t === void 0 ? void 0 : t.clientWidth);
            fe(o === null || o === void 0 ? void 0 : o.clientWidth);
            J(i);
            be(u);
        }), [ f, g ]);
        (0, t.useEffect)((function() {
            var e = Le.current;
            var n = He.current;
            if (e && n) {
                var t = Qt(Le.current);
                if (!v()(G, t)) {
                    en();
                }
            }
            if (Fe === "inline" && g === "inline") {
                en();
            }
            if (false) {}
        }), [ G, g, Fe, e, en ]);
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ z, C ]);
        var nn = (0, t.useCallback)((function(e) {
            var n = Be.current;
            q === null || q === void 0 ? void 0 : q(e);
            if ((Lo(g) || No(g)) && n) {
                ue(-n.scrollLeft);
            }
            if (Ne.current && Ke.current !== "dockedScrollBar" && n) {
                Ke.current = "container";
                Ne.current.scrollLeft = n.scrollLeft;
            } else {
                Ke.current = undefined;
            }
        }), [ g, q ]);
        var tn = function e() {
            var n = Be.current;
            if (Ke.current !== "container" && n && Ne.current) {
                Ke.current = "dockedScrollBar";
                n.scrollLeft = Ne.current.scrollLeft;
            } else {
                Ke.current = undefined;
            }
        };
        var rn = y()(en, 0);
        var on = (0, t.useCallback)((function() {
            en();
        }), [ en ]);
        var ln = (0, t.useMemo)((function() {
            return y()(on, 50);
        }), [ on ]);
        E()(Be, ln);
        (0, t.useEffect)((function() {
            // Using the return value of useCallback to recreate componentWillUnmount
            return function() {
                rn.cancel();
                ln.cancel();
            };
        }));
        var un = function e(n) {
            return function(e) {
                Le.current = e;
                N(n.props.elementRef, e);
            };
        };
        var sn = function e(n) {
            if (n) {
                // On mount, focus the applicable docked head cell if a primary head cell was focused.
                if (We.current) {
                    Xo(He.current, We.current);
                    We.current = undefined;
                }
            } else {
                // On unmount, focus the applicable primary head cell if a docked head cell was focused.
                var t = Ho(He.current);
                if (t) {
                    Xo(Le.current, t);
                }
            }
            He.current = n;
        };
        var dn = function e(n) {
            var t = Be.current;
            if (!He.current || !t || n.key !== "Tab") {
                return;
            }
            var r = He.current.scrollLeft;
            if (r !== 0) {
                He.current.scrollLeft = 0;
                t.scrollLeft += r;
            }
        };
        var fn = function e(n, t) {
            var r, o;
            var i = t.columnId, a = t.index, l = t.id;
            var s = an();
            if (a === undefined) {
                return;
            }
            var c = H === "none" ? 1 : 2;
            var d = (r = Be.current) === null || r === void 0 ? void 0 : r.querySelectorAll("thead th:nth-child(".concat(a + c, "), tbody td:nth-child(").concat(a + c, ")"));
            var f = s.createElement("div");
            // TODO: find a styled-components compliant way to do this
                        f.style["float"] = "left";
            f.style.position = "fixed";
            f.style.top = "-100";
            f.style.left = "0";
            f.style.maxHeight = "10px";
            f.style.overflow = "hidden";
            u()(d, (function(e) {
                var n = e.cloneNode(true);
 // TS: this seems safe to assume?
                                n.style.display = "block";
                n.style.width = "auto";
                f.appendChild(n);
            }));
            s.body.appendChild(f);
            P === null || P === void 0 ? void 0 : P(n, {
                columnId: i,
                id: l,
                index: a,
                width: f.clientWidth + 1
            });
            (o = f.parentNode) === null || o === void 0 ? void 0 : o.removeChild(f);
        };
        var vn = (0, t.useCallback)((function(e) {
            var n = e.active;
            var t = n.id;
            Ae.current = Ze.indexOf(t);
            $e(t);
        }), [ Ze ]);
        var pn = (0, t.useCallback)((function(e) {
            if (e.over) {
                var n = e.over.id;
                var t = Ze.indexOf(n);
                var r = t;
                if (Ae.current !== undefined && t > Ae.current) {
                    r = t + 1;
                }
                Pe(r);
            } else {
                Pe(-1);
            }
        }), [ Ze ]);
        // these are useMemo rather than useCallback so that ESLint can correctly analyze the dependencies
        // see https://github.com/facebook/react/issues/19240#issuecomment-652945246
                var mn = (0, t.useMemo)((function() {
            return y()(pn, 100);
        }), [ pn ]);
        (0, t.useEffect)((function() {
            return function() {
                mn.cancel();
            };
        }), [ mn ]);
        var bn = (0, t.useCallback)((function(e) {
            var n = e.active, t = e.over;
            if (t && n.id !== t.id) {
                var r = Ze.indexOf(n.id);
                var o = Ze.indexOf(t === null || t === void 0 ? void 0 : t.id);
                R === null || R === void 0 ? void 0 : R({
                    fromIndex: r,
                    toIndex: o
                });
            }
            $e(undefined);
            Pe(-1);
            qe.current = undefined;
            Ae.current = undefined;
        }), [ Ze, R ]);
        var hn = (0, t.useCallback)((function(e) {
            var n = e.active, t = e.over;
            xe.current = true;
            if (t && n.id !== t.id) {
                var r = Je.indexOf(n.id);
                var o = Je.indexOf(t === null || t === void 0 ? void 0 : t.id);
                D === null || D === void 0 ? void 0 : D({
                    fromIndex: r,
                    toIndex: o
                });
            }
        }), [ Je, D ]);
        var yn = (0, t.useCallback)((function() {
            xe.current = true;
        }), []);
        var gn = (0, t.useRef)({});
        var wn = function e(n) {
            var r = [];
            var o = [];
            var i = t.Children.toArray(n.props.children).filter(t.isValidElement).map((function(e, n) {
                var i = e;
                var a = i.key || (0, j.createDOMID)("col-");
                a = String(a);
                // `col-` prefix is necessary to distinguish row or column reordering.
                // Append the `col-` prefix if is missing from dragId
                                if (!a.startsWith("col-")) {
                    a = "col-".concat(a);
                }
                r.push(i);
                o.push(a);
                
                return (0, t.cloneElement)(i, {
                    index: n,
                    dragId: a,
                    elementRef: function e(n) {
                        gn.current[a] = n;
                        N(i.props.elementRef, n);
                    }
                });
            }));
            Ye.current = r;
            Ze = o;
            var a = l === null || l === void 0 ? void 0 : l.filter(t.isValidElement);
            var u =  (0, t.cloneElement)(n, {
                actions: a,
                actionsColumnWidth: s,
                hasInfoColumn: H !== "none",
                hasDragColumn: !!D,
                pinnedColumns: z,
                onAutosizeColumn: fn,
                onRequestMoveColumn: R,
                onRequestResizeColumn: P,
                onRequestToggleAllRows: A,
                elementRef: un(n),
                rowSelection: V
            }, i);
            return u;
        };
        var Cn = (0, t.useMemo)((function() {
            var e, n;
            if (De === -1 || !Le.current || !Be.current) return null;
            var t = !!D;
            var o = H !== "none";
            var i = Array.from((e = (n = Le.current.firstElementChild) === null || n === void 0 ? void 0 : n.children) !== null && e !== void 0 ? e : []).slice(o ? 1 : 0).slice(t ? 1 : 0).slice(A ? 1 : 0);
            var a;
            if (De === 0) {
                var l;
                a = ((l = i[0]) === null || l === void 0 ? void 0 : l.offsetLeft) || 0;
            } else if (De >= i.length) {
                var u = i[i.length - 1];
                a = ((u === null || u === void 0 ? void 0 : u.offsetLeft) || 0) + ((u === null || u === void 0 ? void 0 : u.offsetWidth) || 0) - 1;
 // -1 makes the last guideline visible
                        } else {
                var s;
                a = ((s = i[De]) === null || s === void 0 ? void 0 : s.offsetLeft) || 0;
            }
            
            return r().createElement(ie, {
                $height: Be.current.clientHeight,
                $left: a
            });
        }), [ De, D, H, A ]);
        var xn = function e(n) {
            var r = (l === null || l === void 0 ? void 0 : l.filter(t.isValidElement).length) > 0;
            var o = false;
            var i = [];
            var a = [];
            t.Children.toArray(n === null || n === void 0 ? void 0 : n.props.children).filter(t.isValidElement).forEach((function(e, n) {
                var r, l;
                if (((r = e.props) === null || r === void 0 ? void 0 : r.actionPrimary) || ((l = e.props) === null || l === void 0 ? void 0 : l.actionsSecondary)) {
                    o = true;
                }
                var u = e.key || (0, j.createDOMID)("row-");
                u = String(u);
                // `row-` prefix is necessary to distinguish row or column reordering.
                // Append the `row-` prefix if is missing from dragId
                                if (!u.startsWith("row-")) {
                    u = "row-".concat(u);
                }
                i.push( (0, t.cloneElement)(e, {
                    index: n,
                    dragId: u
                }));
                a.push(u);
            }));
            var u = r || o || !!s;
            Qe = i;
            Je = a;
            
            return (0, t.cloneElement)(n, {
                actions: u,
                movableColumns: !!R,
                onRequestMoveRow: D,
                primaryColumnIndex: $,
                rowExpansion: H,
                pinnedColumns: z,
                stripeRows: B
            }, Qe);
        };
        var Sn = function e() {
            var n = Be.current;
            var t = cn();
            if (!No(g) || !Ve.current || !Le.current || !n) {
                return false;
            }
            var r = Wo(n).top;
            return t.pageYOffset >= r - (f || 0);
        };
        var On = function e() {
            var n = cn();
            var t = Be.current;
            if (!p || !Ve.current || !t) {
                return false;
            }
            var r = xo(t);
            return r.bottom > n.innerHeight && r.top < n.innerHeight;
        };
        var kn = function e(n, o) {
            if (Sn()) {
                // FIXME: handle inline
                // Save the current head focus state so that it can be applied to the docked version
                // after the `HeadTable` mounts.
                We.current = Ho(Le.current);
                if (G === undefined || de === undefined || me === undefined || we === undefined) {
                    return false;
                }
                var i =  (0, t.cloneElement)(n);
                
                return r().createElement(Ar, {
                    headType: "docked",
                    tHead: i,
                    cellWidths: G,
                    horizontalOffset: le,
                    tableWidth: de,
                    top: me,
                    width: we,
                    tableStyle: F,
                    isFixedColumn: o,
                    onKeyUp: dn,
                    elementRef: sn,
                    hasDragColumn: !!D,
                    hasRowExpansion: H !== "none",
                    hasRowSelection: !!A
                });
            }
            return false;
        };
        var Rn = (0, t.useMemo)((function() {
            var e, n, t;
            if (!Ne.current || !Le.current || !Ve.current) {
                return 0;
            }
            var r = cn();
            var o = (e = xo(Ve.current).top) !== null && e !== void 0 ? e : 0;
            var i = (n = (t = Le.current) === null || t === void 0 ? void 0 : t.offsetHeight) !== null && n !== void 0 ? n : 0;
            var a = r.innerHeight - o - i;
            return Math.min(a - Ne.current.offsetHeight, 0);
        }), []);
        var En = (0, t.useMemo)((function() {
            return {
                width: we,
                marginBottom: Rn
            };
        }), [ we, Rn ]);
        var In = (0, t.useMemo)((function() {
            return {
                width: de
            };
        }), [ de ]);
        var Tn = function e() {
            if (!On()) {
                return false;
            }
            
            return r().createElement(ne, {
                style: En,
                onScroll: tn,
                "data-test": "docked-scroll-bar",
                ref: Ne
            },  r().createElement(te, {
                style: In,
                "data-test": "docked-scroll-content"
            }));
        };
        var jn = function e() {
            var n = Le.current;
            var t = Be.current;
            /**
     * there is slightly less space for the table than tableContainer's width,
     * because there is a 1 px border added to every column but the first
     */            if (n) {
                var r = Qt(Le.current);
                var o = r === null || r === void 0 ? void 0 : r.length;
                var i = t === null || t === void 0 ? void 0 : t.clientWidth;
                if (i != null) {
                    var a = Math.max(0, o - 1);
                    return i - a;
                }
            }
            return null;
        };
        var Dn;
        var Pn;
        var An;
        var qn = !!P;
        t.Children.toArray(c).forEach((function(e) {
            var n = e.type.splunkUiType;
            if (n === "Table.Head") {
                Dn = wn(e);
                // Check if all HeadCells have a defined width (making the table fixed width)
                                var r = t.Children.toArray(e.props.children).filter(t.isValidElement);
                var o = r.every((function(e) {
                    return e.props.width && e.props.width !== undefined;
                }));
                qn = o;
            } else if (n === "Table.Body") {
                Pn = xn(e);
            } else if (n === "Table.Caption") {
                // the html spec mandates that a caption must be the first table child.
                // catch the caption here so that it can be placed first later. note: a
                // caption can be placed on the bottom side of a table by adjusting the
                // caption's side prop.
                An = e;
            }
        }));
        So(Dn, "A Table.Head is required.");
        var _n = kn(Dn, qn);
        // When a `headTable` is used, suppress focus interactions in the primary `Head`. Focus
        // interactions will be handled by the overlaid `headTable` instead.
                var zn = t.Children.toArray(Dn.props.children).filter(t.isValidElement).map((function(e) {
            
            // TS: onClick is different for regular/dropdown head cells, that's ok
            return (0, t.cloneElement)(e, {
                visible: !_n
            });
        }));
        Dn =  (0, t.cloneElement)(Dn, {}, zn);
        var Mn = Se || cn();
        var $n = Lo(g) ? Be.current : Mn;
        if (L) {
            qn = false;
        }
        var Nn = jn();
        var Ln = (0, t.useMemo)((function() {
            return {
                headType: g,
                horizontalOverflow: C,
                resizableFillLayout: L,
                tableContainerWidthWithoutBorders: Nn
            };
        }), [ g, C, L, Nn ]);
        var Wn = (0, t.useMemo)((function() {
            if (!D) {
                return {
                    onDragStart: function e() {
                        return undefined;
                    },
                    onDragOver: function e() {
                        return undefined;
                    },
                    onDragEnd: function e() {
                        return undefined;
                    },
                    onDragCancel: function e() {
                        return undefined;
                    }
                };
            }
            return {
                onDragStart: function e(n) {
                    var t = n.active.id;
                    if (t === undefined) {
                        return undefined;
                    }
                    var r = Ko(Je, t);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Picked up sortable row in position %(pos)d of %(total)d."), {
                        pos: r,
                        total: Je.length
                    }) : undefined;
                },
                onDragOver: function e(n) {
                    var t = n.active, r = n.over;
                    if (xe.current) {
                        xe.current = false;
                        return undefined;
                    }
                    if (!t.id || !(r === null || r === void 0 ? void 0 : r.id)) {
                        return undefined;
                    }
                    var o = Ko(Je, t.id);
                    var i = Ko(Je, r.id);
                    return (0, I.sprintf)((0, T._)("Row moved from position %(from)d to position %(to)d of %(total)d."), {
                        from: o,
                        to: i,
                        total: Je.length
                    });
                },
                onDragEnd: function e(n) {
                    var t = n.over;
                    if (!(t === null || t === void 0 ? void 0 : t.id)) {
                        return undefined;
                    }
                    var r = Ko(Je, t.id);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Row dropped at position %(pos)d of %(total)d."), {
                        pos: r,
                        total: Je.length
                    }) : undefined;
                },
                onDragCancel: function e(n) {
                    var t = n.active.id;
                    if (t === undefined) {
                        return undefined;
                    }
                    var r = Ko(Je, t);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Row returned to its starting position of %(pos)d."), {
                        pos: r
                    }) : undefined;
                }
            };
        }), [ Je, D ]);
        var Hn = (0, t.useMemo)((function() {
            if (!R) {
                return {
                    onDragStart: function e() {
                        return undefined;
                    },
                    onDragOver: function e() {
                        return undefined;
                    },
                    onDragEnd: function e() {
                        return undefined;
                    },
                    onDragCancel: function e() {
                        return undefined;
                    }
                };
            }
            return {
                onDragStart: function e(n) {
                    var t = n.active.id;
                    if (t === undefined) {
                        return undefined;
                    }
                    var r = Vo(Ze, t);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Picked up sortable column in position %(pos)d of %(total)d."), {
                        pos: r,
                        total: Ze.length
                    }) : undefined;
                },
                onDragOver: function e(n) {
                    var t = n.active, r = n.over;
                    if (xe.current) {
                        xe.current = false;
                        return undefined;
                    }
                    if (!t.id || !(r === null || r === void 0 ? void 0 : r.id)) {
                        return undefined;
                    }
                    var o = Vo(Ze, t.id);
                    var i = Vo(Ze, r.id);
                    if (typeof o !== "number" || typeof i !== "number") return undefined;
                    return (0, I.sprintf)((0, T._)("Column moved from position %(from)d to position %(to)d of %(total)d."), {
                        from: o,
                        to: i,
                        total: Ze.length
                    });
                },
                onDragEnd: function e(n) {
                    var t = n.over;
                    if (!(t === null || t === void 0 ? void 0 : t.id)) {
                        return undefined;
                    }
                    var r = Vo(Ze, t.id);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Column dropped at position %(pos)d of %(total)d."), {
                        pos: r,
                        total: Ze.length
                    }) : undefined;
                },
                onDragCancel: function e(n) {
                    var t = n.active.id;
                    if (t === undefined) {
                        return undefined;
                    }
                    var r = Vo(Ze, t);
                    return typeof r === "number" ? (0, I.sprintf)((0, T._)("Column returned to its starting position of %(pos)d."), {
                        pos: r
                    }) : undefined;
                }
            };
        }), [ Ze, R ]);
        var Kn = (0, t.useMemo)((function() {
            return {
                screenReaderInstructions: D ? Fo : Uo,
                announcements: D ? Wn : Hn
            };
        }), [ Wn, Hn, D ]);
        var Vn = (0, t.useCallback)((function(e) {
            var n = e.active.id;
            if (n.startsWith("col-") && R) {
                Ie([ i.restrictToHorizontalAxis ]);
                vn(e);
            } else if (n.startsWith("row-") && D) {
                Ie([ i.restrictToVerticalAxis ]);
            }
        }), [ R, D, vn ]);
        var Bn = (0, t.useCallback)((function(e) {
            Ie([]);
            var n = e.active.id;
            if (n.startsWith("col-") && R) {
                bn(e);
            } else if (n.startsWith("row-") && D) {
                hn(e);
            }
        }), [ bn, hn, R, D ]);
        var Fn = (0, t.useCallback)((function(e) {
            var n = e.active.id;
            if (n.startsWith("col-") && R) {
                mn(e);
            }
        }), [ mn, R ]);
        var Un = (0, t.useCallback)((function(e) {
            Ie([]);
            var n = e.active.id;
            if (n.startsWith("row-") && D) {
                yn();
            }
        }), [ yn, D ]);
        var Xn = (0, t.useMemo)((function() {
            return function(e) {
                var n = (0, o.pointerWithin)(e);
                // Collision detection algorithms return an array of collisions
                                if (n.length > 0) {
                    return n;
                }
                // If there are no collisions with the pointer, return rectangle intersections
                                return (0, o.rectIntersection)(e);
            };
        }), []);
        var Yn = Me ? gn.current[Me] : null;
        var Gn = Yn ? xo(Yn).width : undefined;
        var Jn = Be.current ? xo(Be.current).height : undefined;
        
        return r().createElement(Q, Oo({
            "data-test": "table",
            ref: m,
            style: _,
            "data-test-row-selection": A ? V : undefined
        }, b()(U, [].concat(ko(Object.keys(Mo)), [ "style" ]))),  r().createElement(x(), {
            target: cn(),
            eventType: "scroll",
            listener: rn,
            options: Yo
        }),  r().createElement(ye.Provider, {
            value: Ln
        }, _n,  r().createElement(Z, {
            "data-test": "scroll-container",
            onScroll: nn,
            ref: Be,
            style: O,
            $dragging: !!Me
        },  r().createElement(S.ScrollContainerProvider, {
            value: $n
        },  r().createElement(o.DndContext, {
            accessibility: Kn,
            collisionDetection: Xn,
            modifiers: Ee,
            onDragCancel: Un,
            onDragEnd: Bn,
            onDragOver: Fn,
            onDragStart: Vn,
            sensors: Oe
        },  r().createElement(ee, {
            ref: Ve,
            "data-test": "main-table",
            style: F,
            "data-fixed-column": qn ? "true" : undefined,
            $resizableFillLayout: L
        }, An, R ?  r().createElement(a.SortableContext, {
            items: Ze,
            strategy: a.horizontalListSortingStrategy
        }, Dn) : Dn, D ?  r().createElement(a.SortableContext, {
            items: Je,
            strategy: a.verticalListSortingStrategy
        }, Pn) : Pn), Me &&  r().createElement(o.DragOverlay, null,  r().createElement(oe, {
            $width: Gn,
            $height: Jn
        })), Me && Cn))), Tn()));
    }
    Jo.propTypes = Mo;
    Jo.Body = J;
    Jo.Caption = de;
    Jo.Cell = Oe;
    Jo.Head = er;
    Jo.HeadCell = $n;
    Jo.HeadDropdownCell = xr;
    Jo.Row = Ht;
    /* harmony default export */    const Qo = Jo;
    // CONCATENATED MODULE: ./src/Table/index.ts
    module.exports = n;
    /******/})();
