// @splunk/react-ui/TabLayout.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Panel: () => /* reexport */ S,
        default: () => /* reexport */ M
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var o =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/TabBar"
    const l = require("@splunk/react-ui/TabBar");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const u = require("@splunk/react-ui/useControlled");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const d = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "lodash/omit"
    const s = require("lodash/omit");
    var f =  e.n(s);
    // CONCATENATED MODULE: external "styled-components"
    const v = require("styled-components");
    var p =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/themes"
    const b = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/TabLayout/TabLayoutStyles.ts
    var y = p().div.withConfig({
        displayName: "TabLayoutStyles__StyledPanel",
        componentId: "lf0quy-0"
    })([ "", " ", "" ], b.mixins.reset("block"), (function(e) {
        var t = e.$inset;
        return t && (0, v.css)([ "padding:", ";" ], b.variables.spacingLarge);
    }));
    var m = p().div.withConfig({
        displayName: "TabLayoutStyles__Styled",
        componentId: "lf0quy-1"
    })([ "text-align:center;", "" ], (function(e) {
        var t = e.$layout;
        return t === "vertical" && (0, v.css)([ "display:flex;& > [role='tablist']{flex:0 0 auto;}& > ", "{width:100%;}" ], /* sc-sel */ y);
    }));
    // CONCATENATED MODULE: ./src/TabLayout/Panel.tsx
    function h() {
        return h = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, h.apply(null, arguments);
    }
    function g(e, t) {
        if (null == e) return {};
        var n, r, a = O(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (r = 0; r < o.length; r++) {
                n = o[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function O(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var P = {
        children: o().node,
        count: o().number,
        disabled: o().bool,
        elementRef: o().oneOfType([ o().func, o().object ]),
        icon: o().node,
        inset: o().bool,
        label: o().oneOfType([ o().string, o().element ]),
        panelId: o().string.isRequired,
        tooltip: o().node
    };
    function I(e) {
        var t = e.children, n = e.disabled, a = n === void 0 ? false : n, o = e.elementRef, l = e.inset, i = l === void 0 ? true : l, u = e.panelId, c = g(e, [ "children", "disabled", "elementRef", "inset", "panelId" ]);
        // @docs-props-type PanelPropsBase
                
        return r().createElement(y, h({
            $inset: i,
            "data-test-panel-id": u,
            "data-test": "panel",
            ref: o,
            role: "tabpanel"
        }, f()(c, [ "icon", "label", "tooltip" ])), t);
    }
    I.propTypes = P;
    /* harmony default export */    const S = I;
    // CONCATENATED MODULE: ./src/TabLayout/TabLayout.tsx
    function T() {
        return T = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, T.apply(null, arguments);
    }
    function j(e, t) {
        return C(e) || q(e, t) || A(e, t) || w();
    }
    function w() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function A(e, t) {
        if (e) {
            if ("string" == typeof e) return x(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? x(e, t) : void 0;
        }
    }
    function x(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function q(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, a, o, l, i = [], u = !0, c = !1;
            try {
                if (o = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = o.call(n)).done) && (i.push(r.value), i.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, a = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (l = n["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw a;
                }
            }
            return i;
        }
    }
    function C(e) {
        if (Array.isArray(e)) return e;
    }
    function E(e, t) {
        if (null == e) return {};
        var n, r, a = _(e, t);
        if (Object.getOwnPropertySymbols) {
            var o = Object.getOwnPropertySymbols(e);
            for (r = 0; r < o.length; r++) {
                n = o[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (a[n] = e[n]);
            }
        }
        return a;
    }
    function _(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    /** @public */    var k = {
        autoActivate: o().bool,
        activePanelId: o().string,
        children: o().node,
        defaultActivePanelId: o().string,
        elementRef: o().oneOfType([ o().func, o().object ]),
        iconPosition: o().oneOf([ "above", "left" ]),
        layout: o().oneOf([ "horizontal", "vertical" ]),
        maxTabWidth: o().number,
        onChange: o().func
    };
    /**
 * The `TabLayout` is a group of managed `Panels`. Only one panel can be open at a time.
 * `TabLayout` supports both the controlled and uncontrolled patterns.
 */    function R(e) {
        var t = e.autoActivate, a = e.activePanelId, o = e.children, l = e.defaultActivePanelId, u = e.elementRef, s = e.iconPosition, f = e.layout, v = f === void 0 ? "horizontal" : f, p = e.maxTabWidth, b = e.onChange, y = E(e, [ "autoActivate", "activePanelId", "children", "defaultActivePanelId", "elementRef", "iconPosition", "layout", "maxTabWidth", "onChange" ]);
        // @docs-props-type TabLayoutPropsBase
                var h = c()({
            componentName: "TabLayout",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultActivePanelId",
            valuePropName: "activePanelId"
        });
        var g = (0, n.useState)(), O = j(g, 2), P = O[0], I = O[1];
        var S = (0, n.useState)(h ? undefined : l), w = j(S, 2), A = w[0], x = w[1];
 // Could use activePanelIdProp for isControlled route but chose not to b/c we always favor direct prop access in below logic in that scenario
                var q = (0, n.useMemo)((function() {
            return h ? a : A;
        }), [ h, a, A ]);
        var C = (0, n.useState)((0, d.createGUID)()), _ = j(C, 1), k = _[0];
        var R = (0, n.useCallback)((function(e, t) {
            var n = t.selectedTabId;
            if (!h) {
                x(n);
            }
            b === null || b === void 0 ? void 0 : b(e, {
                activePanelId: n
            });
        }), [ h, b ]);
        var M = (0, n.useMemo)((function() {
            var e = null;
            var t = n.Children.toArray(o).filter(n.isValidElement).map((function(t) {
                var a = t.props;
                var o = "".concat(k, "-").concat(a.panelId);
                var l = "".concat(k, "-").concat(a.panelId, "-tab");
                if (a.panelId === q) {
                    e =  (0, n.cloneElement)(t, {
                        "aria-labelledby": l,
                        id: o
                    });
                }
                var u = {};
                // tell TS that PanelProps can contain `data-title`
                                var c = a["data-title"];
                if (c != null) {
                    u["data-title"] = c;
                }
                
                return r().createElement(i().Tab, T({
                    count: a.count,
                    icon: a.icon,
                    key: a.panelId,
                    label: a.label,
                    tabId: a.panelId,
                    id: l,
                    ariaControls: o,
                    tooltip: a.tooltip,
                    disabled: a.disabled
                }, u));
            }));
            I(e);
            return t;
        }), [ q, o, k ]);
        (0, n.useEffect)((function() {
            if (false) {}
        }), [ P ]);
        
        return r().createElement(m, T({
            "data-test-active-panel-id": q,
            "data-test": "tab-layout",
            ref: u,
            $layout: v
        }, y),  r().createElement(i(), {
            autoActivate: t,
            activeTabId: q,
            iconPosition: s,
            layout: v,
            maxTabWidth: p,
            onChange: R
        }, M), P);
    }
    R.Panel = S;
    R.propTypes = k;
    /* harmony default export */    const M = R;
    // CONCATENATED MODULE: ./src/TabLayout/index.ts
    module.exports = t;
    /******/})();
