// @splunk/react-ui/Text.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ oe
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "lodash/keys"
    const o = require("lodash/keys");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/omit"
    const i = require("lodash/omit");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/pickBy"
    const u = require("lodash/pickBy");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "prop-types"
    const c = require("prop-types");
    var d =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-icons/Cross"
    const v = require("@splunk/react-icons/Cross");
    var b =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-icons/Eye"
    const p = require("@splunk/react-icons/Eye");
    var f =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-icons/EyeSlashed"
    const y = require("@splunk/react-icons/EyeSlashed");
    var m =  e.n(y);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const g = require("@splunk/react-ui/Button");
    var h =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/ControlGroup"
    const C = require("@splunk/react-ui/ControlGroup");
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const x = require("@splunk/react-ui/useControlled");
    var k =  e.n(x);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const w = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const S = require("styled-components");
    var O =  e.n(S);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const j = require("@splunk/react-ui/Box");
    var E =  e.n(j);
    // CONCATENATED MODULE: external "@splunk/react-ui/ButtonSimple"
    const A = require("@splunk/react-ui/ButtonSimple");
    // CONCATENATED MODULE: external "@splunk/themes"
    const B = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Text/TextStyles.ts
    var I = "230px";
    var P = O()(h()).withConfig({
        displayName: "TextStyles__StyledClearButton",
        componentId: "eg7n6t-0"
    })([ "display:none;visibility:hidden;" ]);
    var T = O()(E()).withConfig({
        displayName: "TextStyles__StyledBox",
        componentId: "eg7n6t-1"
    })([ "display:flex;justify-content:space-between;flex-grow:1;flex-shrink:1;gap:", ";padding-inline:", ";min-height:", ";overflow:hidden;border:", " solid ", ";border-radius:", ";box-sizing:border-box;background-color:", ";box-shadow:inset 0 1px 1px rgba(0,0,0,0.075);align-items:center;cursor:text;&:has(> input:focus){box-shadow:", ";}", " ", " ", " &[data-inline]{width:", ";flex-basis:", ";}", "" ], B.variables.spacingXSmall, B.variables.spacingSmall, B.variables.inputHeight, B.variables.inputBorderWidth, B.variables.interactiveColorBorder, B.variables.borderRadius, B.variables.interactiveColorBackground, B.variables.focusShadow, (function(e) {
        var r = e.$append;
        return r && (0, S.css)([ "margin-right:-1px;border-top-right-radius:0;border-bottom-right-radius:0;border-right:none;" ]);
    }), (function(e) {
        var r = e.$prepend;
        return r && (0, S.css)([ "border-top-left-radius:0;border-bottom-left-radius:0;" ]);
    }), (function(e) {
        var r = e.$disabled, t = e.$error;
        return r ? (0, S.css)([ "border-color:", ";box-shadow:inset 0 1px 2px rgba(0,0,0,0.025);background-color:", ";cursor:not-allowed;" ], t ? B.variables.interactiveColorAccentErrorWeak : B.variables.interactiveColorBorderDisabled, B.variables.interactiveColorBackgroundDisabled) : (0, 
        S.css)([ "", " &:hover{border-color:", ";background-color:", ";}", "{visibility:visible;display:block;}" ], t && (0, 
        S.css)([ "border-color:", ";" ], B.variables.interactiveColorAccentError), t ? B.variables.interactiveColorAccentErrorStrong : B.variables.interactiveColorBorderHover, B.variables.interactiveColorOverlayHover, /* sc-sel */ P);
    }), I, I, (function(e) {
        var r = e.$isTimeInput;
        return r && (0, S.css)([ "@media screen and (min--moz-device-pixel-ratio:0){", "{display:none;}}" ], /* sc-sel */ P);
    }));
    /* Some of these need greater specificity than input[type=text] */    var q = O().input.withConfig({
        displayName: "TextStyles__StyledInput",
        componentId: "eg7n6t-2"
    })([ "", ";", " width:100%;outline-style:none;border:0;background:transparent;flex-grow:1;font-size:inherit;&[type='time']{&::-webkit-calendar-picker-indicator{display:none;}}&::placeholder{color:", ";opacity:1;}&[disabled],&[aria-disabled='true']{color:", ";&::placeholder{color:", ";}}" ], B.mixins.reset("inline-flex"), B.mixins.typography("body", {
        lineHeight: "single",
        color: "active"
    }), B.variables.contentColorMuted, B.variables.contentColorDisabled, B.variables.contentColorDisabled);
    var $ = O().div.withConfig({
        displayName: "TextStyles__StyledAdornment",
        componentId: "eg7n6t-3"
    })([ "display:inline-flex;align-items:center;justify-content:center;pointer-events:none;z-index:1;color:", ";", " ", ";& > button{", "}& > button:focus,& > a:focus{box-shadow:", ";}" ], B.variables.contentColorMuted, (function() {
        return (0, S.css)([ "height:calc(", " - ", " * 2);" ], B.variables.inputHeight, B.variables.inputBorderWidth);
    }), (function(e) {
        var r = e.$disabled;
        return r && (0, S.css)([ "color:", ";" ], B.variables.contentColorDisabled);
    }), (0, A.buttonMixin)({
        borderColor: B.variables.transparent,
        borderColorHover: B.variables.transparent,
        borderColorActive: B.variables.transparent,
        borderColorDisabled: B.variables.transparent,
        backgroundHover: B.variables.interactiveColorOverlayHover,
        backgroundActive: B.variables.interactiveColorOverlayActive
    }), B.variables.focusShadowInset);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function D(e, r) {
        if (e) {
            if (typeof e === "function") {
                e(r);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = r;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Text/Text.tsx
    function _(e) {
        "@babel/helpers - typeof";
        return _ = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, _(e);
    }
    function N(e) {
        return M(e) || H(e) || X(e) || F();
    }
    function F() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function H(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function M(e) {
        if (Array.isArray(e)) return J(e);
    }
    function R(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function z(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? R(Object(t), !0).forEach((function(r) {
                V(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : R(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function V(e, r, t) {
        return (r = K(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function K(e) {
        var r = L(e, "string");
        return "symbol" == _(r) ? r : r + "";
    }
    function L(e, r) {
        if ("object" != _(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != _(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    function W() {
        return W = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, W.apply(null, arguments);
    }
    function G(e, r) {
        return Y(e) || Q(e, r) || X(e, r) || U();
    }
    function U() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function X(e, r) {
        if (e) {
            if ("string" == typeof e) return J(e, r);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? J(e, r) : void 0;
        }
    }
    function J(e, r) {
        (null == r || r > e.length) && (r = e.length);
        for (var t = 0, n = Array(r); t < r; t++) {
            n[t] = e[t];
        }
        return n;
    }
    function Q(e, r) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var n, o, a, i, l = [], u = !0, s = !1;
            try {
                if (a = (t = t.call(e)).next, 0 === r) {
                    if (Object(t) !== t) return;
                    u = !1;
                } else for (;!(u = (n = a.call(t)).done) && (l.push(n.value), l.length !== r); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != t["return"] && (i = t["return"](), Object(i) !== i)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return l;
        }
    }
    function Y(e) {
        if (Array.isArray(e)) return e;
    }
    function Z(e, r) {
        if (null == e) return {};
        var t, n, o = ee(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                t = a[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function ee(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    /** @public */
    /** @public */
    /** @public */    var re = {
        append: d().bool,
        autoCapitalize: d().string,
        autoComplete: d().string,
        autoCorrect: d().string,
        autoFocus: d().bool,
        canClear: d().bool,
        children: d().node,
        defaultValue: d().string,
        describedBy: d().string,
        disabled: d().oneOfType([ d().bool, d().oneOf([ "dimmed" ]) ]),
        elementRef: d().oneOfType([ d().func, d().object ]),
        endAdornment: d().node,
        error: d().bool,
        inline: d().bool,
        /** @private. */
        inputClassName: d().string,
        inputId: d().string,
        inputRef: d().oneOfType([ d().func, d().object ]),
        labelledBy: d().string,
        maxLength: d().number,
        name: d().string,
        onBlur: d().func,
        onChange: d().func,
        onFocus: d().func,
        onKeyDown: d().func,
        onSelect: d().func,
        onClick: d().func,
        /** @private. */
        placeholder: d().string,
        prepend: d().bool,
        /** @private. */
        required: d().bool,
        spellCheck: d().bool,
        tabIndex: d().number,
        startAdornment: d().node,
        /** @private */
        splunkTheme: d().object,
        type: d().string,
        passwordVisibilityToggle: d().bool,
        value: d().string
    };
    var te =  n().createElement(b(), {
        inline: false
    });
    /** Note: Text places role and aria props onto the input. All other props are placed on the wrapper. */    function ne(e) {
        var r;
        var o = e.append, i = o === void 0 ? false : o, u = e.autoCapitalize, c = e.autoComplete, d = e.autoCorrect, v = e.autoFocus, b = v === void 0 ? false : v, p = e.canClear, y = p === void 0 ? false : p, g = e.children, x = e.defaultValue, S = e.disabled, O = S === void 0 ? false : S, j = e.describedBy, E = e.elementRef, A = e.endAdornment, B = e.error, I = B === void 0 ? false : B, _ = e.inline, F = _ === void 0 ? false : _, H = e.inputClassName, M = e.inputId, R = e.inputRef, V = e.labelledBy, K = e.maxLength, L = e.name, U = e.onBlur, X = e.onChange, J = e.onClick, Q = e.onFocus, Y = e.onKeyDown, ee = e.onSelect, re = e.passwordVisibilityToggle, ne = re === void 0 ? false : re, oe = e.placeholder, ae = e.prepend, ie = ae === void 0 ? false : ae, le = e.required, ue = e.spellCheck, se = e.startAdornment, ce = e.tabIndex, de = ce === void 0 ? 0 : ce, ve = e.title, be = e.type, pe = be === void 0 ? "text" : be, fe = e.value, ye = Z(e, [ "append", "autoCapitalize", "autoComplete", "autoCorrect", "autoFocus", "canClear", "children", "defaultValue", "disabled", "describedBy", "elementRef", "endAdornment", "error", "inline", "inputClassName", "inputId", "inputRef", "labelledBy", "maxLength", "name", "onBlur", "onChange", "onClick", "onFocus", "onKeyDown", "onSelect", "passwordVisibilityToggle", "placeholder", "prepend", "required", "spellCheck", "startAdornment", "tabIndex", "title", "type", "value" ]);
        // @docs-props-type TextPropsBase
                var me = (0, t.useContext)(C.ControlGroupContext);
        var ge = k()({
            componentName: "Text",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var he = (0, t.useRef)(null);
        var Ce = (0, t.useState)(x || ""), xe = G(Ce, 2), ke = xe[0], we = xe[1];
        var Se = (0, t.useState)(true), Oe = G(Se, 2), je = Oe[0], Ee = Oe[1];
        var Ae = (0, t.useMemo)((function() {
            return ge ? fe : ke;
        }), [ ge, fe, ke ]);
        var Be = (0, t.useCallback)((function(e) {
            var r = e.target.value;
            U === null || U === void 0 ? void 0 : U(e, {
                value: r,
                name: L
            });
        }), [ L, U ]);
        var Ie = (0, t.useCallback)((function(e) {
            var r = e.target.value;
            if (!ge) {
                we(r);
            }
            X === null || X === void 0 ? void 0 : X(e, {
                value: r,
                name: L
            });
        }), [ ge, L, X ]);
        var Pe = (0, t.useCallback)((function(e) {
            var r;
            e.preventDefault();
            var t = "";
            if (!ge) {
                we(t);
            }
            he === null || he === void 0 ? void 0 : (r = he.current) === null || r === void 0 ? void 0 : r.focus();
            X === null || X === void 0 ? void 0 : X(e, {
                value: t,
                name: L
            });
        }), [ ge, L, X ]);
        var Te = (0, t.useCallback)((function(e) {
            J === null || J === void 0 ? void 0 : J(e);
            e.stopPropagation();
        }), [ J ]);
        var qe = (0, t.useCallback)((function(e) {
            if (document.activeElement !== he.current) {
                var r;
                (r = he.current) === null || r === void 0 ? void 0 : r.focus();
            }
            J === null || J === void 0 ? void 0 : J(e);
        }), [ J ]);
        var $e = (0, t.useCallback)((function(e) {
            e.stopPropagation();
        }), []);
        var De = (0, t.useCallback)((function(e) {
            var r = e.target.value;
            Q === null || Q === void 0 ? void 0 : Q(e, {
                value: r,
                name: L
            });
        }), [ L, Q ]);
        var _e = (0, t.useCallback)((function(e) {
            Y === null || Y === void 0 ? void 0 : Y(e);
        }), [ Y ]);
        var Ne = (0, t.useCallback)((function(e) {
            he.current = e;
            D(R, e);
        }), [ R, he ]);
        var Fe = (0, t.useCallback)((function(e) {
            ee === null || ee === void 0 ? void 0 : ee(e);
        }), [ ee ]);
        var He = (0, t.useCallback)((function() {
            Ee(!je);
        }), [ je ]);
        var Me = (0, t.useCallback)((function() {
            var e;
            var r = {
                $disabled: !!O,
                $position: "end"
            };
            if (A) {
                
                return n().createElement($, W({
                    "data-test": "end-adornment",
                    onClick: $e
                }, r), A);
            }
            if (ne && !O) {
                var t = je ?  n().createElement(f(), null) :  n().createElement(m(), null);
                var o = je ? (0, w._)("Show password") : (0, w._)("Hide password");
                
                return n().createElement($, r,  n().createElement(h(), {
                    appearance: "secondary",
                    "data-test": "password-toggle",
                    inline: false,
                    onClick: He,
                    type: "button",
                    icon: t,
                    "aria-label": o
                }));
            }
            var a = (e = me.labelAttrs) === null || e === void 0 ? void 0 : e.text;
            var i = a ? (0, w._)("Clear ".concat(a, " text field")) : (0, w._)("Clear text field");
            if (!!Ae && y) {
                
                return n().createElement($, r, !O &&  n().createElement(P, {
                    "data-test": "clear",
                    appearance: "secondary",
                    "aria-label": i,
                    onClick: Pe,
                    icon: te
                }));
            }
            return undefined;
        }), [ y, (r = me.labelAttrs) === null || r === void 0 ? void 0 : r.text, O, Ae, A, $e, Pe, He, je, ne ]);
        var Re = (0, t.useCallback)((function() {
            var e = {
                $disabled: !!O,
                $position: "start"
            };
            if (se) {
                
                return n().createElement($, W({
                    "data-test": "start-adornment",
                    onClick: $e
                }, e), se);
            }
            return undefined;
        }), [ O, $e, se ]);
        (0, t.useEffect)((function() {
            if (false) {}
        }), [ y, ne ]);
        var ze = (0, t.useMemo)((function() {
            return z(z({
                role: "textbox"
            }, s()(ye, (function(e, r) {
                return r === "role" || r.indexOf("aria-") === 0;
            }))), {}, {
                "aria-describedby": j,
                "aria-labelledby": V,
                "aria-invalid": I || undefined,
                "aria-disabled": O === "dimmed" || undefined
            });
        }), [ j, O, I, V, ye ]);
        var Ve = z({
            $append: i || undefined,
            $error: I,
            $prepend: ie || undefined
        }, l()(ye, [ "inputRef", "onBlur", "onClick", "onChange", "onFocus", "onKeyDown", "onSelect" ].concat(N(a()(ze)))));
        var Ke = je ? "password" : "text";
        var Le = ne ? Ke : pe;
        var We = z(z({}, ze), {}, {
            "data-test": "textbox",
            autoCapitalize: u,
            autoComplete: c,
            autoCorrect: d,
            autoFocus: b,
            className: H,
            id: M,
            maxLength: K,
            name: L,
            onChange: Ie,
            onKeyDown: _e,
            onSelect: Fe,
            onClick: Te,
            onFocus: De,
            onBlur: Be,
            placeholder: oe,
            ref: Ne,
            required: le,
            spellCheck: ue,
            title: ve,
            tabIndex: de,
            type: Le,
            value: Ae,
            $error: I
        });
        // Do not render value in DOM when type is password, or passwordVisibilityToggle is enabled
                var Ge = pe === "password" || ne ? undefined : Ae;
        // Firefox hack (SUI-2716). Remove when fixed: https://bugzilla.mozilla.org/show_bug.cgi?id=1427173
                var Ue = pe === "time";
        
        return n().createElement(T, W({
            tabIndex: -1,
            $isTimeInput: Ue,
            "data-test": "text",
            "data-test-value": Ge,
            "data-test-disabled": O === "dimmed" && "dimmed" || O && "disabled" || undefined,
            ref: E,
            flex: true,
            inline: F,
            onClick: qe,
            $disabled: !!O
        }, Ve), Re(), O ?  n().createElement(q, W({
            className: H,
            "data-test": "textbox",
            disabled: O === true,
            placeholder: oe,
            readOnly: true,
            type: Le,
            value: Ae
        }, ze)) :  n().createElement(q, W({}, We, {
            onClick: Te
        })), g, Me());
    }
    ne.propTypes = re;
    ne.componentType = "Text";
    /* harmony default export */    const oe = ne;
    // only for styled components using Text
        // CONCATENATED MODULE: ./src/Text/index.ts
    module.exports = r;
    /******/})();
