// @splunk/react-ui/TextArea.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ de
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/keys"
    const o = require("lodash/keys");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "lodash/omit"
    const i = require("lodash/omit");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "lodash/pickBy"
    const u = require("lodash/pickBy");
    var c =  e.n(u);
    // CONCATENATED MODULE: external "lodash/throttle"
    const s = require("lodash/throttle");
    var d =  e.n(s);
    // CONCATENATED MODULE: external "prop-types"
    const f = require("prop-types");
    var p =  e.n(f);
    // CONCATENATED MODULE: external "@splunk/react-ui/EventListener"
    const v = require("@splunk/react-ui/EventListener");
    var b =  e.n(v);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const m = require("@splunk/react-ui/useControlled");
    var y =  e.n(m);
    // CONCATENATED MODULE: ./src/TextArea/syncHeightWithShadow.ts
    // This file was adapted from [Call-Em-All's material-ui](https://github.com/mui-org/material-ui),
    // which is MIT-licensed:
    // The MIT License (MIT)
    // Copyright (c) 2014 Call-Em-All
    // Permission is hereby granted, free of charge, to any person obtaining a copy
    // of this software and associated documentation files (the "Software"), to deal
    // in the Software without restriction, including without limitation the rights
    // to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
    // copies of the Software, and to permit persons to whom the Software is
    // furnished to do so, subject to the following conditions:
    // The above copyright notice and this permission notice shall be included in all
    // copies or substantial portions of the Software.
    // THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
    // IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
    // FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
    // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
    // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
    // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
    // SOFTWARE.
    // End of MIT license text.
    function h(e) {
        var n = e.height, r = e.rowsMax, t = e.rowsMin, o = e.shadow;
        if (o) {
            var a = window.getComputedStyle(o);
            var i = a.getPropertyValue("line-height");
            // jsdom does not properly inherit/re-calculate the line-height property so we skip in these cases SUI-1943, SUI-2428
                        if (i === "inherit" || i === "normal") {
                return false;
            }
            var l = parseInt(i, 10);
            var u = parseInt(a.getPropertyValue("border-top-width"), 10);
            var c = parseInt(a.getPropertyValue("border-bottom-width"), 10);
            var s = o.scrollHeight + u + c;
            /* eslint-disable no-restricted-globals */
            // Leverage the global `isNaN` here for IE support, rather than Number.isNaN.
            // We know newHeight is numeric because we parseInt() above, so `isNaN` should be reliable.
                        if (s === undefined || isNaN(s)) {
                return false;
            }
            /* eslint-enable no-restricted-globals */            if (r >= t) {
                var d = r * l + u + c;
                s = Math.min(d, s);
            }
            var f = t * l + u + c;
            s = Math.max(f, s);
            if (n !== s) {
                return s;
            }
        }
        return false;
    }
    // CONCATENATED MODULE: external "styled-components"
    const g = require("styled-components");
    var w =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Box"
    const C = require("@splunk/react-ui/Box");
    var x =  e.n(C);
    // CONCATENATED MODULE: external "@splunk/themes"
    const S = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/TextArea/TextAreaStyles.ts
    var A = "230px";
    var O = S.variables.spacingSmall;
    // variables.inputHeight can't be used with syncHeight/auto-expanding rows.
    // Use a custom value here to ensure the single-line TextArea matches the Text height
    // and keeps the vertical spacing above the label consistent (SUI-8053).
        var E = (0, S.pick)({
        comfortable: "5px",
        compact: "3px"
    });
    var j = w()(x()).withConfig({
        displayName: "TextAreaStyles__StyledBox",
        componentId: "gfy8yp-0"
    })([ "display:grid;grid-template-columns:[startAdornment] min-content [content] minmax(0,1fr) [endAdornment] min-content;position:relative;min-height:", ";overflow:hidden;border:", " solid ", ";border-radius:", ";background-color:", ";box-shadow:inset 0 1px 1px rgba(0,0,0,0.075);align-items:center;padding:", " ", ";&:hover:not([disabled]){border-color:", ";}&:not([disabled]):has(textarea:focus){box-shadow:", ";color:", ";}", " ", " ", " ", " ", "" ], S.variables.inputHeight, S.variables.inputBorderWidth, S.variables.interactiveColorBorder, S.variables.borderRadius, S.variables.interactiveColorBackground, E, O, S.variables.interactiveColorBorderHover, S.variables.focusShadow, S.variables.contentColorActive, (function(e) {
        var n = e.$error;
        return n && (0, g.css)([ "border-color:", ";&:hover:not([disabled]){border-color:", ";}" ], S.variables.interactiveColorAccentError, S.variables.interactiveColorAccentErrorStrong);
    }), (function(e) {
        var n = e.$append;
        return n && (0, g.css)([ "margin-right:-1px;border-top-right-radius:0;border-bottom-right-radius:0;border-right:none;" ]);
    }), (function(e) {
        var n = e.$prepend;
        return n && (0, g.css)([ "border-top-left-radius:0;border-bottom-left-radius:0;" ]);
    }), (function(e) {
        var n = e.disabled;
        return n && (0, g.css)([ "background-color:", ";border-color:", ";box-shadow:inset 0 1px 2px rgba(0,0,0,0.025);" ], S.variables.interactiveColorBackgroundDisabled, S.variables.interactiveColorBorderDisabled);
    }), (function(e) {
        var n = e.$inline;
        return n && (0, g.css)([ "display:inline-grid;width:", ";" ], A);
    }));
    // This wrapper ensures the textarea and shadow text area are the same width.
    // The wrapper's size is determined by the parent grid layout: the textarea and shadow
    // are both set to be 100% of the wrapper.
    // Having the same width is necessary for the height to be calculated correctly
    // so that the value wrapping behavior is correct
        var k = w().div.withConfig({
        displayName: "TextAreaStyles__StyledTextAreaAndShadowWrapper",
        componentId: "gfy8yp-1"
    })([ "", ";position:relative;overflow:hidden;flex-grow:1;grid-column:content;" ], S.mixins.reset("inline-flex"));
    var I = w().textarea.withConfig({
        displayName: "TextAreaStyles__StyledTextArea",
        componentId: "gfy8yp-2"
    })([ "", ";color:", ";position:relative;resize:none;overflow:auto;white-space:pre-wrap;width:100%;line-height:20px;&::placeholder{color:", ";opacity:1;}&[disabled]{cursor:not-allowed;color:", ";&::placeholder{color:", ";}}" ], S.mixins.reset(), S.variables.contentColorActive, S.variables.contentColorMuted, S.variables.contentColorDisabled, S.variables.contentColorDisabled);
    var T = w()(I).withConfig({
        displayName: "TextAreaStyles__StyledTextAreaShadow",
        componentId: "gfy8yp-3"
    })([ "position:absolute;overflow:hidden;left:-10000px;top:-10000px;visibility:hidden;pointer-events:none;" ]);
    var P = (0, g.css)([ "display:inline-flex;min-height:1lh;align-self:start;align-items:center;justify-content:center;pointer-events:none;color:", ";", ";" ], S.variables.contentColorMuted, (function(e) {
        var n = e.$disabled;
        return n && (0, g.css)([ "cursor:not-allowed;color:", ";" ], S.variables.contentColorDisabled);
    }));
    var B = w().div.withConfig({
        displayName: "TextAreaStyles__StyledStartAdornment",
        componentId: "gfy8yp-4"
    })([ "", " grid-column:startAdornment;margin-inline-end:", ";" ], P, S.variables.spacingSmall);
    var M = w().div.withConfig({
        displayName: "TextAreaStyles__StyledEndAdornment",
        componentId: "gfy8yp-5"
    })([ "", " grid-column:endAdornment;margin-inline-start:", ";" ], P, S.variables.spacingSmall);
    // CONCATENATED MODULE: ./src/utils/ssrDocument.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var N = {
        body: {
            appendChild: function e() {
                return [];
            }
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        activeElement: {
            blur: function e() {},
            nodeName: ""
        },
        querySelector: function e() {
            return null;
        },
        querySelectorAll: function e() {
            return [];
        },
        getElementById: function e() {
            return null;
        },
        createEvent: function e() {
            return {
                initEvent: function e() {}
            };
        },
        createElement: function e() {
            return {
                children: [],
                childNodes: [],
                style: {},
                setAttribute: function e() {},
                getElementsByTagName: function e() {
                    return [];
                }
            };
        },
        createElementNS: function e() {
            return {};
        },
        importNode: function e() {
            return null;
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        }
    };
    function q() {
        var e = typeof document !== "undefined" ? document : N;
        return e;
    }
    var _ = q();
    /* harmony default export */    const D = /* unused pure expression or super */ null && _;
    // CONCATENATED MODULE: ./src/utils/ssrWindow.ts
    /* eslint-disable @typescript-eslint/no-empty-function */
    var $ = {
        document: N,
        navigator: {
            userAgent: ""
        },
        location: {
            hash: "",
            host: "",
            hostname: "",
            href: "",
            origin: "",
            pathname: "",
            protocol: "",
            search: ""
        },
        history: {
            replaceState: function e() {},
            pushState: function e() {},
            go: function e() {},
            back: function e() {}
        },
        CustomEvent: function e() {
            return this;
        },
        addEventListener: function e() {},
        removeEventListener: function e() {},
        getComputedStyle: function e() {
            return {
                getPropertyValue: function e() {
                    return "";
                }
            };
        },
        Image: function e() {},
        Date: function e() {},
        screen: {},
        setTimeout: function e() {},
        clearTimeout: function e() {},
        matchMedia: function e() {
            return {};
        },
        requestAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                n();
                return null;
            }
            return setTimeout(n, 0);
        },
        cancelAnimationFrame: function e(n) {
            if (typeof setTimeout === "undefined") {
                return;
            }
            clearTimeout(n);
        }
    };
    function F() {
        var e = typeof window !== "undefined" ? window : $;
        return e;
    }
    var L = F();
    /* harmony default export */    const R = L;
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function V(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/TextArea/TextArea.tsx
    function z(e) {
        "@babel/helpers - typeof";
        return z = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, z(e);
    }
    function K() {
        return K = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, K.apply(null, arguments);
    }
    function H(e) {
        return G(e) || U(e) || re(e) || W();
    }
    function W() {
        throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function U(e) {
        if ("undefined" != typeof Symbol && null != e[Symbol.iterator] || null != e["@@iterator"]) return Array.from(e);
    }
    function G(e) {
        if (Array.isArray(e)) return te(e);
    }
    function J(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function Q(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? J(Object(r), !0).forEach((function(n) {
                X(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : J(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function X(e, n, r) {
        return (n = Y(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function Y(e) {
        var n = Z(e, "string");
        return "symbol" == z(n) ? n : n + "";
    }
    function Z(e, n) {
        if ("object" != z(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != z(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function ee(e, n) {
        return ae(e) || oe(e, n) || re(e, n) || ne();
    }
    function ne() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function re(e, n) {
        if (e) {
            if ("string" == typeof e) return te(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? te(e, n) : void 0;
        }
    }
    function te(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function oe(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, o, a, i, l = [], u = !0, c = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = a.call(r)).done) && (l.push(t.value), l.length !== n); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (i = r["return"](), Object(i) !== i)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function ae(e) {
        if (Array.isArray(e)) return e;
    }
    function ie(e, n) {
        if (null == e) return {};
        var r, t, o = le(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function le(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */
    /** @public */    var ue = {
        append: p().bool,
        autoCapitalize: p().string,
        autoComplete: p().string,
        autoCorrect: p().string,
        autoFocus: p().bool,
        children: p().node,
        defaultValue: p().string,
        describedBy: p().string,
        disabled: p().bool,
        elementRef: p().oneOfType([ p().func, p().object ]),
        endAdornment: p().node,
        error: p().bool,
        inline: p().bool,
        /** @private. */
        inputClassName: p().string,
        inputId: p().string,
        inputRef: p().oneOfType([ p().func, p().object ]),
        labelledBy: p().string,
        maxLength: p().number,
        name: p().string,
        onBlur: p().func,
        onChange: p().func,
        onFocus: p().func,
        onKeyDown: p().func,
        onSelect: p().func,
        onInputClick: p().func,
        prepend: p().bool,
        /** @private. */
        required: p().bool,
        rowsMax: p().number,
        rowsMin: p().number,
        spellCheck: p().bool,
        tabIndex: p().number,
        startAdornment: p().node,
        value: p().string
    };
    var ce = function e() {};
    /** Note: TextArea places role and aria props onto the input. All other props are placed on the wrapper. */    function se(e) {
        var n = e.append, o = n === void 0 ? false : n, i = e.autoCapitalize, u = e.autoComplete, s = e.autoCorrect, f = e.autoFocus, p = f === void 0 ? false : f, v = e.children, m = e.disabled, g = m === void 0 ? false : m, w = e.describedBy, C = e.defaultValue, x = e.elementRef, S = e.error, A = S === void 0 ? false : S, O = e.endAdornment, E = e.inline, P = E === void 0 ? false : E, N = e.inputClassName, q = e.inputId, _ = e.inputRef, D = e.labelledBy, $ = e.maxLength, F = e.name, L = e.onBlur, z = e.onChange, W = e.onFocus, U = e.onInputClick, G = e.onKeyDown, J = e.onSelect, X = e.placeholder, Y = e.prepend, Z = Y === void 0 ? false : Y, ne = e.required, re = e.rowsMin, te = re === void 0 ? 2 : re, oe = e.rowsMax, ae = oe === void 0 ? 8 : oe, le = e.spellCheck, ue = e.startAdornment, se = e.tabIndex, de = se === void 0 ? 0 : se, fe = e.value, pe = ie(e, [ "append", "autoCapitalize", "autoComplete", "autoCorrect", "autoFocus", "children", "disabled", "describedBy", "defaultValue", "elementRef", "error", "endAdornment", "inline", "inputClassName", "inputId", "inputRef", "labelledBy", "maxLength", "name", "onBlur", "onChange", "onFocus", "onInputClick", "onKeyDown", "onSelect", "placeholder", "prepend", "required", "rowsMin", "rowsMax", "spellCheck", "startAdornment", "tabIndex", "value" ]);
        // @docs-props-type TextAreaPropsBase
                (0, r.useEffect)((function() {
            if (false) {}
        }), [ te, ae ]);
        var ve = y()({
            componentName: "TextArea",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0]
        });
        var be = (0, r.useState)(C || ""), me = ee(be, 2), ye = me[0], he = me[1];
        var ge = (0, r.useState)(), we = ee(ge, 2), Ce = we[0], xe = we[1];
        var Se = (0, r.useRef)(null);
        var Ae = (0, r.useRef)(null);
        var Oe = ve ? fe : ye;
        var Ee = (0, r.useCallback)((function(e) {
            Se.current = e;
            V(_, e);
        }), [ _, Se ]);
        var je = (0, r.useCallback)((function(e) {
            G === null || G === void 0 ? void 0 : G(e);
        }), [ G ]);
        var ke = (0, r.useCallback)((function(e) {
            J === null || J === void 0 ? void 0 : J(e);
        }), [ J ]);
        var Ie = (0, r.useCallback)((function(e) {
            U === null || U === void 0 ? void 0 : U(e);
        }), [ U ]);
        var Te = (0, r.useCallback)((function(e) {
            var n = e.target.value;
            W === null || W === void 0 ? void 0 : W(e, {
                value: n,
                name: F
            });
        }), [ F, W ]);
        var Pe = (0, r.useCallback)((function(e) {
            var n = e.target.value;
            L === null || L === void 0 ? void 0 : L(e, {
                value: n,
                name: F
            });
        }), [ F, L ]);
        var Be = (0, r.useCallback)((function(e) {
            var n = e.target.value;
            if (!ve) {
                he(n);
            }
            z === null || z === void 0 ? void 0 : z(e, {
                value: n,
                name: F
            });
        }), [ z, he, ve, F ]);
        var Me = (0, r.useCallback)((function() {
            var e = h({
                height: Ce,
                rowsMax: ae,
                rowsMin: te,
                shadow: Ae.current
            });
            if (e !== false) {
                xe(e);
            }
        }), [ xe, Ae, ae, te, Ce ]);
        var Ne = (0, r.useMemo)((function() {
            return d()(Me, 100);
        }), [ Me ]);
        (0, r.useLayoutEffect)((function() {
            Me();
        }));
        var qe = Q(Q({}, c()(pe, (function(e, n) {
            return n === "role" || n.indexOf("aria-") === 0;
        }))), {}, {
            "aria-describedby": w,
            "aria-labelledby": D,
            "aria-invalid": A || undefined
        });
        var _e = l()(pe, [ "inputRef", "onBlur", "onChange", "onFocus", "onKeyDown", "onSelect", "onInputClick" ].concat(H(a()(qe))));
        var De = {
            $append: o || undefined,
            $error: A,
            $prepend: Z || undefined
        };
        var $e = Q(Q({}, qe), {}, {
            "data-test": "textbox",
            autoCapitalize: i,
            autoComplete: u,
            autoCorrect: s,
            autoFocus: p,
            className: N,
            disabled: g,
            id: q,
            maxLength: $,
            name: F,
            onBlur: Pe,
            onChange: Be,
            onFocus: Te,
            onKeyDown: je,
            onSelect: ke,
            onClick: Ie,
            placeholder: X,
            // support HTML attribute
            ref: Ee,
            required: ne,
            spellCheck: le,
            style: {
                height: Ce
            },
            tabIndex: de,
            value: Oe,
            $error: A
        });
        
        return t().createElement(t().Fragment, null,  t().createElement(j, K({
            tabIndex: -1,
            "data-test": "text-area",
            "data-test-value": Oe,
            "data-test-disabled": g ? "disabled" : undefined,
            ref: x,
            $inline: P,
            disabled: g
        }, _e, De), ue &&  t().createElement(B, {
            $disabled: !!g
        }, ue),  t().createElement(k, null,  t().createElement(T, {
            "aria-hidden": "true",
            tabIndex: -1,
            onChange: ce,
            rows: te,
            ref: Ae,
            value: X && !Oe ? X : Oe
        }),  t().createElement(I, $e)), v, O &&  t().createElement(M, {
            $disabled: !!g
        }, O)),  t().createElement(b(), {
            target: R,
            eventType: "resize",
            listener: Ne
        }));
    }
    se.propTypes = ue;
    se.componentType = "TextArea";
    /* harmony default export */    const de = se;
    // only for styled components using TextArea
        // CONCATENATED MODULE: ./src/TextArea/index.ts
    module.exports = n;
    /******/})();
