// @splunk/react-ui/Tooltip.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var r = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, r) => {
            /******/ for (var t in r) {
                /******/ if (e.o(r, t) && !e.o(n, t)) {
                    /******/ Object.defineProperty(n, t, {
                        enumerable: true,
                        get: r[t]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ Y
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var t =  e.n(r);
    // CONCATENATED MODULE: external "lodash/throttle"
    const o = require("lodash/throttle");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "prop-types"
    const l = require("prop-types");
    var i =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/react-icons/InformationCircle"
    const u = require("@splunk/react-icons/InformationCircle");
    var s =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/react-ui/Popover"
    const c = require("@splunk/react-ui/Popover");
    var f =  e.n(c);
    // CONCATENATED MODULE: external "@splunk/react-ui/ScreenReaderContent"
    const p = require("@splunk/react-ui/ScreenReaderContent");
    var v =  e.n(p);
    // CONCATENATED MODULE: external "@splunk/react-ui/useControlled"
    const d = require("@splunk/react-ui/useControlled");
    var m =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const b = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "@splunk/ui-utils/id"
    const y = require("@splunk/ui-utils/id");
    // CONCATENATED MODULE: external "styled-components"
    const g = require("styled-components");
    var O =  e.n(g);
    // CONCATENATED MODULE: external "@splunk/react-ui/Button"
    const h = require("@splunk/react-ui/Button");
    var S =  e.n(h);
    // CONCATENATED MODULE: external "@splunk/themes"
    const C = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Tooltip/TooltipStyles.ts
    var k = O().span.withConfig({
        displayName: "TooltipStyles__Styled",
        componentId: "sc-67fvgm-0"
    })([ "", ";position:relative;" ], C.mixins.reset("block"));
    var E = O().span.withConfig({
        displayName: "TooltipStyles__StyledInline",
        componentId: "sc-67fvgm-1"
    })([ "", ";position:relative;" ], C.mixins.reset("inline-block"));
    var T = O().span.withConfig({
        displayName: "TooltipStyles__StyledToggle",
        componentId: "sc-67fvgm-2"
    })([ "display:block;" ]);
    var P = O().span.withConfig({
        displayName: "TooltipStyles__StyledToggleInline",
        componentId: "sc-67fvgm-3"
    })([ "display:inline-block;" ]);
    var j = O().div.withConfig({
        displayName: "TooltipStyles__StyledContent",
        componentId: "sc-67fvgm-4"
    })([ "", ";padding:", ";user-select:none;" ], C.mixins.typography("smallBody", {
        color: "active"
    }), C.variables.spacingSmall);
    var w = O()(S()).withConfig({
        displayName: "TooltipStyles__StyledButton",
        componentId: "sc-67fvgm-5"
    })([ "cursor:help;border-radius:50%;" ]);
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function R(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Tooltip/Tooltip.tsx
    function L(e) {
        "@babel/helpers - typeof";
        return L = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, L(e);
    }
    function q() {
        return q = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var r = arguments[n];
                for (var t in r) {
                    ({}).hasOwnProperty.call(r, t) && (e[t] = r[t]);
                }
            }
            return e;
        }, q.apply(null, arguments);
    }
    function I(e, n) {
        var r = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var t = Object.getOwnPropertySymbols(e);
            n && (t = t.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), r.push.apply(r, t);
        }
        return r;
    }
    function A(e) {
        for (var n = 1; n < arguments.length; n++) {
            var r = null != arguments[n] ? arguments[n] : {};
            n % 2 ? I(Object(r), !0).forEach((function(n) {
                _(e, n, r[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : I(Object(r)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
            }));
        }
        return e;
    }
    function _(e, n, r) {
        return (n = D(n)) in e ? Object.defineProperty(e, n, {
            value: r,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = r, e;
    }
    function D(e) {
        var n = M(e, "string");
        return "symbol" == L(n) ? n : n + "";
    }
    function M(e, n) {
        if ("object" != L(e) || !e) return e;
        var r = e[Symbol.toPrimitive];
        if (void 0 !== r) {
            var t = r.call(e, n || "default");
            if ("object" != L(t)) return t;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    function N(e, n) {
        return K(e) || F(e, n) || B(e, n) || x();
    }
    function x() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function B(e, n) {
        if (e) {
            if ("string" == typeof e) return H(e, n);
            var r = {}.toString.call(e).slice(8, -1);
            return "Object" === r && e.constructor && (r = e.constructor.name), "Map" === r || "Set" === r ? Array.from(e) : "Arguments" === r || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? H(e, n) : void 0;
        }
    }
    function H(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var r = 0, t = Array(n); r < n; r++) {
            t[r] = e[r];
        }
        return t;
    }
    function F(e, n) {
        var r = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != r) {
            var t, o, a, l, i = [], u = !0, s = !1;
            try {
                if (a = (r = r.call(e)).next, 0 === n) {
                    if (Object(r) !== r) return;
                    u = !1;
                } else for (;!(u = (t = a.call(r)).done) && (i.push(t.value), i.length !== n); u = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!u && null != r["return"] && (l = r["return"](), Object(l) !== l)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return i;
        }
    }
    function K(e) {
        if (Array.isArray(e)) return e;
    }
    function W(e, n) {
        if (null == e) return {};
        var r, t, o = V(e, n);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (t = 0; t < a.length; t++) {
                r = a[t], -1 === n.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
            }
        }
        return o;
    }
    function V(e, n) {
        if (null == e) return {};
        var r = {};
        for (var t in e) {
            if ({}.hasOwnProperty.call(e, t)) {
                if (-1 !== n.indexOf(t)) continue;
                r[t] = e[t];
            }
        }
        return r;
    }
    /** @public */
    /** @public */    var U = [ "focusToggle", "mouseEnterPopover", "mouseEnterToggle", "mouseEnterHitArea" ];
    var $ = [ "blurToggle", "clickAway", "escapeKey", "mouseLeaveHitArea", "mouseLeavePopover", "mouseLeaveToggle", "mouseStopHitArea", "offScreen", "tabKey", "toggleClick" ];
    /** @public */
    /** @public */    var z = {
        /** @private */
        appearance: i().string,
        /** @private */
        append: i().bool,
        children: i().node,
        closeDelay: i().number,
        /** @private */
        closeWhen: i().oneOf([ "default", "notOnClick" ]),
        content: i().node,
        contentRelationship: i().oneOf([ "label", "description" ]),
        defaultPlacement: i().oneOf([ "above", "below", "left", "right" ]),
        elementRef: i().oneOfType([ i().func, i().object ]),
        inline: i().bool,
        onRequestClose: i().func,
        onRequestOpen: i().func,
        open: i().bool,
        openDelay: i().oneOfType([ i().oneOf([ "primary", "secondary" ]), i().number ]),
        pointTo: i().shape({
            x: i().number,
            y: i().number
        }),
        /** @private */
        prepend: i().bool,
        renderAnchor: i().func
    };
    var G = {
        tension: 400
    };
    var J = [ "offScreen", "escapeKey" ];
    var Q =  t().createElement(s(), null);
    /**
 * The Tooltip component wraps arbitrary content to be displayed when the target element is hovered
 * or focused.
 */    function X(e) {
        var n = e.appearance, o = e.append, l = e.children, i = e.closeDelay, u = i === void 0 ? 300 : i, s = e.closeWhen, c = s === void 0 ? "default" : s, p = e.content, d = e.contentRelationship, g = e.defaultPlacement, O = g === void 0 ? "above" : g, h = e.elementRef, S = e.inline, C = S === void 0 ? true : S, L = e.onRequestClose, I = e.onRequestOpen, _ = e.open, D = e.openDelay, M = D === void 0 ? "primary" : D, x = e.renderAnchor, B = e.pointTo, H = e.prepend, F = W(e, [ "appearance", "append", "children", "closeDelay", "closeWhen", "content", "contentRelationship", "defaultPlacement", "elementRef", "inline", "onRequestClose", "onRequestOpen", "open", "openDelay", "renderAnchor", "pointTo", "prepend" ]);
        // @docs-props-type TooltipPropsBase
                var K = (0, r.useState)(false), V = N(K, 2), U = V[0], $ = V[1];
        var z = (0, r.useState)(null), X = N(z, 2), Y = X[0], Z = X[1];
        var ee = (0, r.useState)(null), ne = N(ee, 2), re = ne[0], te = ne[1];
        var oe = (0, r.useState)(null), ae = N(oe, 2), le = ae[0], ie = ae[1];
        var ue = (0, r.useState)(null), se = N(ue, 2), ce = se[0], fe = se[1];
        var pe = (0, r.useState)((function() {
            return {
                ariaId: (0, y.createDOMID)("aria-id"),
                popoverId: (0, y.createDOMID)("popover")
            };
        })), ve = N(pe, 1), de = ve[0], me = de.ariaId, be = de.popoverId;
        var ye = m()({
            componentName: "Tooltip",
            /* eslint-disable-next-line prefer-rest-params */
            componentProps: arguments[0],
            // see SUI-7028
            defaultValuePropName: "defaultOpen",
            valuePropName: "open"
        });
        var ge = (0, r.useMemo)((function() {
            var e = !!p;
            return !!Y && ye ? _ : e && U;
        }), [ Y, p, ye, U, _ ]);
        var Oe = (0, r.useRef)();
        (0, r.useEffect)((function() {
            return function() {
                if (Oe.current !== undefined) {
                    clearTimeout(Oe.current);
                }
            };
        }), []);
        var he = (0, r.useCallback)((function(e, n) {
            if (Oe.current !== undefined) {
                clearTimeout(Oe.current);
            }
            Oe.current = setTimeout((function() {
                if (!ye) {
                    $(false);
                }
                L === null || L === void 0 ? void 0 : L(e, n);
            }), u);
        }), [ u, ye, L ]);
        var Se = (0, r.useCallback)((function(e, n) {
            var r = M === "primary" ? 300 : 750;
            var t = typeof M === "number" ? M : r;
            if (Oe.current !== undefined) {
                clearTimeout(Oe.current);
            }
            Oe.current = setTimeout((function() {
                var r = ge;
                if (!ye) {
                    $(true);
                }
                if (!r) {
                    I === null || I === void 0 ? void 0 : I(e, n);
                }
            }), t);
        }), [ ye, ge, I, M ]);
        var Ce = (0, r.useCallback)((function(e) {
            Se(e, {
                reason: "mouseEnterToggle"
            });
        }), [ Se ]);
        var ke = (0, r.useCallback)((function(e) {
            Se(e, {
                reason: "mouseEnterPopover"
            });
        }), [ Se ]);
        var Ee = (0, r.useCallback)((function(e) {
            Se(e, {
                reason: "mouseEnterHitArea"
            });
        }), [ Se ]);
        var Te = (0, r.useCallback)((function(e) {
            var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "mouseLeaveToggle";
            if (e.relatedTarget == null || e.relatedTarget !== le && e.relatedTarget !== ce) {
                he(e, {
                    reason: n
                });
            }
        }), [ he, le, ce ]);
        var Pe = (0, r.useCallback)((function(e) {
            Te(e, "mouseLeavePopover");
        }), [ Te ]);
        var je = (0, r.useCallback)((function(e) {
            Te(e, "mouseLeaveHitArea");
        }), [ Te ]);
        var we = (0, r.useCallback)((function(e) {
            he(e, {
                reason: "mouseStopHitArea"
            });
        }), [ he ]);
        var Re = a()(we, 10);
        var Le = (0, r.useCallback)((function(e) {
            // Handle mouse events with native events due to current issues with how they
            // are handled by react. See https://github.com/facebook/react/issues/4251,
            // https://github.com/facebook/react/issues/19419, and SUI-1232.
            if (e) {
                e.addEventListener("mouseenter", Ce);
                e.addEventListener("mouseleave", Te);
            } else if (Y) {
                Y.removeEventListener("mouseenter", Ce);
                Y.removeEventListener("mouseleave", Te);
            }
            Z(e);
        }), [ Y, Ce, Te ]);
        var qe = (0, r.useCallback)((function(e) {
            if (e) {
                e.addEventListener("mouseenter", ke);
                e.addEventListener("mouseleave", Pe);
            } else if (le) {
                le.removeEventListener("mouseenter", ke);
                le.removeEventListener("mouseleave", Pe);
            }
            ie(e);
        }), [ ke, Pe, le ]);
        var Ie = (0, r.useCallback)((function(e) {
            fe(e);
        }), []);
        var Ae = (0, r.useCallback)((function(e) {
            // Handle mouse events with native events due to current issues with how they
            // are handled by react. See https://github.com/facebook/react/issues/4251,
            // https://github.com/facebook/react/issues/19419, and SUI-1232.
            if (e) {
                e.addEventListener("mouseenter", Ee);
                e.addEventListener("mousemove", Re);
                e.addEventListener("mouseleave", je);
            } else if (re) {
                re.removeEventListener("mouseenter", Ee);
                re.removeEventListener("mousemove", Re);
                re.removeEventListener("mouseleave", je);
            }
            te(e);
        }), [ Ee, je, Re, re ]);
        var _e = (0, r.useCallback)((function(e) {
            if (c !== "notOnClick") {
                he(e, {
                    reason: "toggleClick"
                });
            }
        }), [ c, he ]);
        var De = (0, r.useCallback)((function(e) {
            Se(e, {
                reason: "focusToggle"
            });
        }), [ Se ]);
        var Me = (0, r.useCallback)((function(e) {
            he(e, {
                reason: "blurToggle"
            });
        }), [ he ]);
        var Ne = (0, r.useCallback)((function(e) {
            he(null, e);
        }), [ he ]);
        var xe = (0, r.useCallback)((function(e) {
            Le(e);
            R(h, e);
        }), [ h, Le ]);
        var Be = (0, r.useMemo)((function() {
            return {
                "data-test": "toggle",
                elementRef: xe,
                onBlur: Me,
                onClick: _e,
                onFocus: De
            };
        }), [ Me, _e, De, xe ]);
        var He = (0, r.useMemo)((function() {
            return d === "label" ? {
                "aria-labelledby": me,
                labelledBy: me
            } : {
                "aria-describedby": me,
                describedBy: me
            };
        }), [ d, me ]);
        var Fe = (0, r.useMemo)((function() {
            return {
                append: o,
                prepend: H
            };
        }), [ o, H ]);
        var Ke = (0, r.useMemo)((function() {
            return A(A(A({}, He), Be), Fe);
        }), [ He, Be, Fe ]);
        var We = (0, r.useMemo)((function() {
            return A(A({}, He), Fe);
        }), [ He, Fe ]);
        var Ve = (0, r.useCallback)((function() {
            if (false) {}
            return x === null || x === void 0 ? void 0 : x(A(A(A({}, Ke), He), Fe));
        }), [ He, Ke, l, Fe, x ]);
        var Ue = (0, r.useCallback)((function() {
            var e;
            if ( (0, r.isValidElement)(l)) {
                e =  (0, r.cloneElement)(l, We);
            } else if (l) {
                e = l;
            } else {
                e =  t().createElement(w, {
                    appearance: "subtle",
                    "aria-describedby": me,
                    "aria-label": (0, b._)("More info"),
                    icon: Q
                });
            }
            var n = C ? P : T;
            
            return t().createElement(n, {
                "data-test": "toggle",
                onClick: _e,
                onFocus: De,
                onBlur: Me,
                ref: Le
            }, e);
        }), [ l, C, _e, De, Me, Le, We, me ]);
        var $e = C ? E : k;
        var ze = !!p &&  t().createElement(v(), {
            "aria-hidden": "true",
            id: me
        }, p);
        var Ge =  t().createElement(f(), {
            role: "tooltip",
            align: "center",
            anchor: Y,
            animationConfig: G,
            appearance: n,
            closeReasons: J,
            defaultPlacement: O,
            id: be,
            open: ge,
            onRequestClose: Ne,
            elementRef: qe,
            hitAreaRef: Ae,
            outerRef: Ie,
            pointTo: B
        },  t().createElement(j, null, p));
        if (x) {
            
            return t().createElement(t().Fragment, null, Ve(), ze, Ge);
        }
        
        return t().createElement($e, q({
            "data-test": "tooltip",
            "data-test-open": ge,
            "data-test-popover-id": be,
            ref: h
        }, F), Ue(), ze, Ge);
    }
    X.propTypes = z;
    X.possibleOpenReasons = U;
    X.possibleCloseReasons = $;
    /* harmony default export */    const Y = X;
    // only for styled components using Tooltip
        // CONCATENATED MODULE: ./src/Tooltip/index.ts
    module.exports = n;
    /******/})();
