// @splunk/react-ui/TransitionOpen.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = n => {
            /******/ var t = n && n.__esModule ? 
            /******/ () => n["default"]
            /******/ : () => n
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (n, t) => {
            /******/ for (var r in t) {
                /******/ if (e.o(t, r) && !e.o(n, r)) {
                    /******/ Object.defineProperty(n, r, {
                        enumerable: true,
                        get: t[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, n) => Object.prototype.hasOwnProperty.call(e, n)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var n = {};
    // ESM COMPAT FLAG
        e.r(n);
    // EXPORTS
        e.d(n, {
        default: () => /* reexport */ $
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var r =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "@splunk/react-ui/Animation"
    const a = require("@splunk/react-ui/Animation");
    // CONCATENATED MODULE: external "@splunk/react-ui/usePrevious"
    const u = require("@splunk/react-ui/usePrevious");
    var l =  e.n(u);
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const s = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: external "@splunk/ui-utils/style"
    const c = require("@splunk/ui-utils/style");
    // CONCATENATED MODULE: external "@react-spring/web"
    const f = require("@react-spring/web");
    // CONCATENATED MODULE: external "styled-components"
    const d = require("styled-components");
    var m =  e.n(d);
    // CONCATENATED MODULE: external "@splunk/themes"
    const p = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/TransitionOpen/TransitionOpenStyles.ts
    var v = m()(f.animated.div).withConfig({
        displayName: "TransitionOpenStyles__Styled",
        componentId: "sc-1x58s0g-0"
    })([ "", ";", ";" ], p.mixins.reset("block"), (function(e) {
        var n = e.$hideOverflow;
        return n && (0, d.css)([ "&&{overflow:hidden;}" ]);
    }));
    var y = m()(f.animated.div).withConfig({
        displayName: "TransitionOpenStyles__StyledInner",
        componentId: "sc-1x58s0g-1"
    })([ "", ";", ";width:100%;" ], p.mixins.reset("table"), p.mixins.clearfix());
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function b(e, n) {
        if (e) {
            if (typeof e === "function") {
                e(n);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = n;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/TransitionOpen/TransitionOpen.tsx
    function h() {
        return h = Object.assign ? Object.assign.bind() : function(e) {
            for (var n = 1; n < arguments.length; n++) {
                var t = arguments[n];
                for (var r in t) {
                    ({}).hasOwnProperty.call(t, r) && (e[r] = t[r]);
                }
            }
            return e;
        }, h.apply(null, arguments);
    }
    function g(e) {
        "@babel/helpers - typeof";
        return g = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, g(e);
    }
    function O(e, n) {
        return k(e) || P(e, n) || j(e, n) || S();
    }
    function S() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function j(e, n) {
        if (e) {
            if ("string" == typeof e) return w(e, n);
            var t = {}.toString.call(e).slice(8, -1);
            return "Object" === t && e.constructor && (t = e.constructor.name), "Map" === t || "Set" === t ? Array.from(e) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? w(e, n) : void 0;
        }
    }
    function w(e, n) {
        (null == n || n > e.length) && (n = e.length);
        for (var t = 0, r = Array(n); t < n; t++) {
            r[t] = e[t];
        }
        return r;
    }
    function P(e, n) {
        var t = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != t) {
            var r, o, i, a, u = [], l = !0, s = !1;
            try {
                if (i = (t = t.call(e)).next, 0 === n) {
                    if (Object(t) !== t) return;
                    l = !1;
                } else for (;!(l = (r = i.call(t)).done) && (u.push(r.value), u.length !== n); l = !0) {
                }
            } catch (e) {
                s = !0, o = e;
            } finally {
                try {
                    if (!l && null != t["return"] && (a = t["return"](), Object(a) !== a)) return;
                } finally {
                    if (s) throw o;
                }
            }
            return u;
        }
    }
    function k(e) {
        if (Array.isArray(e)) return e;
    }
    function x(e, n) {
        if (null == e) return {};
        var t, r, o = C(e, n);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                t = i[r], -1 === n.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function C(e, n) {
        if (null == e) return {};
        var t = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== n.indexOf(r)) continue;
                t[r] = e[r];
            }
        }
        return t;
    }
    function F(e, n) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            n && (r = r.filter((function(n) {
                return Object.getOwnPropertyDescriptor(e, n).enumerable;
            }))), t.push.apply(t, r);
        }
        return t;
    }
    function N(e) {
        for (var n = 1; n < arguments.length; n++) {
            var t = null != arguments[n] ? arguments[n] : {};
            n % 2 ? F(Object(t), !0).forEach((function(n) {
                A(e, n, t[n]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : F(Object(t)).forEach((function(n) {
                Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(t, n));
            }));
        }
        return e;
    }
    function A(e, n, t) {
        return (n = T(n)) in e ? Object.defineProperty(e, n, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[n] = t, e;
    }
    function T(e) {
        var n = q(e, "string");
        return "symbol" == g(n) ? n : n + "";
    }
    function q(e, n) {
        if ("object" != g(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var r = t.call(e, n || "default");
            if ("object" != g(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === n ? String : Number)(e);
    }
    /** @public */    var E = [ "slideFromTop", "slideFromRight", "slideFromBottom", "slideFromLeft", "expandHeight", "expandWidth", "none" ];
    /** @public */
    /** @public */    var I = {
        animation: i().oneOfType([ i().oneOf(E), i().shape({
            open: i().oneOf(E).isRequired,
            close: i().oneOf([ "none" ]).isRequired
        }), i().shape({
            open: i().oneOf([ "none" ]).isRequired,
            close: i().oneOf(E).isRequired
        }) ]),
        animateOnMount: i().bool,
        children: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        className: i().string,
        id: i().string,
        innerClassName: i().string,
        innerStyle: i().object,
        onAnimationEnd: i().func,
        onAnimationStart: i().func,
        open: i().bool,
        outerClassName: i().string,
        outerId: i().string,
        outerStyle: i().object,
        renderChildrenWhenCollapsed: i().bool,
        retainFocus: i().bool,
        takeFocus: i().bool
    };
    var R = function e(n) {
        var t;
        switch (n) {
          case "slideFromTop":
          case "slideFromBottom":
          case "expandHeight":
          case "none":
            t = "height";
            break;

          case "slideFromLeft":
          case "slideFromRight":
          case "expandWidth":
            t = "width";
            break;

          default:
 // Intentionally left empty.
                }
        return t;
    };
    var M = function e(n) {
        var t = n.animation, r = n.open, o = n.value;
        var i = {
            value: r ? o : 0
        };
        var a = {
            value: r ? 0 : o
        };
        var u = {
            precision: 1
        };
 // Only use precision when not using transform, otherwise transform animation will break
        // Expansion doesn't require a valid value; the subsequent render animates it.
                if (t === "expandWidth" || t === "expandHeight" || t === "none") {
            return {
                to: i,
                config: u
            };
        }
        // Slide transitions require a valid value, render without transition first.
                if (!o) {
            return null;
        }
        if (t !== "slideFromTop" && t !== "slideFromLeft") {
            // Return the standard transition if not using a transform.
            return {
                to: i,
                from: a,
                config: u
            };
        }
        // The top and left directions require an additional transform of their inner content to match the outer dimension animation.
        // This is needed to achieve a clipping effect from the outer div, rather than the consumer providing their own.
                var l = t === "slideFromTop" ? "Y" : "X";
        var s = "translate".concat(l, "(").concat(r ? 0 : -o, "px)");
        var c = "translate".concat(l, "(").concat(r ? -o : 0, "px)");
        return {
            to: N({
                transform: s
            }, i),
            from: N({
                transform: c
            }, a)
        };
    };
    var _ = function e(n) {
        var t = n.animation, r = n.el, o = n.takeFocusProp;
        var i = 0;
        if (r) {
            if (o) {
                (0, s.takeFocus)(r, "container");
            }
            var a = R(t);
            switch (a) {
              case "height":
                i = r.offsetHeight;
                break;

              case "width":
                i = r.offsetWidth;
                break;

              default:
 // Intentionally left empty.
                        }
        }
        return i;
    };
    var W = {};
    var D = {};
    var H = function e(n) {
        if (typeof n === "string") {
            return n;
        }
        return n.open !== "none" ? n.open : n.close;
    };
    function L(e) {
        var n = e.animation, o = n === void 0 ? "expandHeight" : n, i = e.animateOnMount, u = e.children, f = e.className, d = e.elementRef, m = e.id, p = e.innerClassName, S = e.innerStyle, j = S === void 0 ? W : S, w = e.onAnimationEnd, P = e.onAnimationStart, k = e.open, C = e.outerClassName, F = e.outerId, T = e.outerStyle, q = T === void 0 ? D : T, E = e.renderChildrenWhenCollapsed, I = e.retainFocus, L = e.takeFocus, $ = x(e, [ "animation", "animateOnMount", "children", "className", "elementRef", "id", "innerClassName", "innerStyle", "onAnimationEnd", "onAnimationStart", "open", "outerClassName", "outerId", "outerStyle", "renderChildrenWhenCollapsed", "retainFocus", "takeFocus" ]);
        // @docs-props-type TransitionOpenPropsBase
                var B = l()(k);
        var K = (0, t.useRef)(true);
        var U = (0, t.useState)(i ? k : false), X = O(U, 2), Y = X[0], z = X[1];
        var G = (0, t.useState)(null), J = O(G, 2), Q = J[0], V = J[1];
        var Z = (0, t.useState)(0), ee = O(Z, 2), ne = ee[0], te = ee[1];
        var re = (0, t.useState)(k), oe = O(re, 2), ie = oe[0], ae = oe[1];
        var ue = (0, t.useMemo)((function() {
            return H(o);
        }), [ o ]);
        (0, t.useLayoutEffect)((function() {
            // prevOpen starts undefined, and we don't want to make the comparison in that situation
            if (B !== undefined && k !== B) {
                z(true);
                ae(k);
            }
        }), [ k, B ]);
        // equivalent to componentDidMount and componentWillUnmount
                (0, t.useEffect)((function() {
            // we need to set this to true on every mount because React.StrictMode will call the effect twice
            // and we need to ensure that allowAnimationUpdates is true while the component is mounted
            K.current = true;
            return function() {
                K.current = false;
            };
        }), []);
        var le = (0, t.useCallback)((function(e) {
            V(e);
            te(_({
                animation: ue,
                el: e,
                takeFocusProp: L
            }));
            b(d, e);
        }), [ ue, L, d ]);
        var se = (0, t.useCallback)((function() {
            P === null || P === void 0 ? void 0 : P();
        }), [ P ]);
        var ce = (0, t.useCallback)((function() {
            if (K.current) {
                z(false);
            }
            w === null || w === void 0 ? void 0 : w();
        }), [ w ]);
        var fe = (0, t.useCallback)((function(e) {
            if (Q) {
                (0, s.handleTab)(Q, e);
            }
        }), [ Q ]);
        var de = g(o) === "object" && o.open === "none";
        var me = g(o) === "object" && o.close === "none";
        var pe = (0, a.useAnimation)(N(N({}, M({
            animation: ue,
            open: ie,
            value: ne
        })), {}, {
            immediate: ue === "none" || ie && de || !ie && me,
            onRest: ce,
            onStart: se
        })), ve = pe.value, ye = pe.transform;
        var be;
        if (Y || E) {
            be = R(ue);
        }
        // if renderChildrenWhenCollapsed is true, we hide the content by setting the height of the wrapper to 0
                var he = E && !ie && !Y ? 0 : ve;
        var ge = (0, t.useMemo)((function() {
            return N(N({}, q), be && A({}, be, he));
        }), [ be, he, q ]);
        var Oe = (0, t.useMemo)((function() {
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            return N(N({}, j), {}, {
                transform: ye
            });
        }), [ j, ye ]);
        
        return r().createElement(v, {
            $hideOverflow: !ie || Y || false,
            className: (0, c.toClassName)(f, C),
            style: ge,
            id: F
        }, (ie || Y || E) &&  r().createElement(y, h({
            "data-test": "transition-open"
        }, $, {
            ref: le,
            tabIndex: -1,
            className: p,
            id: m,
            onKeyDown: I ? fe : undefined,
            style: Oe
        }), u));
    }
    L.propTypes = I;
    /* harmony default export */    const $ = L;
    // CONCATENATED MODULE: ./src/TransitionOpen/index.ts
    module.exports = n;
    /******/})();
