// @splunk/react-ui/Tree.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var n = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(n, {
                a: n
            });
            /******/            return n;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, n) => {
            /******/ for (var r in n) {
                /******/ if (e.o(n, r) && !e.o(t, r)) {
                    /******/ Object.defineProperty(t, r, {
                        enumerable: true,
                        get: n[r]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        Item: () => /* reexport */ _,
        default: () => /* reexport */ ue
    });
    // CONCATENATED MODULE: external "react"
    const n = require("react");
    var r =  e.n(n);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var i =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const a = require("styled-components");
    var l =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/themes"
    const u = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/Tree/ItemStyles.ts
    var c = l().span.withConfig({
        displayName: "ItemStyles__StyledTreeItemContentWrapper",
        componentId: "sc-1ig9nmy-0"
    })([ "display:inline-block;min-height:100%;" ]);
    var f = l().span.withConfig({
        displayName: "ItemStyles__StyledTreeItemIndent",
        componentId: "sc-1ig9nmy-1"
    })([ "white-space:pre;" ]);
    var s = l().span.withConfig({
        displayName: "ItemStyles__StyledTreeItemContent",
        componentId: "sc-1ig9nmy-2"
    })([ "display:", ";min-height:100%;&:focus{outline:3px solid ", ";outline-offset:1px;&:active{outline:none;}}&:hover{background-color:", ";cursor:pointer;}" ], (function(e) {
        var t = e.$hasCustomIndent;
        return t ? "inline" : "inline-block";
    }), u.variables.focusColor, u.variables.actionColorBackgroundSecondaryHover);
    // CONCATENATED MODULE: ./src/Tree/TreeContext.tsx
    var d =  (0, n.createContext)({
        defaultIndent: true
    });
    /* harmony default export */    const v = d;
    // CONCATENATED MODULE: ./src/Tree/TreeStyles.ts
    var p = l().ul.withConfig({
        displayName: "TreeStyles__StyledUnorderedList",
        componentId: "sc-13pokwt-0"
    })([ "", ";list-style:none;white-space:inherit;", ";" ], u.mixins.reset("ul"), (function(e) {
        var t = e.$defaultIndent;
        return t && (0, a.css)([ "padding-inline-start:", ";" ], u.variables.spacingXXXLarge);
    }));
    // CONCATENATED MODULE: ./src/utils/updateReactRef.ts
    /**
 * Updates a React ref. Callback refs and object refs (from `createRef` and `useRef`) are supported.
 *
 * @param ref - The React callback or object ref. Can be `null` or `undefined`.
 * @param current - The new value of the ref.
 */
    function y(e, t) {
        if (e) {
            if (typeof e === "function") {
                e(t);
            } else {
                // the public signature of this util uses React.Ref<T> to mirror the way React types refs.
                // the intention here is to signal "we will take care of setting 'current', not you".
                e.current = t;
 // eslint-disable-line no-param-reassign
                        }
        }
    }
    // CONCATENATED MODULE: ./src/Tree/Item.tsx
    function m(e) {
        "@babel/helpers - typeof";
        return m = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, m(e);
    }
    function b() {
        return b = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, b.apply(null, arguments);
    }
    function g(e, t) {
        var n = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var r = Object.getOwnPropertySymbols(e);
            t && (r = r.filter((function(t) {
                return Object.getOwnPropertyDescriptor(e, t).enumerable;
            }))), n.push.apply(n, r);
        }
        return n;
    }
    function h(e) {
        for (var t = 1; t < arguments.length; t++) {
            var n = null != arguments[t] ? arguments[t] : {};
            t % 2 ? g(Object(n), !0).forEach((function(t) {
                I(e, t, n[t]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(n)) : g(Object(n)).forEach((function(t) {
                Object.defineProperty(e, t, Object.getOwnPropertyDescriptor(n, t));
            }));
        }
        return e;
    }
    function I(e, t, n) {
        return (t = O(t)) in e ? Object.defineProperty(e, t, {
            value: n,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[t] = n, e;
    }
    function O(e) {
        var t = w(e, "string");
        return "symbol" == m(t) ? t : t + "";
    }
    function w(e, t) {
        if ("object" != m(e) || !e) return e;
        var n = e[Symbol.toPrimitive];
        if (void 0 !== n) {
            var r = n.call(e, t || "default");
            if ("object" != m(r)) return r;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === t ? String : Number)(e);
    }
    function S(e, t) {
        return A(e) || P(e, t) || k(e, t) || j();
    }
    function j() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function k(e, t) {
        if (e) {
            if ("string" == typeof e) return C(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? C(e, t) : void 0;
        }
    }
    function C(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function P(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = i.call(n)).done) && (l.push(r.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (a = n["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function A(e) {
        if (Array.isArray(e)) return e;
    }
    function E(e, t) {
        if (null == e) return {};
        var n, r, o = T(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function T(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var x = {
        id: i().string,
        children: i().node,
        content: i().node,
        customIndent: i().node,
        elementRef: i().oneOfType([ i().func, i().object ]),
        endAdornment: i().node,
        expanded: i().bool,
        onFocus: i().func,
        onKeyDown: i().func,
        onToggleSelection: i().func,
        onToggleExpansion: i().func
    };
    /** @public */
    /** @public */    function D(e) {
        var t = e.id, o = e.children, i = e.content, a = e.customIndent, l = e.elementRef, u = e.endAdornment, d = e.expanded, m = e.onFocus, g = e.onKeyDown, I = e.onToggleExpansion, O = e.onToggleSelection, w = E(e, [ "id", "children", "content", "customIndent", "elementRef", "endAdornment", "expanded", "onFocus", "onKeyDown", "onToggleExpansion", "onToggleSelection" ]);
        // @docs-props-type TreeItemPropsBase
                var j = (0, n.useContext)(v), k = j.defaultIndent, C = k === void 0 ? true : k, P = j.onItemKeyDown, A = j.onItemClick, T = j.removeNode, x = j.setNode, D = j.itemPaths, _ = j.focusedItemId;
 // make first tree item tabbable
                var $ = (0, n.useState)((D === null || D === void 0 ? void 0 : D.get("".concat(t))) === "0" ? 0 : -1), M = S($, 2), N = M[0], R = M[1];
        var K = (0, n.useRef)(null);
        var F = (0, n.useRef)(null);
        var U = n.Children.toArray(o).filter(n.isValidElement);
        U.forEach((function(e, n) {
            if (e.props.id) {
                if (d && x) {
                    x("".concat(e.props.id), "".concat(D === null || D === void 0 ? void 0 : D.get("".concat(t)), "-").concat(n));
                } else if (T) {
                    T(e.props.id);
                }
            }
        }));
        var q = (0, n.useCallback)((function(e) {
            y(K, e);
            y(l, e);
        }), [ l ]);
        var L = (0, n.useCallback)((function(e) {
            var t;
            var n = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
            // For consistency, if not provided but found in the event then add it to `data` on callback.
            // This allows for common handling of both click and keyboard events in the consumer.
                        var r = h({}, n);
            if (!n.treeItemId && ((t = e.currentTarget) === null || t === void 0 ? void 0 : t.id)) {
                r.treeItemId = e.currentTarget.id;
            }
            I === null || I === void 0 ? void 0 : I(e, r);
        }), [ I ]);
        var X = (0, n.useCallback)((function(e) {
            A === null || A === void 0 ? void 0 : A(e, t);
        }), [ t, A ]);
        var B = (0, n.useCallback)((function(e) {
            var t;
            e.stopPropagation();
            F === null || F === void 0 ? void 0 : (t = F.current) === null || t === void 0 ? void 0 : t.focus();
            m === null || m === void 0 ? void 0 : m(e);
        }), [ m ]);
        var H = (0, n.useCallback)((function() {
            if ((D === null || D === void 0 ? void 0 : D.get("".concat(t))) === "0") {
                R(0);
            }
        }), [ t, D ]);
        (0, n.useEffect)((function() {
            if (_ === t) {
                var e;
                K === null || K === void 0 ? void 0 : (e = K.current) === null || e === void 0 ? void 0 : e.focus();
            }
        }), [ _, t, K ]);
        var V = (0, n.useCallback)((function(e) {
            e.stopPropagation();
        }), []);
        var W = (0, n.useCallback)((function(e) {
            e.stopPropagation();
            if (e.key === " ") {
                e.preventDefault();
                O === null || O === void 0 ? void 0 : O(e, {
                    treeItemId: t
                });
                return;
            }
            // Used to ensure first item in Tree is tab navigable, i.e. in JSONTree when focus is manually set to the root item,
            // this allows user to tab backwards to previous item. Tab index is reset to 0 onBlur.
                        if (e.shiftKey && e.key === "Tab" && (D === null || D === void 0 ? void 0 : D.get("".concat(t))) === "0") {
                R(-1);
            }
            P === null || P === void 0 ? void 0 : P(e, t, d, U, L);
            g === null || g === void 0 ? void 0 : g(e);
        }), [ U, d, L, t, D, P, g, O ]);
        
        return r().createElement("li", b({
            "data-test": "tree-item"
        }, w, {
            id: t,
            ref: q,
            tabIndex: N,
            onBlur: H,
            onFocus: B,
            onKeyDown: W,
            role: "treeitem",
            "aria-expanded": U.length ? d || "false" : undefined
        }),  r().createElement(c, null,  r().createElement(f, null, a),  r().createElement(s, {
            $hasCustomIndent: !!a,
            "data-test": "tree-item-content",
            ref: F,
            tabIndex: -1,
            onFocus: V,
            onClick: X
        }, i, !d && u)), d &&  r().createElement(r().Fragment, null,  r().createElement(p, {
            role: "group",
            $defaultIndent: C
        }, o), u));
    }
    D.propTypes = x;
    /* harmony default export */    const _ = D;
    // CONCATENATED MODULE: ./src/Tree/treeUtils.ts
    function $(e, t) {
        return F(e) || K(e, t) || N(e, t) || M();
    }
    function M() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function N(e, t) {
        if (e) {
            if ("string" == typeof e) return R(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? R(e, t) : void 0;
        }
    }
    function R(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function K(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = i.call(n)).done) && (l.push(r.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (a = n["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function F(e) {
        if (Array.isArray(e)) return e;
    }
    /**
 * A helper function to get first key in Map that matches given value
 */    function U(e, t) {
        var n;
        var r = Array.from(e.entries());
        return (n = r.find((function(e) {
            var n = $(e, 2), r = n[1];
            return r === t;
        }))) === null || n === void 0 ? void 0 : n[0];
    }
    /**
 * A helper function to get parent path of current item path
 */    function q(e) {
        // remove last index number of the current item path string to get parent
        return e.split("-").slice(0, -1).join("-");
    }
    /**
 * A helper function to get next path of current item path in same level
 */    function L(e, t) {
        return e === null || e === void 0 ? void 0 : e.replace(/([^-]*)$/, "".concat(t + 1));
    }
    /**
 * A helper function to get the last index number of an item path string
 */    function X(e) {
        var t = e.split("-").pop();
        if (!t) {
            return 0;
        }
        return parseInt(t, 10) || 0;
    }
    function B(e, t) {
        var n = e.get(t);
        if (!n) {
            return t;
        }
        var r = q(n);
        return U(e, r) || t;
    }
    /**
 * A helper function for getNextTreeItem
 */    function H(e, t) {
        var n = q(t);
        if (!n) {
            return U(e, t);
        }
        // get next node in the parent level
                var r = X(n);
        var o = L(n, r);
        var i = U(e, o);
        if (i) {
            return i;
        }
        // continue getting parents until find next node or reach root
                return H(e, o);
    }
    function V(e, t) {
        var n = e.get(t);
        if (!n) {
            return t;
        }
        // case where next tree item is the first visible child
                var r = U(e, "".concat(n, "-0"));
        if (r) {
            return r;
        }
        // case where next tree item is the next node in the same level
                var o = X(n);
        var i = L(n, o);
        var a = U(e, i);
        if (a) {
            return a;
        }
        // case where next tree item is a higher level or does not exist
                return H(e, n) || t;
    }
    /**
 * A helper function for getPrevTreeItem
 */    function W(e, t) {
        var n = U(e, "".concat(t, "-0"));
        if (!n) {
            return U(e, t);
        }
        var r = U(e, t);
        var o = "".concat(t, "-0");
        // get the last child in current node, and continue to find its children recursively
                while (r) {
            var i = X(o);
            var a = L(o, i);
            r = U(e, a);
            if (r) {
                o = a;
            }
        }
        return W(e, o);
    }
    function z(e, t) {
        var n = e.get(t);
        if (!n) {
            return t;
        }
        var r = X(n);
        // case where current is first child of parent or root item
                if (r === 0) {
            return B(e, t);
        }
        // case where previous tree item is or is a child of previous node in same level
                var o = n.replace(/[^-]+$/, "".concat(r - 1));
        return W(e, o) || t;
    }
    function G(e) {
        return U(e, "0");
    }
    function J(e) {
        var t = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "0";
        var n = U(e, t);
        var r = t;
        // find last child in current node
                while (n) {
            var o = X(r);
            n = U(e, L(r, o));
            if (n) {
                r = L(r, o);
            }
        }
        // check if last child has children
                var i = U(e, "".concat(r, "-0"));
        if (!i) {
            return U(e, r);
        }
        // continue to find its last child recursively
                r = "".concat(r, "-0");
        return J(e, r);
    }
    // CONCATENATED MODULE: ./src/Tree/Tree.tsx
    function Q() {
        return Q = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var n = arguments[t];
                for (var r in n) {
                    ({}).hasOwnProperty.call(n, r) && (e[r] = n[r]);
                }
            }
            return e;
        }, Q.apply(null, arguments);
    }
    function Y(e, t) {
        return re(e) || ne(e, t) || ee(e, t) || Z();
    }
    function Z() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function ee(e, t) {
        if (e) {
            if ("string" == typeof e) return te(e, t);
            var n = {}.toString.call(e).slice(8, -1);
            return "Object" === n && e.constructor && (n = e.constructor.name), "Map" === n || "Set" === n ? Array.from(e) : "Arguments" === n || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n) ? te(e, t) : void 0;
        }
    }
    function te(e, t) {
        (null == t || t > e.length) && (t = e.length);
        for (var n = 0, r = Array(t); n < t; n++) {
            r[n] = e[n];
        }
        return r;
    }
    function ne(e, t) {
        var n = null == e ? null : "undefined" != typeof Symbol && e[Symbol.iterator] || e["@@iterator"];
        if (null != n) {
            var r, o, i, a, l = [], u = !0, c = !1;
            try {
                if (i = (n = n.call(e)).next, 0 === t) {
                    if (Object(n) !== n) return;
                    u = !1;
                } else for (;!(u = (r = i.call(n)).done) && (l.push(r.value), l.length !== t); u = !0) {
                }
            } catch (e) {
                c = !0, o = e;
            } finally {
                try {
                    if (!u && null != n["return"] && (a = n["return"](), Object(a) !== a)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return l;
        }
    }
    function re(e) {
        if (Array.isArray(e)) return e;
    }
    function oe(e, t) {
        if (null == e) return {};
        var n, r, o = ie(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (r = 0; r < i.length; r++) {
                n = i[r], -1 === t.indexOf(n) && {}.propertyIsEnumerable.call(e, n) && (o[n] = e[n]);
            }
        }
        return o;
    }
    function ie(e, t) {
        if (null == e) return {};
        var n = {};
        for (var r in e) {
            if ({}.hasOwnProperty.call(e, r)) {
                if (-1 !== t.indexOf(r)) continue;
                n[r] = e[r];
            }
        }
        return n;
    }
    var ae = {
        children: i().node,
        defaultIndent: i().bool,
        elementRef: i().oneOfType([ i().func, i().object ])
    };
    /**
 * Used to present a hierarchical list.
 */    function le(e) {
        var t = e.children, o = e.defaultIndent, i = o === void 0 ? true : o, a = e.elementRef, l = oe(e, [ "children", "defaultIndent", "elementRef" ]);
        // @docs-props-type TreePropsBase
                var u = (0, n.useState)(), c = Y(u, 2), f = c[0], s = c[1];
        var d = (0, n.useMemo)((function() {
            return new Map;
        }), []);
        var y = (0, n.useCallback)((function(e, t, n, r, o) {
            e.stopPropagation();
            var i = n && (r === null || r === void 0 ? void 0 : r.length);
            switch (e.key) {
              case "ArrowUp":
                e.preventDefault();
                s(z(d, t));
                break;

              case "ArrowDown":
                e.preventDefault();
                s(V(d, t));
                break;

              case "ArrowLeft":
                e.preventDefault();
                if (i) {
                    o(e, {
                        treeItemId: t
                    });
                } else {
                    s(B(d, t));
                }
                break;

              case "ArrowRight":
                e.preventDefault();
                if (i) {
                    s(V(d, t));
                } else {
                    o(e, {
                        treeItemId: t
                    });
                }
                break;

              case "Home":
                e.preventDefault();
                s(G(d));
                break;

              case "End":
                e.preventDefault();
                s(J(d));
                break;

              case "Enter":
                e.preventDefault();
                o(e, {
                    treeItemId: t
                });
                break;

              default:
 // do nothing
                        }
        }), [ d ]);
        var m = function e(t, n) {
            t.preventDefault();
            s(n);
        };
        var b = (0, n.useCallback)((function(e) {
            d["delete"](e);
        }), [ d ]);
        var g = (0, n.useCallback)((function(e, t) {
            d.set(e, t);
        }), [ d ]);
        var h = n.Children.toArray(t).filter(n.isValidElement);
        h.forEach((function(e, t) {
            if (e.props.id && g) {
                g("".concat(e.props.id), "".concat(t));
            }
        }));
        var I = (0, n.useMemo)((function() {
            return {
                defaultIndent: i,
                onItemKeyDown: y,
                onItemClick: m,
                removeNode: b,
                setNode: g,
                itemPaths: d,
                focusedItemId: f
            };
        }), [ i, f, d, y, b, g ]);
        
        return r().createElement(v.Provider, {
            value: I
        },  r().createElement(p, Q({
            "data-test": "tree",
            ref: a,
            role: "tree",
            $defaultIndent: i
        }, l), t));
    }
    le.propTypes = ae;
    le.Item = _;
    /* harmony default export */    const ue = le;
    // CONCATENATED MODULE: ./src/Tree/index.ts
    module.exports = t;
    /******/})();
