// @splunk/react-ui/Typography.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = r => {
            /******/ var t = r && r.__esModule ? 
            /******/ () => r["default"]
            /******/ : () => r
            /******/;
            e.d(t, {
                a: t
            });
            /******/            return t;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        Typography: () => /* reexport */ g,
        default: () => /* reexport */ h
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    var n =  e.n(t);
    // CONCATENATED MODULE: external "prop-types"
    const o = require("prop-types");
    var a =  e.n(o);
    // CONCATENATED MODULE: external "styled-components"
    const i = require("styled-components");
    var l =  e.n(i);
    // CONCATENATED MODULE: external "@splunk/themes/mixins"
    const s = require("@splunk/themes/mixins");
    var u =  e.n(s);
    // CONCATENATED MODULE: ./src/Typography/Typography.tsx
    function f() {
        return f = Object.assign ? Object.assign.bind() : function(e) {
            for (var r = 1; r < arguments.length; r++) {
                var t = arguments[r];
                for (var n in t) {
                    ({}).hasOwnProperty.call(t, n) && (e[n] = t[n]);
                }
            }
            return e;
        }, f.apply(null, arguments);
    }
    function p(e, r) {
        if (null == e) return {};
        var t, n, o = y(e, r);
        if (Object.getOwnPropertySymbols) {
            var a = Object.getOwnPropertySymbols(e);
            for (n = 0; n < a.length; n++) {
                t = a[n], -1 === r.indexOf(t) && {}.propertyIsEnumerable.call(e, t) && (o[t] = e[t]);
            }
        }
        return o;
    }
    function y(e, r) {
        if (null == e) return {};
        var t = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== r.indexOf(n)) continue;
                t[n] = e[n];
            }
        }
        return t;
    }
    var c = {
        as: a().string.isRequired,
        children: a().node.isRequired,
        color: a().oneOf([ "active", "default", "disabled", "inverted", "muted", "inherit" ]),
        elementRef: a().oneOfType([ a().func, a().object ]),
        family: a().oneOf([ "sansSerif", "monospace", "title" ]),
        lineHeight: a().oneOf([ "single", "tight", "snug", "comfortable", "relaxed", "spacious", "normal" ]),
        size: a().oneOf([ 24, 20, 18, 16, 14, 12 ]),
        variant: a().oneOf(s.typographyVariants),
        weight: a().oneOf([ "light", "normal", "semiBold", "bold", "extraBold", "heavy" ]),
        withReset: a().bool
    };
    var d = l().p.withConfig({
        displayName: "Typography__StyledTypography",
        componentId: "cgxs34-0"
    })([ "", "" ], (function(e) {
        var r = e.$variant, t = e.$typographyParams;
        return r ? u().typography(r, t) : u().typography(t);
    }));
    /**
 * Typography renders text content with styling based on the Splunk Design System.
 */    function g(e) {
        var r = e.as, o = e.children, a = e.color, i = e.elementRef, l = e.family, s = e.lineHeight, u = e.size, y = e.variant, c = e.weight, g = e.withReset, h = g === void 0 ? true : g, v = p(e, [ "as", "children", "color", "elementRef", "family", "lineHeight", "size", "variant", "weight", "withReset" ]);
        // @docs-props-type TypographyPropsBase
                var m = (0, t.useMemo)((function() {
            return {
                color: a,
                family: l,
                lineHeight: s,
                size: u,
                weight: c,
                withReset: h
            };
        }), [ a, l, s, u, c, h ]);
        
        return n().createElement(d, f({
            as: r,
            "data-test": "typography",
            ref: i,
            $typographyParams: m,
            $variant: y
        }, v), o);
    }
    g.propTypes = c;
    /* harmony default export */    const h = g;
    // CONCATENATED MODULE: ./src/Typography/index.ts
    module.exports = r;
    /******/})();
