// @splunk/react-ui/useForceUpdate.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var r = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ r.d = (e, t) => {
            /******/ for (var n in t) {
                /******/ if (r.o(t, n) && !r.o(e, n)) {
                    /******/ Object.defineProperty(e, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ r.o = (r, e) => Object.prototype.hasOwnProperty.call(r, e)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ r.r = r => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(r, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(r, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var e = {};
    // ESM COMPAT FLAG
        r.r(e);
    // EXPORTS
        r.d(e, {
        default: () => /* reexport */ f
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    // CONCATENATED MODULE: ./src/useForceUpdate/useForceUpdate.tsx
    function n(r, e) {
        return i(r) || a(r, e) || u(r, e) || o();
    }
    function o() {
        throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
    }
    function u(r, e) {
        if (r) {
            if ("string" == typeof r) return l(r, e);
            var t = {}.toString.call(r).slice(8, -1);
            return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? l(r, e) : void 0;
        }
    }
    function l(r, e) {
        (null == e || e > r.length) && (e = r.length);
        for (var t = 0, n = Array(e); t < e; t++) {
            n[t] = r[t];
        }
        return n;
    }
    function a(r, e) {
        var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"];
        if (null != t) {
            var n, o, u, l, a = [], i = !0, c = !1;
            try {
                if (u = (t = t.call(r)).next, 0 === e) {
                    if (Object(t) !== t) return;
                    i = !1;
                } else for (;!(i = (n = u.call(t)).done) && (a.push(n.value), a.length !== e); i = !0) {
                }
            } catch (r) {
                c = !0, o = r;
            } finally {
                try {
                    if (!i && null != t["return"] && (l = t["return"](), Object(l) !== l)) return;
                } finally {
                    if (c) throw o;
                }
            }
            return a;
        }
    }
    function i(r) {
        if (Array.isArray(r)) return r;
    }
    /**
 * This is a private component not intended for use outside @splunk/react-ui
 *
 * Triggers a rerender of a functional component. Useful as an escape hatch when a render is needed regardless of state.
 * See https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate
 *
 */    function c() {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        var r = (0, t.useReducer)((function(r) {
            return r + 1;
        }), 0), e = n(r, 2), o = e[0], u = e[1];
        return u;
    }
    /* harmony default export */    const f = c;
    // CONCATENATED MODULE: ./src/useForceUpdate/index.ts
    module.exports = e;
    /******/})();
