// @splunk/react-ui/useRovingFocus.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (r, t) => {
            /******/ for (var n in t) {
                /******/ if (e.o(t, n) && !e.o(r, n)) {
                    /******/ Object.defineProperty(r, n, {
                        enumerable: true,
                        get: t[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, r) => Object.prototype.hasOwnProperty.call(e, r)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var r = {};
    // ESM COMPAT FLAG
        e.r(r);
    // EXPORTS
        e.d(r, {
        default: () => /* reexport */ s
    });
    // CONCATENATED MODULE: external "react"
    const t = require("react");
    // CONCATENATED MODULE: external "@splunk/ui-utils/focus"
    const n = require("@splunk/ui-utils/focus");
    // CONCATENATED MODULE: ./src/useRovingFocus/useRovingFocus.tsx
    function o(e) {
        "@babel/helpers - typeof";
        return o = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function(e) {
            return typeof e;
        } : function(e) {
            return e && "function" == typeof Symbol && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
        }, o(e);
    }
    function u(e, r) {
        var t = Object.keys(e);
        if (Object.getOwnPropertySymbols) {
            var n = Object.getOwnPropertySymbols(e);
            r && (n = n.filter((function(r) {
                return Object.getOwnPropertyDescriptor(e, r).enumerable;
            }))), t.push.apply(t, n);
        }
        return t;
    }
    function i(e) {
        for (var r = 1; r < arguments.length; r++) {
            var t = null != arguments[r] ? arguments[r] : {};
            r % 2 ? u(Object(t), !0).forEach((function(r) {
                a(e, r, t[r]);
            })) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : u(Object(t)).forEach((function(r) {
                Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r));
            }));
        }
        return e;
    }
    function a(e, r, t) {
        return (r = c(r)) in e ? Object.defineProperty(e, r, {
            value: t,
            enumerable: !0,
            configurable: !0,
            writable: !0
        }) : e[r] = t, e;
    }
    function c(e) {
        var r = f(e, "string");
        return "symbol" == o(r) ? r : r + "";
    }
    function f(e, r) {
        if ("object" != o(e) || !e) return e;
        var t = e[Symbol.toPrimitive];
        if (void 0 !== t) {
            var n = t.call(e, r || "default");
            if ("object" != o(n)) return n;
            throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return ("string" === r ? String : Number)(e);
    }
    var l = {
        enableLoop: false,
        orientation: "horizontal",
        enableTab: false,
        enableHomeEnd: true
    };
    /**
 * This is a private hook not intended for use outside @splunk/react-ui
 *
 * Handles "roving focus" for a group of elements inside a containing element.
 * Elements inside the element that "rovingFocusProps" are set on will act a single tab stop
 * with arrow keys moving focus between them.
 * See https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#managing_focus_inside_groups
 *
 * ```
 * import React from 'react';
 * import useRovingFocus from '@splunk/react-ui/useRovingFocus';
 *
 * const rovingFocusProps = useRovingFocus();
 *
 * <div {...rovingFocusProps}>
 *     <button type="button">first</button>
 *     <button type="button">second</button>
 * </div>
 * ```
 */    function b() {
        var e = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
        var r = (0, t.useRef)(0);
        var o = (0, t.useRef)([]);
        var u = i(i({}, l), e);
        var a = function e(t) {
            r.current = t;
            o.current.forEach((function(e, t) {
                return t === r.current ? e.removeAttribute("tabindex") : e.setAttribute("tabindex", "-1");
            }));
        };
        var c = function e(t) {
            o.current = t != null ? (0, n.getSortedTabbableElements)(t, {
                ignoreTabIndex: true
            }) : [];
            a(r.current);
        };
        var f = function e(r) {
            var t = r.relatedTarget instanceof HTMLElement && o.current.includes(r.relatedTarget);
            if (!t) {
                a(0);
            }
        };
        var b = function e(r) {
            var t = r.target instanceof HTMLElement ? o.current.indexOf(r.target) : -1;
            if (t >= 0) {
                a(t);
            }
        };
        var s = function e(r) {
            var t = r.nativeEvent.target;
            var i = t instanceof HTMLElement ? o.current.indexOf(t) : -1;
            if (i === -1) {
                return;
            }
            /*  Only apply focus if we are among the navigable items in this container */            (0, 
            n.handleFocus)(r.key, o.current, i, u);
        };
        return {
            ref: c,
            onBlur: f,
            onFocus: b,
            onKeyDown: s
        };
    }
    /* harmony default export */    const s = b;
    // CONCATENATED MODULE: ./src/useRovingFocus/index.ts
    module.exports = r;
    /******/})();
