// @splunk/react-ui/WaitSpinner.js
/******/ (() => {
    // webpackBootstrap
    /******/ "use strict";
    /******/ // The require scope
    /******/    var e = {};
    /******/
    /************************************************************************/
    /******/ /* webpack/runtime/compat get default export */
    /******/    (() => {
        /******/ // getDefaultExport function for compatibility with non-harmony modules
        /******/ e.n = t => {
            /******/ var r = t && t.__esModule ? 
            /******/ () => t["default"]
            /******/ : () => t
            /******/;
            e.d(r, {
                a: r
            });
            /******/            return r;
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/define property getters */
    /******/    (() => {
        /******/ // define getter functions for harmony exports
        /******/ e.d = (t, r) => {
            /******/ for (var n in r) {
                /******/ if (e.o(r, n) && !e.o(t, n)) {
                    /******/ Object.defineProperty(t, n, {
                        enumerable: true,
                        get: r[n]
                    });
                    /******/                }
                /******/            }
            /******/        };
        /******/    })();
    /******/
    /******/ /* webpack/runtime/hasOwnProperty shorthand */
    /******/    (() => {
        /******/ e.o = (e, t) => Object.prototype.hasOwnProperty.call(e, t)
        /******/;
    })();
    /******/
    /******/ /* webpack/runtime/make namespace object */
    /******/    (() => {
        /******/ // define __esModule on exports
        /******/ e.r = e => {
            /******/ if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
                /******/ Object.defineProperty(e, Symbol.toStringTag, {
                    value: "Module"
                });
                /******/            }
            /******/            Object.defineProperty(e, "__esModule", {
                value: true
            });
            /******/        };
        /******/    })();
    /******/
    /************************************************************************/    var t = {};
    // ESM COMPAT FLAG
        e.r(t);
    // EXPORTS
        e.d(t, {
        default: () => /* reexport */ z
    });
    // CONCATENATED MODULE: external "react"
    const r = require("react");
    var n =  e.n(r);
    // CONCATENATED MODULE: external "prop-types"
    const a = require("prop-types");
    var i =  e.n(a);
    // CONCATENATED MODULE: external "@splunk/react-ui/AnimationToggle"
    const s = require("@splunk/react-ui/AnimationToggle");
    // CONCATENATED MODULE: external "@splunk/ui-utils/i18n"
    const o = require("@splunk/ui-utils/i18n");
    // CONCATENATED MODULE: external "styled-components"
    const l = require("styled-components");
    var c =  e.n(l);
    // CONCATENATED MODULE: external "@splunk/themes"
    const f = require("@splunk/themes");
    // CONCATENATED MODULE: ./src/WaitSpinner/WaitSpinnerStyles.ts
    var d = c().div.withConfig({
        displayName: "WaitSpinnerStyles__Styled",
        componentId: "sc-1nu971z-0"
    })([ "", ";align-items:center;" ], f.mixins.reset("inline-flex"));
    var u = (0, l.keyframes)([ "100%{transform:rotate(360deg);}" ]);
    var m = (0, l.keyframes)([ "0%{transform:scale(0);opacity:0;}100%{transform:scale(1);opacity:1;}" ]);
    var p = "2.07s";
    var y = c().svg.withConfig({
        displayName: "WaitSpinnerStyles__StyledSvg",
        componentId: "sc-1nu971z-1"
    })([ "", " ", "" ], (0, f.pickVariant)("$size", {
        small: (0, l.css)([ "width:16px;height:16px;" ]),
        medium: (0, l.css)([ "width:24px;height:24px;" ]),
        large: (0, l.css)([ "width:40px;height:40px;" ])
    }), (function(e) {
        var t = e.$animated;
        return t && (0, l.css)([ "animation:", " ", " infinite linear,", " 500ms cubic-bezier(0.01,0,0,1);" ], u, p, m);
    }));
    var v = (0, l.keyframes)([ "0%{stroke-dasharray:110 10;stroke-dashoffset:-5.5;animation-timing-function:cubic-bezier(0.8,0,0.83,1);}50%{stroke-dasharray:26 94;stroke-dashoffset:-152;}100%{stroke-dasharray:110 10;stroke-dashoffset:-246.5;animation-timing-function:cubic-bezier(0.33,0,0.67,1);}" ]);
    var h = (0, l.keyframes)([ "0%{stroke-dasharray:7 113;stroke-dashoffset:3;animation-timing-function:cubic-bezier(0.8,0,0.83,1);}50%{stroke-dasharray:90 30;stroke-dashoffset:-60;}100%{stroke-dasharray:7 113;stroke-dashoffset:-238;animation-timing-function:cubic-bezier(0.33,0,0.67,1);}" ]);
    var g = c().circle.withConfig({
        displayName: "WaitSpinnerStyles__StyledBasePath",
        componentId: "sc-1nu971z-2"
    })([ "stroke:", ";stroke-width:2;fill:transparent;", " stroke-dasharray:110 10;stroke-dashoffset:-5.5;" ], f.variables.contentColorDisabled, (function(e) {
        var t = e.$animated;
        return t && (0, l.css)([ "animation:", " ", " infinite;animation-fill-mode:backwards;" ], v, p);
    }));
    var b = c().circle.withConfig({
        displayName: "WaitSpinnerStyles__StyledFillPath",
        componentId: "sc-1nu971z-3"
    })([ "fill:transparent;stroke-width:2;stroke:", ";stroke-dasharray:7 113;stroke-dashoffset:3;", "" ], f.variables.contentColorAccent, (function(e) {
        var t = e.$animated;
        return t && (0, l.css)([ "animation:", " ", " infinite;" ], h, p);
    }));
    // CONCATENATED MODULE: ./src/WaitSpinner/WaitSpinner.tsx
    function k() {
        return k = Object.assign ? Object.assign.bind() : function(e) {
            for (var t = 1; t < arguments.length; t++) {
                var r = arguments[t];
                for (var n in r) {
                    ({}).hasOwnProperty.call(r, n) && (e[n] = r[n]);
                }
            }
            return e;
        }, k.apply(null, arguments);
    }
    function w(e, t) {
        if (null == e) return {};
        var r, n, a = S(e, t);
        if (Object.getOwnPropertySymbols) {
            var i = Object.getOwnPropertySymbols(e);
            for (n = 0; n < i.length; n++) {
                r = i[n], -1 === t.indexOf(r) && {}.propertyIsEnumerable.call(e, r) && (a[r] = e[r]);
            }
        }
        return a;
    }
    function S(e, t) {
        if (null == e) return {};
        var r = {};
        for (var n in e) {
            if ({}.hasOwnProperty.call(e, n)) {
                if (-1 !== t.indexOf(n)) continue;
                r[n] = e[n];
            }
        }
        return r;
    }
    var O = {
        elementRef: i().oneOfType([ i().func, i().object ]),
        screenReaderText: i().oneOfType([ i().string, i().oneOf([ "null" ]) ]),
        size: i().oneOf([ "small", "medium", "large" ])
    };
    function x(e) {
        var t = e.elementRef, r = e.screenReaderText, a = r === void 0 ? (0, o._)("Waiting") : r, i = e.size, l = i === void 0 ? "small" : i, c = w(e, [ "elementRef", "screenReaderText", "size" ]);
        // @docs-props-type WaitSpinnerPropsBase
                var f = (0, s.useAnimationToggle)() === "on";
        
        return n().createElement(d, k({
            "data-test": "wait-spinner",
            ref: t
        }, c),  n().createElement(y, {
            viewBox: "0 0 40 40",
            version: "1.1",
            xmlns: "http://www.w3.org/2000/svg",
            $animated: f,
            $size: l
        }, a &&  n().createElement("title", null, a),  n().createElement("g", null,  n().createElement(g, {
            r: "19",
            cx: "20",
            cy: "20",
            $animated: f,
            $size: l
        }),  n().createElement(b, {
            r: "19",
            cx: "20",
            cy: "20",
            $animated: f,
            $size: l
        }))));
    }
    x.propTypes = O;
    /* harmony default export */    const z = x;
    // CONCATENATED MODULE: ./src/WaitSpinner/index.ts
    module.exports = t;
    /******/})();
