// intl-tel-input/vue/build/exports/IntlTelInput.mjs
import { defineComponent as g, mergeModels as m, useModel as b, ref as r, onMounted as h, watch as C, onUnmounted as M, withDirectives as N, createElementBlock as P, openBlock as D, mergeProps as E, vModelText as I } from "vue";
import { i as k } from "../intl-tel-input-f0-6rrpA.mjs";
const B = /* @__PURE__ */ g({
  inheritAttrs: !1,
  __name: "IntlTelInput",
  props: /* @__PURE__ */ m({
    options: { default: () => ({}) },
    disabled: { type: Boolean, default: !1 },
    inputProps: { default: () => ({}) }
  }, {
    modelValue: {
      default: ""
    },
    modelModifiers: {}
  }),
  emits: /* @__PURE__ */ m([
    "changeNumber",
    "changeCountry",
    "changeValidity",
    "changeErrorCode"
  ], ["update:modelValue"]),
  setup(s, { expose: v, emit: f }) {
    const u = b(s, "modelValue"), l = s, o = f, n = r(), t = r(), i = r(!1), d = () => t.value ? l.options.strictMode ? t.value.isValidNumberPrecise() : t.value.isValidNumber() : null, p = () => {
      let e = d();
      i.value !== e && (i.value = e, o("changeValidity", !!e), o(
        "changeErrorCode",
        e ? null : t.value.getValidationError()
      ));
    }, c = () => {
      var e;
      o("changeNumber", ((e = t.value) == null ? void 0 : e.getNumber()) ?? ""), p();
    }, y = () => {
      var e;
      o("changeCountry", ((e = t.value) == null ? void 0 : e.getSelectedCountryData().iso2) ?? ""), c(), p();
    };
    return h(() => {
      n.value && (t.value = k(n.value, l.options), u.value && t.value.setNumber(u.value), l.disabled && t.value.setDisabled(l.disabled), i.value = d());
    }), C(
      () => l.disabled,
      (e) => {
        var a;
        return (a = t.value) == null ? void 0 : a.setDisabled(e);
      }
    ), M(() => {
      var e;
      return (e = t.value) == null ? void 0 : e.destroy();
    }), v({ instance: t, input: n }), (e, a) => N((D(), P("input", E({
      ref_key: "input",
      ref: n,
      "onUpdate:modelValue": a[0] || (a[0] = (V) => u.value = V),
      type: "tel",
      onCountrychange: y,
      onInput: c
    }, e.inputProps), null, 16)), [
      [
        I,
        u.value,
        void 0,
        { lazy: !0 }
      ]
    ]);
  }
});
export {
  B as default
};
