// @splunk/ui-utils/filter.js
"use strict";
/**
 * @file
 * A set of functions for filtering items, often menu items. For simple use cases,
 * use `filterByKeywords`. Use `stringToKeywords` and `testPhrase` when more control
 * of the filtering is necessary.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringToKeywords = stringToKeywords;
exports.testPhrase = testPhrase;
exports.keywordLocationsAll = keywordLocationsAll;
exports.keywordLocations = keywordLocations;
exports.filterByKeywords = filterByKeywords;
const lodash_1 = require("lodash");
/**
 * Converts a string (filterPhrase) to an array of keyword tokens. Tokens are usually words, but can
 * be a multi-word phrase if quotes are used. The output is suitable for the `testPhrase` function.
 *
 * @param {String} filterPhrase - The phrase to be broken into keyword tokens.
 * @returns {String[]}
 * @public
 */
function stringToKeywords(filterPhrase) {
    if (!(0, lodash_1.isString)(filterPhrase)) {
        return [];
    }
    return (filterPhrase.match(/(?:[^\s"]+|"[^"]*")+/g) || []).map((word) => word.replace(/^"(.*)"$/, '$1').toLowerCase());
}
/**
 * Tests if a phrase matches a list of keywords. All keywords must be included in the phrase for a
 * match.
 *
 * Examples:
 * ```js
 * stringToKeywords('Named Bob'); // ['Named', 'Bob']
 * stringToKeywords('"Named Bob"'); // ['Named Bob']
 * stringToKeywords('A Street Cat "Named Bob"'); // ['A', 'Street', 'Cat', 'Named Bob']
 * ```
 *
 * @param {String} phrase - The test phrase.
 * @param {String[]} keywords - An array of keywords, as returned by `stringToKeywords`.
 * @returns {Boolean}
 * @public
 */
function testPhrase(phrase, keywords) {
    if (!(0, lodash_1.isString)(phrase)) {
        return false;
    }
    const p = phrase.toLowerCase();
    return keywords.every((keyword) => (0, lodash_1.includes)(p, keyword));
}
/**
 * Looks for keyword locations in a phrase. Not all keywords need to match for results to be returned.
 *
 * @param {String} phrase - The test phrase.
 * @param {String[]} keywords - An array of keywords, as returned by `stringToKeywords`.
 * @returns {Object[] | false} An array of matches with keyword, start index, and end index. Array is sorted
 * by start and end index. Keyword
 * ranges may overlap. Example:
 * ```js
 * [
 *     {
 *         keyword: 'the dash',
 *         start: 0,
 *         end: 8,
 *     },
 *     {
 *         keyword: 'sales',
 *         start: 12,
 *         end: 17,
 *     },
 * ]
 * ```
 * @private
 */
function keywordLocationsAll(phrase, keywords) {
    if (!(0, lodash_1.isString)(phrase) || !keywords || !keywords.length) {
        return false;
    }
    const p = phrase.toLowerCase();
    const matches = keywords
        .reduce((acc, keyword) => {
        const start = p.indexOf(keyword);
        if (start >= 0) {
            acc.push({
                keyword,
                start,
                end: start + keyword.length,
            });
        }
        return acc;
    }, [])
        .sort((a, b) => {
        if (a.start !== b.start) {
            return a.start > b.start ? 1 : -1;
        }
        return a.end > b.end ? 1 : -1;
    });
    return matches;
}
/**
 * Looks for keyword locations in a phrase and return portions of the string that match one or more
 * keywords. The return value can be used to highlight the matched text.
 *
 * @param {String} phrase - The test phrase.
 * @param {String[]} keywords - An array of keywords, as returned by `stringToKeywords`.
 * @returns {Object[] | false} An array of location with start index and end index. Keyword
 * ranges can overlap. Example:
 * ```js
 * [
 *     {
 *         start: 0,
 *         end: 8,
 *     },
 *     {
 *         start: 12,
 *         end: 17,
 *     },
 * ]
 * ```
 * @public
 */
function keywordLocations(phrase, keywords) {
    const matches = keywordLocationsAll(phrase, keywords);
    if (!matches || matches.length !== keywords.length) {
        return false;
    }
    let { start, end } = matches[0];
    const locations = [];
    function pushLocation() {
        locations.push({
            start,
            end,
        });
    }
    matches.slice(1).forEach((match) => {
        if (match.start > end) {
            pushLocation();
            ({ start, end } = match);
        }
        else if (match.end > end) {
            ({ end } = match);
        }
    });
    pushLocation();
    return locations;
}
/**
 * Filters an array of `items` against the `filterPhrase`.
 *
 * @param {Array} items - An array of strings or objects to filter.
 * @param {String} filterPhrase
 * @param {Function} [valueGetter] - An optional function that returns the property of interest if
 * filtering an array of objects.
 * @returns {Array} A filtered list of items.
 * @public
 */
function filterByKeywords(items, filterPhrase, valueGetter) {
    const keywords = stringToKeywords(filterPhrase);
    return items.filter((item) => testPhrase(valueGetter ? valueGetter(item) : item, keywords));
}
