// @splunk/ui-utils/focus.js
"use strict";
/**
 * @file
 * Utilities for managing focus in the browser.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateTabIndex = exports.getNewIndex = exports.calculateIndex = exports.isHidden = void 0;
exports.getSortedTabbableElements = getSortedTabbableElements;
exports.isTabKey = isTabKey;
exports.handleTab = handleTab;
exports.takeFocus = takeFocus;
exports.handleFocus = handleFocus;
const lodash_1 = require("lodash");
const tabbableSelectors = [
    'a[href]',
    'input:not([disabled])',
    'select:not([disabled])',
    'textarea:not([disabled])',
    'button:not([disabled])',
    '[contenteditable]',
];
const tabbableSelectorsWithTabIndex = tabbableSelectors.concat(['[tabindex]']);
const isStyleVisible = (element) => {
    const style = window.getComputedStyle(element);
    return style.display !== 'none' && style.visibility !== 'hidden';
};
const isAttributeVisible = (element) => !element.hasAttribute('hidden') && !element.hasAttribute('inert');
const isElementVisible = (element) => {
    const visibleByStyle = isStyleVisible(element);
    const visibleByAttribute = isAttributeVisible(element);
    return visibleByStyle && visibleByAttribute;
};
/**
 * Determines whether an html element is considered hidden based on its dimensions or computed visibility style.
 *
 * This function is primarily used in environments where accurate layout-related properties
 * are available, like a real browser environment.
 *
 * @param {HTMLElement} element - The DOM element to check for visibility.
 * @returns {boolean} - Returns true if the element is hidden, otherwise false.
 *
 * @private Exported for testing purposes
 */
const isHidden = (element) => {
    // SUI-6521: `element.getClientRects().length > 0` is used to distinguish between 'actual' 0-size elements and Jest 0-size elements.
    // 'actual' 0-size elements typically still have a value from getClientRects.
    const isZeroSize = element.getClientRects().length > 0 &&
        (element.offsetWidth === 0 || element.offsetHeight === 0);
    return isZeroSize || !isElementVisible(element);
};
exports.isHidden = isHidden;
/**
 * Sorts the tabbable elements in the provided container.
 *
 * @param {Element} container - The target container.
 * @param {Object} [options={ignoreTabIndex: false}] - Configuration options.
 * @param {boolean} [options.ignoreTabIndex=false] - If set to true, the function will neither use "tabindex" as an indicator of selectability nor disqualify elements based on "tabindex=-1".
 * @returns {HTMLElement[]} A sorted array of tabbable elements within the container.
 * @public
 */
function getSortedTabbableElements(container, { ignoreTabIndex } = { ignoreTabIndex: false }) {
    const tabbableElementSelector = (ignoreTabIndex ? tabbableSelectors : tabbableSelectorsWithTabIndex).join(', ');
    const elems = container.querySelectorAll(tabbableElementSelector);
    const tabbableElems = (0, lodash_1.filter)(elems, (element) => {
        const el = element;
        const isValidTabIndex = ignoreTabIndex || el.tabIndex >= 0;
        return (!(0, exports.isHidden)(el) && isValidTabIndex) || el === document.activeElement;
    });
    const reducedElements = tabbableElems.reduce((accumulator, currentElement) => {
        const prevElement = accumulator[accumulator.length - 1];
        const isPrevElementRadio = (prevElement === null || prevElement === void 0 ? void 0 : prevElement.getAttribute('type')) === 'radio';
        const isCurrElementRadio = currentElement.getAttribute('type') === 'radio';
        const sameRadioGroup = currentElement.getAttribute('name') === (prevElement === null || prevElement === void 0 ? void 0 : prevElement.getAttribute('name'));
        // Reduce radio elements to a single Tab stop, ensuring only one per group is focusable
        if (isPrevElementRadio && isCurrElementRadio && sameRadioGroup) {
            if (currentElement.checked) {
                // replace previous radio with current checked radio
                accumulator.pop();
                accumulator.push(currentElement);
            }
        }
        else {
            accumulator.push(currentElement);
        }
        return accumulator;
    }, []);
    return (0, lodash_1.sortBy)(reducedElements, (el) => {
        if (el.tabIndex > 0) {
            return -1 / el.tabIndex;
        }
        return 0;
    });
}
function isTabKey(event) {
    return !(event.key !== 'Tab' || event.metaKey || event.altKey || event.ctrlKey);
}
/**
 * A key event handler that moves focus among tabbable elements within a container.
 *
 * @param {Element} container - The target container.
 * @param {Event} event - The key event to handle.
 * @returns {HTMLElement|null} The element focus was applied to or `null` if focus was not applied.
 * @public
 */
function handleTab(container, event) {
    // Ignore events bubbling up from portals
    if (!container.contains(document.activeElement)) {
        return null;
    }
    // Ensure this is a valid event
    if (!isTabKey(event)) {
        return null;
    }
    const tabbableElements = getSortedTabbableElements(container);
    if (tabbableElements.length === 0) {
        // if the container is focused, don't allow the focus to leave.
        if (document.activeElement === container) {
            event.preventDefault();
            return container;
        }
        return null;
    }
    // Find the current index or set default.
    const currentElement = (event && event.target) || container.querySelector(':focus');
    let currentIndex = tabbableElements.indexOf(currentElement);
    if (currentIndex === -1) {
        currentIndex = event.shiftKey ? 0 : tabbableElements.length - 1;
    }
    // Shift the array instead of the current index.
    if (event.shiftKey) {
        const lastElement = tabbableElements.pop();
        if (lastElement) {
            tabbableElements.unshift(lastElement); // move last to first
        }
    }
    else {
        const firstElement = tabbableElements.shift();
        if (firstElement) {
            tabbableElements.push(firstElement); // move first to last
        }
    }
    // Focus
    event.preventDefault();
    tabbableElements[currentIndex].focus();
    return tabbableElements[currentIndex];
}
/**
 * A helper method that focuses on the first focusable element in a container.
 * If a contained element already has focus, focus does not shift.
 *
 * @param {HTMLElement} container - The container that takes focus.
 * @param {String} [defaultElement='first'] - The target of focus, either 'first' or 'container'.
 * @returns {Element|null} The element that was focused, or null if no element was focused.
 * @public
 */
function takeFocus(container, defaultElement = 'first') {
    const currentElement = container.querySelector(':focus');
    if (currentElement) {
        return currentElement;
    }
    if (defaultElement === 'first') {
        const targetEl = getSortedTabbableElements(container)[0];
        if (targetEl) {
            (0, lodash_1.defer)(() => targetEl.focus());
            return targetEl;
        }
    }
    if (container.hasAttribute('tabIndex')) {
        (0, lodash_1.defer)(() => container.focus());
        return container;
    }
    return null;
}
/**
 * An enumeration of possible navigation directions.
 * @readonly
 * @enum {string}
 */
const focusMoveDirection = Object.freeze({
    NONE: 'none',
    NEXT: 'next',
    PREVIOUS: 'previous',
});
/**
 * Helper function to calculate the new index from the current index.
 *
 * This function determines the new index for focus navigation based on the current index,
 * the direction of movement, and whether looping is enabled. It either caps at the start/end
 * of the array or loops around based on the `enableLoop` flag.
 *
 * @returns {number} - The new index. Returns the current index for invalid direction or out-of-bounds index when looping is disabled.
 */
const calculateIndex = ({ itemsLength, currentIndex, direction, enableLoop, }) => {
    // Check if the direction is valid
    if (direction === focusMoveDirection.NONE ||
        ![focusMoveDirection.NONE, focusMoveDirection.NEXT, focusMoveDirection.PREVIOUS].includes(direction)) {
        return currentIndex; // Return current index for invalid direction
    }
    const increment = direction === focusMoveDirection.NEXT ? 1 : -1;
    if (!enableLoop) {
        const newIndex = currentIndex + increment;
        if (newIndex >= itemsLength || newIndex < 0) {
            return currentIndex;
        }
        return newIndex;
    }
    return (currentIndex + increment + itemsLength) % itemsLength;
};
exports.calculateIndex = calculateIndex;
/**
 * Helper function to get the new focusable index in the items array based on the key.
 *
 * This function calculates the next focusable index in a list of items, depending on the provided key
 * (e.g. arrow keys, Home, End, Tab). It takes into account the current focus index, orientation of navigation
 * (horizontal or vertical), looping behavior, and specific key handling rules.
 *
 * @returns {number} - The new focusable index. Returns -1 if no valid index is found.
 */
const getNewIndex = (key, itemsLength, currentIndex, { enableLoop, orientation, enableTab, enableHomeEnd, }) => {
    // If the current focused item is not in the actionItems, then focus on the first focusable item.
    if (currentIndex < 0 || currentIndex >= itemsLength) {
        return 0;
    }
    if (key === 'Home') {
        return enableHomeEnd ? 0 : -1;
    }
    if (key === 'End') {
        return enableHomeEnd ? itemsLength - 1 : -1;
    }
    let direction = focusMoveDirection.NONE;
    if (orientation === 'horizontal') {
        if (key === 'ArrowLeft') {
            direction = focusMoveDirection.PREVIOUS;
        }
        if (key === 'ArrowRight') {
            direction = focusMoveDirection.NEXT;
        }
    }
    else if (orientation === 'vertical') {
        if (key === 'ArrowUp') {
            direction = focusMoveDirection.PREVIOUS;
        }
        if (key === 'ArrowDown') {
            direction = focusMoveDirection.NEXT;
        }
    }
    if (enableTab) {
        // tabShift is a custom key we defined in Menu
        if (key === 'tabShift') {
            direction = focusMoveDirection.PREVIOUS;
        }
        if (key === 'Tab') {
            direction = focusMoveDirection.NEXT;
        }
    }
    if (direction !== focusMoveDirection.NONE) {
        return (0, exports.calculateIndex)({ itemsLength, currentIndex, direction, enableLoop });
    }
    return -1;
};
exports.getNewIndex = getNewIndex;
/**
 *  Helper function to update the tabindex of action items, setting the active item to be focusable
 */
const updateTabIndex = (activeIndex, actionItems) => {
    actionItems.forEach((el, index) => el.setAttribute('tabindex', index === activeIndex ? '0' : '-1'));
};
exports.updateTabIndex = updateTabIndex;
/**
 * Handles focus navigation for a list of action items based on key presses.
 *
 * @param {string} key - The key value obtained from KeyboardEvent.key, representing the keyboard key pressed.
 * @param {Element[]} actionItems - An array of HTML elements representing the action items.
 * @param {number} currentIndex - The current index of the focused item.
 * @param {Object} [options={ enableLoop: false, orientation: 'horizontal', enableTab: false, enableHomeEnd: true }] - Configuration options.
 * @param {boolean} [options.enableLoop=false] - Whether navigation should loop around the ends of the list.
 * @param {string} [options.orientation='horizontal'] - Navigation orientation: 'horizontal' (left/right arrow keys) or 'vertical' (up/down arrow keys).
 * @param {boolean} [options.enableTab=false] - Enable navigation with the Tab key.
 * @param {boolean} [options.enableHomeEnd=true] - Enable navigation with Home and End keys.
 * @public
 */
function handleFocus(key, actionItems, currentIndex, { enableLoop = false, orientation = 'horizontal', enableTab = false, enableHomeEnd = true, } = {}) {
    var _a, _b;
    const itemsLength = actionItems.length;
    // If tab key navigation is disabled, the group should have only one tab stop.
    // `updateTabIndex` adjusts the tabindex of all items accordingly for forward navigation.
    if (!enableTab && key === 'Tab') {
        (0, exports.updateTabIndex)(0, actionItems);
        return;
    }
    const newIndex = (0, exports.getNewIndex)(key, itemsLength, currentIndex, {
        enableLoop,
        orientation,
        enableTab,
        enableHomeEnd,
    });
    if (newIndex >= 0 && newIndex < itemsLength) {
        (0, exports.updateTabIndex)(newIndex, actionItems);
        (_b = (_a = actionItems[newIndex]).focus) === null || _b === void 0 ? void 0 : _b.call(_a);
    }
    // If tab key navigation is disabled, the group should only have one tab stop.
    // `updateTabIndex` adjusts the tabindex of all items accordingly for the backward navigation.
    // tabShift is a custom key we defined in Menu
    if (newIndex === -1 && !enableTab && key === 'tabShift') {
        (0, exports.updateTabIndex)(currentIndex, actionItems);
    }
}
