// @splunk/ui-utils/keyboard.js
"use strict";
/**
 * @file
 * Utilities for handling keyboard events.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.keycode = keycode;
exports.isNumber = isNumber;
exports.isDecimal = isDecimal;
exports.isMinus = isMinus;
exports.isNumeric = isNumeric;
exports.addsCharacter = addsCharacter;
const keycode_1 = __importDefault(require("keycode"));
const lodash_1 = require("lodash");
// Used to make the code compatible with `userEvent.keyboard`,
// which uses key names like "ArrowDown" that need to be mapped to "down".
const customKeyMap = Object.freeze({
    ArrowDown: 'down',
    ArrowUp: 'up',
    ArrowLeft: 'left',
    ArrowRight: 'right',
    Home: 'home',
    End: 'end',
    Tab: 'tab',
});
function isKeyInCustomKeyMap(key) {
    return key in customKeyMap;
}
/**
 * A utility for mapping key names with their numeric codes. This is an alias for [the third-party
 * library, keycode](https://github.com/timoxley/keycode).
 *
 * @deprecated This function is deprecated and will be removed in a future major version.
 * Use `KeyboardEvent`'s `key` property for identifying key presses instead.
 * @param {KeyboardEvent} event - A keyboard event.
 * @returns {String}
 * @public
 */
function keycode(event) {
    var _a;
    // If kc doesn't return a result, fall back to the custom customKeyMap
    return (_a = (0, keycode_1.default)(event)) !== null && _a !== void 0 ? _a : (isKeyInCustomKeyMap(event.key) ? customKeyMap[event.key] : undefined);
}
/**
 * Whenever possible this library uses event.key because it has fewer issues than event.keyCode.
 * However, Safari 9.0 and earlier only support it for a few keys, such as Enter, and the QA
 * test suite doesn't provide it.
 */
function isValidKey(key) {
    return (0, lodash_1.isUndefined)(key) ? false : key !== 'Unidentified';
}
/**
 * Tests if the event key is a number.
 * @param {Event} event - @deprecated Support for `keyCode` in this function is deprecated and will be removed in a future major version.
 * A keyboard event that includes a `key`, a `keyCode`, or both.
 * @returns {Boolean}
 * @public
 */
function isNumber({ key, keyCode }) {
    if (isValidKey(key)) {
        const keys = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0'];
        return keys.indexOf(key) >= 0;
    }
    return (keyCode >= 48 && keyCode <= 57) || (keyCode >= 96 && keyCode <= 105);
}
/**
 * Tests if the event key is a decimal.
 * @param {Event} event - @deprecated Support for `keyCode` in this function is deprecated and will be removed in a future major version.
 * A keyboard event that includes a `key`, a `keyCode`, or both.
 * @param {Object} [options]
 * @param {String} [locale.string = 'en-US'] -  The locale determines the decimal separator. Supported locale formats are: `xx`, `xx-XX`, and `xx_XX`.
 * @returns {Boolean}
 * @public
 */
function isDecimal({ key, keyCode }, { locale = 'en-US' } = {}) {
    // A more robust and expandable solution would be to use formatToParts() once IE requirements are dropped.
    const formattedNumber = new Intl.NumberFormat(locale.replace('_', '-')).format(1.2);
    const decimalSeparator = formattedNumber.indexOf(',') > -1 ? ',' : '.';
    if (isValidKey(key)) {
        return key === decimalSeparator || key === 'Decimal';
    }
    // keycode doesn't support the string overload for kc, so using it directly here
    if (decimalSeparator === '.' && keyCode === (0, keycode_1.default)('numpad .')) {
        return true;
    }
    // keycode doesn't support the string overload for kc, so using it directly here
    return keyCode === (0, keycode_1.default)(decimalSeparator);
}
/**
 * Tests if the event key is a minus sign.
 * @param {Event} event - @deprecated Support for `keyCode` in this function is deprecated and will be removed in a future major version.
 * A keyboard event that includes a `key`, a `keyCode`, or both.
 * @returns {Boolean}
 * @public
 */
function isMinus({ key, keyCode }) {
    if (isValidKey(key)) {
        return key === '-' || key === 'Subtract';
    }
    // keycode doesn't support the string overload for kc, so using it directly here
    return keyCode === (0, keycode_1.default)('numpad -') || keyCode === (0, keycode_1.default)('-');
}
/**
 * Tests if the event key is a numeric character (number, decimal, or minus).
 * @param {Event} event - @deprecated Support for `keyCode` in this function is deprecated and will be removed in a future major version.
 * A keyboard event that includes a `key`, a `keyCode`, or both.
 * @param {Object} [options]
 * @param {String} [locale.string = 'en-US'] - The locale determines the decimal separator. Supported locale formats are: `xx`, `xx-XX`, and `xx_XX`.
 * @returns {Boolean}
 * @public
 */
function isNumeric(event, options) {
    return isNumber(event) || isDecimal(event, options) || isMinus(event);
}
/**
 * Tests if the event key adds a character. Enter and Tab return false even though they
 * add characters in some situations.
 * Caveat: Safari 9.0 and earlier might return undefined as this cannot be practically
 * determined.
 * @param {Event} event - A keyboard event that includes a key.
 * @returns {Boolean|Undefined}
 * @public
 */
function addsCharacter({ key }) {
    if (isValidKey(key)) {
        // IE 11 named characters
        const names = ['Add', 'Decimal', 'Divide', 'Multiply', 'Spacebar', 'Subtract'];
        // If key is one character it's assumed to be inserting a character rather than some other
        // type of action.
        return key.length === 1 || names.indexOf(key) >= 0;
    }
    // Safari 9.0 and earlier may return undefined
    return undefined;
}
