// @splunk/ui-utils/search.js
"use strict";
/**
 * @file
 * Search helpers that extend keyword tokenization with scoring and ranking across name, keyword, and
 * description fields. Use these utilities when you need ordered results; use `filter` for
 * simple include/exclude filtering without relevance scoring.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.rankItems = exports.getSearchResultScore = void 0;
exports.addSearchData = addSearchData;
exports.search = search;
const parseSearchTokens = (filter) => {
    return filter
        .trim()
        .split(/\s+/)
        .map((token) => token.toLowerCase())
        .filter((token) => token.length > 0);
};
/**
 * Constants used for scoring search matches.
 * Higher scores indicate better matches.
 */
const SEARCH_SCORE = {
    /**
     * Exact match
     */
    EXACT: 100,
    /**
     * Case insensitive substring
     */
    SUBSTR: 10,
    /**
     * No match found
     */
    NONE: 0,
};
/**
 * Sentinel value representing "no match position found".
 *
 * Used as the default nameMatchPosition to ensure items without name matches
 * are sorted after items with name matches during ranking.
 *
 */
const NO_MATCH_POSITION = Number.POSITIVE_INFINITY;
const scoreTerm = (token, searchData) => {
    const { nameLower, keywordsLower, descriptionLower } = searchData;
    const searchScore = {
        hasMatch: false,
        nameScore: SEARCH_SCORE.NONE,
        nameMatchPosition: NO_MATCH_POSITION,
        keywordsScore: SEARCH_SCORE.NONE,
        descriptionScore: SEARCH_SCORE.NONE,
    };
    // Name field
    const nameMatchIndex = nameLower.indexOf(token);
    if (nameMatchIndex > -1) {
        searchScore.hasMatch = true;
        searchScore.nameScore = SEARCH_SCORE.SUBSTR;
        searchScore.nameMatchPosition = Math.min(searchScore.nameMatchPosition, nameMatchIndex);
    }
    // Keywords field
    if (keywordsLower === null || keywordsLower === void 0 ? void 0 : keywordsLower.includes(token)) {
        searchScore.hasMatch = true;
        searchScore.keywordsScore = SEARCH_SCORE.SUBSTR;
    }
    // Description field
    if (descriptionLower === null || descriptionLower === void 0 ? void 0 : descriptionLower.includes(token)) {
        searchScore.hasMatch = true;
        searchScore.descriptionScore = SEARCH_SCORE.SUBSTR;
    }
    return searchScore;
};
const getSearchResultScore = (filter, searchData) => {
    const searchTokens = parseSearchTokens(filter);
    const totalScore = {
        hasMatch: false,
        nameScore: SEARCH_SCORE.NONE,
        keywordsScore: SEARCH_SCORE.NONE,
        descriptionScore: SEARCH_SCORE.NONE,
        nameMatchPosition: NO_MATCH_POSITION,
    };
    // Check for an exact name match first, either the joined search tokens or filter
    const exactNameMatch = searchData.nameLower === searchTokens.join('').toLowerCase() ||
        searchData.nameLower === filter.toLowerCase();
    if (exactNameMatch) {
        totalScore.nameScore = SEARCH_SCORE.EXACT;
        totalScore.nameMatchPosition = 0;
    }
    let allSearchTokensMatch = true;
    searchTokens.forEach((token) => {
        const currentTokenScore = scoreTerm(token, searchData);
        if (!currentTokenScore.hasMatch) {
            allSearchTokensMatch = false;
        }
        // If an exact name match wasn't found, accumulate name scores
        if (!exactNameMatch) {
            totalScore.nameScore += currentTokenScore.nameScore;
        }
        totalScore.keywordsScore += currentTokenScore.keywordsScore;
        totalScore.descriptionScore += currentTokenScore.descriptionScore;
        totalScore.nameMatchPosition = Math.min(totalScore.nameMatchPosition, currentTokenScore.nameMatchPosition);
    });
    totalScore.hasMatch = exactNameMatch || allSearchTokensMatch;
    return totalScore;
};
exports.getSearchResultScore = getSearchResultScore;
const rankItems = (items) => {
    return items.sort((a, b) => {
        const aScore = a.searchScore;
        const bScore = b.searchScore;
        // Priority 1: Items with nameScore come first
        if (aScore.nameScore !== bScore.nameScore) {
            return bScore.nameScore - aScore.nameScore;
        }
        // Priority 2: Then keywords score
        if (aScore.keywordsScore !== bScore.keywordsScore) {
            return bScore.keywordsScore - aScore.keywordsScore;
        }
        // Priority 3: Then description score
        if (aScore.descriptionScore !== bScore.descriptionScore) {
            return bScore.descriptionScore - aScore.descriptionScore;
        }
        // Priority 4: Earlier name match positions come first
        if (aScore.nameMatchPosition !== bScore.nameMatchPosition) {
            return aScore.nameMatchPosition - bScore.nameMatchPosition;
        }
        // Priority 5: Alphabetical tiebreaker
        const aName = a.searchData.nameLower;
        const bName = b.searchData.nameLower;
        return aName.localeCompare(bName);
    });
};
exports.rankItems = rankItems;
/**
 * Returns a new array of items with each item containing precomputed search data to be used with `search`.
 * @param {Array} items - Each item must have `name`. Optionally, they can have `keywords` or `description` properties.
 * @returns {Array} A new array of items, each augmented with `searchData` properties
 * @public
 */
function addSearchData(items) {
    const itemsWithSearchData = items.map((item) => {
        const { name, description, keywords } = item;
        return Object.assign(Object.assign({}, item), { searchData: {
                nameLower: name.toLowerCase(),
                keywordsLower: keywords === null || keywords === void 0 ? void 0 : keywords.toLowerCase(),
                descriptionLower: description === null || description === void 0 ? void 0 : description.toLowerCase(),
            } });
    });
    return itemsWithSearchData;
}
/**
 * Returns filtered and sorted list of data based on provided search term.
 *
 * @param {Array} data - The array of items to search. Each item must include `searchData` property (use `addSearchData` to add it).
 * @param {String} filter - The search term to filter and rank the data by.
 * @returns {Array} An array of items that match the search term, sorted by relevance.
 * @public
 */
function search(filter, data) {
    const scoredData = data.map((item) => {
        return Object.assign(Object.assign({}, item), { searchScore: getSearchResultScore(filter, item.searchData) });
    });
    const dataWithMatches = scoredData.filter((item) => item.searchScore.hasMatch);
    const results = rankItems(dataWithMatches);
    return results;
}
