// @splunk/themes/storybook-addon-splunk-themes/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CHANGED = exports.SET = exports.CONFIGURE = exports.UPDATE = exports.ENABLE_SPLUNK_MAGNETIC_KEY = exports.DISABLE_MULTI_THEME_OPTIONS_KEY = exports.DISABLE_DUAL_THEME_KEY = exports.PARAM_KEY = exports.ADDON_ID = void 0;
var ADDON_ID = '@splunk/themes';
exports.ADDON_ID = ADDON_ID;
var PARAM_KEY = 'splunkThemes';
// TODO(SUI-8284): Remove deprecated `disableDualTheme` in next major version
exports.PARAM_KEY = PARAM_KEY;
var DISABLE_DUAL_THEME_KEY = 'disableDualTheme';
exports.DISABLE_DUAL_THEME_KEY = DISABLE_DUAL_THEME_KEY;
var DISABLE_MULTI_THEME_OPTIONS_KEY = 'disableMultiThemeOptions';
exports.DISABLE_MULTI_THEME_OPTIONS_KEY = DISABLE_MULTI_THEME_OPTIONS_KEY;
var ENABLE_SPLUNK_MAGNETIC_KEY = 'enableSplunkMagneticOption';
exports.ENABLE_SPLUNK_MAGNETIC_KEY = ENABLE_SPLUNK_MAGNETIC_KEY;
var UPDATE = "".concat(ADDON_ID, "/update");
exports.UPDATE = UPDATE;
var CONFIGURE = "".concat(ADDON_ID, "/configure");
exports.CONFIGURE = CONFIGURE;
var SET = "".concat(ADDON_ID, "/default");
exports.SET = SET;
var CHANGED = "".concat(ADDON_ID, "/changed");
exports.CHANGED = CHANGED;
