// @splunk/themes/storybook-addon-splunk-themes/withSplunkTheme.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withSplunkTheme = void 0;
var _react = _interopRequireWildcard(require("react"));
var _lodash = require("lodash");
var _styledComponents = _interopRequireDefault(require("styled-components"));
var _splunkMagnetic = _interopRequireDefault(require("../splunk-magnetic"));
var _SplunkThemeProvider = _interopRequireDefault(require("../SplunkThemeProvider"));
var _variables = _interopRequireDefault(require("../variables"));
var _constants = require("./constants");
var _themeOptions = require("./themeOptions");
var _themes = require("./themes");
var _util = require("./util");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) { "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); } return f; })(e, t); }
function _slicedToArray(r, e) { return _arrayWithHoles(r) || _iterableToArrayLimit(r, e) || _unsupportedIterableToArray(r, e) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0) { ; } } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(r) { if (Array.isArray(r)) return r; }
function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) { ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } } return n; }, _extends.apply(null, arguments); }
function _toConsumableArray(r) { return _arrayWithoutHoles(r) || _iterableToArray(r) || _unsupportedIterableToArray(r) || _nonIterableSpread(); }
function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _iterableToArray(r) { if ("undefined" != typeof Symbol && null != r[Symbol.iterator] || null != r["@@iterator"]) return Array.from(r); }
function _arrayWithoutHoles(r) { if (Array.isArray(r)) return _arrayLikeToArray(r); }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) { n[e] = r[e]; } return n; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n    grid-row: 1;\n    grid-column: labelText / colEnd;\n"]);
  _templateObject7 = function _templateObject7() {
    return data;
  };
  return data;
}
function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    font-size: 68px;\n"]);
  _templateObject6 = function _templateObject6() {
    return data;
  };
  return data;
}
function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    stroke: ", ";\n    fill: ", ";\n"]);
  _templateObject5 = function _templateObject5() {
    return data;
  };
  return data;
}
function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    grid-row: 1;\n    grid-column: colStart / labelText;\n    height: 20px;\n    width: 20px;\n"]);
  _templateObject4 = function _templateObject4() {
    return data;
  };
  return data;
}
function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    display: grid;\n    grid-template-rows: 100%;\n    grid-template-columns: [colStart] min-content [labelText] auto [colEnd];\n    gap: ", ";\n    padding: ", ";\n    background-color: ", ";\n    border-radius: ", ";\n"]);
  _templateObject3 = function _templateObject3() {
    return data;
  };
  return data;
}
function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    display: flex;\n    gap: ", ";\n    flex-direction: column;\n"]);
  _templateObject2 = function _templateObject2() {
    return data;
  };
  return data;
}
function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n    padding: ", ";\n"]);
  _templateObject = function _templateObject() {
    return data;
  };
  return data;
}
function _taggedTemplateLiteral(e, t) { return t || (t = e.slice(0)), Object.freeze(Object.defineProperties(e, { raw: { value: Object.freeze(t) } })); }
var StoryWrapper = _styledComponents["default"].div(_templateObject(), _variables["default"].backgroundColorPage, _variables["default"].spacingLarge);
var MultiStoryContainer = _styledComponents["default"].div(_templateObject2(), _variables["default"].spacingLarge);
var StyledWarning = _styledComponents["default"].div(_templateObject3(), _variables["default"].spacingSmall, _variables["default"].spacingSmall, _variables["default"].neutral200, _variables["default"].borderRadius);
var WarningIcon = _styledComponents["default"].svg(_templateObject4());
var StyledTriangle = _styledComponents["default"].polygon(_templateObject5(), _variables["default"].accentColorWarning, _variables["default"].accentColorWarning);
var StyledText = _styledComponents["default"].text(_templateObject6());
var StyledContent = _styledComponents["default"].span(_templateObject7());
var warningIcon = /*#__PURE__*/_react["default"].createElement(WarningIcon, {
  viewBox: "0 0 100 100",
  "aria-label": "warning icon"
}, /*#__PURE__*/_react["default"].createElement(StyledTriangle, {
  points: "50 15, 100 100, 0 100"
}), /*#__PURE__*/_react["default"].createElement(StyledText, {
  x: "41",
  y: "88"
}, "!"));
var handleSplunkMagneticFamily = function handleSplunkMagneticFamily(themeProps) {
  if (themeProps.family === 'splunk-magnetic') {
    return _objectSpread(_objectSpread({}, themeProps), {}, {
      family: 'prisma',
      customizeTheme: _splunkMagnetic["default"]
    });
  }
  return themeProps;
};

// The Splunk Magnetic option reuses Prisma tokens under the hood. We translate the
// toolbar selection here so the underlying theme provider always receives a Prisma family.
// Multi-render selections ("all" / "both") are treated as undefined because each variant
// gets its own explicit family inside the MultiStory renderer below.
var normalizeFamilyForProvider = function normalizeFamilyForProvider(family) {
  if (!family) {
    return undefined;
  }
  if (family === 'splunk-magnetic') {
    return 'prisma';
  }
  if (family === 'all' || family === 'both') {
    return undefined;
  }
  return family;
};
var shouldRenderMultiThemes = function shouldRenderMultiThemes(_ref) {
  var activeMultiThemeKey = _ref.activeMultiThemeKey,
    currentTheme = _ref.currentTheme,
    hasDisabledMultiThemeOptions = _ref.hasDisabledMultiThemeOptions;
  if (hasDisabledMultiThemeOptions || !activeMultiThemeKey) {
    return false;
  }
  var themeValue = currentTheme[activeMultiThemeKey];

  // TODO(SUI-8284): Remove deprecated `both` value support in next major version
  return themeValue === 'all' || themeValue === 'both';
};

// Renders a story multiple times side-by-side with different theme variants
var MultiStory = function MultiStory(_ref2) {
  var _children$props;
  var children = _ref2.children,
    themeOption = _ref2.themeOption,
    themeProviderProps = _ref2.themeProviderProps,
    hasEnabledSplunkMagneticOption = _ref2.hasEnabledSplunkMagneticOption;
  if (/*#__PURE__*/(0, _react.isValidElement)(children) && (children === null || children === void 0 ? void 0 : (_children$props = children.props) === null || _children$props === void 0 ? void 0 : _children$props.componentId)) {
    var _themeOptions$themeOp;
    var themeOptionValues = ((_themeOptions$themeOp = _themeOptions.themeOptions[themeOption]) === null || _themeOptions$themeOp === void 0 ? void 0 : _themeOptions$themeOp.values) || [];

    // Add splunk magnetic option to family values if enabled
    if (themeOption === 'family' && hasEnabledSplunkMagneticOption) {
      themeOptionValues = [].concat(_toConsumableArray(themeOptionValues), ['splunk-magnetic']);
    }
    return /*#__PURE__*/_react["default"].createElement(MultiStoryContainer, null, themeOptionValues.map(function (themeOptionValue) {
      var variantThemeProviderProps = handleSplunkMagneticFamily(_objectSpread(_objectSpread({}, themeProviderProps), {}, _defineProperty({}, themeOption, themeOptionValue)));
      return /*#__PURE__*/_react["default"].createElement(_SplunkThemeProvider["default"], _extends({
        key: themeOptionValue
      }, variantThemeProviderProps), /*#__PURE__*/_react["default"].createElement(_themes.GlobalStyles, null), /*#__PURE__*/_react["default"].createElement(StoryWrapper, null, children));
    }));
  }

  // Required to render correctly, otherwise MultiStory isn't considered a valid component
  // eslint-disable-next-line react/jsx-no-useless-fragment
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, children);
};
var WithSplunkTheme = function WithSplunkTheme(Story, context) {
  var _parameters$themeSupp;
  var globals = context.globals,
    parameters = context.parameters;
  var globalsTheme = globals[_constants.PARAM_KEY];
  var themesConfig = parameters[_constants.PARAM_KEY];
  var disableDualThemeDeprecated = parameters.disableDualTheme || false;
  var disableMultiThemeOptions = parameters.disableMultiThemeOptions || false;
  var hasEnabledSplunkMagneticOption = parameters.enableSplunkMagneticOption || false;

  // Determine the effective disable state (new param takes precedence)
  var hasDisabledMultiThemeOptions = disableMultiThemeOptions || disableDualThemeDeprecated;

  // Show deprecation warning if deprecated parameter is used
  if ('disableDualTheme' in parameters) {
    // eslint-disable-next-line no-console
    console.warn('The `disableDualTheme` parameter is deprecated and will be removed in the next major version. Use `disableMultiThemeOptions` instead.');
  }

  // Check for contradictory settings between deprecated and new parameters
  if ('disableDualTheme' in parameters && 'disableMultiThemeOptions' in parameters && Boolean(disableDualThemeDeprecated) !== Boolean(disableMultiThemeOptions)) {
    // eslint-disable-next-line no-console
    console.warn('Contradictory theme parameters detected. The new `disableMultiThemeOptions` parameter will take precedence over the deprecated `disableDualTheme` parameter. Please remove the deprecated parameter as it will be removed in the next major version.');
  }
  if (hasDisabledMultiThemeOptions) {
    // eslint-disable-next-line no-console
    console.warn('This story has multi-theme options disabled, so the `all` option will not show in the themes dropdown.');
  }
  var currentTheme = (0, _util.normalizeThemeOptions)(themesConfig, globalsTheme);
  var supportedFamily = parameters === null || parameters === void 0 ? void 0 : (_parameters$themeSupp = parameters.themeSupport) === null || _parameters$themeSupp === void 0 ? void 0 : _parameters$themeSupp.family;
  var activeMultiThemeKey = null;

  // ideally fromPairs would be fromEntries but our ES2015 target doesn't allow this
  var cleanedThemeProviderProps = (0, _lodash.fromPairs)(Object.entries(currentTheme).filter(function (_ref3) {
    var _ref4 = _slicedToArray(_ref3, 2),
      key = _ref4[0],
      value = _ref4[1];
    // TODO(SUI-8284): Remove deprecated `both` value support in next major version
    var isMultiOptionValue = value === 'all' || value === 'both';
    if ((0, _themeOptions.isThemeOptionKey)(key) && isMultiOptionValue) {
      activeMultiThemeKey = key;

      // TODO(SUI-8284): Remove deprecated `both` value support in next major version
      if (currentTheme[activeMultiThemeKey] === 'both') {
        // eslint-disable-next-line no-console
        console.warn('Theme configuration option `both` is deprecated and will be removed in the next major version. Use `all` instead.');
      }
      return false;
    }
    return true;
  }));
  var rawFamily = currentTheme.family;
  var normalizedFamily = normalizeFamilyForProvider(rawFamily);
  var isSplunkMagneticSelection = rawFamily === 'splunk-magnetic';
  var isMultiFamilySelection = rawFamily === 'all' || rawFamily === 'both';
  var displayedFamily = isSplunkMagneticSelection ? 'splunk-magnetic' : normalizedFamily;

  // Warn whenever the selected family isn't supported (including when Magnetic proxies Prisma).
  var shouldShowFamilyWarning = Boolean(supportedFamily && !isMultiFamilySelection && (isSplunkMagneticSelection ? supportedFamily !== 'splunk-magnetic' : normalizedFamily && supportedFamily !== normalizedFamily));
  var resolvedThemeProviderProps = handleSplunkMagneticFamily(cleanedThemeProviderProps);
  var warningContent = shouldShowFamilyWarning ? /*#__PURE__*/_react["default"].createElement(_SplunkThemeProvider["default"], _extends({}, resolvedThemeProviderProps, {
    // Adjust colorScheme so that it matches with the overall storybook background which is light in 'all'/'light' but dark in 'dark'.
    colorScheme: resolvedThemeProviderProps.colorScheme === 'dark' ? 'dark' : 'light'
  }), /*#__PURE__*/_react["default"].createElement(StyledWarning, null, warningIcon, /*#__PURE__*/_react["default"].createElement(StyledContent, null, "This story does not support the theme family ", /*#__PURE__*/_react["default"].createElement("em", null, displayedFamily), ". The supported theme family is ", /*#__PURE__*/_react["default"].createElement("em", null, supportedFamily), ". Select the supported theme family in the toolbar.")), /*#__PURE__*/_react["default"].createElement("hr", null)) : null;
  if (shouldRenderMultiThemes({
    activeMultiThemeKey: activeMultiThemeKey,
    currentTheme: currentTheme,
    hasDisabledMultiThemeOptions: hasDisabledMultiThemeOptions
  })) {
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, warningContent, /*#__PURE__*/_react["default"].createElement(MultiStory, {
      themeOption: activeMultiThemeKey,
      themeProviderProps: cleanedThemeProviderProps,
      hasEnabledSplunkMagneticOption: hasEnabledSplunkMagneticOption
    }, /*#__PURE__*/_react["default"].createElement(Story, context)));
  }
  return /*#__PURE__*/_react["default"].createElement(_SplunkThemeProvider["default"], _extends({}, resolvedThemeProviderProps, {
    colorScheme: resolvedThemeProviderProps.colorScheme
  }), /*#__PURE__*/_react["default"].createElement(_themes.GlobalStyles, null), warningContent, /*#__PURE__*/_react["default"].createElement(Story, context));
};
var withSplunkTheme = WithSplunkTheme;
exports.withSplunkTheme = withSplunkTheme;
