// @splunk/themes/design-tokens/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * #### Action colors
 *
 * Action colors are used **exclusively** for styling Buttons borders and backgrounds.
 * They define interaction states for primary, secondary, subtle, and destructive buttons.
 *
 * @category color
 */
var actionColors = {
  /**
   * Default background for Primary Buttons.
   */
  actionColorBackgroundPrimary: '',
  /**
   * Active (pressed) background for Primary Buttons.
   */
  actionColorBackgroundPrimaryActive: '',
  /**
   * Disabled background for Primary Buttons.
   */
  actionColorBackgroundPrimaryDisabled: '',
  /**
   * Hover background for Primary Buttons.
   */
  actionColorBackgroundPrimaryHover: '',
  /**
   * Default background for Secondary Buttons.
   */
  actionColorBackgroundSecondary: '',
  /**
   * Active (pressed) background for Secondary Buttons.
   */
  actionColorBackgroundSecondaryActive: '',
  /**
   * Disabled background for Secondary Buttons.
   */
  actionColorBackgroundSecondaryDisabled: '',
  /**
   * Hover background for Secondary Buttons.
   */
  actionColorBackgroundSecondaryHover: '',
  /**
   * Default background for Subtle Buttons.
   */
  actionColorBackgroundSubtle: '',
  /**
   * Active (pressed) background for Subtle Buttons.
   */
  actionColorBackgroundSubtleActive: '',
  /**
   * Disabled background for Subtle Buttons.
   */
  actionColorBackgroundSubtleDisabled: '',
  /**
   * Hover background for Subtle Buttons.
   */
  actionColorBackgroundSubtleHover: '',
  /**
   * Default background for Destructive Buttons.
   */
  actionColorBackgroundDestructive: '',
  /**
   * Active (pressed) background for Destructive Buttons.
   */
  actionColorBackgroundDestructiveActive: '',
  /**
   * Disabled background for Destructive Buttons.
   */
  actionColorBackgroundDestructiveDisabled: '',
  /**
   * Hover background for Destructive Buttons.
   */
  actionColorBackgroundDestructiveHover: '',
  /**
   * Default background for Destructive Secondary Buttons.
   */
  actionColorBackgroundDestructiveSecondary: '',
  /**
   * Active (pressed) background for Destructive Secondary Buttons
   */
  actionColorBackgroundDestructiveSecondaryActive: '',
  /**
   * Disabled background for Destructive Secondary Buttons.
   */
  actionColorBackgroundDestructiveSecondaryDisabled: '',
  /**
   * Hover background for Destructive Secondary Buttons.
   */
  actionColorBackgroundDestructiveSecondaryHover: '',
  /**
   * Default border color for Secondary Buttons.
   */
  actionColorBorderSecondary: '',
  /**
   * Active (pressed) border color for Secondary Buttons.
   */
  actionColorBorderSecondaryActive: '',
  /**
   * Disabled border color for Secondary Buttons.
   */
  actionColorBorderSecondaryDisabled: '',
  /**
   * Hover border color for Secondary Buttons.
   */
  actionColorBorderSecondaryHover: '',
  /**
   * Default border color for Destructive Secondary Buttons.
   */
  actionColorBorderDestructiveSecondary: '',
  /**
   * Active (pressed) border color for Destructive Secondary Buttons.
   */
  actionColorBorderDestructiveSecondaryActive: '',
  /**
   * Disabled border color for Destructive Secondary Buttons.
   */
  actionColorBorderDestructiveSecondaryDisabled: '',
  /**
   * Hover border color for Destructive Secondary Buttons.
   */
  actionColorBorderDestructiveSecondaryHover: '',
  /**
   * Default text color for Primary Buttons.
   *
   * @previewType font
   */
  actionColorContentPrimary: '',
  /**
   * Disabled text color for Primary Buttons.
   *
   * @previewType font
   */
  actionColorContentPrimaryDisabled: '',
  /**
   * Default text color for Secondary Buttons.
   *
   * @previewType font
   */
  actionColorContentSecondary: '',
  /**
   * Disabled text color for Secondary Buttons.
   *
   * @previewType font
   */
  actionColorContentSecondaryDisabled: '',
  /**
   * Active text color for Subtle Buttons.
   *
   * @previewType font
   */
  actionColorContentSubtleActive: '',
  /**
   * Hover text color for Subtle Buttons.
   *
   * @previewType font
   */
  actionColorContentSubtleHover: '',
  /**
   * Default text color for Standalone Buttons.
   *
   * @previewType font
   */
  actionColorContentStandalone: '',
  /**
   * Active (pressed) text color for Standalone Buttons.
   *
   * @previewType font
   */
  actionColorContentStandaloneActive: '',
  /**
   * Disabled text color for Standalone Buttons.
   *
   * @previewType font
   */
  actionColorContentStandaloneDisabled: '',
  /**
   * Hover text color for Standalone Buttons.
   *
   * @previewType font
   */
  actionColorContentStandaloneHover: '',
  /**
   * Default text color for Destructive Buttons.
   *
   * @previewType font
   */
  actionColorContentDestructive: '',
  /**
   * Disabled text color for Destructive Buttons.
   *
   * @previewType font
   */
  actionColorContentDestructiveDisabled: '',
  /**
   * Default text color for Destructive Secondary Buttons.
   *
   * @previewType font
   */
  actionColorContentDestructiveSecondary: '',
  /**
   * Disabled text color for Destructive Secondary Buttons.
   *
   * @previewType font
   */
  actionColorContentDestructiveSecondaryDisabled: ''
};

/**
 * #### Interactive colors
 *
 * Interactive colors should be used **exclusively** for styling Data Entry components: Input, Select, etc.
 *
 * "Overlay" colors are intended to be placed over the default background color, such as `interactiveColorAccent`.
 * If the default background color is not transparent, the `blend` mixin can be used to create a new color that combines the two.
 *
 * @category color
 */
var interactiveColors = {
  /**
   * Default border color for Data entry components.
   */
  interactiveColorBorder: '',
  /**
   * Active (pressed) border color for Data entry components.
   */
  interactiveColorBorderActive: '',
  /**
   * Hover border color for Data entry components.
   */
  interactiveColorBorderHover: '',
  /**
   * Disabled border color for Data entry components.
   */
  interactiveColorBorderDisabled: '',
  /**
   * Background overlay color for selected in Data entry components.
   */
  interactiveColorOverlaySelected: '',
  /**
   * Background overlay color for hover in Data entry components.
   */
  interactiveColorOverlayHover: '',
  /**
   * Background overlay color for active (pressed) in Data entry components.
   */
  interactiveColorOverlayActive: '',
  /**
   * Background overlay color for drag in Data entry components.
   */
  interactiveColorOverlayDrag: '',
  /**
   * Default background color for Data entry components.
   */
  interactiveColorBackground: '',
  /**
   * Background color for disabled Data entry components.
   */
  interactiveColorBackgroundDisabled: '',
  /**
   * Brand accent color for Data entry components.
   */
  interactiveColorAccent: '',
  /**
   * Error color for Data entry components.
   */
  interactiveColorAccentError: '',
  /**
   * Stronger error color for emphasis in Data Entry components. Used for hover interaction.
   */
  interactiveColorAccentErrorStrong: '',
  /**
   * Weaker error color for emphasis in Data Entry components. Used for hover interaction.
   */
  interactiveColorAccentErrorWeak: ''
};

/**
 * #### Neutral colors
 *
 * Neutral colors should be used sparingly. Prefer to use more semantic tokens for backgrounds and borders.
 * These colors must not be used for content as they will not pass the required contrast ratio (instead, use [Content colors](#Content Colors)).
 *
 * @category color
 */
var neutralColors = {
  /**
   * Alternate background color in dark color schemes and inverted background in light color schemes.
   */
  black: '',
  /**
   * Least visually emphasized. Do not use this as the background for Data Entry components, does not pass 3:1 graphical contrast.
   */
  neutral50: '',
  /**
   * Do not use this as the background for Data Entry components, does not pass 3:1 graphical contrast.
   */
  neutral100: '',
  /**
   * Not for text, does not pass 3:1 graphical contrast in light & dark color schemes.
   */
  neutral200: '',
  /**
   * Not for text, does not pass 3:1 graphical contrast in light & dark color schemes.
   */
  neutral300: '',
  /**
   * Not for text, does not pass 3:1 graphical contrast in light color schemes.
   */
  neutral400: '',
  /**
   * Most visually emphasized, not for text, does not pass 3:1 graphical contrast in light color schemes.
   */
  neutral500: '',
  /**
   * Alternate background color in light color schemes and inverted background in dark color schemes.
   */
  white: '',
  /**
   * Transparent background color.
   */
  transparent: ''
};

/**
 * #### Background colors
 * Use background colors for page backgrounds and containers such as cards, modals, panels or popups.
 *
 * @category color
 */
var backgroundColors = {
  /**
   * Default background color for the navigation headers.
   */
  backgroundColorNavigation: '',
  /**
   * Primary background color for pages.
   */
  backgroundColorPage: '',
  /**
   * Background color for side panels and navigation panels.
   */
  backgroundColorSidebar: '',
  /**
   * Used for cards or to create visual separation in the UI.
   */
  backgroundColorSection: '',
  /**
   * Default background color for modals.
   */
  backgroundColorDialog: '',
  /**
   * Default background color for popovers and tooltips.
   */
  backgroundColorPopup: '',
  /**
   * Background color for inverted color schemes.
   */
  backgroundColorFloating: '',
  /**
   * Used for page overlays under modals or when masking UI.
   */
  backgroundColorScrim: ''
};

/**
 * #### Content colors
 * Use content colors to create visual hierarchy, convey status or importance for plain text content.
 *
 * @category color
 * @previewType font
 */
var contentColors = {
  /**
   * Text color for links.
   */
  contentColorLink: '',
  /**
   * Text color for visited links.
   */
  contentColorLinkVisited: '',
  /**
   * Text color for disabled links.
   */
  contentColorLinkDisabled: '',
  /**
   * Text color for visited links in a disabled state.
   */
  contentColorLinkVisitedDisabled: '',
  /**
   * Brand accent text color.
   */
  contentColorAccent: '',
  /**
   * Brighter brand accent text color for emphasis.
   */
  contentColorAccentStrong: '',
  /**
   * Brand accent text color that does not pass 4.5:1 contrast ratio.
   */
  contentColorAccentWeak: '',
  /**
   * Text color for headings, titles, and prominent text content.
   */
  contentColorActive: '',
  /**
   * Text color for paragraphs and standard content.
   */
  contentColorDefault: '',
  /**
   * Text color for disabled content that does not pass 4.5:1 contrast.
   */
  contentColorDisabled: '',
  /**
   * Text color used on top of inverted backgrounds.
   */
  contentColorInverted: '',
  /**
   * Subdued text color for less prominent content (use sparingly).
   */
  contentColorMuted: '',
  /**
   * Text color for error or negative status messaging.
   */
  contentColorNegative: '',
  /**
   * Disabled negative text color that does not pass 4.5:1 contrast.
   */
  contentColorNegativeWeak: ''
};

/**
 * #### UI Colors
 *
 * @category color
 */
var uiColors = {
  /**
   * Visual focus ring color used for focus box-shadow. Should not be directly applied to UI elements.
   */
  focusColor: ''
};

/**
 * #### Notification colors
 *
 * Use notifications colors to communicate messages and guidance directly to users.
 * For example in Message Bars to set tone and urgency in a conversational context.
 *
 * Weak variants do not meet graphical or text contrast and should only be used decoratively.
 *
 * @category color
 */
var notificationColors = {
  notificationColorInfoWeak: '',
  notificationColorInfo: '',
  notificationColorInfoStrong: '',
  notificationColorPositiveWeak: '',
  notificationColorPositive: '',
  notificationColorPositiveStrong: '',
  notificationColorCautionWeak: '',
  notificationColorCaution: '',
  notificationColorCautionStrong: '',
  notificationColorNegativeWeak: '',
  notificationColorNegative: '',
  notificationColorNegativeStrong: ''
};

/**
 * #### Severity colors
 *
 * Use severity colors for icons, indicators, and visual signals to help users quickly scan for criticality, risk, or state without relying only on text.
 * Severity colors should be used to aid identifying the semantic meaning of an element; not used to communicate tone.
 *
 * @category color
 */
var severityColors = {
  severityColorCustom: '',
  severityColorUnknown: '',
  severityColorInfo: '',
  severityColorNormal: '',
  severityColorNotice: '',
  severityColorWarning: '',
  severityColorAlert: '',
  severityColorCritical: '',
  severityColorEmergency: ''
};

/**
 * #### Syntax colors
 *
 * These colors should be used for syntax highlighting in programming languages.
 * The colors chosen for tokens depend on the language, its features, and aesthetics.
 * Prefer to use the `Code` component rather than these tokens directly as it does this mapping automatically.
 * See [Content Colors](#Content Colors)
 *
 * @category color
 * @previewType font
 */
var syntaxColors = {
  syntaxBlue: '',
  syntaxBrown: '',
  syntaxGray: '',
  syntaxGreen: '',
  syntaxOrange: '',
  syntaxPink: '',
  syntaxPurple: '',
  syntaxRed: '',
  syntaxTeal: ''
};

/**
 * #### Border colors
 *
 * Border color should be used to outline or divide content and UI elements.
 *
 * @category color
 */
var borderColors = {
  /**
   * Default border color for elements that meets 3:1 graphical contrast (e.g., menu dividers).
   */
  borderColor: '',
  /**
   * Decorative-only border color that does not meet 3:1 graphical contrast.
   */
  borderColorWeak: '',
  /**
   * Border color with stronger visual emphasis that meets 3:1 graphical contrast.
   */
  borderColorStrong: ''
};
var colors = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, actionColors), interactiveColors), neutralColors), backgroundColors), contentColors), uiColors), notificationColors), severityColors), syntaxColors), borderColors);
var _default = colors;
exports["default"] = _default;
