// @splunk/themes/design-tokens/dataViz.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * #### Static colors
 *
 * @category color
 */
var staticColors = {
  static1: '',
  static2: '',
  static3: '',
  static4: '',
  static5: '',
  static6: '',
  static7: '',
  static8: '',
  static9: '',
  static10: '',
  static11: '',
  static12: '',
  static13: '',
  static14: '',
  static15: '',
  static16: '',
  static17: '',
  static18: '',
  static19: '',
  static20: ''
};

/**
 * #### Categorical 1D
 *
 * @category color
 */
var categorical1D = {
  categorical1D1: '',
  categorical1D2: '',
  categorical1D3: '',
  categorical1D4: '',
  categorical1D5: '',
  categorical1D6: '',
  categorical1D7: ''
};

/**
 * #### Categorical 1L
 *
 * @category color
 */
var categorical1L = {
  categorical1L1: '',
  categorical1L2: '',
  categorical1L3: '',
  categorical1L4: '',
  categorical1L5: '',
  categorical1L6: '',
  categorical1L7: ''
};

/**
 * #### Categorical 2D
 *
 * @category color
 */
var categorical2D = {
  categorical2D1: '',
  categorical2D2: '',
  categorical2D3: '',
  categorical2D4: '',
  categorical2D5: '',
  categorical2D6: '',
  categorical2D7: ''
};

/**
 * #### Categorical 2L
 *
 * @category color
 */
var categorical2L = {
  categorical2L1: '',
  categorical2L2: '',
  categorical2L3: '',
  categorical2L4: '',
  categorical2L5: '',
  categorical2L6: '',
  categorical2L7: ''
};
var categorical = _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, categorical1D), categorical1L), categorical2D), categorical2L);

/**
 * #### Divergent 1D
 *
 * @category color
 */
var divergent1D = {
  divergent1D1: '',
  divergent1D2: '',
  divergent1D3: '',
  divergent1D4: '',
  divergent1D5: '',
  divergent1D6: '',
  divergent1D7: ''
};

/**
 * #### Divergent 1L
 *
 * @category color
 */
var divergent1L = {
  divergent1L1: '',
  divergent1L2: '',
  divergent1L3: '',
  divergent1L4: '',
  divergent1L5: '',
  divergent1L6: '',
  divergent1L7: ''
};

/**
 * #### Divergent 2D
 *
 * @category color
 */
var divergent2D = {
  divergent2D1: '',
  divergent2D2: '',
  divergent2D3: '',
  divergent2D4: '',
  divergent2D5: '',
  divergent2D6: '',
  divergent2D7: ''
};

/**
 * #### Divergent 2L
 *
 * @category color
 */
var divergent2L = {
  divergent2L1: '',
  divergent2L2: '',
  divergent2L3: '',
  divergent2L4: '',
  divergent2L5: '',
  divergent2L6: '',
  divergent2L7: ''
};

/**
 * #### Divergent 3D
 *
 * @category color
 */
var divergent3D = {
  divergent3D1: '',
  divergent3D2: '',
  divergent3D3: '',
  divergent3D4: '',
  divergent3D5: '',
  divergent3D6: '',
  divergent3D7: ''
};

/**
 * #### Divergent 3L
 *
 * @category color
 */
var divergent3L = {
  divergent3L1: '',
  divergent3L2: '',
  divergent3L3: '',
  divergent3L4: '',
  divergent3L5: '',
  divergent3L6: '',
  divergent3L7: ''
};

/**
 * #### Divergent 4D
 *
 * @category color
 */
var divergent4D = {
  divergent4D1: '',
  divergent4D2: '',
  divergent4D3: '',
  divergent4D4: '',
  divergent4D5: '',
  divergent4D6: '',
  divergent4D7: ''
};

/**
 * #### Divergent 4L
 *
 * @category color
 */
var divergent4L = {
  divergent4L1: '',
  divergent4L2: '',
  divergent4L3: '',
  divergent4L4: '',
  divergent4L5: '',
  divergent4L6: '',
  divergent4L7: ''
};
var divergent = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, divergent1D), divergent1L), divergent2D), divergent2L), divergent3D), divergent3L), divergent4D), divergent4L);

/**
 * #### Sequential 1D
 *
 * @category color
 */
var sequential1D = {
  sequential1D1: '',
  sequential1D2: '',
  sequential1D3: '',
  sequential1D4: '',
  sequential1D5: '',
  sequential1D6: '',
  sequential1D7: ''
};

/**
 * #### Sequential 1L
 *
 * @category color
 */
var sequential1L = {
  sequential1L1: '',
  sequential1L2: '',
  sequential1L3: '',
  sequential1L4: '',
  sequential1L5: '',
  sequential1L6: '',
  sequential1L7: ''
};

/**
 * #### Sequential 2D
 *
 * @category color
 */
var sequential2D = {
  sequential2D1: '',
  sequential2D2: '',
  sequential2D3: '',
  sequential2D4: '',
  sequential2D5: '',
  sequential2D6: '',
  sequential2D7: ''
};

/**
 * #### Sequential 2L
 *
 * @category color
 */
var sequential2L = {
  sequential2L1: '',
  sequential2L2: '',
  sequential2L3: '',
  sequential2L4: '',
  sequential2L5: '',
  sequential2L6: '',
  sequential2L7: ''
};

/**
 * #### Sequential 3D
 *
 * @category color
 */
var sequential3D = {
  sequential3D1: '',
  sequential3D2: '',
  sequential3D3: '',
  sequential3D4: '',
  sequential3D5: '',
  sequential3D6: '',
  sequential3D7: ''
};

/**
 * #### Sequential 3L
 *
 * @category color
 */
var sequential3L = {
  sequential3L1: '',
  sequential3L2: '',
  sequential3L3: '',
  sequential3L4: '',
  sequential3L5: '',
  sequential3L6: '',
  sequential3L7: ''
};

/**
 * #### Sequential 4D
 *
 * @category color
 */
var sequential4D = {
  sequential4D1: '',
  sequential4D2: '',
  sequential4D3: '',
  sequential4D4: '',
  sequential4D5: '',
  sequential4D6: '',
  sequential4D7: ''
};

/**
 * #### Sequential 4L
 *
 * @category color
 */
var sequential4L = {
  sequential4L1: '',
  sequential4L2: '',
  sequential4L3: '',
  sequential4L4: '',
  sequential4L5: '',
  sequential4L6: '',
  sequential4L7: ''
};

/**
 * #### Sequential 5D
 *
 * @category color
 */
var sequential5D = {
  sequential5D1: '',
  sequential5D2: '',
  sequential5D3: '',
  sequential5D4: '',
  sequential5D5: '',
  sequential5D6: '',
  sequential5D7: ''
};

/**
 * #### Sequential 5L
 *
 * @category color
 */
var sequential5L = {
  sequential5L1: '',
  sequential5L2: '',
  sequential5L3: '',
  sequential5L4: '',
  sequential5L5: '',
  sequential5L6: '',
  sequential5L7: ''
};

/**
 * #### Sequential 6D
 *
 * @category color
 */
var sequential6D = {
  sequential6D1: '',
  sequential6D2: '',
  sequential6D3: '',
  sequential6D4: '',
  sequential6D5: '',
  sequential6D6: '',
  sequential6D7: ''
};

/**
 * #### Sequential 6L
 *
 * @category color
 */
var sequential6L = {
  sequential6L1: '',
  sequential6L2: '',
  sequential6L3: '',
  sequential6L4: '',
  sequential6L5: '',
  sequential6L6: '',
  sequential6L7: ''
};
var sequential = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, sequential1D), sequential1L), sequential2D), sequential2L), sequential3D), sequential3L), sequential4D), sequential4L), sequential5D), sequential5L), sequential6D), sequential6L);

/**
 * #### High low 1D
 *
 * @category color
 */
var highLow1D = {
  highLow1DHigh: '',
  highLow1DLow: ''
};

/**
 * #### High low 1L
 *
 * @category color
 */
var highLow1L = {
  highLow1LHigh: '',
  highLow1LLow: ''
};

/**
 * #### High low 2D
 *
 * @category color
 */
var highLow2D = {
  highLow2DHigh: '',
  highLow2DLow: ''
};

/**
 * #### High low 2L
 *
 * @category color
 */
var highLow2L = {
  highLow2LHigh: '',
  highLow2LLow: ''
};

/**
 * #### High low 3D
 *
 * @category color
 */
var highLow3D = {
  highLow3DHigh: '',
  highLow3DLow: ''
};

/**
 * #### High low 3L
 *
 * @category color
 */
var highLow3L = {
  highLow3LHigh: '',
  highLow3LLow: ''
};

/**
 * #### High low 4D
 *
 * @category color
 */
var highLow4D = {
  highLow4DHigh: '',
  highLow4DLow: ''
};

/**
 * #### High low 4L
 *
 * @category color
 */
var highLow4L = {
  highLow4LHigh: '',
  highLow4LLow: ''
};
var highLow = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, highLow1D), highLow1L), highLow2D), highLow2L), highLow3D), highLow3L), highLow4D), highLow4L);
var dataViz = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, staticColors), categorical), divergent), sequential), highLow);
var _default = dataViz;
exports["default"] = _default;
