// @splunk/themes/design-tokens/deprecated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/*
@splunk/eslint-plugin-splunk-ui/design-token-deprecatedrule auto-fixes tokens
where the deprecation message is exactly "Use `replacementToken`".
*/

/**
 * #### Deprecated tokens
 *
 * @category color
 * @previewType none
 */
var deprecated = {
  /** @deprecated */
  contentColorInfo: '',
  /** @deprecated */
  contentColorWarning: '',
  /** @deprecated */
  contentColorPositive: '',
  /** @deprecated */
  contentColorNegativeStrong: '',
  /**
   * @deprecated  Use a more specific token based on the semantic use case: [`interactiveColorAccent`](#interactiveColorAccent), [`contentColorAccent`](#contentColorAccent), or [`focusColor`](#focusColor).
   */
  interactiveColorPrimary: '',
  /** @deprecated Use `notificationColorPositive` */
  accentColorPositive: '',
  /** @deprecated Use `notificationColorCaution` */
  accentColorWarning: '',
  /** @deprecated Use `notificationColorCaution` */
  accentColorAlert: '',
  /** @deprecated Use `notificationColorNegative` */
  accentColorNegative: '',
  /**
   * @deprecated Use `contentColorDefault`
   */
  textColor: '',
  /**
   * @deprecated Use `contentColorMuted`
   */
  textGray: '',
  /**
   * @deprecated Use `contentColorLink`
   */
  linkColor: '',
  /**
   * @deprecated `Link` hover interaction should use `text-decoration: underline`, color should not change
   */
  linkColorHover: '',
  /**
   * @deprecated Use `borderColorWeak`
   */
  borderLightColor: '',
  /**
   * @deprecated Use `contentColorDisabled`
   */
  textDisabledColor: '',
  /**
   * @deprecated Use `spacingXSmall`
   * @category dimension
   * @previewType spacing
   */
  spacingQuarter: '',
  /**
   * @deprecated Use `spacingSmall`
   * @category dimension
   * @previewType spacing
   */
  spacingHalf: '',
  /**
   * @deprecated Use `spacingLarge`
   * @category dimension
   * @previewType spacing
   */
  spacing: '',
  /**
   * @deprecated Use `backgroundColorPage`
   */
  backgroundColor: '',
  /**
   * @category border
   * @deprecated Set border using CSS and [borderColor](#borderColor).
   */
  border: '',
  /**
   * @category border
   * @deprecated Set border using CSS and [borderColorStrong](#borderColorStrong).
   */
  borderDark: '',
  /**
   * @category border
   * @deprecated Set border using CSS and [borderColorWeak](#borderColorWeak).
   */
  borderLight: '',
  /**
   * @deprecated Use `interactiveColorBorderActive` for Data Entry components or `actionColorBorderSecondaryActive` for Buttons.
   */
  borderActiveColor: '',
  /**
   * @deprecated Use `interactiveColorOverlayHover` for Data Entry components or `actionColorBackgroundSecondaryHover` for Buttons.
   */
  backgroundColorHover: '',
  /**
   * @category shadow
   * @deprecated Don't use hover shadows - see notes on backgroundColorHover for hover affordances.
   */
  hoverShadow: '',
  /**
   * @deprecated Use `notificationColorInfo` or `severityColorInfo`
   */
  statusColorInfo: '',
  /**
   * @deprecated Use `notificationColorInfoWeak`
   */
  statusColorInfoWeak: '',
  /**
   * @deprecated Use `notificationColorInfoStrong`
   */
  statusColorInfoStrong: '',
  /**
   * @deprecated Use `notificationColorPositive` or `severityColorNormal`
   */
  statusColorNormal: '',
  /**
   * @deprecated Use `notificationColorPositiveWeak`
   */
  statusColorNormalWeak: '',
  /**
   * @deprecated Use `notificationColorPositiveStrong`
   */
  statusColorNormalStrong: '',
  /**
   * @deprecated Use `notificationColorCaution` or `severityColorNotice`
   */
  statusColorLow: '',
  /**
   * @deprecated Use `notificationColorCautionWeak`
   */
  statusColorLowWeak: '',
  /**
   * @deprecated Use `notificationColorCautionStrong`
   */
  statusColorLowStrong: '',
  /**
   * @deprecated Use `severityColorWarning`
   */
  statusColorMedium: '',
  /** @deprecated */
  statusColorMediumWeak: '',
  /** @deprecated */
  statusColorMediumStrong: '',
  /**
   * @deprecated Use `notificationColorNegative` or `severityColorAlert`
   */
  statusColorHigh: '',
  /**
   * @deprecated Use `notificationColorNegativeWeak`
   */
  statusColorHighWeak: '',
  /**
   * @deprecated Use `notificationColorNegativeStrong`
   */
  statusColorHighStrong: '',
  /**
   * @deprecated Default color for critical alerts or urgent messages.
   */
  statusColorCritical: '',
  /**
   * @deprecated Weaker, less prominent critical color, does not pass 3:1 graphical or 4.5:1 text contrast.
   */
  statusColorCriticalWeak: '',
  /**
   * @deprecated Brighter, more prominent critical color.
   */
  statusColorCriticalStrong: '',
  /** @deprecated */
  brandColorL50: '',
  /** @deprecated */
  brandColorL40: '',
  /** @deprecated */
  brandColorL30: '',
  /** @deprecated */
  brandColorL20: '',
  /** @deprecated */
  brandColorL10: '',
  /** @deprecated */
  brandColor: '',
  /** @deprecated */
  brandColorD10: '',
  /** @deprecated */
  brandColorD20: '',
  /** @deprecated */
  brandColorD30: '',
  /** @deprecated */
  brandColorD40: '',
  /** @deprecated */
  brandColorD50: '',
  /** @deprecated */
  gray98: '',
  /** @deprecated */
  gray96: '',
  /** @deprecated */
  gray92: '',
  /** @deprecated */
  gray80: '',
  /** @deprecated */
  gray60: '',
  /** @deprecated */
  gray45: '',
  /** @deprecated */
  gray30: '',
  /** @deprecated */
  gray25: '',
  /** @deprecated */
  gray22: '',
  /** @deprecated */
  gray20: '',
  /** @deprecated */
  accentColorL50: '',
  /** @deprecated */
  accentColorL40: '',
  /** @deprecated */
  accentColorL30: '',
  /** @deprecated */
  accentColorL20: '',
  /** @deprecated */
  accentColorL10: '',
  /** @deprecated */
  accentColor: '',
  /** @deprecated */
  accentColorD10: '',
  /** @deprecated */
  accentColorD20: '',
  /** @deprecated */
  accentColorD30: '',
  /** @deprecated */
  accentColorD40: '',
  /** @deprecated */
  accentColorD50: '',
  /** @deprecated */
  errorColorL50: '',
  /** @deprecated */
  errorColorL40: '',
  /** @deprecated */
  errorColorL30: '',
  /** @deprecated */
  errorColorL20: '',
  /** @deprecated */
  errorColorL10: '',
  /**
   * @deprecated Use `notificationColorNegative` or `severityColorAlert`
   */
  errorColor: '',
  /** @deprecated */
  errorColorD10: '',
  /** @deprecated */
  errorColorD20: '',
  /** @deprecated */
  errorColorD30: '',
  /** @deprecated */
  errorColorD40: '',
  /** @deprecated */
  errorColorD50: '',
  /** @deprecated */
  alertColorL50: '',
  /** @deprecated */
  alertColorL40: '',
  /** @deprecated */
  alertColorL30: '',
  /** @deprecated */
  alertColorL20: '',
  /** @deprecated */
  alertColorL10: '',
  /** @deprecated */
  alertColor: '',
  /** @deprecated */
  alertColorD10: '',
  /** @deprecated */
  alertColorD20: '',
  /** @deprecated */
  alertColorD30: '',
  /** @deprecated */
  alertColorD40: '',
  /** @deprecated */
  alertColorD50: '',
  /** @deprecated */
  warningColorL50: '',
  /** @deprecated */
  warningColorL40: '',
  /** @deprecated */
  warningColorL30: '',
  /** @deprecated */
  warningColorL20: '',
  /** @deprecated */
  warningColorL10: '',
  /**
   * @deprecated Use `notificationColorCaution` or `severityColorNotice`
   */
  warningColor: '',
  /** @deprecated */
  warningColorD10: '',
  /** @deprecated */
  warningColorD20: '',
  /** @deprecated */
  warningColorD30: '',
  /** @deprecated */
  warningColorD40: '',
  /** @deprecated */
  warningColorD50: '',
  /** @deprecated */
  successColorL50: '',
  /** @deprecated */
  successColorL40: '',
  /** @deprecated */
  successColorL30: '',
  /** @deprecated */
  successColorL20: '',
  /** @deprecated */
  successColorL10: '',
  /**
   * @deprecated Use `notificationColorPositive` or `severityColorNormal`
   */
  successColor: '',
  /** @deprecated */
  successColorD10: '',
  /** @deprecated */
  successColorD20: '',
  /** @deprecated */
  successColorD30: '',
  /** @deprecated */
  successColorD40: '',
  /** @deprecated */
  successColorD50: '',
  /** @deprecated */
  infoColorL50: '',
  /** @deprecated */
  infoColorL40: '',
  /** @deprecated */
  infoColorL30: '',
  /** @deprecated */
  infoColorL20: '',
  /** @deprecated */
  infoColorL10: '',
  /** @deprecated */
  infoColor: '',
  /** @deprecated */
  infoColorD10: '',
  /** @deprecated */
  infoColorD20: '',
  /** @deprecated */
  infoColorD30: '',
  /** @deprecated */
  infoColorD40: '',
  /** @deprecated */
  infoColorD50: '',
  /** @deprecated */
  diverging1ColorA: '',
  /** @deprecated */
  diverging1ColorB: '',
  /** @deprecated */
  diverging2ColorA: '',
  /** @deprecated */
  diverging2ColorB: '',
  /** @deprecated */
  diverging3ColorA: '',
  /** @deprecated */
  diverging3ColorB: '',
  /** @deprecated */
  diverging4ColorA: '',
  /** @deprecated */
  diverging4ColorB: '',
  /** @deprecated */
  diverging5ColorA: '',
  /** @deprecated */
  diverging5ColorB: '',
  /** @deprecated */
  cat1Color: '',
  /** @deprecated */
  cat1ColorL: '',
  /** @deprecated */
  cat2Color: '',
  /** @deprecated */
  cat2ColorL: '',
  /** @deprecated */
  cat3Color: '',
  /** @deprecated */
  cat3ColorL: '',
  /** @deprecated */
  cat4Color: '',
  /** @deprecated */
  cat4ColorL: '',
  /** @deprecated */
  cat5Color: '',
  /** @deprecated */
  cat5ColorL: '',
  /** @deprecated */
  cat6Color: '',
  /** @deprecated */
  cat6ColorL: '',
  /** @deprecated */
  cat7Color: '',
  /** @deprecated */
  cat7ColorL: '',
  /** @deprecated */
  cat8Color: '',
  /** @deprecated */
  cat8ColorL: '',
  /** @deprecated */
  cat9Color: '',
  /** @deprecated */
  cat9ColorL: '',
  /** @deprecated */
  cat10Color: '',
  /** @deprecated */
  cat10ColorL: '',
  /** @deprecated */
  cat11Color: '',
  /** @deprecated */
  cat11ColorL: '',
  /** @deprecated */
  cat12Color: '',
  /** @deprecated */
  cat12ColorL: '',
  /** @deprecated */
  cat13Color: '',
  /** @deprecated */
  cat13ColorL: '',
  /** @deprecated */
  cat14Color: '',
  /** @deprecated */
  cat14ColorL: '',
  /** @deprecated */
  cat15Color: '',
  /** @deprecated */
  cat15ColorL: '',
  /** @deprecated */
  cat16Color: '',
  /** @deprecated */
  cat16ColorL: '',
  /** @deprecated */
  cat17Color: '',
  /** @deprecated */
  cat17ColorL: '',
  /** @deprecated */
  cat18Color: '',
  /** @deprecated */
  cat18ColorL: '',
  /** @deprecated */
  cat19Color: '',
  /** @deprecated */
  cat19ColorL: '',
  /** @deprecated */
  cat20Color: '',
  /** @deprecated */
  cat20ColorL: '',
  /** @deprecated */
  cat21Color: '',
  /** @deprecated */
  cat21ColorL: '',
  /** @deprecated */
  cat22Color: '',
  /** @deprecated */
  cat22ColorL: '',
  /** @deprecated */
  cat23Color: '',
  /** @deprecated */
  cat23ColorL: '',
  /** @deprecated */
  cat24Color: '',
  /** @deprecated */
  cat24ColorL: '',
  /** @deprecated */
  cat25Color: '',
  /** @deprecated */
  cat25ColorL: '',
  /** @deprecated */
  cat26Color: '',
  /** @deprecated */
  cat26ColorL: '',
  /** @deprecated */
  cat27Color: '',
  /** @deprecated */
  cat27ColorL: '',
  /** @deprecated */
  cat28Color: '',
  /** @deprecated */
  cat28ColorL: '',
  /** @deprecated */
  cat29Color: '',
  /** @deprecated */
  cat29ColorL: '',
  /** @deprecated */
  cat30Color: '',
  /** @deprecated */
  cat30ColorL: '',
  /** @deprecated */
  syntaxBlueLight: '',
  /** @deprecated */
  syntaxGreenLight: '',
  /** @deprecated */
  syntaxPurpleLight: '',
  /** @deprecated */
  syntaxRedLight: '',
  /**
   * @deprecated
   * @category gradient
   */
  draggableBackground: ''
};
var _default = _objectSpread({}, deprecated);
exports["default"] = _default;
