// @splunk/themes/design-tokens/typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * #### Font family
 *
 * @category fontFamily
 */
var fontFamily = {
  fontFamily: '',
  /**
   * Used for Headings and Typography title variants
   */
  fontFamilyTitle: '',
  sansFontFamily: '',
  serifFontFamily: '',
  monoFontFamily: ''
};

/**
 * #### Font weight
 *
 * @category fontWeight
 */
var fontWeight = {
  fontWeightLight: 300,
  fontWeightNormal: 400,
  fontWeightSemiBold: 500,
  fontWeightBold: 700,
  fontWeightHeavy: 800,
  fontWeightExtraBold: 900
};

/**
 * #### Font size
 *
 * @category dimension
 * @previewType font
 */
var fontSize = {
  fontSizeSmall: '',
  fontSize: '',
  fontSizeLarge: '',
  fontSizeXLarge: '',
  fontSizeXXLarge: ''
};

/**
 * #### Font measures
 *
 * @category number
 */
var fontMeasures = {
  /**
   * Use to improve readability and [accessibility](https://www.w3.org/WAI/WCAG21/Understanding/visual-presentation.html)
   * in blocks of text by preventing overly long lines.
   *
   * @category dimension
   * @previewType none
   */
  lineLength: '',
  /**
   * Minimum line length for optimal readability.
   * @category dimension
   * @previewType none
   */
  lineLengthMinimum: '',
  /**
   * Default line height.
   */
  lineHeight: '',
  /**
   * Very tight. Use where space is extremely limited and text is not expected to wrap.
   * Ideal for buttons, badges, input fields, and compact labels.
   */
  lineHeightSingle: '',
  /**
   * Slightly more space than single, still compact
   */
  lineHeightTight: '',
  /**
   * Compact but a bit more readable
   */
  lineHeightSnug: '',
  /**
   * Balanced spacing for smaller text
   */
  lineHeightComfortable: '',
  /**
   * Slightly looser for better readability
   */
  lineHeightRelaxed: '',
  /*
   * Ideal for larger text like headings
   */
  lineHeightSpacious: '',
  /**
   * Default for body text; comfortable, standard line height same as `lineHeight`
   */
  lineHeightNormal: ''
};
var typography = _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, fontFamily), fontWeight), fontSize), fontMeasures);
var _default = typography;
exports["default"] = _default;
