// @splunk/themes/mixins/typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typographyVariants = exports["default"] = void 0;
var _merge = _interopRequireDefault(require("lodash/merge"));
var _styledComponents = require("styled-components");
var _resets = require("./resets");
var _variables = _interopRequireDefault(require("../variables"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n        ", "\n        color: ", ";\n        font-family: ", ";\n        font-size: ", ";\n        font-weight: ", ";\n        line-height: ", ";\n        ", "\n    "]);
  _templateObject8 = function _templateObject8() {
    return data;
  };
  return data;
}
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject7 = function _templateObject7() {
    return data;
  };
  return data;
}
function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject6 = function _templateObject6() {
    return data;
  };
  return data;
}
function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject5 = function _templateObject5() {
    return data;
  };
  return data;
}
function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject4 = function _templateObject4() {
    return data;
  };
  return data;
}
function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject3 = function _templateObject3() {
    return data;
  };
  return data;
}
function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n                text-rendering: optimizelegibility;\n            "]);
  _templateObject2 = function _templateObject2() {
    return data;
  };
  return data;
}
function _templateObject() {
  var data = _taggedTemplateLiteral([""]);
  _templateObject = function _templateObject() {
    return data;
  };
  return data;
}
function _taggedTemplateLiteral(e, t) { return t || (t = e.slice(0)), Object.freeze(Object.defineProperties(e, { raw: { value: Object.freeze(t) } })); }
var typographyVariants = ['body', 'title1', 'title2', 'title3', 'title4', 'title5', 'title6', 'largeBody', 'smallBody', 'monoBody', 'monoSmallBody'];

// Keep in sync with react-ui/Typography for documentation
// TODO: SUI-5042 Improve docs-gen to render extended 3rd party types
// TODO: SUI-5043 UtilPage doesn't render public types
exports.typographyVariants = typographyVariants;
function isTypographyVariant(s) {
  if (typeof s !== 'string') {
    return false;
  }
  return typographyVariants.includes(s);
}
function getParamsForVariant(variant) {
  var family = 'sansSerif';
  var color = 'default';
  var size = 14;
  var lineHeight = 'spacious';
  var weight = 'normal';
  var additionalStyles = (0, _styledComponents.css)(_templateObject());
  switch (variant) {
    case 'title1':
      family = 'title';
      color = 'active';
      size = 24;
      lineHeight = 'spacious';
      weight = 'bold';
      additionalStyles = (0, _styledComponents.css)(_templateObject2());
      break;
    case 'title2':
      family = 'title';
      color = 'active';
      size = 20;
      lineHeight = 'tight';
      weight = 'bold';
      additionalStyles = (0, _styledComponents.css)(_templateObject3());
      break;
    case 'title3':
      family = 'title';
      color = 'active';
      size = 16;
      lineHeight = 'snug';
      weight = 'bold';
      additionalStyles = (0, _styledComponents.css)(_templateObject4());
      break;
    case 'title4':
      family = 'title';
      color = 'active';
      size = 14;
      lineHeight = 'spacious';
      weight = 'bold';
      additionalStyles = (0, _styledComponents.css)(_templateObject5());
      break;
    case 'title5':
      family = 'title';
      color = 'default';
      size = 14;
      lineHeight = 'spacious';
      weight = 'bold';
      additionalStyles = (0, _styledComponents.css)(_templateObject6());
      break;
    case 'title6':
      family = 'title';
      color = 'default';
      size = 14;
      lineHeight = 'spacious';
      weight = 'semiBold';
      additionalStyles = (0, _styledComponents.css)(_templateObject7());
      break;
    case 'largeBody':
      color = 'default';
      size = 16;
      lineHeight = 'snug';
      weight = 'normal';
      break;
    case 'smallBody':
      color = 'default';
      size = 12;
      lineHeight = 'comfortable';
      weight = 'normal';
      break;
    case 'monoBody':
      family = 'monospace';
      break;
    case 'monoSmallBody':
      family = 'monospace';
      color = 'default';
      size = 12;
      lineHeight = 'comfortable';
      weight = 'normal';
      break;
    case 'body':
      // Theme defaults set the 'body' style
      break;
    default:
      {
        if (false) {
          // eslint-disable-next-line no-console
          console.log("Unknown typography variant, ".concat(variant, ". Using default values for theme."));
        }
        // Make sure this "never" happens https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking
        var exhaustiveCheck = variant;
        return exhaustiveCheck;
      }
  }
  return {
    family: family,
    size: size,
    color: color,
    lineHeight: lineHeight,
    weight: weight,
    additionalStyles: additionalStyles
  };
}
var familyPropToVariableMap = {
  sansSerif: _variables["default"].sansFontFamily,
  monospace: _variables["default"].monoFontFamily,
  title: _variables["default"].fontFamilyTitle
};
var colorPropToVariableMap = {
  active: _variables["default"].contentColorActive,
  "default": _variables["default"].contentColorDefault,
  disabled: _variables["default"].contentColorDisabled,
  inverted: _variables["default"].contentColorInverted,
  muted: _variables["default"].contentColorMuted,
  inherit: 'inherit'
};
var fontsizePixelValueToRemMap = {
  24: _variables["default"].fontSizeXXLarge,
  20: _variables["default"].fontSizeXLarge,
  18: '1.125rem',
  16: _variables["default"].fontSizeLarge,
  14: _variables["default"].fontSize,
  12: _variables["default"].fontSizeSmall
};
var lineHeightVariableToValueMap = {
  single: _variables["default"].lineHeightSingle,
  tight: _variables["default"].lineHeightTight,
  snug: _variables["default"].lineHeightSnug,
  comfortable: _variables["default"].lineHeightComfortable,
  relaxed: _variables["default"].lineHeightRelaxed,
  spacious: _variables["default"].lineHeightSpacious,
  normal: _variables["default"].lineHeightNormal
};

// As defined by [font-weight | MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#common_weight_name_mapping)
var weightPropToValueMap = {
  light: _variables["default"].fontWeightLight,
  normal: _variables["default"].fontWeightNormal,
  semiBold: _variables["default"].fontWeightSemiBold,
  bold: _variables["default"].fontWeightBold,
  extraBold: _variables["default"].fontWeightExtraBold,
  heavy: _variables["default"].fontWeightHeavy
};

/**
 * A mixin for styling text content using predefined typography variants
 * and/or customizing font-settings with system parameters: e.g. size, weight, font-family.
 *
 * The default variant is `body` and will be used if no variant or settings
 * are given: i.e. `typography()` or `typography({})`.
 * Variants have the reset applied by default.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { typography } from '@splunk/themes/mixins';
 *
 *  const MyTitle = styled.h1`
 *      ${typography('title1')};
 *  `;
 *
 *  const MyCustomizedTitle = styled.h1`
 *      ${typography('title1', { color: 'inverted' })};
 *  `;
 *
 *  const CustomTitle = styled.h1`
 *      ${typography({size: 56, weight: 'light', color: 'inverted' })};
 *  `;
 *  ```
 * @name typography
 * @kind function
 * @param {string} [variant] Use a predefined typography variant:
 *  `'body'`,
 *  `'title1'`,
 *  `'title2'`,
 *  `'title3'`,
 *  `'title4'`,
 *  `'title5'`,
 *  `'title6'`,
 *  `'largeBody'`,
 *  `'smallBody'`,
 *  `'monoBody'`, or
 *  `'monoSmallBody'`,
 * @param {object} [typographyParams] Customize the font settings or element using system values for: `color`, `family`, `lineHeight`, `size`, `weight` and `withReset`.
 * @public
 */

function typography(variantOrParams, additionalParams) {
  var variant = 'body';
  var params = {};
  if (isTypographyVariant(variantOrParams)) {
    variant = variantOrParams;
    if (additionalParams !== undefined) {
      params = additionalParams;
    }
  } else if (_typeof(variantOrParams) === 'object' && additionalParams === undefined) {
    params = variantOrParams;
  }
  var defaultParams = {
    withReset: false
  };
  var variantParams = getParamsForVariant(variant);
  var mergedParams = (0, _merge["default"])(defaultParams, variantParams, params);

  // Transform params to be ready for css literal below: i.e size="24" -> "24px"
  var finalParams = {
    additionalStyles: mergedParams.additionalStyles,
    color: colorPropToVariableMap[mergedParams.color],
    family: familyPropToVariableMap[mergedParams.family],
    lineHeight: lineHeightVariableToValueMap[mergedParams.lineHeight],
    size: fontsizePixelValueToRemMap[mergedParams.size],
    weight: weightPropToValueMap[mergedParams.weight],
    withReset: mergedParams.withReset
  };
  return function () {
    return (0, _styledComponents.css)(_templateObject8(), finalParams.withReset && (0, _resets.reset)(null), finalParams.color, finalParams.family, finalParams.size, finalParams.weight, finalParams.lineHeight, finalParams.additionalStyles);
  };
}
var _default = typography;
exports["default"] = _default;
