// @splunk/themes/mixins/utilityMixins.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.clearfix = clearfix;
exports.ellipsis = ellipsis;
exports.printWidth100Percent = printWidth100Percent;
exports.printHide = printHide;
exports.printNoBackground = printNoBackground;
exports.printWrapAll = printWrapAll;
exports.screenReaderContent = screenReaderContent;
exports.skipLink = skipLink;
exports.overlayColors = overlayColors;
exports.colorWithAlpha = colorWithAlpha;
exports["default"] = void 0;
var _colorBlend = require("color-blend");
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
/**
 * @file
 * A collection of style-related helper functions. All of them return a single object containing
 * DOM CSS properties, for example: `{ display: …, fontFamily: … }`.
 */

/**
 * `clearfix` is used on a container to ensure its height is at least as tall as any floating
 * children.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { clearfix } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      ${clearfix()};
 *  `
 *  ```
 * @public
 */
function clearfix() {
  return {
    '&::after': {
      display: 'table',
      content: '""',
      clear: 'both'
    }
  };
}

/**
 * Use `ellipsis` for overflowing text. Requires `display` to be set to `inline-block` or `block`.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { ellipsis } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      ${ellipsis()};
 *      width: 300px;
 *  `
 *  ```
 * @public
 */
function ellipsis() {
  return {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  };
}

/**
 * Force an element to be exactly 100% wide so that it doesn't overflow the page.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { printWidth100Percent } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      @media print {
 *          .myElement {
 *             ${printWidth100Percent()};
 *          }
 *      }
 *  }
 *  `
 *  ```
 *  @public
 */
function printWidth100Percent() {
  return {
    maxWidth: '100% !important',
    width: '100% !important',
    overflow: 'hidden !important'
  };
}

/**
 * Hide an element (such as a button).
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { printHide } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      @media print {
 *          .myElement {
 *              ${printHide()};
 *          }
 *      }
 *  }
 *  ```
 * @public
 */
function printHide() {
  return {
    display: 'none !important'
  };
}

/**
 * Remove background gradients and images.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { printNoBackground } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      @media print {
 *          .myElement {
 *              ${printNoBackground()};
 *          }
 *      }
 *  }
 *  ```
 * @public
 */
function printNoBackground() {
  return {
    background: 'none !important'
  };
}

/**
 * Ensure that all text wraps so that it doesn't overflow the page.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { printWrapAll } from '@splunk/themes/mixins';
 *
 *  const myBlock = styled.div`
 *      @media print {
 *          .myElement {
 *              ${printWrapAll()};
 *          }
 *      }
 *  }
 *  ```
 * @public
 */
function printWrapAll() {
  // TS: have to assert as CSSObject because csstype doesn't allow !important
  return {
    wordBreak: 'break-all !important',
    wordWrap: 'break-word !important',
    overflowWrap: 'break-word !important',
    whiteSpace: 'normal !important'
  };
}

/**
 * Visually hide content. Typically used to target content for assistive technologies.
 *
 *  ##### Example
 *  ``` js
 *  import { screenReaderContent } from '@splunk/themes/mixins';
 *
 *  .myElement {
 *      ${screenReaderContent()};
 *  }
 *  ```
 * @public
 */
function screenReaderContent() {
  return {
    position: 'absolute',
    overflow: 'hidden',
    clip: 'rect(0 0 0 0)',
    height: '1px',
    width: '1px',
    margin: '-1px',
    padding: 0,
    border: 0
  };
}

/**
 * Visually hide interactive elements and reveal them on focus. Use this mixin to allow
 * keyboard users to skip over repeated content, such as navigation menus
 *
 * ##### Example
 * ```js
 * import styled from 'styled-components';
 * import Link from '@splunk/react-ui/Link';
 * import mixins from '@splunk/themes';
 *
 * const skipLink = styled(Link)`
 *   ${mixins.skipLink()};
 *  `;
 * // or if you have a container with several links inside it:
 * const skipLinkContainer = styled.div`
 *   ${mixins.skipLink()};
 * `;
 * ```
 * @public
 */
function skipLink() {
  return {
    position: 'absolute',
    '&:not(:focus-within)': {
      overflow: 'hidden',
      clipPath: 'inset(50%)',
      height: '1px',
      minHeight: '1px',
      width: '1px',
      minWidth: '1px',
      padding: '0',
      border: '0',
      whiteSpace: 'nowrap'
    }
  };
}
/**
 * Calculates how one color would appear over another using a normal blend mode.
 * Colors can either be strings or functions, such as variable functions.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { overlayColors } from '@splunk/themes/mixins';
 *  import { interactiveColorAccent, interactiveColorOverlaySelected } from '@splunk/themes/variables';
 *
 *  const myButton = styled.button`
 *      background: ${overlayColors(interactiveColorAccent, interactiveColorOverlaySelected)};
 *  `
 *  ```
 * @name overlayColors
 * @kind function
 * @param {string|function} background
 * @param {string|function} foreground The color to overlay over the background.
 * @public
 */
function overlayColors(c1, c2) {
  return function (props) {
    var c1Value = typeof c1 === 'function' ? c1(props) : c1;
    var c2Value = typeof c2 === 'function' ? c2(props) : c2;
    var c3 = (0, _colorBlend.normal)((0, _tinycolor["default"])(c1Value).toRgb(), (0, _tinycolor["default"])(c2Value).toRgb());
    return (0, _tinycolor["default"])(c3).toRgbString();
  };
}

/**
 * Sets the alpha value on a given color.
 *
 *  ##### Example
 *  ```js
 *  import styled from 'styled-components';
 *  import { colorWithAlpha } from '@splunk/themes/mixins';
 *  import { actionColorBackgroundPrimary } from '@splunk/themes/variables';
 *
 *  const myButton = styled.button`
 *      background: ${colorWithAlpha(actionColorBackgroundPrimary, 0.5)};
 *  `
 *  ```
 * @name colorWithAlpha
 * @kind function
 * @param {string|function} color
 * @param {number} alpha The alpha value accepts range between 0-1.
 * @public
 */
function colorWithAlpha(color, alpha) {
  return function (props) {
    var colorValue = typeof color === 'function' ? color(props) : color;
    if (false && (alpha < 0 || alpha > 1)) {
      // eslint-disable-next-line no-console
      console.warn("The alpha value \"".concat(alpha, "\" is not within acceptable 0-1 range."));
    }
    return (0, _tinycolor["default"])(colorValue).setAlpha(alpha).toRgbString();
  };
}
var _default = {
  clearfix: clearfix,
  ellipsis: ellipsis,
  printWidth100Percent: printWidth100Percent,
  printHide: printHide,
  printNoBackground: printNoBackground,
  printWrapAll: printWrapAll,
  screenReaderContent: screenReaderContent,
  skipLink: skipLink,
  colorWithAlpha: colorWithAlpha,
  overlayColors: overlayColors
};
exports["default"] = _default;
