// @splunk/themes/prisma/dark.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var backgroundColors = {
  backgroundColorPopup: '#27292e',
  backgroundColorSection: '#1a1c20',
  backgroundColorSidebar: '#0b0c0e',
  backgroundColorPage: '#111215',
  backgroundColorNavigation: '#08090a',
  backgroundColorFloating: '#ffffff',
  backgroundColorDialog: '#1e2024',
  backgroundColorScrim: 'rgba(0, 0, 0, 0.8)'
};
var borderColors = {
  borderColor: '#717275',
  borderColorWeak: '#3d3e43',
  borderColorStrong: '#97989b'
};
var contentColors = {
  contentColorAccent: '#3d95ff',
  contentColorAccentStrong: '#9ecaff',
  contentColorAccentWeak: '#14437b',
  contentColorActive: '#fafafa',
  contentColorDefault: '#b5b5b5',
  contentColorDisabled: '#6b6b6b',
  contentColorInverted: '#000000',
  contentColorMuted: '#909090',
  contentColorNegative: '#ff5252',
  contentColorNegativeWeak: '#561a1a',
  contentColorLink: '#3993ff',
  contentColorLinkDisabled: '#14437b',
  contentColorLinkVisited: '#a974f7',
  contentColorLinkVisitedDisabled: '#6732B8'
};
var actionColors = {
  actionColorBackgroundPrimary: '#3993ff',
  actionColorBackgroundPrimaryActive: '#6baeff',
  actionColorBackgroundPrimaryDisabled: '#103460',
  actionColorBackgroundPrimaryHover: '#6baeff',
  actionColorBackgroundSecondary: 'transparent',
  actionColorBackgroundSecondaryActive: 'rgba(225, 225, 225, 0.16)',
  actionColorBackgroundSecondaryDisabled: 'transparent',
  actionColorBackgroundSecondaryHover: 'rgba(255, 255, 255, 0.05)',
  actionColorBackgroundSubtle: 'transparent',
  actionColorBackgroundSubtleActive: '#0a2f5c',
  actionColorBackgroundSubtleDisabled: 'transparent',
  actionColorBackgroundSubtleHover: '#0a2f5c',
  actionColorBackgroundDestructive: '#ff4242',
  actionColorBackgroundDestructiveActive: '#ff6b6b',
  actionColorBackgroundDestructiveDisabled: '#561a1a',
  actionColorBackgroundDestructiveHover: '#ff6b6b',
  actionColorBackgroundDestructiveSecondary: 'transparent',
  actionColorBackgroundDestructiveSecondaryActive: '#471f1f',
  actionColorBackgroundDestructiveSecondaryDisabled: 'transparent',
  actionColorBackgroundDestructiveSecondaryHover: '#471f1f',
  actionColorBorderSecondary: 'rgba(255, 255, 255, 0.5)',
  actionColorBorderSecondaryActive: 'rgba(255, 255, 255, 0.5)',
  actionColorBorderSecondaryDisabled: 'rgba(255, 255, 255, 0.3)',
  actionColorBorderSecondaryHover: 'rgba(255, 255, 255, 0.7)',
  actionColorBorderDestructiveSecondary: '#ff4242',
  actionColorBorderDestructiveSecondaryActive: '#ff6b6b',
  actionColorBorderDestructiveSecondaryDisabled: '#561a1a',
  actionColorBorderDestructiveSecondaryHover: '#ff6b6b',
  actionColorContentPrimary: contentColors.contentColorInverted,
  actionColorContentPrimaryDisabled: contentColors.contentColorInverted,
  actionColorContentSecondary: contentColors.contentColorActive,
  actionColorContentSecondaryDisabled: contentColors.contentColorDisabled,
  actionColorContentSubtleActive: contentColors.contentColorAccentStrong,
  actionColorContentSubtleHover: contentColors.contentColorAccentStrong,
  actionColorContentStandalone: contentColors.contentColorAccent,
  actionColorContentStandaloneActive: contentColors.contentColorAccentStrong,
  actionColorContentStandaloneDisabled: contentColors.contentColorAccentWeak,
  actionColorContentStandaloneHover: contentColors.contentColorAccentStrong,
  actionColorContentDestructive: contentColors.contentColorInverted,
  actionColorContentDestructiveDisabled: contentColors.contentColorInverted,
  actionColorContentDestructiveSecondary: contentColors.contentColorNegative,
  actionColorContentDestructiveSecondaryDisabled: contentColors.contentColorNegativeWeak
};
var interactiveColors = {
  interactiveColorBorder: 'rgba(255, 255, 255, 0.5)',
  interactiveColorBorderActive: 'rgba(225, 225, 225, 0.5)',
  interactiveColorBorderHover: 'rgba(255, 255, 255, 0.7)',
  interactiveColorBorderDisabled: 'rgba(255, 255, 255, 0.30)',
  interactiveColorOverlaySelected: 'rgba(255, 255, 255, 0.1)',
  interactiveColorOverlayHover: 'rgba(255, 255, 255, 0.05)',
  interactiveColorOverlayActive: 'rgba(225, 225, 225, 0.16)',
  interactiveColorOverlayDrag: 'rgba(57, 147, 255, 0.16)',
  interactiveColorBackground: 'transparent',
  interactiveColorBackgroundDisabled: 'transparent',
  interactiveColorAccent: '#3993ff',
  interactiveColorAccentError: '#ff4242',
  interactiveColorAccentErrorStrong: '#ff878b',
  interactiveColorAccentErrorWeak: '#6f261d'
};
var neutralColors = {
  black: '#000000',
  neutral50: '#232429',
  neutral100: '#33343b',
  neutral200: '#43454b',
  neutral300: '#505158',
  neutral400: '#818285',
  neutral500: '#acacad',
  white: '#ffffff',
  transparent: 'transparent'
};
var notificationColors = {
  notificationColorInfoWeak: '#2E1B4B',
  notificationColorInfo: '#AC75FF',
  notificationColorInfoStrong: '#D1B2FF',
  notificationColorPositiveWeak: '#092635',
  notificationColorPositive: '#1AB2FF',
  notificationColorPositiveStrong: '#99DDFF',
  notificationColorCautionWeak: '#331E00',
  notificationColorCaution: '#FFAF0F',
  notificationColorCautionStrong: '#FDDC9B',
  notificationColorNegativeWeak: '#391313',
  notificationColorNegative: '#FF5252',
  notificationColorNegativeStrong: '#FF9999'
};
var severityColors = {
  severityColorCustom: '#868EC2',
  severityColorUnknown: '#889099',
  severityColorInfo: '#A974F7',
  severityColorNormal: '#04A4B0',
  severityColorNotice: '#D6900D',
  severityColorWarning: '#F26722',
  severityColorAlert: '#FA5762',
  severityColorCritical: '#F2638C',
  severityColorEmergency: '#E85FC6'
};
var elevationShadows = {
  embossShadow: '0px 1px 5px rgba(0, 0, 0, 0.35), 0px 0px 1px rgba(0, 0, 0, 0.35)',
  overlayShadow: '0px 26px 103px rgba(0, 0, 0, 0.64), 0px 11px 18px rgba(0, 0, 0, 0.32), 0px 3px 6px rgba(0, 0, 0, 0.3)',
  dragShadow: '0px 26px 103px rgba(0, 0, 0, 0.64), 0px 11px 18px rgba(0, 0, 0, 0.32), 0px 3px 6px rgba(0, 0, 0, 0.3)',
  modalShadow: '0px 50px 200px #000000, 0px 29px 66px rgba(0, 0, 0, 0.41), 0px 14px 47px rgba(0, 0, 0, 0.17), 0px 5px 10px rgba(0, 0, 0, 0.15)'
};
var syntaxColors = {
  syntaxBlue: '#6cd0f0',
  syntaxBrown: '#fccf87',
  syntaxGray: '#909090',
  syntaxGreen: '#cef06c',
  syntaxOrange: '#f7933f',
  syntaxPink: '#f494e5',
  syntaxPurple: '#ab74f1',
  syntaxRed: '#e9627f',
  syntaxTeal: '#45d4ba'
};
var deprecated = {
  contentColorInfo: '#61cafa',
  // @deprecated, SUI-8251
  contentColorNegativeStrong: '#ff9e9e',
  contentColorPositive: '#85f415',
  contentColorWarning: '#f49106',
  accentColorPositive: '#85f415',
  // statusColorNormal
  accentColorWarning: '#f49106',
  // statusColorMedium
  accentColorAlert: '#c6400d',
  accentColorNegative: '#ff4242',
  // statusColorHigh
  interactiveColorPrimary: '#3993FF',
  statusColorInfo: '#61cafa',
  statusColorInfoWeak: (0, _tinycolor["default"])('#61cafa').darken(10).toHexString(),
  statusColorInfoStrong: (0, _tinycolor["default"])('#61cafa').lighten(10).toHexString(),
  statusColorNormal: '#85f415',
  statusColorNormalWeak: (0, _tinycolor["default"])('#85f415').darken(10).toHexString(),
  statusColorNormalStrong: (0, _tinycolor["default"])('#85f415').lighten(10).toHexString(),
  statusColorLow: '#2cbda3',
  statusColorLowWeak: (0, _tinycolor["default"])('#2cbda3').darken(10).toHexString(),
  statusColorLowStrong: (0, _tinycolor["default"])('#2cbda3').lighten(10).toHexString(),
  statusColorMedium: '#f49106',
  statusColorMediumWeak: (0, _tinycolor["default"])('#f49106').darken(10).toHexString(),
  statusColorMediumStrong: (0, _tinycolor["default"])('#f49106').lighten(10).toHexString(),
  statusColorHigh: '#ff4242',
  statusColorHighWeak: (0, _tinycolor["default"])('#ff4242').darken(10).toHexString(),
  statusColorHighStrong: (0, _tinycolor["default"])('#ff4242').lighten(10).toHexString(),
  statusColorCritical: '#ff3361',
  statusColorCriticalWeak: (0, _tinycolor["default"])('#ff3361').darken(10).toHexString(),
  statusColorCriticalStrong: (0, _tinycolor["default"])('#ff3361').lighten(10).toHexString()
};
var theme = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, actionColors), notificationColors), severityColors), elevationShadows), backgroundColors), borderColors), contentColors), neutralColors), interactiveColors), syntaxColors), deprecated);
var _default = theme;
exports["default"] = _default;
