// @splunk/themes/prisma/light.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _tinycolor = _interopRequireDefault(require("tinycolor2"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var backgroundColors = {
  backgroundColorPopup: '#ffffff',
  backgroundColorSection: '#ffffff',
  backgroundColorSidebar: '#f8f8f8',
  backgroundColorPage: '#f9f9f9',
  backgroundColorNavigation: '#f7f7f7',
  backgroundColorFloating: '#000000',
  backgroundColorDialog: '#ffffff',
  backgroundColorScrim: 'rgba(255, 255, 255, 0.75)'
};
var borderColors = {
  borderColor: '#8f8f8f',
  borderColorWeak: '#e0e0e0',
  borderColorStrong: '#6b6b6b'
};
var contentColors = {
  contentColorAccent: '#0264d7',
  contentColorAccentStrong: '#013f88',
  contentColorAccentWeak: '#b3d6fe',
  contentColorActive: '#2c2c2c',
  contentColorDefault: '#4d4d4d',
  contentColorDisabled: '#878787',
  contentColorInverted: '#ffffff',
  contentColorMuted: '#6b6b6b',
  contentColorNegative: '#e00000',
  contentColorNegativeWeak: '#ff9999',
  contentColorLink: '#0264d7',
  contentColorLinkDisabled: '#b3d6fe',
  contentColorLinkVisited: '#864ae0',
  contentColorLinkVisitedDisabled: '#e0ccff'
};
var actionColors = {
  actionColorBackgroundPrimary: '#0264d7',
  actionColorBackgroundPrimaryActive: '#024ba2',
  actionColorBackgroundPrimaryDisabled: '#72b2fe',
  actionColorBackgroundPrimaryHover: '#024ba2',
  actionColorBackgroundSubtle: 'transparent',
  actionColorBackgroundSecondaryActive: 'rgba(0, 0, 0, 0.12)',
  actionColorBackgroundSubtleDisabled: 'transparent',
  actionColorBackgroundSecondaryHover: 'rgba(0, 0, 0, 0.03)',
  actionColorBackgroundSecondary: 'transparent',
  actionColorBackgroundSubtleActive: '#cce3ff',
  actionColorBackgroundSecondaryDisabled: 'transparent',
  actionColorBackgroundSubtleHover: '#cce3ff',
  actionColorBackgroundDestructive: '#e00000',
  actionColorBackgroundDestructiveActive: '#a30000',
  actionColorBackgroundDestructiveDisabled: '#ff9999',
  actionColorBackgroundDestructiveHover: '#a30000',
  actionColorBackgroundDestructiveSecondary: 'transparent',
  actionColorBackgroundDestructiveSecondaryActive: '#ffcccc',
  actionColorBackgroundDestructiveSecondaryDisabled: 'transparent',
  actionColorBackgroundDestructiveSecondaryHover: '#ffcccc',
  actionColorBorderSecondary: 'rgba(0, 0, 0, 0.48)',
  actionColorBorderSecondaryActive: 'rgba(0, 0, 0, 0.5)',
  actionColorBorderSecondaryDisabled: 'rgba(0, 0, 0, 0.3)',
  actionColorBorderSecondaryHover: 'rgba(0, 0, 0, 0.6)',
  actionColorBorderDestructiveSecondary: '#e00000',
  actionColorBorderDestructiveSecondaryActive: '#a30000',
  actionColorBorderDestructiveSecondaryDisabled: '#ff9999',
  actionColorBorderDestructiveSecondaryHover: '#a30000',
  actionColorContentPrimary: contentColors.contentColorInverted,
  actionColorContentPrimaryDisabled: contentColors.contentColorInverted,
  actionColorContentSecondary: contentColors.contentColorActive,
  actionColorContentSecondaryDisabled: contentColors.contentColorDisabled,
  actionColorContentSubtleActive: contentColors.contentColorAccentStrong,
  actionColorContentSubtleHover: contentColors.contentColorAccentStrong,
  actionColorContentStandalone: contentColors.contentColorAccent,
  actionColorContentStandaloneActive: contentColors.contentColorAccentStrong,
  actionColorContentStandaloneDisabled: contentColors.contentColorAccentWeak,
  actionColorContentStandaloneHover: contentColors.contentColorAccentStrong,
  actionColorContentDestructive: contentColors.contentColorInverted,
  actionColorContentDestructiveDisabled: contentColors.contentColorInverted,
  actionColorContentDestructiveSecondary: contentColors.contentColorNegative,
  actionColorContentDestructiveSecondaryDisabled: contentColors.contentColorNegativeWeak
};
var interactiveColors = {
  interactiveColorBorder: 'rgba(0, 0, 0, 0.48)',
  interactiveColorBorderActive: 'rgba(0, 0, 0, 0.5)',
  interactiveColorBorderHover: 'rgba(0, 0, 0, 0.6)',
  interactiveColorBorderDisabled: 'rgba(0, 0, 0, 0.3)',
  interactiveColorOverlaySelected: 'rgba(0, 0, 0, 0.04)',
  interactiveColorOverlayHover: 'rgba(0, 0, 0, 0.03)',
  interactiveColorOverlayActive: 'rgba(0, 0, 0, 0.12)',
  interactiveColorOverlayDrag: 'rgba(2, 100, 215, 0.16)',
  interactiveColorBackground: 'transparent',
  interactiveColorBackgroundDisabled: 'transparent',
  interactiveColorAccent: '#0264d7',
  interactiveColorAccentError: '#e00000',
  interactiveColorAccentErrorStrong: '#a01d26',
  interactiveColorAccentErrorWeak: '#ffb2b5'
};
var neutralColors = {
  white: '#ffffff',
  neutral50: '#f2f2f2',
  neutral100: '#f0f0f0',
  neutral200: '#e6e6e6',
  neutral300: '#dddddd',
  neutral400: '#cacaca',
  neutral500: '#b8b8b8',
  black: '#000000',
  transparent: 'transparent'
};
var notificationColors = {
  notificationColorInfoWeak: '#F0E5FF',
  notificationColorInfo: '#8C4CEB',
  notificationColorInfoStrong: '#40226D',
  notificationColorPositiveWeak: '#CCEEFF',
  notificationColorPositive: '#0076B2',
  notificationColorPositiveStrong: '#054362',
  notificationColorCautionWeak: '#FEE7B9',
  notificationColorCaution: '#996600',
  notificationColorCautionStrong: '#664400',
  notificationColorNegativeWeak: '#FFE5E5',
  notificationColorNegative: '#CF1717',
  notificationColorNegativeStrong: '#5D0A0A'
};
var severityColors = {
  severityColorCustom: '#5D6596',
  severityColorUnknown: '#656C75',
  severityColorInfo: '#864AE0',
  severityColorNormal: '#017580',
  severityColorNotice: '#A65503',
  severityColorWarning: '#BD410B',
  severityColorAlert: '#CC2D37',
  severityColorCritical: '#C2306F',
  severityColorEmergency: '#B53394'
};
var elevationShadows = {
  embossShadow: '0px 1px 5px rgba(0, 0, 0, 0.07), 0px 0px 1px rgba(0, 0, 0, 0.07)',
  overlayShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  dragShadow: '0px 26px 103px rgba(0, 0, 0, 0.13), 0px 11px 18px rgba(0, 0, 0, 0.06), 0px 3px 6px rgba(0, 0, 0, 0.06)',
  modalShadow: '0px 50px 200px rgba(0, 0, 0, 0.3), 0px 29px 66px rgba(0, 0, 0, 0.08), 0px 29px 47px rgba(0, 0, 0, 0.08), 0px 5px 10px rgba(0, 0, 0, 0.03)'
};
var syntaxColors = {
  syntaxBlue: '#0e6e8b',
  syntaxBrown: '#8b5804',
  syntaxGray: '#666666',
  syntaxGreen: '#556f0b',
  syntaxOrange: '#a64708',
  syntaxPink: '#b9139e',
  syntaxPurple: '#703ef4',
  syntaxRed: '#ca1c41',
  syntaxTeal: '#1b7464'
};
var deprecated = {
  contentColorInfo: '#006be5',
  // @deprecated, SUI-8251
  contentColorNegativeStrong: '#7a0000',
  contentColorPositive: '#407a06',
  contentColorWarning: '#c97705',
  accentColorPositive: '#407a06',
  // statusColorNormal
  accentColorWarning: '#c97705',
  // statusColorMedium
  accentColorAlert: '#c6400d',
  accentColorNegative: '#e00000',
  // statusColorHigh
  interactiveColorPrimary: '#0264d7',
  statusColorInfo: '#006be5',
  statusColorInfoWeak: (0, _tinycolor["default"])('#006be5').lighten(10).toHexString(),
  statusColorInfoStrong: (0, _tinycolor["default"])('#006be5').darken(10).toHexString(),
  statusColorNormal: '#407a06',
  statusColorNormalWeak: (0, _tinycolor["default"])('#407a06').lighten(10).toHexString(),
  statusColorNormalStrong: (0, _tinycolor["default"])('#407a06').darken(10).toHexString(),
  statusColorLow: '#155a4e',
  statusColorLowWeak: (0, _tinycolor["default"])('#155a4e').lighten(10).toHexString(),
  statusColorLowStrong: (0, _tinycolor["default"])('#155a4e').darken(10).toHexString(),
  statusColorMedium: '#c97705',
  statusColorMediumWeak: (0, _tinycolor["default"])('#c97705').lighten(10).toHexString(),
  statusColorMediumStrong: (0, _tinycolor["default"])('#c97705').darken(10).toHexString(),
  statusColorHigh: '#e00000',
  statusColorHighWeak: (0, _tinycolor["default"])('#e00000').lighten(10).toHexString(),
  statusColorHighStrong: (0, _tinycolor["default"])('#e00000').darken(10).toHexString(),
  statusColorCritical: '#9e1534',
  statusColorCriticalWeak: (0, _tinycolor["default"])('#9e1534').lighten(10).toHexString(),
  statusColorCriticalStrong: (0, _tinycolor["default"])('#9e1534').darken(10).toHexString()
};
var theme = _objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread(_objectSpread({}, actionColors), notificationColors), severityColors), elevationShadows), backgroundColors), borderColors), contentColors), neutralColors), interactiveColors), syntaxColors), deprecated);
var _default = theme;
exports["default"] = _default;
